{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "0a8ecd31",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install lxml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d2db1bd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using device: cpu\n"
     ]
    }
   ],
   "source": [
    "# --- 0. Environment Setup & Offline Preparation ---\n",
    "\n",
    "# Standard Imports\n",
    "import os\n",
    "import glob\n",
    "import re\n",
    "import pandas as pd\n",
    "import lxml.etree as etree\n",
    "from lxml.etree import _Element as Element # Type hinting for lxml.etree.Element\n",
    "import collections # For deque in parenthesis removal\n",
    "import fitz # PyMuPDF for PDF processing\n",
    "from transformers import AutoModelForCausalLM, AutoTokenizer\n",
    "from transformers.modeling_utils import PreTrainedModel\n",
    "from transformers.tokenization_utils import PreTrainedTokenizer\n",
    "from transformers.utils.quantization_config import BitsAndBytesConfig\n",
    "from transformers.training_args import TrainingArguments\n",
    "from trl import SFTTrainer\n",
    "import torch\n",
    "from datasets import Dataset # Hugging Face datasets library\n",
    "import kagglehub\n",
    "import spacy\n",
    "import json\n",
    "\n",
    "# Set device for PyTorch\n",
    "device = \"cuda\" if torch and torch.cuda.is_available() else \"cpu\"\n",
    "print(f\"Using device: {device}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "b667ac82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Uncomment and run this line once to download the model\n",
    "#!python -m spacy download en_core_web_sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "48e263db",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define constants for file paths and model configurations\n",
    "BASE_INPUT_DIR = './kaggle/input/make-data-count-finding-data-references'\n",
    "ARTICLE_TRAIN_DIR = os.path.join(BASE_INPUT_DIR, 'train')\n",
    "ARTICLE_TEST_DIR = os.path.join(BASE_INPUT_DIR, 'test')\n",
    "\n",
    "# Define directories for articles in train and test sets\n",
    "LABELED_TRAINING_DATA_CSV_PATH = os.path.join(BASE_INPUT_DIR, 'train_labels.csv')\n",
    "\n",
    "# Define the base model path\n",
    "QWEN_BASE_MODEL_PATH = kagglehub.model_download(\"qwen-lm/qwen-3/transformers/0.6b\")\n",
    "\n",
    "# Output directory for the fine-tuned model and results\n",
    "BASE_OUTPUT_DIR = \"./kaggle/working\"\n",
    "FINE_TUNED_MODEL_OUTPUT_DIR = os.path.join(BASE_OUTPUT_DIR, \"qwen_finetuned_dataset_classifier\")\n",
    "FINAL_RESULTS_CSV_PATH = os.path.join(BASE_OUTPUT_DIR, \"article_dataset_classification.csv\")\n",
    "\n",
    "# Load a spaCy model (e.g., 'en_core_web_sm')\n",
    "# python -m spacy download en_core_web_sm \n",
    "NLP_SPACY = spacy.load(\"en_core_web_sm\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2e50e08c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- 2. Information Extraction (IE) - Dataset Identification ---\n",
    "\n",
    "# Regex patterns for common dataset identifiers\n",
    "# DOI_PATTERN = r'10\\.\\d{4,5}/[-._;()/:A-Za-z0-9\\u002D\\u2010\\u2011\\u2012\\u2013\\u2014\\u2015]+'\tDOI_PATTERN\n",
    "# DOI_PATTERN = r'10\\.\\s?\\d{4,5}\\/[-._()<>;\\/:A-Za-z0-9]+\\s?(?:(?![A-Z]+)(?!\\d{1,3}\\.))+[-._()<>;\\/:A-Za-z0-9]+'\n",
    "#DOI_PATTERN = r'\\bhttps://doi.org/10\\.\\d{4,5}\\/[-._\\/:A-Za-z0-9]+'\n",
    "DOI_PATTERN = r'\\b10\\.\\d{4,5}\\/[-._\\/:A-Za-z0-9]+'\n",
    "EPI_PATTERN = r'\\bEPI[-_A-Z0-9]{2,}'\n",
    "SAM_PATTERN = r'\\bSAMN[0-9]{2,}'          # SAMN07159041\n",
    "IPR_PATTERN = r'\\bIPR[0-9]{2,}'\n",
    "CHE_PATTERN = r'\\bCHEMBL[0-9]{2,}'\n",
    "PRJ_PATTERN = r'\\bPRJ[A-Z0-9]{2,}'\n",
    "E_G_PATTERN = r'\\bE-[A-Z]{4}-[0-9]{2,}'   # E-GEOD-19722 or E-PROT-100\n",
    "ENS_PATTERN = r'\\bENS[A-Z]{4}[0-9]{2,}'\n",
    "CVC_PATTERN = r'\\bCVCL_[A-Z0-9]{2,}'\n",
    "EMP_PATTERN = r'\\bEMPIAR-[0-9]{2,}'\n",
    "PXD_PATTERN = r'\\bPXD[0-9]{2,}'\n",
    "HPA_PATTERN = r'\\bHPA[0-9]{2,}'\n",
    "SRR_PATTERN = r'\\bSRR[0-9]{2,}'\n",
    "GSE_PATTERN = r'\\b(GSE|GSM|GDS|GPL)\\d{4,6}\\b' # Example for GEO accession numbers (e.g., GSE12345, GSM12345)\n",
    "GNB_PATTERN = r'\\b[A-Z]{1,2}\\d{5,6}\\b' # GenBank accession numbers (e.g., AB123456, AF000001)\n",
    "CAB_PATTERN = r'\\bCAB[0-9]{2,}'\n",
    "\n",
    "# Combine all patterns into a list\n",
    "DATASET_ID_PATTERNS = [\n",
    "    DOI_PATTERN,\n",
    "    EPI_PATTERN,\n",
    "    SAM_PATTERN,\n",
    "    IPR_PATTERN,\n",
    "    CHE_PATTERN,\n",
    "    PRJ_PATTERN,\n",
    "    E_G_PATTERN,\n",
    "    ENS_PATTERN,\n",
    "    CVC_PATTERN,\n",
    "    EMP_PATTERN,\n",
    "    PXD_PATTERN,\n",
    "    HPA_PATTERN,\n",
    "    SRR_PATTERN,\n",
    "    GSE_PATTERN,\n",
    "    GNB_PATTERN,\n",
    "    CAB_PATTERN,\n",
    "]\n",
    "\n",
    "# Compile all patterns for efficiency\n",
    "COMPILED_DATASET_ID_REGEXES = [re.compile(p) for p in DATASET_ID_PATTERNS]\n",
    "\n",
    "# Data related keywords to look for in the text\n",
    "# These keywords help to ensure that the text is relevant to datasets\n",
    "DATA_RELATED_KEYWORDS = ['data release', 'download', 'program data', 'data availability', 'the data', 'dataset', 'database', 'repository', 'data source', 'data access', 'archive', 'arch.', 'digital']\n",
    "\n",
    "def is_text_data_related(text: str) -> bool:\n",
    "    if not text:\n",
    "        return False\n",
    "    \n",
    "    text_lower = text.lower()\n",
    "    return any(keyword in text_lower for keyword in DATA_RELATED_KEYWORDS)\n",
    "\n",
    "def text_has_dataset_id(text: str) -> bool:\n",
    "    \"\"\"\n",
    "    Check if the given text contains any dataset identifier.\n",
    "    \n",
    "    Args:\n",
    "        text (str): The text to check for dataset identifiers.\n",
    "        \n",
    "    Returns:\n",
    "        bool: True if any dataset identifier is found, False otherwise.\n",
    "    \"\"\"\n",
    "\n",
    "    occurrences_with_context: list[str] = []\n",
    "    for regex in COMPILED_DATASET_ID_REGEXES:\n",
    "        if regex.search(text):\n",
    "            text_lower = text.lower()\n",
    "            # Check for specific keywords in the text\n",
    "            if any(keyword in text_lower for keyword in DATA_RELATED_KEYWORDS):\n",
    "                return True\n",
    "    return False\n",
    "\n",
    "def extract_dataset_ids(text: str, context_chars: int = 250) -> str:\n",
    "    \"\"\"\n",
    "    Extract dataset identifiers with context from the given text.\n",
    "    \n",
    "    Args:\n",
    "        text (str): The text to search for dataset identifiers.\n",
    "        context_chars (int): Number of characters to include before and after the match for context.\n",
    "        \n",
    "    Returns:\n",
    "        list[str]: A list of extracted dataset identifiers with context.\n",
    "    \"\"\"\n",
    "    text = text.replace('\\u200b', '')\n",
    "    is_small_context = len(text) < context_chars * 2\n",
    "    dataset_ids: list[str] = []\n",
    "    occurrences_with_context: list[str] = []\n",
    "    if is_text_data_related(text):\n",
    "        for regex in COMPILED_DATASET_ID_REGEXES:\n",
    "            matches = regex.finditer(text, re.IGNORECASE)\n",
    "            for match in matches:\n",
    "                dataset_id = text[match.start() : match.end()]\n",
    "                if is_small_context:\n",
    "                    dataset_ids.append(dataset_id)\n",
    "                else:\n",
    "                    extracted_snippet = text[max(0, match.start() - context_chars): match.end() + context_chars ]\n",
    "                    if is_text_data_related(extracted_snippet):\n",
    "                        occurrences_with_context.append(\"{\" + f'\"dataset_ids\": {[dataset_id]}, context: \"{extracted_snippet}\"' + \"}\")\n",
    "        if dataset_ids:\n",
    "            occurrences_with_context.append(\"{\" + f'\"dataset_ids\": {dataset_ids}, context: \"{text}\"' + \"}\")\n",
    "    \n",
    "    # If no occurrences found, return an empty string\n",
    "    # Otherwise, join the occurrences with a specific separator\n",
    "    if not occurrences_with_context:\n",
    "        return \"\"\n",
    "    return \",\".join(occurrences_with_context)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "bec3b4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use NLP to get sentences from the given text\n",
    "def get_sentences_from_text(text: str, nlp=NLP_SPACY) -> str:\n",
    "    if not text:\n",
    "        return \"\"\n",
    "    doc_spacy = nlp(text)\n",
    "    return \"\\n\".join([sent.text.replace('-\\n', '-').replace('_\\n', '_').replace('/\\n', '/').replace('\\n', ' ').replace('\\u200b', '') for sent in doc_spacy.sents])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "0b7a53c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def extract_element_text(element: Element | None) -> str:\n",
    "    if element is not None:\n",
    "        # Use itertext() to get all text content from the <p> tag and its descendants\n",
    "        # and join them into a single string.\n",
    "        all_text = \" \".join(element.itertext(tag=None)).replace('\\u200b', '').strip()\n",
    "        return all_text[:2000]\n",
    "    else:\n",
    "        return \"\"\n",
    "    \n",
    "def extract_next_sibling_text(elements: list[Element] | None, sibling_xpath: str) -> str:\n",
    "    \"\"\"\n",
    "    Extracts text from the next sibling of the given XML element.\n",
    "    \n",
    "    Args:\n",
    "        element (Element | None): The XML element whose next sibling's text is to be extracted.\n",
    "        sibling_xpath (str): The XPath expression to find the next sibling element. (eg. \"following-sibling::passage[1]\")\n",
    "        \n",
    "    Returns:\n",
    "        str: A string containing the text from the next sibling element, or an empty string if no sibling exists.\n",
    "\n",
    "    \"\"\"\n",
    "    # Check if the provided elements list is None or empty\n",
    "    if not elements:\n",
    "        return \"\"\n",
    "    \n",
    "    # Assuming there's only one such element, take the first one found\n",
    "    # and find the element immediately following based on the given sibling_xpath.\n",
    "    first_element = elements[0]\n",
    "    sibling_elements = first_element.xpath(sibling_xpath)\n",
    "\n",
    "    if not sibling_elements:\n",
    "        # print(\"DEBUG: No following <passage> element found.\") # Uncomment for debugging\n",
    "        return \"\"\n",
    "    \n",
    "    next_sibling = sibling_elements[0]\n",
    "    if next_sibling is None:\n",
    "        return \"\"\n",
    "    \n",
    "    return extract_element_text(next_sibling)\n",
    "\n",
    "def extract_elements_text(elements: list[Element] | None) -> str:\n",
    "    elements_text = []\n",
    "    if elements is None:\n",
    "        return \"\"\n",
    "    \n",
    "    for element in elements:\n",
    "        text = extract_element_text(element)\n",
    "        if text:\n",
    "            elements_text.append(text)\n",
    "\n",
    "    return \" \".join(elements_text).strip()\n",
    "\n",
    "def extract_text_from_elements_within_element(element: Element | None, child_xpaths: list[str] = [], ns: dict[str, str] | None = None) -> str:\n",
    "    \"\"\"\n",
    "    Extracts text from elements within a given XML element that match the specified tag names.\n",
    "    \n",
    "    Args:\n",
    "        element (Element | None): The XML element to search within.\n",
    "        tag_names (list[str]): A list of tag names to search for.\n",
    "        \n",
    "    Returns:\n",
    "        str: A string containing the extracted text from the matching elements.\n",
    "    \"\"\"\n",
    "    if element is None:\n",
    "        return \"\"\n",
    "    \n",
    "    if not child_xpaths:\n",
    "        # If no child tag names are provided, return the text of the element itself\n",
    "        return extract_element_text(element)\n",
    "    \n",
    "    extracted_text = []\n",
    "    for xpath in child_xpaths:\n",
    "        for child in element.findall(xpath, namespaces=ns):\n",
    "            text = extract_element_text(child)\n",
    "            if text:\n",
    "                extracted_text.append(text)\n",
    "    \n",
    "    return \"|\".join(extracted_text)\n",
    "\n",
    "def extract_data_related_elements_text(elements: list[Element] | None, child_xpaths: list[str] = [], ns: dict[str, str] | None = None) -> list[str]:\n",
    "    elements_text = []\n",
    "    if elements is None:\n",
    "        return elements_text\n",
    "    \n",
    "    for element in elements:\n",
    "        text = extract_dataset_ids(extract_text_from_elements_within_element(element, child_xpaths, ns))\n",
    "        if text:\n",
    "            elements_text.append(text)\n",
    "\n",
    "    return elements_text\n",
    "\n",
    "def extract_data_related_elements_text_from_xpath_list(root: Element | None, xpath_list: list[str], ns: dict[str, str] | None = None) -> list[str]:\n",
    "    \"\"\"\n",
    "    Extracts text from elements in the XML tree that match the provided XPath expressions.\n",
    "    \n",
    "    Args:\n",
    "        root (Element | None): The root element of the XML tree.\n",
    "        xpath_list (list[str]): A list of XPath expressions to search for elements.\n",
    "        \n",
    "    Returns:\n",
    "        list[str]: A list of extracted text from the matching elements.\n",
    "    \"\"\"\n",
    "    elements_text = []\n",
    "    if root is None or not xpath_list:\n",
    "        return elements_text\n",
    "    \n",
    "    for xpath in xpath_list:\n",
    "        primary_xpath, *child_xpath_text = xpath.split('||')\n",
    "        child_xpaths = child_xpath_text[0].split(',') if child_xpath_text else []\n",
    "        elements = root.findall(primary_xpath, namespaces=ns)\n",
    "        if elements:\n",
    "            elements_text.extend(extract_data_related_elements_text(elements, child_xpaths, ns))\n",
    "    return elements_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "02f6ceb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_pdf_text(pdf_doc: fitz.Document)  -> dict[str, str | list[str]]:\n",
    "    \"\"\"\n",
    "    Extracts all text from a PDF document using PyMuPDF.\n",
    "    \n",
    "    Args:\n",
    "        pdf_doc (fitz.Document): The PDF document to extract text from.\n",
    "        \n",
    "    Returns:\n",
    "        str: A JSON string of the article_dict containing specific elements extracted from the PDF.\n",
    "    \"\"\"\n",
    "\n",
    "    # Initialize the article dictionary with empty strings\n",
    "    article_dict = {\n",
    "        'title': '',\n",
    "        'abstract': '',\n",
    "        'data_availability': '',\n",
    "        'other_dataset_citations': []\n",
    "    }\n",
    "\n",
    "    # Initialize variables for text extraction\n",
    "    p1 = None  # Placeholder for the first page text\n",
    "    other_dataset_citations = set()  # Use a set to avoid duplicates\n",
    "    for page in pdf_doc:\n",
    "        # Extract text from the page\n",
    "        textpage = page.get_textpage()\n",
    "        if page.number == 0:\n",
    "            p1 = get_sentences_from_text(textpage.extractTEXT())\n",
    "            p1 = p1[:int(len(p1)/2)]\n",
    "\n",
    "        # Extract text from all blocks that have an abstract or dataset id's\n",
    "        blocks = textpage.extractBLOCKS()\n",
    "        for block in blocks:\n",
    "            block_text = get_sentences_from_text(block[4])\n",
    "            block_text_lower = block_text.lower()\n",
    "            if page.number == 0 and len(block_text) > 100 and \"abstract\" in block_text_lower:\n",
    "                # Add the abstract block text to the article dictionary\n",
    "                article_dict['abstract'] = block_text\n",
    "            elif \"data availability\" in block_text_lower or \"data accessibility\" in block_text_lower:\n",
    "                # Add the data availability block text to the article dictionary\n",
    "                article_dict['data_availability'] = block_text\n",
    "            else:\n",
    "                context_chars = min(250, len(block_text))  # Use a minimum\n",
    "                dataset_ids_found = extract_dataset_ids(block_text, context_chars)  # Extract dataset IDs from the block text\n",
    "                if dataset_ids_found:\n",
    "                    # Append the dataset IDs found in the block to the other_dataset_citations\n",
    "                    other_dataset_citations.add(dataset_ids_found)\n",
    "\n",
    "    article_dict['other_dataset_citations'] = list(other_dataset_citations) if other_dataset_citations else []\n",
    "    # If an abstract was not found, use the first page text as the abstract\n",
    "    if not article_dict['abstract'] and p1:\n",
    "        article_dict['abstract'] = p1\n",
    "\n",
    "    # Return the article dictionary as a JSON string\n",
    "    return article_dict\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "de668c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def extract_xml_text_jats(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availablity info for Journal Archiving and Interchange DTD (JATS)\n",
    "    # The \".//\" ensures it searches anywhere in the document, not just direct children of root.\n",
    "    ns = None  # No namespaces for JATS\n",
    "\n",
    "    xpath_title = \".//article-title\"\n",
    "    xpath_abstract = \".//abstract\"\n",
    "    xpath_data_avail = \".//notes[@notes-type='data-availability']\"\n",
    "    xpath_citations = [\".//element-citation||.article-title,.source,.pub-id\", \".//mixed-citation\"]  # List of XPath expressions for citations\n",
    "        \n",
    "    return {\n",
    "        'title': extract_element_text(root.find(xpath_title, ns)),\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, ns)),\n",
    "        'data_availability': extract_element_text(root.find(xpath_data_avail, ns)),\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "fec2e79e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_tei(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for Text Encoding Initiative (TEI)\n",
    "    # Set the namespace for TEI\n",
    "    ns = {'tei': 'http://www.tei-c.org/ns/1.0'}\n",
    "\n",
    "    xpath_title = \".//tei:title\"\n",
    "    xpath_abstract = \".//tei:abstract\"\n",
    "    xpath_data_avail = \"\" #\".//tei:biblstruct\"\n",
    "    xpath_citations = [\".//tei:biblstruct||.//tei:title,.//tei:idno,.//tei:notes\"]  # List of XPath expressions for citations\n",
    "        \n",
    "    return {\n",
    "        'title': extract_element_text(root.find(xpath_title, namespaces=ns)),\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, namespaces=ns)),\n",
    "        'data_availability': xpath_data_avail,  # No direct extraction for TEI data_availability\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "06311266",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_wiley(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for Wiley XML format\n",
    "    # Set the namespace for Wiley\n",
    "    ns = {'ns': 'http://www.wiley.com/namespaces/wiley'}\n",
    "\n",
    "    xpath_title = \".//ns:publicationMeta[@level='part']/ns:titleGroup\"    #<publicationMeta level=\"part\"><titleGroup><title type=\"main\">\n",
    "    xpath_abstract = \".//ns:abstract[@type='main']\"  #<abstract type=\"main\"\n",
    "    xpath_data_avail = \".//ns:section[@type='dataAvailability']\"  #<section numbered=\"no\" type=\"dataAvailability\"\n",
    "    xpath_citations = [\".//ns:citation||.//ns:articleTitle,.//ns:journalTitle,.//ns:url\"]  # List of XPath expressions for citations\n",
    "        \n",
    "    return {\n",
    "        'title': extract_elements_text(root.findall(xpath_title, namespaces=ns)),\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, namespaces=ns)),\n",
    "        'data_availability': extract_element_text(root.find(xpath_data_avail, namespaces=ns)),\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "dfc6c86c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_bioc(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for BioC-API XML format\n",
    "    ns = None  # No namespaces for BioC\n",
    "\n",
    "    xpath_title = \"string(.//passage[infon[@key='section_type' and text()='TITLE']]/text)\"\n",
    "    xpath_abstract = \"string(.//passage[infon[@key='section_type' and text()='ABSTRACT']]/text)\"\n",
    "    xpath_data_avail = \".//passage[text[text()='DATA ACCESSIBILITY:']]\"\n",
    "    xpath_data_avail_sibling = \"following-sibling::passage[1]\"\n",
    "    xpath_citations = []\n",
    "        \n",
    "    return {\n",
    "        'title': root.xpath(xpath_title, namespaces=ns),\n",
    "        'abstract': root.xpath(xpath_abstract, namespaces=ns)[:2000],  # Limit to 2000 characters\n",
    "        'data_availability': extract_next_sibling_text(root.xpath(xpath_data_avail, namespaces=ns), xpath_data_avail_sibling),\n",
    "        'other_dataset_citations': xpath_citations,\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "58a53b42",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_taxonx(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for TaxonX format\n",
    "    ns = None  # No namespaces for Taxonomic Treatment Publishing DTD\n",
    "\n",
    "    xpath_title = \"string(.//article-meta/title-group/article-title)\"\n",
    "    xpath_abstract = \"string(.//article-meta/abstract)\"\n",
    "    xpath_data_avail = \"\"\n",
    "    xpath_citations = []\n",
    "        \n",
    "    return {\n",
    "        'title': root.xpath(xpath_title, namespaces=ns),\n",
    "        'abstract': root.xpath(xpath_abstract, namespaces=ns)[:2000],  # Limit to 2000 characters\n",
    "        'data_availability': xpath_data_avail,  # No direct extraction for TaxonX data_availability\n",
    "        'other_dataset_citations': xpath_citations,\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "a0a58f0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dictionary mapping XML types to their respective extraction functions\n",
    "XML_TYPE_EXTRACTORS = {\n",
    "    'jats': extract_xml_text_jats,\n",
    "    'tei': extract_xml_text_tei,\n",
    "    'wiley': extract_xml_text_wiley,\n",
    "    'bioc': extract_xml_text_bioc,\n",
    "    'taxonx': extract_xml_text_taxonx,\n",
    "}\n",
    "\n",
    "# --- Data Loading ---\n",
    "def load_file_paths(dataset_type_dir: str) -> pd.DataFrame: \n",
    "    pdf_path = os.path.join(dataset_type_dir, 'PDF')\n",
    "    xml_path = os.path.join(dataset_type_dir, 'XML')\n",
    "    dataset_type = os.path.basename(dataset_type_dir)\n",
    "    pdf_files = [f for f in os.listdir(pdf_path) if f.endswith('.pdf')]\n",
    "    xml_files = [f for f in os.listdir(xml_path) if f.endswith('.xml')]\n",
    "    df_pdf = pd.DataFrame({\n",
    "        'article_id': [f.replace('.pdf', '') for f in pdf_files],\n",
    "        'pdf_file_path': [os.path.join(pdf_path, f) for f in pdf_files]\n",
    "    })\n",
    "    df_xml = pd.DataFrame({\n",
    "        'article_id': [f.replace('.xml', '') for f in xml_files],\n",
    "        'xml_file_path': [os.path.join(xml_path, f) for f in xml_files]\n",
    "    })\n",
    "    merge_df = pd.merge(df_pdf, df_xml, on='article_id', how='outer', suffixes=('_pdf', '_xml'), validate=\"one_to_many\")\n",
    "    merge_df['dataset_type'] = dataset_type\n",
    "    return merge_df\n",
    "\n",
    "def read_pdf_text(pdf_path: str, xml_type: str | None = None) -> str:\n",
    "    \"\"\"Extracts all text from a PDF file using PyMuPDF.\"\"\"\n",
    "    pdf_text = \"\"\n",
    "    if not fitz:\n",
    "        return pdf_text  # Return empty string if fitz is not available\n",
    "    try:\n",
    "        with fitz.open(pdf_path) as doc:\n",
    "            parsed_dict = extract_pdf_text(doc)  # Extract text from the PDF document\n",
    "            pdf_text = json.dumps(parsed_dict, separators=(',', ':'))            \n",
    "    except Exception as e:\n",
    "        print(f\"Error reading PDF {pdf_path}: {e}\")\n",
    "\n",
    "    print(f\"Extracted text from {pdf_path}. Length: {len(pdf_text)} characters\")\n",
    "    return pdf_text\n",
    "\n",
    "def read_xml_text(xml_file_path: str, xml_type: str) -> str:\n",
    "    \"\"\"Reads and concatenates all text content from an XML file.\"\"\"\n",
    "    xml_text = \"\"\n",
    "    if not xml_file_path or not os.path.exists(xml_file_path):\n",
    "        print(f\"XML file not found: {xml_file_path}\")\n",
    "        return xml_text\n",
    "    try:\n",
    "        # Disable external entity resolution for security\n",
    "        parser = etree.XMLParser(resolve_entities=False, no_network=True)\n",
    "        tree = etree.parse(xml_file_path, parser)\n",
    "        root = tree.getroot()\n",
    "\n",
    "        # # Get namespaces from the root element.\n",
    "        # # root.nsmap is a dictionary of {prefix: URI}.\n",
    "        # # If there's a default namespace (e.g., xmlns=\"...\"), its prefix is None.\n",
    "        # # We need to map it to a temporary prefix for XPath.\n",
    "        # ns_map = root.nsmap.copy()\n",
    "        # temp_ns_prefix = None # Initialize to None, will be 'ns' if default namespace found\n",
    "\n",
    "        # if None in ns_map:\n",
    "        #     # If a default namespace exists, assign it a temporary prefix 'ns'\n",
    "        #     # for use in XPath queries.\n",
    "        #     ns_map['ns'] = ns_map.pop(None)\n",
    "        #     temp_ns_prefix = 'ns'\n",
    "        #     print(f\"DEBUG: Default namespace detected and mapped to prefix 'ns': {ns_map['ns']}\")\n",
    "        # else:\n",
    "        #     print(\"DEBUG: No default namespace detected.\")\n",
    "\n",
    "        # Default to JATS if no type is provided\n",
    "        extract_function = XML_TYPE_EXTRACTORS.get(xml_type, extract_xml_text_jats)  \n",
    "        parsed_dict = extract_function(root)\n",
    "        xml_text = json.dumps(parsed_dict, separators=(',', ':'))\n",
    "    except Exception as e:\n",
    "        print(f\"Error reading XML {xml_file_path}: {e}\")\n",
    "        return xml_text\n",
    "    \n",
    "    print(f\"Extracted text from {xml_file_path}. Length: {len(xml_text)} characters\")\n",
    "    return xml_text\n",
    "\n",
    "def process_unsupported_file(file_path: str, xml_type: str | None = None) -> str:\n",
    "    return f\"Unsupported file type for: {file_path}\"\n",
    "\n",
    "# Dictionary mapping file extensions to loading functions\n",
    "FILE_LOADERS = {\n",
    "    '.xml': read_xml_text,\n",
    "    '.pdf': read_pdf_text,\n",
    "}\n",
    "\n",
    "def load_article_text(file_path: str, xml_type: str | None = None) -> str:\n",
    "    \"\"\"\n",
    "    Loads text content from a single article file (PDF or XML).\n",
    "    Returns the text content of the given file.\n",
    "    \"\"\"\n",
    "    text_content = \"\"\n",
    "\n",
    "    # Get the file extension (e.g., '.xml', '.pdf')\n",
    "    _, file_extension = os.path.splitext(file_path)\n",
    "    file_extension = file_extension.lower() # Ensure lowercase for consistent lookup\n",
    "\n",
    "    # Get the appropriate function from the dictionary,\n",
    "    # or fall back to a default 'unsupported' function if not found.\n",
    "    loader_function = FILE_LOADERS.get(file_extension, process_unsupported_file)\n",
    "\n",
    "    # Call the selected function\n",
    "    text_content = loader_function(file_path, xml_type=xml_type)\n",
    "\n",
    "    return text_content\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "5f45bff0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1940"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "'{\"title\":\"Is the spider a good biological control agent for Plutella xylostella (Lepidoptera: Plutellidae)?\",\"abstract\":\"\\\\n        Abstract\\\\n        Spiders, as predators of insects and other invertebrates, are an important part of the natural enemies, and they are recognized as an important biological control agent. Plutella xylostella (Linnaeus, 1758), the diamondback moth (DBM), is a well-known and destructive insect pest of brassicaceous crops worldwide. Here, we analyzed the functional responses of four spiders (Araneae) \\\\u2013 Ebrechtella tricuspidata (Fabricius, 1775) (Thomisidae), Pardosa laura (Karsch, 1879) (Lycosidae), Pardosa astrigera (Koch, 1878) (Lycosidae), and Pardosa pseudoannulata (B\\\\u00f6senberg & Strand, 1906) (Lycosidae) \\\\u2013 on P. xylostella larvae. We also analyzed intraspecific disturbances in the predation reaction and the intensity of scrambling competition of the spiders to P. xylostella larvae. Our results demonstrated that the functional responses of four spiders of different genera were in line with the Holling II model. Two Lycosidae spiders (P. astrigera and P. pseudoannulata) had the potential to control P. xylostella, and female and male spiders that belonged to the same species had different functional responses to P. xylostella. The functional responses of female predation of P. astrigena, P. laura, and P. pseudoannulata was stronger than the males, but male E. tricuspidatus had stronger functional responses to predation than females. We used the Hassell model to describe the intraspecific disturbance experiments of four spiders. There were intraspecific disturbances in the predation reactions of spiders, indicating that the predation ratio of spiders decreased in relation to the increase of its density, and with the increase of spider density, the intensity of scrambling competition of the spider increased.\\\\n      \",\"data_availability\":\"\",\"other_dataset_citations\":[]}'"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Test loading various PDF files\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_2017jc013030.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1017_rdc.2022.19.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1017_s0007123423000601.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.3389_fcimb.2024.1292467.pdf')\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_esp.5058.pdf') # This one is big\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_esp.5059.pdf') # This one is big\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_ece3.4466.pdf') # dryad\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1002_ece3.4466.xml') # dryad\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1002_mp.14424.xml')\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1007_s00259-022-06053-8.xml')    # jats\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1007_s00382-022-06361-7.xml')    # tei\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1111_1365-2435.13431.xml')       # wiley\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1111_mec.16977.xml')             # bioc\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.3897_zoologia.35.e23481.xml')      # taxonx\n",
    "file_text = load_article_text(pdf_file_path, 'taxonx')\n",
    "display(len(file_text))\n",
    "file_text"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45dbdf50",
   "metadata": {},
   "source": [
    "## EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "29d7f58e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading labeled training data from: ./kaggle/input/make-data-count-finding-data-references\\train_labels.csv\n",
      "Training labels shape: (1028, 3)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "type",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "b45e3e90-53b6-4f5f-95c9-764015e6f7db",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "https://doi.org/10.17882/49388",
         "Primary"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "Missing",
         "Missing"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "Missing",
         "Missing"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "Missing",
         "Missing"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "Missing",
         "Missing"
        ]
       ],
       "shape": {
        "columns": 3,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>https://doi.org/10.17882/49388</td>\n",
       "      <td>Primary</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                      dataset_id     type\n",
       "0    10.1002_2017jc013030  https://doi.org/10.17882/49388  Primary\n",
       "1  10.1002_anie.201916483                         Missing  Missing\n",
       "2  10.1002_anie.202005531                         Missing  Missing\n",
       "3  10.1002_anie.202007717                         Missing  Missing\n",
       "4  10.1002_chem.201902131                         Missing  Missing"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Grouped dataset ID counts:\n",
      "   dataset_id_trim  article_id\n",
      "52             htt         325\n",
      "29             Mis         309\n",
      "20             EPI          64\n",
      "47             SAM          41\n",
      "25             IPR          33\n",
      "11             CHE          29\n",
      "41             PRJ          26\n",
      "16             E-G          25\n",
      "19             ENS          21\n",
      "26             K02          20\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id_trim",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "1f1fb964-d582-40aa-b6ca-f085f7e6cc88",
       "rows": [
        [
         "832",
         "10.3389_fcimb.2024.1292467",
         "EPI_ISL_1626969",
         "Secondary",
         "EPI"
        ],
        [
         "910",
         "10.3390_v11060565",
         "EPI1019555",
         "Secondary",
         "EPI"
        ],
        [
         "901",
         "10.3390_v11060565",
         "EPI1018206",
         "Secondary",
         "EPI"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>type</th>\n",
       "      <th>dataset_id_trim</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>10.3389_fcimb.2024.1292467</td>\n",
       "      <td>EPI_ISL_1626969</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>910</th>\n",
       "      <td>10.3390_v11060565</td>\n",
       "      <td>EPI1019555</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>901</th>\n",
       "      <td>10.3390_v11060565</td>\n",
       "      <td>EPI1018206</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     article_id       dataset_id       type dataset_id_trim\n",
       "832  10.3389_fcimb.2024.1292467  EPI_ISL_1626969  Secondary             EPI\n",
       "910           10.3390_v11060565       EPI1019555  Secondary             EPI\n",
       "901           10.3390_v11060565       EPI1018206  Secondary             EPI"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the labeled training data CSV file\n",
    "print(f\"Loading labeled training data from: {LABELED_TRAINING_DATA_CSV_PATH}\")\n",
    "train_labels_df = pd.read_csv(LABELED_TRAINING_DATA_CSV_PATH)\n",
    "\n",
    "print(f\"Training labels shape: {train_labels_df.shape}\")\n",
    "display(train_labels_df.head())\n",
    "\n",
    "# Create a new column 'dataset_id_trim' by extracting the first 3 characters of the 'dataset_id' column\n",
    "train_labels_df['dataset_id_trim'] = train_labels_df['dataset_id'].str[:3]\n",
    "# Find the most frequent types of dataset_id's\n",
    "freq_dataset_id_df = train_labels_df.groupby('dataset_id_trim').count().reset_index()\n",
    "freq_dataset_id_df = freq_dataset_id_df[['dataset_id_trim', 'article_id']].sort_values(by='article_id', ascending=False)\n",
    "print(f\"Grouped dataset ID counts:\\n{freq_dataset_id_df.head(10)}\")\n",
    "\n",
    "# Display the first 10 rows where dataset_id_trim is 'EPI'\n",
    "display(train_labels_df[train_labels_df['dataset_id_trim'] == 'EPI'].sample(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "560a47e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Example grouped training data for article_id '10.1002_2017jc013030': [{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]\n"
     ]
    }
   ],
   "source": [
    "# Group training data by article_id to get all datasets for each article\n",
    "# This creates a dictionary where keys are article_ids and values are lists of dataset dicts\n",
    "grouped_training_data = {}\n",
    "for article_id, group_df in train_labels_df.groupby('article_id'):\n",
    "    grouped_training_data[article_id] = group_df[['dataset_id', 'type']].to_dict('records')\n",
    "\n",
    "# Example usage of grouped_training_data\n",
    "print(f\"Example grouped training data for article_id '10.1002_2017jc013030': {grouped_training_data['10.1002_2017jc013030']}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8718ef7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_first_line_of_xml(file_path: str) -> str | None:\n",
    "    \"\"\"\n",
    "    Reads and returns the first line of an XML file.\n",
    "\n",
    "    Args:\n",
    "        file_path (str): The path to the XML file.\n",
    "\n",
    "    Returns:\n",
    "        str | None: The first line of the file, stripped of leading/trailing whitespace,\n",
    "                    or None if the file cannot be read or is empty.\n",
    "    \"\"\"\n",
    "    if not file_path and not os.path.exists(file_path):\n",
    "        return None\n",
    "    \n",
    "    try:\n",
    "        with open(file_path, 'r', encoding='utf-8') as f:\n",
    "            first_line = f.readline().replace('<?xml version=\"1.0\" encoding=\"UTF-8\"?>', '').replace('<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>', '').strip()\n",
    "            # If the first line is empty, read the next line\n",
    "            if not first_line:\n",
    "                first_line = f.readline()\n",
    "            return first_line.strip()[:90] if first_line else None\n",
    "    except UnicodeDecodeError:\n",
    "        try:\n",
    "            with open(file_path, 'r', encoding='iso-8859-1') as f:\n",
    "                first_line = f.readline().replace('<?xml version=\"1.0\" encoding=\"UTF-8\"?>', '').replace('<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>', '').strip()\n",
    "                if not first_line:\n",
    "                    first_line = f.readline()\n",
    "                return first_line.strip()[:90] if first_line else None\n",
    "        except Exception as e:\n",
    "            return None\n",
    "    except Exception as e:\n",
    "        print(f\"Error reading file '{file_path}': {e}\")\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "71dd0895",
   "metadata": {},
   "outputs": [],
   "source": [
    "def identify_xml_type(first_line: str) -> str:\n",
    "    \"\"\"\n",
    "    Identifies the XML type based on the first line of the XML file.\n",
    "    \n",
    "    Args:\n",
    "        file_path (str): The path to the XML file.\n",
    "        \n",
    "    Returns:\n",
    "        str: The identified XML type ('jats', 'tei', 'wiley', 'bioc', or 'unknown').\n",
    "    \"\"\"\n",
    "    if not first_line:\n",
    "        return \"unknown\"\n",
    "    first_line_lower = first_line.lower()\n",
    "    # Check for specific patterns in the first line\n",
    "    if 'journal archiving and interchange dtd' in first_line_lower:\n",
    "        return \"jats\"\n",
    "    elif 'xmlns=\"http://www.tei-c.org/ns/1.0\"' in first_line_lower:\n",
    "        return \"tei\"\n",
    "    elif 'xmlns=\"http://www.wiley.com/namespaces/wiley\"' in first_line_lower:\n",
    "        return \"wiley\"\n",
    "    elif 'bioc.dtd' in first_line_lower or 'bioc-api' in first_line_lower:\n",
    "        return \"bioc\"\n",
    "    elif 'taxonomic treatment publishing dtd' in first_line_lower:\n",
    "        return \"taxonx\"\n",
    "    \n",
    "    return \"unknown\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "34f61a27",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train files paths shape: (524, 7)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "b37192e8-bf95-417b-bfa1-a580060ad16a",
       "rows": [
        [
         "70",
         "10.1038_hdy.2015.99",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2015.99.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_hdy.2015.99.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.f47f6', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "32",
         "10.1007_jhep12(2018)117",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>10.1038_hdy.2015.99</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>10.1007_jhep12(2018)117</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10.1002_chem.202000235</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "70      10.1038_hdy.2015.99   \n",
       "32  10.1007_jhep12(2018)117   \n",
       "6    10.1002_chem.202000235   \n",
       "\n",
       "                                        pdf_file_path  \\\n",
       "70  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "32  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "6   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \\\n",
       "70  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "32                                                           train   \n",
       "6   ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                         dataset_info  \\\n",
       "70  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "32     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "6      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                       xml_first_line xml_type  \n",
       "70  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei  \n",
       "32                                               None     None  \n",
       "6   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Test files paths shape: (30, 7)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "34ec2142-d05f-4f3d-847b-96c86b28bcad",
       "rows": [
        [
         "22",
         "10.1002_ejic.201900904",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejic.201900904.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc"
        ],
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10.1002_ejic.201900904</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;component xmlns=\"http://www.wiley.com/namespa...</td>\n",
       "      <td>wiley</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                article_id                                      pdf_file_path  \\\n",
       "22  10.1002_ejic.201900904  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2   10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "0     10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \\\n",
       "22  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "0   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "\n",
       "                                         dataset_info  \\\n",
       "22     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "0   [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "\n",
       "                                       xml_first_line xml_type  \n",
       "22  <component xmlns=\"http://www.wiley.com/namespa...    wiley  \n",
       "2   <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc  \n",
       "0   <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load file paths for training and testing datasets\n",
    "train_file_paths_df = load_file_paths(ARTICLE_TRAIN_DIR)\n",
    "test_file_paths_df = load_file_paths(ARTICLE_TEST_DIR)\n",
    "train_file_paths_df['xml_file_path'] = train_file_paths_df['xml_file_path'].fillna('')\n",
    "test_file_paths_df['xml_file_path'] = test_file_paths_df['xml_file_path'].fillna('')\n",
    "\n",
    "# Remove rows in train_file_paths_df that have a corresponding article_id in test_file_paths_df\n",
    "# train_file_paths_df = train_file_paths_df[~train_file_paths_df['article_id'].isin(test_file_paths_df['article_id'])]\n",
    "\n",
    "# Merge the file paths with the grouped_training_data\n",
    "train_file_paths_df['dataset_info'] = train_file_paths_df['article_id'].map(grouped_training_data)\n",
    "test_file_paths_df['dataset_info'] = test_file_paths_df['article_id'].map(grouped_training_data)\n",
    "\n",
    "# Get first line of XML files to determine the type\n",
    "train_file_paths_df['xml_first_line'] = train_file_paths_df['xml_file_path'].apply(read_first_line_of_xml)\n",
    "test_file_paths_df['xml_first_line'] = test_file_paths_df['xml_file_path'].apply(read_first_line_of_xml)\n",
    "\n",
    "# Identify the xml_type based on the first line of the XML file\n",
    "train_file_paths_df['xml_type'] = train_file_paths_df['xml_first_line'].apply(lambda x: identify_xml_type(x) if x else None)\n",
    "test_file_paths_df['xml_type'] = test_file_paths_df['xml_first_line'].apply(lambda x: identify_xml_type(x) if x else None)\n",
    "\n",
    "print(f\"Train files paths shape: {train_file_paths_df.shape}\")\n",
    "display(train_file_paths_df.sample(3))\n",
    "print(f\"Test files paths shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.sample(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "24f16bad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt',\n",
       "       '<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ',\n",
       "       '<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509',\n",
       "       None,\n",
       "       '<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com',\n",
       "       '<!DOCTYPE article PUBLIC \"-//NLM//DTD Journal Archiving and Interchange DTD v3.0 20080202/',\n",
       "       '<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250507',\n",
       "       '<!DOCTYPE article PUBLIC \"-//TaxonX//DTD Taxonomic Treatment Publishing DTD v0 20100105//E'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_file_paths_df['xml_first_line'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "fe55c451",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "320a319a-4491-4f31-95fe-a692d1198e80",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202007717.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201902131.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201902131.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201903120.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001412.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "8",
         "10.1002_chem.202001668",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001668.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001668.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "9",
         "10.1002_chem.202003167",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202003167.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "10",
         "10.1002_cssc.202201821",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_cssc.202201821.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_cssc.202201821.xml",
         "train",
         null,
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "11",
         "10.1002_ece3.3985",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.3985.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "12",
         "10.1002_ece3.4466",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.4466.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.r6nq870', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "13",
         "10.1002_ece3.5260",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5260.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5260.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "14",
         "10.1002_ece3.5395",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5395.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5395.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6144.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "16",
         "10.1002_ece3.6303",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6303.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.37pvmcvgb', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "17",
         "10.1002_ece3.6784",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6784.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "18",
         "10.1002_ece3.961",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.961.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "19",
         "10.1002_ece3.9627",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.9627.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "20",
         "10.1002_ecs2.1280",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.p3fg9', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "21",
         "10.1002_ecs2.4619",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.4619.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ecs2.4619.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.25349/d9qw5x', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "22",
         "10.1002_ejic.201900904",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejic.201900904.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejic.201900904.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "24",
         "10.1002_ejoc.202000916",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejoc.202000916.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "25",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "26",
         "10.1002_esp.5090",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_esp.5090.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9353101', 'type': 'Secondary'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley"
        ],
        [
         "27",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_mp.14424.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "28",
         "10.1002_nafm.10870",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_nafm.10870.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9gtumay', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "29",
         "10.1007_jhep07(2018)134",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep07(2018)134.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "30",
         "10.1007_jhep11(2018)113",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "31",
         "10.1007_jhep11(2018)115",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "32",
         "10.1007_jhep12(2018)117",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "33",
         "10.1007_s00259-022-06053-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00259-022-06053-8.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00259-022-06053-8.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2017.7hs46erv', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2019.30ilqfcl', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "34",
         "10.1007_s00382-012-1636-1",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "35",
         "10.1007_s00382-022-06361-7",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-022-06361-7.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00382-022-06361-7.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6075/j0154fjj', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.6075/j089161b', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "36",
         "10.1007_s00442-022-05201-z",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00442-022-05201-z.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00442-022-05201-z.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.wpzgmsbps', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "37",
         "10.1007_s10904-014-0054-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "38",
         "10.1007_s12263-014-0408-4",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12263-014-0408-4.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "39",
         "10.1007_s12559-020-09751-3",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12559-020-09751-3.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12559-020-09751-3.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.u1x8a5nr', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "40",
         "10.1016_j.ast.2022.107401",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ast.2022.107401.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182.v1', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "41",
         "10.1016_j.cpc.2024.109087",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.cpc.2024.109087.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.cpc.2024.109087.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/9gr6pxhfjm', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "42",
         "10.1016_j.dib.2022.108797",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2022.108797.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2022.108797.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.1594/pangaea.949117', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "43",
         "10.1016_j.dib.2023.109949",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2023.109949.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8338991', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "44",
         "10.1016_j.ecolind.2021.107934",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ecolind.2021.107934.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ecolind.2021.107934.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6073/pasta/be42bb841e696b7bcad9957aed33db5e', 'type': 'Secondary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "45",
         "10.1016_j.fuel.2022.125768",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.fuel.2022.125768.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.11583/dtu.20555586', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.11583/dtu.20555586.v3', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "46",
         "10.1016_j.jlp.2022.104761",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.23642/usn.15134442', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.23642/usn.15134442.v1', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "47",
         "10.1016_j.jobe.2023.107105",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.jobe.2023.107105.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8014149', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.5281/zenodo.8014150', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "48",
         "10.1016_j.jsames.2022.103824",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jsames.2022.103824.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/f7k072r7', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "49",
         "10.1016_j.molcel.2018.11.006",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.molcel.2018.11.006.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.molcel.2018.11.006.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/jb4jjxsbb7.1', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17632/xtb4mkvf8f.1', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 524
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>10.7554_elife.74937</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5281/zenod...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>10.7717_peerj.10452</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'PRJNA664798', 'type': 'Second...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>10.7717_peerj.11352</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7291/d11m3...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>10.7717_peerj.12422</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.15468/dl.c...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>10.7717_peerj.13193</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.6073/pasta...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>524 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "0      10.1002_2017jc013030   \n",
       "1    10.1002_anie.201916483   \n",
       "2    10.1002_anie.202005531   \n",
       "3    10.1002_anie.202007717   \n",
       "4    10.1002_chem.201902131   \n",
       "..                      ...   \n",
       "519     10.7554_elife.74937   \n",
       "520     10.7717_peerj.10452   \n",
       "521     10.7717_peerj.11352   \n",
       "522     10.7717_peerj.12422   \n",
       "523     10.7717_peerj.13193   \n",
       "\n",
       "                                         pdf_file_path  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "..                                                 ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                         xml_file_path dataset_type  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "..                                                 ...          ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                          dataset_info  \\\n",
       "0    [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "3       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "4       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "..                                                 ...   \n",
       "519  [{'dataset_id': 'https://doi.org/10.5281/zenod...   \n",
       "520  [{'dataset_id': 'PRJNA664798', 'type': 'Second...   \n",
       "521  [{'dataset_id': 'https://doi.org/10.7291/d11m3...   \n",
       "522  [{'dataset_id': 'https://doi.org/10.15468/dl.c...   \n",
       "523  [{'dataset_id': 'https://doi.org/10.6073/pasta...   \n",
       "\n",
       "                                        xml_first_line xml_type  \n",
       "0    <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei  \n",
       "1    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "2    <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc  \n",
       "3    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "4    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "..                                                 ...      ...  \n",
       "519  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "520  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "521  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "522  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "523  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "\n",
       "[524 rows x 7 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_file_paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "e30a9aff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf. Length: 4770 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf. Length: 1306 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf. Length: 1597 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf. Length: 1669 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201902131.pdf. Length: 616 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf. Length: 846 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf. Length: 674 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf. Length: 941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001668.pdf. Length: 404 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf. Length: 883 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_cssc.202201821.pdf. Length: 3837 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf. Length: 3599 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf. Length: 2282 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5260.pdf. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5395.pdf. Length: 3908 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6144.pdf. Length: 2895 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf. Length: 3217 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf. Length: 6695 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf. Length: 2786 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf. Length: 2737 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf. Length: 3162 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.4619.pdf. Length: 3416 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejic.201900904.pdf. Length: 586 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf. Length: 924 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf. Length: 272 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf. Length: 4447 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf. Length: 5294 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf. Length: 4496 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_nafm.10870.pdf. Length: 4396 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep07(2018)134.pdf. Length: 849 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf. Length: 1361 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf. Length: 1179 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf. Length: 757 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00259-022-06053-8.pdf. Length: 8511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf. Length: 5502 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-022-06361-7.pdf. Length: 3685 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00442-022-05201-z.pdf. Length: 1651 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf. Length: 905 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf. Length: 5298 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12559-020-09751-3.pdf. Length: 2718 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf. Length: 2529 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.cpc.2024.109087.pdf. Length: 1804 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2022.108797.pdf. Length: 2067 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf. Length: 2173 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ecolind.2021.107934.pdf. Length: 3238 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.fuel.2022.125768.pdf. Length: 1153 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf. Length: 2711 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf. Length: 3556 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jsames.2022.103824.pdf. Length: 3498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.molcel.2018.11.006.pdf. Length: 4432 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.scitotenv.2024.171850.pdf. Length: 3950 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.websem.2024.100815.pdf. Length: 701 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_rdc.2022.19.pdf. Length: 2007 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_s0007123423000601.pdf. Length: 1277 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acs.jcim.9b01185.pdf. Length: 1043 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acscentsci.1c01109.pdf. Length: 1029 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acsomega.3c06074.pdf. Length: 1682 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_jacs.2c06519.pdf. Length: 1901 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2018gl078007.pdf. Length: 4335 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2019jg005297.pdf. Length: 4696 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2019pa003774.pdf. Length: 2497 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2020jf005675.pdf. Length: 2550 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021ea001703.pdf. Length: 2596 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021gl096173.pdf. Length: 760 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021pa004379.pdf. Length: 3410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2022gl100473.pdf. Length: 1648 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2023ea002840.pdf. Length: 2557 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2023wr035126.pdf. Length: 5917 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2013.74.pdf. Length: 3094 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2014.75.pdf. Length: 3197 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2015.99.pdf. Length: 3337 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_ncomms11871.pdf. Length: 1024 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41396-020-00885-8.pdf. Length: 2275 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41437-020-0318-8.pdf. Length: 2026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-04041-x.pdf. Length: 1862 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-07681-1.pdf. Length: 1439 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-019-10357-z.pdf. Length: 2128 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41558-022-01301-z.pdf. Length: 632 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41562-021-01247-w.pdf. Length: 3982 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-019-0101-y.pdf. Length: 2670 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-022-01555-4.pdf. Length: 5297 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-023-02280-2.pdf. Length: 3423 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-017-15852-1.pdf. Length: 2520 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-020-59839-x.pdf. Length: 4196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-021-85671-y.pdf. Length: 6127 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-024-56373-y.pdf. Length: 2275 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41893-020-0559-9.pdf. Length: 6516 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_sdata.2017.167.pdf. Length: 1372 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c5sc01309g.pdf. Length: 2791 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9ce01020c.pdf. Length: 2599 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9ra06638a.pdf. Length: 2790 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9sc02930c.pdf. Length: 2673 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0dt02130j.pdf. Length: 2624 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0gc00363h.pdf. Length: 2738 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01197e.pdf. Length: 2695 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01518k.pdf. Length: 2933 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc02283g.pdf. Length: 2307 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc02676j.pdf. Length: 2407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc03604h.pdf. Length: 2904 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d1ee03696c.pdf. Length: 3049 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d2cc00847e.pdf. Length: 3134 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1051_e3sconf_202017220004.pdf. Length: 2390 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1055_s-0039-1693681.pdf. Length: 1550 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1323607111.pdf. Length: 3995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1705601114.pdf. Length: 4105 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1711872115.pdf. Length: 3930 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1915395117.pdf. Length: 352 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.2306723120.pdf. Length: 4098 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_01621459.2019.1573732.pdf. Length: 587 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_02713683.2019.1607392.pdf. Length: 602 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2019.1669817.pdf. Length: 580 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2020.1714721.pdf. Length: 1865 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_07350015.2020.1766469.pdf. Length: 568 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_13501763.2019.1630470.pdf. Length: 678 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_14756366.2020.1740692.pdf. Length: 760 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_15476286.2016.1232238.pdf. Length: 2950 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_19475683.2023.2226196.pdf. Length: 5309 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_21645515.2023.2189598.pdf. Length: 276 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_22221751.2020.1738277.pdf. Length: 2277 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arad016.pdf. Length: 2917 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arw167.pdf. Length: 3128 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arz101.pdf. Length: 3391 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_bib_bbab395.pdf. Length: 2878 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_evolut_qpad206.pdf. Length: 2053 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_nar_gkp1049.pdf. Length: 2707 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_sysbio_syy011.pdf. Length: 2334 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsbl.2015.0113.pdf. Length: 3034 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.160417.pdf. Length: 4642 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.170796.pdf. Length: 4815 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.190841.pdf. Length: 3764 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.210982.pdf. Length: 3380 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.0371.pdf. Length: 3692 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.1498.pdf. Length: 4078 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.2726.pdf. Length: 2961 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.2764.pdf. Length: 3570 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_096297.pdf. Length: 1315 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_2022.02.10.480011.pdf. Length: 1587 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_2022.07.21.501061.pdf. Length: 938 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1103_physrevresearch.4.023008.pdf. Length: 2961 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s0108270106019688.pdf. Length: 9560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536807066780.pdf. Length: 2469 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536808011148.pdf. Length: 1758 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809014883.pdf. Length: 2173 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809022582.pdf. Length: 2053 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809043700.pdf. Length: 2151 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810036299.pdf. Length: 2851 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810047185.pdf. Length: 2076 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536811003163.pdf. Length: 2169 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s160053681102201x.pdf. Length: 2037 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s160053681103220x.pdf. Length: 1855 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536811042164.pdf. Length: 2735 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812014614.pdf. Length: 2300 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812019885.pdf. Length: 2471 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812024269.pdf. Length: 2194 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812027390.pdf. Length: 2140 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812046892.pdf. Length: 2623 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252514012081.pdf. Length: 2311 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515010271.pdf. Length: 1927 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515011665.pdf. Length: 1970 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515023945.pdf. Length: 1828 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252520013603.pdf. Length: 2063 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2053229620003757.pdf. Length: 2091 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989014026358.pdf. Length: 1632 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989015010257.pdf. Length: 1871 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989015019891.pdf. Length: 2085 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989016016819.pdf. Length: 1970 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989020010658.pdf. Length: 1575 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2059798322005691.pdf. Length: 4835 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616000523.pdf. Length: 1345 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616007033.pdf. Length: 1450 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616007227.pdf. Length: 1478 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314619012446.pdf. Length: 1459 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1109_access.2024.3385658.pdf. Length: 2320 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13087.pdf. Length: 3164 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13431.pdf. Length: 4034 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13569.pdf. Length: 3892 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12382.pdf. Length: 3407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12491.pdf. Length: 3164 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12501.pdf. Length: 3183 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12594.pdf. Length: 3039 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13128.pdf. Length: 3619 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13136.pdf. Length: 3163 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13168.pdf. Length: 2881 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13446.pdf. Length: 2852 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2745.13449.pdf. Length: 3054 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.12453.pdf. Length: 3342 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.13817.pdf. Length: 2948 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_acel.13089.pdf. Length: 4260 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_bph.14747.pdf. Length: 7893 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_cas.12935.pdf. Length: 3470 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ddi.13153.pdf. Length: 5018 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ddi.13161.pdf. Length: 2405 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ejn.13855.pdf. Length: 3293 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12151.pdf. Length: 3109 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12446.pdf. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12768.pdf. Length: 2415 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_evo.13972.pdf. Length: 1341 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_gcb.13914.pdf. Length: 2721 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_gcb.14974.pdf. Length: 2678 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ggr.12517.pdf. Length: 3235 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_mec.16743.pdf. Length: 4767 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_mec.16977.pdf. Length: 2530 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_njb.02077.pdf. Length: 3577 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1128_JVI.01717-21.pdf. Length: 3680 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1128_spectrum.00422-24.pdf. Length: 2957 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1130_ges01387.1.pdf. Length: 7181 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1136_jitc-2021-003114.pdf. Length: 2416 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-016-4366-4.pdf. Length: 1625 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-018-6033-4.pdf. Length: 1256 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-018-6468-7.pdf. Length: 1465 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-019-6583-0.pdf. Length: 2422 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjds_s13688-018-0132-1.pdf. Length: 1830 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1145_3461702.3462538.pdf. Length: 13605 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1155_2013_536562.pdf. Length: 2731 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1177_0018720820970751.pdf. Length: 2304 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_1471-244x-13-259.pdf. Length: 1826 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_1752-153x-7-123.pdf. Length: 1652 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_gb-2010-11-10-r101.pdf. Length: 2362 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s11658-018-0120-2.pdf. Length: 4782 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s11689-019-9287-8.pdf. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0904-1.pdf. Length: 3411 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0922-z.pdf. Length: 6374 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0938-4.pdf. Length: 12271 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-1206-3.pdf. Length: 2200 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-017-1563-6.pdf. Length: 2461 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-017-1952-x.pdf. Length: 4174 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2036-2.pdf. Length: 1723 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2146-x.pdf. Length: 2335 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2263-6.pdf. Length: 5560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2414-9.pdf. Length: 2043 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-020-3415-z.pdf. Length: 2166 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-020-3496-8.pdf. Length: 1929 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12860-020-00261-6.pdf. Length: 3217 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-015-0558-z.pdf. Length: 5430 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-018-1213-2.pdf. Length: 1941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-019-1388-1.pdf. Length: 2441 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-019-1509-x.pdf. Length: 2369 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12863-017-0513-7.pdf. Length: 2264 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12863-019-0790-4.pdf. Length: 2410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-015-2206-9.pdf. Length: 2294 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-4947-8.pdf. Length: 1748 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-5054-6.pdf. Length: 2019 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-5198-4.pdf. Length: 3434 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-5872-1.pdf. Length: 2834 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6099-x.pdf. Length: 2474 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6131-1.pdf. Length: 16101 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6324-7.pdf. Length: 2373 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-020-6657-2.pdf. Length: 2206 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-015-0509-2.pdf. Length: 2333 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-019-1542-3.pdf. Length: 2196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-020-01863-y.pdf. Length: 6054 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12868-018-0468-2.pdf. Length: 2778 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12868-019-0514-8.pdf. Length: 2537 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-018-1332-3.pdf. Length: 1956 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-018-1542-8.pdf. Length: 4421 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1632-2.pdf. Length: 3435 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1889-5.pdf. Length: 2279 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1904-x.pdf. Length: 1981 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1926-4.pdf. Length: 2561 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-2199-7.pdf. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-02651-6.pdf. Length: 2190 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-02692-x.pdf. Length: 1924 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-2295-8.pdf. Length: 4194 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12874-018-0583-x.pdf. Length: 2313 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12874-019-0829-2.pdf. Length: 2365 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12879-018-3065-1.pdf. Length: 2033 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12879-019-3766-0.pdf. Length: 2025 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12881-019-0773-3.pdf. Length: 1866 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12882-019-1494-8.pdf. Length: 1883 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12884-018-1751-z.pdf. Length: 1939 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-017-3237-1.pdf. Length: 2101 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4229-5.pdf. Length: 2466 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4314-9.pdf. Length: 3119 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4768-9.pdf. Length: 2299 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-019-5681-6.pdf. Length: 3968 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-020-06724-5.pdf. Length: 3915 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12903-018-0656-6.pdf. Length: 2137 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12906-019-2769-0.pdf. Length: 4598 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12913-018-3333-1.pdf. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12915-018-0498-3.pdf. Length: 2585 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12915-018-0584-6.pdf. Length: 2008 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12916-019-1469-4.pdf. Length: 2233 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-015-0209-4.pdf. Length: 1862 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-018-0564-z.pdf. Length: 18518 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-018-0620-8.pdf. Length: 2622 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0379-1.pdf. Length: 2325 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0403-5.pdf. Length: 2601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0426-y.pdf. Length: 3744 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0611-7.pdf. Length: 3109 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0628-y.pdf. Length: 6111 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0646-9.pdf. Length: 3572 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-020-00737-6.pdf. Length: 3482 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12931-019-1001-6.pdf. Length: 1891 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12933-019-0813-5.pdf. Length: 2800 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12934-018-1005-9.pdf. Length: 3071 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12935-020-01373-x.pdf. Length: 1815 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12943-018-0878-x.pdf. Length: 2121 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12943-019-1017-z.pdf. Length: 5437 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12967-019-2100-3.pdf. Length: 2269 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12967-020-02311-1.pdf. Length: 9384 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12974-020-01860-y.pdf. Length: 2301 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12977-015-0204-2.pdf. Length: 2357 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13007-019-0403-2.pdf. Length: 1851 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13014-018-1166-z.pdf. Length: 1790 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13020-020-00307-z.pdf. Length: 2621 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0254-8.pdf. Length: 4249 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0266-4.pdf. Length: 4062 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13039-022-00631-z.pdf. Length: 4653 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13046-018-0843-y.pdf. Length: 2545 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-015-0738-7.pdf. Length: 1839 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-015-1052-0.pdf. Length: 3049 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-019-2488-4.pdf. Length: 2044 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13058-015-0618-8.pdf. Length: 2381 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-017-1200-8.pdf. Length: 2545 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-018-1557-3.pdf. Length: 2792 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1749-5.pdf. Length: 2482 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1840-y.pdf. Length: 1527 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1908-8.pdf. Length: 2247 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1918-6.pdf. Length: 2535 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1924-8.pdf. Length: 12778 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-020-02048-6.pdf. Length: 3261 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-020-1949-z.pdf. Length: 7407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-017-0901-2.pdf. Length: 3153 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1078-z.pdf. Length: 2380 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1167-z.pdf. Length: 2094 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1316-4.pdf. Length: 2658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-016-1359-y.pdf. Length: 2293 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-2842-4.pdf. Length: 1859 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-2964-8.pdf. Length: 1885 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-3084-1.pdf. Length: 2209 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-3237-2.pdf. Length: 2188 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13072-019-0285-6.pdf. Length: 1998 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13072-019-0322-5.pdf. Length: 1455 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0674-2.pdf. Length: 2638 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0709-8.pdf. Length: 2702 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-020-00727-4.pdf. Length: 2752 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13100-019-0153-8.pdf. Length: 3786 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-018-4014-1.pdf. Length: 1343 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-019-4058-x.pdf. Length: 1538 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-019-4127-1.pdf. Length: 1383 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13148-018-0581-1.pdf. Length: 2062 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13148-019-0719-9.pdf. Length: 1854 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-015-0110-6.pdf. Length: 1751 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-017-0223-1.pdf. Length: 9834 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-017-0246-7.pdf. Length: 1725 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-018-0298-3.pdf. Length: 5704 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13326-019-0195-3.pdf. Length: 1980 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13568-018-0680-6.pdf. Length: 2366 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13617-019-0084-6.pdf. Length: 2180 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13643-018-0859-6.pdf. Length: 2703 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13750-020-0184-0.pdf. Length: 3382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-017-0338-7.pdf. Length: 3612 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-018-0532-2.pdf. Length: 2338 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-018-0550-0.pdf. Length: 2782 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40169-019-0253-6.pdf. Length: 2822 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40170-018-0185-4.pdf. Length: 2320 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40170-020-00212-x.pdf. Length: 2387 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40425-019-0730-x.pdf. Length: 1994 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40478-018-0561-x.pdf. Length: 7938 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40478-018-0655-5.pdf. Length: 2392 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40657-018-0125-6.pdf. Length: 2153 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40657-020-00194-w.pdf. Length: 3680 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40793-015-0095-9.pdf. Length: 2233 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40851-018-0089-8.pdf. Length: 2520 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1212_wnl.0000000000006035.pdf. Length: 2831 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_bio.035071.pdf. Length: 4039 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_dev.138545.pdf. Length: 4145 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_jeb.211466.pdf. Length: 7030 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.10556.1.pdf. Length: 3375 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.11698.1.pdf. Length: 5087 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13064.1.pdf. Length: 3319 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13483.1.pdf. Length: 3172 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13622.1.pdf. Length: 4142 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13852.1.pdf. Length: 2588 characters\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.4660.1.pdf. Length: 3015 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1364_oe.25.001985.pdf. Length: 742 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pcbi.1011828.pdf. Length: 9397 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pntd.0005385.pdf. Length: 2713 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0070749.pdf. Length: 3545 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0093021.pdf. Length: 3326 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0096457.pdf. Length: 3372 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0117675.pdf. Length: 2160 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0137181.pdf. Length: 2338 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0139215.pdf. Length: 2795 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0146274.pdf. Length: 6533 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0159387.pdf. Length: 2661 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0170126.pdf. Length: 1983 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0188323.pdf. Length: 2226 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0191086.pdf. Length: 2657 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0198382.pdf. Length: 2244 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0200041.pdf. Length: 3159 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0212669.pdf. Length: 4267 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0220399.pdf. Length: 2483 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0253228.pdf. Length: 2577 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0262974.pdf. Length: 2601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0284951.pdf. Length: 2309 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.14358_pers.22-00039r2.pdf. Length: 1911 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.14379_iodp.proc.390393.208.2024.pdf. Length: 470 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_g3.119.400401.pdf. Length: 5374 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_g3.119.400993.pdf. Length: 8640 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_genetics.120.303190.pdf. Length: 2831 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0001-3765201920180768.pdf. Length: 876 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0001-3765201920190822.pdf. Length: 633 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0034-7167-2018-0065.pdf. Length: 1146 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0034-737x201966050002.pdf. Length: 2552 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0047-2085000000239.pdf. Length: 1825 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102-778632120150164.pdf. Length: 2188 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102-77863220007.pdf. Length: 1937 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102.3772e35417.pdf. Length: 1477 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0103-6351_4070.pdf. Length: 1273 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0103-6440201802861.pdf. Length: 2587 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-4060.59642.pdf. Length: 1260 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-530x4900-19.pdf. Length: 873 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-6632.20190362s20180340.pdf. Length: 1531 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1413-785220192703169516.pdf. Length: 2642 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1414-431x20198292.pdf. Length: 2449 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1516-3180.2018.0270071218.pdf. Length: 1782 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1517-869220192505126869.pdf. Length: 3436 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1519-6984.192126.pdf. Length: 2502 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4162-10032.pdf. Length: 1080 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4162-10752.pdf. Length: 2048 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4499.20190067.pdf. Length: 1057 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4685-gmb-2018-0055.pdf. Length: 2354 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1679-78255326.pdf. Length: 1988 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1679-78255753.pdf. Length: 1783 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-90882019000500004.pdf. Length: 1523 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-93042019000400002.pdf. Length: 1778 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-9479.2019.185555.pdf. Length: 375 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1807-3107bor-2019.vol33.0095.pdf. Length: 1530 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-2950_19003927012020.pdf. Length: 2533 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-2950_19008627012020.pdf. Length: 2571 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-4430-eng.agric.v40n1p24-33_2020.pdf. Length: 2287 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-6891v21e-43578.pdf. Length: 1259 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-9823.2007.10034.pdf. Length: 1327 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0766.pdf. Length: 2677 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0921.pdf. Length: 2612 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2019-0338.pdf. Length: 2732 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.032.ao27.pdf. Length: 1132 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.033.ao15.pdf. Length: 1083 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-22562019022.190037.pdf. Length: 1572 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-52712015v43n3rb20180090.pdf. Length: 1645 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-6723.06017.pdf. Length: 1673 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-86372019000243506.pdf. Length: 2250 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1982-0216_20192158519.pdf. Length: 1479 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1983-21252019v32n324rc.pdf. Length: 1495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_2236-3459_83528.pdf. Length: 802 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_2237-6089-2017-0134.pdf. Length: 2925 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_fst.33717.pdf. Length: 1396 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_fst.36617.pdf. Length: 2078 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1677-5538.ibju.2019.0167.pdf. Length: 1751 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1678-86212019000400340.pdf. Length: 2197 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1617_s11527-023-02260-3.pdf. Length: 862 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.17581_bp.2020.09104.pdf. Length: 351 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.17815_jlsrf-2-58.pdf. Length: 1709 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.18438_eblip29674.pdf. Length: 2847 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.20944_preprints202009.0353.v1.pdf. Length: 1932 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21105_joss.04237.pdf. Length: 2363 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21203_rs.3.rs-2814013_v1.pdf. Length: 1933 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21203_rs.3.rs-3338732_v1.pdf. Length: 3941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_cir1497.pdf. Length: 4324 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_fs20233046.pdf. Length: 2245 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20201035.pdf. Length: 1969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20231026.pdf. Length: 1218 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20231027.pdf. Length: 8380 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fcell.2020.618552.pdf. Length: 13637 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fchem.2019.00828.pdf. Length: 2393 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fcimb.2024.1292467.pdf. Length: 2716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_feart.2023.1205211.pdf. Length: 4593 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fevo.2023.1112519.pdf. Length: 2728 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fmicb.2024.1456637.pdf. Length: 2495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fpls.2023.1160645.pdf. Length: 2886 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_cryst2031058.pdf. Length: 1063 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_cryst9060301.pdf. Length: 1245 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_d13010019.pdf. Length: 2296 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_ijms12117360.pdf. Length: 1741 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_inorganics7050058.pdf. Length: 864 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_microorganisms8121872.pdf. Length: 1196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules17077645.pdf. Length: 1493 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules191017026.pdf. Length: 1252 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules200917469.pdf. Length: 1077 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules23123199.pdf. Length: 1235 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules25010053.pdf. Length: 1346 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_rs12121957.pdf. Length: 1840 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_rs4102923.pdf. Length: 6633 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s19030479.pdf. Length: 2020 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s23094491.pdf. Length: 3222 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s23177333.pdf. Length: 2470 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_v11060565.pdf. Length: 977 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3762_bjoc.8.42.pdf. Length: 676 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3762_bjoc.9.92.pdf. Length: 856 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3847_1538-4357_aae92c.pdf. Length: 1539 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_bdj.7.e47369.pdf. Length: 3839 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_jhr.96.111550.pdf. Length: 3637 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_neobiota.82.87455.pdf. Length: 4083 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zookeys.500.9360.pdf. Length: 2191 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.35.e23481.pdf. Length: 1841 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.36.e32053.pdf. Length: 821 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.4025_actasciagron.v40i1.39392.pdf. Length: 2590 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-2021-570.pdf. Length: 4835 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-22-2769-2022.pdf. Length: 2846 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-22-5701-2022.pdf. Length: 6234 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_amt-15-3969-2022.pdf. Length: 2765 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-12-1287-2020.pdf. Length: 4424 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2019-206.pdf. Length: 2036 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-187.pdf. Length: 6713 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-198.pdf. Length: 2138 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-8-663-2016.pdf. Length: 1924 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_gmd-12-4221-2019.pdf. Length: 3360 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_gmd-13-335-2020.pdf. Length: 3185 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_se-13-1541-2022.pdf. Length: 3984 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_tc-17-3617-2023.pdf. Length: 12559 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5937_bnhmb1811227u.pdf. Length: 1914 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.63194.pdf. Length: 2568 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.72626.pdf. Length: 1817 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.29944.pdf. Length: 4436 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.42722.pdf. Length: 1748 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.62329.pdf. Length: 644 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.63455.pdf. Length: 1736 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.73695.pdf. Length: 948 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.74937.pdf. Length: 727 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.10452.pdf. Length: 10028 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.11352.pdf. Length: 2135 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.12422.pdf. Length: 1678 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.13193.pdf. Length: 2907 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml. Length: 1394 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml. Length: 1007 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml. Length: 201 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202007717.xml. Length: 1382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201902131.xml. Length: 1152 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201903120.xml. Length: 1904 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml. Length: 767 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001412.xml. Length: 1135 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001668.xml. Length: 804 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202003167.xml. Length: 1842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_cssc.202201821.xml. Length: 1244 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.3985.xml. Length: 1618 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.4466.xml. Length: 1751 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5260.xml. Length: 1710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5395.xml. Length: 3321 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6144.xml. Length: 1921 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6303.xml. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6784.xml. Length: 2540 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.961.xml. Length: 1674 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.9627.xml. Length: 2255 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ecs2.4619.xml. Length: 1960 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejic.201900904.xml. Length: 1001 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejoc.202000916.xml. Length: 559 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_esp.5090.xml. Length: 2683 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_mp.14424.xml. Length: 2792 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00259-022-06053-8.xml. Length: 3454 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00382-022-06361-7.xml. Length: 1842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00442-022-05201-z.xml. Length: 1887 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12263-014-0408-4.xml. Length: 2200 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12559-020-09751-3.xml. Length: 2918 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ast.2022.107401.xml. Length: 815 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.cpc.2024.109087.xml. Length: 1725 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2022.108797.xml. Length: 1439 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2023.109949.xml. Length: 1623 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ecolind.2021.107934.xml. Length: 1914 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.jobe.2023.107105.xml. Length: 2078 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.molcel.2018.11.006.xml. Length: 1229 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.scitotenv.2024.171850.xml. Length: 769 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1017_s0007123423000601.xml. Length: 1203 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_acscentsci.1c01109.xml. Length: 1602 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_acsomega.3c06074.xml. Length: 2878 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_jacs.2c06519.xml. Length: 1588 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2018gl078007.xml. Length: 2408 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2020jf005675.xml. Length: 2098 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2021gl096173.xml. Length: 364 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2022gl100473.xml. Length: 843 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2023ea002840.xml. Length: 2151 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2023wr035126.xml. Length: 374 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_hdy.2015.99.xml. Length: 1681 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_ncomms11871.xml. Length: 1004 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41396-020-00885-8.xml. Length: 2202 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41437-020-0318-8.xml. Length: 2086 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-018-04041-x.xml. Length: 1950 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-018-07681-1.xml. Length: 1675 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-019-10357-z.xml. Length: 2148 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41558-022-01301-z.xml. Length: 1404 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41562-021-01247-w.xml. Length: 1234 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-019-0101-y.xml. Length: 1908 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-022-01555-4.xml. Length: 3511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-023-02280-2.xml. Length: 1943 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-017-15852-1.xml. Length: 1483 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-020-59839-x.xml. Length: 2063 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-021-85671-y.xml. Length: 2588 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-024-56373-y.xml. Length: 2228 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_sdata.2017.167.xml. Length: 1366 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c5sc01309g.xml. Length: 609 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c9ra06638a.xml. Length: 569 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c9sc02930c.xml. Length: 752 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc01197e.xml. Length: 2133 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc01518k.xml. Length: 1119 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc02283g.xml. Length: 1728 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc02676j.xml. Length: 1422 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc03604h.xml. Length: 1190 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d1ee03696c.xml. Length: 2236 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d2cc00847e.xml. Length: 749 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1055_s-0039-1693681.xml. Length: 1711 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1073_pnas.2306723120.xml. Length: 1040 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_14756366.2020.1740692.xml. Length: 1593 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_15476286.2016.1232238.xml. Length: 2055 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_21645515.2023.2189598.xml. Length: 2499 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_22221751.2020.1738277.xml. Length: 2081 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arad016.xml. Length: 1846 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arw167.xml. Length: 899 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arz101.xml. Length: 2119 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_bib_bbab395.xml. Length: 1205 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_evolut_qpad206.xml. Length: 2437 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_nar_gkp1049.xml. Length: 1451 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_sysbio_syy011.xml. Length: 2094 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsbl.2015.0113.xml. Length: 906 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.160417.xml. Length: 2520 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.170796.xml. Length: 2168 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.190841.xml. Length: 2129 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.210982.xml. Length: 1615 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.0371.xml. Length: 1834 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.1498.xml. Length: 1508 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.2726.xml. Length: 1474 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1101_2022.02.10.480011.xml. Length: 778 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1101_2022.07.21.501061.xml. Length: 1653 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1103_physrevresearch.4.023008.xml. Length: 1710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536807066780.xml. Length: 925 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536808011148.xml. Length: 617 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809014883.xml. Length: 730 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809022582.xml. Length: 642 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809043700.xml. Length: 282 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536810036299.xml. Length: 753 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536810047185.xml. Length: 1071 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536811003163.xml. Length: 714 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s160053681102201x.xml. Length: 498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s160053681103220x.xml. Length: 852 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536811042164.xml. Length: 455 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812014614.xml. Length: 1298 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812019885.xml. Length: 634 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812024269.xml. Length: 626 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812027390.xml. Length: 1084 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812046892.xml. Length: 769 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252514012081.xml. Length: 511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515010271.xml. Length: 367 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515011665.xml. Length: 343 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515023945.xml. Length: 381 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252520013603.xml. Length: 308 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2053229620003757.xml. Length: 530 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989014026358.xml. Length: 545 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989015010257.xml. Length: 540 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989015019891.xml. Length: 714 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989016016819.xml. Length: 920 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989020010658.xml. Length: 560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2059798322005691.xml. Length: 435 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1109_access.2024.3385658.xml. Length: 2003 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2435.13431.xml. Length: 2581 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2435.13569.xml. Length: 2410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2656.12382.xml. Length: 1035 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2656.12491.xml. Length: 2183 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2664.13136.xml. Length: 2393 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_2041-210x.12453.xml. Length: 1904 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_2041-210x.13817.xml. Length: 1623 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_acel.13089.xml. Length: 1646 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_bph.14747.xml. Length: 2009 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_cas.12935.xml. Length: 1815 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ddi.13153.xml. Length: 2252 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ddi.13161.xml. Length: 2316 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ejn.13855.xml. Length: 1980 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12151.xml. Length: 1462 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12446.xml. Length: 1815 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12768.xml. Length: 1838 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_gcb.13914.xml. Length: 2032 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_gcb.14974.xml. Length: 1982 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ggr.12517.xml. Length: 1711 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_mec.16977.xml. Length: 1797 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_njb.02077.xml. Length: 1528 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1128_JVI.01717-21.xml. Length: 2237 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1128_spectrum.00422-24.xml. Length: 2986 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1130_ges01387.1.xml. Length: 2705 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1136_jitc-2021-003114.xml. Length: 1896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1140_epjc_s10052-016-4366-4.xml. Length: 2864 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1140_epjc_s10052-018-6468-7.xml. Length: 2838 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1145_3461702.3462538.xml. Length: 4254 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1155_2013_536562.xml. Length: 1314 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1177_0018720820970751.xml. Length: 1934 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_1471-244x-13-259.xml. Length: 2279 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_1752-153x-7-123.xml. Length: 2245 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_gb-2010-11-10-r101.xml. Length: 2797 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s11658-018-0120-2.xml. Length: 1843 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s11689-019-9287-8.xml. Length: 2862 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0904-1.xml. Length: 1755 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0922-z.xml. Length: 2140 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0938-4.xml. Length: 3184 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-1206-3.xml. Length: 1825 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-017-1563-6.xml. Length: 2356 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-017-1952-x.xml. Length: 2831 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2036-2.xml. Length: 2299 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2146-x.xml. Length: 2940 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2263-6.xml. Length: 1913 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2414-9.xml. Length: 2478 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-020-3415-z.xml. Length: 2615 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-020-3496-8.xml. Length: 3291 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12860-020-00261-6.xml. Length: 2906 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-015-0558-z.xml. Length: 2365 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-018-1213-2.xml. Length: 2237 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-019-1388-1.xml. Length: 2154 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-019-1509-x.xml. Length: 1890 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12863-017-0513-7.xml. Length: 1893 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12863-019-0790-4.xml. Length: 2408 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-015-2206-9.xml. Length: 1986 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-4947-8.xml. Length: 2221 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-5054-6.xml. Length: 2417 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-5198-4.xml. Length: 2641 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-5872-1.xml. Length: 2369 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6099-x.xml. Length: 2657 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6131-1.xml. Length: 2340 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6324-7.xml. Length: 2606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-020-6657-2.xml. Length: 2010 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-015-0509-2.xml. Length: 2193 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-019-1542-3.xml. Length: 2033 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-020-01863-y.xml. Length: 2723 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12868-018-0468-2.xml. Length: 2540 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12868-019-0514-8.xml. Length: 2706 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-018-1332-3.xml. Length: 2218 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-018-1542-8.xml. Length: 2818 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1632-2.xml. Length: 2555 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1889-5.xml. Length: 2415 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1904-x.xml. Length: 2446 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1926-4.xml. Length: 2464 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-2199-7.xml. Length: 2128 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-02651-6.xml. Length: 2680 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-02692-x.xml. Length: 3100 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-2295-8.xml. Length: 2853 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12874-018-0583-x.xml. Length: 2047 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12874-019-0829-2.xml. Length: 1960 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12879-018-3065-1.xml. Length: 2193 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12879-019-3766-0.xml. Length: 2487 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12881-019-0773-3.xml. Length: 2236 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12882-019-1494-8.xml. Length: 2462 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12884-018-1751-z.xml. Length: 2296 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-017-3237-1.xml. Length: 2229 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4229-5.xml. Length: 2197 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4314-9.xml. Length: 2181 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4768-9.xml. Length: 2187 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-019-5681-6.xml. Length: 2359 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-020-06724-5.xml. Length: 2781 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12903-018-0656-6.xml. Length: 1772 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12906-019-2769-0.xml. Length: 3350 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12913-018-3333-1.xml. Length: 2099 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12915-018-0498-3.xml. Length: 2144 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12915-018-0584-6.xml. Length: 2207 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12916-019-1469-4.xml. Length: 2968 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-015-0209-4.xml. Length: 2348 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-018-0564-z.xml. Length: 2173 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-018-0620-8.xml. Length: 3221 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0379-1.xml. Length: 2454 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0403-5.xml. Length: 2266 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0426-y.xml. Length: 2242 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0611-7.xml. Length: 1829 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0628-y.xml. Length: 2413 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0646-9.xml. Length: 2495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-020-00737-6.xml. Length: 2786 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12931-019-1001-6.xml. Length: 2333 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12933-019-0813-5.xml. Length: 2377 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12934-018-1005-9.xml. Length: 2887 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12935-020-01373-x.xml. Length: 1674 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12943-018-0878-x.xml. Length: 1767 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12943-019-1017-z.xml. Length: 2247 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12967-019-2100-3.xml. Length: 2350 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12967-020-02311-1.xml. Length: 2868 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12974-020-01860-y.xml. Length: 2551 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12977-015-0204-2.xml. Length: 2179 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13007-019-0403-2.xml. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13014-018-1166-z.xml. Length: 2276 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13020-020-00307-z.xml. Length: 2325 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13024-018-0254-8.xml. Length: 2447 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13024-018-0266-4.xml. Length: 2616 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13039-022-00631-z.xml. Length: 2985 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13046-018-0843-y.xml. Length: 2430 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-015-0738-7.xml. Length: 2221 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-015-1052-0.xml. Length: 2317 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-019-2488-4.xml. Length: 2346 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13058-015-0618-8.xml. Length: 1694 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-017-1200-8.xml. Length: 1034 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-018-1557-3.xml. Length: 2075 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1749-5.xml. Length: 2383 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1840-y.xml. Length: 1720 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1908-8.xml. Length: 2969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1918-6.xml. Length: 2210 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1924-8.xml. Length: 2868 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-020-02048-6.xml. Length: 2522 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-020-1949-z.xml. Length: 2809 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-017-0901-2.xml. Length: 2528 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1078-z.xml. Length: 2231 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1167-z.xml. Length: 2567 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1316-4.xml. Length: 2083 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-016-1359-y.xml. Length: 2114 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-2842-4.xml. Length: 2215 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-2964-8.xml. Length: 3106 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-3084-1.xml. Length: 2249 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-3237-2.xml. Length: 2129 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13072-019-0285-6.xml. Length: 2061 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13072-019-0322-5.xml. Length: 1926 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-019-0674-2.xml. Length: 2697 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-019-0709-8.xml. Length: 3427 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-020-00727-4.xml. Length: 2176 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13100-019-0153-8.xml. Length: 2701 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-018-4014-1.xml. Length: 1520 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-019-4058-x.xml. Length: 1770 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-019-4127-1.xml. Length: 1687 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13148-018-0581-1.xml. Length: 2145 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13148-019-0719-9.xml. Length: 2395 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-015-0110-6.xml. Length: 2079 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-017-0223-1.xml. Length: 3404 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-017-0246-7.xml. Length: 2660 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-018-0298-3.xml. Length: 2410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13326-019-0195-3.xml. Length: 2133 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13568-018-0680-6.xml. Length: 2258 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13643-018-0859-6.xml. Length: 2592 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-017-0338-7.xml. Length: 2969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-018-0532-2.xml. Length: 3586 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-018-0550-0.xml. Length: 2679 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40169-019-0253-6.xml. Length: 2491 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40170-018-0185-4.xml. Length: 2115 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40170-020-00212-x.xml. Length: 1743 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40425-019-0730-x.xml. Length: 2666 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40478-018-0561-x.xml. Length: 2866 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40478-018-0655-5.xml. Length: 2114 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40793-015-0095-9.xml. Length: 1509 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40851-018-0089-8.xml. Length: 2129 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1212_wnl.0000000000006035.xml. Length: 1761 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1242_bio.035071.xml. Length: 1565 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1242_dev.138545.xml. Length: 2388 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.10556.1.xml. Length: 1671 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.11698.1.xml. Length: 1681 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13064.1.xml. Length: 2190 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13483.1.xml. Length: 2163 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13622.1.xml. Length: 2723 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13852.1.xml. Length: 959 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.4660.1.xml. Length: 1622 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pcbi.1011828.xml. Length: 2891 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pntd.0005385.xml. Length: 1891 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0070749.xml. Length: 1808 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0093021.xml. Length: 2219 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0096457.xml. Length: 1053 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0117675.xml. Length: 1842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0137181.xml. Length: 1822 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0139215.xml. Length: 2176 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0146274.xml. Length: 2180 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0159387.xml. Length: 1830 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0170126.xml. Length: 1409 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0188323.xml. Length: 1856 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0191086.xml. Length: 2019 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0198382.xml. Length: 2187 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0200041.xml. Length: 2394 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0212669.xml. Length: 2174 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0220399.xml. Length: 1590 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0253228.xml. Length: 1539 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0262974.xml. Length: 2613 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0284951.xml. Length: 1266 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1534_g3.119.400401.xml. Length: 2445 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1534_g3.119.400993.xml. Length: 3310 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1413-785220192703169516.xml. Length: 1545 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1414-431x20198292.xml. Length: 1628 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1516-3180.2018.0270071218.xml. Length: 1979 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1678-4685-gmb-2018-0055.xml. Length: 1387 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_s1677-5538.ibju.2019.0167.xml. Length: 1857 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1617_s11527-023-02260-3.xml. Length: 1404 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.17581_bp.2020.09104.xml. Length: 2392 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.20944_preprints202009.0353.v1.xml. Length: 1363 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.21105_joss.04237.xml. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.21203_rs.3.rs-3338732_v1.xml. Length: 2164 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_cir1497.xml. Length: 2217 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_fs20233046.xml. Length: 736 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20201035.xml. Length: 510 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20231026.xml. Length: 288 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20231027.xml. Length: 1172 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fcell.2020.618552.xml. Length: 2475 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fchem.2019.00828.xml. Length: 1076 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fcimb.2024.1292467.xml. Length: 1923 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_feart.2023.1205211.xml. Length: 3480 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fmicb.2024.1456637.xml. Length: 2498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fpls.2023.1160645.xml. Length: 2241 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_d13010019.xml. Length: 1713 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_ijms12117360.xml. Length: 1848 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_inorganics7050058.xml. Length: 956 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_microorganisms8121872.xml. Length: 1490 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules17077645.xml. Length: 1988 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules191017026.xml. Length: 2038 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules200917469.xml. Length: 1447 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules23123199.xml. Length: 1541 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules25010053.xml. Length: 1688 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_rs4102923.xml. Length: 2270 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s19030479.xml. Length: 2120 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s23094491.xml. Length: 2457 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s23177333.xml. Length: 2447 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_v11060565.xml. Length: 1048 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3762_bjoc.8.42.xml. Length: 739 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3762_bjoc.9.92.xml. Length: 962 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_bdj.7.e47369.xml. Length: 1612 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_neobiota.82.87455.xml. Length: 2152 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zookeys.500.9360.xml. Length: 2256 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zoologia.35.e23481.xml. Length: 1940 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zoologia.36.e32053.xml. Length: 945 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-2021-570.xml. Length: 2143 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-22-2769-2022.xml. Length: 2282 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-22-5701-2022.xml. Length: 752 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_amt-15-3969-2022.xml. Length: 310 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-12-1287-2020.xml. Length: 1995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2019-206.xml. Length: 1196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2023-187.xml. Length: 2694 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2023-198.xml. Length: 1461 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_gmd-12-4221-2019.xml. Length: 1905 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_gmd-13-335-2020.xml. Length: 2422 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5937_bnhmb1811227u.xml. Length: 1665 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_eLife.63194.xml. Length: 2246 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_eLife.72626.xml. Length: 2076 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.29944.xml. Length: 1329 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.42722.xml. Length: 1271 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.62329.xml. Length: 1658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.63455.xml. Length: 1919 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.73695.xml. Length: 2194 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.74937.xml. Length: 1556 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.10452.xml. Length: 1307 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.11352.xml. Length: 1672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.12422.xml. Length: 1672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.13193.xml. Length: 2329 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf. Length: 4770 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf. Length: 1306 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf. Length: 1597 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202007717.pdf. Length: 1669 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201902131.pdf. Length: 616 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf. Length: 846 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202000235.pdf. Length: 674 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf. Length: 941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001668.pdf. Length: 404 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202003167.pdf. Length: 883 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_cssc.202201821.pdf. Length: 3837 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.3985.pdf. Length: 3599 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.4466.pdf. Length: 2282 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5395.pdf. Length: 3908 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf. Length: 2895 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6303.pdf. Length: 3217 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6784.pdf. Length: 6695 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.961.pdf. Length: 2786 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.9627.pdf. Length: 2737 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf. Length: 3162 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.4619.pdf. Length: 3416 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf. Length: 586 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf. Length: 924 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000916.pdf. Length: 272 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf. Length: 4447 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5090.pdf. Length: 5294 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf. Length: 4496 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_nafm.10870.pdf. Length: 4396 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1007_jhep07(2018)134.pdf. Length: 849 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml. Length: 1394 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml. Length: 1007 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml. Length: 201 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202007717.xml. Length: 1382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201902131.xml. Length: 1152 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201903120.xml. Length: 1904 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202000235.xml. Length: 767 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001412.xml. Length: 1135 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001668.xml. Length: 804 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202003167.xml. Length: 1842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_cssc.202201821.xml. Length: 1244 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.3985.xml. Length: 1618 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.4466.xml. Length: 1751 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5260.xml. Length: 1710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5395.xml. Length: 3321 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6144.xml. Length: 1921 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6303.xml. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6784.xml. Length: 2540 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.961.xml. Length: 1674 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.9627.xml. Length: 2255 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ecs2.4619.xml. Length: 1960 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejic.201900904.xml. Length: 1001 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejoc.202000916.xml. Length: 559 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_esp.5090.xml. Length: 2683 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_mp.14424.xml. Length: 2792 characters\n",
      "Training file paths DataFrame shape: (524, 9)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "aaf366a3-d922-49d2-8547-376d4f62c45b",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018). \\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database. \\nJournal of Geophysical Research: Oceans, 123, 1229\\u20131250.\\nhttps://doi. \\norg/10.1002/2017JC013030 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557\\u2013575.\\nhttps://doi.org/10.1016/j.rse. \\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field an\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388 Barton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181\\u2013197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB). \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], context: \\\" 15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism. \\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5] \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], context: \\\" 15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 9,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract Characterizin...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: The homolept...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"abstract\":\"Abstract Characterizin...   \n",
       "1  {\"title\":\"\",\"abstract\":\"Abstract: The homolept...   \n",
       "2  {\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...   \n",
       "\n",
       "                                            xml_text  \n",
       "0  {\"title\":\"Assessing the variability in the rel...  \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...  \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Training file paths DataFrame shape: (30, 10)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "947e413f-e1ff-47b1-8a66-58ac0f9c136e",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018). \\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database. \\nJournal of Geophysical Research: Oceans, 123, 1229\\u20131250.\\nhttps://doi. \\norg/10.1002/2017JC013030 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557\\u2013575.\\nhttps://doi.org/10.1016/j.rse. \\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field an\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388 Barton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181\\u2013197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}",
         "{\"title\":\"\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018). \\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database. \\nJournal of Geophysical Research: Oceans, 123, 1229\\u20131250.\\nhttps://doi. \\norg/10.1002/2017JC013030 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557\\u2013575.\\nhttps://doi.org/10.1016/j.rse. \\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field an\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388 Barton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181\\u2013197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"}\"]}\n{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB). \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], context: \\\" 15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB). \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], context: \\\" 15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}\n{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism. \\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5] \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], context: \\\" 15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism. \\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5] \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], context: \\\" 15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}\n{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 10,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract Characterizin...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract Characterizin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: The homolept...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: The homolept...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"abstract\":\"Abstract Characterizin...   \n",
       "1  {\"title\":\"\",\"abstract\":\"Abstract: The homolept...   \n",
       "2  {\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...   \n",
       "\n",
       "                                            xml_text  \\\n",
       "0  {\"title\":\"Assessing the variability in the rel...   \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...   \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...   \n",
       "\n",
       "                                                text  \n",
       "0  {\"title\":\"\",\"abstract\":\"Abstract Characterizin...  \n",
       "1  {\"title\":\"\",\"abstract\":\"Abstract: The homolept...  \n",
       "2  {\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "ename": "OSError",
     "evalue": "Cannot save file into a non-existent directory: 'kaggle\\working'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mOSError\u001b[39m                                   Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[92]\u001b[39m\u001b[32m, line 19\u001b[39m\n\u001b[32m     17\u001b[39m display(test_file_paths_df.head(\u001b[32m3\u001b[39m))\n\u001b[32m     18\u001b[39m \u001b[38;5;66;03m# Save the training and test file paths DataFrames to CSV files\u001b[39;00m\n\u001b[32m---> \u001b[39m\u001b[32m19\u001b[39m \u001b[43mtrain_file_paths_df\u001b[49m\u001b[43m.\u001b[49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mos\u001b[49m\u001b[43m.\u001b[49m\u001b[43mpath\u001b[49m\u001b[43m.\u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m(\u001b[49m\u001b[43mBASE_OUTPUT_DIR\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[33;43m'\u001b[39;49m\u001b[33;43mtrain_file_paths.csv\u001b[39;49m\u001b[33;43m'\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[32m     20\u001b[39m test_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, \u001b[33m'\u001b[39m\u001b[33mtest_file_paths.csv\u001b[39m\u001b[33m'\u001b[39m), index=\u001b[38;5;28;01mFalse\u001b[39;00m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\util\\_decorators.py:333\u001b[39m, in \u001b[36mdeprecate_nonkeyword_arguments.<locals>.decorate.<locals>.wrapper\u001b[39m\u001b[34m(*args, **kwargs)\u001b[39m\n\u001b[32m    327\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(args) > num_allow_args:\n\u001b[32m    328\u001b[39m     warnings.warn(\n\u001b[32m    329\u001b[39m         msg.format(arguments=_format_argument_list(allow_args)),\n\u001b[32m    330\u001b[39m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m,\n\u001b[32m    331\u001b[39m         stacklevel=find_stack_level(),\n\u001b[32m    332\u001b[39m     )\n\u001b[32m--> \u001b[39m\u001b[32m333\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\core\\generic.py:3986\u001b[39m, in \u001b[36mNDFrame.to_csv\u001b[39m\u001b[34m(self, path_or_buf, sep, na_rep, float_format, columns, header, index, index_label, mode, encoding, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, decimal, errors, storage_options)\u001b[39m\n\u001b[32m   3975\u001b[39m df = \u001b[38;5;28mself\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(\u001b[38;5;28mself\u001b[39m, ABCDataFrame) \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28mself\u001b[39m.to_frame()\n\u001b[32m   3977\u001b[39m formatter = DataFrameFormatter(\n\u001b[32m   3978\u001b[39m     frame=df,\n\u001b[32m   3979\u001b[39m     header=header,\n\u001b[32m   (...)\u001b[39m\u001b[32m   3983\u001b[39m     decimal=decimal,\n\u001b[32m   3984\u001b[39m )\n\u001b[32m-> \u001b[39m\u001b[32m3986\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mDataFrameRenderer\u001b[49m\u001b[43m(\u001b[49m\u001b[43mformatter\u001b[49m\u001b[43m)\u001b[49m\u001b[43m.\u001b[49m\u001b[43mto_csv\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m   3987\u001b[39m \u001b[43m    \u001b[49m\u001b[43mpath_or_buf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3988\u001b[39m \u001b[43m    \u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[43m=\u001b[49m\u001b[43mlineterminator\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3989\u001b[39m \u001b[43m    \u001b[49m\u001b[43msep\u001b[49m\u001b[43m=\u001b[49m\u001b[43msep\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3990\u001b[39m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[43m=\u001b[49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3991\u001b[39m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[43m=\u001b[49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3992\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3993\u001b[39m \u001b[43m    \u001b[49m\u001b[43mquoting\u001b[49m\u001b[43m=\u001b[49m\u001b[43mquoting\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3994\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3995\u001b[39m \u001b[43m    \u001b[49m\u001b[43mindex_label\u001b[49m\u001b[43m=\u001b[49m\u001b[43mindex_label\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3996\u001b[39m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3997\u001b[39m \u001b[43m    \u001b[49m\u001b[43mchunksize\u001b[49m\u001b[43m=\u001b[49m\u001b[43mchunksize\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3998\u001b[39m \u001b[43m    \u001b[49m\u001b[43mquotechar\u001b[49m\u001b[43m=\u001b[49m\u001b[43mquotechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   3999\u001b[39m \u001b[43m    \u001b[49m\u001b[43mdate_format\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdate_format\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   4000\u001b[39m \u001b[43m    \u001b[49m\u001b[43mdoublequote\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdoublequote\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   4001\u001b[39m \u001b[43m    \u001b[49m\u001b[43mescapechar\u001b[49m\u001b[43m=\u001b[49m\u001b[43mescapechar\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   4002\u001b[39m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[43m=\u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m   4003\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\io\\formats\\format.py:1014\u001b[39m, in \u001b[36mDataFrameRenderer.to_csv\u001b[39m\u001b[34m(self, path_or_buf, encoding, sep, columns, index_label, mode, compression, quoting, quotechar, lineterminator, chunksize, date_format, doublequote, escapechar, errors, storage_options)\u001b[39m\n\u001b[32m    993\u001b[39m     created_buffer = \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[32m    995\u001b[39m csv_formatter = CSVFormatter(\n\u001b[32m    996\u001b[39m     path_or_buf=path_or_buf,\n\u001b[32m    997\u001b[39m     lineterminator=lineterminator,\n\u001b[32m   (...)\u001b[39m\u001b[32m   1012\u001b[39m     formatter=\u001b[38;5;28mself\u001b[39m.fmt,\n\u001b[32m   1013\u001b[39m )\n\u001b[32m-> \u001b[39m\u001b[32m1014\u001b[39m \u001b[43mcsv_formatter\u001b[49m\u001b[43m.\u001b[49m\u001b[43msave\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m   1016\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m created_buffer:\n\u001b[32m   1017\u001b[39m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(path_or_buf, StringIO)\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\io\\formats\\csvs.py:251\u001b[39m, in \u001b[36mCSVFormatter.save\u001b[39m\u001b[34m(self)\u001b[39m\n\u001b[32m    247\u001b[39m \u001b[38;5;250m\u001b[39m\u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    248\u001b[39m \u001b[33;03mCreate the writer & save.\u001b[39;00m\n\u001b[32m    249\u001b[39m \u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m    250\u001b[39m \u001b[38;5;66;03m# apply compression and byte/text conversion\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m251\u001b[39m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    252\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    253\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    254\u001b[39m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    255\u001b[39m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    256\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mcompression\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    257\u001b[39m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    258\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m handles:\n\u001b[32m    259\u001b[39m     \u001b[38;5;66;03m# Note: self.encoding is irrelevant here\u001b[39;00m\n\u001b[32m    260\u001b[39m     \u001b[38;5;28mself\u001b[39m.writer = csvlib.writer(\n\u001b[32m    261\u001b[39m         handles.handle,\n\u001b[32m    262\u001b[39m         lineterminator=\u001b[38;5;28mself\u001b[39m.lineterminator,\n\u001b[32m   (...)\u001b[39m\u001b[32m    267\u001b[39m         quotechar=\u001b[38;5;28mself\u001b[39m.quotechar,\n\u001b[32m    268\u001b[39m     )\n\u001b[32m    270\u001b[39m     \u001b[38;5;28mself\u001b[39m._save()\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\io\\common.py:749\u001b[39m, in \u001b[36mget_handle\u001b[39m\u001b[34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[39m\n\u001b[32m    747\u001b[39m \u001b[38;5;66;03m# Only for write methods\u001b[39;00m\n\u001b[32m    748\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[33m\"\u001b[39m\u001b[33mr\u001b[39m\u001b[33m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode \u001b[38;5;129;01mand\u001b[39;00m is_path:\n\u001b[32m--> \u001b[39m\u001b[32m749\u001b[39m     \u001b[43mcheck_parent_directory\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mstr\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    751\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m compression:\n\u001b[32m    752\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m compression != \u001b[33m\"\u001b[39m\u001b[33mzstd\u001b[39m\u001b[33m\"\u001b[39m:\n\u001b[32m    753\u001b[39m         \u001b[38;5;66;03m# compression libraries do not like an explicit text-mode\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32mc:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\\pandas\\io\\common.py:616\u001b[39m, in \u001b[36mcheck_parent_directory\u001b[39m\u001b[34m(path)\u001b[39m\n\u001b[32m    614\u001b[39m parent = Path(path).parent\n\u001b[32m    615\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m parent.is_dir():\n\u001b[32m--> \u001b[39m\u001b[32m616\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mOSError\u001b[39;00m(\u001b[33mrf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mCannot save file into a non-existent directory: \u001b[39m\u001b[33m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mparent\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m'\u001b[39m\u001b[33m\"\u001b[39m)\n",
      "\u001b[31mOSError\u001b[39m: Cannot save file into a non-existent directory: 'kaggle\\working'"
     ]
    }
   ],
   "source": [
    "# Load the PDF text content for each article in the training set\n",
    "train_file_paths_df['pdf_text'] = train_file_paths_df['pdf_file_path'].apply(lambda x: load_article_text(x, xml_type=None) if x else \"\")\n",
    "# Load the XML text content for each article in the training set\n",
    "train_file_paths_df['xml_text'] = train_file_paths_df.apply(lambda row: load_article_text(row['xml_file_path'], xml_type=row['xml_type']) if row['xml_file_path'] else \"\", axis=1)\n",
    "# Load the PDF text content for each article in the test set\n",
    "test_file_paths_df['pdf_text'] = test_file_paths_df['pdf_file_path'].apply(lambda x: load_article_text(x, xml_type=None) if x else \"\")\n",
    "# Load the XML text content for each article in the test set\n",
    "test_file_paths_df['xml_text'] = test_file_paths_df.apply(lambda row: load_article_text(row['xml_file_path'], xml_type=row['xml_type']) if row['xml_file_path'] else \"\", axis=1)\n",
    "# Combine the PDF and XML text content into a single 'text' column for training and test sets\n",
    "# train_file_paths_df['text'] = train_file_paths_df['pdf_text'] + \"\\n\" + train_file_paths_df['xml_text']\n",
    "test_file_paths_df['text'] = test_file_paths_df['pdf_text'] + \"\\n\" + test_file_paths_df['xml_text']\n",
    "# Display the first few rows of the training file paths DataFrame\n",
    "print(f\"Training file paths DataFrame shape: {train_file_paths_df.shape}\")\n",
    "display(train_file_paths_df.head(3))\n",
    "# Display the first few rows of the training file paths DataFrame\n",
    "print(f\"Training file paths DataFrame shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.head(3))\n",
    "# Save the training and test file paths DataFrames to CSV files\n",
    "train_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'train_file_paths.csv'), index=False)\n",
    "test_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'test_file_paths.csv'), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "b986f4a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the training and test file paths DataFrames to CSV files\n",
    "train_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'train_file_paths.csv'), index=False)\n",
    "test_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'test_file_paths.csv'), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "15ea73a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "9d63d992-2737-42ea-8fca-408c9f49495b",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018). \\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database. \\nJournal of Geophysical Research: Oceans, 123, 1229\\u20131250.\\nhttps://doi. \\norg/10.1002/2017JC013030 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557\\u2013575.\\nhttps://doi.org/10.1016/j.rse. \\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field an\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388 Barton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181\\u2013197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE. \\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB). \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], context: \\\" 15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism. \\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5] \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], context: \\\" 15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202007717.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: Metal-ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis.\\nBy use of the calix[4]pyrrolato aluminate, [1]\\ufffd, which features a square-planar AlIII, we transfer this concept into the p-block and fully elucidate its mechanisms by experiment and theory.\\nComple-mentary to transition metal-based MLC (aromatization upon substrate binding), substrate binding in [1]\\ufffdoccurs by dearomatization of the ligand.\\nThe aluminate trapps carbonyls by the formation of C\\ufffdC and Al\\ufffdO bonds, but the products maintain full reversibility and outstanding dynamic exchange rates.\\nRemarkably, the C\\ufffdC bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unpre-cedented control over the system\\ufffds constitutional state.\\nMore-over, the metal-ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense.\\nUltimately, this work describes the evolution of an anti-van\\ufffdt Hoff/Le Bel species from their being as a struc-tural curiosity to their application as a reagent and catalyst. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202007717'], context: \\\" 15213773, 2020, 39, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202007717 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Metal\\u2013Ligand Cooperativity of the Calix[4]pyrrolato Aluminate: Triggerable C\\u2212C Bond Formation and Rate Control in Catalysis\",\"abstract\":\"Abstract Metal\\u2010ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis. By use of the calix[4]pyrrolato aluminate, [ 1 ] \\u2212 , which features a square\\u2010planar Al III , we transfer this concept into the  p \\u2010block and fully elucidate its mechanisms by experiment and theory. Complementary to transition metal\\u2010based MLC ( aromatization  upon substrate binding), substrate binding in [ 1 ] \\u2212  occurs by  dearomatization  of the ligand. The aluminate trapps carbonyls by the formation of C\\u2212C and Al\\u2212O bonds, but the products maintain full reversibility and outstanding dynamic exchange rates. Remarkably, the C\\u2212C bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unprecedented control over the system's constitutional state. Moreover, the metal\\u2010ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense. Ultimately, this work describes the evolution of an  anti\\u2010van't Hoff/Le Bel  species from their being as a structural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201902131.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201902131.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pat-tern that was found in a range of compounds featuring geminal diazido units.\\nThrough a simple two-step procedure, the synthesis of structurally diverse target compounds con-taining a tetrazole, such as tetrazoloquinoxalinones, benzoy-laryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"A Synthetic Route Toward Tetrazoles: The Thermolysis of Geminal Diazides\",\"abstract\":\"Abstract A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pattern that was found in a range of compounds featuring geminal diazido units. Through a simple two\\u2010step procedure, the synthesis of structurally diverse target compounds containing a tetrazole, such as tetrazoloquinoxalinones, benzoylaryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and phenanthrols). The initial oxidative diazidation reaction with iodine and sodium azide under mild conditions is followed by the thermal fragmentation under microwave irradiation, leading to the tetrazole products. Noteworthy, an experimental solution is presented in which the potentially hazardous diazide intermediates are not isolated and the concentration of crude reaction mixtures containing diazides is not required to achieve the tetrazoles in good yields.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201903120.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: Systematic variation of the 1,4-dioxane (dx) con-centration during the precipitation of sparingly soluble [MgBr2(dx)2] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R2Mg(dx)n] (n=1, 1.5, 2, and 3), which form during this diox-ane method, depending on the bulkiness of R. The number-ing of the complexes explored in this study is based on the number n of dioxane molecules per magnesium atom, fol-lowed by the substituent R; an apostrophe denotes coordi-nation polymers.\\nThe following derivatives were studied by X-ray crystal-structure determination and NMR spectroscopy: n=1: [Me2Mg(m-dx)]1 (1\\u2019-Me) and [nPr2Mg(m-dx)]1 (1\\u2019-nPr); n=1.5: [{iPr2Mg(dx)}2(m-dx)] (1.5-iPr), [{oTol2Mg(dx)}2(m-dx)] \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Structure\\u2013Solubility Relationship of 1,4\\u2010Dioxane Complexes of Di(hydrocarbyl)magnesium\",\"abstract\":\"Abstract Systematic variation of the 1,4\\u2010dioxane (dx) concentration during the precipitation of sparingly soluble [MgBr 2 (dx) 2 ] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R 2 Mg(dx) n ] ( n =1, 1.5, 2, and 3), which form during this dioxane method, depending on the bulkiness of R. The numbering of the complexes explored in this study is based on the number  n  of dioxane molecules per magnesium atom, followed by the substituent R; an apostrophe denotes coordination polymers. The following derivatives were studied by X\\u2010ray crystal\\u2010structure determination and NMR spectroscopy:  n =1: [Me 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010Me ) and [ n Pr 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010 n Pr );  n =1.5: [{ i Pr 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 i Pr ), [{ o Tol 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 o Tol ), and [(Me 3 Si\\u2010C\\u2261C) 2 Mg(dx) 1.5 ] \\u221e  ( 1.5\\u2032\\u2010C 2 SiMe 3 );  n =2: [ t Bu 2 Mg(dx) 2 ] ( 2\\u2010 t Bu ) and [ o Tol 2 Mg(dx) 2 ] ( 2\\u2010 o Tol );  n =3: [Ph 2 Mg(dx) 3 ] ( 3\\u2010Ph ). In the structure types  1\\u2032 ,  1.5 , and  2 , the magnesium atom exhibits the coordination number 4, whereas pentacoordinate metal atoms are observed in types  3  and  1.5\\u2032 . The structure type  2\\u2032  is realized for [(Ph\\u2010C\\u2261C) 2 Mg(dx) 2 ] \\u221e  ( 2\\u2032\\u2010C 2 Ph ), [MgCl 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Cl ), and [MgBr 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Br ) with hexacoordinate metal atoms. The solubility of the dioxane adducts in common organic solvents strongly depends on the degree of aggregation with the solubility decreasing from molecular to strand to layer structures.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: The reaction of bis(silylenyl)-substituted ferro-cene 1 with two molar equivalents of BPh3 yields the cor-responding bis(silylene\\u2013borane) Lewis adduct 2.\\nThe latter is capable to activate CO2 to furnish the borane-stabilized bis(silanone) 3 through mono-oxygenation of the dative SiII!B silicon centers under release of CO.\\nRemoval of BPh3 from 3 with PMe3 affords the corresponding 1,3,2,4-cyclodisiloxane and the Me3P\\ufffdBPh3 adduct.\\nAll isolated new compounds were characterized and their molecular structures were determined by single-crystal X-ray diffrac-tion analyses. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"An Isolable Bis(Silanone\\u2013Borane) Adduct\",\"abstract\":\"Abstract The reaction of bis(silylenyl)\\u2010substituted ferrocene  1  with two molar equivalents of BPh 3  yields the corresponding bis(silylene\\u2013borane) Lewis adduct  2 . The latter is capable to activate CO 2  to furnish the borane\\u2010stabilized bis(silanone)  3  through mono\\u2010oxygenation of the dative Si II \\u2192B silicon centers under release of CO. Removal of BPh 3  from  3  with PMe 3  affords the corresponding 1,3,2,4\\u2010cyclodisiloxane and the Me 3 P\\u2212BPh 3  adduct. All isolated new compounds were characterized and their molecular structures were determined by single\\u2010crystal X\\u2010ray diffraction analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001412.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: Reactions of a,b-unsaturated aromatic thioke-tones 1 (thiochalcones) with Fe3(CO)12 leading to h4-1-thia-1,3-diene iron tricarbonyl complexes 2, [FeFe] hydroge-nase mimics 3, and the thiopyrane adduct 4 are described. \\nObtained products have been characterized by X-ray crys-tallography and by computational methods.\\nCompletely regio- and diastereoselective formation of the five-mem-bered ring system in products 3, containing four stereo-genic centers, can be explained by an unprecedented, stepwise (3+2)-cycloaddition of two thiochalcone mole-cules mediated by Fe3(CO)12.\\nQuantum chemical calcula-tions aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)-cycloaddition proceeds via sequential intramolecular radical transfer events upon ho-molytic cleavage of one carbon-sulfur bond leading to a diradical intermediate. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Iron(0)\\u2010Mediated Stereoselective (3+2)\\u2010Cycloaddition of Thiochalcones via a Diradical Intermediate\",\"abstract\":\"Abstract Reactions of \\u03b1,\\u03b2\\u2010unsaturated aromatic thioketones  1  (thiochalcones) with Fe 3 (CO) 12  leading to \\u03b7 4 \\u20101\\u2010thia\\u20101,3\\u2010diene iron tricarbonyl complexes\\u2005 2 , [FeFe] hydrogenase mimics  3 , and the thiopyrane adduct  4  are described. Obtained products have been characterized by X\\u2010ray crystallography and by computational methods. Completely regio\\u2010 and diastereoselective formation of the five\\u2010membered ring system in products\\u2005 3 , containing four stereogenic centers, can be explained by an unprecedented, stepwise (3+2)\\u2010cycloaddition of two thiochalcone molecules mediated by Fe 3 (CO) 12 . Quantum chemical calculations aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)\\u2010cycloaddition proceeds via sequential intramolecular radical transfer events upon homolytic cleavage of one carbon\\u2010sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "8",
         "10.1002_chem.202001668",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001668.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001668.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: We have added the {ReIVX5}\\ufffd(X=Br, Cl) synthon to a pocket-based ligand to provide supramolecular design using halogen\\u00b7\\u00b7\\u00b7halogen interactions within an FeIII system that has the potential to undergo spin crossover (SCO).\\nBy removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"The First Use of a ReX 5  Synthon to Modulate Fe III  Spin Crossover via Supramolecular Halogen\\u22c5\\u22c5\\u22c5Halogen Interactions\",\"abstract\":\"Abstract We have added the {Re IV X 5 } \\u2212  (X=Br, Cl) synthon to a pocket\\u2010based ligand to provide supramolecular design using halogen\\u22c5\\u22c5\\u22c5halogen interactions within an Fe III  system that has the potential to undergo spin crossover (SCO). By removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d halogen\\u22c5\\u22c5\\u22c5halogen interactions between neighboring molecules, providing a supramolecular cooperative pathway for SCO. Furthermore, changes to the halogen\\u2010based interaction allow us to modify the temperature and nature of the SCO event.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "9",
         "10.1002_chem.202003167",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202003167.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: Metal-only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M!M bond represent privileged architectures to acquire funda-mental understanding of bimetallic bonding.\\nThis has impor-tant implications in many catalytic processes or supramolec-ular systems that rely on synergistic effects between two metals.\\nHowever, a systematic experimental/computational approach on a well-defined class of compounds is lacking. \\nHere we report a family of MOLPs constructed around the RhI precursor\\n[(h5-C5Me5)Rh(PMe3)2] (1) with a series of s, p and d-block metals, mostly from the main group elements, and investigate their bonding by computational means. \\nAmong the new MOLPs, we have structurally characterized those formed by dative bonding between 1 and MgMeBr, \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Metal\\u2010only Lewis Pairs of Rhodium with  s ,  p  and  d \\u2010Block Metals\",\"abstract\":\"Abstract Metal\\u2010only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M\\u2192M bond represent privileged architectures to acquire fundamental understanding of bimetallic bonding. This has important implications in many catalytic processes or supramolecular systems that rely on synergistic effects between two metals. However, a systematic experimental/computational approach on a well\\u2010defined class of compounds is lacking. Here we report a family of MOLPs constructed around the Rh I  precursor [( \\u03b7 \\n 5 \\u2010C 5 Me 5 )Rh(PMe 3 ) 2 ] ( 1 ) with a series of  s ,  p  and  d \\u2010block metals, mostly from the main group elements, and investigate their bonding by computational means. Among the new MOLPs, we have structurally characterized those formed by dative bonding between  1  and MgMeBr, AlMe 3 , GeCl 2 , SnCl 2 , ZnMe 2  and Zn(C 6 F 5 ) 2,  as well as spectroscopically identified the ones resulting from coordination to MBAr F  (M=Na, Li; BAr F \\n \\u2212 =[B(C 6 H 2 \\u20103,5\\u2010(CF 3 ) 2 ) 4 ] \\u2212 ) and CuCl. Some of these compounds represent unique examples of bimetallic structures, such as the first unambiguous cases of Rh\\u2192Mg dative bonding or base\\u2010free rhodium bound germylene and stannylene species. Multinuclear NMR spectroscopy, including  103 Rh NMR, is used to probe the formation of Rh\\u2192M bonds. A comprehensive theoretical analysis of those provides clear trends. As anticipated, greater bond covalency is found for the more electronegative acids, whereas ionic character dominates for the least electronegative nuclei, though some degree of electron sharing is identified in all cases.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "10",
         "10.1002_cssc.202201821",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_cssc.202201821.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_cssc.202201821.xml",
         "train",
         null,
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium-Ion Batteries** Florian Fiesinger,[a]\\nDaniel Gaissmaier,[a, b, c]\\nMatthias van den Borg,[a] Julian Be\\u00dfner,[a] Adri C. T. van Duin,[d] and Timo Jacob*[a, b, c] One of the key challenges preventing the breakthrough of magnesium-ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode.\\nTo describe the initial steps of Mg anode degradation by O2 impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O2 immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(101\\u03040)A, and Mg(101\\u03041)), heating the surface to several 1000 K. The high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries.\\nAmong the Mg surface terminations, Mg(101\\u03040)A is the most reactive, forming an MgO layer with a thickness of up to 25 \\u00c5. The trained force field can be used to model the ongoing reactions in Mg\\u2013air batteries but also to study the oxidation of magnesium metal in general. \\nIntroduction The search for new energy storage technologies to promote electromobility and reduce global warming is one of the greatest challenges of our time.\\nMagnesium-ion batteries (MIB) are attracting increasing attention as an alternative to conven-tional lithium-ion batteries (LIB) due to the unique properties of magnesium metal anodes, such as high volumetric capacity, low cost, and safe operation.[1\\u20135] However, mass market adoption of MIBs is currently hindered by significant challenges. \\nOne of the biggest challenges is the formation of a passivating interface on the Mg surface which is impermeable to Mg cations and leads to battery failure.[6\\u20138]\\nSun et al. have identified three reasons for Mg degradation in a recent review article.[9] First, Mg has a higher charge density due to its bi-valency, which leads not only to stronger coordination with solvent molecules and salt anions but also facilitates solvent decom-position at the anode.[10] For example, Lu et al. have shown that the salt anion ClO4 is chemically reduced at the Mg anode in carbonate-containing solvents, forming a solid-electrolyte inter-face (SEI) consisting of Mg-halides, MgO, Mg(OH)2, MgCO3, and Mg(ROCO2)2.[11] Second, short circuits are induced by non-uniform Mg growth, which can be further subdivided into hemispherical,[12] porous,[13] or dendritic[14] growth morphologies.[9] Hemispherical 3D particles with different shapes have been detected in TFSI-based electrolytes as a function of the applied current density.[12]\\nIn an Mg(TFSI)2 salt electrolyte with MgCl2 as additive, continuous stripping and plating leads to fracturing and reformation of the SEI \",\"data_availability\":\"Data Availability Statement \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/cssc.202201821'], context: \\\" \\n1864564x, 2023, 3, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/cssc.202201821 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.5281/zenodo.7074790.'], context: \\\"The data that support the findings of this study are openly available in zenodo at https://doi.org/10.5281/zenodo.7074790. \\nFurthermore, the KVIK optimization routine is openly available in the GitHub repository at https://github.com/shk11/KVIKOp-timizer. \\\"}\"]}",
         "{\"title\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium\\u2010Ion Batteries **\",\"abstract\":\"Abstract One of the key challenges preventing the breakthrough of magnesium\\u2010ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode. To describe the initial steps of Mg anode degradation by O 2  impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O 2  immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(10 1 \\u203e \\n0)A, and Mg(10 1 \\u203e \\n1)), heating the surface to several 1000\\u2005K. The high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries. Among the Mg surface terminations, Mg(10 1 \\u203e \\n0)A is the most reactive, forming an MgO layer with a thickness of up to 25\\u2005\\u00c5. The trained force field can be used to model the ongoing reactions in Mg\\u2013air batteries but also to study the oxidation of magnesium metal in general.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "11",
         "10.1002_ece3.3985",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.3985.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2018;8:4525\\u20134533.  \\ufeff\\b   |  4525 www.ecolevol.org 1 | INTRODUCTION Polyandry, where females mate with two or more males, is tax-onomically widespread (Birkhead & M\\u00f8ller, 1998) despite poten-tial costs to females, such as wasted time and energy (Watson,  Arnqvist, & Stallmann, 1998), increased predation risk (Rowe, 1994),  exposure to disease (Thrall, Antonovics, & Dobson, 2000), and risk  of injuries (Crudgington & Siva-\\u00adJothy, 2000).\\nThe maintenance of  polyandry has been attributed to material (direct) and genetic (in-direct) benefits for females.\\nDirect benefits include fertility assur-ance, provision of resources and parental care for the offspring  (Sheldon 1994).\\nIndirect benefits include genetic incompatibility  avoidance, increased genetic diversity of offspring, and the en-hanced survival and reproductive success of offspring (Jennions  & Petrie, 2000; Neff & Pitcher, 2005; Tregenza & Wedell, 2000).  \\nPolyandry may also occur in the absence of benefits to females in    Received: 21 January 2018 | Revised: 7 February 2018 | Accepted: 9 February 2018 DOI: 10.1002/ece3.3985 O R\\nI G I N A L  R E S E A R C H High levels of polyandry, but limited evidence for multiple  paternity, in wild populations of the western rock lobster  (Panulirus cygnus) \\nJacqueline Loo1 | Winn Jason Kennington1 | Simon de Lestang2 | Jason How2 |   Jonathan P. Evans1 This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Centre for Evolutionary Biology, School  of Biological Sciences, The University of  Western Australia, Crawley, WA, Australia 2Western Australian Fisheries and Marine  Research Labo\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1111/j.1469-185X.1970.tb01176.x'], context: \\\"nce of polyandry.\\nBehavioral  Ecology and Sociobiology, 70, 73\\u201381.\\nhttps://doi.org/10.1007/s00265-015-2026-6 Parker, G. A. (1970).\\nSperm competition and its evolutionary conse-quences in the insects.\\nBiological Reviews, 45, 525\\u2013567.\\nhttps://doi. org/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288\\u2013295.  \\nhttps://doi. org/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.)\\n(20\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1471-8286.2005.01155.x'], context: \\\"67.\\nhttps://doi. org/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288\\u2013295.  \\nhttps://doi. org/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.)\\n(2013).\\nLobsters: Biology, management, aquaculture and  fisheries.\\nWest Sussex, UK: Wiley-Blackwell. \\nPhillips, B. F., Cobb, J. S., & George, R. W. (2012).\\nGeneral biology.  \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.3985'], context: \\\" 20457758, 2018, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.3985 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"High levels of polyandry, but limited evidence for multiple paternity, in wild populations of the western rock lobster ( Panulirus cygnus )\",\"abstract\":\"Abstract Polyandry, where multiple mating by females results in the temporal and spatial overlap of ejaculates from two or more males, is taxonomically widespread and occurs in varying frequencies within and among species. In decapods (crabs, lobsters, crayfish, and prawns), rates of polyandry are likely to be variable, but the extent to which patterns of multiple paternity reflect multiple mating, and thus are shaped by postmating processes that bias fertilization toward one or a subset of mated males, is unclear. Here, we use microsatellite markers to examine the frequency of multiple mating (the presence of spermatophores from two or more males) and patterns of paternity in wild populations of western rock lobster ( Panulirus cygnus ). Our data confirm that >45% of females had attached spermatophores arising from at least two males (i.e., confirming polyandry), but we found very limited evidence for multiple paternity; among 24 clutches sampled in this study, only two arose from fertilizations by two or more males. Single inferred paternal genotypes accounted for all remaining progeny genotypes in each clutch, including several instances when the mother had been shown to mate with two or more males. These findings highlight the need for further work to understand whether polyandry is adaptive and to uncover the mechanisms underlying postmating paternity biases in this system.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "12",
         "10.1002_ece3.4466",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.4466.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.r6nq870', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2018;8:9827\\u20139833.  \\ufeff\\b   |  9827 www.ecolevol.org 1 | INTRODUCTION The fundamental biological process of sex determination has been  under intense scientific research for decades, mainly because of its  profound implications on the phenotype of individuals and the for-mation of sex ratios in natural populations (West, Reece, & Sheldon,  2002).\\nIt is therefore a process of major evolutionary significance.  \\nThe mechanisms of sex determination have, for a long time, been  classified discretely as either genetic (GSD) or environmental (ESD).  \\nHowever, researchers are now in agreement that genotypic and en-vironmental sex determination does not necessarily constitute a di-chotomy, but rather two extremes of a continuum (Sarre, Georges, &  Quinn, 2004). \\nSex allocation studies on birds and mammals are numerous  (reviewed in Wapstra et al., 2007) and show that the majority of  these species have genetic sex determination.\\nReptiles (squamates    Received: 11 August 2017 | Revised: 13 June 2018 | Accepted: 6 July 2018 DOI: 10.1002/ece3.4466 O R\\nI G I N A L  R E S E A R C H Incubation temperature and parental identity determine sex in  the Australian agamid lizard Ctenophorus pictus Alexander Hansson  | Mats Olsson This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Department of Biological and Environmental  Sciences, University of Gothenburg,  Gothenburg, Sweden Correspondence Alexander Hansson, Department of  Biological and Environmental Sciences,  University of Gothenburg, Gothenburg,  Sweden. \\nEmail: ha\",\"data_availability\":\"DATA ACCESSIBILITY \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.4466'], context: \\\" 20457758, 2018, 19, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.4466, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Incubation temperature and parental identity determine sex in the Australian agamid lizard  Ctenophorus pictus\",\"abstract\":\"Abstract Sex determination in Australian agamid lizards shows a complex framework of different mechanisms, varying even among closely related taxa. It is clear that discrete classification of these species as either having genetic ( GSD ) or environmental sex determination ( ESD ) does not agree with empirical data. Although many species in this group show temperature\\u2010dependent sex determination ( TSD ), recent evidence suggests additional genetic or epigenetic effects. A proposed model explaining the adaptive significance and evolution of  TSD  in short\\u2010lived agamids predicts that selection will favor temperature\\u2010biased sex ratios in species with intense male\\u2010male competition. Here, we use experimental incubation at (near) constant temperatures to test whether the sex of Australian painted dragons ( Ctenophorus pictus ) is influenced by temperature, building on previous research yet to have reached an agreement regarding the role of temperature in this species. In this study, incubation temperature and parental identity affected hatchling sex suggesting that environment and genetics may work in concert to determine sex in this species. Although our results are consistent with  TSD , our data cannot rule out a temperature\\u2010by\\u2010sex effect on egg or hatchling mortality. However, our findings together with the observed differences of sex determination systems in closely related species within this genus may provide novel opportunities to address fundamental questions in the evolution of sex determination systems.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "13",
         "10.1002_ece3.5260",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5260.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5260.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2019;9:6933\\u20136948.  \\n\\ufeff\\b   |  6933 www.ecolevol.org 1 | INTRODUCTION \\nThe phylogenomics revolution is underway with targeted capture  and enrichment approaches (Mamanova et al., 2010; McCormack,  Hird, Zellmer, Carstens, & Brumfield, 2013) proving an effective  and improved method for phylogenetic studies relative to multilo\\u2010 cus Sanger techniques (Blaimer et al., 2015).\\nAmong the genomic  components that have been targeted, ultraconserved elements  (UCEs sensu Faircloth et al., 2012) have demonstrated their util\\u2010 ity in reconstructing phylogenies across diverse vertebrate\\n[e.g.,  fishes (Faircloth, Sorenson, Santini, & Alfaro, 2013), birds (Hosner,  Tobias, Braun, & Kimball, 2017; McCormack et al., 2013), reptiles  (Crawford et al., 2012; Grismer et al., 2016; Streicher & Wiens,  2017), amphibians (Alexander et al., 2017; Streicher et al., 2018),    Received: 8 April 2019 | Revised: 26 April 2019 | Accepted: 28 April 2019 DOI: 10.1002/ece3.5260   O R\\nI G I N A L  R E S E A R C H Ultraconserved element (UCE) probe set design: Base genome  and initial design parameters critical for optimization Grey T. Gustafson1,2  |   Alana Alexander2,3  |   John S. Sproul4,5  |   James M. Pflug4\\n|   David R. Maddison4  |   Andrew E. Z. Short1,2 This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2019 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Ecology and Evolutionary  Biology, University of Kansas, Lawrence,  Kansas 2Biodiversity Institute, University of Kansas,  Lawrence, Kansas 3Department of Anatomy, School of  Biomedical Sciences, University of Otago,  Dunedin, New Zealand 4Department of Integrative Biology, Oregon  State University, Corvallis, Oregon 5Department of Biology, Unive\",\"data_availability\":\"DATA AVAILABILITY \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.5260'], context: \\\" 20457758, 2019, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5260, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Ultraconserved element (UCE) probe set design: Base genome and initial design parameters critical for optimization\",\"abstract\":\"Abstract Targeted capture and enrichment approaches have proven effective for phylogenetic study. Ultraconserved elements (UCEs) in particular have exhibited great utility for phylogenomic analyses, with the software package  phyluce  being among the most utilized pipelines for UCE phylogenomics, including probe design. Despite the success of UCEs, it is becoming increasing apparent that diverse lineages require probe sets tailored to focal taxa in order to improve locus recovery. However, factors affecting probe design and methods for optimizing probe sets to focal taxa remain underexplored. Here, we use newly available beetle (Coleoptera) genomic resources to investigate factors affecting UCE probe set design using  phyluce . In particular, we explore the effects of stringency during initial design steps, as well as base genome choice on resulting probe sets and locus recovery. We found that both base genome choice and initial bait design stringency parameters greatly alter the number of resultant probes included in final probe sets and strongly affect the number of loci detected and recovered during in silico testing of these probe sets. In addition, we identify attributes of base genomes that correlated with high performance in probe design. Ultimately, we provide a recommended workflow for using  phyluce  to design an optimized UCE probe set that will work across a targeted lineage, and use our findings to develop a new, open\\u2010source UCE probe set for beetles of the suborder Adephaga.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "14",
         "10.1002_ece3.5395",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5395.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5395.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2019;9:8945\\u20138952.  \\n\\ufeff\\b   |  8945 www.ecolevol.org 1 | INTRODUCTION Migration is a ubiquitous phenomenon that has evolved as an ad\\u2010 aptation to seasonally changing environments (Newton, 2008).  \\nBut within the same species, or even population, we often observe  tremendous variation in specific migration features (Chapman,  Br\\u00f6nmark, Nilsson, & Hansson, 2011;\\nFlack et al., 2016).\\nThis may  be the outcome of trade\\u2010offs between the costs and benefits of  different migration strategies, leading to differences in individual  fitness, and subsequently influencing demographic dynamics (Lok,  Overdijk, & Piersma, 2015; Palac\\u00edn, Alonso, Mart\\u00edn, & Alonso,  2017).\\nLife in the Anthropocene is challenging many ancient evo\\u2010 lutionary adaptations including migration.\\nSelection pressures  change quickly, favoring individuals that are better adapted to  human\\u2010modified landscapes (Otto, 2018).\\nThese human\\u2010shaped  selection pressures are caused by hunting, poisoning, electrocu\\u2010 tion, food subsidies or global habitat and climate change, and many  more (Oro, Genovart, Tavecchia, Fowler, & Mart\\u00ednez\\u2010Abra\\u00edn, 2013;    Received: 18 April 2019 | Revised: 28 May 2019 | Accepted: 4 June 2019 DOI: 10.1002/ece3.5395   O R\\nI G I N A L  R E S E A R C H \\u201cCloser\\u2010to\\u2010home\\u201d strategy benefits juvenile survival in a long\\u2010 distance migratory bird Yachang Cheng1,2  |   Wolfgang Fiedler1,2\\n|   Martin Wikelski1,2,3 |   Andrea Flack1,2,3 \\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2019 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Migration, Max Planck  Institute of Animal Behavior, Radolfzell,  Germany 2Department of Biology, University of  Konstanz, Konstanz, Germany 3Centre for the Advanced Study of \",\"data_availability\":\"DATA ACCESSIBILITY \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.5395'], context: \\\" 20457758, 2019, 16, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5395 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], context: \\\"Schmid, H., & Wikelski, M. (2019).\\nData from: LifeTrack White  Stork Rheinland\\u2010Pfalz.\\nMovebank Data Repository, https://doi. org/10.5441/001/1.4192t2j4 Fiedler, W., Leppelsack, E., Leppelsack, H., Stahl, T., Wieding, O., &  \\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], context: \\\"(2019).\\nData from: LifeTrack White Stork Vorarlberg.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.71r7pp6q Flack, A., Fiedler, W., Blas, J., Pokrovsky, I., Kaatz, M., Mitropolsky, M.,  \\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Southwest  Germany.\\nMovebank Data Repository, https://doi.org/10.5441/001/1. \\nck04mn78 Fiedler, W., Flack, A., Schmid, H., Reinhard, U., & Wikelski, M. (2019).  \\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], context: \\\"Data from: LifeTrack White Stork Oberschwaben.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.c42j3js7 Fiedler, W., Hilsendegen, C., Reis, C., Lehmann, J., Hilsendegen, P.,  \\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Bavaria.  \\nMovebank  Data  Repository,  https://doi.org/10.5441/001/1. \\nv1cs4nn0 Fiedler, W., Niederer, W., Sch\\u00f6nenberger, A., Flack, A., & Wikelski, M.  \\\"}\"]}",
         "{\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d strategy benefits juvenile survival in a long\\u2010distance migratory bird\",\"abstract\":\"Abstract Human\\u2010induced changes in the climate and environment that occur at an unprecedented speed are challenging the existence of migratory species. Faced with these new challenges, species with diverse and flexible migratory behaviors may suffer less from population decline, as they may be better at responding to these changes by altering their migratory behavior. At the individual level, variations in migratory behavior may lead to differences in fitness and subsequently influence the population's demographic dynamics. Using lifetime GPS bio\\u2010logging data from 169 white storks ( Ciconia ciconia ), we explore whether the recently shortened migration distance of storks affects their survival during different stages of their juvenile life. We also explore how other variations in migratory decisions (i.e., time, destination), movement activity (measured using overall body dynamic acceleration), and early life conditions influence juvenile survival. We observed that their first autumn migration was the riskiest period for juvenile white storks. Individuals that migrated shorter distances and fledged earlier experienced lower mortality risks. In addition, higher movement activity and overwintering \\u201ccloser\\u2010to\\u2010home\\u201d (with 84.21% of the tracked individuals stayed Europe or North Africa) were associated with higher survival. Our study shows how avian migrants can change life history decisions over only a few decades, and thus it helps us to understand and predict how migrants respond to the rapidly changing world.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.ck04mn78'], context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Sch\\u00e4fle ,  W. \\n ,  \\n Keeves ,  B. \\n ,  \\n Quetting ,  M. \\n ,  \\n Eid ,  B. \\n , \\u2026  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Southwest Germany .  Movebank Data Repository ,  10.5441/001/1.ck04mn78\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Schmid ,  H. \\n ,  \\n Reinhard ,  U. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Oberschwaben .  Movebank Data Repository ,  10.5441/001/1.c42j3js7\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], context: \\\"Fiedler ,  W. \\n ,  \\n Hilsendegen ,  C. \\n ,  \\n Reis ,  C. \\n ,  \\n Lehmann ,  J. \\n ,  \\n Hilsendegen ,  P. \\n ,  \\n Schmid ,  H. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Rheinland\\u2010Pfalz .  Movebank Data Repository ,  10.5441/001/1.4192t2j4\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.v1cs4nn0'], context: \\\"Fiedler ,  W. \\n ,  \\n Leppelsack ,  E. \\n ,  \\n Leppelsack ,  H. \\n ,  \\n Stahl ,  T. \\n ,  \\n Wieding ,  O. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Bavaria .  Movebank Data Repository ,  10.5441/001/1.v1cs4nn0\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], context: \\\"Fiedler ,  W. \\n ,  \\n Niederer ,  W. \\n ,  \\n Sch\\u00f6nenberger ,  A. \\n ,  \\n Flack ,  A. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Vorarlberg .  Movebank Data Repository ,  10.5441/001/1.71r7pp6q\\\"}\"]}"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6144.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2020;10:3463\\u20133476.  \\ufeff\\b   |  3463 www.ecolevol.org 1 | INTRODUCTION Marine ecosystems can change rapidly in response to both natural  and anthropogenic changes in the environment.\\nMonitoring changes  in the abundance and distribution of marine organisms is a chal-lenging but critical component of conservation efforts and natural  resource management.\\nHuman impacts such as overfishing, coastal  development, and pollution can rapidly increase stress on ecosys-tems resulting in an urgent need to develop efficient methods for  monitoring spatial and temporal dynamics of biodiversity (Beng  et al., 2016). \\nFisheries management has long used ichthyoplankton sur-veys as a component of monitoring because sampling the early  life stages of fish reflects the reproductive activities of fish    Received: 17 July 2019 | Revised:\\n31 January 2020 |\\nAccepted: 4 February 2020 DOI: 10.1002/ece3.6144   O R\\nI G I N A L  R E S E A R C H Efficacy of metabarcoding for identification of fish eggs  evaluated with mock communities Elena M. Duke  |   Ronald S. Burton \\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Marine Biology Research Division, Scripps  Institution of Oceanography, University of  California, San Diego, La Jolla, California Correspondence Elena M. Duke, Marine Biology  Research Division, Scripps Institution of  Oceanography, University of California, San  Diego, La Jolla, CA. \\nEmail: emduke@ucsd.edu \\nAbstract There is urgent need for effective and efficient monitoring of marine fish popula-tions.\\nM\",\"data_availability\":\"DATA AVAILABILITY STATEMENT \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1371/journal.pone.0134647'], context: \\\"ermsmeier, M. C., Rogowski, P. A., Terrill,  E., Burton, R. S., & Bernardi, G. (2015).\\nMonitoring spawning ac-tivity in a Southern California marine protected area using molec-ular identification of fish eggs.\\nPLoS ONE, 10(8), 1\\u201321.\\nhttps://doi. org/10.1371/journal.pone.0134647 Hastings, P. A., & Burton, R. S. (2008).\\nEstablishing a DNA Sequence  Database for the Marine Fish Fauna of California.\\nRetrieved from  https://escholarship.org/uc/item/1ck9b3qs Hatzenbuhler, C., Kelly, J. R., Martinson, J., Okum, S., & Pilgrim, E. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6144'], context: \\\" 20457758, 2020, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6144, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "16",
         "10.1002_ece3.6303",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6303.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.37pvmcvgb', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2020;10:5617\\u20135630.  \\ufeff\\b   |  5617 www.ecolevol.org   Received: 26 August 2019 | Revised: 25 March 2020 | Accepted: 30 March 2020 DOI: 10.1002/ece3.6303   O R\\nI G I N A L  R E S E A R C H Can polyploidy confer invasive plants with a wider climatic  tolerance?\\nA test using Solidago canadensis Jizhong Wan1,2 |   Ayub M. O. Oduor1,3  |   \\nRobin Pouteau1  |   Beilei Wang1 |    Luxi Chen1 |   \\nBeifen Yang1 |   Feihai Yu1 |   \\nJunmin Li1 \\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Zhejing Provincial Key Laboratory of Plant  Evolutionary Ecology and Conservation,  Taizhou University, Taizhou, P.R. China 2State Key Laboratory of Plateau Ecology  and Agriculture, Qinghai University, Xining,  \\nP.R. China 3Department of Applied Biology, Technical  University of Kenya, Nairobi, Kenya Correspondence Junmin Li, Zhejiang Provincial Key  Laboratory of Plant Evolutionary Ecology  and Conservation, Taizhou 318000, China. \\nEmail: lijmtzc@126.com Funding information Ten Thousand Talent Program of Zhejiang  Province; National Natural Science  Foundation of China, Grant/Award Number:  31270461; National Key Research and  Development Program, Grant/Award  Number: 2016YFC1201100 Abstract Polyploidy can cause variation in plant functional traits and thereby generate indi-viduals that can adapt to fluctuating environments and exploit new environments.  \\nHowever, few empirical studies have tested for an association between ploidy level  and climatic tolerance of invasive cytotypes relative to conspecific native-range cy-totypes.\\nHere, we used an invasive plant Solidago canadensis to test whether invas\",\"data_availability\":\"DATA AVAILABILITY STATEMENT \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.6303'], context: \\\" 20457758, 2020, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6303, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['MN107151', 'MN335287', 'MN335290'], context: \\\"MN107084\\u2013MN107151, MN335287\\u2013MN335290).\\nThe data that  \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/cyto.a.20253'], context: \\\"Suda, J., & Tr\\u00e1vn\\u00ed\\u010dek, P. (2006).\\nReliable DNA ploidy determination in  dehydrated tissues of vascular plants by DAPI flow cytometry\\u2014new  prospects for plant research.\\nCytometry, 69A, 273\\u2013280.\\nhttps://doi. org/10.1002/cyto.a.20253 Tajima, F. (1989).\\nStatistical method for testing the neutral mutation hy-pothesis by DNA polymorphism.\\nGenetics, 123, 585\\u2013595. \\nTavares, D., Loureiro, J., Martins, A., Castro, M., Roiloa, S., & Castro, S.  (2019).\\nGenetically based phenotypic diffe\\\"}\",\"{\\\"dataset_ids\\\": ['MN107084', 'MN107151'], context: \\\"(NCBI) database (accession numbers: MN107084\\u2013MN107151,  \\\"}\"]}",
         "{\"title\":\"Can polyploidy confer invasive plants with a wider climatic tolerance? A test using  Solidago canadensis\",\"abstract\":\"Abstract Polyploidy can cause variation in plant functional traits and thereby generate individuals that can adapt to fluctuating environments and exploit new environments. However, few empirical studies have tested for an association between ploidy level and climatic tolerance of invasive cytotypes relative to conspecific native\\u2010range cytotypes. Here, we used an invasive plant  Solidago canadensis  to test whether invasive populations had a higher proportion of polyploids, greater height and stem\\u2010base diameter, and occupied a wider range of climatic conditions than conspecific native\\u2010range populations. We also tested whether the invasive populations had overcome genetic founder effects. We sampled a total of 80 populations in parts of the invaded range in China and native range in North America for in situ measurements of plant height and stem\\u2010base diameter in the field and for population genetic and cytotype analyses. To examine climatic correlates, we augmented our field\\u2010sampled data with occurrence records obtained from Global Biodiversity Information Facility. All, except one, of the populations that we sampled in China occurred in a humid subtropical climate. In contrast, the North American populations occurred in humid continental, humid subtropical, and semi\\u2010arid climatic zones. All populations of  S. canadensis  in China were purely hexaploid, while the North American populations were diploid, tetraploid, and hexaploid. The invasive hexaploids were significantly taller and had a\\u00a0larger stem\\u2010base diameter than native hexaploids. Native hexaploids were significantly taller and had larger stem\\u2010base diameter than native diploids. Climatic correlate assessment found that invasive and native populations occupied different climatic envelopes, with invasive populations occurring in warmer and less seasonal climates than native populations. However, there was no significant correlation between ploidy level and climatic envelope of  S. canadensis . Molecular phylogeo\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "17",
         "10.1002_ece3.6784",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6784.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2020;10:11467\\u201311487.  \\ufeff\\b   |  11467 www.ecolevol.org   Received: 25 June 2020 | Revised: 25 August 2020 | Accepted: 27 August 2020 DOI: 10.1002/ece3.6784   O R\\nI G I N A L  R E S E A R C H Ecomorphology of the pectoral girdle in anurans (Amphibia,  Anura): Shape diversity and biomechanical considerations Karolin Engelkes1  \\n|   Lena Kath1 |   Thomas Kleinteich2\\n|   J\\u00f6rg U. Hammel3,4  |    Andr\\u00e9 Beerlink5 |   \\nAlexander Haas1 \\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd 1Center of Natural History (CeNak),  Universit\\u00e4t Hamburg, Hamburg, Germany 2TPW Pr\\u00fcfzentrum GmbH, Neuss, Germany 3Institute of Materials Research, Helmholtz-Zentrum Geesthacht, Geesthacht, Germany 4Institut f\\u00fcr Zoologie und  Evolutionsforschung mit Phyletischem  Museum, Ernst-Hackel-Haus und  Biologiedidaktik, Friedrich-Schiller-Universit\\u00e4t Jena, Jena, Germany \\n5YXLON International GmbH, Hamburg,  Germany Correspondence Karolin Engelkes, Center of Natural History  (CeNak), Universit\\u00e4t Hamburg, Martin-Luther-King-Platz 3, 20146 Hamburg,  Germany. \\nEmail: karolin.engelkes@uni-hamburg.de Funding information Wilhelm-Peters-Fonds of the Deutsche  Gesellschaft f\\u00fcr Herpetologie und  Terrarienkunde e.V (DGHT); Deutsche  Forschungsgemeinschaft, Grant/Award  Number: 387723284 Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that  allow them to inhabit a wide range of habitats.\\nThe different locomotor modes are  likely to be linked to anatomical specializations of the skeleton within the typical frog  Bauplan.\\nWhile such anatomical adaptations of the hind lim\",\"data_availability\":\"DATA AVAILABILITY STATEMENT \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1073/pnas.0800375105'], context: \\\"ak, M., Kadmon, R., Saltz, D.,  & Smouse, P. E. (2008).\\nA movement ecology paradigm for unifying  organismal movement research.\\nProceedings of the National Academy  of Sciences of the United States of America, 105(49), 19052\\u201319059.  \\nhttps://doi.org/10.1073/pnas.0800375105 Nauwelaerts, S., & Aerts, P. (2006).\\nTake-off and landing forces in jump-ing frogs.\\nThe Journal of Experimental Biology, 209(1), 66\\u201377.\\nhttps://doi.org/10.1242/jeb.01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of a\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1469-7580.2007.00691.x'], context: \\\".01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of aquatic and terrestrial locomotion in a semi-aquatic frog, Rana es-culenta: No evidence for a design conflict.\\nJournal of Anatomy, 210(3),  304\\u2013317.\\nhttps://doi.org/10.1111/j.1469-7580.2007.00691.x Noble, G. K., & Jaeckle, M. E. (1928).\\nThe digital pads of the tree frogs.  \\nA study of the phylogenesis of an adaptive structure.\\nJournal of  Morphology, 45(1), 259\\u2013292. \\nNomura, F., Rossa-Feres, D. C., & Langeani, F. (2009).\\nBurrowing behav-ior of\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6784'], context: \\\" 20457758, 2020, 20, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6784 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.13140/RG.2.2.18389.86247'], context: \\\"ge in the evolution of a novel locomotor  type: \\u201cflying\\u201d frogs.\\nEvolution, 44(8), 1931\\u20131946. Engelkes, K. (preprint, in review) Accuracy of bone segmentation and  surface generation strategies analyzed by using synthetic CT vol-umes.\\nhttps://doi.org/10.13140/RG.2.2.18389.86247 Engelkes, K., Friedrich, F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 2\\\"},{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], context: \\\" F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 213\\u2013228.\\nhttps://doi. org/10.1007/s00435-017-0386-3 Engelkes, K., Helfsgott, J., Hammel, J. U., B\\u00fcsse, S., Kleinteich, T., Beerlink,  A., \\u2026 Haas, A. (2019).\\nMeasurement error in \\u03bcCT-based three-dimen-sional geometric morphometrics introduced by surface generation  and landmark data acquisition.\\nJourn\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1550-7408.2008.00342.x'], context: \\\" lineage  of closely related trypanosomes (Trypanosomatidae, Kinetoplastida)  of anurans and sand flies (Psychodidae, Diptera) sharing the same  ecotopes in brazilian amazonia.\\nThe Journal of Eukaryotic Microbiology,  55(5), 427\\u2013435.\\nhttps://doi.org/10.1111/j.1550-7408.2008.00342.x Frost, D. R. (2020) Amphibian Species of the World: an online reference.  \\nVersion 6.1.\\nhttp://research.amnh.org/herpetology/amphibia/index. html. \\nF\\u00fcrbringer, M. (1873).\\nZur vergleichenden Anatomie der Schultermuskeln.  \\nTeil 1.\\nJenaische Zeitschri\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/jeb.12292'], context: \\\" M., Byrne, P. G., Roberts, J. D., & Keogh, J. S. (2014).\\nThe role  of phylogeny and ecology in shaping morphology in 21 genera and 127  species of Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181\\u2013192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825 Wainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright,\\\"},{\\\"dataset_ids\\\": ['10.2307/1444825'], context: \\\"Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181\\u2013192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825 Wainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright, & S. M. Reilly (Eds.), Ecological morphol-ogy: Integrative organismal biology (pp. 42\\u201359).\\nChicago, LondonThe  University of Chicago Press. \\nWain\\\"}\"]}",
         "{\"title\":\"Ecomorphology of the pectoral girdle in anurans (Amphibia, Anura): Shape diversity and biomechanical considerations\",\"abstract\":\"Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that allow them to inhabit a wide range of habitats. The different locomotor modes are likely to be linked to anatomical specializations of the skeleton within the typical frog Bauplan. While such anatomical adaptations of the hind limbs and the pelvic girdle are comparably well understood, the pectoral girdle received much less attention in the past. We tested for locomotor\\u2010mode\\u2010related shape differences in the pectoral girdle bones of 64 anuran species by means of micro\\u2010computed\\u2010tomography\\u2010based geometric morphometrics. The pectoral girdles of selected species were analyzed with regard to the effects of shape differences on muscle moment arms across the shoulder joint and stress dissipation within the coracoid. Phylogenetic relationships, size, and locomotor behavior have an effect on the shape of the pectoral girdle in anurans, but there are differences in the relative impact of these factors between the bones of this skeletal unit. Remarkable shape diversity has been observed within locomotor groups indicating many\\u2010to\\u2010one mapping of form onto function. Significant shape differences have mainly been related to the overall pectoral girdle geometry and the shape of the coracoid. Most prominent shape differences have been found between burrowing and nonburrowing species with headfirst and backward burrowing species significantly differing from one another and from the other locomotor groups. The pectoral girdle shapes of burrowing species have generally larger moment arms for (simulated) humerus retractor muscles across the shoulder joint, which might be an adaptation to the burrowing behavior. The mechanisms of how the moment arms were enlarged differed between species and were associated with differences in the reaction of the coracoid to simulated loading by physiologically relevant forces.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], context: \\\"Engelkes ,  K. \\n ,  \\n Friedrich ,  F. \\n ,  \\n Hammel ,  J. U. \\n , &  \\n Haas ,  A. \\n  ( 2018 ).  A simple setup for episcopic microtomy and a digital image processing workflow to acquire high\\u2010quality volume data and 3D surface models of small vertebrates .  Zoomorphology ,  137 ( 1 ),  213 \\u2013 228 .  10.1007/s00435-017-0386-3\\\"}\"]}"
        ],
        [
         "18",
         "10.1002_ece3.961",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.961.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"The effects of insects, nutrients, and plant invasion on community structure and function above- and belowground Phoebe Wright1, Melissa A. Cregger1,2, Lara Souza1,3, Nathan J. Sanders1,4 & Aim\\ufffdee T. Classen1,4 1Department of Ecology and Evolutionary Biology, University of Tennessee, 569 Dabney Hall, Knoxville, Tennessee 37996 2Institute for Genomic Biology, University of Illinois, 1206 W Gregory Rm 3405, Urbana, Illinois 61801 3Oklahoma Biological Survey and Department Microbiology and Plant Biology, University of Oklahoma, 111 E Chesapeake St., Norman, Oklahoma 73019 4Center for Macroecology Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, Copenhagen DK-2100, Denmark Keywords aboveground biomass, bacteria, fungi, insect, invasive plant, nitrogen-fixer, nutrient amendment, old-field ecosystem, soil enzyme activity. \\nCorrespondence Aim\\ufffdee T. Classen, Center for Macroecology, Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, DK-2100 Copenhagen, Denmark.\\nTel: +45 35-32-22-22; Fax: +45 46-82-74-83; E-mail: atclassen@gmail.com Funding Information The Department of Ecology and Evolutionary Biology and the Office of Research at The University of Tennessee funded this work. \\nReceived:\\n21 November 2013; Revised: 2 December 2013; Accepted: 3 December 2013 Ecology and Evolution 2014; 4(6): 732\\u2013 742 doi: 10.1002/ece3.961 Abstract Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact.\\nIn this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient avail-ability (nitrogen added, nitrogen reduced and control) in a fully crossed, com-pletely randomized plot design.\\nWe found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above- and belowground structure and function at our site.\\nNot surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization.\\nThe introduction of invasi\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.961'], context: \\\" 20457758, 2014, 6, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.961 by California Digital Library University Of California, Wiley Online Library on [08/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"The effects of insects, nutrients, and plant invasion on community structure and function above-and belowground\",\"abstract\":\"Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact. In this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient availability (nitrogen added, nitrogen reduced and control) in a fully crossed, completely randomized plot design. We found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above-and belowground structure and function at our site. Not surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization. The introduction of invasive nitrogen-fixing plant propagules interacted with nutrient amendment and insect presence to alter soil bacterial abundance and the activity of the microbial community. While the larger-scale, longer-term bulk measurements such as biomass production and nutrient mineralization responded to the direct effects of our treatments, the shorter-term and dynamic microbial communities tended to respond to interactions among our treatments. Our results indicate that soil nutrients, invasive plants, and insect herbivores determine both above-and belowground responses, but whether such effects are independent versus interdependent varies with scale.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "19",
         "10.1002_ece3.9627",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.9627.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Ecology and Evolution.\\n2022;12:e9627.  \\n\\ufeff    | 1 of 15 https://doi.org/10.1002/ece3.9627 www.ecolevol.org Received: 5 June 2022 | Revised: 14 October 2022 | Accepted: 20 November 2022 DOI: 10.1002/ece3.9627   R E S E A R C H  A R T I C L E Inferring predator\\u2013\\u00adprey interactions from camera traps:  A Bayesian co-\\u00adabundance modeling approach Zachary Amir1,2  |   Adia Sovie3  |   Matthew Scott Luskin1,2 This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited. \\n\\u00a9 2022 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1School of Biological Sciences, University  of Queensland, St. Lucia, Queensland,  Australia 2Centre for Biodiversity and Conservation  Science, University of Queensland, St.  Lucia, Queensland, Australia 3Department of Fisheries and Wildlife,  Michigan State University, East Lansing,  Michigan, USA Correspondence Zachary Amir, School of Biological  Sciences, University of Queensland,  534 Goddard Hall, St. Lucia 4067, QLD,  Australia. \\nEmail: z.amir@uq.edu.au Abstract Predator\\u2013\\u00adprey dynamics are a fundamental part of ecology, but directly studying inter-actions has proven difficult.\\nThe proliferation of camera trapping has enabled the col-lection of large datasets on wildlife, but researchers face hurdles inferring interactions  from observational data.\\nRecent advances in hierarchical co-\\u00adabundance models infer  species interactions while accounting for two species' detection probabilities, shared  responses to environmental covariates, and propagate uncertainty throughout the  entire modeling process.\\nHowever, current approaches remain unsuitable for interact-ing specie\",\"data_availability\":\"DATA AVAILABILITY STATEMENT \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1086/685095'], context: \\\"interactions on populations of endangered species.\\nThe American  Naturalist, 187, 457\\u2013\\u00ad467. https://doi.org/10.1086/685095 Rovero, F., & Zimmermann, F. (2016).\\nCamera trapping for wildlife research.  \\nPelagic Publishing. \\nRoyle, J. A. (2004).\\nN-\\u00admixture models for estimating population size from  \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.9627'], context: \\\" 20457758, 2022, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.9627 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"Inferring predator\\u2013prey interactions from camera traps: A\\u00a0Bayesian co\\u2010abundance modeling approach\",\"abstract\":\"Abstract Predator\\u2013prey dynamics are a fundamental part of ecology, but directly studying interactions has proven difficult. The proliferation of camera trapping has enabled the collection of large datasets on wildlife, but researchers face hurdles inferring interactions from observational data. Recent advances in hierarchical co\\u2010abundance models infer species interactions while accounting for two species' detection probabilities, shared responses to environmental covariates, and propagate uncertainty throughout the entire modeling process. However, current approaches remain unsuitable for interacting species whose natural densities differ by an order of magnitude and have contrasting detection probabilities, such as predator\\u2013prey interactions, which introduce zero inflation and overdispersion in count histories. Here, we developed a Bayesian hierarchical N\\u2010mixture co\\u2010abundance model that is suitable for inferring predator\\u2013prey interactions. We accounted for excessive zeros in count histories using an informed zero\\u2010inflated Poisson distribution in the abundance formula and accounted for overdispersion in count histories by including a random effect per sampling unit and sampling occasion in the detection probability formula. We demonstrate that models with these modifications outperform alternative approaches, improve model goodness\\u2010of\\u2010fit, and overcome parameter convergence failures. We highlight its utility using 20 camera trapping datasets from 10 tropical forest landscapes in Southeast Asia and estimate four predator\\u2013prey relationships between tigers, clouded leopards, and muntjac and sambar deer. Tigers had a negative effect on muntjac abundance, providing support for top\\u2010down regulation, while clouded leopards had a positive effect on muntjac and sambar deer, likely driven by shared responses to unmodelled covariates like hunting. This Bayesian co\\u2010abundance modeling approach to quantify predator\\u2013prey relationships is widely applicable across species, ecosystems\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "20",
         "10.1002_ecs2.1280",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.p3fg9', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract.   \\nUnderstanding the impact of herbivory on plant populations is a fundamental goal of ecology.  \\nDamage to individual plants can be visually striking and affect the fates of individuals, but these impacts  do not necessarily translate into population-\\u00adlevel differences in vital rates (survival, growth, or fecundity)  or population growth rates.\\nIn biological control of weeds, quantitative assessments of population-\\u00adlevel  impacts of released agents on both target invasive plants and native, nontarget plants are needed to inform  evaluations of the benefits and risks of releasing agents into new regions.\\nHere we present a 3-\\u00adyr exper-imental demographic field study using the European root-\\u00adfeeding biocontrol weevil, Mogulones crucifer,  first released in Canada in 1997 to control the invasive weed Cynoglossum officinale (Boraginaceae).\\nMogu-lones crucifer is an effective \\u201csearch and destroy\\u201d agent in Canada, but sporadically feeds, oviposits, and  develops on native nontarget Boraginaceae.\\nWe investigated the population-\\u00adlevel impacts of this biocon-trol insect on its target weed and a native nontarget plant, Hackelia micrantha (Boraginaceae), by releasing  large numbers of weevils into naturally occurring patches of H. micrantha growing isolated from or inter-spersed with C. officinale.\\nWe followed the fates of individual plants on release and nonrelease (control)  sites for two transition years, developed matrix models to project population growth rates (\\u03bb) for each  plant species, and examined the contributions from differences in vital rates to changes in \\u03bb using life table  response experiments (LTRE).\\nIn contrast to studies of the insect\\u2013plant interaction in its native range, as a  biocontrol agent, M. crucifer increased mortality of C. officinale rosettes in the year immediately following  release, depressing the weed's \\u03bb to below the population replacement level.\\nHowever, \\u03bb for H. micrantha  was never depressed below the replacement level, and any differences between release and nonrelease  sites in the nontarget could not be explained by significant contributions from vital rates in the LTRE.\\nThis  study is the first to simultaneously and experimentally examine target and nontarget population-\\u00adlevel im-pacts of a weed biocontrol insect in the field, and supports the theoretical prediction that plant life history  characteristics and uneven herbivore host preferences can interact to produce differences in population-\\u00ad level impacts between target and nontarget plant species. \",\"data_availability\":\"Data Availability \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.1280'], context: \\\" 21508925, 2016, 5, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.1280 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         ""
        ],
        [
         "21",
         "10.1002_ecs2.4619",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.4619.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ecs2.4619.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.25349/d9qw5x', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"A R T I C L E M a c r o s y s t e m s E c o l o g y Interactions between temperature and predation impact insect emergence in alpine lakes Caroline H. Owens1 | Michelle J. Lee1 \\n| Melissa Grim1,2 | John Schroeder1 | Hillary S. Young1 1Department of Ecology Evolution and Marine Biology, University of California, Santa Barbara, Santa Barbara, California, USA 2Department of Environmental Science and Policy, University of California, Davis, Davis, California, USA Correspondence Caroline H. Owens Email: owens@ucsb.edu Funding information National Science Foundation; University of California Natural Reserve System; University of California, Santa Barbara Handling Editor:\\nManuel T. Lerdau Abstract While insect populations are simultaneously threatened by many local- and global-scale stressors, the interacting effects of these factors remain poorly understood.\\nThese interactions between stressors, whether additive or antagonis-tic, may have profound effects on our assumptions about and predictions for any given system.\\nHere we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes.\\nUsing a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence.\\nBased on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6\\ufffdC of water temperature in oligotrophic alpine lakes.\\nWhile our work confirms pri\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data and novel code (Owens et al., 2023) are available from Dryad: https://doi.org/10.25349/D9QW5X. \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.4619'], context: \\\" 21508925, 2023, 7, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.4619, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/2015GL066235.'], context: \\\" \\nO\\u2019Reilly, C. M., S. Sharma, D. K. Gray, S. E. Hampton, J. S. Read, R. J. Rowley, P. Schneider, et al. 2015.\\n\\u201cRapid and Highly Variable Warming of Lake Surface Waters around the Globe.\\u201d Geophysical Research Letters 42(24): 10773\\u201381.\\nhttps://doi.org/10.1002/2015GL066235. \\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023. \\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset. \\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. K\\\"},{\\\"dataset_ids\\\": ['10.25349/D9QW5X.'], context: \\\"24): 10773\\u201381.\\nhttps://doi.org/10.1002/2015GL066235. \\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023. \\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset. \\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. Knapp, J. Sickman, K. L. Pope, and S. Chandra.\\n2016.\\n\\u201cVariation in Reciprocal Subsidies between Lakes and Land: Perspectives from the Mountains of \\\"}\"]}",
         "{\"title\":\"Interactions between temperature and predation impact insect emergence in alpine lakes\",\"abstract\":\"While insect populations are simultaneously threatened by many local-and global-scale stressors, the interacting effects of these factors remain poorly understood. These interactions between stressors, whether additive or antagonistic, may have profound effects on our assumptions about and predictions for any given system. Here we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes. Using a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence. Based on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6 C of water temperature in oligotrophic alpine lakes. While our work confirms prior findings that predator presence drives strong reductions in insect emergence, we find that the effects of predation are significantly weaker in warmer lakes (2% reduction in warmest lakes studied vs. 75% reduction in coldest). Combining our results with regional studies of predator communities to identify mechanisms suggests that the interaction effects between temperature and predator presence are likely due to changes in predator behavior across temperatures. Critically, to fully understand the multiple stressors that impact insect populations and predict their future effects, we must consider the interactions between stressors across spatial scales.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.25349/D9QW5X'], context: \\\"Data for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes|Dryad. Dataset|10.25349/D9QW5X\\\"}\"]}"
        ],
        [
         "22",
         "10.1002_ejic.201900904",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejic.201900904.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejic.201900904.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley",
         "{\"title\":\"\",\"abstract\":\"Abstract: Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2-(dimethylaminometh-yl)phenyl-C1,N]gold(III),\\n[Au(damp-C1,N)Cl2], and halogenated, potentially tridentate thiosemicarbazones have been studied. \\nThe results of this work show that the complexation of the halo-genated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti-parasitic activity.\\nFluorination in the periphery of the ligands has expect-\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"abstract\":\"Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2\\u2010(dimethylaminomethyl)phenyl\\u2010C 1 ,N]gold(III), [Au(damp\\u2010C 1 ,N)Cl 2 ], and halogenated, potentially tridentate thiosemicarbazones have been studied. The results of this work show that the complexation of the halogenated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti\\u2010parasitic activity. Fluorination in the periphery of the ligands has expectedly no influence on the structural chemistry of the obtained Au(III) complexes, but modulates their biological behaviour. Best results with a remarkably high selectivity index for the trypomastigote form of  Trypanosoma cruzi  were obtained with the complex containing the ligand, which presents a 3,5\\u2010fluorine substitution in meta\\u2010position of an aromatic ring, [Au(dampH)(L\\u20103,5\\u2010F)]Cl.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31\\u201398 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], context: \\\" 10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         ""
        ],
        [
         "24",
         "10.1002_ejoc.202000916",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejoc.202000916.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract: We herein report an ammonium salt-catalyzed proto-col for the regioselective ring opening of aryl-aziridines with \\ufffd-keto esters.\\nThe reaction gives access to a variety of highly \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2010Opening of Aryl\\u2010Aziridines with \\u03b2\\u2010Keto Esters\",\"abstract\":\"We herein report an ammonium salt\\u2010catalyzed protocol for the regioselective ring opening of aryl\\u2010aziridines with \\u03b2\\u2010keto esters. The reaction gives access to a variety of highly functionalized target molecules with two consecutive stereogenic centers and can be rendered enantioselective (up to  e.r . = 91:9) by using bifunctional chiral ammonium salt catalysts.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "25",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"R E S E A R C H A R T I C L E Past, present and future of a meandering river in the Bolivian Amazon basin Kattia Rubi Arnez Ferrel1 | Jonathan Mark Nelson2 | Yasuyuki Shimizu1 | Tomoko Kyuka1 1Graduate School of Engineering, Hokkaido University, Sapporo, Japan 2U.S. Geological Survey, Golden, Colorado, USA Correspondence Kattia Rubi Arnez Ferrel, Graduate School of Engineering, Hokkaido University, Sapporo 060-0808, Japan. \\nEmail: rubikraf@gmail.com Funding information Nitobe School Project Abstract Field observations on small rivers of the Amazon basin are less common due to their remote location and difficult accessibility.\\nHere we show, through remote sensing analysis and field works, the planform evolution and riverbed topography of a small river located in the upper foreland Amazon basin, the Ichilo River.\\nBy tracking plan-form changes over 30 years, we identified the factors that control meander migration rates in the Ichilo River: cutoffs, climate and human interventions.\\nThe data suggest that neck cutoffs are the main controls in the Ichilo River, with an annual density of 0.022 cutoffs/km.\\nIn addition, climate controls have been identified in the form of high-precipitation events that may have promoted cutoffs, an increase in meander migration rate and channel widening.\\nThe width distribution of the Ichilo River is well represented by general extreme value and inverse Gaussian distributions.\\nThe spatio-temporal variability of meandering migration rates in the Ichilo River is analysed in two locations where neck cutoffs are expected.\\nAnalysing the distance across the neck in these two points, we predict the occurrence of a new cutoff.\\nThe combined methodology of bathymetric surveys and structure from motion photogrammetry shows us the Ichilo riverbed topography and banks at high resolution, where two scour holes were identified.\\nFinally, we discuss the impact of planform changes of the Ichilo River on communities that are established along its riverbanks. \\nK E Y W O R D S bathymetric surveys, Bolivian Amazon basin, Ichilo River, meanders, remote sensing, UAV 1 | INTRODUCTION The Amazon basin exhibits a constantly changing landscape m\",\"data_availability\":\"DATA AVAILABILITY STATEMENT \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/esp.5058'], context: \\\" 10969837, 2021, 4, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5058, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.4637'], context: \\\", M.R., Chandler, J.H., Eltner, A., Fraser, C., Miller, P.E., Mills, J.P. et al. (2019) Guidelines on the use of structure-from-motion photo-grammetry in geomorphic research.\\nEarth Surface Processes and Land-forms, 44(10), 2081\\u20132084.\\nhttps://doi.org/10.1002/esp.4637 \\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23\\u201324, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Semi\\\"},{\\\"dataset_ids\\\": ['10.5066/P9FW6E8K'], context: \\\"4.\\nhttps://doi.org/10.1002/esp.4637 \\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23\\u201324, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability. Journal of Geophysical Research - Earth Surface, 111(F4), 1\\u201314. \\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-\\\"},{\\\"dataset_ids\\\": ['10.1029/2005JF000416'], context: \\\" Bolivia, May 23\\u201324, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability. Journal of Geophysical Research - Earth Surface, 111(F4), 1\\u201314. \\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-driven abandonment and sedimentation of meander bends along a fine-grained, non-vegetated, ephemeral river on the Bolivian Alti-plano. \\nGeomorphology, 350, 106917. \\nhttps://d\\\"}\"]}",
         ""
        ],
        [
         "26",
         "10.1002_esp.5090",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_esp.5090.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9353101', 'type': 'Secondary'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley",
         "{\"title\":\"\",\"abstract\":\"R E S E A R C H A R T I C L E Paleoseismology of the 2016 MW 6.1 Petermann earthquake source: Implications for intraplate earthquake behaviour and the geomorphic longevity of bedrock fault scarps in a low strain-rate cratonic region Tamarah Rosellen King1,2 | Mark Quigley1 | Dan Clark3 | Albert Zondervan4 | Jan-Hendrik May5 | Abaz Alimanovic1 1School of Earth Sciences, The University of Melbourne, Melbourne, Victoria, Australia 2COMET, Department of Earth Sciences, University of Oxford, Oxford, UK 3Geoscience Australia, Canberra, Australian Capital Territory, Australia 4Rafter Radiocarbon Laboratory, GNS Science, Lower Hutt, New Zealand 5School of Geography, The University of Melbourne, Melbourne, Victoria, Australia Correspondence Tamarah Rosellen King, School of Earth Sciences, The University of Melbourne, Melbourne, VIC 3010, Australia. \\nEmail: tamarah.king@earth.ox.ac.uk Funding information British Geological Survey; University of Melbourne; Australian Research Council, Grant/Award Number:\\nDP170103350 Abstract The 20 May 2016 MW 6.1 Petermann earthquake in central Australia generated a 21 km surface rupture with 0.1 to 1 m vertical displacements across a low-relief landscape.\\nNo paleo-scarps or potentially analogous topographic features are evident in pre-earthquake Worldview-1 and Worldview-2 satellite data.\\nTwo excavations across the surface rupture expose near-surface fault geometry and mixed aeolian-sheetwash sediment faulted only in the 2016 earthquake.\\nA 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture.\\nSeven cosmogenic beryllium-10 (10Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging-wall and foot-w\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1111/j.1475-4754.1999.tb00987.x'], context: \\\" Laslett, G.M., Yoshida, H. & Olley, J.M. (1999) Optical dating of single and multiple grains of quartz from Jinmium rock shelter, northern Australia: Part I, experimental design and statistical models.\\nArchaeometry, 41(2), 339\\u2013364.\\nhttps://doi. org/10.1111/j.1475-4754.1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104\\u2013117.\\nhttps://doi.org/10.1177/095968\\\"},{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], context: \\\"1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104\\u2013117.\\nhttps://doi.org/10.1177/0959683613515732 Gold, R.D., Briggs, R.W., Crone, A.J. & DuRoss, C.B. (2017) Refining fault slip rates using multiple displaced terrace risers\\u2014an example from the Honey Lake fault, NE California, USA.\\nEarth and Planetary Science Letters, 477, 134\\u2013146.\\nhttps://doi.or\\\"}\",\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], context: \\\" Cummins, P.R. (2009) Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low-T thermochronological data. \\nAustralian Journal of Earth Sciences, 56(2), 99\\u2013110.\\nhttps://doi.org/10.1080/08120090802546977 Calais, E., Camelbeeck, T., Stein, S., Liu, M. & Craig, T.J. (2016)\\nA new para-digm for large earthquakes in stable continental plate interiors. \\nGeophysical Research Letters, 43(20), 10,621\\u201310,637. https://doi.org/10.1002/2016GL070815 Calais, E., M\\\"},{\\\"dataset_ids\\\": ['10.1038/nature04428'], context: \\\" Research Letters, 43(20), 10,621\\u201310,637. https://doi.org/10.1002/2016GL070815 Calais, E., Mattioli, G., DeMets, C., Nocquet, J.-M., Stein, S., Newman, A. & Rydelek, P. (2005)\\nTectonic strain in plate interiors?\\nNature, 438, E9\\u2013E10. \\nhttps://doi.org/10.1038/nature04428 \\n[online] Available from: http://www.nature.com/articles/nature04428 Clark, D. (2012) Neotectonic Features Database. \\nCanberra, ACT: Geoscience Australia, Commonwealth of Australia. \\nClark, D. (2018)\\nWhat have we learned in the 50 years since the 19\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5090'], context: \\\" 10969837, 2021, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5090, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/S0264-3707'], context: \\\"n stable continental regions (USGS Bulletin 2032-B). \\nWashington, DC: US Geological Survey. \\nMachette, M.N. (2000) Active, capable, and potentially active faults \\u2013 a paleoseismic perspective.\\nJournal of Geodynamics, 29(3-5), 387\\u2013392. https://doi.org/10.1016/S0264-3707(99)00060-5 Magee, J.W. (2009)\\nA Literature Review Palaeovalley Groundwater Resources in Arid and Semi-Arid Australia - A literature review.\\nCanberra, Australia: Geoscience Australia. \\nhttp://pid.geoscience.gov.au/dataset/ga/68542 McCalpin, J.P. (200\\\"}\"]}",
         "{\"title\":\"\",\"abstract\":\"Abstract The 20 May 2016  M W  6.1 Petermann earthquake in central Australia generated a 21\\u2009km surface rupture with 0.1 to 1 m vertical displacements across a low\\u2010relief landscape. No paleo\\u2010scarps or potentially analogous topographic features are evident in pre\\u2010earthquake Worldview\\u20101 and Worldview\\u20102 satellite data. Two excavations across the surface rupture expose near\\u2010surface fault geometry and mixed aeolian\\u2010sheetwash sediment faulted only in the 2016 earthquake. A 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture. Seven cosmogenic beryllium\\u201010 ( 10 Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging\\u2010wall and foot\\u2010wall, and three from samples 19 to 50\\u2009km from the surface rupture. No distinction is found between fault proximal rates (1.3 \\u00b1 0.1 to 2.6 \\u00b1 0.2 m Myr \\u22121 ) and distal samples (1.4 \\u00b1 0.1 to 2.3 \\u00b1 0.2 m Myr \\u22121 ). The thickness of rock fragments (2\\u20135 cm) coseismically displaced in the Petermann earthquake perturbs the steady\\u2010state bedrock erosion rate by only 1 to 3%, less than the erosion rate uncertainty estimated for each sample (7\\u201312%). Using  10 Be erosion rates and scarp height measurements we estimate approximately 0.5 to 1 Myr of differential erosion is required to return to pre\\u2010earthquake topography. By inference any pre\\u20102016 fault\\u2010related topography likely required a similar time for removal. We conclude that the Petermann earthquake was the first on this fault in the last ca. 0.5\\u20131 Myr. Extrapolating single nuclide erosion rates across this timescale introduces large uncertainties, and we cannot resolve whether 2016 represents the first ever surface rupture on this fault, or a\\u2009> 1 Myr interseismic period. Either option reinforces the importance of including distributed earthquake sources in fault displacement and seismic hazard analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], context: \\\"Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low\\u2010T thermochronological data|Australian Journal of Earth Sciences|https://doi.org/10.1080/08120090802546977\\\"}\",\"{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], context: \\\"Late\\u2010Holocene climatic variability indicated by three natural archives in arid southern Australia|Holocene|https://doi.org/10.1177/0959683613515732\\\"}\"]}"
        ],
        [
         "27",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_mp.14424.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines Kendall J. Kisera) John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Sara Ahmed and Sonja Stieb Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Abdallah S. R. Mohamed Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Hesham Elhalawani Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA Peter Y. S. Park, Nathan S. Doyle, and Brandon J. Wang Department of Diagnostic and Interventional Imaging, John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Arko Barman, Zhao Li, and W. Jim Zheng Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Clifton D. Fullera) Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Luca Giancardoa) Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA (Received 7 April 2020; revised 22 July 2020; accepted for publication 27 July 2020; published 28 August 2020) \\nThis manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non-small cell lung cancer.\\nThe segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction.\\nBilateral thoracic cavity volumes and pleural effusion volumes were manually seg-men\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39.51'], context: \\\"We describe PleThora, a dataset of 402 expert-vetted tho-racic cavity segmentations, 78 expert-vetted pleural effusion segmentations, and corresponding clinical and technical metadata made available to the public through TCIA at https://doi.org/10.7937/tcia.2020.6c7y-gq39.51 These seg-mentations have value for preprocessing steps in image analy-sis pipelines built for fundamental quantitative imaging tasks, including but not limited to pathologic lung segmentation, lesion detection, and radiomics feature extraction. \\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], context: \\\"hive (TCIA): maintaining and operating a public information repository.\\nJ Digit Imag-ing.\\n2013;26:1045\\u20131057. \\n22.\\nAerts HJWL, Wee L, Rios Velazquez E, et al.\\nData from NSCLC-Radio-mics\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2019. https://doi.org/10.7937/K9/TCIA.2015.PF0M9REI 23.\\nAerts HJ, Velazquez ER, Leijenaar RT, et al.\\nDecoding tumour pheno-type by noninvasive imaging using a quantitative radiomics approach. \\nNat Commun.\\n2014;5:4006. \\n24.\\nLi X, Morgan PS, Ashburner J, Smith J, Rorden C. dcm2niix.exe [com-puter progr\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], context: \\\"51.\\nKiser KJ, Ahmed S, Stieb SM, et al.\\nData from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2020.\\nhttps://doi.org/10.7937/tcia.2020.6c7y-gq39 52.\\nRonneberger O, Fischer P, Brox T. U-Net: Convolutional Networks for Biomedical Image Segmentation.\\nPaper presented at: Medical Image Computing and Computer-Assisted Intervention \\u2013 MICCAI 2015; 5\\u20139 October, 2015; Munich, Germany.\\nhttps://doi.org/\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/mp.14424'], context: \\\" 24734209, 2020, 11,\\nDownloaded from https://aapm.onlinelibrary.wiley.com/doi/10.1002/mp.14424 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License \\\"}\"]}",
         "{\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}"
        ],
        [
         "28",
         "10.1002_nafm.10870",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_nafm.10870.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9gtumay', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract Objective: Silver Chub Macrhybopsis storeriana, a predominately riverine species  throughout its native range, exists within Lake Erie as the only known lake popu-lation.\\nIts population declined in the 1950s and never fully recovered.\\nCanada has  listed Silver Chub in the Great Lakes\\u2013\\u00adSt.\\nLawrence River as endangered and has  initiated a recovery plan that recognized the identification of spawning areas as a  critical component to inform Silver Chub's recovery potential. \\nMethods: We investigated potential spawning locations of Silver Chub using cap-ture records, otolith microchemistry, and daily age analysis.\\nLapillus otolith\\nSr:Ca ra-tios from 27 age-\\u00ad0 Silver Chub were used to identify potential spawning areas.\\nDaily  ages estimated from lapilli were used to calculate hatch dates, which then were com-pared with capture data of adults and river flows to further inform potential spawn-ing areas. \\nResult: The Detroit River (and its nearshore area) was all but ruled out as a potential  spawning location.\\nThe Maumee, Portage, and Sandusky rivers or their nearshore  areas were all possible spawning locations.\\nProjected hatch dates spanned the end  of May through the end of June and occurred across a wide range of flows, although  some peaks in hatch dates corresponded to flow peaks, indicating recruitment is po-tentially enhanced by high flows. \\nConclusion: Silver Chub spawning period and hypothesized spawning rivers or  lacustuaries overlap those of invasive Grass Carp Ctenopharyngodon idella, creat-ing a need to jointly consider Grass Carp control efforts with conservation of Silver  Chub when assessing management alternatives.\\nFurther research on spawning  guild and the use of rivers themselves or nearshore areas influenced by rivers as  spawning areas are required to maximize potential for conservation and recovery  of Silver Chub. \",\"data_availability\":\"DATA AVAILABILITY STATEMENT U.S. Geological Survey data are publicly available at the  websites in the references section.\\nWestern Basin intera-gency August trawl survey data are available from the For-age Task Group of the Lake Erie Committee (http://www. glfc.org/lake-erie-committee.php) by request.\\nData col-lected by the U.S. Fish and Wildlife Service are available  upon request from Anjanette Bowen (anjanette_bowen@ fws.gov).\\nSilver Chub otolith microchemistry data\\nare  available upon request from John Farver or Jeff Miner  (jfarver@bgsu.edu or jminer@bgsu.edu, respectively). \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5066/F75M63X0.'], context: \\\"Fisheries Society, Bethesda, Maryland. \\nUSGS (U.S. Geological Survey). 2019a.\\nGreat Lakes Science Center  Great Lakes research vessel operations 1958\\u2013\\u00ad2018.\\n(version  3.0, April 2019).\\nU.S. Geological Survey data release.\\nAvailable:  https://doi.org/10.5066/F75M63X0. \\nUSGS (U.S. Geological Survey). 2019b.\\nNewly hatched invasive Grass  Carp found in Maumee River, OH.\\nU.S. Geological Survey,  Office of Communications and Publishing, Reston, Virginia.  \\nAvailable: www.usgs.gov/news/newly-\\u00adhatched-\\u00adinvasive-\\u00adgrass -\\\"}\",\"{\\\"dataset_ids\\\": ['10.5066/P9GTUMAY.'], context: \\\"associated species.\\nDoctoral dissertation.\\nThe Ohio State  University, Columbus. \\nKo\\u010dovsk\\u00fd, P. M., and J. Farver. 2019.\\nWater chemistry of Great Lakes  tributaries, 2017\\u2013\\u00ad2018.\\nU.S. Geological Survey data release.  \\nAvailable: https://doi.org/10.5066/P9GTUMAY. \\nKo\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. \\\"},{\\\"dataset_ids\\\": ['10.5066/P94BH3W0.'], context: \\\"\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. org/10.5066/P94BH3W0. \\nKo\\u010dovsk\\u00fd, P. M., N. R. King, E. Weimer, C. Mayer, and S. S. Qian.  2021.\\nValidation of the model-\\u00adpredicted spawning area of Grass  Carp Ctenopharyngodon idella in the Sandusky River.\\nJournal  of Great Lakes Research 47:29\\u201336. \\nKoel, T. M., and R. \\\"}\"]}",
         ""
        ],
        [
         "29",
         "10.1007_jhep07(2018)134",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep07(2018)134.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract: The production cross-sections of \\u03a5(1S), \\u03a5(2S) and \\u03a5(3S) mesons in proton-proton collisions at \\u221as = 13 TeV are measured with a data sample corresponding to an integrated luminosity of 277 \\u00b1 11 pb\\u22121 recorded by the LHCb experiment in 2015.\\nThe \\u03a5 mesons are reconstructed in the decay mode \\u03a5 \\u2192\\u00b5+\\u00b5\\u2212.\\nThe differential production cross-sections times the dimuon\\nbranching fractions are measured as a function of the \\u03a5 transverse momentum, pT, and rapidity, y, over the range 0 < pT < 30 GeV/c and 2.0 < y < 4.5.\\nThe ratios of the cross-sections with respect to the LHCb measurement at \\u221as = 8 TeV are also determined.\\nThe measurements are compared with theoretical predictions based on NRQCD. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "30",
         "10.1007_jhep11(2018)113",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract: Measurements of the differential jet cross section are presented as a function of the jet mass in dijet events, in bins of jet transverse momentum, with and without a jet grooming algorithm.\\nThe data have been recorded by the CMS Collaboration in proton-proton collisions at the LHC at a center-of-mass energy of 13 TeV and correspond to an integrated luminosity of 2.3 fb\\u22121.\\nThe absolute cross sections show slightly different jet transverse momentum spectra in data and Monte Carlo event generators for the settings used.\\nRemoving this transverse momentum dependence, the normalized cross section for ungroomed jets is consistent with the prediction from Monte Carlo event generators for masses below 30% of the transverse momentum.\\nThe normalized cross section for groomed jets is measured with higher precision than the ungroomed cross section.\\nSemi-analytical calculations of the jet mass beyond leading logarithmic accuracy are compared to data, as well as predictions at leading order and next-to-leading order, which include parton showering and hadronization. \\nOverall, in the normalized cross section, the theoretical predictions agree with the measured cross sections within the uncertainties for masses from 10 to 30% of the jet transverse momentum. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "31",
         "10.1007_jhep11(2018)115",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract: A search for charged Higgs boson decaying to a charm and a bottom quark (H+ \\u2192cb) is performed using 19.7 fb\\u22121 of pp collision data at \\u221as = 8 TeV.\\nThe production mechanism investigated in this search is tt pair production in which one top quark decays to a charged Higgs boson and a bottom quark and the other decays to a charged lepton, a neutrino, and a bottom quark. \\nCharged Higgs boson decays to cb are searched for, resulting in a final state containing at least four jets, a charged lepton (muon or electron), and missing transverse momentum.\\nA kinematic fit is performed to identify the pair of jets least likely to be the bottom quarks originating from direct top quark decays and the invariant mass of this pair is used as the final observable in the search.\\nNo evidence for the presence of a charged Higgs boson is observed and upper limits at 95% confidence level of 0.8\\u20130.5% are set on the branching fraction B(t \\u2192H+b), assuming B(H+ \\u2192cb)\\n= 1.0 and B(t \\u2192H+b) + B(t \\u2192Wb)\\n= 1.0, for the charged Higgs boson mass range 90\\u2013150 GeV. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "32",
         "10.1007_jhep12(2018)117",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract: The study of global event shape variables can provide sensitive tests of pre-dictions for multijet production in proton-proton collisions.\\nThis paper presents a study of several event shape variables calculated using jet four momenta in proton-proton collisions at a centre-of-mass energy of 13 TeV and uses data recorded with the CMS detector at the LHC corresponding to an integrated luminosity of 2.2 fb\\u22121.\\nAfter correcting for de-tector effects, the resulting distributions are compared with several theoretical predictions. \\nThe agreement generally improves as the energy, represented by the average transverse momentum of the two leading jets, increases. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "33",
         "10.1007_s00259-022-06053-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00259-022-06053-8.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00259-022-06053-8.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2017.7hs46erv', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2019.30ilqfcl', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract Purpose Attenuation correction and scatter compensation (AC/SC) are two main steps toward quantitative PET imaging,  which remain challenging in PET-only and PET/MRI systems.\\nThese can be effectively tackled via deep learning (DL)  methods.\\nHowever, trustworthy, and generalizable DL models commonly require well-curated, heterogeneous, and large  datasets from multiple clinical centers.\\nAt the same time, owing to legal/ethical issues and privacy concerns, forming a  large collective, centralized dataset poses significant challenges.\\nIn this work, we aimed to develop a DL-based model in a  multicenter setting without direct sharing of data using federated learning (FL) for AC/SC of PET images. \\nMethods Non-attenuation/scatter corrected and CT-based attenuation/scatter corrected (CT-ASC) 18F-FDG PET images  of 300 patients were enrolled in this study.\\nThe dataset consisted of 6 different centers, each with 50 patients, with scan-ner, image acquisition, and reconstruction protocols varying across the centers.\\nCT-based ASC PET images served as the  standard reference.\\nAll images were reviewed to include high-quality and artifact-free PET images.\\nBoth corrected and  uncorrected PET images were converted to standardized uptake values (SUVs).\\nWe used a modified nested U-Net utilizing  residual U-block in a U-shape architecture.\\nWe evaluated two FL models, namely sequential (FL-SQ) and parallel (FL-PL)  and compared their performance with the baseline centralized (CZ) learning model wherein the data were pooled to one  server, as well as center-based (CB) models where for each center the model was built and evaluated separately.\\nData from  each center were divided to contribute to training (30 patients), validation (10 patients), and test sets (10 patients).\\nFinal  evaluations and reports were performed on 60 patients (10 patients from each center). \\nResults In terms of percent SUV absolute relative error (ARE%), both FL-SQ (CI:12.21\\u201314.81%) and FL-PL (CI:11.82\\u2013 13.84%) models demonstrated excellent agreement with the centralized framework (CI:10.32\\u201312.00%), while FL-based  algorithms improved model performance by over 11% compared to CB training strategy (CI: 22.34\\u201326.10%).\\nFurthermore,  the Mann\\u2013Whitney test between different strategies revealed no significant differences between CZ and FL-based algorithms  (p-value > 0.05) in center-categorized mode.\\nAt the same time, a significant difference was observed between the different  training approaches on the overall dataset (p-value < 0.05).\\nIn addition, voxel-wise comparison, with respect to reference  CT-ASC, exhibited similar performance for images predicted by CZ (R2 = 0.94), FL-SQ (R2 = 0.93), and FL-PL (R2 = 0.92),  while CB model achieved a far lower coefficient of determination (R2 = 0.74).\\nDespite the strong correlations between CZ  and FL-based methods compared to reference CT-ASC, a slight underestimation of predicted voxel values was observed. \\nConclusion Deep learning-based models provide promising results toward quantitative PET image reconstruction.\\nSpecifically,  we developed two FL models and compared their performance with center-based and centralized models.\\nThe proposed FL-based models achieved higher performance compared to center-based models, comparable with centralized models.\\nOur work  provided strong empirical evidence that the FL framework can fully benefit from the generalizability and robustness of DL  models used for AC/SC in PET, while obviating the need for the direct sharing of datasets between clinical imaging centers. \",\"data_availability\":\"Data availability Data are available from The Cancer Imaging Archive  (TCIA) from refereces of [20, 39\\u201347]. \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/978-3-030-60548-3_18'], context: \\\"density classification: a real-world  implementation.\\nIn: Domain Adaptation and Representation  Transfer, and Distributed and Collaborative Learning: Lecture  Notes in Computer Science, Vol. 12444.\\nSpringer, Cham. 2020.  \\npp.\\n181\\u201391.\\nhttps://doi.org/10.1007/978-3-030-60548-3_18  39.\\nClark K, Vendt B, Smith K, Freymann J, Kirby J, Koppel P, et al.  \\nThe Cancer Imaging Archive (TCIA): maintaining and operating a  public information repository.\\nJ Digit Imaging.\\n2013;26:1045\\u201357.  \\nhttps://doi.org/10.1007/s10278-013-9622-7.  40\\\"},{\\\"dataset_ids\\\": ['10.1007/s10278-013-9622-7.'], context: \\\"g/10.1007/978-3-030-60548-3_18  39.\\nClark K, Vendt B, Smith K, Freymann J, Kirby J, Koppel P, et al.  \\nThe Cancer Imaging Archive (TCIA): maintaining and operating a  public information repository.\\nJ Digit Imaging.\\n2013;26:1045\\u201357.  \\nhttps://doi.org/10.1007/s10278-013-9622-7.  40.\\nMachtay M, Duan F, Siegel BA, Snyder BS, Gorelick JJ, Red-din JS, et al.\\nPrediction of survival by [18F]fluorodeoxyglucose  positron emission tomography in patients with locally advanced  non-small-cell lung cancer undergoing definitive chemora\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2019.30ilqfcl'], context: \\\" 41.\\nKinahan P, Muzi M, Bialecki B, Herman B, Coombs L. Data from  the ACRIN 6668 Trial NSCLC-FDG-PET.\\nCancer Imaging Arch.  \\n2019;10.\\nhttps://doi.org/10.7937/tcia.2019.30ilqfcl  42.\\nBakr S, Gevaert O, Echegaray S, Ayers K, Zhou M, Shafiq M, et al.  Data for NSCLC radiogenomics collection.\\nThe Cancer Imaging  Archive.\\n2017.\\nhttps://doi.org/10.7937/K9/TCIA.2017.7hs46erv.  43.\\nBakr S, Gevaert O, Echegaray S, Ayers K, Zhou M, \\\"},{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2017.7hs46erv.'], context: \\\"C-FDG-PET.\\nCancer Imaging Arch.  \\n2019;10.\\nhttps://doi.org/10.7937/tcia.2019.30ilqfcl  42.\\nBakr S, Gevaert O, Echegaray S, Ayers K, Zhou M, Shafiq M, et al.  Data for NSCLC radiogenomics collection.\\nThe Cancer Imaging  Archive.\\n2017.\\nhttps://doi.org/10.7937/K9/TCIA.2017.7hs46erv.  43.\\nBakr S, Gevaert O, Echegaray S, Ayers K, Zhou M, Shafiq M,  et al.\\nA radiogenomic dataset of non-small cell lung cancer.\\nSci  Data.\\n2018;5: 180202.\\nhttps://doi.org/10.1038/sdata.2018.202.  44.\\nGevaert O, Xu J, Hoang CD, Leung AN, Xu Y, Quon A, \\\"},{\\\"dataset_ids\\\": ['10.1038/sdata.2018.202.'], context: \\\"he Cancer Imaging  Archive.\\n2017.\\nhttps://doi.org/10.7937/K9/TCIA.2017.7hs46erv.  43.\\nBakr S, Gevaert O, Echegaray S, Ayers K, Zhou M, Shafiq M,  et al.\\nA radiogenomic dataset of non-small cell lung cancer.\\nSci  Data.\\n2018;5: 180202.\\nhttps://doi.org/10.1038/sdata.2018.202.  44.\\nGevaert O, Xu J, Hoang CD, Leung AN, Xu Y, Quon A, et al.  \\nNon-small cell lung cancer: identifying prognostic imaging bio-markers by leveraging public gene expression microarray data\\u2013 methods and preliminary results.\\nRadiology.\\n2012;264:387\\u201396\\\"},{\\\"dataset_ids\\\": ['10.1148/radiol.12111607.'], context: \\\"J, Hoang CD, Leung AN, Xu Y, Quon A, et al.  \\nNon-small cell lung cancer: identifying prognostic imaging bio-markers by leveraging public gene expression microarray data\\u2013 methods and preliminary results.\\nRadiology.\\n2012;264:387\\u201396.  \\nhttps://doi.org/10.1148/radiol.12111607.  45.\\nGrossberg A, Elhalawani H, Mohamed A, Mulder S, Williams  B, White A, et al. MD Anderson Cancer Center Head and Neck  Quantitative Imaging Working Group.(2020) HNSCC .\\nThe  Cancer Imaging Archive. doi:107937/k9/tcia.\\n2020:a8sh-7363.  46.\\nGrossb\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.cpet.2021.06.010.'], context: \\\"00259-020-04852-5.  21.\\nMcMillan AB, Bradshaw TJ.\\nArtificial Intelligence-based data  corrections for attenuation and scatter in position emission tomog-raphy and single-photon emission computed tomography.\\nPET  Clin.\\n2021;16:543\\u201352.\\nhttps://doi.org/10.1016/j.cpet.2021.06.010.  22.\\nRieke N, Hancox J, Li W, Milletar\\u00ec F, Roth HR, Albarqouni S,  et al.\\nThe future of digital health with federated learning.\\nNPJ Digit  Med.\\n2020;3:119.\\nhttps://doi.org/10.1038/s41746-020-00323-1.  23.\\nKaissis GA, Makowski MR, R\\u00fcckert D, Braren R\\\"},{\\\"dataset_ids\\\": ['10.1038/s41746-020-00323-1.'], context: \\\"raphy.\\nPET  Clin.\\n2021;16:543\\u201352.\\nhttps://doi.org/10.1016/j.cpet.2021.06.010.  22.\\nRieke N, Hancox J, Li W, Milletar\\u00ec F, Roth HR, Albarqouni S,  et al.\\nThe future of digital health with federated learning.\\nNPJ Digit  Med.\\n2020;3:119.\\nhttps://doi.org/10.1038/s41746-020-00323-1.  23.\\nKaissis GA, Makowski MR, R\\u00fcckert D, Braren RF.\\nSecure, pri-vacy-preserving and federated machine learning in medical imag-ing.\\nNat Mach Intell.\\n2020;2:305\\u201311. \\\"}\"]}",
         "{\"title\":\"Decentralized collaborative multi-institutional PET attenuation and scatter correction using federated deep learning\",\"abstract\":\"Purpose Attenuation correction and scatter compensation (AC/SC) are two main steps toward quantitative PET imaging, which remain challenging in PET-only and PET/MRI systems. These can be effectively tackled via deep learning (DL) methods. However, trustworthy, and generalizable DL models commonly require well-curated, heterogeneous, and large datasets from multiple clinical centers. At the same time, owing to legal/ethical issues and privacy concerns, forming a large collective, centralized dataset poses significant challenges. In this work, we aimed to develop a DL-based model in a multicenter setting without direct sharing of data using federated learning (FL) for AC/SC of PET images. Methods Non-attenuation/scatter corrected and CT-based attenuation/scatter corrected (CT-ASC)  18 F-FDG PET images of 300 patients were enrolled in this study. The dataset consisted of 6 different centers, each with 50 patients, with scanner, image acquisition, and reconstruction protocols varying across the centers. CT-based ASC PET images served as the standard reference. All images were reviewed to include high-quality and artifact-free PET images. Both corrected and uncorrected PET images were converted to standardized uptake values (SUVs). We used a modified nested U-Net utilizing residual U-block in a U-shape architecture. We evaluated two FL models, namely sequential (FL-SQ) and parallel (FL-PL) and compared their performance with the baseline centralized (CZ) learning model wherein the data were pooled to one server, as well as center-based (CB) models where for each center the model was built and evaluated separately. Data from each center were divided to contribute to training (30 patients), validation (10 patients), and test sets (10 patients). Final evaluations and reports were performed on 60 patients (10 patients from each center). Results In terms of percent SUV absolute relative error (ARE%), both FL-SQ (CI:12.21\\u201314.81%) and FL-PL (CI:11.82\\u201313.84%) models demonstrated\",\"data_availability\":\"Data availability Data are available from The Cancer Imaging Archive (TCIA) from refereces of [ 20 ,  39 \\u2013 47 ].\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1038/s41746-020-00323-1'], context: \\\"The future of digital health with federated learning|NPJ Digit Med|10.1038/s41746-020-00323-1|33015372\\\"}\",\"{\\\"dataset_ids\\\": ['10.1007/s10278-013-9622-7'], context: \\\"The Cancer Imaging Archive (TCIA): maintaining and operating a public information repository|J Digit Imaging|10.1007/s10278-013-9622-7|23884657\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2017.7hs46erv'], context: \\\"Data for NSCLC radiogenomics collection|The Cancer Imaging Archive|10.7937/K9/TCIA.2017.7hs46erv\\\"}\",\"{\\\"dataset_ids\\\": ['10.1038/sdata.2018.202'], context: \\\"A radiogenomic dataset of non-small cell lung cancer|Sci Data|10.1038/sdata.2018.202|30325352\\\"}\",\"{\\\"dataset_ids\\\": ['10.1038/sdata.2018.173'], context: \\\"Imaging and clinical data archive for head and neck squamous cell carcinoma patients treated with radiotherapy|Sci Data|10.1038/sdata.2018.173|30179230\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2019.30ilqfcl'], context: \\\"Kinahan P, Muzi M, Bialecki B, Herman B, Coombs L. Data from the ACRIN 6668 Trial NSCLC-FDG-PET. Cancer Imaging Arch. 2019;10. 10.7937/tcia.2019.30ilqfcl\\\"}\"]}"
        ],
        [
         "34",
         "10.1007_s00382-012-1636-1",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract We present the global general circulation model IPSL-CM5 developed to study the long-term response of the climate system to natural and anthropogenic forcings as part of the 5th Phase of the Coupled Model Intercomparison Project (CMIP5).\\nThis model includes an interactive carbon cycle, a representation of tropospheric and stratospheric chemistry, and a comprehensive representation of aerosols. \\nAs it represents the principal dynamical, physical, and bio-geochemical processes relevant to the climate system, it may be referred to as an Earth System Model.\\nHowever, the IPSL-CM5 model may be used in a multitude of \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1029/2009GL038227'], context: \\\"ys Res Lett 36:L21602.\\ndoi: 10.1029/2009GL040145 Lenton A, Codron F, Bopp L, Metzl N, Cadule P, Tagliabue A, Sommer JL (2009)\\nStratospheric ozone depletion reduces ocean carbon uptake and enhances ocean acidification.\\nGeophys Res Lett 36:L12606.\\ndoi:10.1029/2009GL038227 Liebmann B, Smith CA (1996) Description of a complete (interpo-lated) outgoing longwave radiation dataset.\\nBull Am Meteorol Soc 77(6):1275\\u20131277 Lloyd J, Guilyardi E, Weller H (2011)\\nThe role of atmosphere feedbacks during ENSO in the CMIP3 models.\\nP\\\"},{\\\"dataset_ids\\\": ['10.1007/s00382-005-0064-x'], context: \\\"n-tation of mountain lift forces.\\nMon Wea Rev 127:788\\u2013801 Lott F, Fairhead L, Hourdin F, Levan P (2005)\\nThe stratospheric version of LMDz: dynamical climatologies, arctic oscillation, and impact on the surface climate.\\nClim Dyn 25(7\\u20138):851\\u2013868. \\ndoi:10.1007/s00382-005-0064-x Louis JF (1979)\\nA parametric model of vertical eddy fluxes in the atmosphere.\\nBoundary Layer Meteorol 17:187\\u2013202 Loveland TR, Reed BC, Brown JF, Ohlen DO, Zhu Z, Yang L, Merchant JW (2000) Development of a global land cover characteristics database \\\"},{\\\"dataset_ids\\\": ['10.1080/014311600210191'], context: \\\"ndary Layer Meteorol 17:187\\u2013202 Loveland TR, Reed BC, Brown JF, Ohlen DO, Zhu Z, Yang L, Merchant JW (2000) Development of a global land cover characteristics database and IGBP DISCover from 1 km AV-HRR data.\\nInt J Remote Sens 21(6\\u20137):1303\\u20131330.\\ndoi:10.1080/014311600210191 Ludwig W, Probst J, Kempe S (1996)\\nPredicting the oceanic input of organic carbon by continental erosion.\\nGlobal Biogeochemical Cycles 10(1):23\\u201341.\\ndoi:10.1029/95GB02925 Lyman JM, Good SA, Gouretski VV, Ishii M, Johnson GC, Palmer MD, Smith DM, Will\\\"},{\\\"dataset_ids\\\": ['L12606'], context: \\\"model.\\nGeophys Res Lett 36:L21602.\\ndoi: 10.1029/2009GL040145 Lenton A, Codron F, Bopp L, Metzl N, Cadule P, Tagliabue A, Sommer JL (2009)\\nStratospheric ozone depletion reduces ocean carbon uptake and enhances ocean acidification.\\nGeophys Res Lett 36:L12606.\\ndoi:10.1029/2009GL038227 Liebmann B, Smith CA (1996) Description of a complete (interpo-lated) outgoing longwave radiation dataset.\\nBull Am Meteorol Soc 77(6):1275\\u20131277 Lloyd J, Guilyardi E, Weller H (2011)\\nThe role of atmosphere feedbacks during E\\\"}\",\"{\\\"dataset_ids\\\": ['10.1175/2008JCL'], context: \\\" warming impacts on regional tropical precipitation.\\nJ Climate 17(13):2688\\u2013 2701 Chou C, Neelin JD, Chen CA, Tu JY (2009)\\nEvaluating the \\u2019\\u2019rich-get-richer\\u2019\\u2019 mechanism in tropical precipitation change under global warming.\\nJ Clim 22(8):1982\\u20132005.\\ndoi:10.1175/2008JCL I2471.1 Cionni I, Eyring V, Lamarque JF, Randel WJ, Stevenson DS, Wu F, Bodeker GE, Shepherd TG, Shindell DT, Waugh DW (2011) Ozone database in support of cmip5 simulations: results and corresponding radiative forcing.\\nAtmos Chem Phys 11(21): 11267\\u2013\\\"},{\\\"dataset_ids\\\": ['10.5194/acp-11-11267-2011'], context: \\\"onni I, Eyring V, Lamarque JF, Randel WJ, Stevenson DS, Wu F, Bodeker GE, Shepherd TG, Shindell DT, Waugh DW (2011) Ozone database in support of cmip5 simulations: results and corresponding radiative forcing.\\nAtmos Chem Phys 11(21): 11267\\u201311292.\\ndoi:10.5194/acp-11-11267-2011 Clement A, DiNezio P, Deser C (2011)\\nRethinking the Ocean\\u2019s role in the Southern Oscillation.\\nJ Clim 24(15):4056\\u20134072.\\ndoi: 10.1175/2011JCLI3973.1 Cleveland RB, Cleveland WS, McRae J, Terpenning I (1990)\\nSTL: A seasonal-trend decomposition procedure\\\"}\",\"{\\\"dataset_ids\\\": ['10.1007/BF00211684'], context: \\\"008)\\nNEMO ocean engine.\\nTechnical note, IPSL, available at http://www.nemo-ocean.eu/content/download/11245/56055/file/NEMO_book_v3_2.pdf Madec G, Imbard M (1996)\\nA global ocean mesh to overcome the North Pole singularity.\\nClim Dyn 12(6):381\\u2013388.\\ndoi:10.1007/BF00211684 Madronich S, Flocke S (1998)\\nThe role of solar radiation in atmospheric chemistry.\\nIn: Boule P (eds) Handbook of environ-mental chemistry.\\nSpringer, Heidelberg, pp 1\\u201326 Manabe S, Stouffer RJ (1980) Sensitivity of a global climate model to an increas\\\"},{\\\"dataset_ids\\\": ['10.1175/BAMS-88-9-1383'], context: \\\"arison project.\\nBull Am Meteorol Soc 86(1):89\\u201393 Meehl GA, Covey C, Delworth T, Latif M, McAvaney B, Mitchell JFB, Stouffer RJ (2007)\\nThe WCRP CMIP3 multi-model dataset: a new era in climate change research.\\nBull Am Meteorol Soc 88(9):1383\\u20131394. doi:10.1175/BAMS-88-9-1383 Meehl GA, Stocker TF, Collins WD, Friedlingstein P, Gaye AT, Gregory JM, Kitoh A, Knutti R, Murphy JM, Noda A, Raper SCB, Watterson IG, Weaver AJ, Zhao ZC (2007)\\nGlobal climate projections.\\nIn: Solomon S, Qin D, Manning M, Chen Z, Marquis M, Averyt \\\"}\"]}",
         ""
        ],
        [
         "35",
         "10.1007_s00382-022-06361-7",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-022-06361-7.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00382-022-06361-7.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6075/j0154fjj', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.6075/j089161b', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Abstract Floods caused by atmospheric rivers and wildfires fanned by Santa Ana winds are common occurrences in California with  devastating societal impacts.\\nIn this work, we show that winter weather variability in California, including the occurrence  of extreme and impactful events, is linked to four atmospheric circulation regimes over the North Pacific Ocean previously  named and identified as the \\u201cNP4 modes\\u201d.\\nThese modes come in and out of phase with each other during the season, result-ing in distinct weather patterns that recur throughout the historical record.\\nSome phase combinations favor atmospheric  river landfalls and extreme daily or multi-day precipitation, while other phase combinations favor anomalously hot weather  and drying Santa Ana wind conditions over Southern California.\\nThis historical perspective of atmospheric circulation and  impacts over 70 years reveals that weather patterns are changing in a way that enhances wildfire hazard in California, while  the frequency of weather patterns linked to historical floods is not diminishing.\\nThese changes highlight the rising hazards  of cascading weather extremes in California\\u2019s present and future. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.6075/J0154FJJ.'], context: \\\"des, where we see that together these modes explain  25\\u201390% variance in the portion of the domain most relevant  for west coast weather.\\nThe NP4 dataset used in this study  is described in Guirguis et al. (2020b) and is available at  https://doi.org/10.6075/J0154FJJ. \\\"}\",\"{\\\"dataset_ids\\\": ['10.6075/J0154FJJ'], context: \\\"a and material The winter weather regimes  catalog developed for this study is available from the UC San  Diego Library Digital Collections at https://doi.org/10.6075/ J089161B (Guirguis et al. 2022).\\nThe NP4 dataset is available  at https://doi.org/10.6075/J0154FJJ (Guirguis et al. 2020b).\\nThe  Santa Ana winds regional index and the SIO-R1 AR catalog are  available at https://weclima.ucsd.edu/data-products/.\\nThe Livneh  temperature and precipitation data are available at https://clima tedataguide.ucar.edu/clim\\\"}\",\"{\\\"dataset_ids\\\": ['10.6075/J089161B'], context: \\\"on of that pattern.\\nA catalog of historical  weather regimes observed over 1949\\u20132017 is developed as an  accompaniment to this publication (Guirguis et al. 2022) and is  available from the UC San Diego Library Digital Collections at  https://doi.org/10.6075/J089161B \\\"}\",\"{\\\"dataset_ids\\\": ['10.6075/J089161B'], context: \\\"ales J, Aguilera R, Clemesha R, Cor-ringham TW, Delle Monache L, Reynolds D, Tardy A, Small I,  Ralph FM (2022)\\nHistorical catalog of winter weather regimes  impacting California, 1949-2017.\\nUC San Diego Library Digital  Collections.\\nhttps://doi.org/10.6075/J089161B Guzman Morales J, Gershunov A (2019)\\nClimate change suppresses  Santa Ana Winds of Southern California and sharpens their sea-sonality.\\nGRL.\\nhttps://doi.org/10.1029/2018GL080261 Guzman Morales J, Gershunov A, Theiss J, Li H, Cayan DR (2016)  \\nSanta \\\"},{\\\"dataset_ids\\\": ['10.1029/2018GL080261'], context: \\\"rnia, 1949-2017.\\nUC San Diego Library Digital  Collections.\\nhttps://doi.org/10.6075/J089161B Guzman Morales J, Gershunov A (2019)\\nClimate change suppresses  Santa Ana Winds of Southern California and sharpens their sea-sonality.\\nGRL.\\nhttps://doi.org/10.1029/2018GL080261 Guzman Morales J, Gershunov A, Theiss J, Li H, Cayan DR (2016)  \\nSanta Ana Winds of southern California: their climatology,  extremes, and behavior spanning six and a half decades.\\nGeophys  Res Lett.\\nhttps://doi.org/10.1002/2016GL067887 Hatchett BJ,\\\"}\"]}",
         "{\"title\":\"Winter wet-dry weather patterns driving atmospheric rivers and Santa Ana winds provide evidence for increasing wildfire hazard in California\",\"abstract\":\"Floods caused by atmospheric rivers and wildfires fanned by Santa Ana winds are common occurrences in California with devastating societal impacts. In this work, we show that winter weather variability in California, including the occurrence of extreme and impactful events, is linked to four atmospheric circulation regimes over the North Pacific Ocean previously named and identified as the \\\"NP4 modes\\\". These modes come in and out of phase with each other during the season, resulting in distinct weather patterns that recur throughout the historical record. Some phase combinations favor atmospheric river landfalls and extreme daily or multi-day precipitation, while other phase combinations favor anomalously hot weather and drying Santa Ana wind conditions over Southern California. This historical perspective of atmospheric circulation and impacts over 70 years reveals that weather patterns are changing in a way that enhances wildfire hazard in California, while the frequency of weather patterns linked to historical floods is not diminishing. These changes highlight the rising hazards of cascading weather extremes in California's present and future.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.6075/J0154FJJ'], context: \\\"Data from: Four atmospheric circulation regimes over the North Pacific and their relationship to California precipitation on daily to seasonal timescales|UC San Diego Library Digital Collections|10.6075/J0154FJJ\\\"}\",\"{\\\"dataset_ids\\\": ['10.6075/J089161B'], context: \\\"Historical catalog of winter weather regimes impacting California, 1949-2017|UC San Diego Library Digital Collections|10.6075/J089161B\\\"}\"]}"
        ],
        [
         "36",
         "10.1007_s00442-022-05201-z",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00442-022-05201-z.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00442-022-05201-z.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.wpzgmsbps', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract The width of a population's resource-use niche is determined by individual diet breadth (\\u201cwithin-individual component\\u201d)  and the degree of niche partitioning between individuals (\\u201cbetween-individual component\\u201d).\\nThe balance between these  two factors affects ecological stability and evolutionary trajectories, and may shift as ecological opportunity permits broader  population niches.\\nLakes in California\\u2019s Sierra Nevada Mountains vary in resource diversity for introduced brook trout  (Salvelinus fontinalis) due to elevation, lake morphometry, and watershed features.\\nWe compared the relative contributions  of within- and between-individual niche components to two measures of the dietary niches of thirteen populations of brook  trout: prey taxonomic composition and prey size distribution.\\nFor both taxonomic and size diversity of fish diets, population  niche width was positively related to both the within- and between-individual components.\\nFor taxonomic diversity, the two  components increased in parallel, while for size diversity, the between-individual component became more important relative  to the within-individual component in populations with the greatest niche widths.\\nOur results support the Niche Variation  Hypothesis that populations with broader niches are more heterogeneous among individuals and show that individual niche  width and individual specialization can operate in parallel to expand the population niche. \",\"data_availability\":\"Data availability All data are available at https://doi.org/10.5061/ dryad.wpzgmsbps. \",\"other_dataset_citations\":[]}",
         "{\"title\":\"Population niche width is driven by within-individual niche expansion and individual specialization in introduced brook trout in mountain lakes\",\"abstract\":\"The width of a population's resource-use niche is determined by individual diet breadth (\\u201cwithin-individual component\\u201d) and the degree of niche partitioning between individuals (\\u201cbetween-individual component\\u201d). The balance between these two factors affects ecological stability and evolutionary trajectories, and may shift as ecological opportunity permits broader population niches. Lakes in California\\u2019s Sierra Nevada Mountains vary in resource diversity for introduced brook trout ( Salvelinus fontinalis ) due to elevation, lake morphometry, and watershed features. We compared the relative contributions of within- and between-individual niche components to two measures of the dietary niches of thirteen populations of brook trout: prey taxonomic composition and prey size distribution. For both taxonomic and size diversity of fish diets, population niche width was positively related to both the within- and between-individual components. For taxonomic diversity, the two components increased in parallel, while for size diversity, the between-individual component became more important relative to the within-individual component in populations with the greatest niche widths. Our results support the Niche Variation Hypothesis that populations with broader niches are more heterogeneous among individuals and show that individual niche width and individual specialization can operate in parallel to expand the population niche. Supplementary Information The online version contains supplementary material available at 10.1007/s00442-022-05201-z.\",\"data_availability\":\"Data availability All data are available at  https://doi.org/10.5061/dryad.wpzgmsbps .\",\"other_dataset_citations\":[]}"
        ],
        [
         "37",
         "10.1007_s10904-014-0054-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Abstract Three mixed-metal nitrilotriacetates [Ba(H2O)3 Co(nta)Cl]n (1), [Ba(H2O)3Ni(nta)Cl]n (2), and [Ba(H2O)3 Cu(nta)Cl]n (3) (H3nta = nitrilotriacetic acid) were pre-pared by the reaction of M2?\\nions\\n(M = Co, Ni, and Cu), K3nta, and Ba2? ions in water.\\nThe reaction of complexes 1 and 2 with CuSO4\\ufffd5H2O afforded [Co(H2O)6]n[Cu2 (nta)2]n\\ufffd2nH2O (4) and [Ni(H2O)6]n[Cu2(nta)2]n\\ufffd2nH2O (5), respectively.\\nThe reaction mechanism was elucidated. \\nThe crystal structure analyses indicate that the Ba2? ions formed one-dimensional (1D) zigzag chains in complex 3, and the chains were connected by Cu(nta)- anions to form a three-dimensional network.\\nOn the other hand, in com-plexes 4 and 5, the Cu(nta)- anions formed 1D zigzag chains, and the [Co(H2O)6]2? (or [Ni(H2O)6]2?) cations existed as isolated units. \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "38",
         "10.1007_s12263-014-0408-4",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12263-014-0408-4.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract The discovery of vitamins and clarification of their role in preventing frank essential nutrient deficiencies occurred in the early 1900s.\\nMuch vitamin research has understandably focused on public health and the effects of single nutrients to alleviate acute conditions.\\nThe physio-logical processes for maintaining health, however, are complex systems that depend upon interactions between multiple nutrients, environmental factors, and genetic makeup.\\nTo analyze the relationship between these factors and nutritional health, data were obtained from an \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1038/oby.2009.282'], context: \\\"\\nCell 148:1293\\u20131307. doi:10.1016/j.cell.2012.02.009 Cheng CY, Reich D, Coresh J et al (2010)\\nAdmixture mapping of obesity-related traits in African Americans: the Atherosclerosis Risk in Communities (ARIC) Study.\\nObes (Silver Spring) 18:563\\u2013572. doi:10.1038/oby.2009.282 Csardi G, Nepusz T (2006)\\nThe igraph software package for complex network research.\\nInterJournal Complex Syst 1695.\\nhttp://cran.r-project.org/web/packages/igraph/citation.html Dominguez-Salas P, Moore SE, Cole D et al (2013) DNA methylation potentia\\\"}\",\"{\\\"dataset_ids\\\": ['10.1186/1471-2105-11-S6-S6', '10.1186/1471-2105-13-79'], context: \\\"Xu J, Wise C, Varma V et al (2010)\\nTwo new ArrayTrack libraries for personalized biomedical research.\\nBMC Bioinform 11(Suppl. \\n6):S6.\\ndoi:10.1186/1471-2105-11-S6-S6 Yu X, Wallqvist A, Reifman J (2012) Inferring high-confidence human protein\\u2013protein interactions.\\nBMC Bioinform 13:79. doi:10.1186/1471-2105-13-79 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1186/1755-8794-2-29'], context: \\\"f environmental factors: persistent pollutants and nutrients correlated with serum lipid levels.\\nInt J Epidemiol 41:828\\u2013843. doi:10.1093/ije/dys003 Ramos E, Rotimi C (2009)\\nThe A\\u2019s, G\\u2019s, C\\u2019s, and T\\u2019s of health disparities.\\nBMC Med Genomics 2:29.\\ndoi:10.1186/1755-8794-2-29 Reichardt J, Bornholdt S (2006)\\nStatistical mechanics of community detection.\\nPhys Rev E Stat Nonlin Soft Matter Phys 74:016110 Roux AVD (2011)\\nComplex systems thinking and current impasses in health disparities research.\\nAm J Public Health 101:1627\\\"},{\\\"dataset_ids\\\": ['10.2105/AJPH.2011.300149.Complex'], context: \\\", Bornholdt S (2006)\\nStatistical mechanics of community detection.\\nPhys Rev E Stat Nonlin Soft Matter Phys 74:016110 Roux AVD (2011)\\nComplex systems thinking and current impasses in health disparities research.\\nAm J Public Health 101:1627\\u20131634. \\ndoi:10.2105/AJPH.2011.300149.Complex Schulz LO, Bennett PH, Ravussin E et al (2006) Effects of traditional and western environments on prevalence of type 2 diabetes in Pima Indians in Mexico and the U.S. Diabetes Care 29:1866\\u20131871.\\ndoi:10.2337/dc06-0138 Slikker W Jr, Paule MG, Wright L\\\"},{\\\"dataset_ids\\\": ['10.1038/nature10354'], context: \\\"f-erol.\\nPart 2: calciferol and the significance of nutrient supple-ments.\\nInflamm Allergy Drug Targets 10:64\\u201374 Suhre K, Shin SY, Petersen AK et al (2011)\\nHuman metabolic individuality in biomedical and pharmaceutical research.\\nNature 477:54\\u201360. doi:10.1038/nature10354 Tanaka T, Ngwa JS, van Rooij FJA et al (2013) Genome-wide meta-analysis of observational studies shows common genetic variants associated with macronutrient intake. \\nAm J Clin Nutr 97:1395\\u20131402.\\ndoi:10.3945/ajcn.112.052183 Valentini L, Schaper L, Bu\\\"},{\\\"dataset_ids\\\": ['10.3945/ajcn.112.052183'], context: \\\"esearch.\\nNature 477:54\\u201360. doi:10.1038/nature10354 Tanaka T, Ngwa JS, van Rooij FJA et al (2013) Genome-wide meta-analysis of observational studies shows common genetic variants associated with macronutrient intake. \\nAm J Clin Nutr 97:1395\\u20131402.\\ndoi:10.3945/ajcn.112.052183 Valentini L, Schaper L, Buning C et al (2008) Malnutrition and impaired muscle strength in patients with Crohn\\u2019s disease and ulcerative colitis in remission.\\nNutrition 24:694\\u2013702. doi:10. \\n1016/j.nut.2008.03.018 Von Bertalanffy L (1950)\\nThe theory o\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/j.'], context: \\\"tiplex proteomic assay.\\nPLoS One 6:e26332. doi:10.1371/journal.pone.0026332 Kussmann M, Krause L, Siffert W (2010) Nutrigenomics: where are we with genetic and epigenetic markers for disposition and susceptibility?\\nNutr Rev 68(Suppl.\\n1):S38\\u2013S47.\\ndoi:10.1111/j. 1753-4887.2010.00326.x Lee Y-C, Lai C-Q, Ordovas JM, Parnell LD (2011)\\nA database of gene\\u2013environment interactions pertaining to blood lipid traits, cardiovascular disease and type 2 diabetes.\\nJ Data Min Genomics Proteomics 2:1\\u20138.\\ndoi:10.4172/2153-0\\\"},{\\\"dataset_ids\\\": ['10.4172/2153-0602.1000106'], context: \\\"doi:10.1111/j. 1753-4887.2010.00326.x Lee Y-C, Lai C-Q, Ordovas JM, Parnell LD (2011)\\nA database of gene\\u2013environment interactions pertaining to blood lipid traits, cardiovascular disease and type 2 diabetes.\\nJ Data Min Genomics Proteomics 2:1\\u20138.\\ndoi:10.4172/2153-0602.1000106 Liu JZ, McRae AF, Nyholt DR et al (2010)\\nA versatile gene-based test for genome-wide association studies.\\nAm J Hum Genet 87:139\\u2013145.\\ndoi:10.1016/j.ajhg.2010.06.009 Ma H, Sorokin A, Mazein A et al (2007)\\nThe Edinburgh human metabolic network reconstr\\\"}\"]}",
         "{\"title\":\"Genetic associations with micronutrient levels identified in immune and gastrointestinal networks\",\"abstract\":\"The discovery of vitamins and clarification of their role in preventing frank essential nutrient deficiencies occurred in the early 1900s. Much vitamin research has understandably focused on public health and the effects of single nutrients to alleviate acute conditions. The physiological processes for maintaining health, however, are complex systems that depend upon interactions between multiple nutrients, environmental factors, and genetic makeup. To analyze the relationship between these factors and nutritional health, data were obtained from an observational, community-based participatory research program of children and teens (age 6\\u201314) enrolled in a summer day camp in the Delta region of Arkansas. Assessments of erythrocyte  S -adenosylmethionine (SAM) and  S -adenosylhomocysteine (SAH), plasma homocysteine (Hcy) and 6 organic micronutrients (retinol, 25-hydroxy vitamin D3, pyridoxal, thiamin, riboflavin, and vitamin E), and 1,129 plasma proteins were performed at 3 time points in each of 2\\u00a0years. Genetic makeup was analyzed with 1\\u00a0M SNP genotyping arrays, and nutrient status was assessed with 24-h dietary intake questionnaires. A pattern of metabolites (met_PC1) that included the ratio of erythrocyte SAM/SAH, Hcy, and 5 vitamins were identified by principal component analysis. Met_PC1 levels were significantly associated with (1) single-nucleotide polymorphisms, (2) levels of plasma proteins, and (3) multilocus genotypes coding for gastrointestinal and immune functions, as identified in a global network of metabolic/protein\\u2013protein interactions. Subsequent mining of data from curated pathway, network, and genome-wide association studies identified genetic and functional relationships that may be explained by gene\\u2013nutrient interactions. The systems nutrition strategy described here has thus associated a multivariate metabolite pattern in blood with genes involved in immune and gastrointestinal functions. Electronic supplementary material The online version of \",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "39",
         "10.1007_s12559-020-09751-3",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12559-020-09751-3.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12559-020-09751-3.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.u1x8a5nr', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Abstract The coronavirus disease (COVID-19) caused by a novel coronavirus, SARS-CoV-2, has been declared a global pandemic. \\nDue to its infection rate and severity, it has emerged as one of the major global threats of the current generation.\\nTo support the current combat against the disease, this research aims to propose a machine learning\\u2013based pipeline to detect COVID-19 infection using lung computed tomography scan images (CTI).\\nThis implemented pipeline consists of a number of sub-procedures ranging from segmenting the COVID-19 infection to classifying the segmented regions.\\nThe initial part of the pipeline implements the segmentation of the COVID-19\\u2013affected CTI using social group optimization\\u2013based Kapur\\u2019s entropy thresholding, followed by k-means clustering and morphology-based segmentation.\\nThe next part of the pipeline implements feature extraction, selection, and fusion to classify the infection.\\nPrinciple component analysis\\u2013based serial fusion technique is used in fusing the features and the fused feature vector is then employed to train, test, and validate four different classifiers namely Random Forest, K-Nearest Neighbors (KNN), Support Vector Machine with Radial Basis Function, and Decision Tree.\\nExperimental results using benchmark datasets show a high accuracy (> 91%) for the morphology-based segmentation task; for the classification task, the KNN offers the highest accuracy among the compared classifiers (> 87%).\\nHowever, this should be noted that this method still awaits clinical validation, and therefore should not be used to clinically diagnose ongoing COVID-19 infection. \",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.U1X8A5NR.'], context: \\\"atients with non-small cell lung cancer.\\nRadiology.\\n2009;252(1):263\\u2013272. \\n67.\\nZhao B, Schwartz LH, Kris MG.\\nArchive TCI, editor.\\nData From RIDER Lung CT.\\nThe Cancer Imaging Archive; 2015.\\nLast access date: 17-06-2020.\\nAvailable from: https://doi.org/10.7937/K9/TCIA.2015.U1X8A5NR. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.procs.2018.08.208.'], context: \\\"ical datasets. \\nIn: Proceedings of ICACIN 2020.\\nSpringer; 2020.\\np. 69\\u2013 78. 33.\\nRabby G, Azad S, Mahmud M, Zamli KZ, Rah-man MM. \\nA flexible keyphrase extraction technique for academic literature.\\nProcedia Comput Sci.\\n2018;135:553\\u201363. https://doi.org/10.1016/j.procs.2018.08.208. \\n34.\\nSilver D, et al.\\nMastering the game of Go with deep neural networks and tree search.\\nNature.\\n2016;529(7587):484. \\n35.\\nBhandary A, Prabhu GA, Rajinikanth V, Thanaraj KP, Satapathy SC, Robbins DE, et al.\\nDeep-learning framework to detect lung abn\\\"}\"]}",
         "{\"title\":\"Social Group Optimization\\u2013Assisted Kapur\\u2019s Entropy and Morphological Segmentation for Automated Detection of COVID-19 Infection from Computed Tomography Images\",\"abstract\":\"The coronavirus disease (COVID-19) caused by a novel coronavirus, SARS-CoV-2, has been declared a global pandemic. Due to its infection rate and severity, it has emerged as one of the major global threats of the current generation. To support the current combat against the disease, this research aims to propose a machine learning\\u2013based pipeline to detect COVID-19 infection using lung computed tomography scan images (CTI). This implemented pipeline consists of a number of sub-procedures ranging from segmenting the COVID-19 infection to classifying the segmented regions. The initial part of the pipeline implements the segmentation of the COVID-19\\u2013affected CTI using social group optimization\\u2013based Kapur\\u2019s entropy thresholding, followed by k-means clustering and morphology-based segmentation. The next part of the pipeline implements feature extraction, selection, and fusion to classify the infection. Principle component analysis\\u2013based serial fusion technique is used in fusing the features and the fused feature vector is then employed to train, test, and validate four different classifiers namely Random Forest, K-Nearest Neighbors (KNN), Support Vector Machine with Radial Basis Function, and Decision Tree. Experimental results using benchmark datasets show a high accuracy (> 91%) for the morphology-based segmentation task; for the classification task, the KNN offers the highest accuracy among the compared classifiers (> 87%). However, this should be noted that this method still awaits clinical validation, and therefore should not be used to clinically diagnose ongoing COVID-19 infection.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00521-017-2955-2'], context: \\\"Morphological segmenting and neighborhood pixel-based locality preserving projection on brain fMRI dataset for semantic feature extraction: an affective computing study|Neural Comput Appl|10.1007/s00521-017-2955-2\\\"}\",\"{\\\"dataset_ids\\\": ['10.1007/s10278-013-9622-7'], context: \\\"The Cancer Imaging Archive (TCIA): maintaining and operating a public information repository|J Digit Imaging|10.1007/s10278-013-9622-7|23884657\\\"}\",\"{\\\"dataset_ids\\\": ['10.1118/1.3528204'], context: \\\"The Lung Image Database Consortium (LIDC) and Image Database Resource Initiative (IDRI): a completed reference database of lung nodules on CT scans: The LIDC/IDRI thoracic CT database of lung nodules|Med Phys|10.1118/1.3528204|21452728\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.U1X8A5NR.'], context: \\\"Zhao B, Schwartz LH, Kris MG. Archive TCI, editor. Data From RIDER_Lung CT. The Cancer Imaging Archive; 2015. Last access date: 17-06-2020. Available from: 10.7937/K9/TCIA.2015.U1X8A5NR.\\\"}\"]}"
        ],
        [
         "40",
         "10.1016_j.ast.2022.107401",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ast.2022.107401.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182.v1', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Aerospace Science and Technology 122 (2022) 107401 Contents lists available at ScienceDirect Aerospace Science and Technology www.elsevier.com/locate/aescte Short communication Numerical investigation of full helicopter with and without the ground  effect Paulo A.S.F. Silva a,b,\\u2217, Panagiotis Tsoutsanis a, Antonios F. Antoniadis a a Centre for Computational Engineering Sciences, Cranfield University, Cranfield MK43 0AL, United Kingdom b Federal Institute of Bras\\u00edlia, Bras\\u00edlia, Brazil a r t\\ni c l e  \\ni n f o a b s t r a c t Article history: \\nReceived 21 October 2021 Received in revised form 7 January 2022 Accepted 5 February 2022 Available online 15 February 2022 Communicated by Damiano Casalino Dataset link: https://doi .org /10 .17862\\n/cranfield .rd\\n.19146182 .v1 \\nKeywords: High-order methods Rotorcraft CFD IGE OGE Ground effect In the present work, the aerodynamic performance of the full helicopter PSP in hover flight is investigated  using a simplified concept of multiple reference frame (MRF) technique in the context of high-order  Monotone Upstream Centred Scheme for Conservation Laws (MUSCL) cell-centred finite volume method.  \\nThe predictions were obtained for two ground distances and several collective pitch angle at tip Mach  number of 0.585.\\nThe calculations were made for both out-of-ground-effect (OGE) and in-ground-effect  (IGE) cases and compared with experimental data in terms of pressure distribution and integrated thrust  and torque and vortex system. \\n\\u00a9 2022 The Author(s).\\nPublished by Elsevier Masson SAS.\\nThis is an open access article under the CC BY  license (http://creativecommons.org/licenses/by/4.0/). \\n1.\\nIntroduction The brownout phenomenon is often experienced in helicopters  in ground effect (IGE) when landing, take-off or hovering in dusty  environments, where the particles cloud obscure the pilot\\u2019s vision  of the terrain and consequently decreases substantially the safety.  \\nThis flight operation is usual on medical services on beaches and  army operations on desert environments.\\nUnderstanding the rotor  downwash wake is crucial to improve not only the safety of the  surrounding but also the rotor performance.\\nPredicting helicopter  aerodynamics and the surrounding flowfield challenges most Com-putational Fluid Dynamics (CFD) solvers as it requires a low nu-merical dissipation to accurate predict the blade forces and the  v\",\"data_availability\":\"Data availability \",\"other_dataset_citations\":[]}",
         "{\"title\":\"Numerical investigation of full helicopter with and without the ground effect\",\"abstract\":\"OGE Ground effect In the present work, the aerodynamic performance of the full helicopter PSP in hover flight is investigated using a simplified concept of multiple reference frame (MRF) technique in the context of high-order Monotone Upstream Centred Scheme for Conservation Laws (MUSCL) cell-centred finite volume method. The predictions were obtained for two ground distances and several collective pitch angle at tip Mach number of 0.585. The calculations were made for both out-of-ground-effect (OGE) and in-ground-effect (IGE) cases and compared with experimental data in terms of pressure distribution and integrated thrust and torque and vortex system.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "41",
         "10.1016_j.cpc.2024.109087",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.cpc.2024.109087.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.cpc.2024.109087.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/9gr6pxhfjm', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"H-wave \\u2013 A Python package for the Hartree-Fock approximation and the random phase approximation Tatsumi Aoyamaa,\\u2217, Kazuyoshi Yoshimia, Kota Idoa, Yuichi Motoyamaa, Taiki Kawamurab, Takahiro Misawaa, Takeo Katoa, Akito Kobayashib aInstitute for Solid State Physics, University of Tokyo, Chiba 277-8581, Japan bDepartment of Physics, Nagoya University, Nagoya 464-8602, Japan Abstract H-wave is an open-source software package for performing the Hartree\\u2013Fock approximation (HFA) and ran-dom phase approximation (RPA) for a wide range of Hamiltonians of interacting fermionic systems.\\nIn HFA calculations, H-wave examines the stability of several symmetry-broken phases, such as anti-ferromagnetic and charge-ordered phases, in the given Hamiltonians at zero and finite temperatures.\\nFurthermore, H-wave calculates the dynamical susceptibilities using RPA to examine the instability toward the symmetry-broken phases.\\nBy preparing a simple input file for specifying the Hamiltonians, users can perform HFA and RPA for standard Hamiltonians in condensed matter physics, such as the Hubbard model and its extensions. \\nAdditionally, users can use a Wannier90-like format to specify fermionic Hamiltonians.\\nA Wannier90 format is implemented in RESPACK to derive ab initio Hamiltonians for solids.\\nHFA and RPA for the ab initio Hamiltonians can be easily performed using H-wave.\\nIn this paper, we first explain the basis of HFA and RPA, and the basic usage of H-wave, including download and installation.\\nThereafter, the input file for-mats implemented in H-wave, including the Wannier90-like format for specifying the interacting fermionic Hamiltonians, are discussed.\\nFinally, we present several examples of \",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"H-wave -A Python package for the Hartree-Fock approximation and the random phase approximation\",\"abstract\":\"H-wave is an open-source software package for performing the Hartree-Fock approximation (HFA) and random phase approximation (RPA) for a wide range of Hamiltonians of interacting fermionic systems. In HFA calculations, H-wave examines the stability of several symmetry-broken phases, such as anti-ferromagnetic and charge-ordered phases, in the given Hamiltonians at zero and finite temperatures. Furthermore, H-wave calculates the dynamical susceptibilities using RPA to examine the instability toward the symmetry-broken phases. By preparing a simple input file for specifying the Hamiltonians, users can perform HFA and RPA for standard Hamiltonians in condensed matter physics, such as the Hubbard model and its extensions. Additionally, users can use a Wannier90-like format to specify fermionic Hamiltonians. A Wannier90 format is implemented in RESPACK to derive ab initio Hamiltonians for solids. HFA and RPA for the ab initio Hamiltonians can be easily performed using H-wave. In this paper, we first explain the basis of HFA and RPA, and the basic usage of H-wave, including download and installation. Thereafter, the input file formats implemented in H-wave, including the Wannier90-like format for specifying the interacting fermionic Hamiltonians, are discussed. Finally, we present several examples of H-wave such as zero-temperature HFA calculations for the extended Hubbard model on a square lattice, finite-temperature HFA calculations for the Hubbard model on a cubic lattice, and RPA in the extended Hubbard model on a square lattice.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "42",
         "10.1016_j.dib.2022.108797",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2022.108797.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2022.108797.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.1594/pangaea.949117', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Data in Brief 46 (2023) 108797  Contents lists available at ScienceDirect  Data in Brief  journal homepage: www.elsevier.com/locate/dib  Data Article  Dataset for the establishment of an age model  of marine sediment core KH19-6 Leg.4  PC10/MC14 collected from the Agulhas Ridge  in the South Atlantic Ocean  Kaoru Kubota a , b , \\u2217, Rosaaideihn Tanabe c , Minoru Ikehara c ,  Yukiko Kozaka c , Koji Seike d , Yosuke Miyairi e , Yusuke Yokoyama a , e  a Japan Agency for Marine-Earth Science and Technology, 2-15 Natsushima, Yokosuka, Kanagawa 237-0061, Japan  b Graduate School of Human Development and Environment, Kobe University 3-11 Tsurukabuto, Nada, Kobe, Hyogo  657-8501, Japan  c Center for Advanced Marine Core Research, Kochi University, 200 Monobu-otsu, Nankoku Kochi 783-8502, Japan  d National Institute of Advanced Industrial Science and Technology (AIST), AIST Tsukuba Central 7, 1-1-1 Higashi,  \\nTsukuba 305-8567, Japan  e Atmosphere and Ocean Research Institute, The University of Tokyo, 5-1-5 Kashiwa, Chiba 277-8564, Japan  a r t\\ni c l e  \\ni n f o  Article history:  Received 26 October 2022  Revised 24 November 2022  \\nAccepted 28 November 2022  Available online 5 December 2022  \\nDataset link: Dataset used to obtain an age  model of marine sediment core KH19-6 Leg  4 PC10/MC14 from the Agulhas Ridge in the  South Atlantic Ocean (Original data)  Keywords:  Rad\",\"data_availability\":\"Data Availability  \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1594/PANGAEA.949117'], context: \\\"[1] K. Kubota, R. Tanabe, M. Ikehara, Y. Kozaka, K. Seike, Y. Miyairi, Y. Yokoyama, Dataset used to obtain an age model  of marine sediment core KH19-6 Leg 4 PC10/MC14 from the Agulhas Ridge in the South Atlantic Ocean, PANGAEA  (2022), doi: 10.1594/PANGAEA.949117 .  \\n[2] D.A. Hodell , C.D. Charles , J.H. Curtis , P.G. Mortyn , U.S. Ninnemann , K.A. Venz , Data report: oxygen isotope stratigraphy  of ODP Leg 177 Sites 1088, 1089, 1090, 1093, and 1094, in: R. Gersonde, D.A. Hodell, P. Blum (Eds.), Proceedings \\\"}\"]}",
         "{\"title\":\"Dataset for the establishment of an age model of marine sediment core KH19-6 Leg.4 PC10/MC14 collected from the Agulhas Ridge in the South Atlantic Ocean\",\"abstract\":\"A precise age model of marine sediment core is crucial for environmental studies of the past such as paleoceanography, paleoclimatology, and paleo-hazard studies. Here the geochemical dataset is described that is used to determine the age model of marine sediment cores collected from Agulhas Ridge in the South Atlantic Ocean using piston coring and multiple-coring systems during the 30th Anniversary expeditions of R/V Hakuho Maru in 2019\\u20132020 (KH19-6 Leg.4 PC10/MC14, water depth of 4,604 m). The top 3.27 meter of 12.28-meter-long piston core (PC10) and a whole 0.29-m-long multiple core (MC14) were dated. The dataset includes radiocarbon ages of planktonic foraminifera shells and oxygen isotopes of both planktonic ( Globigerinoides bulloides, Globorotalia inflata ) and benthic ( Gyroidina soldanii ) foraminifera shells. The top 7.5 kyr record was lost, the ages of 3.27 m depth below sea floor was \\u223c140 kyr ago, and sedimentation rates were 0.9\\u20135.5 kyr/cm.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1594/PANGAEA.949117'], context: \\\"Dataset used to obtain an age model of marine sediment core KH19-6 Leg 4 PC10/MC14 from the Agulhas Ridge in the South Atlantic Ocean|PANGAEA|10.1594/PANGAEA.949117\\\"}\"]}"
        ],
        [
         "43",
         "10.1016_j.dib.2023.109949",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2023.109949.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8338991', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Data in Brief 52 (2024) 109949  Contents lists available at ScienceDirect  Data in Brief  journal homepage: www.elsevier.com/locate/dib  Data Article  Volcanic tephra deposition dataset based on  interpolated field measurements following the  2021 Tajogaite Eruption on La Palma, Canary  Islands, Spain  Christopher Shatto a , \\u2217, Frank Weiser a , Anna Walentowitz a ,  Reinhold Stahlmann a , Samip Shrestha b , Mar\\u00eda Guerrero-Campos c ,  F\\u00e9lix Manuel Medina d , Manuel Nogales e , Anke Jentsch f , g ,  Carl Beierkuhnlein\\na , g , h , i  a Department of Biogeography, University of Bayreuth, Universit\\u00e4tsstra\\u00dfe 30, 95447 Bayreuth, Germany  b German Aerospace Center (DLR), German Remote Sensing Data Center (DFD), Oberpfaffenhofen, 82234 Wessling,  Germany  c \\u00c1rea de Medio Ambiente, Gesti\\u00f3n y Planeamiento Territorial y Ambiental (Gesplan S. A.), Avenida 3 de mayo, 71,  Tenerife, Canary Islands, Spain  \\nd Consejer\\u00eda de Medio Ambiente, Cabildo Insular de La Palma, Avenida Los Indianos, 20, 38700 Santa Cruz de La  Palma, Canary Islands, Spain  e Island Ecology and Evolution Research Group, Institute of Natural Products and Agrobiology (IPNA-CSIC), Avenida  Astrof\\u00edsico Francisco S\\u00e1nchez 3, 38206 San Crist\\u00f3bal de La Laguna, Tenerife, Canary Isalnds, Spain  f Disturbance Ecology, University of Bayreuth, Universit\\u00e4tsstra\\u00dfe 30, 95447 Bayreuth, Germany  g Bayreuth Center for Ecology and Environmen\",\"data_availability\":\"Data Availability  \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00445-004-'], context: \\\"ve years later, Eastern Washington  University Press, 1986, pp.\\n108\\u2013111 .  \\n[10] P. Delmelle, F. Villi\\u00e9ras, M. Pelletier, Surface area, porosity and water adsorption properties of fine volcanic ash  particles, Bull.\\nVolcanol. 67 (2005) 160\\u2013169, doi: 10.1007/s00445-004- 0370- x .  \\n[11] R. Civico, T. Ricci, P. Scarlato, J. Taddeucci, D. Andronico, E. Del Bello, . . .\\nN.M P\\u00e9rez, High-resolution digital surface  model of the 2021 eruption deposit of Cumbre Vieja volcano, La Palma, Spain, Sci.\\nData 9 (1) (2022) 1\\u20137, d\\\"}\"]}",
         "{\"title\":\"Volcanic tephra deposition dataset based on interpolated field measurements following the 2021 Tajogaite Eruption on La Palma, Canary Islands, Spain\",\"abstract\":\"In 2021, the Tajogaite Volcano erupted along the western slope of the Cumbre Vieja on the island of La Palma, Canary Islands, Spain. Volcanic tephra blanketed a substantial proportion of the island. By our estimations, approximately 23,000,000 m 3  of pyroclastic ashes and more coarse-grained particles were deposited unto La Palma's land surface in addition to the lava flow. Five months following the initial eruption, we measured the depth of the new ash layer across the island. We combined this data with drone-based observations to compile a dataset comprising the point distribution of ash depth. A spatial interpolation was then performed using Inverse Distance Weighting (IDW) to estimate the ash depth across the island at a 2 m spatial resolution. The interpolation performed well, yielding a root mean squared error (RMSE) value of 0.34 and thus, the dataset offers immense reuse potential for spatial inquiries related to evolutionary traits, vegetation patterns, and vegetation response to disturbance on oceanic islands. In addition, the data can be used to test different spatial interpolation techniques in an effort to improve the accuracy achieved using IDW.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1038/s41597-022-01551-8'], context: \\\"High-resolution digital surface model of the 2021 eruption deposit of Cumbre Vieja volcano, La Palma, Spain|Sci. Data|10.1038/s41597-022-01551-8|35013360\\\"}\"]}"
        ],
        [
         "44",
         "10.1016_j.ecolind.2021.107934",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ecolind.2021.107934.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ecolind.2021.107934.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6073/pasta/be42bb841e696b7bcad9957aed33db5e', 'type': 'Secondary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Ecological Indicators 129 (2021) 107934 Available online 5 July 2021 1470-160X/Crown Copyright \\u00a9 2021 Published by Elsevier Ltd. This is an open access article under the Open Government License (OGL) (http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/). \\nOverwinter sea-ice characteristics important for Antarctic krill recruitment  in the southwest Atlantic  Devi Veytia a,*, Sophie Bestley a,b, So Kawaguchi b,c, Klaus M. Meiners b,c, Eugene J. Murphy d,  Alexander D. Fraser b, Kazuya Kusahara e, Noriaki Kimura f, Stuart Corney a  a Institute for Marine and Antarctic Studies, University of Tasmania, 20 Castray Esp., Hobart, Tasmania 7000,\\nAustralia  b Australian Antarctic Program Partnership, Institute for Marine and Antarctic Studies, University of Tasmania, Hobart, Tasmania 7004, Australia  c Australian Antarctic Division, Department of Agriculture, Water and the Environment, 203 Channel Highway, Kingston, Tasmania 7050, Australia  d British Antarctic Survey, High Cross, Madingley Road, Cambridge CB3 0ET, UK  e Japan Agency for Marine Earth Science and Technology (JAMSTEC), Yokohama, Kanagawa 236-0001, Japan  f Atmosphere and Ocean Research Institute, The University of Tokyo, 5-1-5 Kashiwanoha, Kashiwa 277-8568, Japan    A R T\\nI C L E  I N F O    Keywords:  Climate change  Sea ice  Antarctic krill  Euphausia superba  Overwinter survival  Recruitment  A B S T R A C T    Climate change alters the extent and structure of sea-ice environments, which affects how they function as a  habitat for polar species.\\nIdentifying sea-ice characteristics that serve as indicators of habitat quality will be  crucial to the monitoring and management of climate change impacts.\\nIn the Southern Ocean, Antarctic krill is a  key prey species and fishery target.\\nKrill larvae depend upon sea-ice habitats to survive the winter and recruit to  the population in spring.\\nExisting observations of sea-ice characteristics lack sufficient spatiotemporal coverage  to quantify which ones contribute to favourable overwintering habitat, leading to uncertainties in how current  and future changes in sea ice affect krill populations.\\nHere, we derive regional-scale indices of annual krill  recruitment spanning 35 years across the southwest Atlantic.\\nTo develop meaningful indicators of sea-ice habitat,  we selected variables from a high-resolution sea-ice model that are hypothesized as relevant for larval habitat  use.\\nThe resulting correlations between recruitment and sea-ice indicators vary by regio\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.6073/pasta/be42bb841e696b7bca'], context: \\\".bas.ac.  uk/project/krillbase/).\\nKRILLBASE length frequency data are available  from the British Antarctic Survey by request (contact: Simeon Hill,  sih@bas.ac.uk), and the Palmer LTER length frequency data are avail\\u00ad able  online  (https://doi.org/10.6073/pasta/be42bb841e696b7bca\\u00ad d9957aed33db5e).\\nOcean-sea ice model output are available from K.K.  upon request (kazuya.kusahara@gmail.com).\\nFor sea-ice velocity fields  from the Kimura dataset, contact N.K. (kimura_n@aori.u-tokyo.ac.jp).  \\nThe remotely-sensed surface velocity \\\"}\"]}",
         "{\"title\":\"Overwinter sea-ice characteristics important for Antarctic krill recruitment in the southwest Atlantic\",\"abstract\":\"Climate change alters the extent and structure of sea-ice environments, which affects how they function as a habitat for polar species. Identifying sea-ice characteristics that serve as indicators of habitat quality will be crucial to the monitoring and management of climate change impacts. In the Southern Ocean, Antarctic krill is a key prey species and fishery target. Krill larvae depend upon sea-ice habitats to survive the winter and recruit to the population in spring. Existing observations of sea-ice characteristics lack sufficient spatiotemporal coverage to quantify which ones contribute to favourable overwintering habitat, leading to uncertainties in how current and future changes in sea ice affect krill populations. Here, we derive regional-scale indices of annual krill recruitment spanning 35 years across the southwest Atlantic. To develop meaningful indicators of sea-ice habitat, we selected variables from a high-resolution sea-ice model that are hypothesized as relevant for larval habitat use. The resulting correlations between recruitment and sea-ice indicators vary by region and show remote connections to sea ice that correspond with established theories of larval transport. Through an improved representation of sea-ice habitat quality, as compared with using more traditional satellite-derived variables such as sea-ice extent and duration, we highlight plausible regions of overwintering habitat. Our findings improve current understanding of how krill are likely responding to changing sea ice and support emerging views that larval habitat use is complex. Furthermore, regional variation in larval dependence on sea ice may provide pockets of resilience to change for the broader krill population.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "45",
         "10.1016_j.fuel.2022.125768",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.fuel.2022.125768.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.11583/dtu.20555586', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.11583/dtu.20555586.v3', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"    General rights  Copyright and moral rights for the publications made accessible in the public portal are retained by the authors and/or other copyright  owners and it is a condition of accessing publications that users recognise and abide by the legal requirements associated with these rights.    \\n\\uf0b7\\nUsers may download and print one copy of any publication from the public portal for the purpose of private study or research.  \\n\\uf0b7\\nYou may not further distribute the material or use it for any profit-making activity or commercial gain  \\uf0b7\\nYou may freely distribute the URL identifying the publication in the public portal    If you believe that this document breaches copyright please contact us providing details, and we will remov\",\"data_availability\":\"Data availability \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.11583/DTU.20555586.v2.'], context: \\\"The experimental data used for the results is freely available in a repository at: https://doi.org/10.11583/DTU.20555586.v2. \\\"}\",\"{\\\"dataset_ids\\\": ['10.11583/DTU.20'], context: \\\"Dataset link: https://doi.org/10.11583/DTU.20 555586.v2 \\\"}\"]}",
         ""
        ],
        [
         "46",
         "10.1016_j.jlp.2022.104761",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.23642/usn.15134442', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.23642/usn.15134442.v1', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Journal of Loss Prevention in the Process Industries 77 (2022) 104761 Available online 22 March 2022 0950-4230/\\u00a9 2022 The Author(s).\\nPublished by Elsevier Ltd.\\nThis is an open access article under the CC BY license (http://creativecommons.org/licenses/by/4.0/). \\nNumerical study of premixed gas explosion in a 1-m channel partly filled  with 18650 cell-like cylinders with experiments  M. Henriksen a,*, K. Vaagsaether a, J. Lundberg a, S. Forseth b, D. Bjerketvedt a  a Faculty of Technology, Natural Sciences and Maritime Sciences, University of South-Eastern Norway, Kj\\u00f8lnes Ring 56, Porsgrunn, 3901, Norway  b Norwegian Defence Research Establishment (FFI), Oslo, Norway    A R T\\nI C L E  I N F O    Keywords:  Li-ion battery safety  Open-source software  OpenFOAM  Explosion simulation  Premixed combustion  A B S T R A C T    Abused and defective Li-ion cells can cause a catastrophic failure of a Li-ion battery (LIB), leading to severe fires  and explosions.\\nIn recent years, several numerical and experimental studies have been conducted on the ex\\u00ad plosion hazard related to the vented combustible gases from failed Li-ion cells.\\nExperimentally quantifying  fundamental properties for failing LIBs is essential for understanding safety issues; however, it can be costly,  time-consuming, and can be partly incomplete.\\nComputational fluid dynamic (CFD) simulations have been an  essential tool for studying the risk and consequences in the process industry.\\nIn this study, the prediction ac\\u00ad curacy of the open-source CFD combustion model/solver XiFoam was evaluated by comparing numerical sim\\u00ad ulations and experiments of premixed gas explosions in a 1-m explosion channel partly filled with 18650 cell-like  cylinders.\\nThe prediction accuracy was determined by calculating the mean geometric bias and variance for the  temporal pressure evolution, maximum pressure peak, positive impulse, spatial flame front velocity for two  different channel geometries, in addition to two gas compositions at several fuel-air equivalence ratios.\\nFrom this  method, the XiFoam model/solver gave an overall acceptable model performance for both geometries and gas  composition.    \\n1. Introduction  Over the last decade, several fires and explosions have been caused  by failing Li-ion batteries (LIBs).\\nFor example, in Texas, USA, in April  2017, a train car carrying discarded LIBs for recycling caused an ex\\u00ad plosion, which shattered windows 350 feet from the incident (Sophia  Beausoleil, 2017).\\nIn April 2019, a Li-ion battery energy storage system  (BESS) caught fire in Arizona, USA.\\nThr\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         ""
        ],
        [
         "47",
         "10.1016_j.jobe.2023.107105",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.jobe.2023.107105.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8014149', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.5281/zenodo.8014150', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"abstract\":\"Journal of Building Engineering 76 (2023) 107105 Available online 23 June 2023 2352-7102/\\u00a9 2023 The Author(s). \\nPublished by Elsevier Ltd. This is an open access article under the CC BY-NC-ND license (http://creativecommons.org/licenses/by-nc-nd/4.0/). \\nContents lists available at ScienceDirect Journal of Building Engineering journal homepage: www.elsevier.com/locate/jobe Full length article Deep learning for crack detection on masonry fa\\u00e7ades using limited data and transfer learning Stamos Katsigiannis a,\\u2217, Saleh Seyedzadeh b, Andrew Agapiou c, Naeem Ramzan d a Durham University, UK b University of Edinburgh, UK c University of Strathclyde, UK d University of the West of Scotland, UK A R T I C L E I N F O Keywords: Crack detection Brickwork masonry \\nDeep learning Convolutional neural networks Transfer learning Brickwork dataset A B S T R A C T Crack detection in masonry fa\\u00e7ades is a crucial task for ensuring the safety and longevity of buildings.\\nHowever, traditional methods are often time-consuming, expensive, and labour-intensive.\\nIn recent years, deep learning techniques have been applied to detect cracks in masonry images, but these models often require large amounts of annotated data to achieve high accuracy, which can be difficult to obtain.\\nIn this article, we propose a deep learning approach for crack detection on brickwork masonry fa\\u00e7ades using transfer learning with limited annotated data.\\nOur approach uses a pre-trained deep convolutional neural network model as a feature extractor, which is then optimised specifically for crack detection.\\nTo evaluate the effectiveness of our proposed method, we created and curated a dataset of 700 brickwork masonry fa\\u00e7ade images, an\",\"data_availability\":\"Data availability \",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1109/ICCV.2017.74.'], context: \\\"in: 2017 IEEE International Conference on Computer Vision, ICCV, 2017, pp.\\n618\\u2013626, http://dx.doi.org/10.1109/ICCV.2017.74. \\n[55] J. Deng, W. Dong, R. Socher, L.-J. Li, K. Li, L. Fei-Fei, Imagenet: A large-scale hierarchical image database, in: 2009 IEEE Conference on Computer Vision \\\"}\",\"{\\\"dataset_ids\\\": ['10.5281/zenodo.8014150.'], context: \\\"1 Dataset download link: https://doi.org/10.5281/zenodo.8014150. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.conbuildmat.2020.119397.'], context: \\\"http://dx.doi.org/10.1016/j.conbuildmat.2020.119397. \\n[32] M. Maguire, S. Dorafshan, R.J. Thomas, SDNET2018:\\nA concrete crack image dataset for machine learning applications, 2018, http://dx.doi.org/10.15142/\\\"}\",\"{\\\"dataset_ids\\\": ['10.17265/1934-7359/2022.03.004.'], context: \\\"Eng.\\nArchit. 16 (2022) 150\\u2013171, http://dx.doi.org/10.17265/1934-7359/2022.03.004. \\n[41] S. Katsigiannis, S. Seyedzadeh, A. Agapiou, N. Ramzan, Brickwork cracks dataset, 2023, http://dx.doi.org/10.5281/zenodo.8014150. \\n[42] L. Perez, J. Wang, The effectiveness of data augmentation in image classification using deep learning, 2017,\\\"},{\\\"dataset_ids\\\": ['10.5281/zenodo.8014150.'], context: \\\"Eng.\\nArchit. 16 (2022) 150\\u2013171, http://dx.doi.org/10.17265/1934-7359/2022.03.004. \\n[41] S. Katsigiannis, S. Seyedzadeh, A. Agapiou, N. Ramzan, Brickwork cracks dataset, 2023, http://dx.doi.org/10.5281/zenodo.8014150. \\n[42] L. Perez, J. Wang, The effectiveness of data augmentation in image classification using deep learning, 2017, arXiv preprint, arXiv:1712.04621. \\n[43] C. Shorten, T.M. Khoshgoftaar, A survey on image data augmentation for deep learning, J. Big D\\\"}\"]}",
         "{\"title\":\"Deep learning for crack detection on masonry fa\\u00e7ades using limited data and transfer learning\",\"abstract\":\"Crack detection in masonry fa\\u00e7ades is a crucial task for ensuring the safety and longevity of buildings. However, traditional methods are often time-consuming, expensive, and labourintensive. In recent years, deep learning techniques have been applied to detect cracks in masonry images, but these models often require large amounts of annotated data to achieve high accuracy, which can be difficult to obtain. In this article, we propose a deep learning approach for crack detection on brickwork masonry fa\\u00e7ades using transfer learning with limited annotated data. Our approach uses a pre-trained deep convolutional neural network model as a feature extractor, which is then optimised specifically for crack detection. To evaluate the effectiveness of our proposed method, we created and curated a dataset of 700 brickwork masonry fa\\u00e7ade images, and used 500 images for training, 100 for validation, and the remaining 100 images for testing. Results showed that our approach is very effective in detecting cracks, achieving an accuracy and F1-score of up to 100% when following end-to-end training of the neural network, thus being a promising solution for building inspection and maintenance, particularly in situations where annotated data is limited. Moreover, the transfer learning approach can be easily adapted to different types of masonry fa\\u00e7ades, making it a versatile tool for building inspection and maintenance.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.15142/T3TD19'], context: \\\"SDNET2018: A concrete crack image dataset for machine learning applications|10.15142/T3TD19\\\"}\",\"{\\\"dataset_ids\\\": ['10.5281/zenodo.8014150'], context: \\\"Brickwork cracks dataset|10.5281/zenodo.8014150\\\"}\",\"{\\\"dataset_ids\\\": ['10.1109/CVPR.2009.5206848'], context: \\\"Imagenet: A large-scale hierarchical image database|IEEE Conference on Computer Vision and Pattern Recognition|10.1109/CVPR.2009.5206848\\\"}\"]}"
        ],
        [
         "48",
         "10.1016_j.jsames.2022.103824",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jsames.2022.103824.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/f7k072r7', 'type': 'Primary'}]",
         null,
         null,
         "{\"title\":\"\",\"abstract\":\"Journal of South American Earth Sciences 116 (2022) 103824 Available online 27 April 2022 0895-9811/\\u00a9 2022 Elsevier Ltd.\\nAll rights reserved. \\nInventory of large landslides along the Central Western Andes (ca. 15\\u25e6\\u201320\\u25e6 S): Landslide distribution patterns and insights on controlling factors  Fabrizio Delgado a,b, Swann Zerathe b,*, St\\u00b4ephane Schwartz b, Bastien Mathieux b,  Carlos Benavente a,c  a Especialidad Ingenier\\u00eda Geol\\u00b4ogica, Facultad de Ciencias e Ingenier\\u00eda.\\nPontificia Universidad Cat\\u00b4olica del Per\\u00fa, Av.\\nUniversitaria 1801, San Miguel, Lima, 15088, Peru  b Univ.\\nGrenoble Alpes, Univ.\\nSavoie Mont Blanc, CNRS, IRD, IFSTTAR, ISTerre, 38000, Grenoble, France  c Instituto Geol\\u00b4ogico, Minero y Metal\\u00fargico INGEMMET, Av. Canad\\u00b4a, 1470, Lima, Peru    A R T I C L E  I N F O    Keywords:  Central Western Andes  Atacama desert  Large landslides  Inventory  Controlling factors  A B S T R A C T    The western flank of the Central Andes hosts some of the largest terrestrial landslides (v > km3), which mor\\u00ad phologies are particularly well-preserved due to low erosion rates related to the hyper-arid climate prevailing in  this region since the Miocene.\\nFirst-order questions are pending about the factors controlling the development  and the triggering of those large-scale slope failures.\\nPrevious studies provided some geomorphological analysis  and dating on individual study cases, but a regional-scale vision of landslide processes long the Central Western  Andes is missing.  \\nHere we report an original inventory of large landslides (areas from 0.1 to 180 km2) established along the  western flank of the Central Andes between latitudes ca.\\n15 and 20\\u25e6S, and from the Pacific coast to the Altiplano.  \\nBased on manual mapping (using satellite images analysis, Google Earth and DEMs analysis) and a compilation of  previous works, we inventoried more than a thousand large landslides in this region.\\nWe then statistically  explored the database according to the landslides typology, size, abundance and relation to geologic, tectonic  and climatic settings of the Central Western Andes in order to provide a first insight on their controlling factors.  \\nLandslide size-frequency distribution follows a power-law with an exponent of 2.31 \\u00b1 0.16 and a cut-off of 4.0 \\u00b1 1.9 km2 showing a strong contribution of the largest landslides to the cumulated landslide area.\\nWe r\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5066/F7K072R7'], context: \\\"a database described in detail below (Supplement, Table 1).\\nThis  database was further used for statistical analysis of landslides patterns.  \\nAll calculations were made using ArcGIS software and using the SRTM  DEM of 30 m of resolution (SRTM, DOI:/10.5066/F7K072R7).\\nThe  \\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/esurf-2018-20.'], context: \\\"  extendidos.   \\nMarc, O., Stumpf, A., Malet, J.P., Gosset, M., Uchida, T., Chiang, S.H., 2018.\\nTowards a  global database of rainfall-induced landslide inventories: first insights from past and  new events.\\nEarth Surf.\\nDyn.\\nDiscuss.\\nhttps://doi.org/10.5194/esurf-2018-20.  \\nMargirier, A., Audin, L., Carcaillet, J., Schwartz, S., 2015.\\nTectonic and climatic controls  on the Chuquibamba landslide (western Andes, southern Peru).\\nEarth Surf.\\nDynam.  \\nDiscuss. 2, 1129\\u20131153.  \\nMari\\u02dcno, J., Samaniego, P., Manrique, N., Vald\\\"}\"]}",
         ""
        ],
        [
         "49",
         "10.1016_j.molcel.2018.11.006",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.molcel.2018.11.006.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.molcel.2018.11.006.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/jb4jjxsbb7.1', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17632/xtb4mkvf8f.1', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"abstract\":\"Resource Interactome Rewiring Following Pharmacological Targeting of BET Bromodomains Graphical Abstract Highlights d\\nTreatment with JQ1 induces an extensive BET proteins interactome rewiring d\\nStructural and biophysical studies expand the target space for BET bromodomains \\nd Two distinct short linear motifs mediate BET ET domain interactions d BRD3 negatively regulates proliferation through Pol I and II mechanisms Authors Jean-Philippe Lambert, Sarah Picaud, Takao Fujisawa, ..., Michael D. Wilson, Panagis Filippakopoulos, Anne-Claude Gingras Correspondence\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['E-GEOD-44672'], context: \\\"ChIP-Seq Data Analysis For GFP-BRD3 ChIP-seq experiments, 50bp single-end reads were obtained.\\nFastq files of other U-2 OS ChIP-seq datasets were downloaded through ArrayExpress (E-GEOD-44672, Walz et al., 2014).\\nFastq files of TCOF1 ChIP-seq and the corresponding input in HeLa cells were obtained from GSE89420 (Calo et al., 2018).\\nRaw reads were first trimmed using Trimmomatic (v.0.36) with default parameters.\\nAs reads from E-GEOD-44672\\\"},{\\\"dataset_ids\\\": ['GSE89420'], context: \\\"ts, 50bp single-end reads were obtained.\\nFastq files of other U-2 OS ChIP-seq datasets were downloaded through ArrayExpress (E-GEOD-44672, Walz et al., 2014).\\nFastq files of TCOF1 ChIP-seq and the corresponding input in HeLa cells were obtained from GSE89420 (Calo et al., 2018).\\nRaw reads were first trimmed using Trimmomatic (v.0.36) with default parameters.\\nAs reads from E-GEOD-44672 are only 33bp in length they were not trimmed.\\nReads were then aligned to the human genome (hg19) using bwa (v.0.7.8, Li\\\"},{\\\"dataset_ids\\\": ['GSE69140'], context: \\\"orrelation heatmap was generated based on the Spearman cor-relations between each pairs of samples on the rDNA region with 1bp resolution.\\nPositions were no reads bind to in all samples were excluded. \\nSRA files of published datasets (BRD4 in mESCs: GSE69140; CHD4 in mESCs: GSE61188; BRD4 in mouse leukemic cells: GSE52279; BRD9 in mouse leukemic cells: GSE79360) were obtained from NCBI\\u2019s Gene Expression Omnibus using the SRA toolkit (v.2.9.0) and were mapped to the mouse reference genome (mm9) with bowt\\\"},{\\\"dataset_ids\\\": ['GSE61188'], context: \\\"nerated based on the Spearman cor-relations between each pairs of samples on the rDNA region with 1bp resolution.\\nPositions were no reads bind to in all samples were excluded. \\nSRA files of published datasets (BRD4 in mESCs: GSE69140; CHD4 in mESCs: GSE61188; BRD4 in mouse leukemic cells: GSE52279; BRD9 in mouse leukemic cells: GSE79360) were obtained from NCBI\\u2019s Gene Expression Omnibus using the SRA toolkit (v.2.9.0) and were mapped to the mouse reference genome (mm9) with bowtie2 (v.2.2.3.4.1) using d\\\"},{\\\"dataset_ids\\\": ['GSE52279'], context: \\\"ons between each pairs of samples on the rDNA region with 1bp resolution.\\nPositions were no reads bind to in all samples were excluded. \\nSRA files of published datasets (BRD4 in mESCs: GSE69140; CHD4 in mESCs: GSE61188; BRD4 in mouse leukemic cells: GSE52279; BRD9 in mouse leukemic cells: GSE79360) were obtained from NCBI\\u2019s Gene Expression Omnibus using the SRA toolkit (v.2.9.0) and were mapped to the mouse reference genome (mm9) with bowtie2 (v.2.2.3.4.1) using default parameters. \\\"},{\\\"dataset_ids\\\": ['GSE79360'], context: \\\" rDNA region with 1bp resolution.\\nPositions were no reads bind to in all samples were excluded. \\nSRA files of published datasets (BRD4 in mESCs: GSE69140; CHD4 in mESCs: GSE61188; BRD4 in mouse leukemic cells: GSE52279; BRD9 in mouse leukemic cells: GSE79360) were obtained from NCBI\\u2019s Gene Expression Omnibus using the SRA toolkit (v.2.9.0) and were mapped to the mouse reference genome (mm9) with bowtie2 (v.2.2.3.4.1) using default parameters. \\\"}\",\"{\\\"dataset_ids\\\": ['E-MTAB-5670'], context: \\\"NNE; 5NNF; 5NNG; 6G0P; 6G0O; 6G0Q; 6G0R; and 6G0S. Small-angle X-Ray scattering data and models reported in this paper are SASD: SASDCT2; SASDCS2; SASDCR2; SASDCU2 (https://www.sasbdb.org).\\nChIP-seq data reported in this paper were deposited with as E-MTAB-5670 at ArrayExpress (https://www.ebi.ac.uk/arrayexpress).\\nAll original source files (SPOT arrays, microscopy images and western blots) have been deposited to Mendeley Data and are available at https://data.mendeley.com/datasets/ under ascensions xtb4mkv\\\"}\"]}",
         "{\"title\":\"Interactome Rewiring Following Pharmacological Targeting of BET Bromodomains\",\"abstract\":\"Summary Targeting bromodomains (BRDs) of the bromo-and-extra-terminal (BET) family offers opportunities for therapeutic intervention in cancer and other diseases. Here, we profile the interactomes of BRD2, BRD3, BRD4, and BRDT following treatment with the pan-BET BRD inhibitor JQ1, revealing broad rewiring of the interaction landscape, with three distinct classes of behavior for the 603 unique interactors identified. A group of proteins associate in a JQ1-sensitive manner with BET BRDs through canonical and new binding modes, while two classes of extra-terminal (ET)-domain binding motifs mediate acetylation-independent interactions. Last, we identify an unexpected increase in several interactions following JQ1 treatment that define negative functions for BRD3 in the regulation of rRNA synthesis and potentially RNAPII-dependent gene expression that result in decreased cell proliferation. Together, our data highlight the contributions of BET protein modules to their interactomes allowing for a better understanding of pharmacological rewiring in response to JQ1.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 9,
        "rows": 524
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract Characterizin...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: The homolept...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: Metal-ligand...</td>\n",
       "      <td>{\"title\":\"Metal\\u2013Ligand Cooperativity of t...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Abstract: A new synthe...</td>\n",
       "      <td>{\"title\":\"A Synthetic Route Toward Tetrazoles:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>10.7554_elife.74937</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5281/zenod...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"  1  First-order visua...</td>\n",
       "      <td>{\"title\":\"First-order visual interneurons dist...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>10.7717_peerj.10452</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'PRJNA664798', 'type': 'Second...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Submitted 10 August 20...</td>\n",
       "      <td>{\"title\":\"Bacterial community diversity, ligno...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>10.7717_peerj.11352</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7291/d11m3...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Variation in purple se...</td>\n",
       "      <td>{\"title\":\"Variation in purple sea urchin ( Str...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>10.7717_peerj.12422</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.15468/dl.c...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Spatial and temporal d...</td>\n",
       "      <td>{\"title\":\"Spatial and temporal distribution pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>10.7717_peerj.13193</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.6073/pasta...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"abstract\":\"Submitted 21 June 2021...</td>\n",
       "      <td>{\"title\":\"Foliar nutrient concentrations of si...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>524 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "0      10.1002_2017jc013030   \n",
       "1    10.1002_anie.201916483   \n",
       "2    10.1002_anie.202005531   \n",
       "3    10.1002_anie.202007717   \n",
       "4    10.1002_chem.201902131   \n",
       "..                      ...   \n",
       "519     10.7554_elife.74937   \n",
       "520     10.7717_peerj.10452   \n",
       "521     10.7717_peerj.11352   \n",
       "522     10.7717_peerj.12422   \n",
       "523     10.7717_peerj.13193   \n",
       "\n",
       "                                         pdf_file_path  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "..                                                 ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                         xml_file_path dataset_type  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "..                                                 ...          ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                          dataset_info  \\\n",
       "0    [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "3       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "4       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "..                                                 ...   \n",
       "519  [{'dataset_id': 'https://doi.org/10.5281/zenod...   \n",
       "520  [{'dataset_id': 'PRJNA664798', 'type': 'Second...   \n",
       "521  [{'dataset_id': 'https://doi.org/10.7291/d11m3...   \n",
       "522  [{'dataset_id': 'https://doi.org/10.15468/dl.c...   \n",
       "523  [{'dataset_id': 'https://doi.org/10.6073/pasta...   \n",
       "\n",
       "                                        xml_first_line xml_type  \\\n",
       "0    <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2    <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "3    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "4    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "..                                                 ...      ...   \n",
       "519  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "520  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "521  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "522  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "523  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "\n",
       "                                              pdf_text  \\\n",
       "0    {\"title\":\"\",\"abstract\":\"Abstract Characterizin...   \n",
       "1    {\"title\":\"\",\"abstract\":\"Abstract: The homolept...   \n",
       "2    {\"title\":\"\",\"abstract\":\"Abstract: Trifluoromet...   \n",
       "3    {\"title\":\"\",\"abstract\":\"Abstract: Metal-ligand...   \n",
       "4    {\"title\":\"\",\"abstract\":\"Abstract: A new synthe...   \n",
       "..                                                 ...   \n",
       "519  {\"title\":\"\",\"abstract\":\"  1  First-order visua...   \n",
       "520  {\"title\":\"\",\"abstract\":\"Submitted 10 August 20...   \n",
       "521  {\"title\":\"\",\"abstract\":\"Variation in purple se...   \n",
       "522  {\"title\":\"\",\"abstract\":\"Spatial and temporal d...   \n",
       "523  {\"title\":\"\",\"abstract\":\"Submitted 21 June 2021...   \n",
       "\n",
       "                                              xml_text  \n",
       "0    {\"title\":\"Assessing the variability in the rel...  \n",
       "1    {\"title\":\"Effective and Reversible Carbon Diox...  \n",
       "2    {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...  \n",
       "3    {\"title\":\"Metal\\u2013Ligand Cooperativity of t...  \n",
       "4    {\"title\":\"A Synthetic Route Toward Tetrazoles:...  \n",
       "..                                                 ...  \n",
       "519  {\"title\":\"First-order visual interneurons dist...  \n",
       "520  {\"title\":\"Bacterial community diversity, ligno...  \n",
       "521  {\"title\":\"Variation in purple sea urchin ( Str...  \n",
       "522  {\"title\":\"Spatial and temporal distribution pa...  \n",
       "523  {\"title\":\"Foliar nutrient concentrations of si...  \n",
       "\n",
       "[524 rows x 9 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_file_paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "id": "2640c46b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- QwenModelEval Class ---\n",
    "# kagglehub.model_download(\"qwen-lm/qwen-3/transformers/0.6b\")\n",
    "#max_new_tokens=32768\n",
    "class QwenModelEval:\n",
    "    def __init__(self, model_name, sys_prompt, enable_thinking=True, max_new_tokens=1024):\n",
    "        print(f\"Loading Qwen model and tokenizer from: {model_name}\")\n",
    "        self.model_name = model_name\n",
    "        self.sys_prompt = sys_prompt\n",
    "        self.enable_thinking = enable_thinking  # Enable or disable thinking mode\n",
    "        self.max_new_tokens = max_new_tokens  # Set the maximum number of new tokens to generate\n",
    "        # Load the tokenizer and model\n",
    "        # Using trust_remote_code=True to allow custom model code execution\n",
    "        self.tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)\n",
    "        self.model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=\"auto\", device_map=\"auto\", trust_remote_code=True)\n",
    "        self.model.eval() # Set the model to evaluation mode here.\n",
    "\n",
    "    def generate_response(self, user_input):  \n",
    "        inputs = self._get_inputs(user_input)\n",
    "        # Disable gradient calculation during inference\n",
    "        # Generate the response using the model\n",
    "        with torch.no_grad(): \n",
    "            generated_ids = self.model.generate(\n",
    "                **inputs,\n",
    "                max_new_tokens=self.max_new_tokens,\n",
    "                pad_token_id=self.tokenizer.eos_token_id,\n",
    "                eos_token_id=self.tokenizer.convert_tokens_to_ids(\"<|im_end|>\"),\n",
    "                # do_sample=False, # Use greedy decoding (fastest)\n",
    "                # num_beams=1,     # Do not use beam search (fastest)\n",
    "                # temperature=0.0, # Make output deterministic (if do_sample=False, this has no effect)                \n",
    "                temperature=0.6 if self.enable_thinking else 0.7,\n",
    "                top_p=0.95 if self.enable_thinking else 0.8,\n",
    "                top_k=20,\n",
    "                min_p=0\n",
    "            )\n",
    "        # Parse the response and thinking content\n",
    "        return self._parse_response(inputs, generated_ids)\n",
    "\n",
    "    def _get_inputs(self, user_input):\n",
    "        \"\"\"Prepare the input for the model based on user input.\"\"\"\n",
    "        # Trim the user input to a maximum length for better performance\n",
    "        user_input = user_input[:4096]  # Limit input length to 4096 characters\n",
    "        print(f\"Preparing input with length: {len(user_input)}\")\n",
    "        # Create the messages for the chat template\n",
    "        messages = [\n",
    "            {\"role\": \"system\", \"content\": self.sys_prompt},\n",
    "            {\"role\": \"user\", \"content\": user_input}\n",
    "        ]\n",
    "        text = self.tokenizer.apply_chat_template(\n",
    "            messages,\n",
    "            tokenize=False,\n",
    "            add_generation_prompt=True,\n",
    "            enable_thinking=self.enable_thinking\n",
    "        )\n",
    "        return self.tokenizer(text, return_tensors=\"pt\").to(self.model.device)\n",
    "    \n",
    "    def _parse_response(self, inputs, generated_ids):\n",
    "        print(\"Parsing response from generated IDs...\")\n",
    "        # Extract the output IDs from the generated IDs\n",
    "        output_ids = generated_ids[0][len(inputs.input_ids[0]):].tolist()\n",
    "        try:\n",
    "            index = len(output_ids) - output_ids[::-1].index(151668)\n",
    "        except ValueError:\n",
    "            index = 0\n",
    "\n",
    "        thinking_content = self.tokenizer.decode(output_ids[:index], skip_special_tokens=True).strip(\"\\n\")\n",
    "        raw_response = self.tokenizer.decode(output_ids[index:], skip_special_tokens=True).strip(\"\\n\")\n",
    "        response = self._parse_json(raw_response)\n",
    "        return response, thinking_content\n",
    "    \n",
    "    def _parse_json(self, raw_response: str) -> list[dict[str,str]]:\n",
    "        # Remove code block markers and leading/trailing whitespace\n",
    "        cleaned = raw_response.strip()\n",
    "        if cleaned.startswith(\"```json\"):\n",
    "            cleaned = cleaned[len(\"```json\"):].strip()\n",
    "        if cleaned.endswith(\"```\"):\n",
    "            cleaned = cleaned[:-3].strip()\n",
    "\n",
    "        # Now parse as JSON\n",
    "        try:\n",
    "            return json.loads(cleaned)\n",
    "        except json.JSONDecodeError as e:\n",
    "            return []        \n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "id": "74e5a664",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the one-shot reasoning, task, and example prompt\n",
    "# This prompt is designed to guide the model through a structured reasoning process\n",
    "\n",
    "# reason_prompt =  '''\n",
    "# You are an advanced AI reasoning assistant tasked with delivering a comprehensive analysis of a specific problem or question.  Your goal is to outline your reasoning process in a structured and transparent manner, with each step reflecting a thorough examination of the issue at hand, culminating in a well-reasoned conclusion.\n",
    "\n",
    "# ### Key Instructions:\n",
    "# 1.  Conduct **at least 5 distinct reasoning steps**, each building on the previous one.\n",
    "# 2.  **Acknowledge the limitations** inherent to AI, specifically what you can accurately assess and what you may struggle with.\n",
    "# 3.  **Adopt multiple reasoning frameworks** to resolve the problem or derive conclusions, such as:\n",
    "# - **Deductive reasoning** (drawing specific conclusions from general principles)\n",
    "# - **Inductive reasoning** (deriving broader generalizations from specific observations)\n",
    "# - **Abductive reasoning** (choosing the best possible explanation for the given evidence)\n",
    "# - **Analogical reasoning** (solving problems through comparisons and analogies)\n",
    "# 4.  **Critically analyze your reasoning** to identify potential flaws, biases, or gaps in logic.\n",
    "# 5.  When reviewing, apply a **fundamentally different perspective or approach** to enhance your analysis.\n",
    "# 6.  **Employ at least 2 distinct reasoning methods** to derive or verify the accuracy of your conclusions.\n",
    "# 7.  **Incorporate relevant domain knowledge** and **best practices** where applicable, ensuring your reasoning aligns with established standards.\n",
    "# 8.  **Quantify certainty levels** for each step and your final conclusion, where applicable.\n",
    "# 9.  Consider potential **edge cases or exceptions** that could impact the outcome of your reasoning.\n",
    "# 10.  Provide **clear justifications** for dismissing alternative hypotheses or solutions that arise during your analysis.\n",
    "# '''\n",
    "\n",
    "reason_prompt =  '''\n",
    "You are an advanced AI research assistant that is skilled in identifying and classifying datasets used within academic research papers.\n",
    "Be as accurate as possible but don't over think it.\n",
    "'''\n",
    "\n",
    "task_prompt = '''\n",
    "You are given an article_id and the associated text of an academic research paper.\n",
    "Within the text of the paper, you are given an abstract and a list of potential_dataset_ids that contains dataset_ids and their associated context within the paper.\n",
    "You have 3 tasks:\n",
    "\n",
    "1. Your first task is to identify all citations of datasets used in the research for this article. An article may cite zero or many datasets.\n",
    "Datasets in an article can be cited within the context using various terms such as \"data release\", \"data availability\", \"dataset\", \"database\", \"repository\", \"data source\", \"data access\", \"data archive\".\n",
    "Each dataset has a unique, persistent identifier to represent it called a dataset_id. If you find more than one citation of the same dataset_id, only process the first one.\n",
    "\n",
    "There are 2 ways to identify a dataset_id:\n",
    "The first way to identify a dataset_id is via a Digital Object Identifier (DOI). DOIs are used for all papers and some datasets. We want to identify DOIs that are used as dataset_id's, not the DOI of the paper itself or any other papers.\n",
    "They take the following form: https://doi.org/[prefix]/[suffix]. The prefix always starts with \"10.\" and is followed by a 4 or 5 digit number. The suffix can contain letters, numbers, and special characters.\n",
    "Examples of DOI dataset_ids (just for reference):\n",
    "https://doi.org/10.12345/12345\n",
    "https://doi.org/10.1234/xxxx.1x1x-xx11\n",
    "\n",
    "The second way to identify a dataset_id is via an Accession ID. Accession ID's vary in form by individual data repository where the data live. \n",
    "Examples of Accession ID's (just for reference):\n",
    "\"EPI_ISL_10271777\" (EPI dataset)\n",
    "\"IPR000264\" (InterPro dataset)\n",
    "\"SAMN07159041\" (NCBI Sequence Read Archive dataset)\n",
    "\"CHEMBL1782574\" (ChEMBL dataset)\n",
    "\n",
    "\n",
    "2. Your second task is to classify the type of each dataset_id that you find as \"Primary\" or \"Secondary\" as it is used within the context of the paper.\n",
    "Primary - raw or processed data generated as part of this paper, specifically for this study\n",
    "Secondary - raw or processed data derived or reused from existing records or published data\n",
    "\n",
    "3. Your third task is to return your results in a JSON format.\n",
    "If an article does not refer to any dataset_id's, return a single JSON object with the following structure:\n",
    "```json\n",
    "[\n",
    "    {\n",
    "        \"dataset_id\": \"Missing\",\n",
    "        \"type\": \"Missing\"\n",
    "    }\n",
    "]\n",
    "If an article refers to one or more dataset_id's, you need to classify the type of each dataset as \"Primary\" or \"Secondary\" and\n",
    "return every dataset found in a JSON array of objects, where each object has the following structure:\n",
    "```json\n",
    "[\n",
    "    {\n",
    "        \"dataset_id\": dataset_id here,\n",
    "        \"type\": type here\n",
    "    },\n",
    "    ...\n",
    "]\n",
    "'''\n",
    "\n",
    "SYS_PROMPT = reason_prompt + task_prompt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "a43a2780",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "08c2451a-0774-4674-8339-871085f8d6c1",
       "rows": [
        [
         "52",
         "10.1017_rdc.2022.19",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_rdc.2022.19.pdf",
         null,
         "train",
         "[{'dataset_id': 'https://doi.org/10.11588/data/10100', 'type': 'Secondary'}]"
        ],
        [
         "53",
         "10.1017_s0007123423000601",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_s0007123423000601.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1017_s0007123423000601.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7910/dvn/fwckse', 'type': 'Primary'}]"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 2
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>10.1017_rdc.2022.19</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.11588/data...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>10.1017_s0007123423000601</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7910/dvn/f...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   article_id  \\\n",
       "52        10.1017_rdc.2022.19   \n",
       "53  10.1017_s0007123423000601   \n",
       "\n",
       "                                        pdf_file_path  \\\n",
       "52  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "53  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \\\n",
       "52                                                NaN        train   \n",
       "53  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                         dataset_info  \n",
       "52  [{'dataset_id': 'https://doi.org/10.11588/data...  \n",
       "53  [{'dataset_id': 'https://doi.org/10.7910/dvn/f...  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# train_file_paths_df['text'] = train_file_paths_df['pdf_file_path'].apply(load_article_text)\n",
    "# test_file_paths_df['text'] = test_file_paths_df['pdf_file_path'].apply(load_article_text)\n",
    "\n",
    "#Create a new df from train_file_paths_df where the string representation of dataset_info does not contain 'Missing'.\n",
    "train_file_paths_df_2 = train_file_paths_df[~train_file_paths_df['dataset_info'].astype(str).str.contains('Missing')]\n",
    "train_file_paths_df_3 = train_file_paths_df_2[train_file_paths_df_2['article_id'].astype(str).str.contains('10.1017_')]\n",
    "train_file_paths_df_3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "id": "f8a8a117",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Qwen model and tokenizer from: C:\\Users\\jim\\.cache\\kagglehub\\models\\qwen-lm\\qwen-3\\transformers\\0.6b\\1\n"
     ]
    }
   ],
   "source": [
    "# Instantiate the QwenModelEval class with the model path and system prompt\n",
    "inference_model = QwenModelEval(QWEN_BASE_MODEL_PATH, sys_prompt=SYS_PROMPT, enable_thinking=True, max_new_tokens=1576)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "749ca3d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_articles(file_paths_df: pd.DataFrame, model) -> pd.DataFrame:\n",
    "    results = []\n",
    "    for i, row in file_paths_df.iterrows():\n",
    "        article_id = row['article_id']\n",
    "        pdf_file_path = row['pdf_file_path']\n",
    "        xml_file_path = row['xml_file_path']\n",
    "\n",
    "        print(f\"Processing article {i}/{len(file_paths_df)}: {article_id}\")\n",
    "\n",
    "        # Load the text content from the PDF or XML file\n",
    "        text_content = load_article_text(pdf_file_path) if pdf_file_path else load_article_text(xml_file_path)\n",
    "\n",
    "        # Prepare the user input for the model\n",
    "        user_input = f\"Article ID: {article_id}\\nText Content: {text_content}\\n\"\n",
    "\n",
    "        # Generate response from the model\n",
    "        response, thinking_content = model.generate_response(user_input)\n",
    "\n",
    "        results.append({\n",
    "            'article_id': article_id,\n",
    "            'llm_input': user_input,\n",
    "            'llm_response': response,\n",
    "            'llm_thinking_content': thinking_content\n",
    "        })\n",
    "\n",
    "    return pd.DataFrame(results).sort_values(by=[\"article_id\"]).reset_index(drop=True)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "4d2524bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing article 52/2: 10.1017_rdc.2022.19\n",
      "Total blocks with data related dataset IDs found: 1\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_rdc.2022.19.pdf. Length: 1934 characters\n",
      "Preparing input with length: 1982\n",
      "Parsing response from generated IDs...\n",
      "Processing article 53/2: 10.1017_s0007123423000601\n",
      "Total blocks with data related dataset IDs found: 1\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_s0007123423000601.pdf. Length: 1248 characters\n",
      "Preparing input with length: 1302\n",
      "Parsing response from generated IDs...\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_input",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_response",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "llm_thinking_content",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "056093f4-8300-4674-8aee-aadd2246cb29",
       "rows": [
        [
         "0",
         "10.1017_rdc.2022.19",
         "Article ID: 10.1017_rdc.2022.19\nText Content: ABSTRACT.\nIn the frame of the IAEA-CRP (Coordinated Research Projects): Enhancing Nuclear Analytical Techniques to Meet the Needs of Forensic Sciences, an intercomparison exercise was organized between three AMS laboratories.\nAim of the program is to promote the use of nuclear and accelerator-based techniques in routine forensics practice.\nIn this view, one of the key points is the assessment of the precision and accuracy levels achievable on material of forensic interest.\nWe review the general structure and status of the project, with emphasis on results obtained in the analysis of wines of different grape varieties and grounded coffee beans from different locations such as Brazil, Spain, and Italy.\nThe three laboratories processed the samples according to different chemical protocols and performed the 14C measurements using different systems: MICADAS in Zurich and Debrecen and a HVEE 4130HC 3 MV Tandetron in Lecce.\nWithin the quoted uncertainty, the results showed good reproducibility, indicating that uncertainty level of the order of 0.3% are achieved by AMS on a single sample while multiple sample analyses results in precision down to 0.10.2%.\nThe measured 14C concentrations on coffee and wine samples resulted to be consistent with atmospheric 14C levels in the growing years. .\n potential_dataset_ids: [{\"dataset_ids\": ['https://doi.org/10.11588/data/10100'], context: \" L, Fedi M, Friedrich R, Maspero F, Sava T. 2019.\nradiocarbon dating and the protection of cultural heritage.\nRadiocarbon 61(5):11331134. \nHammer S, Levin I. 2017. \nMonthly mean atmospheric D14CO2 at Jungfraujoch and Schauinsland from 1986 to 2016.\nhttps://doi.org/10.11588/data/10100 heiDATA: Heidelberg Research Data Repository [Distributor] V2 [Version]. \nHandlos P, Svetlik I, Horkov L, Fejgl M, Kotik L, Brychov V, Megisova N, Marecov K. 2018. \nBomb peak: radiocarbon dating of skeletal remains in routine forensic medical p\"},]\n",
         "[{'dataset_id': 'https://doi.org/10.11588/data/10100', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem step by step. The user provided an article with an ID and a text abstract, along with a list of potential dataset_ids and their context. The goal is to identify all citations of datasets, classify their type, and return the results in JSON.\n\nFirst, I need to look through the text of the paper to find any mentions of datasets. The text mentions three laboratories processing samples using different chemical protocols and measuring 14C. The DOI provided is \"https://doi.org/10.11588/data/10100\". \n\nNext, I need to check if this DOI matches the format for a dataset ID. The DOI format is 10.1... with a prefix of 10. So the example given in the problem has \"https://doi.org/10.11588/data/10100\". The DOI here starts with \"10.\" followed by a 4 or 5-digit number. The suffix part includes \"data\" and \"10100\". The problem mentions that the dataset_id is derived from the DOI. \n\nNow, looking at the potential_dataset_ids, the first one is the DOI. The context says that this is a dataset related to radiocarbon dating and cultural heritage. The other entries in the potential_dataset_ids are Accession IDs like \"EPI_ISL_10271777\" and \"IPR000264\", but those seem to be from different repositories. The user's text mentions that the three laboratories used 14C measurements. \n\nWait, the problem's text says that the three laboratories processed samples using different systems. The 14C measurements are part of the analysis. The DOI in the text is likely a dataset related to the radiocarbon dating. The other Accession IDs might be from other datasets, but the user's context includes the DOI as part of the dataset. \n\nSo, the dataset_id here is \"https://doi.org/10.11588/data/10100\". The type is \"Primary\" because it's a dataset used in the analysis, not just raw data. The other Accession IDs in the potential_dataset_ids are probably not relevant here. \n\nTherefore, the JSON should include this dataset_id and type as Primary. Since there's only one dataset found, the JSON array will have that object. The user's example response shows a similar structure, so I need to follow that format.\n</think>"
        ],
        [
         "1",
         "10.1017_s0007123423000601",
         "Article ID: 10.1017_s0007123423000601\nText Content: Abstract Like much of the European centre-left, Britains Labour Party has struggled to appeal to its former core working class support base in recent years.\nHowever, this is largely a failure to connect with the white working class (WWC) specifically, whereas support among ethnic minorities remains robust.\nWe hypothesise that Labour could be experiencing a trade-off, whereby efforts to cater to minorities harm its perceived ability to represent WWC interests.\nWe test this thesis by examining whether WWC voters are more likely to view minority and working class representation in zero-sum terms and shun Labour when they associate the party with minority interests.\nWe show that the WWC are somewhat less likely to view working class and ethnic minority representation as strongly correlated, and Labours perceived ability to represent minorities is negatively associated with WWC support.\nThis is not (primarily) about ethnocentrism.\nInstead, we suggest that relative political deprivation is crucial. .\n potential_dataset_ids: [{\"dataset_ids\": ['https://doi.org/10.7910/DVN/FWCKSE'], context: \"Data availability statement.\nReplication data for this article can be found in Harvard Dataverse at: https://doi.org/10.7910/DVN/FWCKSE \"},]\n",
         "[{'dataset_id': 'https://doi.org/10.7910/DVN/FWCKSE', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this step by step. The user provided an article with an ID and a text content. The potential_dataset_ids include a DOI and an Accession ID. First, I need to check if the DOI in the text is a dataset_id. The DOI given is https://doi.org/10.7910/DVN/FWCKSE. The format starts with 10. followed by a number, which matches the structure of a DOI. So that's a valid dataset_id. \n\nNext, the Accession ID mentioned is \"SAMN07159041\" and \"IPR000264\". The user's potential_dataset_ids list has one of these, but I need to check if they are present in the text. The text mentions \"data availability statement\" and \"Harvard Dataverse\" but doesn't directly mention the Accession ID. However, the example given in the potential_dataset_ids includes \"SAMN07159041\" as an Accession ID. Since the text does not explicitly state the Accession ID, but the example includes it, I should consider that as a possible dataset_id. But the user's potential_dataset_ids list has only one entry, so maybe the Accession ID is included there. However, the user's input might have a mistake in the potential_dataset_ids list. Let me double-check.\n\nThe user's potential_dataset_ids list shows ['https://doi.org/10.7910/DVN/FWCKSE', 'SAMN07159041'] as the dataset_ids. The context for the first one is \"Data availability statement\" and \"Harvard Dataverse\". The second one's context is not provided, but the user's example includes it. However, the user's actual input might have the Accession ID in the potential_dataset_ids. Since the user's text doesn't mention the Accession ID, but the example does, perhaps the correct dataset_id is the DOI. But the user's potential_dataset_ids list includes both. However, the user's actual input might have the Accession ID in the potential_dataset_ids list. Wait, looking back, the user's potential_dataset_ids list has 'SAMN07159041' as an Accession ID. The context for that is not provided, but the example given in the potential_dataset_ids includes it. So, perhaps the correct dataset_id is the Accession ID. But since the user's text doesn't mention it, maybe the correct answer is to include both? Wait, the user's potential_dataset_ids list has both entries. The first is a DOI, the second is an Accession ID. But the user's text mentions the DOI and the Accession ID in the context. However, the user's potential_dataset_ids list includes only one entry. Wait, no, the user's potential_dataset_ids list shows both entries. But maybe the Accession ID is not present in the text. Let me check again.\n\nThe text content mentions \"data availability statement\" and \"Harvard Dataverse\" but does not mention the Accession ID. However, the potential_dataset_ids list includes \"SAMN07159041\" as an Accession ID. So, even though the text doesn't mention it, the user's potential_dataset_ids list includes it. Therefore, I should process both entries. But the user's potential_dataset_ids list has both. So, in the JSON, both entries would be included, with their respective types. However, the user's potential_dataset_ids list has only one entry. Wait, no, looking back, the user's potential_dataset_ids list is:\n\npotential_dataset_ids: [{\"dataset_ids\": ['https://doi.org/10.7910/DVN/FWCKSE'], context: \"Data availability statement. Replication data for this article can be found in Harvard Dataverse at: https://doi.org/10.7910/DVN/FWCKSE \"},]\n\nand the other entry is SAMN07159041. Wait, no, the user's potential_dataset_ids list is only one entry. Wait, the user's input shows the potential_dataset_ids as:\n\npotential_dataset_ids: [{\"dataset_ids\": ['https://doi.org/10.7910/DVN/FWCKSE'], context: \"Data availability statement. Replication data for this article can be found in Harvard Dataverse at: https://doi.org/10.7910/DVN/FWCKSE \"},]\n\nand then maybe another entry? Wait, no, looking again, the user's potential_dataset_ids list is only the first entry. So, perhaps the Accession ID is not present in the text. Therefore, the only dataset_id found is the DOI. Therefore, the JSON should include that dataset_id and its type. However, the user's potential_dataset_ids list includes the Accession ID, but the text doesn't mention it. So, perhaps the correct answer is only the DOI. But the user's potential_dataset_ids list includes both entries. This is a bit confusing. Alternatively, maybe the Accession ID is part of the dataset_ids provided, but the text doesn't mention it. Therefore, the correct answer is to process only the DOI. But since the user's potential_dataset_ids list includes both, perhaps I should check again. Let me confirm once more.\n\nThe user's potential_dataset_ids list is:\n\npotential_dataset_ids: [{\"dataset_ids\": ['https://doi.org/10.7910/DVN/FWCKSE'], context: \"Data availability statement. Replication data for this article can be found in Harvard Dataverse at: https://doi.org/10.7910/DVN/FWCKSE \"},]\n\nand then maybe another entry? Wait, no, the user's input shows only the first entry. Therefore, only the DOI is present. So, the dataset_id is the DOI, and the type is \"Primary\" since it's a dataset for data availability. The Accession ID might not be present, so the JSON should include only that. Therefore, the final answer is a JSON with the dataset_id and type.\n</think>"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 2
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>llm_input</th>\n",
       "      <th>llm_response</th>\n",
       "      <th>llm_thinking_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1017_rdc.2022.19</td>\n",
       "      <td>Article ID: 10.1017_rdc.2022.19\\nText Content:...</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.11588/data...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem step ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1017_s0007123423000601</td>\n",
       "      <td>Article ID: 10.1017_s0007123423000601\\nText Co...</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7910/DVN/F...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this step by step....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  article_id  \\\n",
       "0        10.1017_rdc.2022.19   \n",
       "1  10.1017_s0007123423000601   \n",
       "\n",
       "                                           llm_input  \\\n",
       "0  Article ID: 10.1017_rdc.2022.19\\nText Content:...   \n",
       "1  Article ID: 10.1017_s0007123423000601\\nText Co...   \n",
       "\n",
       "                                        llm_response  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.11588/data...   \n",
       "1  [{'dataset_id': 'https://doi.org/10.7910/DVN/F...   \n",
       "\n",
       "                                llm_thinking_content  \n",
       "0  <think>\\nOkay, let's tackle this problem step ...  \n",
       "1  <think>\\nOkay, let's tackle this step by step....  "
      ]
     },
     "execution_count": 399,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "processed_articles_df = process_articles(train_file_paths_df_3, inference_model)\n",
    "processed_articles_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "e9bb547b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save processed_articles_df to CSV\n",
    "processed_articles_df.to_csv(\"processed_articles.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "c263cb88",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'dataset_id': 'https://doi.org/10.11588/data/10100', 'type': 'Primary'}]"
      ]
     },
     "execution_count": 401,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import json\n",
    "#response_text = processed_articles_df[processed_articles_df['article_id']=='10.1017_rdc.2022.19']['llm_response']\n",
    "response_text = processed_articles_df.loc[processed_articles_df['article_id'] == '10.1017_rdc.2022.19', 'llm_response'].values\n",
    "response_text[0]\n",
    "# parsed_json = json.loads(response_text[0])\n",
    "# print(parsed_json)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ae6e4d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing article 0/30: 10.1002_2017jc013030\n",
      "Total blocks with data related dataset IDs found: 3\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf. Length: 4335 characters\n",
      "Preparing input with length: 4096\n",
      "Parsing response from generated IDs...\n",
      "Processing article 1/30: 10.1002_anie.201916483\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf. Length: 789 characters\n",
      "Preparing input with length: 840\n",
      "Parsing response from generated IDs...\n",
      "Processing article 2/30: 10.1002_anie.202005531\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf. Length: 1088 characters\n",
      "Preparing input with length: 1139\n",
      "Parsing response from generated IDs...\n",
      "Processing article 3/30: 10.1002_anie.202007717\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202007717.pdf. Length: 1130 characters\n",
      "Preparing input with length: 1181\n",
      "Parsing response from generated IDs...\n",
      "Processing article 4/30: 10.1002_chem.201902131\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201902131.pdf. Length: 564 characters\n",
      "Preparing input with length: 615\n",
      "Parsing response from generated IDs...\n",
      "Processing article 5/30: 10.1002_chem.201903120\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf. Length: 784 characters\n",
      "Preparing input with length: 835\n",
      "Parsing response from generated IDs...\n",
      "Processing article 6/30: 10.1002_chem.202000235\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202000235.pdf. Length: 610 characters\n",
      "Preparing input with length: 661\n",
      "Parsing response from generated IDs...\n",
      "Processing article 7/30: 10.1002_chem.202001412\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf. Length: 887 characters\n",
      "Preparing input with length: 938\n",
      "Parsing response from generated IDs...\n",
      "Processing article 8/30: 10.1002_chem.202001668\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001668.pdf. Length: 322 characters\n",
      "Preparing input with length: 373\n",
      "Parsing response from generated IDs...\n",
      "Processing article 9/30: 10.1002_chem.202003167\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202003167.pdf. Length: 827 characters\n",
      "Preparing input with length: 878\n",
      "Parsing response from generated IDs...\n",
      "Processing article 10/30: 10.1002_cssc.202201821\n",
      "Total blocks with data related dataset IDs found: 1\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_cssc.202201821.pdf. Length: 3253 characters\n",
      "Preparing input with length: 3304\n",
      "Parsing response from generated IDs...\n",
      "Processing article 11/30: 10.1002_ece3.3985\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.3985.pdf. Length: 1881 characters\n",
      "Preparing input with length: 1927\n",
      "Parsing response from generated IDs...\n",
      "Processing article 12/30: 10.1002_ece3.4466\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.4466.pdf. Length: 1804 characters\n",
      "Preparing input with length: 1850\n",
      "Parsing response from generated IDs...\n",
      "Processing article 13/30: 10.1002_ece3.5260\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf. Length: 1945 characters\n",
      "Preparing input with length: 1991\n",
      "Parsing response from generated IDs...\n",
      "Processing article 14/30: 10.1002_ece3.5395\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5395.pdf. Length: 1940 characters\n",
      "Preparing input with length: 1986\n",
      "Parsing response from generated IDs...\n",
      "Processing article 15/30: 10.1002_ece3.6144\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf. Length: 1799 characters\n",
      "Preparing input with length: 1845\n",
      "Parsing response from generated IDs...\n",
      "Processing article 16/30: 10.1002_ece3.6303\n",
      "Total blocks with data related dataset IDs found: 2\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6303.pdf. Length: 2121 characters\n",
      "Preparing input with length: 2167\n",
      "Parsing response from generated IDs...\n",
      "Processing article 17/30: 10.1002_ece3.6784\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6784.pdf. Length: 1859 characters\n",
      "Preparing input with length: 1905\n",
      "Parsing response from generated IDs...\n",
      "Processing article 18/30: 10.1002_ece3.961\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.961.pdf. Length: 2278 characters\n",
      "Preparing input with length: 2323\n",
      "Parsing response from generated IDs...\n",
      "Processing article 19/30: 10.1002_ece3.9627\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.9627.pdf. Length: 1800 characters\n",
      "Preparing input with length: 1846\n",
      "Parsing response from generated IDs...\n",
      "Processing article 20/30: 10.1002_ecs2.1280\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf. Length: 2565 characters\n",
      "Preparing input with length: 2611\n",
      "Parsing response from generated IDs...\n",
      "Processing article 21/30: 10.1002_ecs2.4619\n",
      "Total blocks with data related dataset IDs found: 2\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.4619.pdf. Length: 3015 characters\n",
      "Preparing input with length: 3061\n",
      "Parsing response from generated IDs...\n",
      "Processing article 22/30: 10.1002_ejic.201900904\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf. Length: 532 characters\n",
      "Preparing input with length: 583\n",
      "Parsing response from generated IDs...\n",
      "Processing article 23/30: 10.1002_ejoc.202000139\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf. Length: 402 characters\n",
      "Preparing input with length: 453\n",
      "Parsing response from generated IDs...\n",
      "Processing article 24/30: 10.1002_ejoc.202000916\n",
      "Total blocks with data related dataset IDs found: 0\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000916.pdf. Length: 215 characters\n",
      "Preparing input with length: 266\n",
      "Parsing response from generated IDs...\n",
      "Processing article 25/30: 10.1002_esp.5058\n",
      "Total blocks with data related dataset IDs found: 1\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf. Length: 4011 characters\n",
      "Preparing input with length: 4056\n"
     ]
    }
   ],
   "source": [
    "test_set_processed_articles_df = process_articles(test_file_paths_df, inference_model)\n",
    "test_set_processed_articles_df.to_csv(\"test_set_processed_articles.csv\", index=False)\n",
    "test_set_processed_articles_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4f2525aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Qwen model and tokenizer from: C:\\Users\\jim\\.cache\\kagglehub\\models\\qwen-lm\\qwen-3\\transformers\\0.6b\\1\n"
     ]
    }
   ],
   "source": [
    "eval_model = QwenModelEval(QWEN_BASE_MODEL_PATH, sys_prompt=\"You are a chatbot.\",)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c8bee422",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following generation flags are not valid and may be ignored: ['temperature', 'top_p', 'top_k']. Set `TRANSFORMERS_VERBOSITY=info` for more details.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "User: How many r's in strawberries?\n",
      "Preparing input for user: How many r's in strawberries?\n",
      "Generating response for user input:\n",
      "Parsing response from generated IDs...\n",
      "Bot: ('There are 2 \\'r\\'s in the word \"strawberries\".', '<think>\\nOkay, the user is asking how many \\'r\\'s are in \"strawberries\". Let me start by breaking down the word. \"Strawberries\" is spelled S-T-R-A-W-B-E-R-R-I-N-G-S. Let me count each \\'r\\' here. \\n\\nFirst letter: S, no \\'r\\'. Second: T, no. Third: R, one \\'r\\'. Fourth: A, no. Fifth: W, no. Sixth: B, no. Seventh: E, no. Eighth: R, another \\'r\\'. Ninth: I, no. Tenth: N, no. Eleventh: G, no. So that\\'s two \\'r\\'s in total. \\n\\nWait, maybe I should double-check. Let me write it out again: S-T-R-A-W-B-E-R-R-I-N-G-S. Yes, the third and eighth letters are both \\'r\\'s. So two \\'r\\'s. The user might be testing if I can count them correctly. I should make sure there\\'s no other \\'r\\'s I missed. No, I think that\\'s all. So the answer is two.\\n</think>')\n",
      "----------------------\n"
     ]
    }
   ],
   "source": [
    "# --- Example Usage of eval_model ---\n",
    "\n",
    "# First input (without /think or /no_think tags, thinking mode is enabled by default)\n",
    "user_input_1 = \"How many r's in strawberries?\"\n",
    "print(f\"User: {user_input_1}\")\n",
    "response_1 = eval_model.generate_response(user_input_1)\n",
    "print(f\"Bot: {response_1}\")\n",
    "print(\"----------------------\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "959c69e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# --- 2. Data Preparation for LLM Training (Revised for Combined Task) ---\n",
    "\n",
    "def load_base_llm_for_training():\n",
    "    \"\"\"Loads the base Qwen model and tokenizer for fine-tuning.\"\"\"\n",
    "    global llm_tokenizer, llm_model\n",
    "    if not AutoModelForCausalLM or not QWEN_BASE_MODEL_PATH:\n",
    "        print(\"LLM components not available or base model path not set. Skipping LLM loading.\")\n",
    "        return False\n",
    "    try:\n",
    "        print(f\"Loading Qwen tokenizer from: {QWEN_BASE_MODEL_PATH}\")\n",
    "        llm_tokenizer = AutoTokenizer.from_pretrained(QWEN_BASE_MODEL_PATH, trust_remote_code=True)\n",
    "        if llm_tokenizer.pad_token is None:\n",
    "            llm_tokenizer.pad_token = llm_tokenizer.eos_token\n",
    "            print(\"Set tokenizer.pad_token to tokenizer.eos_token\")\n",
    "\n",
    "        print(f\"Loading Qwen model from: {QWEN_BASE_MODEL_PATH}\")\n",
    "        llm_model = AutoModelForCausalLM.from_pretrained(\n",
    "            QWEN_BASE_MODEL_PATH,\n",
    "            torch_dtype=torch.bfloat16 if torch.cuda.is_available() and torch.cuda.is_bf16_supported() else torch.float32,\n",
    "            device_map=\"auto\", # Automatically uses GPU if available\n",
    "            trust_remote_code=True,\n",
    "            # load_in_8bit=True if bnb else False # Uncomment if bitsandbytes is used\n",
    "        )\n",
    "        print(f\"Base LLM loaded successfully on {llm_model.device}.\")\n",
    "        return True\n",
    "    except Exception as e:\n",
    "        print(f\"Error loading base LLM for training: {e}\")\n",
    "        llm_tokenizer, llm_model = None, None # Reset to None on failure\n",
    "        return False\n",
    "\n",
    "def prepare_training_data_for_llm(\n",
    "    training_df: pd.DataFrame,\n",
    "    all_article_texts: dict[str, str],\n",
    "    tokenizer_max_length: int\n",
    ") -> Dataset:\n",
    "    \"\"\"\n",
    "    Prepares training data for LLM fine-tuning, aggregating dataset IDs and classifications\n",
    "    per article and formatting into ChatML JSON output.\n",
    "    \"\"\"\n",
    "    formatted_examples = []\n",
    "\n",
    "    # Group training data by article_id to get all datasets for each article\n",
    "    # This creates a dictionary where keys are article_ids and values are lists of dataset dicts\n",
    "    grouped_training_data = training_df.groupby('article_id').apply(\n",
    "        lambda x: [{\"dataset_id\": row['dataset_id'], \"classification\": row['label']} for _, row in x]\n",
    "    ).to_dict()\n",
    "\n",
    "    # Get all article IDs for which we have text content\n",
    "    all_article_ids_with_text = set(all_article_texts.keys())\n",
    "    \n",
    "    # Iterate through all articles for which we have text (these are our potential training examples)\n",
    "    for article_id in all_article_ids_with_text:\n",
    "        article_text = all_article_texts.get(article_id, \"\")\n",
    "        if not article_text:\n",
    "            print(f\"Warning: Article text for {article_id} not found. Skipping training example.\")\n",
    "            continue\n",
    "\n",
    "        # Truncate article text to fit within the model's context window\n",
    "        # Reserve tokens for the prompt and the expected JSON response.\n",
    "        # A typical Qwen 1.5 model has 32768 max_seq_length.\n",
    "        # 512 tokens for prompt/response is a safe estimate.\n",
    "        truncated_article_text = article_text[:tokenizer_max_length - 512] \n",
    "\n",
    "        # Determine the ground truth output for this article\n",
    "        if article_id in grouped_training_data:\n",
    "            # Article has datasets, format them as JSON\n",
    "            ground_truth_datasets = grouped_training_data[article_id]\n",
    "            assistant_response_json = json.dumps(ground_truth_datasets, ensure_ascii=False)\n",
    "        else:\n",
    "            # Article has no datasets in training data, so the model should output an empty list.\n",
    "            # This explicitly trains the model to output '[]' for \"Missing\" cases.\n",
    "            assistant_response_json = \"[]\"\n",
    "            # print(f\"Info: Article {article_id} has no datasets in training data. Training to output '[]'.\")\n",
    "\n",
    "        # Construct the user message for the LLM\n",
    "        user_message = f\"\"\"\n",
    "Article Text:\n",
    "{truncated_article_text}\n",
    "\n",
    "Task: Identify all datasets or databases used in this research article and classify each as \"Primary\" (if created by the authors for this research) or \"Secondary\" (if an existing dataset used in this research).\n",
    "\n",
    "Output Format: Provide a JSON list of objects. Each object should have \"dataset_id\" and \"classification\" keys. If no datasets are identified, return an empty JSON list: [].\n",
    "\"\"\"\n",
    "        # Construct the full ChatML formatted string for SFTTrainer\n",
    "        # The trainer will use this entire string as the 'text' field.\n",
    "        chatml_formatted_string = f\"<|im_start|>system\\nYou are an expert research assistant. Your task is to extract and classify datasets from scientific articles.<|im_end|>\\n<|im_start|>user\\n{user_message.strip()}<|im_end|>\\n<|im_start|>assistant\\n{assistant_response_json}<|im_end|>\"\n",
    "        \n",
    "        formatted_examples.append({\"text\": chatml_formatted_string})\n",
    "\n",
    "    if not formatted_examples:\n",
    "        raise ValueError(\"No training examples could be prepared. Check your data and article texts.\")\n",
    "\n",
    "    return Dataset.from_list(formatted_examples)\n",
    "\n",
    "# --- 3. LLM Model Training (Fine-tuning) ---\n",
    "\n",
    "# Attempt to load tokenizer and model if not already loaded (e.g., if previous training failed or was skipped)\n",
    "if llm_model is None:\n",
    "    load_base_llm_for_training()\n",
    "\n",
    "if llm_model and not training_df.empty and Dataset: # Ensure Dataset is imported\n",
    "    print(\"\\n--- Preparing data for Fine-tuning (Combined Task) ---\")\n",
    "    # Use the model's max_length for context, or a reasonable default if tokenizer isn't loaded\n",
    "    max_len = llm_tokenizer.model_max_length if llm_tokenizer else 4096 \n",
    "    train_dataset = prepare_training_data_for_llm(training_df, all_article_texts, max_len)\n",
    "    \n",
    "    print(f\"Prepared {len(train_dataset)} examples for fine-tuning.\")\n",
    "    print(\"Example formatted training instance (first 500 chars):\")\n",
    "    print(train_dataset[0]['text'][:500])\n",
    "\n",
    "    print(\"\\n--- Starting Fine-tuning (Combined Task) ---\")\n",
    "    try:\n",
    "        training_args = TrainingArguments(\n",
    "            output_dir=f\"{FINE_TUNED_MODEL_OUTPUT_DIR}/checkpoints\",\n",
    "            num_train_epochs=1,  # Start with 1 epoch, adjust as needed\n",
    "            per_device_train_batch_size=1, # Adjust based on VRAM\n",
    "            gradient_accumulation_steps=4, # Effective batch size = 1 * 4 = 4\n",
    "            learning_rate=2e-5,\n",
    "            logging_steps=10,\n",
    "            save_steps=50, # Save checkpoints periodically\n",
    "            fp16=torch.cuda.is_available() and not torch.cuda.is_bf16_supported(),\n",
    "            bf16=torch.cuda.is_available() and torch.cuda.is_bf16_supported(),\n",
    "            optim=\"paged_adamw_8bit\", # Good for memory efficiency if bitsandbytes is installed\n",
    "            # report_to=\"none\", # Disable logging to external services\n",
    "            # max_steps=100, # For quick testing\n",
    "        )\n",
    "\n",
    "        trainer = SFTTrainer(\n",
    "            model=llm_model,\n",
    "            tokenizer=llm_tokenizer,\n",
    "            train_dataset=train_dataset,\n",
    "            dataset_text_field=\"text\", # This field contains the full ChatML string\n",
    "            args=training_args,\n",
    "            max_seq_length=max_len, # Use the model's full context length\n",
    "            packing=False, # Set to True if your inputs are much shorter than max_seq_length\n",
    "        )\n",
    "\n",
    "        trainer.train()\n",
    "        print(\"Fine-tuning completed.\")\n",
    "\n",
    "        print(f\"Saving fine-tuned model to: {FINE_TUNED_MODEL_OUTPUT_DIR}\")\n",
    "        trainer.save_model(FINE_TUNED_MODEL_OUTPUT_DIR)\n",
    "        print(\"Model and tokenizer saved.\")\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"An error occurred during fine-tuning: {e}\")\n",
    "        import traceback\n",
    "        traceback.print_exc()\n",
    "        llm_model = None # Mark model as failed to load/train\n",
    "else:\n",
    "    print(\"Skipping LLM fine-tuning due to missing training data or LLM components.\")\n",
    "\n",
    "\n",
    "# --- 4. LLM-based Extraction & Classification (Inference) ---\n",
    "\n",
    "# Load the fine-tuned model for inference (if training was successful)\n",
    "# If training was skipped or failed, this will attempt to load from the base path or fail.\n",
    "if inference_model is None: # Only load if not already loaded\n",
    "    if AutoModelForCausalLM: # Check if transformers is available\n",
    "        if os.path.exists(FINE_TUNED_MODEL_OUTPUT_DIR) and os.path.isdir(FINE_TUNED_MODEL_OUTPUT_DIR):\n",
    "            MODEL_TO_LOAD = FINE_TUNED_MODEL_OUTPUT_DIR\n",
    "            print(f\"Loading fine-tuned model for inference from: {MODEL_TO_LOAD}\")\n",
    "        else:\n",
    "            MODEL_TO_LOAD = QWEN_BASE_MODEL_PATH\n",
    "            print(f\"Fine-tuned model not found. Loading base model for inference from: {MODEL_TO_LOAD}\")\n",
    "\n",
    "        try:\n",
    "            inference_tokenizer = AutoTokenizer.from_pretrained(MODEL_TO_LOAD, trust_remote_code=True)\n",
    "            if inference_tokenizer.pad_token is None:\n",
    "                inference_tokenizer.pad_token = inference_tokenizer.eos_token\n",
    "            inference_model = AutoModelForCausalLM.from_pretrained(\n",
    "                MODEL_TO_LOAD,\n",
    "                torch_dtype=torch.bfloat16 if torch.cuda.is_available() and torch.cuda.is_bf16_supported() else torch.float32,\n",
    "                device_map=\"auto\",\n",
    "                trust_remote_code=True\n",
    "            ).eval() # Set to evaluation mode\n",
    "            print(f\"Inference LLM loaded successfully on {inference_model.device}.\")\n",
    "        except Exception as e:\n",
    "            print(f\"Error loading inference LLM from {MODEL_TO_LOAD}: {e}\")\n",
    "            inference_model, inference_tokenizer = None, None\n",
    "    else:\n",
    "        print(\"Transformers library not available. Cannot load LLM for inference.\")\n",
    "\n",
    "\n",
    "def extract_and_classify_with_llm(article_text: str) -> list[dict]:\n",
    "    \"\"\"\n",
    "    Uses the loaded LLM to extract dataset IDs and classify them.\n",
    "    Returns a list of dictionaries like [{\"dataset_id\": \"...\", \"classification\": \"...\"}].\n",
    "    Returns an empty list if LLM is unavailable or parsing fails.\n",
    "    \"\"\"\n",
    "    if not inference_model or not inference_tokenizer:\n",
    "        print(\"  LLM unavailable for extraction/classification.\")\n",
    "        return [] # Return empty list if LLM is not loaded\n",
    "\n",
    "    # Truncate article text for inference if it exceeds model's context window\n",
    "    # Use the same max_length as during training for consistency\n",
    "    max_inference_context_length = inference_tokenizer.model_max_length - 256 # Reserve tokens for prompt and response\n",
    "    truncated_article_text = article_text[:max_inference_context_length]\n",
    "\n",
    "    user_message = f\"\"\"\n",
    "Article Text:\n",
    "{truncated_article_text}\n",
    "\n",
    "Task: Identify all datasets or databases used in this research article and classify each as \"Primary\" (if created by the authors for this research) or \"Secondary\" (if an existing dataset used in this research).\n",
    "\n",
    "Output Format: Provide a JSON list of objects. Each object should have \"dataset_id\" and \"classification\" keys. If no datasets are identified, return an empty JSON list: [].\n",
    "\"\"\"\n",
    "    messages = [\n",
    "        {\"role\": \"system\", \"content\": \"You are an expert research assistant. Your task is to extract and classify datasets from scientific articles.\"},\n",
    "        {\"role\": \"user\", \"content\": user_message.strip()}\n",
    "    ]\n",
    "    \n",
    "    input_ids = inference_tokenizer.apply_chat_template(\n",
    "        messages,\n",
    "        tokenize=True,\n",
    "        add_generation_prompt=True,\n",
    "        return_tensors=\"pt\"\n",
    "    ).to(inference_model.device)\n",
    "\n",
    "    try:\n",
    "        with torch.no_grad():\n",
    "            outputs = inference_model.generate(\n",
    "                input_ids,\n",
    "                max_new_tokens=512, # Allow more tokens for multiple dataset outputs\n",
    "                pad_token_id=inference_tokenizer.eos_token_id,\n",
    "                eos_token_id=inference_tokenizer.convert_tokens_to_ids(\"<|im_end|>\")\n",
    "            )\n",
    "        \n",
    "        response_text = inference_tokenizer.decode(\n",
    "            outputs[0][input_ids.shape[1]:],\n",
    "            skip_special_tokens=False # Keep special tokens to remove <|im_end|> explicitly\n",
    "        ).strip()\n",
    "        response_text = response_text.replace(\"<|im_end|>\", \"\").strip()\n",
    "        \n",
    "        print(f\"  LLM raw response: '{response_text}'\")\n",
    "\n",
    "        # Attempt to parse the JSON output\n",
    "        try:\n",
    "            parsed_data = json.loads(response_text)\n",
    "            if isinstance(parsed_data, list):\n",
    "                # Validate structure: each item should be a dict with 'dataset_id' and 'classification'\n",
    "                valid_datasets = []\n",
    "                for item in parsed_data:\n",
    "                    if isinstance(item, dict) and 'dataset_id' in item and 'classification' in item:\n",
    "                        # Basic validation for classification label\n",
    "                        if item['classification'] in [\"Primary\", \"Secondary\"]:\n",
    "                            valid_datasets.append(item)\n",
    "                        else:\n",
    "                            print(f\"  Warning: Invalid classification '{item['classification']}' for dataset '{item.get('dataset_id', 'N/A')}'. Skipping.\")\n",
    "                    else:\n",
    "                        print(f\"  Warning: Malformed JSON object: {item}. Skipping.\")\n",
    "                return valid_datasets\n",
    "            else:\n",
    "                print(f\"  Warning: LLM did not return a JSON list: {response_text}\")\n",
    "                return []\n",
    "        except json.JSONDecodeError as jde:\n",
    "            print(f\"  Error decoding JSON from LLM response: {jde}. Raw response: '{response_text}'\")\n",
    "            return []\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"  Error during LLM generation: {e}\")\n",
    "        return []\n",
    "\n",
    "# --- Main Processing Loop for all articles (Revised) ---\n",
    "print(\"\\n--- Starting Article Processing and Classification (LLM-driven) ---\")\n",
    "final_results = []\n",
    "\n",
    "for article_id, article_text in all_article_texts.items():\n",
    "    print(f\"\\nProcessing article: {article_id}\")\n",
    "    \n",
    "    # LLM directly extracts and classifies\n",
    "    identified_datasets = extract_and_classify_with_llm(article_text)\n",
    "    \n",
    "    if not identified_datasets:\n",
    "        # If LLM returns an empty list, classify the article as \"Missing\"\n",
    "        print(f\"  LLM identified no datasets for {article_id}. Classifying as 'Missing'.\")\n",
    "        final_results.append({\n",
    "            \"article_id\": article_id,\n",
    "            \"dataset_id\": \"N/A\", # Indicate no specific dataset ID\n",
    "            \"classification_label\": \"Missing\"\n",
    "        })\n",
    "    else:\n",
    "        print(f\"  LLM identified {len(identified_datasets)} dataset(s) for {article_id}.\")\n",
    "        for item in identified_datasets:\n",
    "            final_results.append({\n",
    "                \"article_id\": article_id,\n",
    "                \"dataset_id\": item.get(\"dataset_id\", \"Unknown\"), # Use .get() for safety\n",
    "                \"classification_label\": item.get(\"classification\", \"Uncertain_LLM\")\n",
    "            })\n",
    "\n",
    "\n",
    "# --- 5. Results & Output ---\n",
    "\n",
    "print(\"\\n--- Final Results ---\")\n",
    "if final_results:\n",
    "    results_df = pd.DataFrame(final_results)\n",
    "    print(results_df.head(10)) # Print first 10 rows\n",
    "    \n",
    "    # Save to CSV\n",
    "    results_df.to_csv(FINAL_RESULTS_CSV_PATH, index=False)\n",
    "    print(f\"\\nResults saved to: {FINAL_RESULTS_CSV_PATH}\")\n",
    "else:\n",
    "    print(\"No results generated.\")\n",
    "\n",
    "print(\"\\nProcessing complete, Jim!\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
