{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3195093d",
   "metadata": {},
   "source": [
    "#### 1. Setup and Dependencies\n",
    "\n",
    "First, ensure all necessary libraries are installed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7c498a2e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name: numpy\n",
      "Version: 2.3.0\n",
      "Summary: Fundamental package for array computing in Python\n",
      "Home-page: https://numpy.org\n",
      "Author: Travis E. Oliphant et al.\n",
      "Author-email: \n",
      "License: Copyright (c) 2005-2025, NumPy Developers.\n",
      " All rights reserved.\n",
      "\n",
      " Redistribution and use in source and binary forms, with or without\n",
      " modification, are permitted provided that the following conditions are\n",
      " met:\n",
      "\n",
      "     * Redistributions of source code must retain the above copyright\n",
      "        notice, this list of conditions and the following disclaimer.\n",
      "\n",
      "     * Redistributions in binary form must reproduce the above\n",
      "        copyright notice, this list of conditions and the following\n",
      "        disclaimer in the documentation and/or other materials provided\n",
      "        with the distribution.\n",
      "\n",
      "     * Neither the name of the NumPy Developers nor the names of any\n",
      "        contributors may be used to endorse or promote products derived\n",
      "        from this software without specific prior written permission.\n",
      "\n",
      " THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS\n",
      " \"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT\n",
      " LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR\n",
      " A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT\n",
      " OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,\n",
      " SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT\n",
      " LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,\n",
      " DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY\n",
      " THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      " (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE\n",
      " OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "\n",
      " ----\n",
      "\n",
      " The NumPy repository and source distributions bundle several libraries that are\n",
      " compatibly licensed.  We list these here.\n",
      "\n",
      " Name: lapack-lite\n",
      " Files: numpy/linalg/lapack_lite/*\n",
      " License: BSD-3-Clause\n",
      "   For details, see numpy/linalg/lapack_lite/LICENSE.txt\n",
      "\n",
      " Name: dragon4\n",
      " Files: numpy/_core/src/multiarray/dragon4.c\n",
      " License: MIT\n",
      "   For license text, see numpy/_core/src/multiarray/dragon4.c\n",
      "\n",
      " Name: libdivide\n",
      " Files: numpy/_core/include/numpy/libdivide/*\n",
      " License: Zlib\n",
      "   For license text, see numpy/_core/include/numpy/libdivide/LICENSE.txt\n",
      "\n",
      "\n",
      " Note that the following files are vendored in the repository and sdist but not\n",
      " installed in built numpy packages:\n",
      "\n",
      " Name: Meson\n",
      " Files: vendored-meson/meson/*\n",
      " License: Apache 2.0\n",
      "   For license text, see vendored-meson/meson/COPYING\n",
      "\n",
      " Name: spin\n",
      " Files: .spin/cmds.py\n",
      " License: BSD-3\n",
      "   For license text, see .spin/LICENSE\n",
      "\n",
      " Name: tempita\n",
      " Files: numpy/_build_utils/tempita/*\n",
      " License: MIT\n",
      "   For details, see numpy/_build_utils/tempita/LICENCE.txt\n",
      "\n",
      " ----\n",
      "\n",
      " This binary distribution of NumPy also bundles the following software:\n",
      "\n",
      "\n",
      " Name: OpenBLAS\n",
      " Files: numpy.libs\\libscipy_openblas*.dll\n",
      " Description: bundled as a dynamically linked library\n",
      " Availability: https://github.com/OpenMathLib/OpenBLAS/\n",
      " License: BSD-3-Clause\n",
      "   Copyright (c) 2011-2014, The OpenBLAS Project\n",
      "   All rights reserved.\n",
      "\n",
      "   Redistribution and use in source and binary forms, with or without\n",
      "   modification, are permitted provided that the following conditions are\n",
      "   met:\n",
      "\n",
      "      1. Redistributions of source code must retain the above copyright\n",
      "         notice, this list of conditions and the following disclaimer.\n",
      "\n",
      "      2. Redistributions in binary form must reproduce the above copyright\n",
      "         notice, this list of conditions and the following disclaimer in\n",
      "         the documentation and/or other materials provided with the\n",
      "         distribution.\n",
      "      3. Neither the name of the OpenBLAS project nor the names of\n",
      "         its contributors may be used to endorse or promote products\n",
      "         derived from this software without specific prior written\n",
      "         permission.\n",
      "\n",
      "   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS \"AS IS\"\n",
      "   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE\n",
      "   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE\n",
      "   ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE\n",
      "   LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL\n",
      "   DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR\n",
      "   SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER\n",
      "   CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,\n",
      "   OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE\n",
      "   USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "\n",
      "\n",
      " Name: LAPACK\n",
      " Files: numpy.libs\\libscipy_openblas*.dll\n",
      " Description: bundled in OpenBLAS\n",
      " Availability: https://github.com/OpenMathLib/OpenBLAS/\n",
      " License: BSD-3-Clause-Open-MPI\n",
      "   Copyright (c) 1992-2013 The University of Tennessee and The University\n",
      "                           of Tennessee Research Foundation.  All rights\n",
      "                           reserved.\n",
      "   Copyright (c) 2000-2013 The University of California Berkeley. All\n",
      "                           rights reserved.\n",
      "   Copyright (c) 2006-2013 The University of Colorado Denver.  All rights\n",
      "                           reserved.\n",
      "\n",
      "   $COPYRIGHT$\n",
      "\n",
      "   Additional copyrights may follow\n",
      "\n",
      "   $HEADER$\n",
      "\n",
      "   Redistribution and use in source and binary forms, with or without\n",
      "   modification, are permitted provided that the following conditions are\n",
      "   met:\n",
      "\n",
      "   - Redistributions of source code must retain the above copyright\n",
      "     notice, this list of conditions and the following disclaimer.\n",
      "\n",
      "   - Redistributions in binary form must reproduce the above copyright\n",
      "     notice, this list of conditions and the following disclaimer listed\n",
      "     in this license in the documentation and/or other materials\n",
      "     provided with the distribution.\n",
      "\n",
      "   - Neither the name of the copyright holders nor the names of its\n",
      "     contributors may be used to endorse or promote products derived from\n",
      "     this software without specific prior written permission.\n",
      "\n",
      "   The copyright holders provide no reassurances that the source code\n",
      "   provided does not infringe any patent, copyright, or any other\n",
      "   intellectual property rights of third parties.  The copyright holders\n",
      "   disclaim any liability to any recipient for claims brought against\n",
      "   recipient by any third party for infringement of that parties\n",
      "   intellectual property rights.\n",
      "\n",
      "   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS\n",
      "   \"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT\n",
      "   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR\n",
      "   A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT\n",
      "   OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,\n",
      "   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT\n",
      "   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,\n",
      "   DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY\n",
      "   THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n",
      "   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE\n",
      "   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n",
      "\n",
      "\n",
      " Name: GCC runtime library\n",
      " Files: numpy.libs\\libscipy_openblas*.dll\n",
      " Description: statically linked to files compiled with gcc\n",
      " Availability: https://gcc.gnu.org/git/?p=gcc.git;a=tree;f=libgfortran\n",
      " License: GPL-3.0-or-later WITH GCC-exception-3.1\n",
      "   Copyright (C) 2002-2017 Free Software Foundation, Inc.\n",
      "\n",
      "   Libgfortran is free software; you can redistribute it and/or modify\n",
      "   it under the terms of the GNU General Public License as published by\n",
      "   the Free Software Foundation; either version 3, or (at your option)\n",
      "   any later version.\n",
      "\n",
      "   Libgfortran is distributed in the hope that it will be useful,\n",
      "   but WITHOUT ANY WARRANTY; without even the implied warranty of\n",
      "   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n",
      "   GNU General Public License for more details.\n",
      "\n",
      "   Under Section 7 of GPL version 3, you are granted additional\n",
      "   permissions described in the GCC Runtime Library Exception, version\n",
      "   3.1, as published by the Free Software Foundation.\n",
      "\n",
      "   You should have received a copy of the GNU General Public License and\n",
      "   a copy of the GCC Runtime Library Exception along with this program;\n",
      "   see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see\n",
      "   <http://www.gnu.org/licenses/>.\n",
      "\n",
      " ----\n",
      "\n",
      " Full text of license texts referred to above follows (that they are\n",
      " listed below does not necessarily imply the conditions apply to the\n",
      " present binary release):\n",
      "\n",
      " ----\n",
      "\n",
      " GCC RUNTIME LIBRARY EXCEPTION\n",
      "\n",
      " Version 3.1, 31 March 2009\n",
      "\n",
      " Copyright (C) 2009 Free Software Foundation, Inc. <https://fsf.org/>\n",
      "\n",
      " Everyone is permitted to copy and distribute verbatim copies of this\n",
      " license document, but changing it is not allowed.\n",
      "\n",
      " This GCC Runtime Library Exception (\"Exception\") is an additional\n",
      " permission under section 7 of the GNU General Public License, version\n",
      " 3 (\"GPLv3\"). It applies to a given file (the \"Runtime Library\") that\n",
      " bears a notice placed by the copyright holder of the file stating that\n",
      " the file is governed by GPLv3 along with this Exception.\n",
      "\n",
      " When you use GCC to compile a program, GCC may combine portions of\n",
      " certain GCC header files and runtime libraries with the compiled\n",
      " program. The purpose of this Exception is to allow compilation of\n",
      " non-GPL (including proprietary) programs to use, in this way, the\n",
      " header files and runtime libraries covered by this Exception.\n",
      "\n",
      " 0. Definitions.\n",
      "\n",
      " A file is an \"Independent Module\" if it either requires the Runtime\n",
      " Library for execution after a Compilation Process, or makes use of an\n",
      " interface provided by the Runtime Library, but is not otherwise based\n",
      " on the Runtime Library.\n",
      "\n",
      " \"GCC\" means a version of the GNU Compiler Collection, with or without\n",
      " modifications, governed by version 3 (or a specified later version) of\n",
      " the GNU General Public License (GPL) with the option of using any\n",
      " subsequent versions published by the FSF.\n",
      "\n",
      " \"GPL-compatible Software\" is software whose conditions of propagation,\n",
      " modification and use would permit combination with GCC in accord with\n",
      " the license of GCC.\n",
      "\n",
      " \"Target Code\" refers to output from any compiler for a real or virtual\n",
      " target processor architecture, in executable form or suitable for\n",
      " input to an assembler, loader, linker and/or execution\n",
      " phase. Notwithstanding that, Target Code does not include data in any\n",
      " format that is used as a compiler intermediate representation, or used\n",
      " for producing a compiler intermediate representation.\n",
      "\n",
      " The \"Compilation Process\" transforms code entirely represented in\n",
      " non-intermediate languages designed for human-written code, and/or in\n",
      " Java Virtual Machine byte code, into Target Code. Thus, for example,\n",
      " use of source code generators and preprocessors need not be considered\n",
      " part of the Compilation Process, since the Compilation Process can be\n",
      " understood as starting with the output of the generators or\n",
      " preprocessors.\n",
      "\n",
      " A Compilation Process is \"Eligible\" if it is done using GCC, alone or\n",
      " with other GPL-compatible software, or if it is done without using any\n",
      " work based on GCC. For example, using non-GPL-compatible Software to\n",
      " optimize any GCC intermediate representations would not qualify as an\n",
      " Eligible Compilation Process.\n",
      "\n",
      " 1. Grant of Additional Permission.\n",
      "\n",
      " You have permission to propagate a work of Target Code formed by\n",
      " combining the Runtime Library with Independent Modules, even if such\n",
      " propagation would otherwise violate the terms of GPLv3, provided that\n",
      " all Target Code was generated by Eligible Compilation Processes. You\n",
      " may then convey such a combination under terms of your choice,\n",
      " consistent with the licensing of the Independent Modules.\n",
      "\n",
      " 2. No Weakening of GCC Copyleft.\n",
      "\n",
      " The availability of this Exception does not imply any general\n",
      " presumption that third-party software is unaffected by the copyleft\n",
      " requirements of the license of GCC.\n",
      "\n",
      " ----\n",
      "\n",
      "                     GNU GENERAL PUBLIC LICENSE\n",
      "                        Version 3, 29 June 2007\n",
      "\n",
      "  Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>\n",
      "  Everyone is permitted to copy and distribute verbatim copies\n",
      "  of this license document, but changing it is not allowed.\n",
      "\n",
      "                             Preamble\n",
      "\n",
      "   The GNU General Public License is a free, copyleft license for\n",
      " software and other kinds of works.\n",
      "\n",
      "   The licenses for most software and other practical works are designed\n",
      " to take away your freedom to share and change the works.  By contrast,\n",
      " the GNU General Public License is intended to guarantee your freedom to\n",
      " share and change all versions of a program--to make sure it remains free\n",
      " software for all its users.  We, the Free Software Foundation, use the\n",
      " GNU General Public License for most of our software; it applies also to\n",
      " any other work released this way by its authors.  You can apply it to\n",
      " your programs, too.\n",
      "\n",
      "   When we speak of free software, we are referring to freedom, not\n",
      " price.  Our General Public Licenses are designed to make sure that you\n",
      " have the freedom to distribute copies of free software (and charge for\n",
      " them if you wish), that you receive source code or can get it if you\n",
      " want it, that you can change the software or use pieces of it in new\n",
      " free programs, and that you know you can do these things.\n",
      "\n",
      "   To protect your rights, we need to prevent others from denying you\n",
      " these rights or asking you to surrender the rights.  Therefore, you have\n",
      " certain responsibilities if you distribute copies of the software, or if\n",
      " you modify it: responsibilities to respect the freedom of others.\n",
      "\n",
      "   For example, if you distribute copies of such a program, whether\n",
      " gratis or for a fee, you must pass on to the recipients the same\n",
      " freedoms that you received.  You must make sure that they, too, receive\n",
      " or can get the source code.  And you must show them these terms so they\n",
      " know their rights.\n",
      "\n",
      "   Developers that use the GNU GPL protect your rights with two steps:\n",
      " (1) assert copyright on the software, and (2) offer you this License\n",
      " giving you legal permission to copy, distribute and/or modify it.\n",
      "\n",
      "   For the developers' and authors' protection, the GPL clearly explains\n",
      " that there is no warranty for this free software.  For both users' and\n",
      " authors' sake, the GPL requires that modified versions be marked as\n",
      " changed, so that their problems will not be attributed erroneously to\n",
      " authors of previous versions.\n",
      "\n",
      "   Some devices are designed to deny users access to install or run\n",
      " modified versions of the software inside them, although the manufacturer\n",
      " can do so.  This is fundamentally incompatible with the aim of\n",
      " protecting users' freedom to change the software.  The systematic\n",
      " pattern of such abuse occurs in the area of products for individuals to\n",
      " use, which is precisely where it is most unacceptable.  Therefore, we\n",
      " have designed this version of the GPL to prohibit the practice for those\n",
      " products.  If such problems arise substantially in other domains, we\n",
      " stand ready to extend this provision to those domains in future versions\n",
      " of the GPL, as needed to protect the freedom of users.\n",
      "\n",
      "   Finally, every program is threatened constantly by software patents.\n",
      " States should not allow patents to restrict development and use of\n",
      " software on general-purpose computers, but in those that do, we wish to\n",
      " avoid the special danger that patents applied to a free program could\n",
      " make it effectively proprietary.  To prevent this, the GPL assures that\n",
      " patents cannot be used to render the program non-free.\n",
      "\n",
      "   The precise terms and conditions for copying, distribution and\n",
      " modification follow.\n",
      "\n",
      "                        TERMS AND CONDITIONS\n",
      "\n",
      "   0. Definitions.\n",
      "\n",
      "   \"This License\" refers to version 3 of the GNU General Public License.\n",
      "\n",
      "   \"Copyright\" also means copyright-like laws that apply to other kinds of\n",
      " works, such as semiconductor masks.\n",
      "\n",
      "   \"The Program\" refers to any copyrightable work licensed under this\n",
      " License.  Each licensee is addressed as \"you\".  \"Licensees\" and\n",
      " \"recipients\" may be individuals or organizations.\n",
      "\n",
      "   To \"modify\" a work means to copy from or adapt all or part of the work\n",
      " in a fashion requiring copyright permission, other than the making of an\n",
      " exact copy.  The resulting work is called a \"modified version\" of the\n",
      " earlier work or a work \"based on\" the earlier work.\n",
      "\n",
      "   A \"covered work\" means either the unmodified Program or a work based\n",
      " on the Program.\n",
      "\n",
      "   To \"propagate\" a work means to do anything with it that, without\n",
      " permission, would make you directly or secondarily liable for\n",
      " infringement under applicable copyright law, except executing it on a\n",
      " computer or modifying a private copy.  Propagation includes copying,\n",
      " distribution (with or without modification), making available to the\n",
      " public, and in some countries other activities as well.\n",
      "\n",
      "   To \"convey\" a work means any kind of propagation that enables other\n",
      " parties to make or receive copies.  Mere interaction with a user through\n",
      " a computer network, with no transfer of a copy, is not conveying.\n",
      "\n",
      "   An interactive user interface displays \"Appropriate Legal Notices\"\n",
      " to the extent that it includes a convenient and prominently visible\n",
      " feature that (1) displays an appropriate copyright notice, and (2)\n",
      " tells the user that there is no warranty for the work (except to the\n",
      " extent that warranties are provided), that licensees may convey the\n",
      " work under this License, and how to view a copy of this License.  If\n",
      " the interface presents a list of user commands or options, such as a\n",
      " menu, a prominent item in the list meets this criterion.\n",
      "\n",
      "   1. Source Code.\n",
      "\n",
      "   The \"source code\" for a work means the preferred form of the work\n",
      " for making modifications to it.  \"Object code\" means any non-source\n",
      " form of a work.\n",
      "\n",
      "   A \"Standard Interface\" means an interface that either is an official\n",
      " standard defined by a recognized standards body, or, in the case of\n",
      " interfaces specified for a particular programming language, one that\n",
      " is widely used among developers working in that language.\n",
      "\n",
      "   The \"System Libraries\" of an executable work include anything, other\n",
      " than the work as a whole, that (a) is included in the normal form of\n",
      " packaging a Major Component, but which is not part of that Major\n",
      " Component, and (b) serves only to enable use of the work with that\n",
      " Major Component, or to implement a Standard Interface for which an\n",
      " implementation is available to the public in source code form.  A\n",
      " \"Major Component\", in this context, means a major essential component\n",
      " (kernel, window system, and so on) of the specific operating system\n",
      " (if any) on which the executable work runs, or a compiler used to\n",
      " produce the work, or an object code interpreter used to run it.\n",
      "\n",
      "   The \"Corresponding Source\" for a work in object code form means all\n",
      " the source code needed to generate, install, and (for an executable\n",
      " work) run the object code and to modify the work, including scripts to\n",
      " control those activities.  However, it does not include the work's\n",
      " System Libraries, or general-purpose tools or generally available free\n",
      " programs which are used unmodified in performing those activities but\n",
      " which are not part of the work.  For example, Corresponding Source\n",
      " includes interface definition files associated with source files for\n",
      " the work, and the source code for shared libraries and dynamically\n",
      " linked subprograms that the work is specifically designed to require,\n",
      " such as by intimate data communication or control flow between those\n",
      " subprograms and other parts of the work.\n",
      "\n",
      "   The Corresponding Source need not include anything that users\n",
      " can regenerate automatically from other parts of the Corresponding\n",
      " Source.\n",
      "\n",
      "   The Corresponding Source for a work in source code form is that\n",
      " same work.\n",
      "\n",
      "   2. Basic Permissions.\n",
      "\n",
      "   All rights granted under this License are granted for the term of\n",
      " copyright on the Program, and are irrevocable provided the stated\n",
      " conditions are met.  This License explicitly affirms your unlimited\n",
      " permission to run the unmodified Program.  The output from running a\n",
      " covered work is covered by this License only if the output, given its\n",
      " content, constitutes a covered work.  This License acknowledges your\n",
      " rights of fair use or other equivalent, as provided by copyright law.\n",
      "\n",
      "   You may make, run and propagate covered works that you do not\n",
      " convey, without conditions so long as your license otherwise remains\n",
      " in force.  You may convey covered works to others for the sole purpose\n",
      " of having them make modifications exclusively for you, or provide you\n",
      " with facilities for running those works, provided that you comply with\n",
      " the terms of this License in conveying all material for which you do\n",
      " not control copyright.  Those thus making or running the covered works\n",
      " for you must do so exclusively on your behalf, under your direction\n",
      " and control, on terms that prohibit them from making any copies of\n",
      " your copyrighted material outside their relationship with you.\n",
      "\n",
      "   Conveying under any other circumstances is permitted solely under\n",
      " the conditions stated below.  Sublicensing is not allowed; section 10\n",
      " makes it unnecessary.\n",
      "\n",
      "   3. Protecting Users' Legal Rights From Anti-Circumvention Law.\n",
      "\n",
      "   No covered work shall be deemed part of an effective technological\n",
      " measure under any applicable law fulfilling obligations under article\n",
      " 11 of the WIPO copyright treaty adopted on 20 December 1996, or\n",
      " similar laws prohibiting or restricting circumvention of such\n",
      " measures.\n",
      "\n",
      "   When you convey a covered work, you waive any legal power to forbid\n",
      " circumvention of technological measures to the extent such circumvention\n",
      " is effected by exercising rights under this License with respect to\n",
      " the covered work, and you disclaim any intention to limit operation or\n",
      " modification of the work as a means of enforcing, against the work's\n",
      " users, your or third parties' legal rights to forbid circumvention of\n",
      " technological measures.\n",
      "\n",
      "   4. Conveying Verbatim Copies.\n",
      "\n",
      "   You may convey verbatim copies of the Program's source code as you\n",
      " receive it, in any medium, provided that you conspicuously and\n",
      " appropriately publish on each copy an appropriate copyright notice;\n",
      " keep intact all notices stating that this License and any\n",
      " non-permissive terms added in accord with section 7 apply to the code;\n",
      " keep intact all notices of the absence of any warranty; and give all\n",
      " recipients a copy of this License along with the Program.\n",
      "\n",
      "   You may charge any price or no price for each copy that you convey,\n",
      " and you may offer support or warranty protection for a fee.\n",
      "\n",
      "   5. Conveying Modified Source Versions.\n",
      "\n",
      "   You may convey a work based on the Program, or the modifications to\n",
      " produce it from the Program, in the form of source code under the\n",
      " terms of section 4, provided that you also meet all of these conditions:\n",
      "\n",
      "     a) The work must carry prominent notices stating that you modified\n",
      "     it, and giving a relevant date.\n",
      "\n",
      "     b) The work must carry prominent notices stating that it is\n",
      "     released under this License and any conditions added under section\n",
      "     7.  This requirement modifies the requirement in section 4 to\n",
      "     \"keep intact all notices\".\n",
      "\n",
      "     c) You must license the entire work, as a whole, under this\n",
      "     License to anyone who comes into possession of a copy.  This\n",
      "     License will therefore apply, along with any applicable section 7\n",
      "     additional terms, to the whole of the work, and all its parts,\n",
      "     regardless of how they are packaged.  This License gives no\n",
      "     permission to license the work in any other way, but it does not\n",
      "     invalidate such permission if you have separately received it.\n",
      "\n",
      "     d) If the work has interactive user interfaces, each must display\n",
      "     Appropriate Legal Notices; however, if the Program has interactive\n",
      "     interfaces that do not display Appropriate Legal Notices, your\n",
      "     work need not make them do so.\n",
      "\n",
      "   A compilation of a covered work with other separate and independent\n",
      " works, which are not by their nature extensions of the covered work,\n",
      " and which are not combined with it such as to form a larger program,\n",
      " in or on a volume of a storage or distribution medium, is called an\n",
      " \"aggregate\" if the compilation and its resulting copyright are not\n",
      " used to limit the access or legal rights of the compilation's users\n",
      " beyond what the individual works permit.  Inclusion of a covered work\n",
      " in an aggregate does not cause this License to apply to the other\n",
      " parts of the aggregate.\n",
      "\n",
      "   6. Conveying Non-Source Forms.\n",
      "\n",
      "   You may convey a covered work in object code form under the terms\n",
      " of sections 4 and 5, provided that you also convey the\n",
      " machine-readable Corresponding Source under the terms of this License,\n",
      " in one of these ways:\n",
      "\n",
      "     a) Convey the object code in, or embodied in, a physical product\n",
      "     (including a physical distribution medium), accompanied by the\n",
      "     Corresponding Source fixed on a durable physical medium\n",
      "     customarily used for software interchange.\n",
      "\n",
      "     b) Convey the object code in, or embodied in, a physical product\n",
      "     (including a physical distribution medium), accompanied by a\n",
      "     written offer, valid for at least three years and valid for as\n",
      "     long as you offer spare parts or customer support for that product\n",
      "     model, to give anyone who possesses the object code either (1) a\n",
      "     copy of the Corresponding Source for all the software in the\n",
      "     product that is covered by this License, on a durable physical\n",
      "     medium customarily used for software interchange, for a price no\n",
      "     more than your reasonable cost of physically performing this\n",
      "     conveying of source, or (2) access to copy the\n",
      "     Corresponding Source from a network server at no charge.\n",
      "\n",
      "     c) Convey individual copies of the object code with a copy of the\n",
      "     written offer to provide the Corresponding Source.  This\n",
      "     alternative is allowed only occasionally and noncommercially, and\n",
      "     only if you received the object code with such an offer, in accord\n",
      "     with subsection 6b.\n",
      "\n",
      "     d) Convey the object code by offering access from a designated\n",
      "     place (gratis or for a charge), and offer equivalent access to the\n",
      "     Corresponding Source in the same way through the same place at no\n",
      "     further charge.  You need not require recipients to copy the\n",
      "     Corresponding Source along with the object code.  If the place to\n",
      "     copy the object code is a network server, the Corresponding Source\n",
      "     may be on a different server (operated by you or a third party)\n",
      "     that supports equivalent copying facilities, provided you maintain\n",
      "     clear directions next to the object code saying where to find the\n",
      "     Corresponding Source.  Regardless of what server hosts the\n",
      "     Corresponding Source, you remain obligated to ensure that it is\n",
      "     available for as long as needed to satisfy these requirements.\n",
      "\n",
      "     e) Convey the object code using peer-to-peer transmission, provided\n",
      "     you inform other peers where the object code and Corresponding\n",
      "     Source of the work are being offered to the general public at no\n",
      "     charge under subsection 6d.\n",
      "\n",
      "   A separable portion of the object code, whose source code is excluded\n",
      " from the Corresponding Source as a System Library, need not be\n",
      " included in conveying the object code work.\n",
      "\n",
      "   A \"User Product\" is either (1) a \"consumer product\", which means any\n",
      " tangible personal property which is normally used for personal, family,\n",
      " or household purposes, or (2) anything designed or sold for incorporation\n",
      " into a dwelling.  In determining whether a product is a consumer product,\n",
      " doubtful cases shall be resolved in favor of coverage.  For a particular\n",
      " product received by a particular user, \"normally used\" refers to a\n",
      " typical or common use of that class of product, regardless of the status\n",
      " of the particular user or of the way in which the particular user\n",
      " actually uses, or expects or is expected to use, the product.  A product\n",
      " is a consumer product regardless of whether the product has substantial\n",
      " commercial, industrial or non-consumer uses, unless such uses represent\n",
      " the only significant mode of use of the product.\n",
      "\n",
      "   \"Installation Information\" for a User Product means any methods,\n",
      " procedures, authorization keys, or other information required to install\n",
      " and execute modified versions of a covered work in that User Product from\n",
      " a modified version of its Corresponding Source.  The information must\n",
      " suffice to ensure that the continued functioning of the modified object\n",
      " code is in no case prevented or interfered with solely because\n",
      " modification has been made.\n",
      "\n",
      "   If you convey an object code work under this section in, or with, or\n",
      " specifically for use in, a User Product, and the conveying occurs as\n",
      " part of a transaction in which the right of possession and use of the\n",
      " User Product is transferred to the recipient in perpetuity or for a\n",
      " fixed term (regardless of how the transaction is characterized), the\n",
      " Corresponding Source conveyed under this section must be accompanied\n",
      " by the Installation Information.  But this requirement does not apply\n",
      " if neither you nor any third party retains the ability to install\n",
      " modified object code on the User Product (for example, the work has\n",
      " been installed in ROM).\n",
      "\n",
      "   The requirement to provide Installation Information does not include a\n",
      " requirement to continue to provide support service, warranty, or updates\n",
      " for a work that has been modified or installed by the recipient, or for\n",
      " the User Product in which it has been modified or installed.  Access to a\n",
      " network may be denied when the modification itself materially and\n",
      " adversely affects the operation of the network or violates the rules and\n",
      " protocols for communication across the network.\n",
      "\n",
      "   Corresponding Source conveyed, and Installation Information provided,\n",
      " in accord with this section must be in a format that is publicly\n",
      " documented (and with an implementation available to the public in\n",
      " source code form), and must require no special password or key for\n",
      " unpacking, reading or copying.\n",
      "\n",
      "   7. Additional Terms.\n",
      "\n",
      "   \"Additional permissions\" are terms that supplement the terms of this\n",
      " License by making exceptions from one or more of its conditions.\n",
      " Additional permissions that are applicable to the entire Program shall\n",
      " be treated as though they were included in this License, to the extent\n",
      " that they are valid under applicable law.  If additional permissions\n",
      " apply only to part of the Program, that part may be used separately\n",
      " under those permissions, but the entire Program remains governed by\n",
      " this License without regard to the additional permissions.\n",
      "\n",
      "   When you convey a copy of a covered work, you may at your option\n",
      " remove any additional permissions from that copy, or from any part of\n",
      " it.  (Additional permissions may be written to require their own\n",
      " removal in certain cases when you modify the work.)  You may place\n",
      " additional permissions on material, added by you to a covered work,\n",
      " for which you have or can give appropriate copyright permission.\n",
      "\n",
      "   Notwithstanding any other provision of this License, for material you\n",
      " add to a covered work, you may (if authorized by the copyright holders of\n",
      " that material) supplement the terms of this License with terms:\n",
      "\n",
      "     a) Disclaiming warranty or limiting liability differently from the\n",
      "     terms of sections 15 and 16 of this License; or\n",
      "\n",
      "     b) Requiring preservation of specified reasonable legal notices or\n",
      "     author attributions in that material or in the Appropriate Legal\n",
      "     Notices displayed by works containing it; or\n",
      "\n",
      "     c) Prohibiting misrepresentation of the origin of that material, or\n",
      "     requiring that modified versions of such material be marked in\n",
      "     reasonable ways as different from the original version; or\n",
      "\n",
      "     d) Limiting the use for publicity purposes of names of licensors or\n",
      "     authors of the material; or\n",
      "\n",
      "     e) Declining to grant rights under trademark law for use of some\n",
      "     trade names, trademarks, or service marks; or\n",
      "\n",
      "     f) Requiring indemnification of licensors and authors of that\n",
      "     material by anyone who conveys the material (or modified versions of\n",
      "     it) with contractual assumptions of liability to the recipient, for\n",
      "     any liability that these contractual assumptions directly impose on\n",
      "     those licensors and authors.\n",
      "\n",
      "   All other non-permissive additional terms are considered \"further\n",
      " restrictions\" within the meaning of section 10.  If the Program as you\n",
      " received it, or any part of it, contains a notice stating that it is\n",
      " governed by this License along with a term that is a further\n",
      " restriction, you may remove that term.  If a license document contains\n",
      " a further restriction but permits relicensing or conveying under this\n",
      " License, you may add to a covered work material governed by the terms\n",
      " of that license document, provided that the further restriction does\n",
      " not survive such relicensing or conveying.\n",
      "\n",
      "   If you add terms to a covered work in accord with this section, you\n",
      " must place, in the relevant source files, a statement of the\n",
      " additional terms that apply to those files, or a notice indicating\n",
      " where to find the applicable terms.\n",
      "\n",
      "   Additional terms, permissive or non-permissive, may be stated in the\n",
      " form of a separately written license, or stated as exceptions;\n",
      " the above requirements apply either way.\n",
      "\n",
      "   8. Termination.\n",
      "\n",
      "   You may not propagate or modify a covered work except as expressly\n",
      " provided under this License.  Any attempt otherwise to propagate or\n",
      " modify it is void, and will automatically terminate your rights under\n",
      " this License (including any patent licenses granted under the third\n",
      " paragraph of section 11).\n",
      "\n",
      "   However, if you cease all violation of this License, then your\n",
      " license from a particular copyright holder is reinstated (a)\n",
      " provisionally, unless and until the copyright holder explicitly and\n",
      " finally terminates your license, and (b) permanently, if the copyright\n",
      " holder fails to notify you of the violation by some reasonable means\n",
      " prior to 60 days after the cessation.\n",
      "\n",
      "   Moreover, your license from a particular copyright holder is\n",
      " reinstated permanently if the copyright holder notifies you of the\n",
      " violation by some reasonable means, this is the first time you have\n",
      " received notice of violation of this License (for any work) from that\n",
      " copyright holder, and you cure the violation prior to 30 days after\n",
      " your receipt of the notice.\n",
      "\n",
      "   Termination of your rights under this section does not terminate the\n",
      " licenses of parties who have received copies or rights from you under\n",
      " this License.  If your rights have been terminated and not permanently\n",
      " reinstated, you do not qualify to receive new licenses for the same\n",
      " material under section 10.\n",
      "\n",
      "   9. Acceptance Not Required for Having Copies.\n",
      "\n",
      "   You are not required to accept this License in order to receive or\n",
      " run a copy of the Program.  Ancillary propagation of a covered work\n",
      " occurring solely as a consequence of using peer-to-peer transmission\n",
      " to receive a copy likewise does not require acceptance.  However,\n",
      " nothing other than this License grants you permission to propagate or\n",
      " modify any covered work.  These actions infringe copyright if you do\n",
      " not accept this License.  Therefore, by modifying or propagating a\n",
      " covered work, you indicate your acceptance of this License to do so.\n",
      "\n",
      "   10. Automatic Licensing of Downstream Recipients.\n",
      "\n",
      "   Each time you convey a covered work, the recipient automatically\n",
      " receives a license from the original licensors, to run, modify and\n",
      " propagate that work, subject to this License.  You are not responsible\n",
      " for enforcing compliance by third parties with this License.\n",
      "\n",
      "   An \"entity transaction\" is a transaction transferring control of an\n",
      " organization, or substantially all assets of one, or subdividing an\n",
      " organization, or merging organizations.  If propagation of a covered\n",
      " work results from an entity transaction, each party to that\n",
      " transaction who receives a copy of the work also receives whatever\n",
      " licenses to the work the party's predecessor in interest had or could\n",
      " give under the previous paragraph, plus a right to possession of the\n",
      " Corresponding Source of the work from the predecessor in interest, if\n",
      " the predecessor has it or can get it with reasonable efforts.\n",
      "\n",
      "   You may not impose any further restrictions on the exercise of the\n",
      " rights granted or affirmed under this License.  For example, you may\n",
      " not impose a license fee, royalty, or other charge for exercise of\n",
      " rights granted under this License, and you may not initiate litigation\n",
      " (including a cross-claim or counterclaim in a lawsuit) alleging that\n",
      " any patent claim is infringed by making, using, selling, offering for\n",
      " sale, or importing the Program or any portion of it.\n",
      "\n",
      "   11. Patents.\n",
      "\n",
      "   A \"contributor\" is a copyright holder who authorizes use under this\n",
      " License of the Program or a work on which the Program is based.  The\n",
      " work thus licensed is called the contributor's \"contributor version\".\n",
      "\n",
      "   A contributor's \"essential patent claims\" are all patent claims\n",
      " owned or controlled by the contributor, whether already acquired or\n",
      " hereafter acquired, that would be infringed by some manner, permitted\n",
      " by this License, of making, using, or selling its contributor version,\n",
      " but do not include claims that would be infringed only as a\n",
      " consequence of further modification of the contributor version.  For\n",
      " purposes of this definition, \"control\" includes the right to grant\n",
      " patent sublicenses in a manner consistent with the requirements of\n",
      " this License.\n",
      "\n",
      "   Each contributor grants you a non-exclusive, worldwide, royalty-free\n",
      " patent license under the contributor's essential patent claims, to\n",
      " make, use, sell, offer for sale, import and otherwise run, modify and\n",
      " propagate the contents of its contributor version.\n",
      "\n",
      "   In the following three paragraphs, a \"patent license\" is any express\n",
      " agreement or commitment, however denominated, not to enforce a patent\n",
      " (such as an express permission to practice a patent or covenant not to\n",
      " sue for patent infringement).  To \"grant\" such a patent license to a\n",
      " party means to make such an agreement or commitment not to enforce a\n",
      " patent against the party.\n",
      "\n",
      "   If you convey a covered work, knowingly relying on a patent license,\n",
      " and the Corresponding Source of the work is not available for anyone\n",
      " to copy, free of charge and under the terms of this License, through a\n",
      " publicly available network server or other readily accessible means,\n",
      " then you must either (1) cause the Corresponding Source to be so\n",
      " available, or (2) arrange to deprive yourself of the benefit of the\n",
      " patent license for this particular work, or (3) arrange, in a manner\n",
      " consistent with the requirements of this License, to extend the patent\n",
      " license to downstream recipients.  \"Knowingly relying\" means you have\n",
      " actual knowledge that, but for the patent license, your conveying the\n",
      " covered work in a country, or your recipient's use of the covered work\n",
      " in a country, would infringe one or more identifiable patents in that\n",
      " country that you have reason to believe are valid.\n",
      "\n",
      "   If, pursuant to or in connection with a single transaction or\n",
      " arrangement, you convey, or propagate by procuring conveyance of, a\n",
      " covered work, and grant a patent license to some of the parties\n",
      " receiving the covered work authorizing them to use, propagate, modify\n",
      " or convey a specific copy of the covered work, then the patent license\n",
      " you grant is automatically extended to all recipients of the covered\n",
      " work and works based on it.\n",
      "\n",
      "   A patent license is \"discriminatory\" if it does not include within\n",
      " the scope of its coverage, prohibits the exercise of, or is\n",
      " conditioned on the non-exercise of one or more of the rights that are\n",
      " specifically granted under this License.  You may not convey a covered\n",
      " work if you are a party to an arrangement with a third party that is\n",
      " in the business of distributing software, under which you make payment\n",
      " to the third party based on the extent of your activity of conveying\n",
      " the work, and under which the third party grants, to any of the\n",
      " parties who would receive the covered work from you, a discriminatory\n",
      " patent license (a) in connection with copies of the covered work\n",
      " conveyed by you (or copies made from those copies), or (b) primarily\n",
      " for and in connection with specific products or compilations that\n",
      " contain the covered work, unless you entered into that arrangement,\n",
      " or that patent license was granted, prior to 28 March 2007.\n",
      "\n",
      "   Nothing in this License shall be construed as excluding or limiting\n",
      " any implied license or other defenses to infringement that may\n",
      " otherwise be available to you under applicable patent law.\n",
      "\n",
      "   12. No Surrender of Others' Freedom.\n",
      "\n",
      "   If conditions are imposed on you (whether by court order, agreement or\n",
      " otherwise) that contradict the conditions of this License, they do not\n",
      " excuse you from the conditions of this License.  If you cannot convey a\n",
      " covered work so as to satisfy simultaneously your obligations under this\n",
      " License and any other pertinent obligations, then as a consequence you may\n",
      " not convey it at all.  For example, if you agree to terms that obligate you\n",
      " to collect a royalty for further conveying from those to whom you convey\n",
      " the Program, the only way you could satisfy both those terms and this\n",
      " License would be to refrain entirely from conveying the Program.\n",
      "\n",
      "   13. Use with the GNU Affero General Public License.\n",
      "\n",
      "   Notwithstanding any other provision of this License, you have\n",
      " permission to link or combine any covered work with a work licensed\n",
      " under version 3 of the GNU Affero General Public License into a single\n",
      " combined work, and to convey the resulting work.  The terms of this\n",
      " License will continue to apply to the part which is the covered work,\n",
      " but the special requirements of the GNU Affero General Public License,\n",
      " section 13, concerning interaction through a network will apply to the\n",
      " combination as such.\n",
      "\n",
      "   14. Revised Versions of this License.\n",
      "\n",
      "   The Free Software Foundation may publish revised and/or new versions of\n",
      " the GNU General Public License from time to time.  Such new versions will\n",
      " be similar in spirit to the present version, but may differ in detail to\n",
      " address new problems or concerns.\n",
      "\n",
      "   Each version is given a distinguishing version number.  If the\n",
      " Program specifies that a certain numbered version of the GNU General\n",
      " Public License \"or any later version\" applies to it, you have the\n",
      " option of following the terms and conditions either of that numbered\n",
      " version or of any later version published by the Free Software\n",
      " Foundation.  If the Program does not specify a version number of the\n",
      " GNU General Public License, you may choose any version ever published\n",
      " by the Free Software Foundation.\n",
      "\n",
      "   If the Program specifies that a proxy can decide which future\n",
      " versions of the GNU General Public License can be used, that proxy's\n",
      " public statement of acceptance of a version permanently authorizes you\n",
      " to choose that version for the Program.\n",
      "\n",
      "   Later license versions may give you additional or different\n",
      " permissions.  However, no additional obligations are imposed on any\n",
      " author or copyright holder as a result of your choosing to follow a\n",
      " later version.\n",
      "\n",
      "   15. Disclaimer of Warranty.\n",
      "\n",
      "   THERE IS NO WARRANTY FOR THE PROGRAM, TO THE EXTENT PERMITTED BY\n",
      " APPLICABLE LAW.  EXCEPT WHEN OTHERWISE STATED IN WRITING THE COPYRIGHT\n",
      " HOLDERS AND/OR OTHER PARTIES PROVIDE THE PROGRAM \"AS IS\" WITHOUT WARRANTY\n",
      " OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO,\n",
      " THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR\n",
      " PURPOSE.  THE ENTIRE RISK AS TO THE QUALITY AND PERFORMANCE OF THE PROGRAM\n",
      " IS WITH YOU.  SHOULD THE PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF\n",
      " ALL NECESSARY SERVICING, REPAIR OR CORRECTION.\n",
      "\n",
      "   16. Limitation of Liability.\n",
      "\n",
      "   IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING\n",
      " WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MODIFIES AND/OR CONVEYS\n",
      " THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES, INCLUDING ANY\n",
      " GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING OUT OF THE\n",
      " USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED TO LOSS OF\n",
      " DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY YOU OR THIRD\n",
      " PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER PROGRAMS),\n",
      " EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF\n",
      " SUCH DAMAGES.\n",
      "\n",
      "   17. Interpretation of Sections 15 and 16.\n",
      "\n",
      "   If the disclaimer of warranty and limitation of liability provided\n",
      " above cannot be given local legal effect according to their terms,\n",
      " reviewing courts shall apply local law that most closely approximates\n",
      " an absolute waiver of all civil liability in connection with the\n",
      " Program, unless a warranty or assumption of liability accompanies a\n",
      " copy of the Program in return for a fee.\n",
      "\n",
      "                      END OF TERMS AND CONDITIONS\n",
      "\n",
      "             How to Apply These Terms to Your New Programs\n",
      "\n",
      "   If you develop a new program, and you want it to be of the greatest\n",
      " possible use to the public, the best way to achieve this is to make it\n",
      " free software which everyone can redistribute and change under these terms.\n",
      "\n",
      "   To do so, attach the following notices to the program.  It is safest\n",
      " to attach them to the start of each source file to most effectively\n",
      " state the exclusion of warranty; and each file should have at least\n",
      " the \"copyright\" line and a pointer to where the full notice is found.\n",
      "\n",
      "     <one line to give the program's name and a brief idea of what it does.>\n",
      "     Copyright (C) <year>  <name of author>\n",
      "\n",
      "     This program is free software: you can redistribute it and/or modify\n",
      "     it under the terms of the GNU General Public License as published by\n",
      "     the Free Software Foundation, either version 3 of the License, or\n",
      "     (at your option) any later version.\n",
      "\n",
      "     This program is distributed in the hope that it will be useful,\n",
      "     but WITHOUT ANY WARRANTY; without even the implied warranty of\n",
      "     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n",
      "     GNU General Public License for more details.\n",
      "\n",
      "     You should have received a copy of the GNU General Public License\n",
      "     along with this program.  If not, see <https://www.gnu.org/licenses/>.\n",
      "\n",
      " Also add information on how to contact you by electronic and paper mail.\n",
      "\n",
      "   If the program does terminal interaction, make it output a short\n",
      " notice like this when it starts in an interactive mode:\n",
      "\n",
      "     <program>  Copyright (C) <year>  <name of author>\n",
      "     This program comes with ABSOLUTELY NO WARRANTY; for details type `show w'.\n",
      "     This is free software, and you are welcome to redistribute it\n",
      "     under certain conditions; type `show c' for details.\n",
      "\n",
      " The hypothetical commands `show w' and `show c' should show the appropriate\n",
      " parts of the General Public License.  Of course, your program's commands\n",
      " might be different; for a GUI interface, you would use an \"about box\".\n",
      "\n",
      "   You should also get your employer (if you work as a programmer) or school,\n",
      " if any, to sign a \"copyright disclaimer\" for the program, if necessary.\n",
      " For more information on this, and how to apply and follow the GNU GPL, see\n",
      " <https://www.gnu.org/licenses/>.\n",
      "\n",
      "   The GNU General Public License does not permit incorporating your program\n",
      " into proprietary programs.  If your program is a subroutine library, you\n",
      " may consider it more useful to permit linking proprietary applications with\n",
      " the library.  If this is what you want to do, use the GNU Lesser General\n",
      " Public License instead of this License.  But first, please read\n",
      " <https://www.gnu.org/licenses/why-not-lgpl.html>.\n",
      "\n",
      "\n",
      "Location: c:\\Data\\Projects\\make-data-count-classifier\\.venv\\Lib\\site-packages\n",
      "Requires: \n",
      "Required-by: accelerate, bitsandbytes, blis, datasets, fitz, nibabel, nipype, pandas, peft, scikit-learn, scipy, spacy, thinc, torchvision, transformers, xformers\n"
     ]
    }
   ],
   "source": [
    "# !pip show transformers\n",
    "!pip show numpy\n",
    "\n",
    "# Name: transformers\n",
    "# Version: 4.52.4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "62ca2607",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CUDA is available! Using GPU: NVIDIA GeForce RTX 3050 Laptop GPU\n"
     ]
    }
   ],
   "source": [
    "# 1.1. Install necessary libraries\n",
    "# Use !pip install for notebook environment\n",
    "# !pip install transformers trl accelerate bitsandbytes sentencepiece lxml PyMuPDF spacy peft\n",
    "# !python -m spacy download en_core_web_sm # Download a small spaCy model\n",
    "\n",
    "# 1.2. Import Libraries\n",
    "import os\n",
    "import re\n",
    "import json\n",
    "import pandas as pd\n",
    "from dataclasses import dataclass, field, asdict\n",
    "from typing import Set, List, Optional, Dict, Any\n",
    "\n",
    "import fitz # PyMuPDF\n",
    "from lxml import etree # For XML parsing\n",
    "import spacy\n",
    "import kagglehub\n",
    "\n",
    "import torch\n",
    "from transformers import AutoTokenizer, AutoModelForCausalLM\n",
    "from transformers.utils.quantization_config import BitsAndBytesConfig\n",
    "from trl import SFTTrainer, SFTConfig\n",
    "from datasets import Dataset, concatenate_datasets\n",
    "from peft import LoraConfig, PeftModel, prepare_model_for_kbit_training\n",
    "\n",
    "from tqdm.auto import tqdm\n",
    "import gc # For garbage collection\n",
    "\n",
    "# For KaggleHub integration (assuming it's set up or models are downloaded)\n",
    "# You might need to install kagglehub if you plan to use it directly for model download\n",
    "# !pip install kagglehub\n",
    "\n",
    "# 1.3. Configure CUDA for local GPU\n",
    "if torch.cuda.is_available():\n",
    "    print(f\"CUDA is available! Using GPU: {torch.cuda.get_device_name(0)}\")\n",
    "    device = torch.device(\"cuda\")\n",
    "    torch.cuda.empty_cache() # Clear GPU memory\n",
    "else:\n",
    "    print(\"CUDA is not available. Using CPU.\")\n",
    "    device = torch.device(\"cpu\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c1ab706e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import classes from local utility file\n",
    "import mdc_data_processing_utils\n",
    "\n",
    "# If mdc_data_processing_utils.py has been changed and saved.\n",
    "# To load the changes without restarting the kernel:\n",
    "import importlib\n",
    "importlib.reload(mdc_data_processing_utils)\n",
    "\n",
    "# Now, any calls to functions from mdc_data_processing_utils\n",
    "# will use the newly reloaded code.\n",
    "from mdc_data_processing_utils import (\n",
    "    ArticleData,\n",
    "    DatasetCitation,\n",
    "    LlmTrainingData,\n",
    "    SubmissionData,\n",
    "    MdcFileTextExtractor,\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b7b02acd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define constants for file paths and model configurations\n",
    "BASE_INPUT_DIR = './kaggle/input/make-data-count-finding-data-references'\n",
    "BASE_OUTPUT_DIR = \"./kaggle/working\"\n",
    "\n",
    "# Define directories for articles in train and test sets\n",
    "TRAIN_DATA_DIR = os.path.join(BASE_INPUT_DIR, 'train')\n",
    "TEST_DATA_DIR = os.path.join(BASE_INPUT_DIR, 'test')\n",
    "TRAIN_LABELS_PATH = os.path.join(BASE_INPUT_DIR, 'train_labels.csv')\n",
    "\n",
    "# Define the path to the few-shot examples CSV\n",
    "FEW_SHOT_CSV_PATH = os.path.join(BASE_OUTPUT_DIR, \"few_shot_examples.csv\")\n",
    "\n",
    "# Define the base model path\n",
    "QWEN_BASE_MODEL_PATH = kagglehub.model_download(\"qwen-lm/qwen-3/transformers/0.6b\")\n",
    "\n",
    "# Output directory for the fine-tuned model and results\n",
    "FINE_TUNED_MODEL_OUTPUT_DIR = os.path.join(BASE_OUTPUT_DIR, \"results\")\n",
    "SAMPLE_SUBMISSION_PATH = os.path.join(BASE_OUTPUT_DIR, \"submission.csv\")\n",
    "\n",
    "# Load spaCy model for sentence segmentation and potentially other NLP tasks\n",
    "# python -m spacy download en_core_web_sm \n",
    "NLP_SPACY = spacy.load(\"en_core_web_sm\")\n",
    "\n",
    "FEW_SHOT_EXAMPLES = \"\"\"\n",
    "### Example 1 (Primary)\n",
    "Article Abstract: We present a novel dataset of forest growth measurements collected over 5 years in the Amazon rainforest. This data was used to develop a new model of carbon sequestration.\n",
    "Dataset ID: 10.5061/dryad.r6nq870\n",
    "Data Citation Context: The raw data for this study, including all measurements and derived variables, has been deposited in the Dryad Digital Repository (dryad.2f62927). This newly generated dataset supports the findings of our research.\n",
    "Classification: Primary\n",
    "\n",
    "### Example 2 (Primary - Another Example)\n",
    "Article Abstract: We developed a new computational model for protein folding and generated a large dataset of simulated protein structures.\n",
    "Dataset ID: PDB12345\n",
    "Data Citation Context: The simulated protein structures (PDB12345) are available in the Protein Data Bank and were generated as part of this study's novel computational approach.\n",
    "Classification: Primary\n",
    "\n",
    "### Example 3 (Primary - Ambiguous Case)\n",
    "Article Abstract: This study utilized both newly collected field data and some previously published climate data to analyze ecosystem changes.\n",
    "Dataset ID: 10.5281/zenodo.7074790\n",
    "Data Citation Context: Data Availability Statement The data that support the findings of this study are openly available in zenodo at https://doi.org/10.5281/zenodo.7074790.(https://doi.org/10.5281/zenodo.7074790).\n",
    "Classification: Primary\n",
    "\n",
    "### Example 4 (Secondary)\n",
    "Article Abstract: We re-analyzed publicly available gene expression data to identify new biomarkers for cancer.\n",
    "Dataset ID: GSE12345\n",
    "Data Citation Context: We downloaded the gene expression profiles from the GEO database (GSE12345), which were previously published by Smith et al. (2020). This existing dataset was re-analyzed for our current study.\n",
    "Classification: Secondary\n",
    "\n",
    "### Example 5 (Missing - Irrelevant Context)\n",
    "Article Abstract: Implications for intraplate earthquake behavior and the geomorphic longevity of bedrock fault scarps in a low strain-rate cratonic region.\n",
    "Dataset ID: 10.1080/08120090802546977\n",
    "Data Citation Context: (2009) Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low-T thermochronological data. Australian Journal of Earth Sciences, 56(2), 99-110. https://doi.org/(https://doi.org/10.1080/08120090802546977).\n",
    "Classification: Missing\n",
    "\n",
    "### Example 6 (Missing - No context)\n",
    "Article Abstract: Our study investigates the social dynamics of ant colonies.\n",
    "Dataset ID: 10.1038/s41586-023-06000-0\n",
    "Data Citation Context: \n",
    "Classification: Missing\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f5b4259",
   "metadata": {},
   "source": [
    "#### 3. Data Loading and Initial Preprocessing\n",
    "\n",
    "This section will cover how to load the raw competition data (full text articles and labels) and begin structuring it."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09a5d12b",
   "metadata": {},
   "source": [
    "#### Load Labeled Training Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "438f57fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading labeled training data from: ./kaggle/input/make-data-count-finding-data-references\\train_labels.csv\n",
      "Training labels shape: (1028, 3)\n",
      "Example grouped training data for article_id '10.1002_2017jc013030': [{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]\n",
      "Files paths shape: (262, 5)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ground_truth_dataset_info",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "f4c8cda8-73ca-4e39-ba27-50b2ea0d4f85",
       "rows": [
        [
         "17",
         "10.1107_s2052252514012081",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252514012081.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252514012081.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]"
        ],
        [
         "158",
         "10.3897_neobiota.82.87455",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_neobiota.82.87455.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_neobiota.82.87455.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.15468/dl.yjd4cg', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.15468/dl.yu2xpv', 'type': 'Secondary'}]"
        ],
        [
         "240",
         "10.12688_f1000research.13483.1",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13483.1.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13483.1.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5256/f1000research.13483.d188591', 'type': 'Primary'}]"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>ground_truth_dataset_info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10.1107_s2052252514012081</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>10.3897_neobiota.82.87455</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.15468/dl.y...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>10.12688_f1000research.13483.1</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5256/f1000...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         article_id  \\\n",
       "17        10.1107_s2052252514012081   \n",
       "158       10.3897_neobiota.82.87455   \n",
       "240  10.12688_f1000research.13483.1   \n",
       "\n",
       "                                         pdf_file_path  \\\n",
       "17   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "158  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "240  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                         xml_file_path dataset_type  \\\n",
       "17   ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "158  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "240  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                             ground_truth_dataset_info  \n",
       "17      [{'dataset_id': 'Missing', 'type': 'Missing'}]  \n",
       "158  [{'dataset_id': 'https://doi.org/10.15468/dl.y...  \n",
       "240  [{'dataset_id': 'https://doi.org/10.5256/f1000...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "def load_file_paths(dataset_type_dir: str) -> pd.DataFrame: \n",
    "    pdf_path = os.path.join(dataset_type_dir, 'PDF')\n",
    "    xml_path = os.path.join(dataset_type_dir, 'XML')\n",
    "    dataset_type = os.path.basename(dataset_type_dir)\n",
    "    pdf_files = [f for f in os.listdir(pdf_path) if f.endswith('.pdf')]\n",
    "    xml_files = [f for f in os.listdir(xml_path) if f.endswith('.xml')]\n",
    "    df_pdf = pd.DataFrame({\n",
    "        'article_id': [f.replace('.pdf', '') for f in pdf_files],\n",
    "        'pdf_file_path': [os.path.join(pdf_path, f) for f in pdf_files]\n",
    "    })\n",
    "    df_xml = pd.DataFrame({\n",
    "        'article_id': [f.replace('.xml', '') for f in xml_files],\n",
    "        'xml_file_path': [os.path.join(xml_path, f) for f in xml_files]\n",
    "    })\n",
    "    merge_df = pd.merge(df_pdf, df_xml, on='article_id', how='outer', suffixes=('_pdf', '_xml'), validate=\"one_to_many\")\n",
    "    merge_df['dataset_type'] = dataset_type\n",
    "    return merge_df\n",
    "\n",
    "# Load the labeled training data CSV file\n",
    "print(f\"Loading labeled training data from: {TRAIN_LABELS_PATH}\")\n",
    "train_labels_df = pd.read_csv(TRAIN_LABELS_PATH)\n",
    "print(f\"Training labels shape: {train_labels_df.shape}\")\n",
    "\n",
    "# Group training data by article_id to get all datasets for each article\n",
    "# This creates a dictionary where keys are article_ids and values are lists of dataset dicts\n",
    "grouped_training_data = {}\n",
    "for article_id, group_df in train_labels_df.groupby('article_id'):\n",
    "    grouped_training_data[article_id] = group_df[['dataset_id', 'type']].to_dict('records')\n",
    "\n",
    "# Example usage of grouped_training_data\n",
    "print(f\"Example grouped training data for article_id '10.1002_2017jc013030': {grouped_training_data['10.1002_2017jc013030']}\")\n",
    "\n",
    "# Just for testing, always set to the TEST_DATA_DIR\n",
    "base_file_dir = TRAIN_DATA_DIR\n",
    "\n",
    "# Load file paths for base directory\n",
    "file_paths_df = load_file_paths(base_file_dir)\n",
    "file_paths_df['pdf_file_path'] = file_paths_df['pdf_file_path'].fillna('')\n",
    "file_paths_df['xml_file_path'] = file_paths_df['xml_file_path'].fillna('')\n",
    "file_paths_df = file_paths_df[file_paths_df['article_id'] != '10.20944_preprints202009.0353.v1']\n",
    "\n",
    "# Merge the file paths with the grouped_training_data\n",
    "file_paths_df['ground_truth_dataset_info'] = file_paths_df['article_id'].map(grouped_training_data)\n",
    "file_paths_df['ground_truth_dataset_info'] = file_paths_df['ground_truth_dataset_info'].fillna('')\n",
    "\n",
    "# Reduce the file paths DataFrame to only those with ground truth dataset info and get a sample\n",
    "# This is to ensure we have a manageable dataset for training\n",
    "file_paths_df = file_paths_df[file_paths_df['ground_truth_dataset_info'].astype(bool)]\n",
    "file_paths_df = file_paths_df.reset_index(drop=True)\n",
    "file_paths_df = file_paths_df.sample(frac=.5, random_state=42).reset_index(drop=True)  # Shuffle the DataFrame\n",
    "print(f\"Files paths shape: {file_paths_df.shape}\")\n",
    "display(file_paths_df.sample(3))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a2f0763",
   "metadata": {},
   "source": [
    "#### Define Training Data Extract Function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8d2b2277",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_training_data_for_llm(file_paths_df: pd.DataFrame) -> list[dict[str, str]]:\n",
    "    \"\"\"\n",
    "    Extracts article data for training set with ground truth.\n",
    "    \n",
    "    Args:\n",
    "        file_paths_df (pd.DataFrame): DataFrame containing file paths and ground truth info.\n",
    "        \n",
    "    Returns:\n",
    "        Dict[str, ArticleData]: Dictionary mapping article IDs to ArticleData objects.\n",
    "    \"\"\"\n",
    "    training_data_for_llm: list[dict[str, str]] = [] # This will be a list of LlmTrainingData for the LLM training dataset\n",
    "    for i, row in tqdm(file_paths_df.iterrows(), total=len(file_paths_df)):\n",
    "        article_id = row['article_id']\n",
    "        filepath = row['pdf_file_path'] if row['pdf_file_path'] else row['xml_file_path']\n",
    "        ground_truth_list = row['ground_truth_dataset_info'] if 'ground_truth_dataset_info' in row else []\n",
    "\n",
    "        file_extractor = MdcFileTextExtractor(article_id, filepath)\n",
    "        article_data = file_extractor.extract_article_data_for_training(NLP_SPACY, ground_truth_list)\n",
    "        training_data_for_llm.extend(article_data.get_data_for_llm())\n",
    "\n",
    "    print(f\"Loaded training data for {len(training_data_for_llm)} articles.\")\n",
    "    return training_data_for_llm\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3aeda081",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "ground_truth_dataset_info",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "deb41cdf-991d-4258-afa4-8dc623421de7",
       "rows": [
        [
         "154",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 1
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>ground_truth_dataset_info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>10.1002_esp.5058</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           article_id                                      pdf_file_path  \\\n",
       "154  10.1002_esp.5058  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "    xml_file_path dataset_type  \\\n",
       "154                      train   \n",
       "\n",
       "                             ground_truth_dataset_info  \n",
       "154  [{'dataset_id': 'https://doi.org/10.5061/dryad...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For testing, let's extract training data for a specific article\n",
    "sample_file_paths_df = file_paths_df.loc[file_paths_df['article_id'] == '10.1002_esp.5058']\n",
    "sample_file_paths_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef756a4e",
   "metadata": {},
   "source": [
    "#### 4. Advanced Preprocessing: Extracting Dataset Mentions and Context (Training)\n",
    "\n",
    "Use regex to find the given dataset IDs from the training_labels and then use spaCy to extract surrounding sentences as context.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5f23263",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "41c96cc841b8426abe8543c2a6429f3d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/262 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.12422.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0198382.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.12453.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-019-0101-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12974-020-01860-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.13193.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s23177333.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_22221751.2020.1738277.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1103_physrevresearch.4.023008.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_tc-17-3617-2023.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-019-10357-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.033.ao15.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules191017026.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536808011148.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252514012081.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1364_oe.25.001985.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.1498.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13046-018-0843-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1414-431x20198292.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-020-00737-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12916-019-1469-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13072-019-0322-5.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0426-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12868-018-0468-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20231027.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0921.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-8-663-2016.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.74937.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acsomega.3c06074.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5937_bnhmb1811227u.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-018-4014-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-04041-x.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.29944.xml\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2014.75.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-022-01555-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1055_s-0039-1693681.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1711872115.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12913-018-3333-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.2726.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.032.ao27.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-015-0209-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41437-020-0318-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2414-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40170-020-00212-x.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41562-021-01247-w.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.10452.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9sc02930c.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1677-5538.ibju.2019.0167.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.63194.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s11689-019-9287-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjds_s13688-018-0132-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_sysbio_syy011.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13136.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20201035.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13617-019-0084-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-020-00727-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-019-1388-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1908-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12931-019-1001-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_ncomms11871.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.14379_iodp.proc.390393.208.2024.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s160053681103220x.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-024-56373-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021pa004379.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pcbi.1011828.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9ra06638a.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6324-7.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12501.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41558-022-01301-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-020-59839-x.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-015-0509-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-018-1542-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-90882019000500004.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2036-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.63455.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515023945.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-93042019000400002.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13446.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1316-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0001-3765201920180768.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_ijms12117360.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01197e.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616000523.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-019-6583-0.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.11698.1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13750-020-0184-0.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0253228.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0159387.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-4060.59642.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13007-019-0403-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-2950_19008627012020.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812027390.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0922-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2015.99.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40657-020-00194-w.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0266-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13431.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1617_s11527-023-02260-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12860-020-00261-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1145_3461702.3462538.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12151.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13100-019-0153-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1167-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-1206-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812046892.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4499.20190067.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12491.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_dev.138545.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102.3772e35417.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1130_ges01387.1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-019-4127-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12915-018-0498-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.18438_eblip29674.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40851-018-0089-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_sdata.2017.167.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fchem.2019.00828.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_evo.13972.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0047-2085000000239.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13568-018-0680-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_jacs.2c06519.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13064.1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40793-015-0095-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.17581_bp.2020.09104.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-019-1542-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1678-86212019000400340.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1924-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-2295-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13087.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0646-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.160417.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13148-019-0719-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_14756366.2020.1740692.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-020-3415-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2263-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13643-018-0859-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2018gl078007.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_amt-15-3969-2022.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12879-019-3766-0.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2023wr035126.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_neobiota.82.87455.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fcimb.2024.1292467.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01518k.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1519-6984.192126.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-018-0550-0.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-020-01863-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4768-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0284951.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_21645515.2023.2189598.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-2021-570.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0611-7.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3762_bjoc.8.42.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.4660.1.pdf\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_d13010019.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_fst.33717.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2020jf005675.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2020.1714721.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0674-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-020-06724-5.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_v11060565.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.13817.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13168.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s19030479.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_02713683.2019.1607392.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515011665.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-12-1287-2020.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arad016.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_15476286.2016.1232238.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4314-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12382.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-018-6468-7.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1128_spectrum.00422-24.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810036299.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12768.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0188323.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_g3.119.400993.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-9479.2019.185555.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.36.e32053.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-015-0110-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fevo.2023.1112519.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809014883.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0139215.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2013.74.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3847_1538-4357_aae92c.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616007033.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_cas.12935.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zookeys.500.9360.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0262974.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12863-019-0790-4.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12446.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-017-15852-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2019.1669817.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.72626.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810047185.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0gc00363h.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812024269.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6131-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-6632.20190362s20180340.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-187.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-021-85671-y.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1705601114.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989020010658.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989015019891.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_mec.16743.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1889-5.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-07681-1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1078-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-015-2206-9.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.35.e23481.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arw167.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-22-5701-2022.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13058-015-0618-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12881-019-0773-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13483.1.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_2236-3459_83528.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_bdj.7.e47369.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0254-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0212669.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-198.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0766.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0709-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-3237-2.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_gcb.13914.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536807066780.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0137181.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12874-018-0583-x.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12943-019-1017-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12967-019-2100-3.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2019pa003774.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_07350015.2020.1766469.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12903-018-0656-6.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12884-018-1751-z.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0070749.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812014614.pdf\n",
      "Loaded training data for 324 articles.\n",
      "Prepared 324 training examples for the LLM.\n",
      "Training set size: 291 examples\n",
      "Validation set size: 33 examples\n"
     ]
    }
   ],
   "source": [
    "# This take 10+ minutes\n",
    "\n",
    "# 4.3. Populate ArticleData with DatasetCitation objects and ground truth\n",
    "training_data_for_llm = extract_training_data_for_llm(file_paths_df)\n",
    "print(f\"Prepared {len(training_data_for_llm)} training examples for the LLM.\")\n",
    "\n",
    "# Convert the list of LlmTrainingData to a DataFrame and save it\n",
    "training_data_for_llm_df = pd.DataFrame(training_data_for_llm)\n",
    "training_data_for_llm_df.to_csv(os.path.join(BASE_OUTPUT_DIR, \"training_data_for_llm.csv\"), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a2f99b3b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Training set size: 261 examples\n",
      "Validation set size: 29 examples\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "952"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def last_of_string(s: str, length: int = 400) -> str:\n",
    "    return s[-length:]\n",
    "\n",
    "# Load data from the csv file\n",
    "training_data_for_llm_df = pd.read_csv(os.path.join(BASE_OUTPUT_DIR, \"training_data_for_llm.csv\"))\n",
    "training_data_for_llm_df = training_data_for_llm_df.sample(290, random_state=42)\n",
    "training_data_for_llm_df['citation_context'] = training_data_for_llm_df['citation_context'].apply(last_of_string)\n",
    "\n",
    "# Convert to Hugging Face Dataset format\n",
    "train_dataset = Dataset.from_pandas(training_data_for_llm_df)\n",
    "\n",
    "# Split into train/validation\n",
    "train_test_split = train_dataset.train_test_split(test_size=0.1)\n",
    "train_dataset = train_test_split['train']\n",
    "eval_dataset = train_test_split['test']\n",
    "print(f\"Training set size: {len(train_dataset)} examples\")\n",
    "print(f\"Validation set size: {len(eval_dataset)} examples\")\n",
    "\n",
    "# Clean up\n",
    "del training_data_for_llm_df\n",
    "del train_test_split\n",
    "gc.collect()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3726457c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded 16 few-shot examples.\n",
      "New training set size: 277 examples\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2e557d60b5bb4c0cb816ed7bd70955f1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Creating CSV from Arrow format:   0%|          | 0/1 [00:00<?, ?ba/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "720ba376265240d8b7376260654aa111",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Creating CSV from Arrow format:   0%|          | 0/1 [00:00<?, ?ba/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "19113"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Concatenate the few-shot examples with the main training dataset\n",
    "# This will add the few-shot examples as additional rows to the training data\n",
    "few_shot_dataset = Dataset.from_csv(FEW_SHOT_CSV_PATH)\n",
    "print(f\"Loaded {len(few_shot_dataset)} few-shot examples.\")\n",
    "train_dataset = concatenate_datasets([train_dataset, few_shot_dataset])\n",
    "print(f\"New training set size: {len(train_dataset)} examples\")\n",
    "\n",
    "# Save off the datasets to CSV for later use\n",
    "train_dataset.to_csv(os.path.join(BASE_OUTPUT_DIR, \"train_dataset.csv\"), index=False)\n",
    "eval_dataset.to_csv(os.path.join(BASE_OUTPUT_DIR, \"eval_dataset.csv\"), index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c96a7203",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Load datasets from CSV\n",
    "# train_dataset = Dataset.from_csv(os.path.join(BASE_OUTPUT_DIR, \"train_dataset.csv\"))\n",
    "# eval_dataset = Dataset.from_csv(os.path.join(BASE_OUTPUT_DIR, \"eval_dataset.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "21b7ee0d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Dataset({\n",
       "    features: ['article_id', 'article_doi', 'article_abstract', 'dataset_id', 'citation_context', 'label', '__index_level_0__'],\n",
       "    num_rows: 277\n",
       "})"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "7ae25fcd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Dataset({\n",
       "    features: ['article_id', 'article_doi', 'article_abstract', 'dataset_id', 'citation_context', 'label', '__index_level_0__'],\n",
       "    num_rows: 29\n",
       "})"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eval_dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0066f50c",
   "metadata": {},
   "source": [
    "#### 5. Model Selection and Configuration\n",
    "\n",
    "We'll use a Qwen model."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "12e2982e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model C:\\Users\\jim\\.cache\\kagglehub\\models\\qwen-lm\\qwen-3\\transformers\\0.6b\\1 loaded with 4-bit quantization.\n"
     ]
    }
   ],
   "source": [
    "# 5.1. Choose a Model from KaggleHub\n",
    "# Example: Qwen/Qwen1.5-0.5B-Chat (or 1.8B-Chat if 0.5B is too small/performs poorly)\n",
    "# You can find these on KaggleHub or Hugging Face Hub.\n",
    "model_name = QWEN_BASE_MODEL_PATH\n",
    "\n",
    "# 5.2. Load Tokenizer\n",
    "tokenizer = AutoTokenizer.from_pretrained(model_name)\n",
    "tokenizer.pad_token = tokenizer.eos_token # Qwen uses EOS for padding\n",
    "\n",
    "# 5.3. Load Model with Quantization (4-bit)\n",
    "nf4_config = BitsAndBytesConfig(\n",
    "    load_in_4bit=True,\n",
    "    bnb_4bit_quant_type=\"nf4\",\n",
    "    bnb_4bit_use_double_quant=True,\n",
    "    bnb_4bit_compute_dtype=torch.bfloat16 # Or torch.float16 if bfloat16 is not supported by your GPU\n",
    ")\n",
    "\n",
    "model = AutoModelForCausalLM.from_pretrained(\n",
    "    model_name,\n",
    "    quantization_config=nf4_config,\n",
    "    torch_dtype=torch.bfloat16, # Match compute_dtype\n",
    "    device_map=\"auto\", # Automatically maps model to available devices\n",
    "    trust_remote_code=True # Required for some models like Qwen\n",
    ")\n",
    "\n",
    "# Prepare model for k-bit training (LoRA compatible)\n",
    "model.config.use_cache = False\n",
    "model.config.pretraining_tp = 1\n",
    "model = prepare_model_for_kbit_training(model)\n",
    "\n",
    "print(f\"Model {model_name} loaded with 4-bit quantization.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24a83df8",
   "metadata": {},
   "source": [
    "#### 6. Dataset Preparation for Training\n",
    "\n",
    "Format the extracted data into instruction-tuning prompts using the ChatML format, which Qwen models are trained on."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a008679d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Example of formatted training data (string output):\n",
      "Length of the full prompt in tokens: 496\n",
      "<|im_start|>system\n",
      "You are an expert assistant for classifying research data citations. /no_think<|im_end|>\n",
      "<|im_start|>user\n",
      "\n",
      "Given the following 'Article Abstract' and a specific data citation ('Dataset ID' and 'Data Citation Context' combination), classify the data citation as either: \n",
      "'Primary' (if the data citation refers to raw or processed **data created/generated as part of the paper**, specifically for this study), \n",
      "'Secondary' (if the data citation refers to raw or processed **data derived/reused from existing records** or previously published data), or \n",
      "'Missing' (if the data citation refers to another **article/paper/journal**, a **figure, software, or other non-data entity**, or the 'Data Citation Context' is **empty or irrelevant**).\n",
      "\n",
      "If the data citation refers to raw or processed **data** but the distinction between 'Primary' and 'Secondary' is ambiguous, then default to 'Primary'.\n",
      "\n",
      "Now, classify the following:\n",
      "\n",
      "Article Abstract: The genome of a laboratory-adapted strain of Leptospira interrogans serovar Hardjo was sequenced and analyzed. Comparison of the sequenced genome with that recently pub lished for a field isolate of the same serovar revealed relatively high sequence conservation at the nucleotide level, despite the different biological background of both samples. Con versely, comparison of both serovar Hardjo geno\n",
      "Dataset ID: PRJNA74167\n",
      "Data Citation Context: RJNA74167 L. santarosai Shermani LT 821 LSS PRJNA47139 L. weilii --- UI 13098 LEP1GSC108 PRJNA74123 doi:10.1371/journal.pone.0159387.t001 PLOS ONE | DOI:10.1371/journal.pone.0159387 July 21, 2016 3 / 12 Phylogenomic Analysis of Leptospira interrogans Serovar Hardjo corresponding to chromosomes I (4.34 Mb) and II (353 kb), by using ABACAS and the genome of L. interrogans serovar Lai as a reference.\n",
      "\n",
      "Classification:<|im_end|>\n",
      "<|im_start|>assistant\n",
      "<think>\n",
      "\n",
      "</think>\n",
      "\n",
      "Secondary<|im_end|>\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# 6.1. Define the formatting function for ChatML (Corrected for trl 0.19.1)\n",
    "def format_example(example):\n",
    "    messages = [\n",
    "        {\"role\": \"system\", \"content\": \"You are an expert assistant for classifying research data citations. /no_think\"},\n",
    "        {\"role\": \"user\", \"content\": (\n",
    "            f\"\"\"\n",
    "Given the following 'Article Abstract' and a specific data citation ('Dataset ID' and 'Data Citation Context' combination), classify the data citation as either: \n",
    "'Primary' (if the data citation refers to raw or processed **data created/generated as part of the paper**, specifically for this study), \n",
    "'Secondary' (if the data citation refers to raw or processed **data derived/reused from existing records** or previously published data), or \n",
    "'Missing' (if the data citation refers to another **article/paper/journal**, a **figure, software, or other non-data entity**, or the 'Data Citation Context' is **empty or irrelevant**).\\n\\n\"\"\"\n",
    "            f\"If the data citation refers to raw or processed **data** but the distinction between 'Primary' and 'Secondary' is ambiguous, then default to 'Primary'.\\n\\n\"\n",
    "            # f\"{FEW_SHOT_EXAMPLES.strip()}\\n\\n\" # Add the examples here\n",
    "            f\"Now, classify the following:\\n\\n\" # Add a clear separator            \n",
    "            f\"Article Abstract: {example['article_abstract']}\\n\" \n",
    "            f\"Dataset ID: {example['dataset_id']}\\n\"\n",
    "            f\"Data Citation Context: {example['citation_context']}\\n\\n\"\n",
    "            f\"Classification:\"\n",
    "        )}\n",
    "    ]\n",
    "    # The target output for the model is just \"Primary\", \"Secondary, or \"Missing\"\n",
    "    messages.append({\"role\": \"assistant\", \"content\": example['label']})\n",
    "    \n",
    "    # Apply chat template and return the string directly\n",
    "    # <--- IMPORTANT CHANGE: Directly return the string, not a dictionary\n",
    "    return tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=False, enable_thinking=False)\n",
    "\n",
    "# Apply the formatting to the dataset\n",
    "# IMPORTANT: When formatting_func returns a string directly, you typically don't\n",
    "# need to call .map() on the dataset beforehand if SFTTrainer handles it internally.\n",
    "# However, if you want to inspect the formatted text, you can still do this:\n",
    "# formatted_train_dataset = train_dataset.map(format_example)\n",
    "# But for SFTTrainer, you pass the original `train_dataset` and the `formatting_func`\n",
    "# and `dataset_text_field` (which will be ignored if formatting_func is used to generate the text).\n",
    "\n",
    "# Print an example to verify (you'll need to call format_example directly for this)\n",
    "print(\"\\nExample of formatted training data (string output):\")\n",
    "# You can't directly print from formatted_train_dataset if you don't map it first.\n",
    "# Let's print by calling the function on a sample:\n",
    "if len(train_dataset) > 0:\n",
    "    sample_formatted_text = format_example(train_dataset[35])\n",
    "    tokenized_input = tokenizer(sample_formatted_text, return_tensors=\"pt\")\n",
    "    prompt_length = tokenized_input.input_ids.shape[1]\n",
    "    print(f\"Length of the full prompt in tokens: {prompt_length}\")    \n",
    "    print(sample_formatted_text)\n",
    "else:\n",
    "    print(\"No training data to display example.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "542d74e3",
   "metadata": {},
   "source": [
    "### 7. Train the Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "3d8fdd82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f886a0b7addf4492b2cb12f80a636ab7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Applying formatting function to train dataset:   0%|          | 0/277 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0a80cd0a235244a59b314895d0dc358d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Adding EOS to train dataset:   0%|          | 0/277 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f0bde8cd6c4c46ae82bf95124b9a6053",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tokenizing train dataset:   0%|          | 0/277 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fbbed3abcc8e4906b169bce4e39e7fbd",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Truncating train dataset:   0%|          | 0/277 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0129f58e4c9746ed9654ff458123dab6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Applying formatting function to eval dataset:   0%|          | 0/29 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6288bfa3d2894f8e87dd020993ad1ee2",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Adding EOS to eval dataset:   0%|          | 0/29 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "01534cf4249d485a9ee33142513b661b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Tokenizing eval dataset:   0%|          | 0/29 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a69daf00ac3747538915c452aa6eddd2",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Truncating eval dataset:   0%|          | 0/29 [00:00<?, ? examples/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# ---------------------------------------------------------\n",
    "# This version uses the evaluation dataset in the SFTTrainer\n",
    "# ---------------------------------------------------------\n",
    "\n",
    "import numpy as np\n",
    "from sklearn.metrics import accuracy_score, f1_score, precision_score, recall_score\n",
    "from transformers.trainer_utils import EvalPrediction # Import this for type hinting if desired\n",
    "\n",
    "# Assuming your model outputs logits for 3 classes (Primary, Secondary, Missing)\n",
    "# And your labels are integers (e.g., 0, 1, 2) corresponding to these classes.\n",
    "\n",
    "def compute_classification_metrics(eval_pred: EvalPrediction):\n",
    "    logits, labels = eval_pred.predictions, eval_pred.label_ids\n",
    "    \n",
    "    # For classification, you typically take the argmax of the logits to get predicted class IDs\n",
    "    predictions = np.argmax(logits, axis=-1)\n",
    "    \n",
    "    # Calculate desired metrics\n",
    "    accuracy = accuracy_score(labels, predictions)\n",
    "    f1_macro = f1_score(labels, predictions, average='macro') # Macro F1 treats all classes equally\n",
    "    f1_weighted = f1_score(labels, predictions, average='weighted') # Weighted F1 accounts for class imbalance\n",
    "    \n",
    "    # You might also want precision and recall per class, or just overall\n",
    "    # precision_macro = precision_score(labels, predictions, average='macro')\n",
    "    # recall_macro = recall_score(labels, predictions, average='macro')\n",
    "\n",
    "    return {\n",
    "        \"accuracy\": accuracy,\n",
    "        \"f1_macro\": f1_macro,\n",
    "        \"f1_weighted\": f1_weighted,\n",
    "        # You can also include 'eval_loss' if you want, but Trainer computes it by default\n",
    "        # \"eval_loss\": eval_pred.metrics.get(\"eval_loss\", None) # Access it if Trainer provides it\n",
    "    }\n",
    "\n",
    "# 7.1. Configure LoRA\n",
    "peft_config = LoraConfig(\n",
    "    lora_alpha=16,\n",
    "    lora_dropout=0.1,\n",
    "    r=64,\n",
    "    bias=\"none\",\n",
    "    task_type=\"CAUSAL_LM\",\n",
    "    target_modules=\"all-linear\", # Adjust based on model architecture if needed\n",
    ")\n",
    "\n",
    "# 7.2. Configure Training Arguments (now using SFTConfig)\n",
    "training_args = SFTConfig(\n",
    "    output_dir=FINE_TUNED_MODEL_OUTPUT_DIR,\n",
    "    per_device_train_batch_size=1,\n",
    "    gradient_accumulation_steps=16,\n",
    "    learning_rate=2e-4,\n",
    "    num_train_epochs=3,\n",
    "    logging_steps=10,\n",
    "    save_steps=25,\n",
    "    optim=\"paged_adamw_8bit\",\n",
    "    fp16=True,\n",
    "    bf16=False,\n",
    "    max_grad_norm=0.3,\n",
    "    warmup_ratio=0.03,\n",
    "    lr_scheduler_type=\"constant\",\n",
    "    report_to=\"none\",\n",
    "    disable_tqdm=False,\n",
    "    remove_unused_columns=False,\n",
    "    label_names=['labels'],\n",
    "    \n",
    "    # SFTTrainer-specific parameters moved into SFTConfig\n",
    "    max_seq_length=512,\n",
    "    packing=False,\n",
    "    dataset_text_field=\"text\",\n",
    "    gradient_checkpointing=True,\n",
    "    gradient_checkpointing_kwargs={'use_reentrant':False},\n",
    "\n",
    "    # --- NEW: Evaluation Parameters ---\n",
    "    eval_strategy=\"steps\", # Evaluate every 'eval_steps'. You can also use \"epoch\" for evaluation_strategy\n",
    "    eval_steps=25,               # How often to run evaluation (e.g., every 25 steps)\n",
    "    save_strategy=\"steps\",       # How often to save checkpoints\n",
    "    save_total_limit=1,          # Only keep the best model checkpoint\n",
    "    load_best_model_at_end=True, # Load the model with the best validation metric at the end of training\n",
    "    greater_is_better=False,     # For loss, lower is better\n",
    ")\n",
    "\n",
    "# 7.3. Initialize SFTTrainer\n",
    "trainer = SFTTrainer(\n",
    "    model=model,\n",
    "    processing_class=tokenizer,\n",
    "    train_dataset=train_dataset,\n",
    "    eval_dataset=eval_dataset, # <--- Pass the evaluation dataset here\n",
    "    # compute_metrics=compute_classification_metrics,\n",
    "    peft_config=peft_config,\n",
    "    args=training_args,\n",
    "    formatting_func=format_example\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45982673",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Starting model training...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "    <div>\n",
       "      \n",
       "      <progress value='26' max='54' style='width:300px; height:20px; vertical-align: middle;'></progress>\n",
       "      [26/54 06:42 < 07:49, 0.06 it/s, Epoch 1.40/3]\n",
       "    </div>\n",
       "    <table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       " <tr style=\"text-align: left;\">\n",
       "      <th>Step</th>\n",
       "      <th>Training Loss</th>\n",
       "      <th>Validation Loss</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>1.267800</td>\n",
       "      <td>1.078030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table><p>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "# 7.4. Start Training\n",
    "print(\"\\nStarting model training...\")\n",
    "trainer.train()\n",
    "print(\"Training complete!\")\n",
    "\n",
    "# Save the fine-tuned model (LoRA adapters)\n",
    "trainer.save_model(os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, \"final_model\"))\n",
    "print(f\"Fine-tuned model saved to {os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, 'final_model')}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89ce92dd",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6dba261d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Initiating GPU memory cleanup...\n",
      "GPU memory cleanup complete. Please check nvidia-smi to confirm.\n"
     ]
    }
   ],
   "source": [
    "# --- Explicit GPU Memory Cleanup ---\n",
    "print(\"\\nInitiating GPU memory cleanup...\")\n",
    "\n",
    "# 1. Explicitly delete large objects that consume GPU memory\n",
    "#    This removes references, allowing Python's garbage collector to act.\n",
    "# if 'trainer' in locals() and trainer is not None:\n",
    "#     del trainer\n",
    "# if 'model' in locals() and model is not None:\n",
    "#     del model\n",
    "# if 'tokenizer' in locals() and tokenizer is not None:\n",
    "#     del tokenizer\n",
    "# If you had other large tensors or datasets explicitly moved to GPU,\n",
    "# you would delete them here too. For Hugging Face datasets, they are usually\n",
    "# on CPU unless you manually call .to('cuda').\n",
    "\n",
    "# 2. Force Python's garbage collection\n",
    "#    This helps ensure that deleted objects are immediately cleaned up.\n",
    "# gc.collect()\n",
    "\n",
    "# 3. Clear PyTorch's CUDA memory cache\n",
    "#    This tells PyTorch to release any cached memory back to the OS/driver.\n",
    "# if torch.cuda.is_available():\n",
    "#     torch.cuda.empty_cache()\n",
    "\n",
    "print(\"GPU memory cleanup complete. Please check nvidia-smi to confirm.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd1e3154",
   "metadata": {},
   "source": [
    "#### 8. Inference and Evaluation\n",
    "\n",
    "After training, load the best model (or the final one) and apply it to the test data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "583fed9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Model loaded for inference.\n"
     ]
    }
   ],
   "source": [
    "# 8.1. Load the Trained Model (or merge LoRA adapters for full model)\n",
    "# If you saved LoRA adapters, you'll need to load the base model and then the adapters.\n",
    "# For inference, it's often easier to merge them.\n",
    "# model = AutoModelForCausalLM.from_pretrained(\n",
    "#     model_name,\n",
    "#     quantization_config=nf4_config, # Use the same config as training\n",
    "#     torch_dtype=torch.bfloat16,\n",
    "#     device_map=\"auto\",\n",
    "#     trust_remote_code=True\n",
    "# )\n",
    "# model = PeftModel.from_pretrained(model, os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, \"final_model\"))\n",
    "# model = model.merge_and_unload() # Merge LoRA adapters into the base model\n",
    "\n",
    "# For simplicity, if you just want to test the last saved checkpoint:\n",
    "# You can also load the model directly from the checkpoint if it's a full save\n",
    "# model = AutoModelForCausalLM.from_pretrained(os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, \"final_model\"), device_map=\"auto\")\n",
    "# tokenizer = AutoTokenizer.from_pretrained(os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, \"final_model\"))\n",
    "\n",
    "# If you want to load the base model and then the adapters for inference:\n",
    "model = AutoModelForCausalLM.from_pretrained(\n",
    "    model_name,\n",
    "    quantization_config=nf4_config,\n",
    "    torch_dtype=torch.bfloat16,\n",
    "    device_map=\"auto\",\n",
    "    trust_remote_code=True\n",
    ")\n",
    "model = PeftModel.from_pretrained(model, os.path.join(FINE_TUNED_MODEL_OUTPUT_DIR, \"final_model\"))\n",
    "model.eval() # Set to evaluation mode\n",
    "\n",
    "print(\"Model loaded for inference.\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "2bc475f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Files paths shape: (30, 4)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "c60aaf83-ec3a-40f2-9abb-a58c8b7d6ffc",
       "rows": [
        [
         "27",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_mp.14424.xml",
         "test"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6144.xml",
         "test"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf",
         "",
         "test"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10.1002_mp.14424</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10.1002_ece3.6144</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10.1002_ejoc.202000139</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                article_id                                      pdf_file_path  \\\n",
       "27        10.1002_mp.14424  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "15       10.1002_ece3.6144  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "23  10.1002_ejoc.202000139  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \n",
       "27  ./kaggle/input/make-data-count-finding-data-re...         test  \n",
       "15  ./kaggle/input/make-data-count-finding-data-re...         test  \n",
       "23                                                            test  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# For testing, always set to the TEST_DATA_DIR\n",
    "base_file_dir = TEST_DATA_DIR\n",
    "\n",
    "# Load file paths for base directory\n",
    "test_file_paths_df = load_file_paths(base_file_dir)\n",
    "test_file_paths_df['xml_file_path'] = test_file_paths_df['xml_file_path'].fillna('')\n",
    "\n",
    "print(f\"Files paths shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.sample(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b06b8228",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def invoke_model_for_inference(tokenizer, article_data: ArticleData) -> list[SubmissionData]:\n",
    "    submission_data_list = []\n",
    "    article_id = article_data.article_id\n",
    "    dataset_citations = article_data.dataset_citations\n",
    "    if not dataset_citations:\n",
    "        submission_data_list.append(SubmissionData(article_id, dataset_id=\"Missing\", type=\"Missing\"))\n",
    "        return submission_data_list\n",
    "\n",
    "    print(f\"Found {len(dataset_citations)} citations for {article_id}\")\n",
    "    for dc in dataset_citations:\n",
    "        # Create the prompt for inference\n",
    "        messages = [\n",
    "            {\"role\": \"system\", \"content\": \"You are an expert assistant for classifying research data citations. /no_think\"},\n",
    "            {\"role\": \"user\", \"content\": (\n",
    "                f\"\"\"\n",
    "Given the following 'Article Abstract' and a specific data citation ('Dataset ID' and 'Data Citation Context' combination), classify the data citation as either: \n",
    "'Primary' (if the data citation refers to raw or processed **data created/generated as part of the paper**, specifically for this study), \n",
    "'Secondary' (if the data citation refers to raw or processed **data derived/reused from existing records** or previously published data), or \n",
    "'Missing' (if the data citation refers to another **article/paper/journal**, a **figure, software, or other non-data entity**, or the 'Data Citation Context' is **empty or irrelevant**).\\n\\n\"\"\"\n",
    "                f\"If the data citation refers to raw or processed **data** but the distinction between 'Primary' and 'Secondary' is ambiguous, then default to 'Primary'.\\n\\n\"\n",
    "                # f\"{FEW_SHOT_EXAMPLES.strip()}\\n\\n\" # Add the examples here\n",
    "                f\"Now, classify the following:\\n\\n\" # Add a clear separator            \n",
    "                f\"Article Abstract: {article_data.abstract}\\n\"\n",
    "                f\"Dataset ID: {dc.dataset_id}\\n\"                \n",
    "                f\"Data Citation Context: {dc.citation_context}\\n\\n\"\n",
    "                f\"Classification:\"\n",
    "            )}\n",
    "        ]\n",
    "\n",
    "        # --- CHANGE STARTS HERE ---\n",
    "        # Tokenize and get both input_ids and attention_mask\n",
    "        input_text = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)\n",
    "        inputs = tokenizer(input_text, return_tensors=\"pt\", max_length=512, truncation=True).to(device)\n",
    "\n",
    "        with torch.no_grad():\n",
    "            output = model.generate(\n",
    "                **inputs, # <--- Pass the entire dictionary (includes input_ids and attention_mask)\n",
    "                max_new_tokens=10, # Expecting \"Primary\" or \"Secondary\"\n",
    "                do_sample=True,    # <--- Enable sampling\n",
    "                temperature=0.7,   # <--- Adjust temperature (0.7-0.9 is common)\n",
    "                top_p=0.9,         # <--- Top-p sampling (consider tokens that sum to 90% probability)\n",
    "                top_k=50,          # <--- Top-k sampling (consider only the top 50 most probable tokens)                \n",
    "                pad_token_id=tokenizer.eos_token_id,\n",
    "                eos_token_id=tokenizer.eos_token_id\n",
    "            )\n",
    "        # --- CHANGE ENDS HERE ---        \n",
    "\n",
    "        generated_text = tokenizer.decode(output[0][inputs['input_ids'].shape[1]:], skip_special_tokens=True).strip() # Use inputs['input_ids']\n",
    "        # print(f\"LLM Resp: {generated_text}\")        \n",
    "        \n",
    "        # Post-process the generated text to get the classification\n",
    "        predicted_type = \"Missing\"\n",
    "        if \"Primary\" in generated_text:\n",
    "            predicted_type = \"Primary\"\n",
    "        elif \"Secondary\" in generated_text:\n",
    "            predicted_type = \"Secondary\"\n",
    "        \n",
    "        submission_data_list.append(SubmissionData(article_id, dataset_id=dc.dataset_id, type=predicted_type, context=dc.citation_context))\n",
    "\n",
    "    return submission_data_list\n",
    "\n",
    "def process_test_articles(tokenizer, file_paths_df: pd.DataFrame) -> list[SubmissionData]:\n",
    "    \"\"\"\n",
    "    Extracts article data for testing set without ground truth.\n",
    "    \n",
    "    Args:\n",
    "        file_paths_df (pd.DataFrame): DataFrame containing file paths and ground truth info.\n",
    "        \n",
    "    Returns:\n",
    "        Dict[str, ArticleData]: Dictionary mapping article IDs to ArticleData objects.\n",
    "    \"\"\"\n",
    "    submission_data_list = []\n",
    "    for i, row in tqdm(file_paths_df.iterrows(), total=len(file_paths_df)):\n",
    "        article_id = row['article_id']\n",
    "        filepath = row['pdf_file_path'] if row['pdf_file_path'] else row['xml_file_path']\n",
    "        file_extractor = MdcFileTextExtractor(article_id, filepath)\n",
    "        \n",
    "        # Extract article data\n",
    "        article_data = file_extractor.extract_article_data_for_inference(NLP_SPACY)\n",
    "\n",
    "        # Invoke the model with the collected article_data\n",
    "        submission_data_list.extend(invoke_model_for_inference(tokenizer, article_data))\n",
    "\n",
    "    print(f\"Processed testing data for {len(submission_data_list)} article and dataset_id combos.\")\n",
    "    return submission_data_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "903dc0ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "81d840bd-8a04-43e2-9138-998c09241cee",
       "rows": [
        [
         "20",
         "10.1002_ecs2.1280",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf",
         "",
         "test"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 1
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10.1002_ecs2.1280</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           article_id                                      pdf_file_path  \\\n",
       "20  10.1002_ecs2.1280  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "   xml_file_path dataset_type  \n",
       "20                       test  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_test_file_paths_df = test_file_paths_df.sample(2, random_state=42)\n",
    "sample_test_file_paths_df = test_file_paths_df.loc[test_file_paths_df['article_id']=='10.1002_mp.14424']\n",
    "sample_test_file_paths_df = test_file_paths_df.loc[test_file_paths_df['article_id']=='10.1002_cssc.202201821']\n",
    "sample_test_file_paths_df = test_file_paths_df.loc[test_file_paths_df['article_id']=='10.1002_ecs2.1280']\n",
    "# sample_test_file_paths_df = test_file_paths_df.loc[test_file_paths_df['article_id']=='10.1002_esp.5090']\n",
    "sample_test_file_paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "efde80b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "23153ae59b0f4509a1e5442a9df008de",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/30 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf\n",
      "Found 2 citations for 10.1002_2017jc013030\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202007717.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201902131.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202000235.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001668.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202003167.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_cssc.202201821.pdf\n",
      "Found 1 citations for 10.1002_cssc.202201821\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.3985.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.4466.pdf\n",
      "Found 1 citations for 10.1002_ece3.4466\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf\n",
      "Found 3 citations for 10.1002_ece3.5260\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5395.pdf\n",
      "Found 5 citations for 10.1002_ece3.5395\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf\n",
      "Found 2 citations for 10.1002_ece3.6144\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6303.pdf\n",
      "Found 5 citations for 10.1002_ece3.6303\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6784.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.961.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.9627.pdf\n",
      "Found 3 citations for 10.1002_ece3.9627\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf\n",
      "Found 1 citations for 10.1002_ecs2.1280\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.4619.pdf\n",
      "Found 1 citations for 10.1002_ecs2.4619\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000916.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf\n",
      "Found 3 citations for 10.1002_esp.5058\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5090.pdf\n",
      "Found 1 citations for 10.1002_esp.5090\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf\n",
      "Found 2 citations for 10.1002_mp.14424\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_nafm.10870.pdf\n",
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1007_jhep07(2018)134.pdf\n",
      "Processed testing data for 47 article and dataset_id combos.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[SubmissionData(article_id='10.1002_2017jc013030', dataset_id='10.17882/49388', type='Secondary', context='Data referring to Organelli et al. (2016a; https://doi.org/10.17882/(https://doi.org/10.17882/47142) 47142) and Barbieux et al. (2017;(https://doi.org/10.17882/47142) https://doi.org/10.17882/49388) are(https://doi.org/10.17882/49388) freely available on SEANOE.'),\n",
       " SubmissionData(article_id='10.1002_2017jc013030', dataset_id='10.17882/47142', type='Primary', context='Approches Numeriques); Pierre-Marie Poulain (National Institute of Oceanography and Experimental Geophysics, Italy; ArgoItaly); Sabrina Speich (Laboratoire de Meteorologie Dynamique, France; LEFEGMMC); Virginie Thierry (Ifremer, France; LEFE-GMMC); Pascal Conan (Observatoire Oceanologique de Banyuls sur mer, France; LEFE-GMMC); Laurent Coppola (Laboratoire dOceanographie de Villefranche, France; LEFE-GMMC); Anne Petrenko (Mediterranean Institute of Oceanography, France; LEFE-GMMC); and Jean-Baptiste Sallee (Laboratoire dOceanographie et du Climat, France; LEFE-GMMC). Collin Roesler (Bowdoin College, USA) and Yannick Huot (University of Sherbrooke, Canada) are acknowledged for useful comments and fruitful discussion. We also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available. Data referring to Organelli et al. (2016a; https://doi.org/10.17882/(https://doi.org/10.17882/47142)'),\n",
       " SubmissionData(article_id='10.1002_anie.201916483', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_anie.202005531', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_anie.202007717', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.201902131', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.201903120', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.202000235', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.202001412', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.202001668', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_chem.202003167', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_cssc.202201821', dataset_id='10.5281/zenodo.7074790', type='Primary', context='*Conflict of Interest**_ The authors declare no conflict of interest. **Data Availability Statement** The data that support the findings of this study are openly available in zenodo at https://doi.org/10.5281/zenodo.7074790.(https://doi.org/10.5281/zenodo.7074790)'),\n",
       " SubmissionData(article_id='10.1002_ece3.3985', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_ece3.4466', dataset_id='10.5061/dryad.r6nq870', type='Secondary', context='**AUTHOR CONTRIBUTIONS** AH and MO conceived the ideas and designed methodology; AH col lected the data; AH and MO analyzed the data; AH led the writing **9832** HANSSON and OLSSON **|** of the manuscript. All authors contributed critically to the drafts and gave final approval for publication. **DATA ACCESSIBILITY** The dataset supporting this article are available in Dryad https://doi.(https://doi.org/10.5061/dryad.r6nq870)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5260', dataset_id='MK838494', type='Secondary', context='J.S.S., J.M.P., and D.R.M. ob- tained and provided the Illumina and Sanger sequence data, assem- bled the base genomes, and produced genomic assembly metrics. All authors contributed to writing of the manuscript. **DATA AVAILABILITY** DNA sequences: GenBank MK838494 to MK838511; P(info:x-wiley/peptideatlas/MK838494) hyluce BED/ BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad https://doi.org/10.5061/dryad.2f62927;(https://doi.org/10.5061/dryad.2f62927)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5260', dataset_id='10.5061/dryad.2f62927', type='Secondary', context='J.S.S., J.M.P., and D.R.M. ob- tained and provided the Illumina and Sanger sequence data, assem- bled the base genomes, and produced genomic assembly metrics. All authors contributed to writing of the manuscript. **DATA AVAILABILITY** DNA sequences: GenBank MK838494 to MK838511; P(info:x-wiley/peptideatlas/MK838494) hyluce BED/ BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad https://doi.org/10.5061/dryad.2f62927;(https://doi.org/10.5061/dryad.2f62927)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5260', dataset_id='MK838511', type='Secondary', context='J.S.S., J.M.P., and D.R.M. ob- tained and provided the Illumina and Sanger sequence data, assem- bled the base genomes, and produced genomic assembly metrics. All authors contributed to writing of the manuscript. **DATA AVAILABILITY** DNA sequences: GenBank MK838494 to MK838511; P(info:x-wiley/peptideatlas/MK838494) hyluce BED/ BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad https://doi.org/10.5061/dryad.2f62927;(https://doi.org/10.5061/dryad.2f62927)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5395', dataset_id='10.5441/001/1', type='Secondary', context='**DATA ACCESSIBILITY** The data used for this study are available through the Movebank Data Repository (https://www.movebank.org): with https://doi.org/10.5441(https://www.movebank.org) /001/1.v1cs4nn0, https://doi.org/10.5441/001/1.c42j3js7, https://(https://doi.org/10.5441/001/1.v1cs4nn0)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5395', dataset_id='10.5441/001/1.c42j3js7', type='Secondary', context='**DATA ACCESSIBILITY** The data used for this study are available through the Movebank Data Repository (https://www.movebank.org): with https://doi.org/10.5441(https://www.movebank.org) /001/1.v1cs4nn0, https://doi.org/10.5441/001/1.c42j3js7, https://(https://doi.org/10.5441/001/1.v1cs4nn0)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5395', dataset_id='10.5441/001/1.4192t2j4', type='Secondary', context='The data used for this study are available through the Movebank Data Repository (https://www.movebank.org): with https://doi.org/10.5441(https://www.movebank.org) /001/1.v1cs4nn0, https://doi.org/10.5441/001/1.c42j3js7, https://(https://doi.org/10.5441/001/1.v1cs4nn0) doi.org/10.5441/001/1.4192t2j4,(https://doi.org/10.5441/001/1.4192t2j4) https://doi.org/10.5441/001/1.(https://doi.org/10.5441/001/1.ck04mn78)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5395', dataset_id='10.5441/001/1.v1cs4nn0', type='Secondary', context='**DATA ACCESSIBILITY** The data used for this study are available through the Movebank Data Repository (https://www.movebank.org): with https://doi.org/10.5441(https://www.movebank.org) /001/1.v1cs4nn0, https://doi.org/10.5441/001/1.c42j3js7, https://(https://doi.org/10.5441/001/1.v1cs4nn0)'),\n",
       " SubmissionData(article_id='10.1002_ece3.5395', dataset_id='10.5441/001/1.ck04mn78', type='Secondary', context='The data used for this study are available through the Movebank Data Repository (https://www.movebank.org): with https://doi.org/10.5441(https://www.movebank.org) /001/1.v1cs4nn0, https://doi.org/10.5441/001/1.c42j3js7, https://(https://doi.org/10.5441/001/1.v1cs4nn0) doi.org/10.5441/001/1.4192t2j4,(https://doi.org/10.5441/001/1.4192t2j4) https://doi.org/10.5441/001/1.(https://doi.org/10.5441/001/1.ck04mn78)'),\n",
       " SubmissionData(article_id='10.1002_ece3.6144', dataset_id='10.5061/dryad.zw3r22854', type='Secondary', context='Elena Duke performed research, analyzed data, and wrote the paper. Ron Burton designed research, wrote the paper, and is the principal investigator. **DATA AVAILABILITY STATEMENT** Data are available at Dryad Digital Repository at: https://doi.(https://doi.org/10.5061/dryad.zw3r22854)'),\n",
       " SubmissionData(article_id='10.1002_ece3.6144', dataset_id='MH718435', type='Secondary', context='Two species ( _Seriphus politus_ and _Etrumeus acuminatus_ ) included in our constructed libraries did not have 16S rRNA sequences in NCBI. We obtained tissue samples from voucher specimens for these species from the Scripps Institution of Oceanography Marine Vertebrate Collection, extracted and sequenced their DNA and submitted these sequences to GenBank (accession numbers: MH714866 and(info:ddbj-embl-genbank/MH714866) MH718435). The total number of reads per species was mapped and(info:ddbj-embl-genbank/MH718435) counted using RNA-seq analysis tools in CLC at 97% similarity to a local reference database.'),\n",
       " SubmissionData(article_id='10.1002_ece3.6303', dataset_id='MN335287', type='Secondary', context='The congener was collected from Kuocangshan Mountain in Linhai City, Zhejiang Province, China, and identified by Prof. Ming Jiang in Taizhou University, China. All phylogenetic analyses were performed with MEGA v 10.0.5 (Kumar et al., 2018). Sequences of the 34 haplotypes and the two outgroups have been **5622** WAN et al . **|** submitted to the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084-MN107151, MN335287-MN335290).'),\n",
       " SubmissionData(article_id='10.1002_ece3.6303', dataset_id='MN335290', type='Secondary', context='The congener was collected from Kuocangshan Mountain in Linhai City, Zhejiang Province, China, and identified by Prof. Ming Jiang in Taizhou University, China. All phylogenetic analyses were performed with MEGA v 10.0.5 (Kumar et al., 2018). Sequences of the 34 haplotypes and the two outgroups have been **5622** WAN et al . **|** submitted to the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084-MN107151, MN335287-MN335290).'),\n",
       " SubmissionData(article_id='10.1002_ece3.6303', dataset_id='10.5061/dryad.37pvmcvgb', type='Primary', context='**Feihai Yu:** Conceptualization (supporting); Writing-review & editing (supporting). **Junmin Li:** Conceptualization (lead); Funding acquisition (lead); Investigation (lead); Methodology (lead); Project administration (lead); Validation (lead); Writing-original draft (lead); Writing-review & editing (lead). **DATA AVAILABILITY STATEMENT** DNA sequences have been deposited in the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084-MN107151, MN335287-MN335290). The data that(info:ddbj-embl-genbank/MN107084) support the findings of this study have been deposited in Dryad with doi:https://doi.org/10.5061/dryad.37pvmcvgb.(https://doi.org/10.5061/dryad.37pvmcvgb) **ORCID** _'),\n",
       " SubmissionData(article_id='10.1002_ece3.6303', dataset_id='MN107084', type='Secondary', context='The congener was collected from Kuocangshan Mountain in Linhai City, Zhejiang Province, China, and identified by Prof. Ming Jiang in Taizhou University, China. All phylogenetic analyses were performed with MEGA v 10.0.5 (Kumar et al., 2018). Sequences of the 34 haplotypes and the two outgroups have been **5622** WAN et al . **|** submitted to the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084-MN107151, MN335287-MN335290).'),\n",
       " SubmissionData(article_id='10.1002_ece3.6303', dataset_id='MN107151', type='Secondary', context='The congener was collected from Kuocangshan Mountain in Linhai City, Zhejiang Province, China, and identified by Prof. Ming Jiang in Taizhou University, China. All phylogenetic analyses were performed with MEGA v 10.0.5 (Kumar et al., 2018). Sequences of the 34 haplotypes and the two outgroups have been **5622** WAN et al . **|** submitted to the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084-MN107151, MN335287-MN335290).'),\n",
       " SubmissionData(article_id='10.1002_ece3.6784', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_ece3.961', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_ece3.9627', dataset_id='10.6084/m9.figshare.21619055', type='Secondary', context='This article has earned Open Data and Open Materials badges. Data and materials are available at https://doi.org/10.5061/dryad.(https://doi.org/10.5061/dryad.b8gtht7h3) b8gtht7h3.(https://doi.org/10.5061/dryad.b8gtht7h3) **DATA AVAILABILITY STATEMENT** All R code and data used to implement our Bayesian co-\\xadabundance models can be accessed at this Figshare repository: https://doi.(https://doi.org/10.6084/m9.figshare.21619055)'),\n",
       " SubmissionData(article_id='10.1002_ece3.9627', dataset_id='10.5061/dryad', type='Secondary', context='** This article has earned Open Data and Open Materials badges. Data and materials are available at https://doi.org/10.5061/dryad.(https://doi.org/10.5061/dryad.b8gtht7h3) b8gtht7h3.(https://doi.org/10.5061/dryad.b8gtht7h3) **DATA AVAILABILITY STATEMENT**'),\n",
       " SubmissionData(article_id='10.1002_ece3.9627', dataset_id='10.5061/dryad.b8gtht7h3', type='Secondary', context='** This article has earned Open Data and Open Materials badges. Data and materials are available at https://doi.org/10.5061/dryad.(https://doi.org/10.5061/dryad.b8gtht7h3) b8gtht7h3.(https://doi.org/10.5061/dryad.b8gtht7h3) **DATA AVAILABILITY STATEMENT**'),\n",
       " SubmissionData(article_id='10.1002_ecs2.1280', dataset_id='10.5061/dryad.p3fg9', type='Secondary', context='Additional Supporting Information may be found online at: http://onlinelibrary.wiley.com/doi/10.1002/(http://onlinelibrary.wiley.com/doi/10.1002/ecs2.1280/supinfo) ecs2.1280/supinfo(http://onlinelibrary.wiley.com/doi/10.1002/ecs2.1280/supinfo) ## D ata A vailability Data associated with this paper have been deposited in Dryad: https://doi.org/10.5061/dryad.p3fg9(https://doi.org/10.5061/dryad.p3fg9) v www.esajournals.org 17 May 2016 v Volume 7(5) v'),\n",
       " SubmissionData(article_id='10.1002_ecs2.4619', dataset_id='10.25349/D9QW5X', type='Primary', context='The manuscript was greatly improved thanks to the attention and edits of John Melack, Tom Smith, and Jennifer King. CONFLICT OF INTEREST STATEMENT The authors declare no conflicts of interest. DATA AVAILABILITY STATEMENT Data and novel code (Owens et al., 2023) are available from Dryad: https://doi.org/10.25349/D9QW5X.(https://doi.org/10.25349/D9QW5X) ORCID'),\n",
       " SubmissionData(article_id='10.1002_ejic.201900904', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_ejoc.202000139', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_ejoc.202000916', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1002_esp.5058', dataset_id='10.5066/P9FW6E8K', type='Primary', context='Any use of trade, firm, or product names is for descriptive pur poses only and does not imply endorsement by the U.S. Government. CONFLICT OF INTEREST There is no conflict of interest in this paper. DATA AVAILABILITY STATEMENT The dataset of the bathymetric survey and structure from motion of the Ichilo River is openly available USGS data release at http://doi.(http://doi.org/10.5066/P9FW6E8K)'),\n",
       " SubmissionData(article_id='10.1002_esp.5058', dataset_id='10.5061/dryad', type='Secondary', context='CONFLICT OF INTEREST There is no conflict of interest in this paper. DATA AVAILABILITY STATEMENT The dataset of the bathymetric survey and structure from motion of the Ichilo River is openly available USGS data release at http://doi.(http://doi.org/10.5066/P9FW6E8K) org/10.5066/P9FW6E8K and Dryad https://doi.org/10.5061/dryad.(http://doi.org/10.5066/P9FW6E8K)'),\n",
       " SubmissionData(article_id='10.1002_esp.5058', dataset_id='10.5061/dryad.jh9w0vt9t', type='Secondary', context='DATA AVAILABILITY STATEMENT The dataset of the bathymetric survey and structure from motion of the Ichilo River is openly available USGS data release at http://doi.(http://doi.org/10.5066/P9FW6E8K) org/10.5066/P9FW6E8K and Dryad https://doi.org/10.5061/dryad.(http://doi.org/10.5066/P9FW6E8K) jh9w0vt9t, respectively.(https://doi.org/10.5061/dryad.jh9w0vt9t) ORCID Kattia Rubi Arnez Ferrel https://orcid.org/0000-0003-1670-3272(https://orcid.org/0000-0003-1670-3272) Jonathan Mark Nelson'),\n",
       " SubmissionData(article_id='10.1002_esp.5090', dataset_id='10.5066/P9353101', type='Secondary', context='2 | METHODS 2.1 | Analysis of Worldview-1 imagery and elevation model For our topographic analysis we use the pre- and post-Petermann earthquake digital elevation models (DEMs) produced by Gold et al. (2019). These were derived using pre-event (March and June 2014) and post-event (October and November 2016) in-track stereo 0.5 m resolution panchromatic WorldView-1 and WorldView-2 images (2019, DigitalGlobe) using the Surface Extraction from TIN-based Searchspace Minimization software (Noh & Howat, 2015) running on the University of Iowa Argon supercomputer. The DEMs are available at https://doi.org/10.5066/P9353101, and the processing techniques(https://doi.org/10.5066/P9353101) are described in the supplementary information provided by Gold'),\n",
       " SubmissionData(article_id='10.1002_mp.14424', dataset_id='10.7937/tcia.2020.6c7y-gq39', type='Secondary', context='Four hundred and two thoracic segmentations were first generated automatically by a U-Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy-eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert-vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at https://doi.org/10.7937/tcia.2020.6c7y-gq39.'),\n",
       " SubmissionData(article_id='10.1002_mp.14424', dataset_id='10.7937/K9/TCIA.2015.PF0M9REI', type='Primary', context='22. Aerts HJWL, Wee L, Rios Velazquez E, et al. Data from NSCLC-Radiomics Dataset. In: The Cancer Imaging Archive; 2019. https://doi.org/(https://doi.org/10.7937/K9/TCIA.2015.PF0M9REI)'),\n",
       " SubmissionData(article_id='10.1002_nafm.10870', dataset_id='Missing', type='Missing', context=''),\n",
       " SubmissionData(article_id='10.1007_jhep07(2018)134', dataset_id='Missing', type='Missing', context='')]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sample_sub = process_test_articles(tokenizer, test_file_paths_df)\n",
    "display(sample_sub)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "36ef4754",
   "metadata": {},
   "source": [
    "#### 9. Submission File Generation (Kaggle Specific)\n",
    "\n",
    "Finally, format your predictions into the required `submission.csv` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "b3aac269",
   "metadata": {},
   "outputs": [],
   "source": [
    "def format_dataset_id(dataset_id: str) -> str:\n",
    "    \"\"\"\n",
    "    Formats the dataset_id by removing any leading/trailing whitespace and ensuring it is a string.\n",
    "    \n",
    "    Args:\n",
    "        dataset_id (str): The dataset identifier to format.\n",
    "        \n",
    "    Returns:\n",
    "        str: The formatted dataset identifier.\n",
    "    \"\"\"\n",
    "    if dataset_id and dataset_id.startswith(\"10.\") and len(dataset_id) > 10:\n",
    "        # If the dataset_id starts with \"10.\" and is longer than 10 characters, it's likely a DOI\n",
    "        dataset_id = \"https://doi.org/\" + dataset_id.lower().strip()\n",
    "    return dataset_id\n",
    "\n",
    "def prepare_for_submission(submission_list: list[SubmissionData]) -> pd.DataFrame:\n",
    "    \"\"\"\n",
    "    Prepares the submission_list for submission by ensuring the correct columns and formatting.\n",
    "    \n",
    "    Args:\n",
    "        expanded_df (pd.DataFrame): The DataFrame containing expanded dataset information.\n",
    "        \n",
    "    Returns:\n",
    "        pd.DataFrame: A DataFrame ready for submission with 'article_id', 'dataset_id', and 'type' columns.\n",
    "    \"\"\"\n",
    "    submission_df = pd.DataFrame(sample_sub)\n",
    "    # Ensure the DataFrame has the correct columns\n",
    "    submission_df = submission_df[['article_id', 'dataset_id', 'type']].copy()\n",
    "\n",
    "    # Format dataset_id\n",
    "    submission_df['dataset_id'] = submission_df['dataset_id'].apply(format_dataset_id)  \n",
    "\n",
    "    # Remove rows where type is 'Missing' and reset index\n",
    "    submission_df = submission_df[submission_df['type'] != 'Missing'].reset_index(drop=True)\n",
    "    submission_df['row_id'] = range(len(submission_df))\n",
    "\n",
    "    # Reorder columns to match the submission format\n",
    "    submission_df = submission_df[['row_id', 'article_id', 'dataset_id', 'type']]\n",
    "    \n",
    "    return submission_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f10b8cb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submission file 'submission_df.csv' created successfully!\n"
     ]
    }
   ],
   "source": [
    "# 9.1. Create Submission DataFrame\n",
    "\n",
    "submission_df = prepare_for_submission(sample_sub)\n",
    "submission_df.to_csv(\"submission_df.csv\", index=False)\n",
    "print(\"Submission file 'submission_df.csv' created successfully!\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "bf893421",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TP: 4\n",
      "FP: 26\n",
      "FN: 10\n",
      "F1 Score: 0.182\n"
     ]
    }
   ],
   "source": [
    "def f1_score(tp, fp, fn):\n",
    "    return 2 * tp / (2 * tp + fp + fn) if (2 * tp + fp + fn) != 0 else 0.0\n",
    "    \n",
    "    \n",
    "# if not os.getenv('KAGGLE_IS_COMPETITION_RERUN'):\n",
    "pred_df = submission_df.copy()\n",
    "label_df = pd.read_csv(\"./kaggle/input/make-data-count-finding-data-references/sample_submission.csv\")\n",
    "label_df = label_df[label_df['type'] != 'Missing'].reset_index(drop=True)\n",
    "\n",
    "hits_df = label_df.merge(pred_df, on=[\"article_id\", \"dataset_id\", \"type\"])\n",
    "\n",
    "tp = hits_df.shape[0]\n",
    "fp = pred_df.shape[0] - tp\n",
    "fn = label_df.shape[0] - tp\n",
    "\n",
    "\n",
    "print(\"TP:\", tp)\n",
    "print(\"FP:\", fp)\n",
    "print(\"FN:\", fn)\n",
    "print(\"F1 Score:\", round(f1_score(tp, fp, fn), 3))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
