{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "0a8ecd31",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install lxml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "d2db1bd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using device: cpu\n"
     ]
    }
   ],
   "source": [
    "# --- 0. Environment Setup & Offline Preparation ---\n",
    "\n",
    "# Standard Imports\n",
    "import os\n",
    "import glob\n",
    "import re\n",
    "import pandas as pd\n",
    "import lxml.etree as etree\n",
    "from lxml.etree import _Element as Element # Type hinting for lxml.etree.Element\n",
    "import collections # For deque in parenthesis removal\n",
    "import fitz # PyMuPDF for PDF processing\n",
    "from transformers import AutoModelForCausalLM, AutoTokenizer\n",
    "from transformers.modeling_utils import PreTrainedModel\n",
    "from transformers.tokenization_utils import PreTrainedTokenizer\n",
    "from transformers.utils.quantization_config import BitsAndBytesConfig\n",
    "from transformers.training_args import TrainingArguments\n",
    "from trl import SFTTrainer\n",
    "import torch\n",
    "from datasets import Dataset # Hugging Face datasets library\n",
    "import kagglehub\n",
    "import spacy\n",
    "import json\n",
    "\n",
    "# Set device for PyTorch\n",
    "device = \"cuda\" if torch and torch.cuda.is_available() else \"cpu\"\n",
    "print(f\"Using device: {device}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "b667ac82",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Uncomment and run this line once to download the model\n",
    "#!python -m spacy download en_core_web_sm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "48e263db",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define constants for file paths and model configurations\n",
    "BASE_INPUT_DIR = './kaggle/input/make-data-count-finding-data-references'\n",
    "ARTICLE_TRAIN_DIR = os.path.join(BASE_INPUT_DIR, 'train')\n",
    "ARTICLE_TEST_DIR = os.path.join(BASE_INPUT_DIR, 'test')\n",
    "\n",
    "# Define directories for articles in train and test sets\n",
    "LABELED_TRAINING_DATA_CSV_PATH = os.path.join(BASE_INPUT_DIR, 'train_labels.csv')\n",
    "\n",
    "# Define the base model path\n",
    "QWEN_BASE_MODEL_PATH = kagglehub.model_download(\"qwen-lm/qwen-3/transformers/0.6b\")\n",
    "\n",
    "# Output directory for the fine-tuned model and results\n",
    "BASE_OUTPUT_DIR = \"./kaggle/working\"\n",
    "FINE_TUNED_MODEL_OUTPUT_DIR = os.path.join(BASE_OUTPUT_DIR, \"qwen_finetuned_dataset_classifier\")\n",
    "FINAL_RESULTS_CSV_PATH = os.path.join(BASE_OUTPUT_DIR, \"article_dataset_classification.csv\")\n",
    "\n",
    "# Load a spaCy model (e.g., 'en_core_web_sm')\n",
    "# python -m spacy download en_core_web_sm \n",
    "NLP_SPACY = spacy.load(\"en_core_web_sm\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "id": "2e50e08c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- 2. Information Extraction (IE) - Dataset Identification ---\n",
    "\n",
    "# Regex patterns for common dataset identifiers\n",
    "# DOI_PATTERN = r'10\\.\\d{4,5}/[-._;()/:A-Za-z0-9\\u002D\\u2010\\u2011\\u2012\\u2013\\u2014\\u2015]+'\tDOI_PATTERN\n",
    "# DOI_PATTERN = r'10\\.\\s?\\d{4,5}\\/[-._()<>;\\/:A-Za-z0-9]+\\s?(?:(?![A-Z]+)(?!\\d{1,3}\\.))+[-._()<>;\\/:A-Za-z0-9]+'\n",
    "#DOI_PATTERN = r'\\bhttps://doi.org/10\\.\\d{4,5}\\/[-._\\/:A-Za-z0-9]+'\n",
    "DOI_PATTERN = r'\\b10\\.\\d{4,5}\\/[-._\\/:A-Za-z0-9]+'\n",
    "EPI_PATTERN = r'\\bEPI[-_A-Z0-9]{2,}'\n",
    "SAM_PATTERN = r'\\bSAMN[0-9]{2,}'          # SAMN07159041\n",
    "IPR_PATTERN = r'\\bIPR[0-9]{2,}'\n",
    "CHE_PATTERN = r'\\bCHEMBL[0-9]{2,}'\n",
    "PRJ_PATTERN = r'\\bPRJ[A-Z0-9]{2,}'\n",
    "E_G_PATTERN = r'\\bE-[A-Z]{4}-[0-9]{2,}'   # E-GEOD-19722 or E-PROT-100\n",
    "ENS_PATTERN = r'\\bENS[A-Z]{4}[0-9]{2,}'\n",
    "CVC_PATTERN = r'\\bCVCL_[A-Z0-9]{2,}'\n",
    "EMP_PATTERN = r'\\bEMPIAR-[0-9]{2,}'\n",
    "PXD_PATTERN = r'\\bPXD[0-9]{2,}'\n",
    "HPA_PATTERN = r'\\bHPA[0-9]{2,}'\n",
    "SRR_PATTERN = r'\\bSRR[0-9]{2,}'\n",
    "GSE_PATTERN = r'\\b(GSE|GSM|GDS|GPL)\\d{4,6}\\b' # Example for GEO accession numbers (e.g., GSE12345, GSM12345)\n",
    "GNB_PATTERN = r'\\b[A-Z]{1,2}\\d{5,6}\\b' # GenBank accession numbers (e.g., AB123456, AF000001)\n",
    "CAB_PATTERN = r'\\bCAB[0-9]{2,}'\n",
    "\n",
    "# Combine all patterns into a list\n",
    "DATASET_ID_PATTERNS = [\n",
    "    DOI_PATTERN,\n",
    "    EPI_PATTERN,\n",
    "    SAM_PATTERN,\n",
    "    IPR_PATTERN,\n",
    "    CHE_PATTERN,\n",
    "    PRJ_PATTERN,\n",
    "    E_G_PATTERN,\n",
    "    ENS_PATTERN,\n",
    "    CVC_PATTERN,\n",
    "    EMP_PATTERN,\n",
    "    PXD_PATTERN,\n",
    "    HPA_PATTERN,\n",
    "    SRR_PATTERN,\n",
    "    GSE_PATTERN,\n",
    "    GNB_PATTERN,\n",
    "    CAB_PATTERN,\n",
    "]\n",
    "\n",
    "# Compile all patterns for efficiency\n",
    "COMPILED_DATASET_ID_REGEXES = [re.compile(p) for p in DATASET_ID_PATTERNS]\n",
    "\n",
    "# Data related keywords to look for in the text\n",
    "# These keywords help to ensure that the text is relevant to datasets\n",
    "DATA_RELATED_KEYWORDS = ['data release', 'download', 'program data', 'data availability', 'the data', 'dataset', 'database', 'repository', 'data source', 'data access', 'archive', 'arch.', 'digital']\n",
    "\n",
    "def is_text_data_related(text: str) -> bool:\n",
    "    if not text:\n",
    "        return False\n",
    "    \n",
    "    text_lower = text.lower()\n",
    "    return any(keyword in text_lower for keyword in DATA_RELATED_KEYWORDS)\n",
    "\n",
    "def text_has_dataset_id(text: str) -> bool:\n",
    "    \"\"\"\n",
    "    Check if the given text contains any dataset identifier.\n",
    "    \n",
    "    Args:\n",
    "        text (str): The text to check for dataset identifiers.\n",
    "        \n",
    "    Returns:\n",
    "        bool: True if any dataset identifier is found, False otherwise.\n",
    "    \"\"\"\n",
    "\n",
    "    occurrences_with_context: list[str] = []\n",
    "    for regex in COMPILED_DATASET_ID_REGEXES:\n",
    "        if regex.search(text):\n",
    "            text_lower = text.lower()\n",
    "            # Check for specific keywords in the text\n",
    "            if any(keyword in text_lower for keyword in DATA_RELATED_KEYWORDS):\n",
    "                return True\n",
    "    return False\n",
    "\n",
    "def extract_dataset_ids(text: str, context_chars: int = 250) -> str:\n",
    "    \"\"\"\n",
    "    Extract dataset identifiers with context from the given text.\n",
    "    \n",
    "    Args:\n",
    "        text (str): The text to search for dataset identifiers.\n",
    "        context_chars (int): Number of characters to include before and after the match for context.\n",
    "        \n",
    "    Returns:\n",
    "        list[str]: A list of extracted dataset identifiers with context.\n",
    "    \"\"\"\n",
    "    text = text.replace('\\u200b', '')\n",
    "    is_small_context = len(text) < context_chars * 2\n",
    "    dataset_ids: list[str] = []\n",
    "    occurrences_with_context: list[str] = []\n",
    "    if is_text_data_related(text):\n",
    "        for regex in COMPILED_DATASET_ID_REGEXES:\n",
    "            matches = regex.finditer(text, re.IGNORECASE)\n",
    "            for match in matches:\n",
    "                dataset_id = text[match.start() : match.end()]\n",
    "                if is_small_context:\n",
    "                    dataset_ids.append(dataset_id)\n",
    "                else:\n",
    "                    extracted_snippet = text[max(0, match.start() - context_chars): match.end() + context_chars ]\n",
    "                    extracted_snippet = extracted_snippet.replace('\\n', '').replace('[', '').replace(']', '')\n",
    "                    extracted_snippet = re.sub(r'\\s+', ' ', extracted_snippet).strip()\n",
    "                    if is_text_data_related(extracted_snippet):\n",
    "                        occurrences_with_context.append(\"{\" + f'\"dataset_ids\": {[dataset_id]}, citation_context: \"{extracted_snippet}\"' + \"}\")\n",
    "        if dataset_ids:\n",
    "            occurrences_with_context.append(\"{\" + f'\"dataset_ids\": {dataset_ids}, citation_context: \"{text}\"' + \"}\")\n",
    "    \n",
    "    # If no occurrences found, return an empty string\n",
    "    # Otherwise, join the occurrences with a specific separator\n",
    "    if not occurrences_with_context:\n",
    "        return \"\"\n",
    "    return \",\".join(occurrences_with_context)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "bec3b4d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use NLP to get sentences from the given text\n",
    "NON_STD_UNICODE_DASHES = re.compile(r'[\\u2010\\u2011\\u2012\\u2013\\u2014]')\n",
    "def get_sentences_from_text(text: str, nlp=NLP_SPACY) -> str:\n",
    "    if not text:\n",
    "        return \"\"\n",
    "    \n",
    "    # Replace all non-standard unicode dashes with '-'\n",
    "    text = NON_STD_UNICODE_DASHES.sub('-', text)\n",
    "    text = text.replace('-\\n', '-').replace('_\\n', '_').replace('/\\n', '/').replace('\\n', ' ').replace('\\u200b', '').strip()\n",
    "    doc_spacy = nlp(text)\n",
    "    return \"\\n\".join([sent.text for sent in doc_spacy.sents])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "8dc17d8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_author_names(full_text: str, nlp=NLP_SPACY) -> str:\n",
    "    \"\"\"\n",
    "    Extracts potential author names from the beginning of a research article's text\n",
    "    using spaCy's Named Entity Recognition. It attempts to isolate the author section\n",
    "    and applies heuristics to filter out non-author entities.\n",
    "\n",
    "    Args:\n",
    "        full_text (str): The complete text content of the research article,\n",
    "                         typically extracted from a PDF.\n",
    "\n",
    "    Returns:\n",
    "        List[str]: A list of unique strings, each representing a potential author name,\n",
    "                   sorted alphabetically. Returns an empty list if no authors are found.\n",
    "    \"\"\"\n",
    "    if not full_text or not full_text.strip():\n",
    "        return []\n",
    "\n",
    "    full_text = full_text.replace('1\\n,', ',').replace('1,', ',').replace('\\u2019', \"'\")\n",
    "\n",
    "    # 1. Isolate the potential author section\n",
    "    # Authors are typically at the very beginning, before the abstract or introduction.\n",
    "    # We'll search for common section headers to define the end of the author block.\n",
    "    # Using regex for case-insensitive search and handling various newline/spacing.\n",
    "    header_patterns = [\n",
    "        r\"\\n\\s*Abstract\\s*\\n\",\n",
    "        r\"\\n\\s*Introduction\\s*\\n\",\n",
    "        r\"\\n\\s*Summary\\s*\\n\",\n",
    "        r\"\\n\\s*Keywords\\s*\\n\",\n",
    "        r\"\\n\\s*Graphical Abstract\\s*\\n\",\n",
    "        r\"\\n\\s*1\\.\\s*Introduction\\s*\\n\", # Common for numbered sections\n",
    "        r\"\\n\\s*DOI:\\s*\\n\" # Sometimes DOI appears before abstract\n",
    "    ]\n",
    "\n",
    "    author_section_end_index = len(full_text)\n",
    "    for pattern in header_patterns:\n",
    "        match = re.search(pattern, full_text, re.IGNORECASE)\n",
    "        if match:\n",
    "            # Take text up to the start of the found header\n",
    "            author_section_end_index = min(author_section_end_index, match.start())\n",
    "            break\n",
    "    \n",
    "    # As a fallback or if no header is found early, limit the search to the first\n",
    "    # 2500 characters. This prevents processing the entire document for authors.\n",
    "    author_section_text = full_text[:min(author_section_end_index, 2500)]\n",
    "\n",
    "    if not author_section_text.strip():\n",
    "        return []\n",
    "\n",
    "    # 2. Process the isolated author section with spaCy\n",
    "    doc = nlp(author_section_text)\n",
    "\n",
    "    # 3. Extract PERSON entities and apply initial filtering\n",
    "    potential_authors: set[str] = set()\n",
    "    for ent in doc.ents:\n",
    "        if ent.label_ == \"PERSON\":\n",
    "            name = ent.text.strip()\n",
    "            # Basic filtering to reduce false positives:\n",
    "            # - Exclude very short strings (e.g., single letters, common conjunctions)\n",
    "            # - Exclude common stop words (e.g., \"The\", \"And\")\n",
    "            # - Exclude all-uppercase strings that might be acronyms (e.g., \"WHO\", \"NASA\")\n",
    "            # - Ensure it contains at least one space (e.g., \"John Doe\") or is a capitalized\n",
    "            #   single word that's longer than 2 characters (e.g., \"Smith\").\n",
    "            if (len(name) > 1 and\n",
    "                name.lower() not in nlp.Defaults.stop_words and\n",
    "                not name.isupper() and\n",
    "                (' ' in name or (name[0].isupper() and len(name) > 2))):\n",
    "                \n",
    "                potential_authors.add(name)\n",
    "\n",
    "    # 4. Apply more advanced heuristics to filter out non-author names\n",
    "    # This step is crucial for accuracy and often requires tuning.\n",
    "    filtered_authors = set()\n",
    "    for author in potential_authors:\n",
    "        # Heuristic 1: Filter out names that contain common affiliation keywords.\n",
    "        # This is a simple check; more robust solutions might use spaCy's dependency\n",
    "        # parsing to check if a PERSON entity is part of an ORG entity.\n",
    "        affiliation_keywords = [\"univ\", \"observ\", \"institute\", \"department\", \"center\", \"lab\",\n",
    "                                \"hospital\", \"college\", \"school\", \"inc.\", \"ltd.\", \"company\",\n",
    "                                \"corp.\", \"group\", \"foundation\", \"research\"]\n",
    "        if any(keyword in author.lower() for keyword in affiliation_keywords):\n",
    "            continue # Skip if it looks like an affiliation\n",
    "\n",
    "        # Heuristic 2: Filter out names that contain email patterns or ORCID patterns.\n",
    "        if '@' in author or re.search(r'\\b\\d{4}-\\d{4}-\\d{4}-\\d{3}[\\dX]\\b', author):\n",
    "            continue # Skip if it contains an email or ORCID\n",
    "\n",
    "        # Heuristic 3: Filter out names that are likely just initials or very short.\n",
    "        # This is partially covered by initial filtering, but can be refined.\n",
    "        # E.g., \"J. D.\" might be an author, but \"J.\" alone is unlikely.\n",
    "        if len(author.split()) == 1 and len(author) <= 2 and author.isupper():\n",
    "            continue # Skip single-letter or two-letter uppercase (e.g., \"JD\")\n",
    "\n",
    "        filtered_authors.add(author)\n",
    "\n",
    "    # Convert to list and sort for consistent output\n",
    "    return \", \".join(list(filtered_authors))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "0b7a53c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def extract_element_text(element: Element | None) -> str:\n",
    "    if element is not None:\n",
    "        # Use itertext() to get all text content from the <p> tag and its descendants\n",
    "        # and join them into a single string.\n",
    "        all_text = \" \".join(element.itertext(tag=None)).replace('\\u200b', '').strip()\n",
    "        return all_text[:2000]\n",
    "    else:\n",
    "        return \"\"\n",
    "    \n",
    "def extract_next_sibling_text(elements: list[Element] | None, sibling_xpath: str) -> str:\n",
    "    \"\"\"\n",
    "    Extracts text from the next sibling of the given XML element.\n",
    "    \n",
    "    Args:\n",
    "        element (Element | None): The XML element whose next sibling's text is to be extracted.\n",
    "        sibling_xpath (str): The XPath expression to find the next sibling element. (eg. \"following-sibling::passage[1]\")\n",
    "        \n",
    "    Returns:\n",
    "        str: A string containing the text from the next sibling element, or an empty string if no sibling exists.\n",
    "\n",
    "    \"\"\"\n",
    "    # Check if the provided elements list is None or empty\n",
    "    if not elements:\n",
    "        return \"\"\n",
    "    \n",
    "    # Assuming there's only one such element, take the first one found\n",
    "    # and find the element immediately following based on the given sibling_xpath.\n",
    "    first_element = elements[0]\n",
    "    sibling_elements = first_element.xpath(sibling_xpath)\n",
    "\n",
    "    if not sibling_elements:\n",
    "        # print(\"DEBUG: No following <passage> element found.\") # Uncomment for debugging\n",
    "        return \"\"\n",
    "    \n",
    "    next_sibling = sibling_elements[0]\n",
    "    if next_sibling is None:\n",
    "        return \"\"\n",
    "    \n",
    "    return extract_element_text(next_sibling)\n",
    "\n",
    "def extract_elements_text(elements: list[Element] | None, sep: str = \" \") -> str:\n",
    "    elements_text = []\n",
    "    if elements is None:\n",
    "        return \"\"\n",
    "    \n",
    "    for element in elements:\n",
    "        text = extract_element_text(element)\n",
    "        if text:\n",
    "            elements_text.append(text)\n",
    "\n",
    "    return sep.join(elements_text).strip()\n",
    "\n",
    "def extract_elements_text_from_xpath_list(root: Element | None, xpath_list: list[str], ns: dict[str, str] | None = None) -> str:\n",
    "    elements_text = \"\"\n",
    "    if root is None or not xpath_list:\n",
    "        return \"\"\n",
    "    \n",
    "    for xpath in xpath_list:\n",
    "        element = root.find(xpath, namespaces=ns)\n",
    "        elements_text += extract_element_text(element)\n",
    "    return elements_text\n",
    "\n",
    "def extract_text_from_elements_within_element(element: Element | None, child_xpaths: list[str] = [], ns: dict[str, str] | None = None) -> str:\n",
    "    \"\"\"\n",
    "    Extracts text from elements within a given XML element that match the specified tag names.\n",
    "    \n",
    "    Args:\n",
    "        element (Element | None): The XML element to search within.\n",
    "        tag_names (list[str]): A list of tag names to search for.\n",
    "        \n",
    "    Returns:\n",
    "        str: A string containing the extracted text from the matching elements.\n",
    "    \"\"\"\n",
    "    if element is None:\n",
    "        return \"\"\n",
    "    \n",
    "    if not child_xpaths:\n",
    "        # If no child tag names are provided, return the text of the element itself\n",
    "        return extract_element_text(element)\n",
    "    \n",
    "    extracted_text = []\n",
    "    for xpath in child_xpaths:\n",
    "        for child in element.findall(xpath, namespaces=ns):\n",
    "            text = extract_element_text(child)\n",
    "            if text:\n",
    "                extracted_text.append(text)\n",
    "    \n",
    "    return \"|\".join(extracted_text)\n",
    "\n",
    "def extract_data_related_elements_text(elements: list[Element] | None, child_xpaths: list[str] = [], ns: dict[str, str] | None = None) -> list[str]:\n",
    "    elements_text = []\n",
    "    if elements is None:\n",
    "        return elements_text\n",
    "    \n",
    "    for element in elements:\n",
    "        text = extract_dataset_ids(extract_text_from_elements_within_element(element, child_xpaths, ns))\n",
    "        if text:\n",
    "            elements_text.append(text)\n",
    "\n",
    "    return elements_text\n",
    "\n",
    "def extract_data_related_elements_text_from_xpath_list(root: Element | None, xpath_list: list[str], ns: dict[str, str] | None = None) -> list[str]:\n",
    "    \"\"\"\n",
    "    Extracts text from elements in the XML tree that match the provided XPath expressions.\n",
    "    \n",
    "    Args:\n",
    "        root (Element | None): The root element of the XML tree.\n",
    "        xpath_list (list[str]): A list of XPath expressions to search for elements.\n",
    "        \n",
    "    Returns:\n",
    "        list[str]: A list of extracted text from the matching elements.\n",
    "    \"\"\"\n",
    "    elements_text = []\n",
    "    if root is None or not xpath_list:\n",
    "        return elements_text\n",
    "    \n",
    "    for xpath in xpath_list:\n",
    "        primary_xpath, *child_xpath_text = xpath.split('||')\n",
    "        child_xpaths = child_xpath_text[0].split(',') if child_xpath_text else []\n",
    "        elements = root.findall(primary_xpath, namespaces=ns)\n",
    "        if elements:\n",
    "            elements_text.extend(extract_data_related_elements_text(elements, child_xpaths, ns))\n",
    "    return elements_text\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "02f6ceb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_pdf_text(pdf_doc: fitz.Document)  -> dict[str, str | list[str]]:\n",
    "    \"\"\"\n",
    "    Extracts all text from a PDF document using PyMuPDF.\n",
    "    \n",
    "    Args:\n",
    "        pdf_doc (fitz.Document): The PDF document to extract text from.\n",
    "        \n",
    "    Returns:\n",
    "        str: A JSON string of the article_dict containing specific elements extracted from the PDF.\n",
    "    \"\"\"\n",
    "\n",
    "    # Initialize the article dictionary with empty strings\n",
    "    article_dict = {\n",
    "        'title': '',\n",
    "        'authors': '',\n",
    "        'abstract': '',\n",
    "        'data_availability': '',\n",
    "        'other_dataset_citations': []\n",
    "    }\n",
    "\n",
    "    # Initialize variables for text extraction\n",
    "    p1 = None  # Placeholder for the first page text\n",
    "    other_dataset_citations = set()  # Use a set to avoid duplicates\n",
    "    for page in pdf_doc:\n",
    "        # Extract text from the page\n",
    "        textpage = page.get_textpage()\n",
    "        if page.number == 0:\n",
    "            p1_txt = textpage.extractTEXT()\n",
    "            p1 = get_sentences_from_text(p1_txt)\n",
    "            p1 = p1[:int(len(p1)/2)]\n",
    "            article_dict['authors'] = extract_author_names(p1_txt, nlp=NLP_SPACY)\n",
    "\n",
    "        # Extract text from all blocks that have an abstract or dataset id's\n",
    "        blocks = textpage.extractBLOCKS()\n",
    "        for block in blocks:\n",
    "            block_text = get_sentences_from_text(block[4])\n",
    "            block_text_lower = block_text.lower()\n",
    "            if page.number == 0 and len(block_text) > 100 and \"abstract\" in block_text_lower:\n",
    "                # Add the abstract block text to the article dictionary\n",
    "                article_dict['abstract'] = block_text\n",
    "            elif \"data availability\" in block_text_lower or \"data accessibility\" in block_text_lower:\n",
    "                # Add the data availability block text to the article dictionary\n",
    "                article_dict['data_availability'] = block_text\n",
    "            else:\n",
    "                context_chars = min(250, len(block_text))  # Use a minimum\n",
    "                dataset_ids_found = extract_dataset_ids(block_text, context_chars)  # Extract dataset IDs from the block text\n",
    "                if dataset_ids_found:\n",
    "                    # Append the dataset IDs found in the block to the other_dataset_citations\n",
    "                    other_dataset_citations.add(dataset_ids_found)\n",
    "\n",
    "    article_dict['other_dataset_citations'] = list(other_dataset_citations) if other_dataset_citations else []\n",
    "    # If an abstract was not found, use the first page text as the abstract\n",
    "    if not article_dict['abstract'] and p1:\n",
    "        article_dict['abstract'] = p1\n",
    "\n",
    "    # Return the article dictionary as a JSON string\n",
    "    return article_dict\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "id": "de668c1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "def extract_xml_text_jats(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availablity info for Journal Archiving and Interchange DTD (JATS)\n",
    "    # The \".//\" ensures it searches anywhere in the document, not just direct children of root.\n",
    "    ns = None  # No namespaces for JATS\n",
    "\n",
    "    xpath_title = \".//article-title\"\n",
    "    xpath_authors_1 = \".//contrib-group/contrib[@contrib-type='author']/name\"\n",
    "    xpath_authors_2 = \".//biblstruct/analytic/author[@role='corresp']/persname\"\n",
    "    authors = extract_elements_text(root.findall(xpath_authors_1, namespaces=ns), sep=\", \")\n",
    "    if not authors:\n",
    "        authors = extract_elements_text(root.findall(xpath_authors_2, namespaces=ns), sep=\", \")\n",
    "    xpath_abstract = \".//abstract\"\n",
    "    xpath_data_avails = [\".//notes[@notes-type='data-availability']\", \".//sec[@sec-type='data-availability']\"]\n",
    "    xpath_citations = [\".//element-citation||.article-title,.source,.pub-id\", \".//mixed-citation\"]  # List of XPath expressions for citations\n",
    "\n",
    "    return {\n",
    "        'title': extract_element_text(root.find(xpath_title, ns)),\n",
    "        'authors': authors,\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, ns)),\n",
    "        'data_availability': extract_elements_text_from_xpath_list(root, xpath_data_avails, ns=ns),\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "fec2e79e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_tei(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for Text Encoding Initiative (TEI)\n",
    "    # Set the namespace for TEI\n",
    "    ns = {'tei': 'http://www.tei-c.org/ns/1.0'}\n",
    "\n",
    "    xpath_title = \".//tei:title\"\n",
    "    xpath_authors = \".//tei:sourcedesc/tei:biblstruct/tei:analytic/tei:author/tei:persname\"\n",
    "    xpath_abstract = \".//tei:abstract\"\n",
    "    xpath_data_avail = \"\" #\".//tei:biblstruct\"\n",
    "    xpath_citations = [\".//tei:biblstruct||.//tei:title,.//tei:idno,.//tei:notes\"]  # List of XPath expressions for citations\n",
    "        \n",
    "    return {\n",
    "        'title': extract_element_text(root.find(xpath_title, namespaces=ns)),\n",
    "        'authors': extract_elements_text(root.findall(xpath_authors, namespaces=ns), sep=\", \"),\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, namespaces=ns)),\n",
    "        'data_availability': xpath_data_avail,  # No direct extraction for TEI data_availability\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "06311266",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_wiley(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for Wiley XML format\n",
    "    # Set the namespace for Wiley\n",
    "    ns = {'ns': 'http://www.wiley.com/namespaces/wiley'}\n",
    "\n",
    "    xpath_title = \".//ns:publicationMeta[@level='part']/ns:titleGroup\"    #<publicationMeta level=\"part\"><titleGroup><title type=\"main\">\n",
    "    xpath_authors = \".//selfCitationGroup/citation[@type='self']/author\"\n",
    "    xpath_abstract = \".//ns:abstract[@type='main']\"  #<abstract type=\"main\"\n",
    "    xpath_data_avail = \".//ns:section[@type='dataAvailability']\"  #<section numbered=\"no\" type=\"dataAvailability\"\n",
    "    xpath_citations = [\".//ns:citation||.//ns:articleTitle,.//ns:journalTitle,.//ns:url\"]  # List of XPath expressions for citations\n",
    "        \n",
    "    return {\n",
    "        'title': extract_elements_text(root.findall(xpath_title, namespaces=ns)),\n",
    "        'authors': extract_elements_text(root.findall(xpath_authors, namespaces=ns), sep=\", \"),\n",
    "        'abstract': extract_element_text(root.find(xpath_abstract, namespaces=ns)),\n",
    "        'data_availability': extract_element_text(root.find(xpath_data_avail, namespaces=ns)),\n",
    "        'other_dataset_citations': extract_data_related_elements_text_from_xpath_list(root, xpath_citations, ns=ns),\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "dfc6c86c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_bioc(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for BioC-API XML format\n",
    "    ns = None  # No namespaces for BioC\n",
    "\n",
    "    xpath_title = \"string(.//passage[infon[@key='section_type' and text()='TITLE']]/text)\"\n",
    "    xpath_authors = \"string(.//infon[@key='name_0'] | .//infon[@key='name_1'] | .//infon[@key='name_2'] | .//infon[@key='name_3'])\"\n",
    "    xpath_abstract = \"string(.//passage[infon[@key='section_type' and text()='ABSTRACT']]/text)\"\n",
    "    xpath_data_avail = \".//passage[text[text()='DATA ACCESSIBILITY:']]\"\n",
    "    xpath_data_avail_sibling = \"following-sibling::passage[1]\"\n",
    "    xpath_citations = []\n",
    "        \n",
    "    return {\n",
    "        'title': root.xpath(xpath_title, namespaces=ns),\n",
    "        'authors': root.xpath(xpath_authors, namespaces=ns).strip().replace('surname:', '').replace(';given-names:', ' '),\n",
    "        'abstract': root.xpath(xpath_abstract, namespaces=ns)[:2000],  # Limit to 2000 characters\n",
    "        'data_availability': extract_next_sibling_text(root.xpath(xpath_data_avail, namespaces=ns), xpath_data_avail_sibling),\n",
    "        'other_dataset_citations': xpath_citations,\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "id": "58a53b42",
   "metadata": {},
   "outputs": [],
   "source": [
    "def extract_xml_text_taxonx(root: Element) -> dict[str, str | list[str]]:\n",
    "    # Find the title, abstract, and data availability info for TaxonX format\n",
    "    ns = None  # No namespaces for Taxonomic Treatment Publishing DTD\n",
    "\n",
    "    xpath_title = \"string(.//article-meta/title-group/article-title)\"\n",
    "    xpath_authors = \"\"\n",
    "    xpath_abstract = \"string(.//article-meta/abstract)\"\n",
    "    xpath_data_avail = \"\"\n",
    "    xpath_citations = []\n",
    "        \n",
    "    return {\n",
    "        'title': root.xpath(xpath_title, namespaces=ns),\n",
    "        'authors': xpath_authors,\n",
    "        'abstract': root.xpath(xpath_abstract, namespaces=ns)[:2000],  # Limit to 2000 characters\n",
    "        'data_availability': xpath_data_avail,  # No direct extraction for TaxonX data_availability\n",
    "        'other_dataset_citations': xpath_citations,\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "a0a58f0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dictionary mapping XML types to their respective extraction functions\n",
    "XML_TYPE_EXTRACTORS = {\n",
    "    'jats': extract_xml_text_jats,\n",
    "    'tei': extract_xml_text_tei,\n",
    "    'wiley': extract_xml_text_wiley,\n",
    "    'bioc': extract_xml_text_bioc,\n",
    "    'taxonx': extract_xml_text_taxonx,\n",
    "}\n",
    "\n",
    "# --- Data Loading ---\n",
    "def load_file_paths(dataset_type_dir: str) -> pd.DataFrame: \n",
    "    pdf_path = os.path.join(dataset_type_dir, 'PDF')\n",
    "    xml_path = os.path.join(dataset_type_dir, 'XML')\n",
    "    dataset_type = os.path.basename(dataset_type_dir)\n",
    "    pdf_files = [f for f in os.listdir(pdf_path) if f.endswith('.pdf')]\n",
    "    xml_files = [f for f in os.listdir(xml_path) if f.endswith('.xml')]\n",
    "    df_pdf = pd.DataFrame({\n",
    "        'article_id': [f.replace('.pdf', '') for f in pdf_files],\n",
    "        'pdf_file_path': [os.path.join(pdf_path, f) for f in pdf_files]\n",
    "    })\n",
    "    df_xml = pd.DataFrame({\n",
    "        'article_id': [f.replace('.xml', '') for f in xml_files],\n",
    "        'xml_file_path': [os.path.join(xml_path, f) for f in xml_files]\n",
    "    })\n",
    "    merge_df = pd.merge(df_pdf, df_xml, on='article_id', how='outer', suffixes=('_pdf', '_xml'), validate=\"one_to_many\")\n",
    "    merge_df['dataset_type'] = dataset_type\n",
    "    return merge_df\n",
    "\n",
    "def read_pdf_text(file_path: str, xml_type: str | None = None) -> dict[str, str | list[str]]:\n",
    "    \"\"\"Extracts all text from a PDF file using PyMuPDF.\"\"\"\n",
    "    article_dict = {}\n",
    "    if file_path and os.path.exists(file_path):\n",
    "        try:\n",
    "            with fitz.open(file_path) as doc:\n",
    "                article_dict = extract_pdf_text(doc)  # Extract text from the PDF document\n",
    "        except Exception as e:\n",
    "            print(f\"Error reading PDF {file_path}: {e}\")\n",
    "    else:\n",
    "        print(f\"PDF file not found: {file_path}\")\n",
    "    \n",
    "    return article_dict\n",
    "\n",
    "def read_xml_text(file_path: str, xml_type: str) -> dict[str, str | list[str]]:\n",
    "    \"\"\"Reads and extracts text from an XML file based on the specified XML type.\n",
    "    Args:\n",
    "        file_path (str): The path to the XML file.\n",
    "        xml_type (str): The type of XML format (e.g., 'jats', 'tei', 'wiley', 'bioc', 'taxonx').\n",
    "    Returns:\n",
    "        dict: A dictionary containing the extracted text from the XML file.\n",
    "    \"\"\"\n",
    "    # Initialize the article dictionary\n",
    "    article_dict = {}\n",
    "    if file_path and os.path.exists(file_path):\n",
    "        # Disable external entity resolution for security\n",
    "        parser = etree.XMLParser(resolve_entities=False, no_network=True)\n",
    "        try:\n",
    "            tree = etree.parse(file_path, parser)\n",
    "            root = tree.getroot()\n",
    "            # Use the appropriate extraction function based on the xml_type\n",
    "            extract_function = XML_TYPE_EXTRACTORS.get(xml_type, extract_xml_text_jats)  \n",
    "            article_dict = extract_function(root)\n",
    "        except Exception as e:\n",
    "            print(f\"Error reading XML {file_path}: {e}\")\n",
    "    else:\n",
    "        print(f\"XML file not found: {file_path}\")    \n",
    "    return article_dict\n",
    "\n",
    "def process_unsupported_file(file_path: str, xml_type: str | None = None) -> str:\n",
    "    return f\"Unsupported file type for: {file_path}\"\n",
    "\n",
    "# Dictionary mapping file extensions to loading functions\n",
    "FILE_LOADERS = {\n",
    "    '.xml': read_xml_text,\n",
    "    '.pdf': read_pdf_text,\n",
    "}\n",
    "\n",
    "def load_article_text(file_path: str, xml_type: str | None = None) -> str:\n",
    "    \"\"\"\n",
    "    Loads text content from a single article file (PDF or XML).\n",
    "    Returns the text content of the given file.\n",
    "    \"\"\"\n",
    "    text_content = \"\"\n",
    "\n",
    "    # Get the file extension (e.g., '.xml', '.pdf')\n",
    "    _, file_extension = os.path.splitext(file_path)\n",
    "    file_extension = file_extension.lower() # Ensure lowercase for consistent lookup\n",
    "\n",
    "    # Get the appropriate function from the dictionary,\n",
    "    # or fall back to a default 'unsupported' function if not found.\n",
    "    loader_function = FILE_LOADERS.get(file_extension, process_unsupported_file)\n",
    "\n",
    "    # Call the selected function\n",
    "    article_dict = loader_function(file_path, xml_type=xml_type)\n",
    "    text_content = json.dumps(article_dict, separators=(',', ':'))\n",
    "    print(f\"Extracted text from {file_path}. Length: {len(text_content)} characters\")\n",
    "\n",
    "    return text_content\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "5f45bff0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_mec.16977.xml. Length: 1824 characters\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1824"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "'{\"title\":\"Genetic basis of ecologically relevant body shape variation among four genera of cichlid fishes\",\"authors\":\"DeLorenzo Leah\",\"abstract\":\"Divergence in body shape is one of the most widespread and repeated patterns of morphological variation in fishes and is associated with habitat specification and swimming mechanics. Such ecological diversification is the first stage of the explosive adaptive radiation of cichlid fishes in the East African Rift Lakes. We use two hybrid crosses of cichlids (Metriaclima sp. \\\\u00c3\\\\u0097 Aulonocara sp. and Labidochromis sp. \\\\u00c3\\\\u0097 Labeotropheus sp., >975 animals total) to determine the genetic basis of body shape diversification that is similar to benthic-pelagic divergence across fishes. Using a series of both linear and geometric shape measurements, we identify 34 quantitative trait loci (QTL) that underlie various aspects of body shape variation. These QTL are spread throughout the genome, each explain 3.2\\\\u00e2\\\\u0080\\\\u00938.6% of phenotypic variation, and are largely modular. Further, QTL are distinct both between these two crosses of Lake Malawi cichlids and compared to previously identified QTL for body shape in fishes such as sticklebacks. We find that body shape is controlled by many genes of small effect. In all, we find that convergent body shape phenotypes commonly observed across fish clades are most likely due to distinct genetic and molecular mechanisms.\",\"data_availability\":\"SUPPL paragraph 43163 Raw sequence data are available at https://www.ncbi.nlm.nih.gov/bioproject/PRJNA955776. Additional data are available at Dryad https://doi.org/10.5061/dryad.4mw6m90cz. Dryad files include phenotypic measures, TPS files for geometric morphometric analysis, and genotypes used for quantitative trait loci mapping.\",\"other_dataset_citations\":[]}'"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Test loading various PDF files\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_2017jc013030.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1017_rdc.2022.19.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1017_s0007123423000601.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.3389_fcimb.2024.1292467.pdf')\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_esp.5058.pdf') # This one is big\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_esp.5059.pdf') # This one is big\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'PDF', '10.1002_ece3.4466.pdf') # dryad\n",
    "#pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1002_ece3.4466.xml') # dryad\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1002_mp.14424.xml')\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1007_s00259-022-06053-8.xml')    # jats\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1007_s00382-022-06361-7.xml')    # tei\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1111_1365-2435.13431.xml')       # wiley\n",
    "pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1111_mec.16977.xml')             # bioc\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.3897_zoologia.35.e23481.xml')      # taxonx\n",
    "# pdf_file_path = os.path.join(ARTICLE_TRAIN_DIR, 'XML', '10.1002_ece3.6144.xml')               # jats\n",
    "file_text = load_article_text(pdf_file_path, 'bioc')\n",
    "display(len(file_text))\n",
    "file_text"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45dbdf50",
   "metadata": {},
   "source": [
    "## EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "29d7f58e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading labeled training data from: ./kaggle/input/make-data-count-finding-data-references\\train_labels.csv\n",
      "Training labels shape: (1028, 3)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "type",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "fc0c67c7-c210-49d4-b670-8c34485397b0",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "https://doi.org/10.17882/49388",
         "Primary"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "Missing",
         "Missing"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "Missing",
         "Missing"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "Missing",
         "Missing"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "Missing",
         "Missing"
        ]
       ],
       "shape": {
        "columns": 3,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>https://doi.org/10.17882/49388</td>\n",
       "      <td>Primary</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>Missing</td>\n",
       "      <td>Missing</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                      dataset_id     type\n",
       "0    10.1002_2017jc013030  https://doi.org/10.17882/49388  Primary\n",
       "1  10.1002_anie.201916483                         Missing  Missing\n",
       "2  10.1002_anie.202005531                         Missing  Missing\n",
       "3  10.1002_anie.202007717                         Missing  Missing\n",
       "4  10.1002_chem.201902131                         Missing  Missing"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Grouped dataset ID counts:\n",
      "   dataset_id_trim  article_id\n",
      "52             htt         325\n",
      "29             Mis         309\n",
      "20             EPI          64\n",
      "47             SAM          41\n",
      "25             IPR          33\n",
      "11             CHE          29\n",
      "41             PRJ          26\n",
      "16             E-G          25\n",
      "19             ENS          21\n",
      "26             K02          20\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_id_trim",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "341f70e6-078f-4124-a23d-d9bed174ed99",
       "rows": [
        [
         "923",
         "10.3390_v11060565",
         "EPI954553",
         "Secondary",
         "EPI"
        ],
        [
         "382",
         "10.1128_JVI.01717-21",
         "EPI_ISL_332399",
         "Primary",
         "EPI"
        ],
        [
         "925",
         "10.3390_v11060565",
         "EPI954555",
         "Secondary",
         "EPI"
        ]
       ],
       "shape": {
        "columns": 4,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>type</th>\n",
       "      <th>dataset_id_trim</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>923</th>\n",
       "      <td>10.3390_v11060565</td>\n",
       "      <td>EPI954553</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>10.1128_JVI.01717-21</td>\n",
       "      <td>EPI_ISL_332399</td>\n",
       "      <td>Primary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>925</th>\n",
       "      <td>10.3390_v11060565</td>\n",
       "      <td>EPI954555</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>EPI</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id      dataset_id       type dataset_id_trim\n",
       "923     10.3390_v11060565       EPI954553  Secondary             EPI\n",
       "382  10.1128_JVI.01717-21  EPI_ISL_332399    Primary             EPI\n",
       "925     10.3390_v11060565       EPI954555  Secondary             EPI"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the labeled training data CSV file\n",
    "print(f\"Loading labeled training data from: {LABELED_TRAINING_DATA_CSV_PATH}\")\n",
    "train_labels_df = pd.read_csv(LABELED_TRAINING_DATA_CSV_PATH)\n",
    "\n",
    "print(f\"Training labels shape: {train_labels_df.shape}\")\n",
    "display(train_labels_df.head())\n",
    "\n",
    "# Create a new column 'dataset_id_trim' by extracting the first 3 characters of the 'dataset_id' column\n",
    "train_labels_df['dataset_id_trim'] = train_labels_df['dataset_id'].str[:3]\n",
    "# Find the most frequent types of dataset_id's\n",
    "freq_dataset_id_df = train_labels_df.groupby('dataset_id_trim').count().reset_index()\n",
    "freq_dataset_id_df = freq_dataset_id_df[['dataset_id_trim', 'article_id']].sort_values(by='article_id', ascending=False)\n",
    "print(f\"Grouped dataset ID counts:\\n{freq_dataset_id_df.head(10)}\")\n",
    "\n",
    "# Display the first 10 rows where dataset_id_trim is 'EPI'\n",
    "display(train_labels_df[train_labels_df['dataset_id_trim'] == 'EPI'].sample(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "560a47e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Example grouped training data for article_id '10.1002_2017jc013030': [{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]\n"
     ]
    }
   ],
   "source": [
    "# Group training data by article_id to get all datasets for each article\n",
    "# This creates a dictionary where keys are article_ids and values are lists of dataset dicts\n",
    "grouped_training_data = {}\n",
    "for article_id, group_df in train_labels_df.groupby('article_id'):\n",
    "    grouped_training_data[article_id] = group_df[['dataset_id', 'type']].to_dict('records')\n",
    "\n",
    "# Example usage of grouped_training_data\n",
    "print(f\"Example grouped training data for article_id '10.1002_2017jc013030': {grouped_training_data['10.1002_2017jc013030']}\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "8718ef7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_first_line_of_xml(file_path: str) -> str | None:\n",
    "    \"\"\"\n",
    "    Reads and returns the first line of an XML file.\n",
    "\n",
    "    Args:\n",
    "        file_path (str): The path to the XML file.\n",
    "\n",
    "    Returns:\n",
    "        str | None: The first line of the file, stripped of leading/trailing whitespace,\n",
    "                    or None if the file cannot be read or is empty.\n",
    "    \"\"\"\n",
    "    if not file_path and not os.path.exists(file_path):\n",
    "        return None\n",
    "    \n",
    "    try:\n",
    "        with open(file_path, 'r', encoding='utf-8') as f:\n",
    "            first_line = f.readline().replace('<?xml version=\"1.0\" encoding=\"UTF-8\"?>', '').replace('<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>', '').strip()\n",
    "            # If the first line is empty, read the next line\n",
    "            if not first_line:\n",
    "                first_line = f.readline()\n",
    "            return first_line.strip()[:90] if first_line else None\n",
    "    except UnicodeDecodeError:\n",
    "        try:\n",
    "            with open(file_path, 'r', encoding='iso-8859-1') as f:\n",
    "                first_line = f.readline().replace('<?xml version=\"1.0\" encoding=\"UTF-8\"?>', '').replace('<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>', '').strip()\n",
    "                if not first_line:\n",
    "                    first_line = f.readline()\n",
    "                return first_line.strip()[:90] if first_line else None\n",
    "        except Exception as e:\n",
    "            return None\n",
    "    except Exception as e:\n",
    "        print(f\"Error reading file '{file_path}': {e}\")\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "id": "71dd0895",
   "metadata": {},
   "outputs": [],
   "source": [
    "def identify_xml_type(first_line: str) -> str:\n",
    "    \"\"\"\n",
    "    Identifies the XML type based on the first line of the XML file.\n",
    "    \n",
    "    Args:\n",
    "        file_path (str): The path to the XML file.\n",
    "        \n",
    "    Returns:\n",
    "        str: The identified XML type ('jats', 'tei', 'wiley', 'bioc', or 'unknown').\n",
    "    \"\"\"\n",
    "    if not first_line:\n",
    "        return \"unknown\"\n",
    "    first_line_lower = first_line.lower()\n",
    "    # Check for specific patterns in the first line\n",
    "    if 'journal archiving and interchange dtd' in first_line_lower:\n",
    "        return \"jats\"\n",
    "    elif 'xmlns=\"http://www.tei-c.org/ns/1.0\"' in first_line_lower:\n",
    "        return \"tei\"\n",
    "    elif 'xmlns=\"http://www.wiley.com/namespaces/wiley\"' in first_line_lower:\n",
    "        return \"wiley\"\n",
    "    elif 'bioc.dtd' in first_line_lower or 'bioc-api' in first_line_lower:\n",
    "        return \"bioc\"\n",
    "    elif 'taxonomic treatment publishing dtd' in first_line_lower:\n",
    "        return \"taxonx\"\n",
    "    \n",
    "    return \"unknown\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "34f61a27",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train files paths shape: (524, 7)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "8bba6aaf-285e-4ec4-ae36-d8a2b4a0a6fb",
       "rows": [
        [
         "405",
         "10.1590_0047-2085000000239",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0047-2085000000239.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "259",
         "10.1186_s12870-019-2199-7",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-2199-7.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-2199-7.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "476",
         "10.3390_microorganisms8121872",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_microorganisms8121872.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_microorganisms8121872.xml",
         "train",
         "[{'dataset_id': 'CHEMBL1568820', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL1652119', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL198796', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL243644', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL2436978', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL3109593', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL3311228', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL3752642', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL3960662', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL402487', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL403', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL4065616', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL408500', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL561', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL64391', 'type': 'Secondary'}, {'dataset_id': 'CHEMBL76', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>10.1590_0047-2085000000239</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>10.1186_s12870-019-2199-7</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>10.3390_microorganisms8121872</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'CHEMBL1568820', 'type': 'Seco...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        article_id  \\\n",
       "405     10.1590_0047-2085000000239   \n",
       "259      10.1186_s12870-019-2199-7   \n",
       "476  10.3390_microorganisms8121872   \n",
       "\n",
       "                                         pdf_file_path  \\\n",
       "405  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "259  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "476  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                         xml_file_path dataset_type  \\\n",
       "405                                                           train   \n",
       "259  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "476  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                          dataset_info  \\\n",
       "405     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "259     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "476  [{'dataset_id': 'CHEMBL1568820', 'type': 'Seco...   \n",
       "\n",
       "                                        xml_first_line xml_type  \n",
       "405                                               None     None  \n",
       "259  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "476  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Test files paths shape: (30, 7)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "ddd2b2fe-5ab4-49e9-99c1-2534d03f3270",
       "rows": [
        [
         "25",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf",
         "",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201903120.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001412.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10.1002_esp.5058</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td></td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10.1002_chem.201903120</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10.1002_chem.202001412</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                article_id                                      pdf_file_path  \\\n",
       "25        10.1002_esp.5058  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "5   10.1002_chem.201903120  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "7   10.1002_chem.202001412  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \\\n",
       "25                                                            test   \n",
       "5   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "7   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "\n",
       "                                         dataset_info  \\\n",
       "25  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "5      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "7      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                       xml_first_line xml_type  \n",
       "25                                               None     None  \n",
       "5   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "7   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load file paths for training and testing datasets\n",
    "train_file_paths_df = load_file_paths(ARTICLE_TRAIN_DIR)\n",
    "test_file_paths_df = load_file_paths(ARTICLE_TEST_DIR)\n",
    "train_file_paths_df['xml_file_path'] = train_file_paths_df['xml_file_path'].fillna('')\n",
    "test_file_paths_df['xml_file_path'] = test_file_paths_df['xml_file_path'].fillna('')\n",
    "\n",
    "# Remove rows in train_file_paths_df that have a corresponding article_id in test_file_paths_df\n",
    "# train_file_paths_df = train_file_paths_df[~train_file_paths_df['article_id'].isin(test_file_paths_df['article_id'])]\n",
    "\n",
    "# Merge the file paths with the grouped_training_data\n",
    "train_file_paths_df['dataset_info'] = train_file_paths_df['article_id'].map(grouped_training_data)\n",
    "test_file_paths_df['dataset_info'] = test_file_paths_df['article_id'].map(grouped_training_data)\n",
    "\n",
    "# Get first line of XML files to determine the type\n",
    "train_file_paths_df['xml_first_line'] = train_file_paths_df['xml_file_path'].apply(read_first_line_of_xml)\n",
    "test_file_paths_df['xml_first_line'] = test_file_paths_df['xml_file_path'].apply(read_first_line_of_xml)\n",
    "\n",
    "# Identify the xml_type based on the first line of the XML file\n",
    "train_file_paths_df['xml_type'] = train_file_paths_df['xml_first_line'].apply(lambda x: identify_xml_type(x) if x else None)\n",
    "test_file_paths_df['xml_type'] = test_file_paths_df['xml_first_line'].apply(lambda x: identify_xml_type(x) if x else None)\n",
    "\n",
    "print(f\"Train files paths shape: {train_file_paths_df.shape}\")\n",
    "display(train_file_paths_df.sample(3))\n",
    "print(f\"Test files paths shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.sample(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "24f16bad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt',\n",
       "       '<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ',\n",
       "       '<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509',\n",
       "       None,\n",
       "       '<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com',\n",
       "       '<!DOCTYPE article PUBLIC \"-//NLM//DTD Journal Archiving and Interchange DTD v3.0 20080202/',\n",
       "       '<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250507',\n",
       "       '<!DOCTYPE article PUBLIC \"-//TaxonX//DTD Taxonomic Treatment Publishing DTD v0 20100105//E'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 278,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_file_paths_df['xml_first_line'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "fe55c451",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "ref": "d35d2bb7-4010-4c75-b5b5-01e75ab1892c",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202007717.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201902131.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201902131.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201903120.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001412.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "8",
         "10.1002_chem.202001668",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001668.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001668.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "9",
         "10.1002_chem.202003167",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202003167.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "10",
         "10.1002_cssc.202201821",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_cssc.202201821.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_cssc.202201821.xml",
         "train",
         null,
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "11",
         "10.1002_ece3.3985",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.3985.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "12",
         "10.1002_ece3.4466",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.4466.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.r6nq870', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "13",
         "10.1002_ece3.5260",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5260.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5260.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "14",
         "10.1002_ece3.5395",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5395.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5395.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6144.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "16",
         "10.1002_ece3.6303",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6303.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.37pvmcvgb', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "17",
         "10.1002_ece3.6784",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6784.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "18",
         "10.1002_ece3.961",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.961.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "19",
         "10.1002_ece3.9627",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.9627.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "20",
         "10.1002_ecs2.1280",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.p3fg9', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "21",
         "10.1002_ecs2.4619",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.4619.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ecs2.4619.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.25349/d9qw5x', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "22",
         "10.1002_ejic.201900904",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejic.201900904.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejic.201900904.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "24",
         "10.1002_ejoc.202000916",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejoc.202000916.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "25",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "26",
         "10.1002_esp.5090",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_esp.5090.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9353101', 'type': 'Secondary'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley"
        ],
        [
         "27",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_mp.14424.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "28",
         "10.1002_nafm.10870",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_nafm.10870.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/p9gtumay', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "29",
         "10.1007_jhep07(2018)134",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep07(2018)134.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "30",
         "10.1007_jhep11(2018)113",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "31",
         "10.1007_jhep11(2018)115",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "32",
         "10.1007_jhep12(2018)117",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "33",
         "10.1007_s00259-022-06053-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00259-022-06053-8.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00259-022-06053-8.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2017.7hs46erv', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2019.30ilqfcl', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "34",
         "10.1007_s00382-012-1636-1",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "35",
         "10.1007_s00382-022-06361-7",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-022-06361-7.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00382-022-06361-7.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6075/j0154fjj', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.6075/j089161b', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "36",
         "10.1007_s00442-022-05201-z",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00442-022-05201-z.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00442-022-05201-z.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.wpzgmsbps', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "37",
         "10.1007_s10904-014-0054-8",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf",
         "",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         null
        ],
        [
         "38",
         "10.1007_s12263-014-0408-4",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12263-014-0408-4.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "39",
         "10.1007_s12559-020-09751-3",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12559-020-09751-3.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12559-020-09751-3.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.u1x8a5nr', 'type': 'Secondary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "40",
         "10.1016_j.ast.2022.107401",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ast.2022.107401.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17862/cranfield.rd.19146182.v1', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "41",
         "10.1016_j.cpc.2024.109087",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.cpc.2024.109087.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.cpc.2024.109087.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/9gr6pxhfjm', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "42",
         "10.1016_j.dib.2022.108797",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2022.108797.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2022.108797.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.1594/pangaea.949117', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "43",
         "10.1016_j.dib.2023.109949",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2023.109949.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8338991', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ],
        [
         "44",
         "10.1016_j.ecolind.2021.107934",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ecolind.2021.107934.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ecolind.2021.107934.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.6073/pasta/be42bb841e696b7bcad9957aed33db5e', 'type': 'Secondary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "45",
         "10.1016_j.fuel.2022.125768",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.fuel.2022.125768.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.11583/dtu.20555586', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.11583/dtu.20555586.v3', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "46",
         "10.1016_j.jlp.2022.104761",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.23642/usn.15134442', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.23642/usn.15134442.v1', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "47",
         "10.1016_j.jobe.2023.107105",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.jobe.2023.107105.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5281/zenodo.8014149', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.5281/zenodo.8014150', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei"
        ],
        [
         "48",
         "10.1016_j.jsames.2022.103824",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jsames.2022.103824.pdf",
         "",
         "train",
         "[{'dataset_id': 'https://doi.org/10.5066/f7k072r7', 'type': 'Primary'}]",
         null,
         null
        ],
        [
         "49",
         "10.1016_j.molcel.2018.11.006",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.molcel.2018.11.006.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.molcel.2018.11.006.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17632/jb4jjxsbb7.1', 'type': 'Primary'}, {'dataset_id': 'https://doi.org/10.17632/xtb4mkvf8f.1', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats"
        ]
       ],
       "shape": {
        "columns": 7,
        "rows": 524
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>10.7554_elife.74937</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5281/zenod...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>10.7717_peerj.10452</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'PRJNA664798', 'type': 'Second...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>10.7717_peerj.11352</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7291/d11m3...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>10.7717_peerj.12422</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.15468/dl.c...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>523</th>\n",
       "      <td>10.7717_peerj.13193</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.6073/pasta...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>524 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "0      10.1002_2017jc013030   \n",
       "1    10.1002_anie.201916483   \n",
       "2    10.1002_anie.202005531   \n",
       "3    10.1002_anie.202007717   \n",
       "4    10.1002_chem.201902131   \n",
       "..                      ...   \n",
       "519     10.7554_elife.74937   \n",
       "520     10.7717_peerj.10452   \n",
       "521     10.7717_peerj.11352   \n",
       "522     10.7717_peerj.12422   \n",
       "523     10.7717_peerj.13193   \n",
       "\n",
       "                                         pdf_file_path  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...   \n",
       "..                                                 ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                         xml_file_path dataset_type  \\\n",
       "0    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "3    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "4    ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "..                                                 ...          ...   \n",
       "519  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "520  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "521  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "522  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "523  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                          dataset_info  \\\n",
       "0    [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "3       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "4       [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "..                                                 ...   \n",
       "519  [{'dataset_id': 'https://doi.org/10.5281/zenod...   \n",
       "520  [{'dataset_id': 'PRJNA664798', 'type': 'Second...   \n",
       "521  [{'dataset_id': 'https://doi.org/10.7291/d11m3...   \n",
       "522  [{'dataset_id': 'https://doi.org/10.15468/dl.c...   \n",
       "523  [{'dataset_id': 'https://doi.org/10.6073/pasta...   \n",
       "\n",
       "                                        xml_first_line xml_type  \n",
       "0    <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei  \n",
       "1    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "2    <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc  \n",
       "3    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "4    <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "..                                                 ...      ...  \n",
       "519  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "520  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "521  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "522  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "523  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats  \n",
       "\n",
       "[524 rows x 7 columns]"
      ]
     },
     "execution_count": 279,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_file_paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "e30a9aff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf. Length: 4956 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf. Length: 1355 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf. Length: 1736 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202007717.pdf. Length: 1734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201902131.pdf. Length: 671 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.201903120.pdf. Length: 941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202000235.pdf. Length: 734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001412.pdf. Length: 1057 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202001668.pdf. Length: 506 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_chem.202003167.pdf. Length: 955 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_cssc.202201821.pdf. Length: 3920 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.3985.pdf. Length: 3707 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.4466.pdf. Length: 2364 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5260.pdf. Length: 2692 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.5395.pdf. Length: 4026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6144.pdf. Length: 2964 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6303.pdf. Length: 3312 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.6784.pdf. Length: 6896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.961.pdf. Length: 2882 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ece3.9627.pdf. Length: 2825 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.1280.pdf. Length: 3339 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ecs2.4619.pdf. Length: 3555 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejic.201900904.pdf. Length: 606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000139.pdf. Length: 970 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_ejoc.202000916.pdf. Length: 284 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5058.pdf. Length: 4494 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_esp.5090.pdf. Length: 5378 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_mp.14424.pdf. Length: 4658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_nafm.10870.pdf. Length: 4574 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep07(2018)134.pdf. Length: 879 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)113.pdf. Length: 1399 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep11(2018)115.pdf. Length: 1259 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_jhep12(2018)117.pdf. Length: 802 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00259-022-06053-8.pdf. Length: 8627 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-012-1636-1.pdf. Length: 5569 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00382-022-06361-7.pdf. Length: 3928 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s00442-022-05201-z.pdf. Length: 1710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s10904-014-0054-8.pdf. Length: 917 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12263-014-0408-4.pdf. Length: 4218 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1007_s12559-020-09751-3.pdf. Length: 2733 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ast.2022.107401.pdf. Length: 2606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.cpc.2024.109087.pdf. Length: 1907 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2022.108797.pdf. Length: 2209 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.dib.2023.109949.pdf. Length: 2506 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.ecolind.2021.107934.pdf. Length: 3386 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.fuel.2022.125768.pdf. Length: 1241 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jlp.2022.104761.pdf. Length: 2783 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jobe.2023.107105.pdf. Length: 3667 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.jsames.2022.103824.pdf. Length: 3635 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.molcel.2018.11.006.pdf. Length: 4505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.scitotenv.2024.171850.pdf. Length: 4588 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1016_j.websem.2024.100815.pdf. Length: 748 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_rdc.2022.19.pdf. Length: 2041 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1017_s0007123423000601.pdf. Length: 1321 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acs.jcim.9b01185.pdf. Length: 1165 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acscentsci.1c01109.pdf. Length: 1114 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_acsomega.3c06074.pdf. Length: 1730 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1021_jacs.2c06519.pdf. Length: 1991 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2018gl078007.pdf. Length: 4366 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2019jg005297.pdf. Length: 4623 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2019pa003774.pdf. Length: 2559 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2020jf005675.pdf. Length: 2655 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021ea001703.pdf. Length: 2702 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021gl096173.pdf. Length: 821 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2021pa004379.pdf. Length: 3519 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2022gl100473.pdf. Length: 1694 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2023ea002840.pdf. Length: 2557 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1029_2023wr035126.pdf. Length: 5995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2013.74.pdf. Length: 3107 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2014.75.pdf. Length: 3228 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_hdy.2015.99.pdf. Length: 3317 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_ncomms11871.pdf. Length: 1041 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41396-020-00885-8.pdf. Length: 2458 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41437-020-0318-8.pdf. Length: 2108 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-04041-x.pdf. Length: 2118 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-018-07681-1.pdf. Length: 1570 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41467-019-10357-z.pdf. Length: 2333 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41558-022-01301-z.pdf. Length: 754 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41562-021-01247-w.pdf. Length: 4021 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-019-0101-y.pdf. Length: 2765 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-022-01555-4.pdf. Length: 5480 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41597-023-02280-2.pdf. Length: 3477 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-017-15852-1.pdf. Length: 2516 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-020-59839-x.pdf. Length: 4336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-021-85671-y.pdf. Length: 6299 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41598-024-56373-y.pdf. Length: 2440 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_s41893-020-0559-9.pdf. Length: 6524 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1038_sdata.2017.167.pdf. Length: 1418 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c5sc01309g.pdf. Length: 2772 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9ce01020c.pdf. Length: 2620 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9ra06638a.pdf. Length: 2851 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_c9sc02930c.pdf. Length: 2898 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0dt02130j.pdf. Length: 2716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0gc00363h.pdf. Length: 2843 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01197e.pdf. Length: 2793 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc01518k.pdf. Length: 3097 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc02283g.pdf. Length: 2407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc02676j.pdf. Length: 2444 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d0sc03604h.pdf. Length: 2939 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d1ee03696c.pdf. Length: 3208 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1039_d2cc00847e.pdf. Length: 3314 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1051_e3sconf_202017220004.pdf. Length: 2428 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1055_s-0039-1693681.pdf. Length: 1753 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1323607111.pdf. Length: 4085 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1705601114.pdf. Length: 4305 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1711872115.pdf. Length: 4031 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.1915395117.pdf. Length: 398 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1073_pnas.2306723120.pdf. Length: 4251 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_01621459.2019.1573732.pdf. Length: 636 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_02713683.2019.1607392.pdf. Length: 741 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2019.1669817.pdf. Length: 685 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_0284186x.2020.1714721.pdf. Length: 2014 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_07350015.2020.1766469.pdf. Length: 601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_13501763.2019.1630470.pdf. Length: 710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_14756366.2020.1740692.pdf. Length: 902 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_15476286.2016.1232238.pdf. Length: 3137 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_19475683.2023.2226196.pdf. Length: 5436 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_21645515.2023.2189598.pdf. Length: 293 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1080_22221751.2020.1738277.pdf. Length: 2476 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arad016.pdf. Length: 3026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arw167.pdf. Length: 3254 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_beheco_arz101.pdf. Length: 3529 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_bib_bbab395.pdf. Length: 2916 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_evolut_qpad206.pdf. Length: 2166 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_nar_gkp1049.pdf. Length: 2752 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1093_sysbio_syy011.pdf. Length: 2392 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsbl.2015.0113.pdf. Length: 3155 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.160417.pdf. Length: 4813 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.170796.pdf. Length: 5013 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.190841.pdf. Length: 3891 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rsos.210982.pdf. Length: 3504 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.0371.pdf. Length: 3934 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.1498.pdf. Length: 4113 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.2726.pdf. Length: 3154 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1098_rspb.2015.2764.pdf. Length: 3763 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_096297.pdf. Length: 1414 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_2022.02.10.480011.pdf. Length: 1722 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1101_2022.07.21.501061.pdf. Length: 993 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1103_physrevresearch.4.023008.pdf. Length: 3088 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s0108270106019688.pdf. Length: 9573 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536807066780.pdf. Length: 2527 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536808011148.pdf. Length: 1813 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809014883.pdf. Length: 2277 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809022582.pdf. Length: 2132 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536809043700.pdf. Length: 2222 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810036299.pdf. Length: 2911 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536810047185.pdf. Length: 2072 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536811003163.pdf. Length: 2190 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s160053681102201x.pdf. Length: 2213 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s160053681103220x.pdf. Length: 1925 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536811042164.pdf. Length: 2862 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812014614.pdf. Length: 2413 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812019885.pdf. Length: 2602 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812024269.pdf. Length: 2277 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812027390.pdf. Length: 2231 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s1600536812046892.pdf. Length: 2693 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252514012081.pdf. Length: 2367 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515010271.pdf. Length: 2014 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515011665.pdf. Length: 2002 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252515023945.pdf. Length: 1873 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2052252520013603.pdf. Length: 2170 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2053229620003757.pdf. Length: 2192 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989014026358.pdf. Length: 1777 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989015010257.pdf. Length: 1974 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989015019891.pdf. Length: 2232 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989016016819.pdf. Length: 2096 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2056989020010658.pdf. Length: 1814 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2059798322005691.pdf. Length: 5072 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616000523.pdf. Length: 1565 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616007033.pdf. Length: 1652 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314616007227.pdf. Length: 1547 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1107_s2414314619012446.pdf. Length: 1514 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1109_access.2024.3385658.pdf. Length: 2415 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13087.pdf. Length: 3312 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13431.pdf. Length: 4190 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2435.13569.pdf. Length: 3967 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12382.pdf. Length: 3523 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12491.pdf. Length: 3206 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12501.pdf. Length: 3290 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2656.12594.pdf. Length: 2881 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13128.pdf. Length: 3759 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13136.pdf. Length: 3315 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13168.pdf. Length: 3045 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2664.13446.pdf. Length: 2898 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_1365-2745.13449.pdf. Length: 3299 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.12453.pdf. Length: 3451 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_2041-210x.13817.pdf. Length: 3030 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_acel.13089.pdf. Length: 4464 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_bph.14747.pdf. Length: 8358 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_cas.12935.pdf. Length: 3576 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ddi.13153.pdf. Length: 5357 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ddi.13161.pdf. Length: 2481 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ejn.13855.pdf. Length: 3354 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12151.pdf. Length: 3243 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12446.pdf. Length: 2380 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_eva.12768.pdf. Length: 2578 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_evo.13972.pdf. Length: 1410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_gcb.13914.pdf. Length: 2807 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_gcb.14974.pdf. Length: 2760 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_ggr.12517.pdf. Length: 3770 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_mec.16743.pdf. Length: 4981 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_mec.16977.pdf. Length: 2734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1111_njb.02077.pdf. Length: 3449 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1128_JVI.01717-21.pdf. Length: 4023 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1128_spectrum.00422-24.pdf. Length: 3061 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1130_ges01387.1.pdf. Length: 7325 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1136_jitc-2021-003114.pdf. Length: 2498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-016-4366-4.pdf. Length: 1634 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-018-6033-4.pdf. Length: 1263 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-018-6468-7.pdf. Length: 1456 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjc_s10052-019-6583-0.pdf. Length: 3668 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1140_epjds_s13688-018-0132-1.pdf. Length: 1857 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1145_3461702.3462538.pdf. Length: 13842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1155_2013_536562.pdf. Length: 2779 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1177_0018720820970751.pdf. Length: 2406 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_1471-244x-13-259.pdf. Length: 1923 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_1752-153x-7-123.pdf. Length: 1688 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_gb-2010-11-10-r101.pdf. Length: 2515 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s11658-018-0120-2.pdf. Length: 5501 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s11689-019-9287-8.pdf. Length: 2383 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0904-1.pdf. Length: 3511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0922-z.pdf. Length: 6607 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-0938-4.pdf. Length: 11774 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-016-1206-3.pdf. Length: 2264 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-017-1563-6.pdf. Length: 2526 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-017-1952-x.pdf. Length: 4252 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2036-2.pdf. Length: 1788 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2146-x.pdf. Length: 2385 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2263-6.pdf. Length: 5729 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-018-2414-9.pdf. Length: 2113 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-020-3415-z.pdf. Length: 2378 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12859-020-3496-8.pdf. Length: 1995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12860-020-00261-6.pdf. Length: 3350 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-015-0558-z.pdf. Length: 5611 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-018-1213-2.pdf. Length: 1990 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-019-1388-1.pdf. Length: 2503 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12862-019-1509-x.pdf. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12863-017-0513-7.pdf. Length: 2338 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12863-019-0790-4.pdf. Length: 2463 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-015-2206-9.pdf. Length: 2382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-4947-8.pdf. Length: 1836 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-5054-6.pdf. Length: 2159 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-018-5198-4.pdf. Length: 3726 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-5872-1.pdf. Length: 2890 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6099-x.pdf. Length: 2516 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6131-1.pdf. Length: 16405 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-019-6324-7.pdf. Length: 2500 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12864-020-6657-2.pdf. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-015-0509-2.pdf. Length: 2437 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-019-1542-3.pdf. Length: 2482 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12866-020-01863-y.pdf. Length: 6203 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12868-018-0468-2.pdf. Length: 2875 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12868-019-0514-8.pdf. Length: 2601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-018-1332-3.pdf. Length: 2060 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-018-1542-8.pdf. Length: 4521 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1632-2.pdf. Length: 3490 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1889-5.pdf. Length: 2422 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1904-x.pdf. Length: 2109 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-1926-4.pdf. Length: 2658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-019-2199-7.pdf. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-02651-6.pdf. Length: 2276 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-02692-x.pdf. Length: 2027 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12870-020-2295-8.pdf. Length: 4306 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12874-018-0583-x.pdf. Length: 2475 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12874-019-0829-2.pdf. Length: 2407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12879-018-3065-1.pdf. Length: 2096 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12879-019-3766-0.pdf. Length: 2109 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12881-019-0773-3.pdf. Length: 2115 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12882-019-1494-8.pdf. Length: 1896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12884-018-1751-z.pdf. Length: 2096 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-017-3237-1.pdf. Length: 2308 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4229-5.pdf. Length: 2554 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4314-9.pdf. Length: 3258 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-018-4768-9.pdf. Length: 2385 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-019-5681-6.pdf. Length: 4322 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12885-020-06724-5.pdf. Length: 3937 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12903-018-0656-6.pdf. Length: 2227 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12906-019-2769-0.pdf. Length: 4746 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12913-018-3333-1.pdf. Length: 2294 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12915-018-0498-3.pdf. Length: 2725 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12915-018-0584-6.pdf. Length: 2210 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12916-019-1469-4.pdf. Length: 2317 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-015-0209-4.pdf. Length: 2000 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-018-0564-z.pdf. Length: 18856 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12918-018-0620-8.pdf. Length: 2716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0379-1.pdf. Length: 2700 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0403-5.pdf. Length: 2739 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-018-0426-y.pdf. Length: 3854 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0611-7.pdf. Length: 3223 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0628-y.pdf. Length: 6176 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-019-0646-9.pdf. Length: 3672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12920-020-00737-6.pdf. Length: 3573 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12931-019-1001-6.pdf. Length: 2011 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12933-019-0813-5.pdf. Length: 2843 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12934-018-1005-9.pdf. Length: 3165 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12935-020-01373-x.pdf. Length: 1928 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12943-018-0878-x.pdf. Length: 2219 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12943-019-1017-z.pdf. Length: 5582 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12967-019-2100-3.pdf. Length: 2527 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12967-020-02311-1.pdf. Length: 9555 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12974-020-01860-y.pdf. Length: 2431 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s12977-015-0204-2.pdf. Length: 2495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13007-019-0403-2.pdf. Length: 1899 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13014-018-1166-z.pdf. Length: 1842 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13020-020-00307-z.pdf. Length: 2693 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0254-8.pdf. Length: 4444 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13024-018-0266-4.pdf. Length: 4287 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13039-022-00631-z.pdf. Length: 4874 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13046-018-0843-y.pdf. Length: 2606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-015-0738-7.pdf. Length: 1925 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-015-1052-0.pdf. Length: 3323 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13054-019-2488-4.pdf. Length: 2170 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13058-015-0618-8.pdf. Length: 2579 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-017-1200-8.pdf. Length: 2558 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-018-1557-3.pdf. Length: 2954 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1749-5.pdf. Length: 2640 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1840-y.pdf. Length: 1738 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1908-8.pdf. Length: 2511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1918-6.pdf. Length: 2610 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-019-1924-8.pdf. Length: 13025 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-020-02048-6.pdf. Length: 3432 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13059-020-1949-z.pdf. Length: 7657 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-017-0901-2.pdf. Length: 3319 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1078-z.pdf. Length: 2466 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1167-z.pdf. Length: 2242 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13068-018-1316-4.pdf. Length: 2761 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-016-1359-y.pdf. Length: 2346 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-2842-4.pdf. Length: 1938 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-2964-8.pdf. Length: 1969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-3084-1.pdf. Length: 2345 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13071-018-3237-2.pdf. Length: 2354 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13072-019-0285-6.pdf. Length: 2096 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13072-019-0322-5.pdf. Length: 1534 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0674-2.pdf. Length: 2952 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-019-0709-8.pdf. Length: 2796 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13073-020-00727-4.pdf. Length: 2998 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13100-019-0153-8.pdf. Length: 3869 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-018-4014-1.pdf. Length: 1472 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-019-4058-x.pdf. Length: 1562 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13104-019-4127-1.pdf. Length: 1377 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13148-018-0581-1.pdf. Length: 2342 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13148-019-0719-9.pdf. Length: 1961 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-015-0110-6.pdf. Length: 1816 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-017-0223-1.pdf. Length: 9931 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-017-0246-7.pdf. Length: 1746 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13321-018-0298-3.pdf. Length: 5183 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13326-019-0195-3.pdf. Length: 2102 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13568-018-0680-6.pdf. Length: 2463 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13617-019-0084-6.pdf. Length: 2245 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13643-018-0859-6.pdf. Length: 2827 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s13750-020-0184-0.pdf. Length: 3602 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-017-0338-7.pdf. Length: 3762 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-018-0532-2.pdf. Length: 2401 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40168-018-0550-0.pdf. Length: 2911 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40169-019-0253-6.pdf. Length: 2930 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40170-018-0185-4.pdf. Length: 2335 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40170-020-00212-x.pdf. Length: 2526 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40425-019-0730-x.pdf. Length: 2366 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40478-018-0561-x.pdf. Length: 8016 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40478-018-0655-5.pdf. Length: 2658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40657-018-0125-6.pdf. Length: 2312 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40657-020-00194-w.pdf. Length: 3783 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40793-015-0095-9.pdf. Length: 2318 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1186_s40851-018-0089-8.pdf. Length: 2606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1212_wnl.0000000000006035.pdf. Length: 3101 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_bio.035071.pdf. Length: 3449 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_dev.138545.pdf. Length: 4239 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1242_jeb.211466.pdf. Length: 6551 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.10556.1.pdf. Length: 3505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.11698.1.pdf. Length: 5432 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13064.1.pdf. Length: 3440 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13483.1.pdf. Length: 3298 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13622.1.pdf. Length: 4341 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.13852.1.pdf. Length: 2840 characters\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "MuPDF error: unsupported error: cannot create appearance stream for  annotations\n",
      "\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.12688_f1000research.4660.1.pdf. Length: 3146 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1364_oe.25.001985.pdf. Length: 1098 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pcbi.1011828.pdf. Length: 9585 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pntd.0005385.pdf. Length: 3032 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0070749.pdf. Length: 3610 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0093021.pdf. Length: 3444 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0096457.pdf. Length: 3385 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0117675.pdf. Length: 2257 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0137181.pdf. Length: 2380 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0139215.pdf. Length: 2846 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0146274.pdf. Length: 6655 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0159387.pdf. Length: 2735 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0170126.pdf. Length: 2065 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0188323.pdf. Length: 2279 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0191086.pdf. Length: 2732 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0198382.pdf. Length: 2319 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0200041.pdf. Length: 3217 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0212669.pdf. Length: 4308 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0220399.pdf. Length: 2807 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0253228.pdf. Length: 2648 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0262974.pdf. Length: 2641 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1371_journal.pone.0284951.pdf. Length: 2388 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.14358_pers.22-00039r2.pdf. Length: 1931 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.14379_iodp.proc.390393.208.2024.pdf. Length: 631 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_g3.119.400401.pdf. Length: 5542 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_g3.119.400993.pdf. Length: 8766 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1534_genetics.120.303190.pdf. Length: 2995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0001-3765201920180768.pdf. Length: 996 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0001-3765201920190822.pdf. Length: 706 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0034-7167-2018-0065.pdf. Length: 1172 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0034-737x201966050002.pdf. Length: 2570 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0047-2085000000239.pdf. Length: 2066 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102-778632120150164.pdf. Length: 2498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102-77863220007.pdf. Length: 2253 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0102.3772e35417.pdf. Length: 1939 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0103-6351_4070.pdf. Length: 1283 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0103-6440201802861.pdf. Length: 2776 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-4060.59642.pdf. Length: 1569 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-530x4900-19.pdf. Length: 1030 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_0104-6632.20190362s20180340.pdf. Length: 1654 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1413-785220192703169516.pdf. Length: 2868 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1414-431x20198292.pdf. Length: 2708 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1516-3180.2018.0270071218.pdf. Length: 1879 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1517-869220192505126869.pdf. Length: 3431 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1519-6984.192126.pdf. Length: 2634 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4162-10032.pdf. Length: 1418 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4162-10752.pdf. Length: 2311 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4499.20190067.pdf. Length: 1292 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1678-4685-gmb-2018-0055.pdf. Length: 2469 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1679-78255326.pdf. Length: 2155 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1679-78255753.pdf. Length: 1852 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-90882019000500004.pdf. Length: 1694 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-93042019000400002.pdf. Length: 1819 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1806-9479.2019.185555.pdf. Length: 586 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1807-3107bor-2019.vol33.0095.pdf. Length: 1676 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-2950_19003927012020.pdf. Length: 2590 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-2950_19008627012020.pdf. Length: 2649 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-4430-eng.agric.v40n1p24-33_2020.pdf. Length: 2442 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-6891v21e-43578.pdf. Length: 1389 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1809-9823.2007.10034.pdf. Length: 1634 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0766.pdf. Length: 2763 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2018-0921.pdf. Length: 2691 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5373-mr-2019-0338.pdf. Length: 2860 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.032.ao27.pdf. Length: 1311 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1980-5918.033.ao15.pdf. Length: 1224 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-22562019022.190037.pdf. Length: 1594 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-52712015v43n3rb20180090.pdf. Length: 2053 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-6723.06017.pdf. Length: 1873 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1981-86372019000243506.pdf. Length: 2757 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1982-0216_20192158519.pdf. Length: 1611 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_1983-21252019v32n324rc.pdf. Length: 1536 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_2236-3459_83528.pdf. Length: 969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_2237-6089-2017-0134.pdf. Length: 3068 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_fst.33717.pdf. Length: 1623 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_fst.36617.pdf. Length: 2236 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1677-5538.ibju.2019.0167.pdf. Length: 1851 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1590_s1678-86212019000400340.pdf. Length: 2704 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1617_s11527-023-02260-3.pdf. Length: 984 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.17581_bp.2020.09104.pdf. Length: 370 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.17815_jlsrf-2-58.pdf. Length: 1805 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.18438_eblip29674.pdf. Length: 2896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.20944_preprints202009.0353.v1.pdf. Length: 2078 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21105_joss.04237.pdf. Length: 2453 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21203_rs.3.rs-2814013_v1.pdf. Length: 1997 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.21203_rs.3.rs-3338732_v1.pdf. Length: 4039 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_cir1497.pdf. Length: 4454 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_fs20233046.pdf. Length: 2243 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20201035.pdf. Length: 1993 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20231026.pdf. Length: 1258 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3133_ofr20231027.pdf. Length: 8319 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fcell.2020.618552.pdf. Length: 14192 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fchem.2019.00828.pdf. Length: 2484 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fcimb.2024.1292467.pdf. Length: 2803 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_feart.2023.1205211.pdf. Length: 4837 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fevo.2023.1112519.pdf. Length: 2895 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fmicb.2024.1456637.pdf. Length: 2728 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3389_fpls.2023.1160645.pdf. Length: 2954 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_cryst2031058.pdf. Length: 1208 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_cryst9060301.pdf. Length: 1463 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_d13010019.pdf. Length: 2367 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_ijms12117360.pdf. Length: 1791 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_inorganics7050058.pdf. Length: 988 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_microorganisms8121872.pdf. Length: 1298 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules17077645.pdf. Length: 1573 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules191017026.pdf. Length: 1431 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules200917469.pdf. Length: 1164 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules23123199.pdf. Length: 1446 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_molecules25010053.pdf. Length: 1474 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_rs12121957.pdf. Length: 1848 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_rs4102923.pdf. Length: 6726 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s19030479.pdf. Length: 2121 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s23094491.pdf. Length: 3366 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_s23177333.pdf. Length: 2554 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3390_v11060565.pdf. Length: 1278 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3762_bjoc.8.42.pdf. Length: 771 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3762_bjoc.9.92.pdf. Length: 992 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3847_1538-4357_aae92c.pdf. Length: 1601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_bdj.7.e47369.pdf. Length: 3967 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_jhr.96.111550.pdf. Length: 3948 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_neobiota.82.87455.pdf. Length: 4167 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zookeys.500.9360.pdf. Length: 2290 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.35.e23481.pdf. Length: 2027 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.3897_zoologia.36.e32053.pdf. Length: 833 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.4025_actasciagron.v40i1.39392.pdf. Length: 2865 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-2021-570.pdf. Length: 5055 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-22-2769-2022.pdf. Length: 2972 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_acp-22-5701-2022.pdf. Length: 6388 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_amt-15-3969-2022.pdf. Length: 2859 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-12-1287-2020.pdf. Length: 4499 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2019-206.pdf. Length: 2173 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-187.pdf. Length: 6958 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-2023-198.pdf. Length: 2304 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_essd-8-663-2016.pdf. Length: 1998 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_gmd-12-4221-2019.pdf. Length: 3504 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_gmd-13-335-2020.pdf. Length: 3426 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_se-13-1541-2022.pdf. Length: 4020 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5194_tc-17-3617-2023.pdf. Length: 12649 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.5937_bnhmb1811227u.pdf. Length: 2047 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.63194.pdf. Length: 2773 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_eLife.72626.pdf. Length: 1987 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.29944.pdf. Length: 4743 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.42722.pdf. Length: 1849 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.62329.pdf. Length: 716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.63455.pdf. Length: 1804 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.73695.pdf. Length: 995 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7554_elife.74937.pdf. Length: 796 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.10452.pdf. Length: 10807 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.11352.pdf. Length: 2187 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.12422.pdf. Length: 1752 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.7717_peerj.13193.pdf. Length: 3022 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml. Length: 1633 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml. Length: 1124 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml. Length: 234 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202007717.xml. Length: 1444 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201902131.xml. Length: 1262 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.201903120.xml. Length: 2036 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202000235.xml. Length: 867 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001412.xml. Length: 1382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202001668.xml. Length: 992 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_chem.202003167.xml. Length: 1983 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_cssc.202201821.xml. Length: 1690 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.3985.xml. Length: 1716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.4466.xml. Length: 1794 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5260.xml. Length: 2560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.5395.xml. Length: 4026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6144.xml. Length: 2188 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6303.xml. Length: 2672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.6784.xml. Length: 2900 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.961.xml. Length: 1771 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ece3.9627.xml. Length: 2505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ecs2.4619.xml. Length: 2080 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejic.201900904.xml. Length: 1014 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_ejoc.202000916.xml. Length: 669 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_esp.5090.xml. Length: 2714 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_mp.14424.xml. Length: 3030 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00259-022-06053-8.xml. Length: 3756 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00382-022-06361-7.xml. Length: 2142 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s00442-022-05201-z.xml. Length: 1945 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12263-014-0408-4.xml. Length: 2466 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1007_s12559-020-09751-3.xml. Length: 3045 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ast.2022.107401.xml. Length: 891 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.cpc.2024.109087.xml. Length: 1860 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2022.108797.xml. Length: 1751 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.dib.2023.109949.xml. Length: 2019 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.ecolind.2021.107934.xml. Length: 2070 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.jobe.2023.107105.xml. Length: 2185 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.molcel.2018.11.006.xml. Length: 3145 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1016_j.scitotenv.2024.171850.xml. Length: 1489 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1017_s0007123423000601.xml. Length: 1242 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_acscentsci.1c01109.xml. Length: 1806 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_acsomega.3c06074.xml. Length: 2991 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1021_jacs.2c06519.xml. Length: 1680 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2018gl078007.xml. Length: 2470 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2020jf005675.xml. Length: 2111 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2021gl096173.xml. Length: 461 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2022gl100473.xml. Length: 896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2023ea002840.xml. Length: 2231 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1029_2023wr035126.xml. Length: 672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_hdy.2015.99.xml. Length: 1753 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_ncomms11871.xml. Length: 1081 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41396-020-00885-8.xml. Length: 2450 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41437-020-0318-8.xml. Length: 2180 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-018-04041-x.xml. Length: 2254 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-018-07681-1.xml. Length: 1817 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41467-019-10357-z.xml. Length: 2307 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41558-022-01301-z.xml. Length: 1505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41562-021-01247-w.xml. Length: 1757 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-019-0101-y.xml. Length: 2022 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-022-01555-4.xml. Length: 3788 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41597-023-02280-2.xml. Length: 2020 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-017-15852-1.xml. Length: 1536 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-020-59839-x.xml. Length: 2230 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-021-85671-y.xml. Length: 2877 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_s41598-024-56373-y.xml. Length: 2370 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1038_sdata.2017.167.xml. Length: 1417 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c5sc01309g.xml. Length: 683 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c9ra06638a.xml. Length: 663 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_c9sc02930c.xml. Length: 1022 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc01197e.xml. Length: 2261 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc01518k.xml. Length: 1283 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc02283g.xml. Length: 1888 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc02676j.xml. Length: 1497 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d0sc03604h.xml. Length: 1271 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d1ee03696c.xml. Length: 2400 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1039_d2cc00847e.xml. Length: 942 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1055_s-0039-1693681.xml. Length: 1923 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1073_pnas.2306723120.xml. Length: 1389 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_14756366.2020.1740692.xml. Length: 1743 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_15476286.2016.1232238.xml. Length: 2172 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_21645515.2023.2189598.xml. Length: 2742 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1080_22221751.2020.1738277.xml. Length: 2244 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arad016.xml. Length: 1922 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arw167.xml. Length: 1027 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_beheco_arz101.xml. Length: 2221 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_bib_bbab395.xml. Length: 1239 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_evolut_qpad206.xml. Length: 2600 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_nar_gkp1049.xml. Length: 1511 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1093_sysbio_syy011.xml. Length: 2157 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsbl.2015.0113.xml. Length: 994 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.160417.xml. Length: 2588 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.170796.xml. Length: 2279 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.190841.xml. Length: 2224 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rsos.210982.xml. Length: 1710 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.0371.xml. Length: 2097 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.1498.xml. Length: 1530 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1098_rspb.2015.2726.xml. Length: 1601 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1101_2022.02.10.480011.xml. Length: 899 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1101_2022.07.21.501061.xml. Length: 1736 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1103_physrevresearch.4.023008.xml. Length: 1805 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536807066780.xml. Length: 974 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536808011148.xml. Length: 675 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809014883.xml. Length: 790 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809022582.xml. Length: 741 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536809043700.xml. Length: 365 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536810036299.xml. Length: 812 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536810047185.xml. Length: 1092 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536811003163.xml. Length: 747 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s160053681102201x.xml. Length: 571 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s160053681103220x.xml. Length: 911 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536811042164.xml. Length: 595 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812014614.xml. Length: 1382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812019885.xml. Length: 717 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812024269.xml. Length: 702 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812027390.xml. Length: 1130 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s1600536812046892.xml. Length: 852 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252514012081.xml. Length: 595 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515010271.xml. Length: 456 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515011665.xml. Length: 427 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252515023945.xml. Length: 440 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2052252520013603.xml. Length: 399 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2053229620003757.xml. Length: 639 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989014026358.xml. Length: 641 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989015010257.xml. Length: 628 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989015019891.xml. Length: 781 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989016016819.xml. Length: 1031 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2056989020010658.xml. Length: 692 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1107_s2059798322005691.xml. Length: 626 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1109_access.2024.3385658.xml. Length: 2161 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2435.13431.xml. Length: 2603 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2435.13569.xml. Length: 2432 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2656.12382.xml. Length: 1048 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2656.12491.xml. Length: 2196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_1365-2664.13136.xml. Length: 2415 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_2041-210x.12453.xml. Length: 1917 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_2041-210x.13817.xml. Length: 1645 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_acel.13089.xml. Length: 2022 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_bph.14747.xml. Length: 2448 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_cas.12935.xml. Length: 1838 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ddi.13153.xml. Length: 2265 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ddi.13161.xml. Length: 2329 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ejn.13855.xml. Length: 2201 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12151.xml. Length: 1564 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12446.xml. Length: 1917 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_eva.12768.xml. Length: 1984 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_gcb.13914.xml. Length: 2045 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_gcb.14974.xml. Length: 3772 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_ggr.12517.xml. Length: 2130 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_mec.16977.xml. Length: 1824 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1111_njb.02077.xml. Length: 1541 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1128_JVI.01717-21.xml. Length: 2575 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1128_spectrum.00422-24.xml. Length: 3468 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1130_ges01387.1.xml. Length: 2877 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1136_jitc-2021-003114.xml. Length: 2196 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1140_epjc_s10052-016-4366-4.xml. Length: 40549 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1140_epjc_s10052-018-6468-7.xml. Length: 42181 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1145_3461702.3462538.xml. Length: 4495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1155_2013_536562.xml. Length: 1327 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1177_0018720820970751.xml. Length: 2064 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_1471-244x-13-259.xml. Length: 2371 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_1752-153x-7-123.xml. Length: 2317 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_gb-2010-11-10-r101.xml. Length: 2990 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s11658-018-0120-2.xml. Length: 1987 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s11689-019-9287-8.xml. Length: 2924 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0904-1.xml. Length: 1824 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0922-z.xml. Length: 2378 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-0938-4.xml. Length: 3345 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-016-1206-3.xml. Length: 1894 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-017-1563-6.xml. Length: 2439 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-017-1952-x.xml. Length: 2962 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2036-2.xml. Length: 2397 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2146-x.xml. Length: 3076 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2263-6.xml. Length: 2004 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-018-2414-9.xml. Length: 2554 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-020-3415-z.xml. Length: 2869 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12859-020-3496-8.xml. Length: 3403 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12860-020-00261-6.xml. Length: 3053 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-015-0558-z.xml. Length: 2560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-018-1213-2.xml. Length: 2319 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-019-1388-1.xml. Length: 2517 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12862-019-1509-x.xml. Length: 1969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12863-017-0513-7.xml. Length: 1999 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12863-019-0790-4.xml. Length: 2500 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-015-2206-9.xml. Length: 2095 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-4947-8.xml. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-5054-6.xml. Length: 2577 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-018-5198-4.xml. Length: 3022 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-5872-1.xml. Length: 2449 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6099-x.xml. Length: 2748 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6131-1.xml. Length: 2482 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-019-6324-7.xml. Length: 2776 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12864-020-6657-2.xml. Length: 2177 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-015-0509-2.xml. Length: 2277 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-019-1542-3.xml. Length: 2338 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12866-020-01863-y.xml. Length: 2858 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12868-018-0468-2.xml. Length: 2680 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12868-019-0514-8.xml. Length: 2789 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-018-1332-3.xml. Length: 2340 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-018-1542-8.xml. Length: 2903 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1632-2.xml. Length: 3098 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1889-5.xml. Length: 2589 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1904-x.xml. Length: 2577 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-1926-4.xml. Length: 2645 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-019-2199-7.xml. Length: 2253 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-02651-6.xml. Length: 2790 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-02692-x.xml. Length: 3214 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12870-020-2295-8.xml. Length: 2966 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12874-018-0583-x.xml. Length: 2246 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12874-019-0829-2.xml. Length: 2052 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12879-018-3065-1.xml. Length: 2358 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12879-019-3766-0.xml. Length: 2734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12881-019-0773-3.xml. Length: 2656 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12882-019-1494-8.xml. Length: 2542 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12884-018-1751-z.xml. Length: 2512 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-017-3237-1.xml. Length: 2435 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4229-5.xml. Length: 2322 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4314-9.xml. Length: 2353 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-018-4768-9.xml. Length: 2346 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-019-5681-6.xml. Length: 3158 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12885-020-06724-5.xml. Length: 2870 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12903-018-0656-6.xml. Length: 1894 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12906-019-2769-0.xml. Length: 3498 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12913-018-3333-1.xml. Length: 2229 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12915-018-0498-3.xml. Length: 2295 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12915-018-0584-6.xml. Length: 2429 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12916-019-1469-4.xml. Length: 3136 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-015-0209-4.xml. Length: 2475 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-018-0564-z.xml. Length: 2246 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12918-018-0620-8.xml. Length: 3423 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0379-1.xml. Length: 2939 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0403-5.xml. Length: 2448 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-018-0426-y.xml. Length: 2356 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0611-7.xml. Length: 1984 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0628-y.xml. Length: 2503 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-019-0646-9.xml. Length: 2595 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12920-020-00737-6.xml. Length: 2901 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12931-019-1001-6.xml. Length: 2643 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12933-019-0813-5.xml. Length: 2697 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12934-018-1005-9.xml. Length: 3005 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12935-020-01373-x.xml. Length: 1901 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12943-018-0878-x.xml. Length: 1932 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12943-019-1017-z.xml. Length: 2764 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12967-019-2100-3.xml. Length: 2775 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12967-020-02311-1.xml. Length: 2969 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12974-020-01860-y.xml. Length: 2764 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s12977-015-0204-2.xml. Length: 2335 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13007-019-0403-2.xml. Length: 2288 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13014-018-1166-z.xml. Length: 2331 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13020-020-00307-z.xml. Length: 2428 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13024-018-0254-8.xml. Length: 2663 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13024-018-0266-4.xml. Length: 2843 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13039-022-00631-z.xml. Length: 3334 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13046-018-0843-y.xml. Length: 2525 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-015-0738-7.xml. Length: 2313 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-015-1052-0.xml. Length: 2608 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13054-019-2488-4.xml. Length: 2495 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13058-015-0618-8.xml. Length: 1942 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-017-1200-8.xml. Length: 1087 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-018-1557-3.xml. Length: 2358 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1749-5.xml. Length: 2620 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1840-y.xml. Length: 2024 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1908-8.xml. Length: 3320 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1918-6.xml. Length: 2328 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-019-1924-8.xml. Length: 2953 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-020-02048-6.xml. Length: 2723 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13059-020-1949-z.xml. Length: 3067 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-017-0901-2.xml. Length: 2675 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1078-z.xml. Length: 2363 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1167-z.xml. Length: 2832 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13068-018-1316-4.xml. Length: 2182 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-016-1359-y.xml. Length: 2178 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-2842-4.xml. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-2964-8.xml. Length: 3272 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-3084-1.xml. Length: 2393 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13071-018-3237-2.xml. Length: 2317 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13072-019-0285-6.xml. Length: 2170 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13072-019-0322-5.xml. Length: 2018 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-019-0674-2.xml. Length: 3122 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-019-0709-8.xml. Length: 3577 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13073-020-00727-4.xml. Length: 3472 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13100-019-0153-8.xml. Length: 2908 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-018-4014-1.xml. Length: 1670 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-019-4058-x.xml. Length: 2014 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13104-019-4127-1.xml. Length: 1893 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13148-018-0581-1.xml. Length: 2504 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13148-019-0719-9.xml. Length: 2525 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-015-0110-6.xml. Length: 2182 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-017-0223-1.xml. Length: 3609 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-017-0246-7.xml. Length: 2746 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13321-018-0298-3.xml. Length: 2505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13326-019-0195-3.xml. Length: 2556 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13568-018-0680-6.xml. Length: 2357 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s13643-018-0859-6.xml. Length: 2748 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-017-0338-7.xml. Length: 3144 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-018-0532-2.xml. Length: 3741 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40168-018-0550-0.xml. Length: 2921 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40169-019-0253-6.xml. Length: 2703 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40170-018-0185-4.xml. Length: 2168 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40170-020-00212-x.xml. Length: 1916 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40425-019-0730-x.xml. Length: 3142 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40478-018-0561-x.xml. Length: 3066 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40478-018-0655-5.xml. Length: 2446 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40793-015-0095-9.xml. Length: 1679 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1186_s40851-018-0089-8.xml. Length: 2254 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1212_wnl.0000000000006035.xml. Length: 2297 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1242_bio.035071.xml. Length: 1605 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1242_dev.138545.xml. Length: 2573 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.10556.1.xml. Length: 1787 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.11698.1.xml. Length: 1937 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13064.1.xml. Length: 2530 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13483.1.xml. Length: 2379 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13622.1.xml. Length: 2933 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.13852.1.xml. Length: 1242 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.12688_f1000research.4660.1.xml. Length: 1733 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pcbi.1011828.xml. Length: 3006 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pntd.0005385.xml. Length: 2166 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0070749.xml. Length: 1906 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0093021.xml. Length: 2336 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0096457.xml. Length: 1081 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0117675.xml. Length: 1942 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0137181.xml. Length: 1868 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0139215.xml. Length: 2229 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0146274.xml. Length: 2264 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0159387.xml. Length: 1910 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0170126.xml. Length: 1502 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0188323.xml. Length: 1986 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0191086.xml. Length: 2129 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0198382.xml. Length: 2246 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0200041.xml. Length: 2451 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0212669.xml. Length: 2252 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0220399.xml. Length: 1823 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0253228.xml. Length: 1612 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0262974.xml. Length: 2678 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1371_journal.pone.0284951.xml. Length: 1435 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1534_g3.119.400401.xml. Length: 3180 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1534_g3.119.400993.xml. Length: 4701 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1413-785220192703169516.xml. Length: 1715 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1414-431x20198292.xml. Length: 1729 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1516-3180.2018.0270071218.xml. Length: 2076 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_1678-4685-gmb-2018-0055.xml. Length: 1506 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1590_s1677-5538.ibju.2019.0167.xml. Length: 1968 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1617_s11527-023-02260-3.xml. Length: 1589 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.17581_bp.2020.09104.xml. Length: 2547 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.20944_preprints202009.0353.v1.xml. Length: 1522 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.21105_joss.04237.xml. Length: 2533 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.21203_rs.3.rs-3338732_v1.xml. Length: 2291 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_cir1497.xml. Length: 2297 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_fs20233046.xml. Length: 767 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20201035.xml. Length: 541 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20231026.xml. Length: 310 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3133_ofr20231027.xml. Length: 1204 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fcell.2020.618552.xml. Length: 3644 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fchem.2019.00828.xml. Length: 1251 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fcimb.2024.1292467.xml. Length: 2450 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_feart.2023.1205211.xml. Length: 3742 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fmicb.2024.1456637.xml. Length: 2780 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3389_fpls.2023.1160645.xml. Length: 2616 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_d13010019.xml. Length: 1757 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_ijms12117360.xml. Length: 1877 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_inorganics7050058.xml. Length: 1045 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_microorganisms8121872.xml. Length: 1591 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules17077645.xml. Length: 2120 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules191017026.xml. Length: 2170 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules200917469.xml. Length: 1566 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules23123199.xml. Length: 1715 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_molecules25010053.xml. Length: 1787 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_rs4102923.xml. Length: 2361 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s19030479.xml. Length: 2209 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s23094491.xml. Length: 2566 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_s23177333.xml. Length: 2553 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3390_v11060565.xml. Length: 1365 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3762_bjoc.8.42.xml. Length: 792 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3762_bjoc.9.92.xml. Length: 1067 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_bdj.7.e47369.xml. Length: 1706 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_neobiota.82.87455.xml. Length: 2253 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zookeys.500.9360.xml. Length: 2361 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zoologia.35.e23481.xml. Length: 1953 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.3897_zoologia.36.e32053.xml. Length: 958 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-2021-570.xml. Length: 2312 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-22-2769-2022.xml. Length: 2423 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_acp-22-5701-2022.xml. Length: 913 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_amt-15-3969-2022.xml. Length: 416 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-12-1287-2020.xml. Length: 2085 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2019-206.xml. Length: 1365 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2023-187.xml. Length: 2998 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_essd-2023-198.xml. Length: 1675 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_gmd-12-4221-2019.xml. Length: 2027 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5194_gmd-13-335-2020.xml. Length: 2645 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.5937_bnhmb1811227u.xml. Length: 2110 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_eLife.63194.xml. Length: 3418 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_eLife.72626.xml. Length: 3638 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.29944.xml. Length: 2371 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.42722.xml. Length: 1783 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.62329.xml. Length: 2189 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.63455.xml. Length: 3524 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.73695.xml. Length: 3656 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7554_elife.74937.xml. Length: 2337 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.10452.xml. Length: 1410 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.11352.xml. Length: 1719 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.12422.xml. Length: 1724 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.7717_peerj.13193.xml. Length: 2407 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf. Length: 4956 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf. Length: 1355 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf. Length: 1736 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202007717.pdf. Length: 1734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201902131.pdf. Length: 671 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf. Length: 941 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202000235.pdf. Length: 734 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf. Length: 1057 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001668.pdf. Length: 506 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202003167.pdf. Length: 955 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_cssc.202201821.pdf. Length: 3920 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.3985.pdf. Length: 3707 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.4466.pdf. Length: 2364 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf. Length: 2692 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5395.pdf. Length: 4026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf. Length: 2964 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6303.pdf. Length: 3312 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6784.pdf. Length: 6896 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.961.pdf. Length: 2882 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.9627.pdf. Length: 2825 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf. Length: 3339 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.4619.pdf. Length: 3555 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf. Length: 606 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf. Length: 970 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000916.pdf. Length: 284 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf. Length: 4494 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5090.pdf. Length: 5378 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf. Length: 4658 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_nafm.10870.pdf. Length: 4574 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1007_jhep07(2018)134.pdf. Length: 879 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml. Length: 1633 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml. Length: 1124 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml. Length: 234 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202007717.xml. Length: 1444 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201902131.xml. Length: 1262 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201903120.xml. Length: 2036 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202000235.xml. Length: 867 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001412.xml. Length: 1382 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001668.xml. Length: 992 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202003167.xml. Length: 1983 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_cssc.202201821.xml. Length: 1690 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.3985.xml. Length: 1716 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.4466.xml. Length: 1794 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5260.xml. Length: 2560 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5395.xml. Length: 4026 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6144.xml. Length: 2188 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6303.xml. Length: 2672 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6784.xml. Length: 2900 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.961.xml. Length: 1771 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.9627.xml. Length: 2505 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ecs2.4619.xml. Length: 2080 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejic.201900904.xml. Length: 1014 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejoc.202000916.xml. Length: 669 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_esp.5090.xml. Length: 2714 characters\n",
      "Extracted text from ./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_mp.14424.xml. Length: 3030 characters\n",
      "Training file paths DataFrame shape: (524, 9)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "5ea21167-7eb2-447d-ab55-51cdbffdcf5f",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 9,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...   \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...   \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...   \n",
       "\n",
       "                                            xml_text  \n",
       "0  {\"title\":\"Assessing the variability in the rel...  \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...  \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Training file paths DataFrame shape: (30, 10)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "781399ea-2353-4af7-86c3-11895d22f95f",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}\n{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 10,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...   \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...   \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...   \n",
       "\n",
       "                                            xml_text  \\\n",
       "0  {\"title\":\"Assessing the variability in the rel...   \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...   \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...   \n",
       "\n",
       "                                                text  \n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...  \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...  \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the PDF text content for each article in the training set\n",
    "train_file_paths_df['pdf_text'] = train_file_paths_df['pdf_file_path'].apply(lambda x: load_article_text(x, xml_type=None) if x else \"\")\n",
    "# Load the XML text content for each article in the training set\n",
    "train_file_paths_df['xml_text'] = train_file_paths_df.apply(lambda row: load_article_text(row['xml_file_path'], xml_type=row['xml_type']) if row['xml_file_path'] else \"\", axis=1)\n",
    "# Load the PDF text content for each article in the test set\n",
    "test_file_paths_df['pdf_text'] = test_file_paths_df['pdf_file_path'].apply(lambda x: load_article_text(x, xml_type=None) if x else \"\")\n",
    "# Load the XML text content for each article in the test set\n",
    "test_file_paths_df['xml_text'] = test_file_paths_df.apply(lambda row: load_article_text(row['xml_file_path'], xml_type=row['xml_type']) if row['xml_file_path'] else \"\", axis=1)\n",
    "# Combine the PDF and XML text content into a single 'text' column for training and test sets\n",
    "# train_file_paths_df['text'] = train_file_paths_df['pdf_text'] + \"\\n\" + train_file_paths_df['xml_text']\n",
    "test_file_paths_df['text'] = test_file_paths_df['pdf_text'] + \"\\n\" + test_file_paths_df['xml_text']\n",
    "# Display the first few rows of the training file paths DataFrame\n",
    "print(f\"Training file paths DataFrame shape: {train_file_paths_df.shape}\")\n",
    "display(train_file_paths_df.head(3))\n",
    "# Display the first few rows of the training file paths DataFrame\n",
    "print(f\"Training file paths DataFrame shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.head(3))\n",
    "# Save the training and test file paths DataFrames to CSV files\n",
    "train_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'train_file_paths.csv'), index=False)\n",
    "test_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'test_file_paths.csv'), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "b986f4a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the training and test file paths DataFrames to CSV files\n",
    "train_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'train_file_paths.csv'), index=False)\n",
    "test_file_paths_df.to_csv(os.path.join(BASE_OUTPUT_DIR, 'test_file_paths.csv'), index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "2640c46b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- QwenModelEval Class ---\n",
    "# kagglehub.model_download(\"qwen-lm/qwen-3/transformers/0.6b\")\n",
    "#max_new_tokens=32768\n",
    "class QwenModelEval:\n",
    "    def __init__(self, model_name, sys_prompt, enable_thinking=True, max_new_tokens=1024, max_input_length=8200):\n",
    "        print(f\"Loading Qwen model and tokenizer from: {model_name}\")\n",
    "        self.model_name = model_name\n",
    "        self.sys_prompt = sys_prompt\n",
    "        self.enable_thinking = enable_thinking  # Enable or disable thinking mode\n",
    "        self.max_new_tokens = max_new_tokens  # Set the maximum number of new tokens to generate\n",
    "        self.max_input_length = max_input_length  # Set the maximum input length for the model\n",
    "        # Load the tokenizer and model\n",
    "        # Using trust_remote_code=True to allow custom model code execution\n",
    "        self.tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)\n",
    "        self.model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=\"auto\", device_map=\"auto\", trust_remote_code=True)\n",
    "        self.model.eval() # Set the model to evaluation mode here.\n",
    "\n",
    "    def generate_response(self, user_input):  \n",
    "        inputs = self._get_inputs(user_input)\n",
    "        # Disable gradient calculation during inference\n",
    "        # Generate the response using the model\n",
    "        with torch.no_grad(): \n",
    "            generated_ids = self.model.generate(\n",
    "                **inputs,\n",
    "                max_new_tokens=self.max_new_tokens,\n",
    "                pad_token_id=self.tokenizer.eos_token_id,\n",
    "                eos_token_id=self.tokenizer.convert_tokens_to_ids(\"<|im_end|>\"),\n",
    "                # do_sample=False, # Use greedy decoding (fastest)\n",
    "                # num_beams=1,     # Do not use beam search (fastest)\n",
    "                # temperature=0.0, # Make output deterministic (if do_sample=False, this has no effect)                \n",
    "                temperature=0.6 if self.enable_thinking else 0.7,\n",
    "                top_p=0.95 if self.enable_thinking else 0.8,\n",
    "                top_k=20,\n",
    "                min_p=0\n",
    "            )\n",
    "        # Parse the response and thinking content\n",
    "        return self._parse_response(inputs, generated_ids)\n",
    "\n",
    "    def _get_inputs(self, user_input):\n",
    "        \"\"\"Prepare the input for the model based on user input.\"\"\"\n",
    "        # Trim the user input to a maximum length for better performance\n",
    "        user_input = user_input[:self.max_input_length]  # Limit input length to 4096 characters\n",
    "        print(f\"Preparing input with length: {len(user_input)}\")\n",
    "        # Create the messages for the chat template\n",
    "        messages = [\n",
    "            {\"role\": \"system\", \"content\": self.sys_prompt},\n",
    "            {\"role\": \"user\", \"content\": user_input}\n",
    "        ]\n",
    "        text = self.tokenizer.apply_chat_template(\n",
    "            messages,\n",
    "            tokenize=False,\n",
    "            add_generation_prompt=True,\n",
    "            enable_thinking=self.enable_thinking\n",
    "        )\n",
    "        return self.tokenizer(text, return_tensors=\"pt\").to(self.model.device)\n",
    "    \n",
    "    def _parse_response(self, inputs, generated_ids):\n",
    "        print(\"Parsing response from generated IDs...\")\n",
    "        # Extract the output IDs from the generated IDs\n",
    "        output_ids = generated_ids[0][len(inputs.input_ids[0]):].tolist()\n",
    "        try:\n",
    "            index = len(output_ids) - output_ids[::-1].index(151668)\n",
    "        except ValueError:\n",
    "            index = 0\n",
    "\n",
    "        thinking_content = self.tokenizer.decode(output_ids[:index], skip_special_tokens=True).strip(\"\\n\")\n",
    "        raw_response = self.tokenizer.decode(output_ids[index:], skip_special_tokens=True).strip(\"\\n\")\n",
    "        response = self._parse_json(raw_response)\n",
    "        return response, thinking_content\n",
    "    \n",
    "    def _parse_json(self, raw_response: str) -> list[dict[str,str]]:\n",
    "        # Remove code block markers and leading/trailing whitespace\n",
    "        cleaned = raw_response.strip()\n",
    "        if cleaned.startswith(\"```json\"):\n",
    "            cleaned = cleaned[len(\"```json\"):].strip()\n",
    "        if cleaned.endswith(\"```\"):\n",
    "            cleaned = cleaned[:-3].strip()\n",
    "\n",
    "        # Now parse as JSON\n",
    "        try:\n",
    "            return json.loads(cleaned)\n",
    "        except json.JSONDecodeError as e:\n",
    "            return []        \n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "id": "74e5a664",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the one-shot reasoning and task prompt\n",
    "# This prompt is designed to guide the model through a structured reasoning process\n",
    "SYS_PROMPT = \"\"\"\n",
    "You are an advanced AI research assistant specialized in identifying and classifying datasets used within academic research papers.\n",
    "Your primary goal is to accurately extract and categorize dataset identifiers (dataset_ids) from provided paper sections.\n",
    "\n",
    "---\n",
    "\n",
    "### Input Data Structure\n",
    "\n",
    "You will receive a JSON string representing key sections of an academic paper, structured as follows:\n",
    "\n",
    "```json\n",
    "{\n",
    "    \"title\": \"Title of the paper\",\n",
    "    \"authors\": \"List of authors, e.g., 'Author A, Author B'\",\n",
    "    \"abstract\": \"Abstract of the paper\",\n",
    "    \"data_availability\": \"Data availability information\",\n",
    "    \"other_dataset_citations\": [\n",
    "        {\"dataset_ids\": [\"10.12345/12345\"], \"citation_context\": \"Dataset citation context 1\"},\n",
    "        {\"dataset_ids\": [\"10.1234/xxxx.1x1x-xx11\", \"EPI_ISL_12345678\"], \"citation_context\": \"Dataset citation context 2\"},\n",
    "        ...\n",
    "    ]\n",
    "}\n",
    "```\n",
    "\n",
    "**Guidance on Input Sections:**\n",
    "*   **`title`**: Provides general context for the paper's topic.\n",
    "*   **`authors`**: Lists the authors of the paper, which can help in classifying datasets as Primary or Secondary.\n",
    "*   **`abstract`**: **Crucial** for understanding the research scope and classifying datasets as Primary or Secondary. Use it to determine if a dataset is *directly related* to *this paper's* specific research.\n",
    "*   **`data_availability`**: This section, if present, often contains **Primary** datasets directly used or generated by the authors for the research of this paper.\n",
    "*   **`other_dataset_citations`**: A list of potential dataset citations. The `citation_context` is vital to confirm if a `dataset_id` truly refers to a dataset and to aid in classification.\n",
    "\n",
    "---\n",
    "\n",
    "### Core Objective & Critical Exclusion\n",
    "\n",
    "Your overarching objective is to identify and classify **only valid, data-related `dataset_id`s**.\n",
    "\n",
    "**CRITICAL EXCLUSION**: You **MUST NOT** extract any `dataset_id`s that refer to other academic papers, articles, or the paper itself. Focus strictly on identifiers for *datasets*.\n",
    "\n",
    "---\n",
    "\n",
    "### Key Definitions\n",
    "\n",
    "*   **`dataset_id`**: A unique, persistent identifier for a dataset. There are two main types:\n",
    "\n",
    "    1.  **Digital Object Identifier (DOI)**:\n",
    "        *   **Format**: `[prefix]/[suffix]`. The prefix always starts with \"10.\" and is followed by a 4 or 5 digit number. The suffix can contain letters, numbers, and special characters.\n",
    "        *   May or may not start with \"https://doi.org/\" or \"doi:\".\n",
    "        *   **IMPORTANT DOI VALIDATION RULE**:\n",
    "            *   Only identify DOIs that are explicitly used as `dataset_id`s.\n",
    "            *   **DO NOT extract DOIs for academic papers/articles.**\n",
    "            *   **If a DOI is presented as a reference to a publication (e.g., \"as described in [DOI]\", \"cited in [DOI]\", \"see [DOI] for details on the method\"), it is NOT a dataset_id.**\n",
    "            *   A DOI is a `dataset_id` ONLY if the surrounding `citation_context` or `data_availability` section clearly indicates it refers to a dataset, data repository, data archive, or similar data-specific entity.\n",
    "\n",
    "    2.  **Accession ID**:\n",
    "        *   Typically alphanumeric strings that uniquely identify a dataset within a specific data repository.\n",
    "        *   Can be found in both `data_availability` and `other_dataset_citations` sections.\n",
    "        *   *Examples*: `\"EPI_ISL_12345678\"` (EPI dataset), `\"IPR000264\"` (InterPro dataset), `\"SAMN07159041\"` (NCBI Sequence Read Archive dataset), `\"CHEMBL1782574\"` (ChEMBL dataset)\n",
    "\n",
    "*   **Dataset Type Classification**:\n",
    "\n",
    "    *   **Primary**: Raw or processed data *generated or created by the **authors** as part of the research specifically related to the topic of this paper*. This data is central to the findings presented in *this* article.\n",
    "        *   *Keywords often associated with Primary*: \"generated in this study\", \"our data\", \"newly sequenced\", \"collected for this project\".\n",
    "    *   **Secondary**: Raw or processed data *reused from existing records or published data that was only used by the authors and not created by the **authors** for this paper*. This data might be used for comparison, background, or as a starting point, but it's not the novel data produced by *this* specific research.\n",
    "        *   *Keywords often associated with Secondary*: \"publicly available data\", \"previously published\", \"re-analyzed data\", \"obtained from\", \"data from [another study/database]\", \"existing dataset\", \"external database\".\n",
    "    *   **Fallback Rule**: If, after careful consideration of the `abstract` and `citation_context`, the relationship of the dataset to *this paper's* specific research remains ambiguous, then default to \"Primary\".\n",
    "\n",
    "---\n",
    "\n",
    "### Tasks: Step-by-Step Instructions\n",
    "\n",
    "Follow these three tasks in order:\n",
    "\n",
    "**Task 1: Identify Valid Dataset IDs**\n",
    "\n",
    "1.  **Search Priority**: Begin by searching the `data_availability` section. Then, search the `other_dataset_citations` section.\n",
    "2.  **Validation**: For each potential `dataset_id` (DOI or Accession ID), confirm it is truly data-related.\n",
    "    *   **For DOIs**: Strictly apply the **IMPORTANT DOI VALIDATION RULE** defined above. If it refers to a publication, **DO NOT** extract it.\n",
    "    *   **For all IDs**: Look for surrounding terms like \"data release\", \"data availability\", \"dataset\", \"database\", \"repository\", \"data source\", \"data access\", or \"data archive\" within the `data_availability` section or the `citation_context`.\n",
    "3.  **Deduplication**: If the same `dataset_id` is found multiple times, **only process the first instance encountered**.\n",
    "4.  **Conditional Proceeding**:\n",
    "    *   If **no valid `dataset_id`s are found** after searching both sections, **skip directly to Task 3** and output the \"Missing\" JSON structure.\n",
    "    *   If one or more valid `dataset_id`s are found, proceed to Task 2.\n",
    "\n",
    "**Task 2: Classify Dataset Types**\n",
    "\n",
    "1.  For each valid `dataset_id` identified in Task 1, classify its type as either \"Primary\" or \"Secondary\".\n",
    "2.  **Crucially, use the `abstract` section and the `citation_context` (if applicable) to determine if the dataset is *directly related* to the novel research presented in *this paper*.**\n",
    "3.  Apply the \"Key Definitions\" for Primary and Secondary types, paying close attention to the associated keywords.\n",
    "4.  Remember the \"Fallback Rule\": Only default to \"Primary\" if, after careful consideration, the classification remains truly ambiguous.\n",
    "\n",
    "**Task 3: Format and Return Results**\n",
    "\n",
    "Return your final results as a JSON array of objects.\n",
    "\n",
    "1.  **Scenario A: No Valid Datasets Found**\n",
    "    If Task 1 resulted in no valid `dataset_id`s, return a single JSON object with the following structure:\n",
    "    ```json\n",
    "    [\n",
    "        {\n",
    "            \"dataset_id\": \"Missing\",\n",
    "            \"type\": \"Missing\"\n",
    "        }\n",
    "    ]\n",
    "    ```\n",
    "2.  **Scenario B: One or More Valid Datasets Found**\n",
    "    If Task 1 identified one or more valid `dataset_id`s, return every valid dataset found in a JSON array of objects, where each object has the following structure:\n",
    "    ```json\n",
    "    [\n",
    "        {\n",
    "            \"dataset_id\": \"example_id_1\",\n",
    "            \"type\": \"Primary\"\n",
    "        },\n",
    "        {\n",
    "            \"dataset_id\": \"example_id_2\",\n",
    "            \"type\": \"Secondary\"\n",
    "        },\n",
    "        ...\n",
    "    ]\n",
    "    ```\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "f8a8a117",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Qwen model and tokenizer from: C:\\Users\\jim\\.cache\\kagglehub\\models\\qwen-lm\\qwen-3\\transformers\\0.6b\\1\n"
     ]
    }
   ],
   "source": [
    "# Instantiate the QwenModelEval class with the model path and system prompt\n",
    "inference_model = QwenModelEval(QWEN_BASE_MODEL_PATH, sys_prompt=SYS_PROMPT, enable_thinking=True, max_new_tokens=1576)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "id": "749ca3d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_articles(file_paths_df: pd.DataFrame, model) -> pd.DataFrame:\n",
    "    results = []\n",
    "    for i, row in file_paths_df.iterrows():\n",
    "        article_id = row['article_id']\n",
    "        dataset_info = row['dataset_info']\n",
    "        pdf_text = row['pdf_text']\n",
    "        xml_text = row['xml_text']\n",
    "        text_type = \"XML\" if xml_text else \"PDF\"\n",
    "        text_content = xml_text if xml_text else pdf_text\n",
    "        response = \"\"\n",
    "        thinking_content = \"\"\n",
    "\n",
    "        # Prepare the user input for the model\n",
    "        user_input = f\"Text Content: {text_content}\\n\"\n",
    "        \n",
    "        if text_content:\n",
    "            print(f\"Processing article {i}/{len(file_paths_df)}: {article_id}, type: {text_type}\")\n",
    "            # Generate response from the model\n",
    "            response, thinking_content = model.generate_response(user_input)\n",
    "\n",
    "        results.append({\n",
    "            'article_id': article_id,\n",
    "            'dataset_info': dataset_info,\n",
    "            'text_type': text_type,\n",
    "            'llm_input': user_input,\n",
    "            'llm_response': response,\n",
    "            'llm_thinking_content': thinking_content\n",
    "        })\n",
    "\n",
    "    return pd.DataFrame(results).sort_values(by=[\"article_id\"]).reset_index(drop=True)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "dc937f32",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Training file paths DataFrame shape: (524, 9)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "b3f20f5b-e4c0-47e6-91de-fbb319a5e238",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_2017jc013030.xml",
         "train",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.201916483.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\train\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\train\\XML\\10.1002_anie.202005531.xml",
         "train",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 9,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>train</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...        train   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...   \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...   \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...   \n",
       "\n",
       "                                            xml_text  \n",
       "0  {\"title\":\"Assessing the variability in the rel...  \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...  \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Test file paths DataFrame shape: (30, 10)\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "ef80da02-6d22-4bb8-811a-215e9a9fea19",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}\n{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ]
       ],
       "shape": {
        "columns": 10,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0    10.1002_2017jc013030  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1  10.1002_anie.201916483  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_anie.202005531  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2  <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...   \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...   \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...   \n",
       "\n",
       "                                            xml_text  \\\n",
       "0  {\"title\":\"Assessing the variability in the rel...   \n",
       "1  {\"title\":\"Effective and Reversible Carbon Diox...   \n",
       "2  {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...   \n",
       "\n",
       "                                                text  \n",
       "0  {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...  \n",
       "1  {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...  \n",
       "2  {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Load the training file paths DataFrame from the CSV file\n",
    "train_file_paths_df = pd.read_csv(os.path.join(BASE_OUTPUT_DIR, 'train_file_paths.csv'))\n",
    "# Load the test file paths DataFrame from the CSV file\n",
    "test_file_paths_df = pd.read_csv(os.path.join(BASE_OUTPUT_DIR, 'test_file_paths.csv'))\n",
    "# Display the first few rows of the training file paths DataFrame\n",
    "print(f\"Training file paths DataFrame shape: {train_file_paths_df.shape}\")\n",
    "display(train_file_paths_df.head(3))\n",
    "# Display the first few rows of the test file paths DataFrame\n",
    "print(f\"Test file paths DataFrame shape: {test_file_paths_df.shape}\")\n",
    "display(test_file_paths_df.head(3))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "id": "71a285a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "119cd934-0108-49fa-b539-658c2a5e6844",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_2017jc013030.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_2017jc013030.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}",
         "{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard Gentili, Chla, Grigor, Antoine Poteau, Edouard Leymarie, Catherine, Marie Barbieux, Fabrizio D'Ortenzio, Julia Uitz\",\"abstract\":\"Abstract Characterizing phytoplankton distribution and dynamics in the world\\u2019s open oceans requires in situ observations over a broad range of space and time scales.\\nIn addition to temperature/salinity measure-ments, Biogeochemical-Argo (BGC-Argo) profiling floats are capable of autonomously observing at high-frequency bio-optical properties such as the chlorophyll fluorescence, a proxy of the chlorophyll a concen-tration (Chla), the particulate backscattering coefficient (bbp), a proxy of the stock of particulate organic car-bon, and the light available for photosynthesis.\\nWe analyzed an unprecedented BGC-Argo database of more than 8,500 multivariable profiles collected in various oceanic conditions, from subpolar waters to subtropical gyres.\\nOur objective is to refine previously established Chla versus bbp relationships and gain insights into the sources of vertical, seasonal, and regional variability in this relationship.\\nDespite some regional, seasonal and vertical variations, a general covariation occurs at a global scale.\\nWe distinguish two main contrasted situations: (1) concomitant changes in Chla and bbp that correspond to actual variations in phytoplankton biomass, e.g., in subpolar regimes; (2) a decoupling between the two variables attributed to photoacclima-tion or changes in the relative abundance of nonalgal particles, e.g., in subtropical regimes.\\nThe variability in the bbp:Chla ratio in the surface layer appears to be essentially influenced by the type of particles and by photoacclimation processes.\\nThe large BGC-Argo database helps identifying the spatial and temporal scales at which this ratio is predominantly driven by one or the other of these two factors.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/2017JC013030'], citation_context: \\\"Citation: Barbieux, M., Uitz, J., Bricaud, A., Organelli, E., Poteau, A., Schmechtig, C., . . .\\nClaustre, H. (2018).\\nAssessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database.\\nJournal of Geophysical Research: Oceans, 123, 1229-1250.\\nhttps://doi.\\norg/10.1002/2017JC013030\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"nelli, E., Claustre, H., Schmechtig, C., Poteau, A., Boss, E., . . .\\nXing, X. (2017).\\nA global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications.\\nSEANOE.\\nhttps://doi.org/10.17882/49388\\nBarton, A. D., Lozier, M. S., & Williams, R. G. (2015).\\nPhysical controls of variability in North Atlantic phytoplankton communities.\\nLimnology and Oceanography, 60(1), 181-197.\\nhttps://doi.org/10.1002/lno.10011 Beckmann, A., & Hense, I. (2007).\\nBen\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/j.rse.'], citation_context: \\\", Alvain, S., Vantrepotte, V., & Huertas, I. E. (2014).\\nIdentification of dominant phytoplankton functional types in the Mediterra-nean Sea based on a regionalized remote sensing approach.\\nRemote Sensing of Environment, 152, 557-575.\\nhttps://doi.org/10.1016/j.rse.\\n2014.06.029 Organelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and\\\"},{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"mechtig, C., Poteau, A., Bricaud, A., . . .\\nDall\\u2019Olmo, G. (2016a).\\nA global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications.\\nSEANOE.\\nhttps://doi.org/10.17882/47142\\nOrganelli, E., Barbieux, M., Claustre, H., Schmechtig, C., Poteau, A., Bricaud, A., . . .\\nXing, X. (2017b).\\nTwo databases derived from BGC-Argo float measurements for marine biogeochemical and bio-optical applications.\\nEarth System Science Data, 9, \\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"cknowledged for useful comments and fruitful discussion.\\nWe also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"},{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\" also thank the International Argo Program and the CORIOLIS project that contribute to make the data freely and publicly available.\\nData referring to Organelli et al. (2016a; https://doi.org/10.17882/47142) and Barbieux et al. (2017; https://doi.org/10.17882/49388) are freely available on SEANOE.\\\"}\"]}\n{\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.201916483.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.201916483.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel Werner\",\"abstract\":\"Abstract: The homoleptic pyrazolate complexes [CeIII 4-(Me2pz)12] and [CeIV(Me2pz)4]2 quantitatively insert CO2 to give [CeIII 4(Me2pz\\u00b7CO2)12] and [CeIV(Me2pz\\u00b7CO2)4], respec-tively (Me2pz = 3,5-dimethylpyrazolato).\\nThis process is rever-sible for both complexes, as observed by in situ IR and NMR spectroscopy in solution and by TGA in the solid state.\\nBy adjusting the molar ratio, one molecule of CO2 per [CeIV-(Me2pz)4] complex could be inserted to give trimetallic [Ce3-(Me2pz)9(Me2pz\\u00b7CO2)3(thf)].\\nBoth the cerous and ceric in-sertion products catalyze the formation of cyclic carbonates from epoxides and CO2 under mild conditions.\\nIn the absence of epoxide, the ceric catalyst is prone to reduction by the co-catalyst tetra-n-butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.201916483'], citation_context: \\\"15213773, 2020, 14, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.201916483 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202005531.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202005531.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE collection SYSTEM \"BioC.dtd\"><collection><source>BioC-API</source><date>20250509",
         "bioc",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Trifluoromethyl, Dong Wang, C. Grace Carlton, Masanori Tayu, Joseph J. W. McDouall, David J. Procter\",\"abstract\":\"Abstract: Trifluoromethyl sulfoxides are a new class of trifluoromethylthiolating reagent.\\nThe sulfoxides engage in metal-free C\\ufffdH trifluoromethylthiolation with a range of (hetero)arenes.\\nThe method is also applicable to the function-alization of important compound classes, such as ligand derivatives and polyaromatics, and in the late-stage trifluor-omethylthiolation of medicines and agrochemicals.\\nThe iso-lation and characterization of a sulfonium salt intermediate supports an interrupted Pummerer reaction mechanism.\\nIncorporating fluorine into organic compounds is a useful tool in drug design and development.\\nThe fluoro group is well known to improve the pharmacokinetic properties of a mol-ecule and fluorine-18 is an important radioisotope in molec-ular imaging.[1,2] Trifluoromethylthio (SCF3) groups are commonly found in drug molecules and veterinary medi-cines.[3,4]\\nBy combining a fluorinated moiety with a hetero-atom, many have turned to the SCF3 group to impart useful properties, such as high lipophilicity, to a compound of interest.[5]\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202005531'], citation_context: \\\"15213773, 2020, 37, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202005531 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_anie.202007717.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_anie.202007717.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximilian Sigmund, Lutz Greb\",\"abstract\":\"Abstract: Metal-ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis.\\nBy use of the calix[4]pyrrolato aluminate, [1]\\ufffd, which features a square-planar AlIII, we transfer this concept into the p-block and fully elucidate its mechanisms by experiment and theory.\\nComple-mentary to transition metal-based MLC (aromatization upon substrate binding), substrate binding in [1]\\ufffdoccurs by dearomatization of the ligand.\\nThe aluminate trapps carbonyls by the formation of C\\ufffdC and Al\\ufffdO bonds, but the products maintain full reversibility and outstanding dynamic exchange rates.\\nRemarkably, the C\\ufffdC bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unpre-cedented control over the system\\ufffds constitutional state.\\nMore-over, the metal-ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense.\\nUltimately, this work describes the evolution of an anti-van\\ufffdt Hoff/Le Bel species from their being as a struc-tural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202007717'], citation_context: \\\"15213773, 2020, 39, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202007717 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Metal\\u2013Ligand Cooperativity of the Calix[4]pyrrolato Aluminate: Triggerable C\\u2212C Bond Formation and Rate Control in Catalysis\",\"authors\":\"Ebner Fabian, Sigmund Lukas Maximilian, Greb Lutz\",\"abstract\":\"Abstract Metal\\u2010ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis. By use of the calix[4]pyrrolato aluminate, [ 1 ] \\u2212 , which features a square\\u2010planar Al III , we transfer this concept into the  p \\u2010block and fully elucidate its mechanisms by experiment and theory. Complementary to transition metal\\u2010based MLC ( aromatization  upon substrate binding), substrate binding in [ 1 ] \\u2212  occurs by  dearomatization  of the ligand. The aluminate trapps carbonyls by the formation of C\\u2212C and Al\\u2212O bonds, but the products maintain full reversibility and outstanding dynamic exchange rates. Remarkably, the C\\u2212C bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unprecedented control over the system's constitutional state. Moreover, the metal\\u2010ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense. Ultimately, this work describes the evolution of an  anti\\u2010van't Hoff/Le Bel  species from their being as a structural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximilian Sigmund, Lutz Greb\",\"abstract\":\"Abstract: Metal-ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis.\\nBy use of the calix[4]pyrrolato aluminate, [1]\\ufffd, which features a square-planar AlIII, we transfer this concept into the p-block and fully elucidate its mechanisms by experiment and theory.\\nComple-mentary to transition metal-based MLC (aromatization upon substrate binding), substrate binding in [1]\\ufffdoccurs by dearomatization of the ligand.\\nThe aluminate trapps carbonyls by the formation of C\\ufffdC and Al\\ufffdO bonds, but the products maintain full reversibility and outstanding dynamic exchange rates.\\nRemarkably, the C\\ufffdC bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unpre-cedented control over the system\\ufffds constitutional state.\\nMore-over, the metal-ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense.\\nUltimately, this work describes the evolution of an anti-van\\ufffdt Hoff/Le Bel species from their being as a struc-tural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/anie.202007717'], citation_context: \\\"15213773, 2020, 39, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/anie.202007717 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Metal\\u2013Ligand Cooperativity of the Calix[4]pyrrolato Aluminate: Triggerable C\\u2212C Bond Formation and Rate Control in Catalysis\",\"authors\":\"Ebner Fabian, Sigmund Lukas Maximilian, Greb Lutz\",\"abstract\":\"Abstract Metal\\u2010ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis. By use of the calix[4]pyrrolato aluminate, [ 1 ] \\u2212 , which features a square\\u2010planar Al III , we transfer this concept into the  p \\u2010block and fully elucidate its mechanisms by experiment and theory. Complementary to transition metal\\u2010based MLC ( aromatization  upon substrate binding), substrate binding in [ 1 ] \\u2212  occurs by  dearomatization  of the ligand. The aluminate trapps carbonyls by the formation of C\\u2212C and Al\\u2212O bonds, but the products maintain full reversibility and outstanding dynamic exchange rates. Remarkably, the C\\u2212C bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unprecedented control over the system's constitutional state. Moreover, the metal\\u2010ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense. Ultimately, this work describes the evolution of an  anti\\u2010van't Hoff/Le Bel  species from their being as a structural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201902131.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201902131.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, Stefan F. Kirsch*[a]\",\"abstract\":\"Abstract: A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pat-tern that was found in a range of compounds featuring geminal diazido units.\\nThrough a simple two-step procedure, the synthesis of structurally diverse target compounds con-taining a tetrazole, such as tetrazoloquinoxalinones, benzoy-laryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"A Synthetic Route Toward Tetrazoles: The Thermolysis of Geminal Diazides\",\"authors\":\"Holzschneider Kristina, Tong My Linh, Mohr Fabian Prof.\\u2005Dr., Kirsch Stefan F. Prof.\\u2005Dr.\",\"abstract\":\"Abstract A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pattern that was found in a range of compounds featuring geminal diazido units. Through a simple two\\u2010step procedure, the synthesis of structurally diverse target compounds containing a tetrazole, such as tetrazoloquinoxalinones, benzoylaryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and phenanthrols). The initial oxidative diazidation reaction with iodine and sodium azide under mild conditions is followed by the thermal fragmentation under microwave irradiation, leading to the tetrazole products. Noteworthy, an experimental solution is presented in which the potentially hazardous diazide intermediates are not isolated and the concentration of crude reaction mixtures containing diazides is not required to achieve the tetrazoles in good yields.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, Stefan F. Kirsch*[a]\",\"abstract\":\"Abstract: A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pat-tern that was found in a range of compounds featuring geminal diazido units.\\nThrough a simple two-step procedure, the synthesis of structurally diverse target compounds con-taining a tetrazole, such as tetrazoloquinoxalinones, benzoy-laryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"A Synthetic Route Toward Tetrazoles: The Thermolysis of Geminal Diazides\",\"authors\":\"Holzschneider Kristina, Tong My Linh, Mohr Fabian Prof.\\u2005Dr., Kirsch Stefan F. Prof.\\u2005Dr.\",\"abstract\":\"Abstract A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pattern that was found in a range of compounds featuring geminal diazido units. Through a simple two\\u2010step procedure, the synthesis of structurally diverse target compounds containing a tetrazole, such as tetrazoloquinoxalinones, benzoylaryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and phenanthrols). The initial oxidative diazidation reaction with iodine and sodium azide under mild conditions is followed by the thermal fragmentation under microwave irradiation, leading to the tetrazole products. Noteworthy, an experimental solution is presented in which the potentially hazardous diazide intermediates are not isolated and the concentration of crude reaction mixtures containing diazides is not required to achieve the tetrazoles in good yields.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.201903120.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.201903120.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Matthias, Philippe R. Meisinger, Reinald Fischer, Regina Suxdorf, Helmar G\\u00e7rls\",\"abstract\":\"Abstract: Systematic variation of the 1,4-dioxane (dx) con-centration during the precipitation of sparingly soluble [MgBr2(dx)2] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R2Mg(dx)n] (n=1, 1.5, 2, and 3), which form during this diox-ane method, depending on the bulkiness of R. The number-ing of the complexes explored in this study is based on the number n of dioxane molecules per magnesium atom, fol-lowed by the substituent R; an apostrophe denotes coordi-nation polymers.\\nThe following derivatives were studied by X-ray crystal-structure determination and NMR spectroscopy: n=1: [Me2Mg(m-dx)]1 (1\\u2019-Me) and [nPr2Mg(m-dx)]1 (1\\u2019-nPr); n=1.5: [{iPr2Mg(dx)}2(m-dx)] (1.5-iPr), [{oTol2Mg(dx)}2(m-dx)]\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Structure\\u2013Solubility Relationship of 1,4\\u2010Dioxane Complexes of Di(hydrocarbyl)magnesium\",\"authors\":\"Fischer Reinald Dr., G\\u00f6rls Helmar Dr., Meisinger Philippe R., Suxdorf Regina, Westerhausen Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract Systematic variation of the 1,4\\u2010dioxane (dx) concentration during the precipitation of sparingly soluble [MgBr 2 (dx) 2 ] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R 2 Mg(dx) n ] ( n =1, 1.5, 2, and 3), which form during this dioxane method, depending on the bulkiness of R. The numbering of the complexes explored in this study is based on the number  n  of dioxane molecules per magnesium atom, followed by the substituent R; an apostrophe denotes coordination polymers. The following derivatives were studied by X\\u2010ray crystal\\u2010structure determination and NMR spectroscopy:  n =1: [Me 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010Me ) and [ n Pr 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010 n Pr );  n =1.5: [{ i Pr 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 i Pr ), [{ o Tol 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 o Tol ), and [(Me 3 Si\\u2010C\\u2261C) 2 Mg(dx) 1.5 ] \\u221e  ( 1.5\\u2032\\u2010C 2 SiMe 3 );  n =2: [ t Bu 2 Mg(dx) 2 ] ( 2\\u2010 t Bu ) and [ o Tol 2 Mg(dx) 2 ] ( 2\\u2010 o Tol );  n =3: [Ph 2 Mg(dx) 3 ] ( 3\\u2010Ph ). In the structure types  1\\u2032 ,  1.5 , and  2 , the magnesium atom exhibits the coordination number 4, whereas pentacoordinate metal atoms are observed in types  3  and  1.5\\u2032 . The structure type  2\\u2032  is realized for [(Ph\\u2010C\\u2261C) 2 Mg(dx) 2 ] \\u221e  ( 2\\u2032\\u2010C 2 Ph ), [MgCl 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Cl ), and [MgBr 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Br ) with hexacoordinate metal atoms. The solubility of the dioxane adducts in common organic solvents strongly depends on the degree of aggregation with the solubility decreasing from molecular to strand to layer structures.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Matthias, Philippe R. Meisinger, Reinald Fischer, Regina Suxdorf, Helmar G\\u00e7rls\",\"abstract\":\"Abstract: Systematic variation of the 1,4-dioxane (dx) con-centration during the precipitation of sparingly soluble [MgBr2(dx)2] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R2Mg(dx)n] (n=1, 1.5, 2, and 3), which form during this diox-ane method, depending on the bulkiness of R. The number-ing of the complexes explored in this study is based on the number n of dioxane molecules per magnesium atom, fol-lowed by the substituent R; an apostrophe denotes coordi-nation polymers.\\nThe following derivatives were studied by X-ray crystal-structure determination and NMR spectroscopy: n=1: [Me2Mg(m-dx)]1 (1\\u2019-Me) and [nPr2Mg(m-dx)]1 (1\\u2019-nPr); n=1.5: [{iPr2Mg(dx)}2(m-dx)] (1.5-iPr), [{oTol2Mg(dx)}2(m-dx)]\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"Structure\\u2013Solubility Relationship of 1,4\\u2010Dioxane Complexes of Di(hydrocarbyl)magnesium\",\"authors\":\"Fischer Reinald Dr., G\\u00f6rls Helmar Dr., Meisinger Philippe R., Suxdorf Regina, Westerhausen Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract Systematic variation of the 1,4\\u2010dioxane (dx) concentration during the precipitation of sparingly soluble [MgBr 2 (dx) 2 ] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R 2 Mg(dx) n ] ( n =1, 1.5, 2, and 3), which form during this dioxane method, depending on the bulkiness of R. The numbering of the complexes explored in this study is based on the number  n  of dioxane molecules per magnesium atom, followed by the substituent R; an apostrophe denotes coordination polymers. The following derivatives were studied by X\\u2010ray crystal\\u2010structure determination and NMR spectroscopy:  n =1: [Me 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010Me ) and [ n Pr 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010 n Pr );  n =1.5: [{ i Pr 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 i Pr ), [{ o Tol 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 o Tol ), and [(Me 3 Si\\u2010C\\u2261C) 2 Mg(dx) 1.5 ] \\u221e  ( 1.5\\u2032\\u2010C 2 SiMe 3 );  n =2: [ t Bu 2 Mg(dx) 2 ] ( 2\\u2010 t Bu ) and [ o Tol 2 Mg(dx) 2 ] ( 2\\u2010 o Tol );  n =3: [Ph 2 Mg(dx) 3 ] ( 3\\u2010Ph ). In the structure types  1\\u2032 ,  1.5 , and  2 , the magnesium atom exhibits the coordination number 4, whereas pentacoordinate metal atoms are observed in types  3  and  1.5\\u2032 . The structure type  2\\u2032  is realized for [(Ph\\u2010C\\u2261C) 2 Mg(dx) 2 ] \\u221e  ( 2\\u2032\\u2010C 2 Ph ), [MgCl 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Cl ), and [MgBr 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Br ) with hexacoordinate metal atoms. The solubility of the dioxane adducts in common organic solvents strongly depends on the degree of aggregation with the solubility decreasing from molecular to strand to layer structures.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202000235.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202000235.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3, Matthias, Shenglai Yao, Lewis\",\"abstract\":\"Abstract: The reaction of bis(silylenyl)-substituted ferro-cene 1 with two molar equivalents of BPh3 yields the cor-responding bis(silylene-borane) Lewis adduct 2.\\nThe latter is capable to activate CO2 to furnish the borane-stabilized bis(silanone) 3 through mono-oxygenation of the dative SiII!B silicon centers under release of CO.\\nRemoval of BPh3 from 3 with PMe3 affords the corresponding 1,3,2,4-cyclodisiloxane and the Me3P\\ufffdBPh3 adduct.\\nAll isolated new compounds were characterized and their molecular structures were determined by single-crystal X-ray diffrac-tion analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"An Isolable Bis(Silanone\\u2013Borane) Adduct\",\"authors\":\"Luecke Marcel\\u2010Philip, Pens Elron, Yao Shenglai Dr., Driess Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract The reaction of bis(silylenyl)\\u2010substituted ferrocene  1  with two molar equivalents of BPh 3  yields the corresponding bis(silylene\\u2013borane) Lewis adduct  2 . The latter is capable to activate CO 2  to furnish the borane\\u2010stabilized bis(silanone)  3  through mono\\u2010oxygenation of the dative Si II \\u2192B silicon centers under release of CO. Removal of BPh 3  from  3  with PMe 3  affords the corresponding 1,3,2,4\\u2010cyclodisiloxane and the Me 3 P\\u2212BPh 3  adduct. All isolated new compounds were characterized and their molecular structures were determined by single\\u2010crystal X\\u2010ray diffraction analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3, Matthias, Shenglai Yao, Lewis\",\"abstract\":\"Abstract: The reaction of bis(silylenyl)-substituted ferro-cene 1 with two molar equivalents of BPh3 yields the cor-responding bis(silylene-borane) Lewis adduct 2.\\nThe latter is capable to activate CO2 to furnish the borane-stabilized bis(silanone) 3 through mono-oxygenation of the dative SiII!B silicon centers under release of CO.\\nRemoval of BPh3 from 3 with PMe3 affords the corresponding 1,3,2,4-cyclodisiloxane and the Me3P\\ufffdBPh3 adduct.\\nAll isolated new compounds were characterized and their molecular structures were determined by single-crystal X-ray diffrac-tion analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"An Isolable Bis(Silanone\\u2013Borane) Adduct\",\"authors\":\"Luecke Marcel\\u2010Philip, Pens Elron, Yao Shenglai Dr., Driess Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract The reaction of bis(silylenyl)\\u2010substituted ferrocene  1  with two molar equivalents of BPh 3  yields the corresponding bis(silylene\\u2013borane) Lewis adduct  2 . The latter is capable to activate CO 2  to furnish the borane\\u2010stabilized bis(silanone)  3  through mono\\u2010oxygenation of the dative Si II \\u2192B silicon centers under release of CO. Removal of BPh 3  from  3  with PMe 3  affords the corresponding 1,3,2,4\\u2010cyclodisiloxane and the Me 3 P\\u2212BPh 3  adduct. All isolated new compounds were characterized and their molecular structures were determined by single\\u2010crystal X\\u2010ray diffraction analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001412.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001412.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u00e7rls,[a, Wolfgang, Lawesson, Hassan Abul-Futouh,[c, Rolf Huisgen, Stefanie, FeFe\",\"abstract\":\"Abstract: Reactions of a,b-unsaturated aromatic thioke-tones 1 (thiochalcones) with Fe3(CO)12 leading to h4-1-thia-1,3-diene iron tricarbonyl complexes 2, [FeFe] hydroge-nase mimics 3, and the thiopyrane adduct 4 are described.\\nObtained products have been characterized by X-ray crys-tallography and by computational methods.\\nCompletely regio- and diastereoselective formation of the five-mem-bered ring system in products 3, containing four stereo-genic centers, can be explained by an unprecedented, stepwise (3+2)-cycloaddition of two thiochalcone mole-cules mediated by Fe3(CO)12.\\nQuantum chemical calcula-tions aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)-cycloaddition proceeds via sequential intramolecular radical transfer events upon ho-molytic cleavage of one carbon-sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Iron(0)\\u2010Mediated Stereoselective (3+2)\\u2010Cycloaddition of Thiochalcones via a Diradical Intermediate\",\"authors\":\"Buday Philipp, Seeber Phillip, Zens Clara, Abul\\u2010Futouh Hassan Dr., G\\u00f6rls Helmar Dr., Gr\\u00e4fe Stefanie Prof.\\u2005Dr., Matczak Piotr Dr., Kupfer Stephan Dr., Weigand Wolfgang Prof.\\u2005Dr., Mloston Grzegorz Prof.\\u2005Dr.\",\"abstract\":\"Abstract Reactions of \\u03b1,\\u03b2\\u2010unsaturated aromatic thioketones  1  (thiochalcones) with Fe 3 (CO) 12  leading to \\u03b7 4 \\u20101\\u2010thia\\u20101,3\\u2010diene iron tricarbonyl complexes\\u2005 2 , [FeFe] hydrogenase mimics  3 , and the thiopyrane adduct  4  are described. Obtained products have been characterized by X\\u2010ray crystallography and by computational methods. Completely regio\\u2010 and diastereoselective formation of the five\\u2010membered ring system in products\\u2005 3 , containing four stereogenic centers, can be explained by an unprecedented, stepwise (3+2)\\u2010cycloaddition of two thiochalcone molecules mediated by Fe 3 (CO) 12 . Quantum chemical calculations aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)\\u2010cycloaddition proceeds via sequential intramolecular radical transfer events upon homolytic cleavage of one carbon\\u2010sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u00e7rls,[a, Wolfgang, Lawesson, Hassan Abul-Futouh,[c, Rolf Huisgen, Stefanie, FeFe\",\"abstract\":\"Abstract: Reactions of a,b-unsaturated aromatic thioke-tones 1 (thiochalcones) with Fe3(CO)12 leading to h4-1-thia-1,3-diene iron tricarbonyl complexes 2, [FeFe] hydroge-nase mimics 3, and the thiopyrane adduct 4 are described.\\nObtained products have been characterized by X-ray crys-tallography and by computational methods.\\nCompletely regio- and diastereoselective formation of the five-mem-bered ring system in products 3, containing four stereo-genic centers, can be explained by an unprecedented, stepwise (3+2)-cycloaddition of two thiochalcone mole-cules mediated by Fe3(CO)12.\\nQuantum chemical calcula-tions aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)-cycloaddition proceeds via sequential intramolecular radical transfer events upon ho-molytic cleavage of one carbon-sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"Iron(0)\\u2010Mediated Stereoselective (3+2)\\u2010Cycloaddition of Thiochalcones via a Diradical Intermediate\",\"authors\":\"Buday Philipp, Seeber Phillip, Zens Clara, Abul\\u2010Futouh Hassan Dr., G\\u00f6rls Helmar Dr., Gr\\u00e4fe Stefanie Prof.\\u2005Dr., Matczak Piotr Dr., Kupfer Stephan Dr., Weigand Wolfgang Prof.\\u2005Dr., Mloston Grzegorz Prof.\\u2005Dr.\",\"abstract\":\"Abstract Reactions of \\u03b1,\\u03b2\\u2010unsaturated aromatic thioketones  1  (thiochalcones) with Fe 3 (CO) 12  leading to \\u03b7 4 \\u20101\\u2010thia\\u20101,3\\u2010diene iron tricarbonyl complexes\\u2005 2 , [FeFe] hydrogenase mimics  3 , and the thiopyrane adduct  4  are described. Obtained products have been characterized by X\\u2010ray crystallography and by computational methods. Completely regio\\u2010 and diastereoselective formation of the five\\u2010membered ring system in products\\u2005 3 , containing four stereogenic centers, can be explained by an unprecedented, stepwise (3+2)\\u2010cycloaddition of two thiochalcone molecules mediated by Fe 3 (CO) 12 . Quantum chemical calculations aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)\\u2010cycloaddition proceeds via sequential intramolecular radical transfer events upon homolytic cleavage of one carbon\\u2010sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "8",
         "10.1002_chem.202001668",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202001668.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202001668.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, Christopher E. Anson,[a, Annie K. Powell*[a, Irina A. K\\ufffdhne,[a\",\"abstract\":\"Abstract: We have added the {ReIVX5}\\ufffd(X=Br, Cl) synthon to a pocket-based ligand to provide supramolecular design using halogen\\u00b7\\u00b7\\u00b7halogen interactions within an FeIII system that has the potential to undergo spin crossover (SCO).\\nBy removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"The First Use of a ReX 5  Synthon to Modulate Fe III  Spin Crossover via Supramolecular Halogen\\u22c5\\u22c5\\u22c5Halogen Interactions\",\"authors\":\"Busch Rebecca Dr., Carter Anthony B., Konidaris Konstantis F. Dr., K\\u00fchne Irina A. Dr., Gonz\\u00e1lez Ricardo Dr., Anson Christopher E. Dr., Powell Annie K. Prof.\\u2005Dr.\",\"abstract\":\"Abstract We have added the {Re IV X 5 } \\u2212  (X=Br, Cl) synthon to a pocket\\u2010based ligand to provide supramolecular design using halogen\\u22c5\\u22c5\\u22c5halogen interactions within an Fe III  system that has the potential to undergo spin crossover (SCO). By removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d halogen\\u22c5\\u22c5\\u22c5halogen interactions between neighboring molecules, providing a supramolecular cooperative pathway for SCO. Furthermore, changes to the halogen\\u2010based interaction allow us to modify the temperature and nature of the SCO event.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, Christopher E. Anson,[a, Annie K. Powell*[a, Irina A. K\\ufffdhne,[a\",\"abstract\":\"Abstract: We have added the {ReIVX5}\\ufffd(X=Br, Cl) synthon to a pocket-based ligand to provide supramolecular design using halogen\\u00b7\\u00b7\\u00b7halogen interactions within an FeIII system that has the potential to undergo spin crossover (SCO).\\nBy removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"The First Use of a ReX 5  Synthon to Modulate Fe III  Spin Crossover via Supramolecular Halogen\\u22c5\\u22c5\\u22c5Halogen Interactions\",\"authors\":\"Busch Rebecca Dr., Carter Anthony B., Konidaris Konstantis F. Dr., K\\u00fchne Irina A. Dr., Gonz\\u00e1lez Ricardo Dr., Anson Christopher E. Dr., Powell Annie K. Prof.\\u2005Dr.\",\"abstract\":\"Abstract We have added the {Re IV X 5 } \\u2212  (X=Br, Cl) synthon to a pocket\\u2010based ligand to provide supramolecular design using halogen\\u22c5\\u22c5\\u22c5halogen interactions within an Fe III  system that has the potential to undergo spin crossover (SCO). By removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d halogen\\u22c5\\u22c5\\u22c5halogen interactions between neighboring molecules, providing a supramolecular cooperative pathway for SCO. Furthermore, changes to the halogen\\u2010based interaction allow us to modify the temperature and nature of the SCO event.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "9",
         "10.1002_chem.202003167",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_chem.202003167.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_chem.202003167.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bonding, Lewis, Macarena G. Alf\\ufffdrez+\",\"abstract\":\"Abstract: Metal-only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M!M bond represent privileged architectures to acquire funda-mental understanding of bimetallic bonding.\\nThis has impor-tant implications in many catalytic processes or supramolec-ular systems that rely on synergistic effects between two metals.\\nHowever, a systematic experimental/computational approach on a well-defined class of compounds is lacking.\\nHere we report a family of MOLPs constructed around the RhI precursor\\n[(h5-C5Me5)Rh(PMe3)2] (1) with a series of s, p and d-block metals, mostly from the main group elements, and investigate their bonding by computational means.\\nAmong the new MOLPs, we have structurally characterized those formed by dative bonding between 1 and MgMeBr,\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Metal\\u2010only Lewis Pairs of Rhodium with  s ,  p  and  d \\u2010Block Metals\",\"authors\":\"Bajo Sonia Dr., Alf\\u00e9rez Macarena G., Alcaide Mar\\u00eda M., L\\u00f3pez\\u2010Serrano Joaqu\\u00edn Dr., Campos Jes\\u00fas Dr.\",\"abstract\":\"Abstract Metal\\u2010only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M\\u2192M bond represent privileged architectures to acquire fundamental understanding of bimetallic bonding. This has important implications in many catalytic processes or supramolecular systems that rely on synergistic effects between two metals. However, a systematic experimental/computational approach on a well\\u2010defined class of compounds is lacking. Here we report a family of MOLPs constructed around the Rh I  precursor [( \\u03b7 \\n 5 \\u2010C 5 Me 5 )Rh(PMe 3 ) 2 ] ( 1 ) with a series of  s ,  p  and  d \\u2010block metals, mostly from the main group elements, and investigate their bonding by computational means. Among the new MOLPs, we have structurally characterized those formed by dative bonding between  1  and MgMeBr, AlMe 3 , GeCl 2 , SnCl 2 , ZnMe 2  and Zn(C 6 F 5 ) 2,  as well as spectroscopically identified the ones resulting from coordination to MBAr F  (M=Na, Li; BAr F \\n \\u2212 =[B(C 6 H 2 \\u20103,5\\u2010(CF 3 ) 2 ) 4 ] \\u2212 ) and CuCl. Some of these compounds represent unique examples of bimetallic structures, such as the first unambiguous cases of Rh\\u2192Mg dative bonding or base\\u2010free rhodium bound germylene and stannylene species. Multinuclear NMR spectroscopy, including  103 Rh NMR, is used to probe the formation of Rh\\u2192M bonds. A comprehensive theoretical analysis of those provides clear trends. As anticipated, greater bond covalency is found for the more electronegative acids, whereas ionic character dominates for the least electronegative nuclei, though some degree of electron sharing is identified in all cases.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bonding, Lewis, Macarena G. Alf\\ufffdrez+\",\"abstract\":\"Abstract: Metal-only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M!M bond represent privileged architectures to acquire funda-mental understanding of bimetallic bonding.\\nThis has impor-tant implications in many catalytic processes or supramolec-ular systems that rely on synergistic effects between two metals.\\nHowever, a systematic experimental/computational approach on a well-defined class of compounds is lacking.\\nHere we report a family of MOLPs constructed around the RhI precursor\\n[(h5-C5Me5)Rh(PMe3)2] (1) with a series of s, p and d-block metals, mostly from the main group elements, and investigate their bonding by computational means.\\nAmong the new MOLPs, we have structurally characterized those formed by dative bonding between 1 and MgMeBr,\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"Metal\\u2010only Lewis Pairs of Rhodium with  s ,  p  and  d \\u2010Block Metals\",\"authors\":\"Bajo Sonia Dr., Alf\\u00e9rez Macarena G., Alcaide Mar\\u00eda M., L\\u00f3pez\\u2010Serrano Joaqu\\u00edn Dr., Campos Jes\\u00fas Dr.\",\"abstract\":\"Abstract Metal\\u2010only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M\\u2192M bond represent privileged architectures to acquire fundamental understanding of bimetallic bonding. This has important implications in many catalytic processes or supramolecular systems that rely on synergistic effects between two metals. However, a systematic experimental/computational approach on a well\\u2010defined class of compounds is lacking. Here we report a family of MOLPs constructed around the Rh I  precursor [( \\u03b7 \\n 5 \\u2010C 5 Me 5 )Rh(PMe 3 ) 2 ] ( 1 ) with a series of  s ,  p  and  d \\u2010block metals, mostly from the main group elements, and investigate their bonding by computational means. Among the new MOLPs, we have structurally characterized those formed by dative bonding between  1  and MgMeBr, AlMe 3 , GeCl 2 , SnCl 2 , ZnMe 2  and Zn(C 6 F 5 ) 2,  as well as spectroscopically identified the ones resulting from coordination to MBAr F  (M=Na, Li; BAr F \\n \\u2212 =[B(C 6 H 2 \\u20103,5\\u2010(CF 3 ) 2 ) 4 ] \\u2212 ) and CuCl. Some of these compounds represent unique examples of bimetallic structures, such as the first unambiguous cases of Rh\\u2192Mg dative bonding or base\\u2010free rhodium bound germylene and stannylene species. Multinuclear NMR spectroscopy, including  103 Rh NMR, is used to probe the formation of Rh\\u2192M bonds. A comprehensive theoretical analysis of those provides clear trends. As anticipated, greater bond covalency is found for the more electronegative acids, whereas ionic character dominates for the least electronegative nuclei, though some degree of electron sharing is identified in all cases.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "10",
         "10.1002_cssc.202201821",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_cssc.202201821.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_cssc.202201821.xml",
         "test",
         null,
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u03041, Matthias, Adri C. T. van, Daniel Gaissmaier,[a\",\"abstract\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium-Ion Batteries** Florian Fiesinger,[a]\\nDaniel Gaissmaier,[a, b, c]\\nMatthias van den Borg,[a] Julian Be\\u00dfner,[a]\\nAdri C. T. van Duin,[d] and Timo Jacob*[a, b, c] One of the key challenges preventing the breakthrough of magnesium-ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode.\\nTo describe the initial steps of Mg anode degradation by O2 impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O2 immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(101\\u03040)A, and Mg(101\\u03041)), heating the surface to several 1000 K.\\nThe high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries.\\nAmong the Mg surface terminations, Mg(101\\u03040)A is the most reactive, forming an MgO layer with a thickness of up to 25 \\u00c5. The trained force field can be used to model the ongoing reactions in Mg-air batteries but also to study the oxidation of magnesium metal in general.\\nIntroduction The search for new energy storage technologies to promote electromobility and reduce global warming is one of the greatest challenges of our time.\\nMagnesium-ion batteries (MIB) are attracting increasing attention as an alternative to conven-tional lithium-ion batteries (LIB) due to the unique properties of magnesium metal anodes, such as high volumetric capacity, low cost, and safe operation.[1-5] However, mass market adoption of MIBs is currently hindered by significant challenges.\\nOne of the biggest challenges is the formation of a passivating interface on the Mg surface which is impermeable to Mg cations and leads to battery failure.[6-8] Sun et al. have identified three reasons for Mg degradation in a recent review article.[9] First, Mg has a higher charge density due to its bi-valency, which leads not only to stronger coordination with solvent molecules and salt anions but also facilitates solvent decom-position at the anode.[10] For example, Lu et al. have shown that the salt anion ClO4 is chemically reduced at the Mg anode in carbonate-containing solvents, forming a solid-electrolyte inter-face (SEI) consisting of Mg-halides, MgO, Mg(OH)2, MgCO3, and Mg(ROCO2)2.[11]\\nSecond, short circuits are induced by non-uniform Mg growth, which can be further subdivided into hemispherical,[12] porous,[13] or dendritic[14] growth morphologies.[9] Hemispherical 3D particles with different shapes have been detected in TFSI-based electrolytes as a function of the applied current density.[12]\\nIn an Mg(TFSI)2 salt electrolyte with MgCl2 as additive, continuous stripping and plating leads to fracturing and reformation of the SE\",\"data_availability\":\"Data Availability Statement\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5281/zenodo.7074790.'], citation_context: \\\"The data that support the findings of this study are openly available in zenodo at https://doi.org/10.5281/zenodo.7074790.\\nFurthermore, the KVIK optimization routine is openly available in the GitHub repository at https://github.com/shk11/KVIKOp-timizer.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/cssc.202201821'], citation_context: \\\"1864564x, 2023, 3, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/cssc.202201821 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium\\u2010Ion Batteries **\",\"authors\":\"Fiesinger Florian, Gaissmaier Daniel Dr., van den Borg Matthias, Be\\u00dfner Julian, van Duin Adri C. T. Prof.\\u2005Dr., Jacob Timo Prof.\\u2005Dr.\",\"abstract\":\"Abstract One of the key challenges preventing the breakthrough of magnesium\\u2010ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode. To describe the initial steps of Mg anode degradation by O 2  impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O 2  immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(10 1 \\u203e \\n0)A, and Mg(10 1 \\u203e \\n1)), heating the surface to several 1000\\u2005K. The high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries. Among the Mg surface terminations, Mg(10 1 \\u203e \\n0)A is the most reactive, forming an MgO layer with a thickness of up to 25\\u2005\\u00c5. The trained force field can be used to model the ongoing reactions in Mg\\u2013air batteries but also to study the oxidation of magnesium metal in general.\",\"data_availability\":\"Data Availability Statement The data that support the findings of this study are openly available in zenodo at  https://doi.org/10.5281/zenodo.7074790 . Furthermore, the  KVIK  optimization routine is openly available in the GitHub repository at  https://github.com/shk11/KVIKOptimizer .\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u03041, Matthias, Adri C. T. van, Daniel Gaissmaier,[a\",\"abstract\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium-Ion Batteries** Florian Fiesinger,[a]\\nDaniel Gaissmaier,[a, b, c]\\nMatthias van den Borg,[a] Julian Be\\u00dfner,[a]\\nAdri C. T. van Duin,[d] and Timo Jacob*[a, b, c] One of the key challenges preventing the breakthrough of magnesium-ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode.\\nTo describe the initial steps of Mg anode degradation by O2 impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O2 immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(101\\u03040)A, and Mg(101\\u03041)), heating the surface to several 1000 K.\\nThe high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries.\\nAmong the Mg surface terminations, Mg(101\\u03040)A is the most reactive, forming an MgO layer with a thickness of up to 25 \\u00c5. The trained force field can be used to model the ongoing reactions in Mg-air batteries but also to study the oxidation of magnesium metal in general.\\nIntroduction The search for new energy storage technologies to promote electromobility and reduce global warming is one of the greatest challenges of our time.\\nMagnesium-ion batteries (MIB) are attracting increasing attention as an alternative to conven-tional lithium-ion batteries (LIB) due to the unique properties of magnesium metal anodes, such as high volumetric capacity, low cost, and safe operation.[1-5] However, mass market adoption of MIBs is currently hindered by significant challenges.\\nOne of the biggest challenges is the formation of a passivating interface on the Mg surface which is impermeable to Mg cations and leads to battery failure.[6-8] Sun et al. have identified three reasons for Mg degradation in a recent review article.[9] First, Mg has a higher charge density due to its bi-valency, which leads not only to stronger coordination with solvent molecules and salt anions but also facilitates solvent decom-position at the anode.[10] For example, Lu et al. have shown that the salt anion ClO4 is chemically reduced at the Mg anode in carbonate-containing solvents, forming a solid-electrolyte inter-face (SEI) consisting of Mg-halides, MgO, Mg(OH)2, MgCO3, and Mg(ROCO2)2.[11]\\nSecond, short circuits are induced by non-uniform Mg growth, which can be further subdivided into hemispherical,[12] porous,[13] or dendritic[14] growth morphologies.[9] Hemispherical 3D particles with different shapes have been detected in TFSI-based electrolytes as a function of the applied current density.[12]\\nIn an Mg(TFSI)2 salt electrolyte with MgCl2 as additive, continuous stripping and plating leads to fracturing and reformation of the SE\",\"data_availability\":\"Data Availability Statement\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5281/zenodo.7074790.'], citation_context: \\\"The data that support the findings of this study are openly available in zenodo at https://doi.org/10.5281/zenodo.7074790.\\nFurthermore, the KVIK optimization routine is openly available in the GitHub repository at https://github.com/shk11/KVIKOp-timizer.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/cssc.202201821'], citation_context: \\\"1864564x, 2023, 3, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/cssc.202201821 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium\\u2010Ion Batteries **\",\"authors\":\"Fiesinger Florian, Gaissmaier Daniel Dr., van den Borg Matthias, Be\\u00dfner Julian, van Duin Adri C. T. Prof.\\u2005Dr., Jacob Timo Prof.\\u2005Dr.\",\"abstract\":\"Abstract One of the key challenges preventing the breakthrough of magnesium\\u2010ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode. To describe the initial steps of Mg anode degradation by O 2  impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O 2  immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(10 1 \\u203e \\n0)A, and Mg(10 1 \\u203e \\n1)), heating the surface to several 1000\\u2005K. The high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries. Among the Mg surface terminations, Mg(10 1 \\u203e \\n0)A is the most reactive, forming an MgO layer with a thickness of up to 25\\u2005\\u00c5. The trained force field can be used to model the ongoing reactions in Mg\\u2013air batteries but also to study the oxidation of magnesium metal in general.\",\"data_availability\":\"Data Availability Statement The data that support the findings of this study are openly available in zenodo at  https://doi.org/10.5281/zenodo.7074790 . Furthermore, the  KVIK  optimization routine is openly available in the GitHub repository at  https://github.com/shk11/KVIKOptimizer .\",\"other_dataset_citations\":[]}"
        ],
        [
         "11",
         "10.1002_ece3.3985",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.3985.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.3985.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Jonathan P. Evans, Watson, Revised, Evolution, Jason How2, Winn Jason, Jonathan P. Evans1, Simon\",\"abstract\":\"Ecology and Evolution.\\n2018;8:4525-4533.  \\n\\ufeff\\b   |  4525 www.ecolevol.org 1 | INTRODUCTION Polyandry, where females mate with two or more males, is tax-onomically widespread (Birkhead & M\\u00f8ller, 1998) despite poten-tial costs to females, such as wasted time and energy (Watson,  Arnqvist, & Stallmann, 1998), increased predation risk (Rowe, 1994),  exposure to disease (Thrall, Antonovics, & Dobson, 2000), and risk  of injuries (Crudgington & Siva-\\u00adJothy, 2000).\\nThe maintenance of  polyandry has been attributed to material (direct) and genetic (in-direct) benefits for females.\\nDirect benefits include fertility assur-ance, provision of resources and parental care for the offspring  (Sheldon 1994).\\nIndirect benefits include genetic incompatibility  avoidance, increased genetic diversity of offspring, and the en-hanced survival and reproductive success of offspring (Jennions  & Petrie, 2000; Neff & Pitcher, 2005; Tregenza & Wedell, 2000).  \\nPolyandry may also occur in the absence of benefits to females in    Received: 21 January 2018 | Revised: 7 February 2018 | Accepted: 9 February 2018 DOI: 10.1002/ece3.3985 O R\\nI G I N A L  R E S E A R C H High levels of polyandry, but limited evidence for multiple  paternity, in wild populations of the western rock lobster  (Panulirus cygnus)\\nJacqueline Loo1 | Winn Jason Kennington1 | Simon de Lestang2 | Jason How2 |   Jonathan P. Evans1\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Centre for Evolutionary Biology, School  of Biological Sciences, The University of  Western Australia, Crawley, WA, Australia 2Western Australian Fisheries and Marine  Research Lab\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.3985'], citation_context: \\\"20457758, 2018, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.3985 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/j.1469-185X.1970.tb01176.x'], citation_context: \\\"nce of polyandry.\\nBehavioral  Ecology and Sociobiology, 70, 73-81. https://doi.org/10.1007/s00265-015-2026-6\\nParker, G. A. (1970).\\nSperm competition and its evolutionary conse-quences in the insects.\\nBiological Reviews, 45, 525-567.\\nhttps://doi.\\norg/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288-295.  \\nhttps://doi.\\norg/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.) (20\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1471-8286.2005.01155.x'], citation_context: \\\"67.\\nhttps://doi.\\norg/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288-295.  \\nhttps://doi.\\norg/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.) (2013).\\nLobsters: Biology, management, aquaculture and  fisheries.\\nWest Sussex, UK: Wiley-Blackwell.\\nPhillips, B. F., Cobb, J. S., & George, R. W. (2012).\\nGeneral biology.\\\"}\"]}",
         "{\"title\":\"High levels of polyandry, but limited evidence for multiple paternity, in wild populations of the western rock lobster ( Panulirus cygnus )\",\"authors\":\"Loo Jacqueline, Kennington Winn Jason, de Lestang Simon, How Jason, Evans Jonathan P.\",\"abstract\":\"Abstract Polyandry, where multiple mating by females results in the temporal and spatial overlap of ejaculates from two or more males, is taxonomically widespread and occurs in varying frequencies within and among species. In decapods (crabs, lobsters, crayfish, and prawns), rates of polyandry are likely to be variable, but the extent to which patterns of multiple paternity reflect multiple mating, and thus are shaped by postmating processes that bias fertilization toward one or a subset of mated males, is unclear. Here, we use microsatellite markers to examine the frequency of multiple mating (the presence of spermatophores from two or more males) and patterns of paternity in wild populations of western rock lobster ( Panulirus cygnus ). Our data confirm that >45% of females had attached spermatophores arising from at least two males (i.e., confirming polyandry), but we found very limited evidence for multiple paternity; among 24 clutches sampled in this study, only two arose from fertilizations by two or more males. Single inferred paternal genotypes accounted for all remaining progeny genotypes in each clutch, including several instances when the mother had been shown to mate with two or more males. These findings highlight the need for further work to understand whether polyandry is adaptive and to uncover the mechanisms underlying postmating paternity biases in this system.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Jonathan P. Evans, Watson, Revised, Evolution, Jason How2, Winn Jason, Jonathan P. Evans1, Simon\",\"abstract\":\"Ecology and Evolution.\\n2018;8:4525-4533.  \\n\\ufeff\\b   |  4525 www.ecolevol.org 1 | INTRODUCTION Polyandry, where females mate with two or more males, is tax-onomically widespread (Birkhead & M\\u00f8ller, 1998) despite poten-tial costs to females, such as wasted time and energy (Watson,  Arnqvist, & Stallmann, 1998), increased predation risk (Rowe, 1994),  exposure to disease (Thrall, Antonovics, & Dobson, 2000), and risk  of injuries (Crudgington & Siva-\\u00adJothy, 2000).\\nThe maintenance of  polyandry has been attributed to material (direct) and genetic (in-direct) benefits for females.\\nDirect benefits include fertility assur-ance, provision of resources and parental care for the offspring  (Sheldon 1994).\\nIndirect benefits include genetic incompatibility  avoidance, increased genetic diversity of offspring, and the en-hanced survival and reproductive success of offspring (Jennions  & Petrie, 2000; Neff & Pitcher, 2005; Tregenza & Wedell, 2000).  \\nPolyandry may also occur in the absence of benefits to females in    Received: 21 January 2018 | Revised: 7 February 2018 | Accepted: 9 February 2018 DOI: 10.1002/ece3.3985 O R\\nI G I N A L  R E S E A R C H High levels of polyandry, but limited evidence for multiple  paternity, in wild populations of the western rock lobster  (Panulirus cygnus)\\nJacqueline Loo1 | Winn Jason Kennington1 | Simon de Lestang2 | Jason How2 |   Jonathan P. Evans1\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Centre for Evolutionary Biology, School  of Biological Sciences, The University of  Western Australia, Crawley, WA, Australia 2Western Australian Fisheries and Marine  Research Lab\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.3985'], citation_context: \\\"20457758, 2018, 9, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.3985 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/j.1469-185X.1970.tb01176.x'], citation_context: \\\"nce of polyandry.\\nBehavioral  Ecology and Sociobiology, 70, 73-81. https://doi.org/10.1007/s00265-015-2026-6\\nParker, G. A. (1970).\\nSperm competition and its evolutionary conse-quences in the insects.\\nBiological Reviews, 45, 525-567.\\nhttps://doi.\\norg/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288-295.  \\nhttps://doi.\\norg/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.) (20\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1471-8286.2005.01155.x'], citation_context: \\\"67.\\nhttps://doi.\\norg/10.1111/j.1469-185X.1970.tb01176.x Peakall, R., & Smouse, P. E. (2006).\\nGENALEX 6: Genetic analy-sis in Excel.\\nPopulation genetic software for teaching and re-search.  \\nMolecular  Ecology  Notes,  6,  288-295.  \\nhttps://doi.\\norg/10.1111/j.1471-8286.2005.01155.x Phillips, B. (Ed.) (2013).\\nLobsters: Biology, management, aquaculture and  fisheries.\\nWest Sussex, UK: Wiley-Blackwell.\\nPhillips, B. F., Cobb, J. S., & George, R. W. (2012).\\nGeneral biology.\\\"}\"]}\n{\"title\":\"High levels of polyandry, but limited evidence for multiple paternity, in wild populations of the western rock lobster ( Panulirus cygnus )\",\"authors\":\"Loo Jacqueline, Kennington Winn Jason, de Lestang Simon, How Jason, Evans Jonathan P.\",\"abstract\":\"Abstract Polyandry, where multiple mating by females results in the temporal and spatial overlap of ejaculates from two or more males, is taxonomically widespread and occurs in varying frequencies within and among species. In decapods (crabs, lobsters, crayfish, and prawns), rates of polyandry are likely to be variable, but the extent to which patterns of multiple paternity reflect multiple mating, and thus are shaped by postmating processes that bias fertilization toward one or a subset of mated males, is unclear. Here, we use microsatellite markers to examine the frequency of multiple mating (the presence of spermatophores from two or more males) and patterns of paternity in wild populations of western rock lobster ( Panulirus cygnus ). Our data confirm that >45% of females had attached spermatophores arising from at least two males (i.e., confirming polyandry), but we found very limited evidence for multiple paternity; among 24 clutches sampled in this study, only two arose from fertilizations by two or more males. Single inferred paternal genotypes accounted for all remaining progeny genotypes in each clutch, including several instances when the mother had been shown to mate with two or more males. These findings highlight the need for further work to understand whether polyandry is adaptive and to uncover the mechanisms underlying postmating paternity biases in this system.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "12",
         "10.1002_ece3.4466",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.4466.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.4466.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.r6nq870', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Mats Olsson, Revised, Sarre, Alexander Hansson, Evolution, Georges\",\"abstract\":\"Ecology and Evolution.\\n2018;8:9827-9833.  \\n\\ufeff\\b   |  9827 www.ecolevol.org\\n1 | INTRODUCTION\\nThe fundamental biological process of sex determination has been  under intense scientific research for decades, mainly because of its  profound implications on the phenotype of individuals and the for-mation of sex ratios in natural populations (West, Reece, & Sheldon,  2002).\\nIt is therefore a process of major evolutionary significance.  \\nThe mechanisms of sex determination have, for a long time, been  classified discretely as either genetic (GSD) or environmental (ESD).  \\nHowever, researchers are now in agreement that genotypic and en-vironmental sex determination does not necessarily constitute a di-chotomy, but rather two extremes of a continuum (Sarre, Georges, &  Quinn, 2004).\\nSex allocation studies on birds and mammals are numerous  (reviewed in Wapstra et al., 2007) and show that the majority of  these species have genetic sex determination.\\nReptiles (squamates    Received: 11 August 2017 | Revised: 13 June 2018 | Accepted: 6 July 2018 DOI: 10.1002/ece3.4466 O R\\nI G I N A L  R E S E A R C H Incubation temperature and parental identity determine sex in  the Australian agamid lizard Ctenophorus pictus Alexander Hansson  | Mats Olsson\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Department of Biological and Environmental  Sciences, University of Gothenburg,  Gothenburg, Sweden Correspondence Alexander Hansson, Department of  Biological and Environmental Sciences,  University of Gothenburg, Gothenburg,  Sweden.\\nEmail: ha\",\"data_availability\":\"DATA ACCESSIBILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.4466'], citation_context: \\\"20457758, 2018, 19, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.4466, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Incubation temperature and parental identity determine sex in the Australian agamid lizard  Ctenophorus pictus\",\"authors\":\"Hansson Alexander, Olsson Mats\",\"abstract\":\"Abstract Sex determination in Australian agamid lizards shows a complex framework of different mechanisms, varying even among closely related taxa. It is clear that discrete classification of these species as either having genetic ( GSD ) or environmental sex determination ( ESD ) does not agree with empirical data. Although many species in this group show temperature\\u2010dependent sex determination ( TSD ), recent evidence suggests additional genetic or epigenetic effects. A proposed model explaining the adaptive significance and evolution of  TSD  in short\\u2010lived agamids predicts that selection will favor temperature\\u2010biased sex ratios in species with intense male\\u2010male competition. Here, we use experimental incubation at (near) constant temperatures to test whether the sex of Australian painted dragons ( Ctenophorus pictus ) is influenced by temperature, building on previous research yet to have reached an agreement regarding the role of temperature in this species. In this study, incubation temperature and parental identity affected hatchling sex suggesting that environment and genetics may work in concert to determine sex in this species. Although our results are consistent with  TSD , our data cannot rule out a temperature\\u2010by\\u2010sex effect on egg or hatchling mortality. However, our findings together with the observed differences of sex determination systems in closely related species within this genus may provide novel opportunities to address fundamental questions in the evolution of sex determination systems.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Mats Olsson, Revised, Sarre, Alexander Hansson, Evolution, Georges\",\"abstract\":\"Ecology and Evolution.\\n2018;8:9827-9833.  \\n\\ufeff\\b   |  9827 www.ecolevol.org\\n1 | INTRODUCTION\\nThe fundamental biological process of sex determination has been  under intense scientific research for decades, mainly because of its  profound implications on the phenotype of individuals and the for-mation of sex ratios in natural populations (West, Reece, & Sheldon,  2002).\\nIt is therefore a process of major evolutionary significance.  \\nThe mechanisms of sex determination have, for a long time, been  classified discretely as either genetic (GSD) or environmental (ESD).  \\nHowever, researchers are now in agreement that genotypic and en-vironmental sex determination does not necessarily constitute a di-chotomy, but rather two extremes of a continuum (Sarre, Georges, &  Quinn, 2004).\\nSex allocation studies on birds and mammals are numerous  (reviewed in Wapstra et al., 2007) and show that the majority of  these species have genetic sex determination.\\nReptiles (squamates    Received: 11 August 2017 | Revised: 13 June 2018 | Accepted: 6 July 2018 DOI: 10.1002/ece3.4466 O R\\nI G I N A L  R E S E A R C H Incubation temperature and parental identity determine sex in  the Australian agamid lizard Ctenophorus pictus Alexander Hansson  | Mats Olsson\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2018 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Department of Biological and Environmental  Sciences, University of Gothenburg,  Gothenburg, Sweden Correspondence Alexander Hansson, Department of  Biological and Environmental Sciences,  University of Gothenburg, Gothenburg,  Sweden.\\nEmail: ha\",\"data_availability\":\"DATA ACCESSIBILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.4466'], citation_context: \\\"20457758, 2018, 19, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.4466, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Incubation temperature and parental identity determine sex in the Australian agamid lizard  Ctenophorus pictus\",\"authors\":\"Hansson Alexander, Olsson Mats\",\"abstract\":\"Abstract Sex determination in Australian agamid lizards shows a complex framework of different mechanisms, varying even among closely related taxa. It is clear that discrete classification of these species as either having genetic ( GSD ) or environmental sex determination ( ESD ) does not agree with empirical data. Although many species in this group show temperature\\u2010dependent sex determination ( TSD ), recent evidence suggests additional genetic or epigenetic effects. A proposed model explaining the adaptive significance and evolution of  TSD  in short\\u2010lived agamids predicts that selection will favor temperature\\u2010biased sex ratios in species with intense male\\u2010male competition. Here, we use experimental incubation at (near) constant temperatures to test whether the sex of Australian painted dragons ( Ctenophorus pictus ) is influenced by temperature, building on previous research yet to have reached an agreement regarding the role of temperature in this species. In this study, incubation temperature and parental identity affected hatchling sex suggesting that environment and genetics may work in concert to determine sex in this species. Although our results are consistent with  TSD , our data cannot rule out a temperature\\u2010by\\u2010sex effect on egg or hatchling mortality. However, our findings together with the observed differences of sex determination systems in closely related species within this genus may provide novel opportunities to address fundamental questions in the evolution of sex determination systems.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "13",
         "10.1002_ece3.5260",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5260.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, John S. Sproul4,5, Grey T. Gustafson,2, David R. Maddison4\\n |, Dunedin, Evolution, Hird, Hosner, Streicher, Alana Alexander2,3\\n |, Alexander et al., Sorenson, Mamanova, Lawrence, Grey T. Gustafson, James M. Pflug4, Sanger, Crawford et al.\",\"abstract\":\"Ecology and Evolution.\\n2019;9:6933-6948.  \\n\\ufeff\\b   |  6933 www.ecolevol.org\\n1 | INTRODUCTION\\nThe phylogenomics revolution is underway with targeted capture  and enrichment approaches (Mamanova et al., 2010; McCormack,  Hird, Zellmer, Carstens, & Brumfield, 2013) proving an effective  and improved method for phylogenetic studies relative to multilo-cus Sanger techniques (Blaimer et al., 2015).\\nAmong the genomic  components that have been targeted, ultraconserved elements  (UCEs sensu Faircloth et al., 2012) have demonstrated their util-ity in reconstructing phylogenies across diverse vertebrate\\n[e.g.,  fishes (Faircloth, Sorenson, Santini, & Alfaro, 2013), birds (Hosner,  Tobias, Braun, & Kimball, 2017; McCormack et al., 2013), reptiles  (Crawford et al., 2012; Grismer et al., 2016; Streicher & Wiens,  2017), amphibians (Alexander et al., 2017; Streicher et al., 2018),    Received: 8 April 2019 | Revised: 26 April 2019 | Accepted: 28 April 2019 DOI: 10.1002/ece3.5260   O R\\nI G I N A L  R E S E A R C H Ultraconserved element (UCE) probe set design: Base genome  and initial design parameters critical for optimization Grey T. Gustafson1,2  |   Alana Alexander2,3  |   John S. Sproul4,5  |   James M. Pflug4 |   David R. Maddison4  |   Andrew E. Z. Short1,2\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2019\\nThe Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Ecology and Evolutionary  Biology, University of Kansas, Lawrence,  Kansas 2Biodiversity Institute, University of Kansas,  Lawrence, Kansas 3Department of Anatomy, School of  Biomedical Sciences, University of Otago,  Dunedin, New Zealand 4Department of Integrative Biology, Oregon  State University, Corvallis, Oregon 5Department of Biology, Univ\",\"data_availability\":\"DATA AVAILABILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.5260'], citation_context: \\\"20457758, 2019, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5260, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Ultraconserved element (UCE) probe set design: Base genome and initial design parameters critical for optimization\",\"authors\":\"Gustafson Grey T., Alexander Alana, Sproul John S., Pflug James M., Maddison David R., Short Andrew E. Z.\",\"abstract\":\"Abstract Targeted capture and enrichment approaches have proven effective for phylogenetic study. Ultraconserved elements (UCEs) in particular have exhibited great utility for phylogenomic analyses, with the software package  phyluce  being among the most utilized pipelines for UCE phylogenomics, including probe design. Despite the success of UCEs, it is becoming increasing apparent that diverse lineages require probe sets tailored to focal taxa in order to improve locus recovery. However, factors affecting probe design and methods for optimizing probe sets to focal taxa remain underexplored. Here, we use newly available beetle (Coleoptera) genomic resources to investigate factors affecting UCE probe set design using  phyluce . In particular, we explore the effects of stringency during initial design steps, as well as base genome choice on resulting probe sets and locus recovery. We found that both base genome choice and initial bait design stringency parameters greatly alter the number of resultant probes included in final probe sets and strongly affect the number of loci detected and recovered during in silico testing of these probe sets. In addition, we identify attributes of base genomes that correlated with high performance in probe design. Ultimately, we provide a recommended workflow for using  phyluce  to design an optimized UCE probe set that will work across a targeted lineage, and use our findings to develop a new, open\\u2010source UCE probe set for beetles of the suborder Adephaga.\",\"data_availability\":\"DATA AVAILABILITY DNA sequences: GenBank MK838494 to MK838511;  phyluce  BED/BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad  https://doi.org/10.5061/dryad.2f62927 ; Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fastas available from;  https://github.com/laninsky/comparing_monolithic_UCE_fastas ; Alexander 2018b: Phase_target available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target ; Alexander 2018c: distance_calcs available from:  https://github.com/laninsky/distance_calcs ; Alexander 2018d: Phase_everyone available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone .\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, John S. Sproul4,5, Grey T. Gustafson,2, David R. Maddison4\\n |, Dunedin, Evolution, Hird, Hosner, Streicher, Alana Alexander2,3\\n |, Alexander et al., Sorenson, Mamanova, Lawrence, Grey T. Gustafson, James M. Pflug4, Sanger, Crawford et al.\",\"abstract\":\"Ecology and Evolution.\\n2019;9:6933-6948.  \\n\\ufeff\\b   |  6933 www.ecolevol.org\\n1 | INTRODUCTION\\nThe phylogenomics revolution is underway with targeted capture  and enrichment approaches (Mamanova et al., 2010; McCormack,  Hird, Zellmer, Carstens, & Brumfield, 2013) proving an effective  and improved method for phylogenetic studies relative to multilo-cus Sanger techniques (Blaimer et al., 2015).\\nAmong the genomic  components that have been targeted, ultraconserved elements  (UCEs sensu Faircloth et al., 2012) have demonstrated their util-ity in reconstructing phylogenies across diverse vertebrate\\n[e.g.,  fishes (Faircloth, Sorenson, Santini, & Alfaro, 2013), birds (Hosner,  Tobias, Braun, & Kimball, 2017; McCormack et al., 2013), reptiles  (Crawford et al., 2012; Grismer et al., 2016; Streicher & Wiens,  2017), amphibians (Alexander et al., 2017; Streicher et al., 2018),    Received: 8 April 2019 | Revised: 26 April 2019 | Accepted: 28 April 2019 DOI: 10.1002/ece3.5260   O R\\nI G I N A L  R E S E A R C H Ultraconserved element (UCE) probe set design: Base genome  and initial design parameters critical for optimization Grey T. Gustafson1,2  |   Alana Alexander2,3  |   John S. Sproul4,5  |   James M. Pflug4 |   David R. Maddison4  |   Andrew E. Z. Short1,2\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2019\\nThe Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Ecology and Evolutionary  Biology, University of Kansas, Lawrence,  Kansas 2Biodiversity Institute, University of Kansas,  Lawrence, Kansas 3Department of Anatomy, School of  Biomedical Sciences, University of Otago,  Dunedin, New Zealand 4Department of Integrative Biology, Oregon  State University, Corvallis, Oregon 5Department of Biology, Univ\",\"data_availability\":\"DATA AVAILABILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.5260'], citation_context: \\\"20457758, 2019, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5260, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Ultraconserved element (UCE) probe set design: Base genome and initial design parameters critical for optimization\",\"authors\":\"Gustafson Grey T., Alexander Alana, Sproul John S., Pflug James M., Maddison David R., Short Andrew E. Z.\",\"abstract\":\"Abstract Targeted capture and enrichment approaches have proven effective for phylogenetic study. Ultraconserved elements (UCEs) in particular have exhibited great utility for phylogenomic analyses, with the software package  phyluce  being among the most utilized pipelines for UCE phylogenomics, including probe design. Despite the success of UCEs, it is becoming increasing apparent that diverse lineages require probe sets tailored to focal taxa in order to improve locus recovery. However, factors affecting probe design and methods for optimizing probe sets to focal taxa remain underexplored. Here, we use newly available beetle (Coleoptera) genomic resources to investigate factors affecting UCE probe set design using  phyluce . In particular, we explore the effects of stringency during initial design steps, as well as base genome choice on resulting probe sets and locus recovery. We found that both base genome choice and initial bait design stringency parameters greatly alter the number of resultant probes included in final probe sets and strongly affect the number of loci detected and recovered during in silico testing of these probe sets. In addition, we identify attributes of base genomes that correlated with high performance in probe design. Ultimately, we provide a recommended workflow for using  phyluce  to design an optimized UCE probe set that will work across a targeted lineage, and use our findings to develop a new, open\\u2010source UCE probe set for beetles of the suborder Adephaga.\",\"data_availability\":\"DATA AVAILABILITY DNA sequences: GenBank MK838494 to MK838511;  phyluce  BED/BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad  https://doi.org/10.5061/dryad.2f62927 ; Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fastas available from;  https://github.com/laninsky/comparing_monolithic_UCE_fastas ; Alexander 2018b: Phase_target available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target ; Alexander 2018c: distance_calcs available from:  https://github.com/laninsky/distance_calcs ; Alexander 2018d: Phase_everyone available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone .\",\"other_dataset_citations\":[]}"
        ],
        [
         "14",
         "10.1002_ece3.5395",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5395.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.5395.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Evolution, Andrea Flack, Chapman, Konstanz, Alonso, Wolfgang Fiedler,2, Yachang Cheng, Yachang Cheng,2, Lok, Andrea, Radolfzell\",\"abstract\":\"Ecology and Evolution.\\n2019;9:8945-8952.  \\n\\ufeff\\b   |  8945 www.ecolevol.org 1 | INTRODUCTION Migration is a ubiquitous phenomenon that has evolved as an ad-aptation to seasonally changing environments (Newton, 2008).  \\nBut within the same species, or even population, we often observe  tremendous variation in specific migration features (Chapman,  Br\\u00f6nmark, Nilsson, & Hansson, 2011;\\nFlack et al., 2016).\\nThis may  be the outcome of trade-offs between the costs and benefits of  different migration strategies, leading to differences in individual  fitness, and subsequently influencing demographic dynamics (Lok,  Overdijk, & Piersma, 2015; Palac\\u00edn, Alonso, Mart\\u00edn, & Alonso,  2017).\\nLife in the Anthropocene is challenging many ancient evo-lutionary adaptations including migration.\\nSelection pressures  change quickly, favoring individuals that are better adapted to  human-modified landscapes (Otto, 2018).\\nThese human-shaped  selection pressures are caused by hunting, poisoning, electrocu-tion, food subsidies or global habitat and climate change, and many  more (Oro, Genovart, Tavecchia, Fowler, & Mart\\u00ednez-Abra\\u00edn, 2013;    \\nReceived: 18 April 2019 | Revised: 28 May 2019 | Accepted: 4 June 2019 DOI: 10.1002/ece3.5395   O R\\nI G I N A L  R E S E A R C H \\u201cCloser-to-home\\u201d strategy benefits juvenile survival in a long-distance migratory bird Yachang Cheng1,2  |   \\nWolfgang Fiedler1,2\\n|   Martin Wikelski1,2,3 |   \\nAndrea Flack1,2,3\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2019\\nThe Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Migration, Max Planck  Institute of Animal Behavior, Radolfzell,  Germany 2Department of Biology, University of  Konstanz, Konstanz, Germany 3Centre for the Advanced Study \",\"data_availability\":\"DATA ACCESSIBILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], citation_context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Southwest  Germany.\\nMovebank Data Repository, https://doi.org/10.5441/001/1.\\nck04mn78 Fiedler, W., Flack, A., Schmid, H., Reinhard, U., & Wikelski, M. (2019).\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], citation_context: \\\"(2019).\\nData from: LifeTrack White Stork Vorarlberg.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.71r7pp6q Flack, A., Fiedler, W., Blas, J., Pokrovsky, I., Kaatz, M., Mitropolsky, M.,\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], citation_context: \\\"Schmid, H., & Wikelski, M. (2019).\\nData from: LifeTrack White  Stork Rheinland-Pfalz.\\nMovebank Data Repository, https://doi.\\norg/10.5441/001/1.4192t2j4 Fiedler, W., Leppelsack, E., Leppelsack, H., Stahl, T., Wieding, O., &\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], citation_context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Bavaria.  \\nMovebank  Data  Repository,  https://doi.org/10.5441/001/1.\\nv1cs4nn0 Fiedler, W., Niederer, W., Sch\\u00f6nenberger, A., Flack, A., & Wikelski, M.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.5395'], citation_context: \\\"20457758, 2019, 16, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5395 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], citation_context: \\\"Data from: LifeTrack White Stork Oberschwaben.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.c42j3js7 Fiedler, W., Hilsendegen, C., Reis, C., Lehmann, J., Hilsendegen, P.,\\\"}\"]}",
         "{\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d strategy benefits juvenile survival in a long\\u2010distance migratory bird\",\"authors\":\"Cheng Yachang, Fiedler Wolfgang, Wikelski Martin, Flack Andrea\",\"abstract\":\"Abstract Human\\u2010induced changes in the climate and environment that occur at an unprecedented speed are challenging the existence of migratory species. Faced with these new challenges, species with diverse and flexible migratory behaviors may suffer less from population decline, as they may be better at responding to these changes by altering their migratory behavior. At the individual level, variations in migratory behavior may lead to differences in fitness and subsequently influence the population's demographic dynamics. Using lifetime GPS bio\\u2010logging data from 169 white storks ( Ciconia ciconia ), we explore whether the recently shortened migration distance of storks affects their survival during different stages of their juvenile life. We also explore how other variations in migratory decisions (i.e., time, destination), movement activity (measured using overall body dynamic acceleration), and early life conditions influence juvenile survival. We observed that their first autumn migration was the riskiest period for juvenile white storks. Individuals that migrated shorter distances and fledged earlier experienced lower mortality risks. In addition, higher movement activity and overwintering \\u201ccloser\\u2010to\\u2010home\\u201d (with 84.21% of the tracked individuals stayed Europe or North Africa) were associated with higher survival. Our study shows how avian migrants can change life history decisions over only a few decades, and thus it helps us to understand and predict how migrants respond to the rapidly changing world.\",\"data_availability\":\"DATA ACCESSIBILITY The data used for this study are available through the Movebank Data Repository ( https://www.movebank.org ): with  https://doi.org/10.5441/001/1.v1cs4nn0 ,  https://doi.org/10.5441/001/1.c42j3js7 ,  https://doi.org/10.5441/001/1.4192t2j4 ,  https://doi.org/10.5441/001/1.ck04mn78 ,  https://doi.org/10.5441/001/1.71r7pp6q  (Fiedler, Flack, Sch\\u00e4fle, et al.,  2019 ; Fiedler, Flack, Schmid, Reinhard, & Wikelski,  2019 ; Fiedler, Hilsendegen, et al.,  2019 ; Fiedler, Leppelsack, et al.,  2019 ; Fiedler, Niederer, Sch\\u00f6nenberger, Flack, & Wikelski,  2019 ).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.ck04mn78'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Sch\\u00e4fle ,  W. \\n ,  \\n Keeves ,  B. \\n ,  \\n Quetting ,  M. \\n ,  \\n Eid ,  B. \\n , \\u2026  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Southwest Germany .  Movebank Data Repository ,  10.5441/001/1.ck04mn78\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Schmid ,  H. \\n ,  \\n Reinhard ,  U. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Oberschwaben .  Movebank Data Repository ,  10.5441/001/1.c42j3js7\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Hilsendegen ,  C. \\n ,  \\n Reis ,  C. \\n ,  \\n Lehmann ,  J. \\n ,  \\n Hilsendegen ,  P. \\n ,  \\n Schmid ,  H. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Rheinland\\u2010Pfalz .  Movebank Data Repository ,  10.5441/001/1.4192t2j4\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.v1cs4nn0'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Leppelsack ,  E. \\n ,  \\n Leppelsack ,  H. \\n ,  \\n Stahl ,  T. \\n ,  \\n Wieding ,  O. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Bavaria .  Movebank Data Repository ,  10.5441/001/1.v1cs4nn0\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Niederer ,  W. \\n ,  \\n Sch\\u00f6nenberger ,  A. \\n ,  \\n Flack ,  A. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Vorarlberg .  Movebank Data Repository ,  10.5441/001/1.71r7pp6q\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Evolution, Andrea Flack, Chapman, Konstanz, Alonso, Wolfgang Fiedler,2, Yachang Cheng, Yachang Cheng,2, Lok, Andrea, Radolfzell\",\"abstract\":\"Ecology and Evolution.\\n2019;9:8945-8952.  \\n\\ufeff\\b   |  8945 www.ecolevol.org 1 | INTRODUCTION Migration is a ubiquitous phenomenon that has evolved as an ad-aptation to seasonally changing environments (Newton, 2008).  \\nBut within the same species, or even population, we often observe  tremendous variation in specific migration features (Chapman,  Br\\u00f6nmark, Nilsson, & Hansson, 2011;\\nFlack et al., 2016).\\nThis may  be the outcome of trade-offs between the costs and benefits of  different migration strategies, leading to differences in individual  fitness, and subsequently influencing demographic dynamics (Lok,  Overdijk, & Piersma, 2015; Palac\\u00edn, Alonso, Mart\\u00edn, & Alonso,  2017).\\nLife in the Anthropocene is challenging many ancient evo-lutionary adaptations including migration.\\nSelection pressures  change quickly, favoring individuals that are better adapted to  human-modified landscapes (Otto, 2018).\\nThese human-shaped  selection pressures are caused by hunting, poisoning, electrocu-tion, food subsidies or global habitat and climate change, and many  more (Oro, Genovart, Tavecchia, Fowler, & Mart\\u00ednez-Abra\\u00edn, 2013;    \\nReceived: 18 April 2019 | Revised: 28 May 2019 | Accepted: 4 June 2019 DOI: 10.1002/ece3.5395   O R\\nI G I N A L  R E S E A R C H \\u201cCloser-to-home\\u201d strategy benefits juvenile survival in a long-distance migratory bird Yachang Cheng1,2  |   \\nWolfgang Fiedler1,2\\n|   Martin Wikelski1,2,3 |   \\nAndrea Flack1,2,3\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2019\\nThe Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Department of Migration, Max Planck  Institute of Animal Behavior, Radolfzell,  Germany 2Department of Biology, University of  Konstanz, Konstanz, Germany 3Centre for the Advanced Study \",\"data_availability\":\"DATA ACCESSIBILITY\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], citation_context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Southwest  Germany.\\nMovebank Data Repository, https://doi.org/10.5441/001/1.\\nck04mn78 Fiedler, W., Flack, A., Schmid, H., Reinhard, U., & Wikelski, M. (2019).\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], citation_context: \\\"(2019).\\nData from: LifeTrack White Stork Vorarlberg.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.71r7pp6q Flack, A., Fiedler, W., Blas, J., Pokrovsky, I., Kaatz, M., Mitropolsky, M.,\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], citation_context: \\\"Schmid, H., & Wikelski, M. (2019).\\nData from: LifeTrack White  Stork Rheinland-Pfalz.\\nMovebank Data Repository, https://doi.\\norg/10.5441/001/1.4192t2j4 Fiedler, W., Leppelsack, E., Leppelsack, H., Stahl, T., Wieding, O., &\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.'], citation_context: \\\"Wikelski, M. (2019).\\nData from: LifeTrack White Stork Bavaria.  \\nMovebank  Data  Repository,  https://doi.org/10.5441/001/1.\\nv1cs4nn0 Fiedler, W., Niederer, W., Sch\\u00f6nenberger, A., Flack, A., & Wikelski, M.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.5395'], citation_context: \\\"20457758, 2019, 16, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.5395 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], citation_context: \\\"Data from: LifeTrack White Stork Oberschwaben.\\nMovebank Data  Repository, https://doi.org/10.5441/001/1.c42j3js7 Fiedler, W., Hilsendegen, C., Reis, C., Lehmann, J., Hilsendegen, P.,\\\"}\"]}\n{\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d strategy benefits juvenile survival in a long\\u2010distance migratory bird\",\"authors\":\"Cheng Yachang, Fiedler Wolfgang, Wikelski Martin, Flack Andrea\",\"abstract\":\"Abstract Human\\u2010induced changes in the climate and environment that occur at an unprecedented speed are challenging the existence of migratory species. Faced with these new challenges, species with diverse and flexible migratory behaviors may suffer less from population decline, as they may be better at responding to these changes by altering their migratory behavior. At the individual level, variations in migratory behavior may lead to differences in fitness and subsequently influence the population's demographic dynamics. Using lifetime GPS bio\\u2010logging data from 169 white storks ( Ciconia ciconia ), we explore whether the recently shortened migration distance of storks affects their survival during different stages of their juvenile life. We also explore how other variations in migratory decisions (i.e., time, destination), movement activity (measured using overall body dynamic acceleration), and early life conditions influence juvenile survival. We observed that their first autumn migration was the riskiest period for juvenile white storks. Individuals that migrated shorter distances and fledged earlier experienced lower mortality risks. In addition, higher movement activity and overwintering \\u201ccloser\\u2010to\\u2010home\\u201d (with 84.21% of the tracked individuals stayed Europe or North Africa) were associated with higher survival. Our study shows how avian migrants can change life history decisions over only a few decades, and thus it helps us to understand and predict how migrants respond to the rapidly changing world.\",\"data_availability\":\"DATA ACCESSIBILITY The data used for this study are available through the Movebank Data Repository ( https://www.movebank.org ): with  https://doi.org/10.5441/001/1.v1cs4nn0 ,  https://doi.org/10.5441/001/1.c42j3js7 ,  https://doi.org/10.5441/001/1.4192t2j4 ,  https://doi.org/10.5441/001/1.ck04mn78 ,  https://doi.org/10.5441/001/1.71r7pp6q  (Fiedler, Flack, Sch\\u00e4fle, et al.,  2019 ; Fiedler, Flack, Schmid, Reinhard, & Wikelski,  2019 ; Fiedler, Hilsendegen, et al.,  2019 ; Fiedler, Leppelsack, et al.,  2019 ; Fiedler, Niederer, Sch\\u00f6nenberger, Flack, & Wikelski,  2019 ).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.ck04mn78'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Sch\\u00e4fle ,  W. \\n ,  \\n Keeves ,  B. \\n ,  \\n Quetting ,  M. \\n ,  \\n Eid ,  B. \\n , \\u2026  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Southwest Germany .  Movebank Data Repository ,  10.5441/001/1.ck04mn78\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Schmid ,  H. \\n ,  \\n Reinhard ,  U. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Oberschwaben .  Movebank Data Repository ,  10.5441/001/1.c42j3js7\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Hilsendegen ,  C. \\n ,  \\n Reis ,  C. \\n ,  \\n Lehmann ,  J. \\n ,  \\n Hilsendegen ,  P. \\n ,  \\n Schmid ,  H. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Rheinland\\u2010Pfalz .  Movebank Data Repository ,  10.5441/001/1.4192t2j4\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.v1cs4nn0'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Leppelsack ,  E. \\n ,  \\n Leppelsack ,  H. \\n ,  \\n Stahl ,  T. \\n ,  \\n Wieding ,  O. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Bavaria .  Movebank Data Repository ,  10.5441/001/1.v1cs4nn0\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Niederer ,  W. \\n ,  \\n Sch\\u00f6nenberger ,  A. \\n ,  \\n Flack ,  A. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Vorarlberg .  Movebank Data Repository ,  10.5441/001/1.71r7pp6q\\\"}\"]}"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6144.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |, Ronald S. Burton, Evolution\",\"abstract\":\"Ecology and Evolution.\\n2020;10:3463-3476.  \\n\\ufeff\\b   |  3463 www.ecolevol.org 1 | INTRODUCTION Marine ecosystems can change rapidly in response to both natural  and anthropogenic changes in the environment.\\nMonitoring changes  in the abundance and distribution of marine organisms is a chal-lenging but critical component of conservation efforts and natural  resource management.\\nHuman impacts such as overfishing, coastal  development, and pollution can rapidly increase stress on ecosys-tems resulting in an urgent need to develop efficient methods for  monitoring spatial and temporal dynamics of biodiversity (Beng  et al., 2016).\\nFisheries management has long used ichthyoplankton sur-veys as a component of monitoring because sampling the early  life stages of fish reflects the reproductive activities of fish    Received: 17 July 2019 | Revised:\\n31 January 2020 |\\nAccepted: 4 February 2020 DOI: 10.1002/ece3.6144   O R\\nI G I N A L  R E S E A R C H Efficacy of metabarcoding for identification of fish eggs  evaluated with mock communities Elena M. Duke  |   Ronald S. Burton\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Marine Biology Research Division, Scripps  Institution of Oceanography, University of  California, San Diego, La Jolla, California Correspondence Elena M. Duke, Marine Biology  Research Division, Scripps Institution of  Oceanography, University of California, San  Diego, La Jolla, CA.\\nEmail: emduke@ucsd.edu\\nAbstract There is urgent need for effective and efficient monitoring of marine fish popula-tions.\\nMo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1371/journal.pone.0134647'], citation_context: \\\"ermsmeier, M. C., Rogowski, P. A., Terrill,  E., Burton, R. S., & Bernardi, G. (2015).\\nMonitoring spawning ac-tivity in a Southern California marine protected area using molec-ular identification of fish eggs.\\nPLoS ONE, 10(8), 1-21.\\nhttps://doi.\\norg/10.1371/journal.pone.0134647 Hastings, P. A., & Burton, R. S. (2008).\\nEstablishing a DNA Sequence  Database for the Marine Fish Fauna of California.\\nRetrieved from  https://escholarship.org/uc/item/1ck9b3qs Hatzenbuhler, C., Kelly, J. R., Martinson, J., Okum, S., & Pilgrim, E. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6144'], citation_context: \\\"20457758, 2020, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6144, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |, Ronald S. Burton, Evolution\",\"abstract\":\"Ecology and Evolution.\\n2020;10:3463-3476.  \\n\\ufeff\\b   |  3463 www.ecolevol.org 1 | INTRODUCTION Marine ecosystems can change rapidly in response to both natural  and anthropogenic changes in the environment.\\nMonitoring changes  in the abundance and distribution of marine organisms is a chal-lenging but critical component of conservation efforts and natural  resource management.\\nHuman impacts such as overfishing, coastal  development, and pollution can rapidly increase stress on ecosys-tems resulting in an urgent need to develop efficient methods for  monitoring spatial and temporal dynamics of biodiversity (Beng  et al., 2016).\\nFisheries management has long used ichthyoplankton sur-veys as a component of monitoring because sampling the early  life stages of fish reflects the reproductive activities of fish    Received: 17 July 2019 | Revised:\\n31 January 2020 |\\nAccepted: 4 February 2020 DOI: 10.1002/ece3.6144   O R\\nI G I N A L  R E S E A R C H Efficacy of metabarcoding for identification of fish eggs  evaluated with mock communities Elena M. Duke  |   Ronald S. Burton\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Marine Biology Research Division, Scripps  Institution of Oceanography, University of  California, San Diego, La Jolla, California Correspondence Elena M. Duke, Marine Biology  Research Division, Scripps Institution of  Oceanography, University of California, San  Diego, La Jolla, CA.\\nEmail: emduke@ucsd.edu\\nAbstract There is urgent need for effective and efficient monitoring of marine fish popula-tions.\\nMo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1371/journal.pone.0134647'], citation_context: \\\"ermsmeier, M. C., Rogowski, P. A., Terrill,  E., Burton, R. S., & Bernardi, G. (2015).\\nMonitoring spawning ac-tivity in a Southern California marine protected area using molec-ular identification of fish eggs.\\nPLoS ONE, 10(8), 1-21.\\nhttps://doi.\\norg/10.1371/journal.pone.0134647 Hastings, P. A., & Burton, R. S. (2008).\\nEstablishing a DNA Sequence  Database for the Marine Fish Fauna of California.\\nRetrieved from  https://escholarship.org/uc/item/1ck9b3qs Hatzenbuhler, C., Kelly, J. R., Martinson, J., Okum, S., & Pilgrim, E. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6144'], citation_context: \\\"20457758, 2020, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6144, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}"
        ],
        [
         "16",
         "10.1002_ece3.6303",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6303.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6303.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.37pvmcvgb', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. O. Oduor,3, Robin, Evolution, Luxi Chen1, Junmin Li, Beifen Yang1\",\"abstract\":\"Ecology and Evolution.\\n2020;10:5617-5630.  \\n\\ufeff\\b   |  5617 www.ecolevol.org   Received: 26 August 2019 | Revised: 25 March 2020 | Accepted: 30 March 2020 DOI: 10.1002/ece3.6303   O R\\nI G I N A L  R E S E A R C H Can polyploidy confer invasive plants with a wider climatic  tolerance?\\nA test using Solidago canadensis Jizhong Wan1,2\\n|   \\nAyub M. O. Oduor1,3  |   Robin Pouteau1  |   Beilei Wang1 |    Luxi Chen1 |   Beifen Yang1 |   Feihai Yu1\\n|   Junmin Li1\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Zhejing Provincial Key Laboratory of Plant  Evolutionary Ecology and Conservation,  Taizhou University, Taizhou, P.R. China 2State Key Laboratory of Plateau Ecology  and Agriculture, Qinghai University, Xining,  P.R. China 3Department of Applied Biology, Technical  University of Kenya, Nairobi, Kenya Correspondence Junmin Li, Zhejiang Provincial Key  Laboratory of Plant Evolutionary Ecology  and Conservation, Taizhou 318000, China.\\nEmail: lijmtzc@126.com\\nFunding information Ten Thousand Talent Program of Zhejiang  Province; National Natural Science  Foundation of China, Grant/Award Number:  31270461; National Key Research and  Development Program, Grant/Award  Number: 2016YFC1201100\\nAbstract Polyploidy can cause variation in plant functional traits and thereby generate indi-viduals that can adapt to fluctuating environments and exploit new environments.  \\nHowever, few empirical studies have tested for an association between ploidy level  and climatic tolerance of invasive cytotypes relative to conspecific native-range cy-totypes.\\nHere, we used an invasive plant Solidago canadensis to test whether invasiv\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.6303'], citation_context: \\\"20457758, 2020, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6303, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/cyto.a.20253'], citation_context: \\\"Suda, J., & Tr\\u00e1vn\\u00ed\\u010dek, P. (2006).\\nReliable DNA ploidy determination in  dehydrated tissues of vascular plants by DAPI flow cytometry-new  prospects for plant research.\\nCytometry, 69A, 273-280.\\nhttps://doi.\\norg/10.1002/cyto.a.20253 Tajima, F. (1989).\\nStatistical method for testing the neutral mutation hy-pothesis by DNA polymorphism.\\nGenetics, 123, 585-595.\\nTavares, D., Loureiro, J., Martins, A., Castro, M., Roiloa, S., & Castro, S.  (2019).\\nGenetically based phenotypic differ\\\"}\",\"{\\\"dataset_ids\\\": ['MN107151', 'MN335287', 'MN335290'], citation_context: \\\"MN107084-MN107151, MN335287-MN335290).\\nThe data that\\\"}\",\"{\\\"dataset_ids\\\": ['MN107084', 'MN107151'], citation_context: \\\"(NCBI) database (accession numbers: MN107084-MN107151,\\\"}\"]}",
         "{\"title\":\"Can polyploidy confer invasive plants with a wider climatic tolerance? A test using  Solidago canadensis\",\"authors\":\"Wan Jizhong, Oduor Ayub M. O., Pouteau Robin, Wang Beilei, Chen Luxi, Yang Beifen, Yu Feihai, Li Junmin\",\"abstract\":\"Abstract Polyploidy can cause variation in plant functional traits and thereby generate individuals that can adapt to fluctuating environments and exploit new environments. However, few empirical studies have tested for an association between ploidy level and climatic tolerance of invasive cytotypes relative to conspecific native\\u2010range cytotypes. Here, we used an invasive plant  Solidago canadensis  to test whether invasive populations had a higher proportion of polyploids, greater height and stem\\u2010base diameter, and occupied a wider range of climatic conditions than conspecific native\\u2010range populations. We also tested whether the invasive populations had overcome genetic founder effects. We sampled a total of 80 populations in parts of the invaded range in China and native range in North America for in situ measurements of plant height and stem\\u2010base diameter in the field and for population genetic and cytotype analyses. To examine climatic correlates, we augmented our field\\u2010sampled data with occurrence records obtained from Global Biodiversity Information Facility. All, except one, of the populations that we sampled in China occurred in a humid subtropical climate. In contrast, the North American populations occurred in humid continental, humid subtropical, and semi\\u2010arid climatic zones. All populations of  S. canadensis  in China were purely hexaploid, while the North American populations were diploid, tetraploid, and hexaploid. The invasive hexaploids were significantly taller and had a\\u00a0larger stem\\u2010base diameter than native hexaploids. Native hexaploids were significantly taller and had larger stem\\u2010base diameter than native diploids. Climatic correlate assessment found that invasive and native populations occupied different climatic envelopes, with invasive populations occurring in warmer and less seasonal climates than native populations. However, there was no significant correlation between ploidy level and climatic envelope of  S. canadensis . Molecular phylogeo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT DNA sequences have been deposited in the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084\\u2013MN107151, MN335287\\u2013MN335290). The data that support the findings of this study have been deposited in Dryad with doi: https://doi.org/10.5061/dryad.37pvmcvgb .\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. O. Oduor,3, Robin, Evolution, Luxi Chen1, Junmin Li, Beifen Yang1\",\"abstract\":\"Ecology and Evolution.\\n2020;10:5617-5630.  \\n\\ufeff\\b   |  5617 www.ecolevol.org   Received: 26 August 2019 | Revised: 25 March 2020 | Accepted: 30 March 2020 DOI: 10.1002/ece3.6303   O R\\nI G I N A L  R E S E A R C H Can polyploidy confer invasive plants with a wider climatic  tolerance?\\nA test using Solidago canadensis Jizhong Wan1,2\\n|   \\nAyub M. O. Oduor1,3  |   Robin Pouteau1  |   Beilei Wang1 |    Luxi Chen1 |   Beifen Yang1 |   Feihai Yu1\\n|   Junmin Li1\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1Zhejing Provincial Key Laboratory of Plant  Evolutionary Ecology and Conservation,  Taizhou University, Taizhou, P.R. China 2State Key Laboratory of Plateau Ecology  and Agriculture, Qinghai University, Xining,  P.R. China 3Department of Applied Biology, Technical  University of Kenya, Nairobi, Kenya Correspondence Junmin Li, Zhejiang Provincial Key  Laboratory of Plant Evolutionary Ecology  and Conservation, Taizhou 318000, China.\\nEmail: lijmtzc@126.com\\nFunding information Ten Thousand Talent Program of Zhejiang  Province; National Natural Science  Foundation of China, Grant/Award Number:  31270461; National Key Research and  Development Program, Grant/Award  Number: 2016YFC1201100\\nAbstract Polyploidy can cause variation in plant functional traits and thereby generate indi-viduals that can adapt to fluctuating environments and exploit new environments.  \\nHowever, few empirical studies have tested for an association between ploidy level  and climatic tolerance of invasive cytotypes relative to conspecific native-range cy-totypes.\\nHere, we used an invasive plant Solidago canadensis to test whether invasiv\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.6303'], citation_context: \\\"20457758, 2020, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6303, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/cyto.a.20253'], citation_context: \\\"Suda, J., & Tr\\u00e1vn\\u00ed\\u010dek, P. (2006).\\nReliable DNA ploidy determination in  dehydrated tissues of vascular plants by DAPI flow cytometry-new  prospects for plant research.\\nCytometry, 69A, 273-280.\\nhttps://doi.\\norg/10.1002/cyto.a.20253 Tajima, F. (1989).\\nStatistical method for testing the neutral mutation hy-pothesis by DNA polymorphism.\\nGenetics, 123, 585-595.\\nTavares, D., Loureiro, J., Martins, A., Castro, M., Roiloa, S., & Castro, S.  (2019).\\nGenetically based phenotypic differ\\\"}\",\"{\\\"dataset_ids\\\": ['MN107151', 'MN335287', 'MN335290'], citation_context: \\\"MN107084-MN107151, MN335287-MN335290).\\nThe data that\\\"}\",\"{\\\"dataset_ids\\\": ['MN107084', 'MN107151'], citation_context: \\\"(NCBI) database (accession numbers: MN107084-MN107151,\\\"}\"]}\n{\"title\":\"Can polyploidy confer invasive plants with a wider climatic tolerance? A test using  Solidago canadensis\",\"authors\":\"Wan Jizhong, Oduor Ayub M. O., Pouteau Robin, Wang Beilei, Chen Luxi, Yang Beifen, Yu Feihai, Li Junmin\",\"abstract\":\"Abstract Polyploidy can cause variation in plant functional traits and thereby generate individuals that can adapt to fluctuating environments and exploit new environments. However, few empirical studies have tested for an association between ploidy level and climatic tolerance of invasive cytotypes relative to conspecific native\\u2010range cytotypes. Here, we used an invasive plant  Solidago canadensis  to test whether invasive populations had a higher proportion of polyploids, greater height and stem\\u2010base diameter, and occupied a wider range of climatic conditions than conspecific native\\u2010range populations. We also tested whether the invasive populations had overcome genetic founder effects. We sampled a total of 80 populations in parts of the invaded range in China and native range in North America for in situ measurements of plant height and stem\\u2010base diameter in the field and for population genetic and cytotype analyses. To examine climatic correlates, we augmented our field\\u2010sampled data with occurrence records obtained from Global Biodiversity Information Facility. All, except one, of the populations that we sampled in China occurred in a humid subtropical climate. In contrast, the North American populations occurred in humid continental, humid subtropical, and semi\\u2010arid climatic zones. All populations of  S. canadensis  in China were purely hexaploid, while the North American populations were diploid, tetraploid, and hexaploid. The invasive hexaploids were significantly taller and had a\\u00a0larger stem\\u2010base diameter than native hexaploids. Native hexaploids were significantly taller and had larger stem\\u2010base diameter than native diploids. Climatic correlate assessment found that invasive and native populations occupied different climatic envelopes, with invasive populations occurring in warmer and less seasonal climates than native populations. However, there was no significant correlation between ploidy level and climatic envelope of  S. canadensis . Molecular phylogeo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT DNA sequences have been deposited in the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084\\u2013MN107151, MN335287\\u2013MN335290). The data that support the findings of this study have been deposited in Dryad with doi: https://doi.org/10.5061/dryad.37pvmcvgb .\",\"other_dataset_citations\":[]}"
        ],
        [
         "17",
         "10.1002_ece3.6784",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6784.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6784.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Thomas, Evolutionsforschung, Biologiedidaktik, Evolution, Lena Kath1, CeNak, Luther-King-Platz, Email, Helmholtz-\\nZentrum Geesthacht, Neuss, Alexander, Andr\\u00e9, Friedrich\",\"abstract\":\"Ecology and Evolution. 2020;10:11467-11487.  \\n\\ufeff\\b   |  11467 www.ecolevol.org   Received: 25 June 2020 | Revised: 25 August 2020 | Accepted: 27 August 2020 DOI: 10.1002/ece3.6784   O R\\nI G I N A L  R E S E A R C H Ecomorphology of the pectoral girdle in anurans (Amphibia,  Anura): Shape diversity and biomechanical considerations Karolin Engelkes1  |   Lena Kath1\\n|   Thomas Kleinteich2 |   J\\u00f6rg U. Hammel3,4  |    \\nAndr\\u00e9 Beerlink5 |   \\nAlexander Haas1 This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd 1Center of Natural History (CeNak),  Universit\\u00e4t Hamburg, Hamburg, Germany 2TPW Pr\\u00fcfzentrum GmbH, Neuss, Germany 3Institute of Materials Research, Helmholtz-Zentrum Geesthacht, Geesthacht, Germany 4Institut f\\u00fcr Zoologie und  Evolutionsforschung mit Phyletischem  Museum, Ernst-Hackel-Haus und  Biologiedidaktik, Friedrich-Schiller-Universit\\u00e4t Jena, Jena, Germany 5YXLON International GmbH, Hamburg,  Germany Correspondence Karolin Engelkes, Center of Natural History  (CeNak), Universit\\u00e4t Hamburg, Martin-Luther-King-Platz 3, 20146 Hamburg,  Germany.\\nEmail: karolin.engelkes@uni-hamburg.de Funding information Wilhelm-Peters-Fonds of the Deutsche  Gesellschaft f\\u00fcr Herpetologie und  Terrarienkunde e.V (DGHT); Deutsche  Forschungsgemeinschaft, Grant/Award  Number: 387723284 Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that  allow them to inhabit a wide range of habitats.\\nThe different locomotor modes are  likely to be linked to anatomical specializations of the skeleton within the typical frog  Bauplan.\\nWhile such anatomical adaptations of the hind li\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.6784'], citation_context: \\\"20457758, 2020, 20, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6784 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/jeb.12292'], citation_context: \\\" M., Byrne, P. G., Roberts, J. D., & Keogh, J. S. (2014).\\nThe role  of phylogeny and ecology in shaping morphology in 21 genera and 127  species of Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181-192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825\\nWainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright,\\\"},{\\\"dataset_ids\\\": ['10.2307/1444825'], citation_context: \\\"Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181-192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825\\nWainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright, & S. M. Reilly (Eds.), Ecological morphol-ogy: Integrative organismal biology (pp. 42-59).\\nChicago, LondonThe  University of Chicago Press.\\nWainw\\\"}\",\"{\\\"dataset_ids\\\": ['10.13140/RG.2.2.18389.86247'], citation_context: \\\"ge in the evolution of a novel locomotor  type: \\u201cflying\\u201d frogs.\\nEvolution, 44(8), 1931-1946.\\nEngelkes, K. (preprint, in review) Accuracy of bone segmentation and  surface generation strategies analyzed by using synthetic CT vol-umes.\\nhttps://doi.org/10.13140/RG.2.2.18389.86247 Engelkes, K., Friedrich, F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 2\\\"},{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], citation_context: \\\" F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 213-228.\\nhttps://doi.\\norg/10.1007/s00435-017-0386-3 Engelkes, K., Helfsgott, J., Hammel, J. U., B\\u00fcsse, S., Kleinteich, T., Beerlink,  A., \\u2026 Haas, A. (2019).\\nMeasurement error in \\u03bcCT-based three-dimen-sional geometric morphometrics introduced by surface generation  and landmark data acquisition.\\nJourn\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1550-7408.2008.00342.x'], citation_context: \\\" lineage  of closely related trypanosomes (Trypanosomatidae, Kinetoplastida)  of anurans and sand flies (Psychodidae, Diptera) sharing the same  ecotopes in brazilian amazonia.\\nThe Journal of Eukaryotic Microbiology,  55(5), 427-435. https://doi.org/10.1111/j.1550-7408.2008.00342.x\\nFrost, D. R. (2020) Amphibian Species of the World: an online reference.  \\nVersion 6.1.\\nhttp://research.amnh.org/herpetology/amphibia/index.\\nhtml.\\nF\\u00fcrbringer, M. (1873).\\nZur vergleichenden Anatomie der Schultermuskeln.  \\nTeil 1.\\nJenaische Zeitschrif\\\"}\",\"{\\\"dataset_ids\\\": ['10.1073/pnas.0800375105'], citation_context: \\\"ak, M., Kadmon, R., Saltz, D.,  & Smouse, P. E. (2008).\\nA movement ecology paradigm for unifying  organismal movement research.\\nProceedings of the National Academy  of Sciences of the United States of America, 105(49), 19052-19059.  \\nhttps://doi.org/10.1073/pnas.0800375105 Nauwelaerts, S., & Aerts, P. (2006).\\nTake-off and landing forces in jump-ing frogs.\\nThe Journal of Experimental Biology, 209(1), 66-77.\\nhttps://doi.org/10.1242/jeb.01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of a\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1469-7580.2007.00691.x'], citation_context: \\\".01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of aquatic and terrestrial locomotion in a semi-aquatic frog, Rana es-culenta: No evidence for a design conflict.\\nJournal of Anatomy, 210(3),  304-317.\\nhttps://doi.org/10.1111/j.1469-7580.2007.00691.x\\nNoble, G. K., & Jaeckle, M. E. (1928).\\nThe digital pads of the tree frogs.  \\nA study of the phylogenesis of an adaptive structure.\\nJournal of  Morphology, 45(1), 259-292.\\nNomura, F., Rossa-Feres, D. C., & Langeani, F. (2009).\\nBurrowing behav-ior of \\\"}\"]}",
         "{\"title\":\"Ecomorphology of the pectoral girdle in anurans (Amphibia, Anura): Shape diversity and biomechanical considerations\",\"authors\":\"Engelkes Karolin, Kath Lena, Kleinteich Thomas, Hammel J\\u00f6rg U., Beerlink Andr\\u00e9, Haas Alexander\",\"abstract\":\"Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that allow them to inhabit a wide range of habitats. The different locomotor modes are likely to be linked to anatomical specializations of the skeleton within the typical frog Bauplan. While such anatomical adaptations of the hind limbs and the pelvic girdle are comparably well understood, the pectoral girdle received much less attention in the past. We tested for locomotor\\u2010mode\\u2010related shape differences in the pectoral girdle bones of 64 anuran species by means of micro\\u2010computed\\u2010tomography\\u2010based geometric morphometrics. The pectoral girdles of selected species were analyzed with regard to the effects of shape differences on muscle moment arms across the shoulder joint and stress dissipation within the coracoid. Phylogenetic relationships, size, and locomotor behavior have an effect on the shape of the pectoral girdle in anurans, but there are differences in the relative impact of these factors between the bones of this skeletal unit. Remarkable shape diversity has been observed within locomotor groups indicating many\\u2010to\\u2010one mapping of form onto function. Significant shape differences have mainly been related to the overall pectoral girdle geometry and the shape of the coracoid. Most prominent shape differences have been found between burrowing and nonburrowing species with headfirst and backward burrowing species significantly differing from one another and from the other locomotor groups. The pectoral girdle shapes of burrowing species have generally larger moment arms for (simulated) humerus retractor muscles across the shoulder joint, which might be an adaptation to the burrowing behavior. The mechanisms of how the moment arms were enlarged differed between species and were associated with differences in the reaction of the coracoid to simulated loading by physiologically relevant forces.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT CT volumes can be downloaded from  https://www.fdr.uni\\u2010hamburg.de/search?page=1&size=20&q=keywords:%22pectoral%20girdle%20morphometrics%20project%22 ; DOI numbers are provided in Appendix S1: Table A1.\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], citation_context: \\\"Engelkes ,  K. \\n ,  \\n Friedrich ,  F. \\n ,  \\n Hammel ,  J. U. \\n , &  \\n Haas ,  A. \\n  ( 2018 ).  A simple setup for episcopic microtomy and a digital image processing workflow to acquire high\\u2010quality volume data and 3D surface models of small vertebrates .  Zoomorphology ,  137 ( 1 ),  213 \\u2013 228 .  10.1007/s00435-017-0386-3\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Thomas, Evolutionsforschung, Biologiedidaktik, Evolution, Lena Kath1, CeNak, Luther-King-Platz, Email, Helmholtz-\\nZentrum Geesthacht, Neuss, Alexander, Andr\\u00e9, Friedrich\",\"abstract\":\"Ecology and Evolution. 2020;10:11467-11487.  \\n\\ufeff\\b   |  11467 www.ecolevol.org   Received: 25 June 2020 | Revised: 25 August 2020 | Accepted: 27 August 2020 DOI: 10.1002/ece3.6784   O R\\nI G I N A L  R E S E A R C H Ecomorphology of the pectoral girdle in anurans (Amphibia,  Anura): Shape diversity and biomechanical considerations Karolin Engelkes1  |   Lena Kath1\\n|   Thomas Kleinteich2 |   J\\u00f6rg U. Hammel3,4  |    \\nAndr\\u00e9 Beerlink5 |   \\nAlexander Haas1 This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd 1Center of Natural History (CeNak),  Universit\\u00e4t Hamburg, Hamburg, Germany 2TPW Pr\\u00fcfzentrum GmbH, Neuss, Germany 3Institute of Materials Research, Helmholtz-Zentrum Geesthacht, Geesthacht, Germany 4Institut f\\u00fcr Zoologie und  Evolutionsforschung mit Phyletischem  Museum, Ernst-Hackel-Haus und  Biologiedidaktik, Friedrich-Schiller-Universit\\u00e4t Jena, Jena, Germany 5YXLON International GmbH, Hamburg,  Germany Correspondence Karolin Engelkes, Center of Natural History  (CeNak), Universit\\u00e4t Hamburg, Martin-Luther-King-Platz 3, 20146 Hamburg,  Germany.\\nEmail: karolin.engelkes@uni-hamburg.de Funding information Wilhelm-Peters-Fonds of the Deutsche  Gesellschaft f\\u00fcr Herpetologie und  Terrarienkunde e.V (DGHT); Deutsche  Forschungsgemeinschaft, Grant/Award  Number: 387723284 Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that  allow them to inhabit a wide range of habitats.\\nThe different locomotor modes are  likely to be linked to anatomical specializations of the skeleton within the typical frog  Bauplan.\\nWhile such anatomical adaptations of the hind li\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.6784'], citation_context: \\\"20457758, 2020, 20, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6784 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/jeb.12292'], citation_context: \\\" M., Byrne, P. G., Roberts, J. D., & Keogh, J. S. (2014).\\nThe role  of phylogeny and ecology in shaping morphology in 21 genera and 127  species of Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181-192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825\\nWainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright,\\\"},{\\\"dataset_ids\\\": ['10.2307/1444825'], citation_context: \\\"Australo-Papuan myobatrachid frogs.\\nJournal of Evolutionary  Biology, 27(1), 181-192.\\nhttps://doi.org/10.1111/jeb.12292 Videler, J. J., & Jorna, J. T. (1985).\\n(1985)\\nFunctions of the sliding pelvis  in Xenopus laevis.\\nCopeia, 1, 251.\\nhttps://doi.org/10.2307/1444825\\nWainwright, P. C. (1994).\\nFunctional morphology as a tool in ecological  research.\\nIn P. C. Wainwright, & S. M. Reilly (Eds.), Ecological morphol-ogy: Integrative organismal biology (pp. 42-59).\\nChicago, LondonThe  University of Chicago Press.\\nWainw\\\"}\",\"{\\\"dataset_ids\\\": ['10.13140/RG.2.2.18389.86247'], citation_context: \\\"ge in the evolution of a novel locomotor  type: \\u201cflying\\u201d frogs.\\nEvolution, 44(8), 1931-1946.\\nEngelkes, K. (preprint, in review) Accuracy of bone segmentation and  surface generation strategies analyzed by using synthetic CT vol-umes.\\nhttps://doi.org/10.13140/RG.2.2.18389.86247 Engelkes, K., Friedrich, F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 2\\\"},{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], citation_context: \\\" F., Hammel, J. U., & Haas, A. (2018).\\nA simple  setup for episcopic microtomy and a digital image processing work-flow to acquire high-quality volume data and 3D surface models  of small vertebrates.\\nZoomorphology, 137(1), 213-228.\\nhttps://doi.\\norg/10.1007/s00435-017-0386-3 Engelkes, K., Helfsgott, J., Hammel, J. U., B\\u00fcsse, S., Kleinteich, T., Beerlink,  A., \\u2026 Haas, A. (2019).\\nMeasurement error in \\u03bcCT-based three-dimen-sional geometric morphometrics introduced by surface generation  and landmark data acquisition.\\nJourn\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1550-7408.2008.00342.x'], citation_context: \\\" lineage  of closely related trypanosomes (Trypanosomatidae, Kinetoplastida)  of anurans and sand flies (Psychodidae, Diptera) sharing the same  ecotopes in brazilian amazonia.\\nThe Journal of Eukaryotic Microbiology,  55(5), 427-435. https://doi.org/10.1111/j.1550-7408.2008.00342.x\\nFrost, D. R. (2020) Amphibian Species of the World: an online reference.  \\nVersion 6.1.\\nhttp://research.amnh.org/herpetology/amphibia/index.\\nhtml.\\nF\\u00fcrbringer, M. (1873).\\nZur vergleichenden Anatomie der Schultermuskeln.  \\nTeil 1.\\nJenaische Zeitschrif\\\"}\",\"{\\\"dataset_ids\\\": ['10.1073/pnas.0800375105'], citation_context: \\\"ak, M., Kadmon, R., Saltz, D.,  & Smouse, P. E. (2008).\\nA movement ecology paradigm for unifying  organismal movement research.\\nProceedings of the National Academy  of Sciences of the United States of America, 105(49), 19052-19059.  \\nhttps://doi.org/10.1073/pnas.0800375105 Nauwelaerts, S., & Aerts, P. (2006).\\nTake-off and landing forces in jump-ing frogs.\\nThe Journal of Experimental Biology, 209(1), 66-77.\\nhttps://doi.org/10.1242/jeb.01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of a\\\"},{\\\"dataset_ids\\\": ['10.1111/j.1469-7580.2007.00691.x'], citation_context: \\\".01969 Nauwelaerts, S., Ramsay, J., & Aerts, P. (2007).\\nMorphological correlates  of aquatic and terrestrial locomotion in a semi-aquatic frog, Rana es-culenta: No evidence for a design conflict.\\nJournal of Anatomy, 210(3),  304-317.\\nhttps://doi.org/10.1111/j.1469-7580.2007.00691.x\\nNoble, G. K., & Jaeckle, M. E. (1928).\\nThe digital pads of the tree frogs.  \\nA study of the phylogenesis of an adaptive structure.\\nJournal of  Morphology, 45(1), 259-292.\\nNomura, F., Rossa-Feres, D. C., & Langeani, F. (2009).\\nBurrowing behav-ior of \\\"}\"]}\n{\"title\":\"Ecomorphology of the pectoral girdle in anurans (Amphibia, Anura): Shape diversity and biomechanical considerations\",\"authors\":\"Engelkes Karolin, Kath Lena, Kleinteich Thomas, Hammel J\\u00f6rg U., Beerlink Andr\\u00e9, Haas Alexander\",\"abstract\":\"Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that allow them to inhabit a wide range of habitats. The different locomotor modes are likely to be linked to anatomical specializations of the skeleton within the typical frog Bauplan. While such anatomical adaptations of the hind limbs and the pelvic girdle are comparably well understood, the pectoral girdle received much less attention in the past. We tested for locomotor\\u2010mode\\u2010related shape differences in the pectoral girdle bones of 64 anuran species by means of micro\\u2010computed\\u2010tomography\\u2010based geometric morphometrics. The pectoral girdles of selected species were analyzed with regard to the effects of shape differences on muscle moment arms across the shoulder joint and stress dissipation within the coracoid. Phylogenetic relationships, size, and locomotor behavior have an effect on the shape of the pectoral girdle in anurans, but there are differences in the relative impact of these factors between the bones of this skeletal unit. Remarkable shape diversity has been observed within locomotor groups indicating many\\u2010to\\u2010one mapping of form onto function. Significant shape differences have mainly been related to the overall pectoral girdle geometry and the shape of the coracoid. Most prominent shape differences have been found between burrowing and nonburrowing species with headfirst and backward burrowing species significantly differing from one another and from the other locomotor groups. The pectoral girdle shapes of burrowing species have generally larger moment arms for (simulated) humerus retractor muscles across the shoulder joint, which might be an adaptation to the burrowing behavior. The mechanisms of how the moment arms were enlarged differed between species and were associated with differences in the reaction of the coracoid to simulated loading by physiologically relevant forces.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT CT volumes can be downloaded from  https://www.fdr.uni\\u2010hamburg.de/search?page=1&size=20&q=keywords:%22pectoral%20girdle%20morphometrics%20project%22 ; DOI numbers are provided in Appendix S1: Table A1.\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], citation_context: \\\"Engelkes ,  K. \\n ,  \\n Friedrich ,  F. \\n ,  \\n Hammel ,  J. U. \\n , &  \\n Haas ,  A. \\n  ( 2018 ).  A simple setup for episcopic microtomy and a digital image processing workflow to acquire high\\u2010quality volume data and 3D surface models of small vertebrates .  Zoomorphology ,  137 ( 1 ),  213 \\u2013 228 .  10.1007/s00435-017-0386-3\\\"}\"]}"
        ],
        [
         "18",
         "10.1002_ece3.961",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.961.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.961.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney Hall, Nathan J. Sanders,4 & Aim, Melissa A. Cregger,2, Norman\",\"abstract\":\"The effects of insects, nutrients, and plant invasion on community structure and function above- and belowground Phoebe Wright1, Melissa A. Cregger1,2, Lara Souza1,3, Nathan J. Sanders1,4 & Aim\\ufffdee T. Classen1,4 1Department of Ecology and Evolutionary Biology, University of Tennessee, 569 Dabney Hall, Knoxville, Tennessee 37996 2Institute for Genomic Biology, University of Illinois, 1206 W Gregory Rm 3405, Urbana, Illinois 61801 3Oklahoma Biological Survey and Department Microbiology and Plant Biology, University of Oklahoma, 111 E Chesapeake St., Norman, Oklahoma 73019 4Center for Macroecology Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, Copenhagen DK-2100, Denmark Keywords aboveground biomass, bacteria, fungi, insect, invasive plant, nitrogen-fixer, nutrient amendment, old-field ecosystem, soil enzyme activity.\\nCorrespondence Aim\\ufffdee T. Classen, Center for Macroecology, Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, DK-2100 Copenhagen, Denmark.\\nTel: +45 35-32-22-22; Fax: +45 46-82-74-83; E-mail: atclassen@gmail.com\\nFunding Information\\nThe Department of Ecology and Evolutionary Biology and the Office of Research at The University of Tennessee funded this work.\\nReceived:\\n21 November 2013; Revised: 2 December 2013; Accepted: 3 December 2013 Ecology and Evolution 2014; 4(6): 732-742 doi: 10.1002/ece3.961 Abstract Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact.\\nIn this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient avail-ability (nitrogen added, nitrogen reduced and control) in a fully crossed, com-pletely randomized plot design.\\nWe found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above- and belowground structure and function at our site.\\nNot surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization.\\nThe introduction of invas\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.961'], citation_context: \\\"20457758, 2014, 6, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.961 by California Digital Library University Of California, Wiley Online Library on [08/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"The effects of insects, nutrients, and plant invasion on community structure and function above-and belowground\",\"authors\":\"Wright Phoebe, Cregger Melissa A, Souza Lara, Sanders Nathan J, Classen Aim\\u00e9e T\",\"abstract\":\"Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact. In this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient availability (nitrogen added, nitrogen reduced and control) in a fully crossed, completely randomized plot design. We found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above-and belowground structure and function at our site. Not surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization. The introduction of invasive nitrogen-fixing plant propagules interacted with nutrient amendment and insect presence to alter soil bacterial abundance and the activity of the microbial community. While the larger-scale, longer-term bulk measurements such as biomass production and nutrient mineralization responded to the direct effects of our treatments, the shorter-term and dynamic microbial communities tended to respond to interactions among our treatments. Our results indicate that soil nutrients, invasive plants, and insect herbivores determine both above-and belowground responses, but whether such effects are independent versus interdependent varies with scale.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney Hall, Nathan J. Sanders,4 & Aim, Melissa A. Cregger,2, Norman\",\"abstract\":\"The effects of insects, nutrients, and plant invasion on community structure and function above- and belowground Phoebe Wright1, Melissa A. Cregger1,2, Lara Souza1,3, Nathan J. Sanders1,4 & Aim\\ufffdee T. Classen1,4 1Department of Ecology and Evolutionary Biology, University of Tennessee, 569 Dabney Hall, Knoxville, Tennessee 37996 2Institute for Genomic Biology, University of Illinois, 1206 W Gregory Rm 3405, Urbana, Illinois 61801 3Oklahoma Biological Survey and Department Microbiology and Plant Biology, University of Oklahoma, 111 E Chesapeake St., Norman, Oklahoma 73019 4Center for Macroecology Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, Copenhagen DK-2100, Denmark Keywords aboveground biomass, bacteria, fungi, insect, invasive plant, nitrogen-fixer, nutrient amendment, old-field ecosystem, soil enzyme activity.\\nCorrespondence Aim\\ufffdee T. Classen, Center for Macroecology, Evolution and Climate, Natural History Museum of Denmark, University of Copenhagen, DK-2100 Copenhagen, Denmark.\\nTel: +45 35-32-22-22; Fax: +45 46-82-74-83; E-mail: atclassen@gmail.com\\nFunding Information\\nThe Department of Ecology and Evolutionary Biology and the Office of Research at The University of Tennessee funded this work.\\nReceived:\\n21 November 2013; Revised: 2 December 2013; Accepted: 3 December 2013 Ecology and Evolution 2014; 4(6): 732-742 doi: 10.1002/ece3.961 Abstract Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact.\\nIn this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient avail-ability (nitrogen added, nitrogen reduced and control) in a fully crossed, com-pletely randomized plot design.\\nWe found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above- and belowground structure and function at our site.\\nNot surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization.\\nThe introduction of invas\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.961'], citation_context: \\\"20457758, 2014, 6, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.961 by California Digital Library University Of California, Wiley Online Library on [08/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"The effects of insects, nutrients, and plant invasion on community structure and function above-and belowground\",\"authors\":\"Wright Phoebe, Cregger Melissa A, Souza Lara, Sanders Nathan J, Classen Aim\\u00e9e T\",\"abstract\":\"Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact. In this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient availability (nitrogen added, nitrogen reduced and control) in a fully crossed, completely randomized plot design. We found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above-and belowground structure and function at our site. Not surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization. The introduction of invasive nitrogen-fixing plant propagules interacted with nutrient amendment and insect presence to alter soil bacterial abundance and the activity of the microbial community. While the larger-scale, longer-term bulk measurements such as biomass production and nutrient mineralization responded to the direct effects of our treatments, the shorter-term and dynamic microbial communities tended to respond to interactions among our treatments. Our results indicate that soil nutrients, invasive plants, and insect herbivores determine both above-and belowground responses, but whether such effects are independent versus interdependent varies with scale.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "19",
         "10.1002_ece3.9627",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.9627.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.9627.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2, Revised, Evolution, Email, Zachary Amir, Goddard Hall\",\"abstract\":\"Ecology and Evolution.\\n2022;12:e9627.  \\n\\ufeff    | 1 of 15 https://doi.org/10.1002/ece3.9627 www.ecolevol.org\\nReceived: 5 June 2022 | Revised: 14 October 2022 | Accepted: 20 November 2022 DOI: 10.1002/ece3.9627   R E S E A R C H  A R T I C L E Inferring predator-\\u00adprey interactions from camera traps:  A Bayesian co-\\u00adabundance modeling approach Zachary Amir1,2  |   Adia Sovie3  |   Matthew Scott Luskin1,2\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2022 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1School of Biological Sciences, University  of Queensland, St. Lucia, Queensland,  Australia 2Centre for Biodiversity and Conservation  Science, University of Queensland, St.  Lucia, Queensland, Australia 3Department of Fisheries and Wildlife,  Michigan State University, East Lansing,  Michigan, USA Correspondence Zachary Amir, School of Biological  Sciences, University of Queensland,  534 Goddard Hall, St. Lucia 4067, QLD,  Australia.\\nEmail: z.amir@uq.edu.au Abstract Predator-\\u00adprey dynamics are a fundamental part of ecology, but directly studying inter-actions has proven difficult.\\nThe proliferation of camera trapping has enabled the col-lection of large datasets on wildlife, but researchers face hurdles inferring interactions  from observational data.\\nRecent advances in hierarchical co-\\u00adabundance models infer  species interactions while accounting for two species' detection probabilities, shared  responses to environmental covariates, and propagate uncertainty throughout the  entire modeling process.\\nHowever, current approaches remain unsuitable for interact-ing specie\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.9627'], citation_context: \\\"20457758, 2022, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.9627 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1086/685095'], citation_context: \\\"interactions on populations of endangered species.\\nThe American  Naturalist, 187, 457-\\u00ad467. https://doi.org/10.1086/685095\\nRovero, F., & Zimmermann, F. (2016).\\nCamera trapping for wildlife research.  \\nPelagic Publishing.\\nRoyle, J. A. (2004).\\nN-\\u00admixture models for estimating population size from\\\"}\"]}",
         "{\"title\":\"Inferring predator\\u2013prey interactions from camera traps: A\\u00a0Bayesian co\\u2010abundance modeling approach\",\"authors\":\"Amir Zachary, Sovie Adia, Luskin Matthew Scott\",\"abstract\":\"Abstract Predator\\u2013prey dynamics are a fundamental part of ecology, but directly studying interactions has proven difficult. The proliferation of camera trapping has enabled the collection of large datasets on wildlife, but researchers face hurdles inferring interactions from observational data. Recent advances in hierarchical co\\u2010abundance models infer species interactions while accounting for two species' detection probabilities, shared responses to environmental covariates, and propagate uncertainty throughout the entire modeling process. However, current approaches remain unsuitable for interacting species whose natural densities differ by an order of magnitude and have contrasting detection probabilities, such as predator\\u2013prey interactions, which introduce zero inflation and overdispersion in count histories. Here, we developed a Bayesian hierarchical N\\u2010mixture co\\u2010abundance model that is suitable for inferring predator\\u2013prey interactions. We accounted for excessive zeros in count histories using an informed zero\\u2010inflated Poisson distribution in the abundance formula and accounted for overdispersion in count histories by including a random effect per sampling unit and sampling occasion in the detection probability formula. We demonstrate that models with these modifications outperform alternative approaches, improve model goodness\\u2010of\\u2010fit, and overcome parameter convergence failures. We highlight its utility using 20 camera trapping datasets from 10 tropical forest landscapes in Southeast Asia and estimate four predator\\u2013prey relationships between tigers, clouded leopards, and muntjac and sambar deer. Tigers had a negative effect on muntjac abundance, providing support for top\\u2010down regulation, while clouded leopards had a positive effect on muntjac and sambar deer, likely driven by shared responses to unmodelled covariates like hunting. This Bayesian co\\u2010abundance modeling approach to quantify predator\\u2013prey relationships is widely applicable across species, ecosystems\",\"data_availability\":\"DATA AVAILABILITY STATEMENT All R code and data used to implement our Bayesian co\\u2010abundance models can be accessed at this Figshare repository:  https://doi.org/10.5061/dryad.b8gtht7h3 .\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2, Revised, Evolution, Email, Zachary Amir, Goddard Hall\",\"abstract\":\"Ecology and Evolution.\\n2022;12:e9627.  \\n\\ufeff    | 1 of 15 https://doi.org/10.1002/ece3.9627 www.ecolevol.org\\nReceived: 5 June 2022 | Revised: 14 October 2022 | Accepted: 20 November 2022 DOI: 10.1002/ece3.9627   R E S E A R C H  A R T I C L E Inferring predator-\\u00adprey interactions from camera traps:  A Bayesian co-\\u00adabundance modeling approach Zachary Amir1,2  |   Adia Sovie3  |   Matthew Scott Luskin1,2\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2022 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. 1School of Biological Sciences, University  of Queensland, St. Lucia, Queensland,  Australia 2Centre for Biodiversity and Conservation  Science, University of Queensland, St.  Lucia, Queensland, Australia 3Department of Fisheries and Wildlife,  Michigan State University, East Lansing,  Michigan, USA Correspondence Zachary Amir, School of Biological  Sciences, University of Queensland,  534 Goddard Hall, St. Lucia 4067, QLD,  Australia.\\nEmail: z.amir@uq.edu.au Abstract Predator-\\u00adprey dynamics are a fundamental part of ecology, but directly studying inter-actions has proven difficult.\\nThe proliferation of camera trapping has enabled the col-lection of large datasets on wildlife, but researchers face hurdles inferring interactions  from observational data.\\nRecent advances in hierarchical co-\\u00adabundance models infer  species interactions while accounting for two species' detection probabilities, shared  responses to environmental covariates, and propagate uncertainty throughout the  entire modeling process.\\nHowever, current approaches remain unsuitable for interact-ing specie\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ece3.9627'], citation_context: \\\"20457758, 2022, 12, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.9627 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1086/685095'], citation_context: \\\"interactions on populations of endangered species.\\nThe American  Naturalist, 187, 457-\\u00ad467. https://doi.org/10.1086/685095\\nRovero, F., & Zimmermann, F. (2016).\\nCamera trapping for wildlife research.  \\nPelagic Publishing.\\nRoyle, J. A. (2004).\\nN-\\u00admixture models for estimating population size from\\\"}\"]}\n{\"title\":\"Inferring predator\\u2013prey interactions from camera traps: A\\u00a0Bayesian co\\u2010abundance modeling approach\",\"authors\":\"Amir Zachary, Sovie Adia, Luskin Matthew Scott\",\"abstract\":\"Abstract Predator\\u2013prey dynamics are a fundamental part of ecology, but directly studying interactions has proven difficult. The proliferation of camera trapping has enabled the collection of large datasets on wildlife, but researchers face hurdles inferring interactions from observational data. Recent advances in hierarchical co\\u2010abundance models infer species interactions while accounting for two species' detection probabilities, shared responses to environmental covariates, and propagate uncertainty throughout the entire modeling process. However, current approaches remain unsuitable for interacting species whose natural densities differ by an order of magnitude and have contrasting detection probabilities, such as predator\\u2013prey interactions, which introduce zero inflation and overdispersion in count histories. Here, we developed a Bayesian hierarchical N\\u2010mixture co\\u2010abundance model that is suitable for inferring predator\\u2013prey interactions. We accounted for excessive zeros in count histories using an informed zero\\u2010inflated Poisson distribution in the abundance formula and accounted for overdispersion in count histories by including a random effect per sampling unit and sampling occasion in the detection probability formula. We demonstrate that models with these modifications outperform alternative approaches, improve model goodness\\u2010of\\u2010fit, and overcome parameter convergence failures. We highlight its utility using 20 camera trapping datasets from 10 tropical forest landscapes in Southeast Asia and estimate four predator\\u2013prey relationships between tigers, clouded leopards, and muntjac and sambar deer. Tigers had a negative effect on muntjac abundance, providing support for top\\u2010down regulation, while clouded leopards had a positive effect on muntjac and sambar deer, likely driven by shared responses to unmodelled covariates like hunting. This Bayesian co\\u2010abundance modeling approach to quantify predator\\u2013prey relationships is widely applicable across species, ecosystems\",\"data_availability\":\"DATA AVAILABILITY STATEMENT All R code and data used to implement our Bayesian co\\u2010abundance models can be accessed at this Figshare repository:  https://doi.org/10.5061/dryad.b8gtht7h3 .\",\"other_dataset_citations\":[]}"
        ],
        [
         "20",
         "10.1002_ecs2.1280",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.1280.pdf",
         null,
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.p3fg9', 'type': 'Primary'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De \\u00adClerck-Floate3, R. G. Lalonde, Yvonne M. Buckley,2, R. A. De Clerck-Floate, Haley A. Catton,,\\u2020 Robert G. Lalonde,1, Y. M. Buckley\",\"abstract\":\"Abstract.   \\nUnderstanding the impact of herbivory on plant populations is a fundamental goal of ecology.  \\nDamage to individual plants can be visually striking and affect the fates of individuals, but these impacts  do not necessarily translate into population-\\u00adlevel differences in vital rates (survival, growth, or fecundity)  or population growth rates.\\nIn biological control of weeds, quantitative assessments of population-\\u00adlevel  impacts of released agents on both target invasive plants and native, nontarget plants are needed to inform  evaluations of the benefits and risks of releasing agents into new regions.\\nHere we present a 3-\\u00adyr exper-imental demographic field study using the European root-\\u00adfeeding biocontrol weevil, Mogulones crucifer,  first released in Canada in 1997 to control the invasive weed Cynoglossum officinale (Boraginaceae).\\nMogu-lones crucifer is an effective \\u201csearch and destroy\\u201d agent in Canada, but sporadically feeds, oviposits, and  develops on native nontarget Boraginaceae.\\nWe investigated the population-\\u00adlevel impacts of this biocon-trol insect on its target weed and a native nontarget plant, Hackelia micrantha (Boraginaceae), by releasing  large numbers of weevils into naturally occurring patches of H. micrantha growing isolated from or inter-spersed with C. officinale.\\nWe followed the fates of individual plants on release and nonrelease (control)  sites for two transition years, developed matrix models to project population growth rates (\\u03bb) for each  plant species, and examined the contributions from differences in vital rates to changes in \\u03bb using life table  response experiments (LTRE).\\nIn contrast to studies of the insect-plant interaction in its native range, as a  biocontrol agent, M. crucifer increased mortality of C. officinale rosettes in the year immediately following  release, depressing the weed's \\u03bb to below the population replacement level.\\nHowever, \\u03bb for H. micrantha  was never depressed below the replacement level, and any differences between release and nonrelease  sites in the nontarget could not be explained by significant contributions from vital rates in the LTRE.\\nThis  study is the first to simultaneously and experimentally examine target and nontarget population-\\u00adlevel im-pacts of a weed biocontrol insect in the field, and supports the theoretical prediction that plant life history  characteristics and uneven herbivore host preferences can interact to produce differences in population-\\u00ad level impacts between target and nontarget plant species.\",\"data_availability\":\"Data Availability\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.1280'], citation_context: \\\"21508925, 2016, 5, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.1280 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "",
         "{\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De \\u00adClerck-Floate3, R. G. Lalonde, Yvonne M. Buckley,2, R. A. De Clerck-Floate, Haley A. Catton,,\\u2020 Robert G. Lalonde,1, Y. M. Buckley\",\"abstract\":\"Abstract.   \\nUnderstanding the impact of herbivory on plant populations is a fundamental goal of ecology.  \\nDamage to individual plants can be visually striking and affect the fates of individuals, but these impacts  do not necessarily translate into population-\\u00adlevel differences in vital rates (survival, growth, or fecundity)  or population growth rates.\\nIn biological control of weeds, quantitative assessments of population-\\u00adlevel  impacts of released agents on both target invasive plants and native, nontarget plants are needed to inform  evaluations of the benefits and risks of releasing agents into new regions.\\nHere we present a 3-\\u00adyr exper-imental demographic field study using the European root-\\u00adfeeding biocontrol weevil, Mogulones crucifer,  first released in Canada in 1997 to control the invasive weed Cynoglossum officinale (Boraginaceae).\\nMogu-lones crucifer is an effective \\u201csearch and destroy\\u201d agent in Canada, but sporadically feeds, oviposits, and  develops on native nontarget Boraginaceae.\\nWe investigated the population-\\u00adlevel impacts of this biocon-trol insect on its target weed and a native nontarget plant, Hackelia micrantha (Boraginaceae), by releasing  large numbers of weevils into naturally occurring patches of H. micrantha growing isolated from or inter-spersed with C. officinale.\\nWe followed the fates of individual plants on release and nonrelease (control)  sites for two transition years, developed matrix models to project population growth rates (\\u03bb) for each  plant species, and examined the contributions from differences in vital rates to changes in \\u03bb using life table  response experiments (LTRE).\\nIn contrast to studies of the insect-plant interaction in its native range, as a  biocontrol agent, M. crucifer increased mortality of C. officinale rosettes in the year immediately following  release, depressing the weed's \\u03bb to below the population replacement level.\\nHowever, \\u03bb for H. micrantha  was never depressed below the replacement level, and any differences between release and nonrelease  sites in the nontarget could not be explained by significant contributions from vital rates in the LTRE.\\nThis  study is the first to simultaneously and experimentally examine target and nontarget population-\\u00adlevel im-pacts of a weed biocontrol insect in the field, and supports the theoretical prediction that plant life history  characteristics and uneven herbivore host preferences can interact to produce differences in population-\\u00ad level impacts between target and nontarget plant species.\",\"data_availability\":\"Data Availability\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.1280'], citation_context: \\\"21508925, 2016, 5, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.1280 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n"
        ],
        [
         "21",
         "10.1002_ecs2.4619",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ecs2.4619.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ecs2.4619.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.25349/d9qw5x', 'type': 'Primary'}]",
         "<html><body><tei xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" xmlns:xlink=\"htt",
         "tei",
         "{\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis, John, Michelle J. Lee1, Caroline H. Owens1, Hillary S., Email, Caroline H. Owens, Melissa\",\"abstract\":\"A R T I C L E M a c r o s y s t e m s E c o l o g y Interactions between temperature and predation impact insect emergence in alpine lakes Caroline H. Owens1 | Michelle J. Lee1\\n| Melissa Grim1,2 | John Schroeder1 | Hillary S. Young1 1Department of Ecology Evolution and Marine Biology, University of California, Santa Barbara, Santa Barbara, California, USA 2Department of Environmental Science and Policy, University of California, Davis, Davis, California, USA Correspondence Caroline H. Owens Email:\\nowens@ucsb.edu\\nFunding information National Science Foundation; University of California Natural Reserve System; University of California, Santa Barbara Handling Editor: Manuel T. Lerdau Abstract While insect populations are simultaneously threatened by many local- and global-scale stressors, the interacting effects of these factors remain poorly understood.\\nThese interactions between stressors, whether additive or antagonis-tic, may have profound effects on our assumptions about and predictions for any given system.\\nHere we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes.\\nUsing a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence.\\nBased on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6\\ufffdC of water temperature in oligotrophic alpine lakes.\\nWhile our work confirms p\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data and novel code (Owens et al., 2023) are available from Dryad: https://doi.org/10.25349/D9QW5X.\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.4619'], citation_context: \\\"21508925, 2023, 7, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.4619, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/2015GL066235.'], citation_context: \\\".\\nO\\u2019Reilly, C. M., S. Sharma, D. K. Gray, S. E. Hampton, J. S. Read, R. J. Rowley, P. Schneider, et al. 2015.\\n\\u201cRapid and Highly Variable Warming of Lake Surface Waters around the Globe.\\u201d\\nGeophysical Research Letters 42(24): 10773-81.\\nhttps://doi.org/10.1002/2015GL066235.\\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023.\\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset.\\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. Knap\\\"},{\\\"dataset_ids\\\": ['10.25349/D9QW5X.'], citation_context: \\\"42(24): 10773-81.\\nhttps://doi.org/10.1002/2015GL066235.\\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023.\\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset.\\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. Knapp, J. Sickman, K. L. Pope, and S. Chandra.\\n2016.\\n\\u201cVariation in Reciprocal Subsidies between Lakes and Land: Perspectives from the Mountains of\\\"}\"]}",
         "{\"title\":\"Interactions between temperature and predation impact insect emergence in alpine lakes\",\"authors\":\"Caroline H Owens, Michelle J Lee, | Melissa Grim, John Schroeder, Hillary S Young, Manuel T Lerdau\",\"abstract\":\"While insect populations are simultaneously threatened by many local-and global-scale stressors, the interacting effects of these factors remain poorly understood. These interactions between stressors, whether additive or antagonistic, may have profound effects on our assumptions about and predictions for any given system. Here we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes. Using a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence. Based on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6 C of water temperature in oligotrophic alpine lakes. While our work confirms prior findings that predator presence drives strong reductions in insect emergence, we find that the effects of predation are significantly weaker in warmer lakes (2% reduction in warmest lakes studied vs. 75% reduction in coldest). Combining our results with regional studies of predator communities to identify mechanisms suggests that the interaction effects between temperature and predator presence are likely due to changes in predator behavior across temperatures. Critically, to fully understand the multiple stressors that impact insect populations and predict their future effects, we must consider the interactions between stressors across spatial scales.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.25349/D9QW5X'], citation_context: \\\"Data for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes|Dryad. Dataset|10.25349/D9QW5X\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis, John, Michelle J. Lee1, Caroline H. Owens1, Hillary S., Email, Caroline H. Owens, Melissa\",\"abstract\":\"A R T I C L E M a c r o s y s t e m s E c o l o g y Interactions between temperature and predation impact insect emergence in alpine lakes Caroline H. Owens1 | Michelle J. Lee1\\n| Melissa Grim1,2 | John Schroeder1 | Hillary S. Young1 1Department of Ecology Evolution and Marine Biology, University of California, Santa Barbara, Santa Barbara, California, USA 2Department of Environmental Science and Policy, University of California, Davis, Davis, California, USA Correspondence Caroline H. Owens Email:\\nowens@ucsb.edu\\nFunding information National Science Foundation; University of California Natural Reserve System; University of California, Santa Barbara Handling Editor: Manuel T. Lerdau Abstract While insect populations are simultaneously threatened by many local- and global-scale stressors, the interacting effects of these factors remain poorly understood.\\nThese interactions between stressors, whether additive or antagonis-tic, may have profound effects on our assumptions about and predictions for any given system.\\nHere we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes.\\nUsing a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence.\\nBased on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6\\ufffdC of water temperature in oligotrophic alpine lakes.\\nWhile our work confirms p\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data and novel code (Owens et al., 2023) are available from Dryad: https://doi.org/10.25349/D9QW5X.\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.4619'], citation_context: \\\"21508925, 2023, 7, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.4619, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/2015GL066235.'], citation_context: \\\".\\nO\\u2019Reilly, C. M., S. Sharma, D. K. Gray, S. E. Hampton, J. S. Read, R. J. Rowley, P. Schneider, et al. 2015.\\n\\u201cRapid and Highly Variable Warming of Lake Surface Waters around the Globe.\\u201d\\nGeophysical Research Letters 42(24): 10773-81.\\nhttps://doi.org/10.1002/2015GL066235.\\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023.\\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset.\\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. Knap\\\"},{\\\"dataset_ids\\\": ['10.25349/D9QW5X.'], citation_context: \\\"42(24): 10773-81.\\nhttps://doi.org/10.1002/2015GL066235.\\nOwens, C., M. Lee, M. Grim, J. Schroeder, and H. Young. 2023.\\n\\u201cData for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes.\\u201d\\nDryad.\\nDataset.\\nhttps://doi.org/10.25349/D9QW5X. Piovia-Scott, J., S. Sadro, R. A. Knapp, J. Sickman, K. L. Pope, and S. Chandra.\\n2016.\\n\\u201cVariation in Reciprocal Subsidies between Lakes and Land: Perspectives from the Mountains of\\\"}\"]}\n{\"title\":\"Interactions between temperature and predation impact insect emergence in alpine lakes\",\"authors\":\"Caroline H Owens, Michelle J Lee, | Melissa Grim, John Schroeder, Hillary S Young, Manuel T Lerdau\",\"abstract\":\"While insect populations are simultaneously threatened by many local-and global-scale stressors, the interacting effects of these factors remain poorly understood. These interactions between stressors, whether additive or antagonistic, may have profound effects on our assumptions about and predictions for any given system. Here we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes. Using a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence. Based on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6 C of water temperature in oligotrophic alpine lakes. While our work confirms prior findings that predator presence drives strong reductions in insect emergence, we find that the effects of predation are significantly weaker in warmer lakes (2% reduction in warmest lakes studied vs. 75% reduction in coldest). Combining our results with regional studies of predator communities to identify mechanisms suggests that the interaction effects between temperature and predator presence are likely due to changes in predator behavior across temperatures. Critically, to fully understand the multiple stressors that impact insect populations and predict their future effects, we must consider the interactions between stressors across spatial scales.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.25349/D9QW5X'], citation_context: \\\"Data for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes|Dryad. Dataset|10.25349/D9QW5X\\\"}\"]}"
        ],
        [
         "22",
         "10.1002_ejic.201900904",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejic.201900904.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejic.201900904.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley",
         "{\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"Abstract: Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2-(dimethylaminometh-yl)phenyl-C1,N]gold(III),\\n[Au(damp-C1,N)Cl2], and halogenated, potentially tridentate thiosemicarbazones have been studied.\\nThe results of this work show that the complexation of the halo-genated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti-parasitic activity.\\nFluorination in the periphery of the ligands has expect-\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2\\u2010(dimethylaminomethyl)phenyl\\u2010C 1 ,N]gold(III), [Au(damp\\u2010C 1 ,N)Cl 2 ], and halogenated, potentially tridentate thiosemicarbazones have been studied. The results of this work show that the complexation of the halogenated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti\\u2010parasitic activity. Fluorination in the periphery of the ligands has expectedly no influence on the structural chemistry of the obtained Au(III) complexes, but modulates their biological behaviour. Best results with a remarkably high selectivity index for the trypomastigote form of  Trypanosoma cruzi  were obtained with the complex containing the ligand, which presents a 3,5\\u2010fluorine substitution in meta\\u2010position of an aromatic ring, [Au(dampH)(L\\u20103,5\\u2010F)]Cl.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"Abstract: Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2-(dimethylaminometh-yl)phenyl-C1,N]gold(III),\\n[Au(damp-C1,N)Cl2], and halogenated, potentially tridentate thiosemicarbazones have been studied.\\nThe results of this work show that the complexation of the halo-genated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti-parasitic activity.\\nFluorination in the periphery of the ligands has expect-\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2\\u2010(dimethylaminomethyl)phenyl\\u2010C 1 ,N]gold(III), [Au(damp\\u2010C 1 ,N)Cl 2 ], and halogenated, potentially tridentate thiosemicarbazones have been studied. The results of this work show that the complexation of the halogenated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti\\u2010parasitic activity. Fluorination in the periphery of the ligands has expectedly no influence on the structural chemistry of the obtained Au(III) complexes, but modulates their biological behaviour. Best results with a remarkably high selectivity index for the trypomastigote form of  Trypanosoma cruzi  were obtained with the complex containing the ligand, which presents a 3,5\\u2010fluorine substitution in meta\\u2010position of an aromatic ring, [Au(dampH)(L\\u20103,5\\u2010F)]Cl.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf",
         null,
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "",
         "{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n"
        ],
        [
         "24",
         "10.1002_ejoc.202000916",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000916.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ejoc.202000916.xml",
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract: We herein report an ammonium salt-catalyzed proto-col for the regioselective ring opening of aryl-aziridines with \\ufffd-keto esters.\\nThe reaction gives access to a variety of highly\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2010Opening of Aryl\\u2010Aziridines with \\u03b2\\u2010Keto Esters\",\"authors\":\"Haider Victoria, Kreuzer Viktoria, Tiffner Maximilian Dr., Spingler Bernhard Dr., Waser Mario Dr.\",\"abstract\":\"We herein report an ammonium salt\\u2010catalyzed protocol for the regioselective ring opening of aryl\\u2010aziridines with \\u03b2\\u2010keto esters. The reaction gives access to a variety of highly functionalized target molecules with two consecutive stereogenic centers and can be rendered enantioselective (up to  e.r . = 91:9) by using bifunctional chiral ammonium salt catalysts.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract: We herein report an ammonium salt-catalyzed proto-col for the regioselective ring opening of aryl-aziridines with \\ufffd-keto esters.\\nThe reaction gives access to a variety of highly\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n{\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2010Opening of Aryl\\u2010Aziridines with \\u03b2\\u2010Keto Esters\",\"authors\":\"Haider Victoria, Kreuzer Viktoria, Tiffner Maximilian Dr., Spingler Bernhard Dr., Waser Mario Dr.\",\"abstract\":\"We herein report an ammonium salt\\u2010catalyzed protocol for the regioselective ring opening of aryl\\u2010aziridines with \\u03b2\\u2010keto esters. The reaction gives access to a variety of highly functionalized target molecules with two consecutive stereogenic centers and can be rendered enantioselective (up to  e.r . = 91:9) by using bifunctional chiral ammonium salt catalysts.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}"
        ],
        [
         "25",
         "10.1002_esp.5058",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5058.pdf",
         null,
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark Nelson2\",\"abstract\":\"R E S E A R C H A R T I C L E Past, present and future of a meandering river in the Bolivian Amazon basin Kattia Rubi Arnez Ferrel1 |\\nJonathan Mark Nelson2 | Yasuyuki Shimizu1 | Tomoko Kyuka1 1Graduate School of Engineering, Hokkaido University, Sapporo, Japan 2U.S. Geological Survey, Golden, Colorado, USA Correspondence Kattia Rubi Arnez Ferrel, Graduate School of Engineering, Hokkaido University, Sapporo 060-0808, Japan.\\nEmail: rubikraf@gmail.com\\nFunding information Nitobe School Project Abstract Field observations on small rivers of the Amazon basin are less common due to their remote location and difficult accessibility.\\nHere we show, through remote sensing analysis and field works, the planform evolution and riverbed topography of a small river located in the upper foreland Amazon basin, the Ichilo River.\\nBy tracking plan-form changes over 30 years, we identified the factors that control meander migration rates in the Ichilo River: cutoffs, climate and human interventions.\\nThe data suggest that neck cutoffs are the main controls in the Ichilo River, with an annual density of 0.022 cutoffs/km.\\nIn addition, climate controls have been identified in the form of high-precipitation events that may have promoted cutoffs, an increase in meander migration rate and channel widening.\\nThe width distribution of the Ichilo River is well represented by general extreme value and inverse Gaussian distributions.\\nThe spatio-temporal variability of meandering migration rates in the Ichilo River is analysed in two locations where neck cutoffs are expected.\\nAnalysing the distance across the neck in these two points, we predict the occurrence of a new cutoff.\\nThe combined methodology of bathymetric surveys and structure from motion photogrammetry shows us the Ichilo riverbed topography and banks at high resolution, where two scour holes were identified.\\nFinally, we discuss the impact of planform changes of the Ichilo River on communities that are established along its riverbanks.\\nK E Y W O R D S bathymetric surveys, Bolivian Amazon basin, Ichilo River, meanders, remote sensing, UAV 1 | INTRODUCTION\\nThe Amazon basin exhibits a constantly changing landscap\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/esp.4637'], citation_context: \\\", M.R., Chandler, J.H., Eltner, A., Fraser, C., Miller, P.E., Mills, J.P. et al. (2019) Guidelines on the use of structure-from-motion photo-grammetry in geomorphic research.\\nEarth Surface Processes and Land-forms, 44(10), 2081-2084.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Semin\\\"},{\\\"dataset_ids\\\": ['10.5066/P9FW6E8K'], citation_context: \\\"84.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-d\\\"},{\\\"dataset_ids\\\": ['10.1029/2005JF000416'], citation_context: \\\", Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-driven abandonment and sedimentation of meander bends along a fine-grained, non-vegetated, ephemeral river on the Bolivian Alti-plano.\\nGeomorphology, 350, 106917.\\nhttps://doi\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5058'], citation_context: \\\"10969837, 2021, 4, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5058, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "",
         "{\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark Nelson2\",\"abstract\":\"R E S E A R C H A R T I C L E Past, present and future of a meandering river in the Bolivian Amazon basin Kattia Rubi Arnez Ferrel1 |\\nJonathan Mark Nelson2 | Yasuyuki Shimizu1 | Tomoko Kyuka1 1Graduate School of Engineering, Hokkaido University, Sapporo, Japan 2U.S. Geological Survey, Golden, Colorado, USA Correspondence Kattia Rubi Arnez Ferrel, Graduate School of Engineering, Hokkaido University, Sapporo 060-0808, Japan.\\nEmail: rubikraf@gmail.com\\nFunding information Nitobe School Project Abstract Field observations on small rivers of the Amazon basin are less common due to their remote location and difficult accessibility.\\nHere we show, through remote sensing analysis and field works, the planform evolution and riverbed topography of a small river located in the upper foreland Amazon basin, the Ichilo River.\\nBy tracking plan-form changes over 30 years, we identified the factors that control meander migration rates in the Ichilo River: cutoffs, climate and human interventions.\\nThe data suggest that neck cutoffs are the main controls in the Ichilo River, with an annual density of 0.022 cutoffs/km.\\nIn addition, climate controls have been identified in the form of high-precipitation events that may have promoted cutoffs, an increase in meander migration rate and channel widening.\\nThe width distribution of the Ichilo River is well represented by general extreme value and inverse Gaussian distributions.\\nThe spatio-temporal variability of meandering migration rates in the Ichilo River is analysed in two locations where neck cutoffs are expected.\\nAnalysing the distance across the neck in these two points, we predict the occurrence of a new cutoff.\\nThe combined methodology of bathymetric surveys and structure from motion photogrammetry shows us the Ichilo riverbed topography and banks at high resolution, where two scour holes were identified.\\nFinally, we discuss the impact of planform changes of the Ichilo River on communities that are established along its riverbanks.\\nK E Y W O R D S bathymetric surveys, Bolivian Amazon basin, Ichilo River, meanders, remote sensing, UAV 1 | INTRODUCTION\\nThe Amazon basin exhibits a constantly changing landscap\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/esp.4637'], citation_context: \\\", M.R., Chandler, J.H., Eltner, A., Fraser, C., Miller, P.E., Mills, J.P. et al. (2019) Guidelines on the use of structure-from-motion photo-grammetry in geomorphic research.\\nEarth Surface Processes and Land-forms, 44(10), 2081-2084.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Semin\\\"},{\\\"dataset_ids\\\": ['10.5066/P9FW6E8K'], citation_context: \\\"84.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-d\\\"},{\\\"dataset_ids\\\": ['10.1029/2005JF000416'], citation_context: \\\", Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-driven abandonment and sedimentation of meander bends along a fine-grained, non-vegetated, ephemeral river on the Bolivian Alti-plano.\\nGeomorphology, 350, 106917.\\nhttps://doi\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5058'], citation_context: \\\"10969837, 2021, 4, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5058, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n"
        ],
        [
         "26",
         "10.1002_esp.5090",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_esp.5090.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_esp.5090.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5066/p9353101', 'type': 'Secondary'}]",
         "<component xmlns=\"http://www.wiley.com/namespaces/wiley\" xmlns:wiley=\"http://www.wiley.com",
         "wiley",
         "{\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2\\n|, Mark, Tamarah Rosellen King, Albert, Dan, Abaz\",\"abstract\":\"R E S E A R C H A R T I C L E Paleoseismology of the 2016 MW 6.1 Petermann earthquake source: Implications for intraplate earthquake behaviour and the geomorphic longevity of bedrock fault scarps in a low strain-rate cratonic region Tamarah Rosellen King1,2 | Mark Quigley1 | Dan Clark3 | Albert Zondervan4 | Jan-Hendrik May5 | Abaz Alimanovic1 1School of Earth Sciences, The University of Melbourne, Melbourne, Victoria, Australia 2COMET, Department of Earth Sciences, University of Oxford, Oxford, UK 3Geoscience Australia, Canberra, Australian Capital Territory, Australia 4Rafter Radiocarbon Laboratory, GNS Science, Lower Hutt, New Zealand 5School of Geography, The University of Melbourne, Melbourne, Victoria, Australia Correspondence Tamarah Rosellen King, School of Earth Sciences, The University of Melbourne, Melbourne, VIC 3010, Australia.\\nEmail: tamarah.king@earth.ox.ac.uk Funding information British Geological Survey; University of Melbourne; Australian Research Council, Grant/Award Number: DP170103350\\nAbstract The 20 May 2016 MW 6.1 Petermann earthquake in central Australia generated a 21 km surface rupture with 0.1 to 1 m vertical displacements across a low-relief landscape.\\nNo paleo-scarps or potentially analogous topographic features are evident in pre-earthquake Worldview-1 and Worldview-2 satellite data.\\nTwo excavations across the surface rupture expose near-surface fault geometry and mixed aeolian-sheetwash sediment faulted only in the 2016 earthquake.\\nA 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture.\\nSeven cosmogenic beryllium-10 (10Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging-wall and foot-\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], citation_context: \\\"& Cummins, P.R. (2009) Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low-T thermochronological data.\\nAustralian Journal of Earth Sciences, 56(2), 99-110.\\nhttps://doi.org/10.1080/08120090802546977\\nCalais, E., Camelbeeck, T., Stein, S., Liu, M. & Craig, T.J. (2016)\\nA new para-digm for large earthquakes in stable continental plate interiors.\\nGeophysical Research Letters, 43(20), 10,621-10,637.\\nhttps://doi.org/10.1002/2016GL070815 Calais, E., Ma\\\"},{\\\"dataset_ids\\\": ['10.1038/nature04428'], citation_context: \\\"l Research Letters, 43(20), 10,621-10,637.\\nhttps://doi.org/10.1002/2016GL070815 Calais, E., Mattioli, G., DeMets, C., Nocquet, J.-M., Stein, S., Newman, A. & Rydelek, P. (2005) Tectonic strain in plate interiors?\\nNature, 438, E9-E10.\\nhttps://doi.org/10.1038/nature04428\\n[online] Available from: http://www.nature.com/articles/nature04428 Clark, D. (2012) Neotectonic Features Database.\\nCanberra, ACT: Geoscience Australia, Commonwealth of Australia.\\nClark, D. (2018)\\nWhat have we learned in the 50 years since the 1968 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/j.1475-4754.1999.tb00987.x'], citation_context: \\\" Laslett, G.M., Yoshida, H. & Olley, J.M. (1999) Optical dating of single and multiple grains of quartz from Jinmium rock shelter, northern Australia: Part I, experimental design and statistical models.\\nArchaeometry, 41(2), 339-364.\\nhttps://doi.\\norg/10.1111/j.1475-4754.1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104-117. https://doi.org/10.1177/095968\\\"},{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], citation_context: \\\"1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104-117. https://doi.org/10.1177/0959683613515732\\nGold, R.D., Briggs, R.W., Crone, A.J. & DuRoss, C.B. (2017) Refining fault slip rates using multiple displaced terrace risers-an example from the Honey Lake fault, NE California, USA.\\nEarth and Planetary Science Letters, 477, 134-146. https://doi.or\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/S0264-3707'], citation_context: \\\" in stable continental regions (USGS Bulletin 2032-B).\\nWashington, DC: US Geological Survey.\\nMachette, M.N. (2000) Active, capable, and potentially active faults - a paleoseismic perspective.\\nJournal of Geodynamics, 29(3-5), 387-392.\\nhttps://doi.org/10.1016/S0264-3707(99)00060-5 Magee, J.W. (2009)\\nA Literature Review Palaeovalley Groundwater Resources in Arid and Semi-Arid Australia - A literature review.\\nCanberra, Australia: Geoscience Australia.\\nhttp://pid.geoscience.gov.au/dataset/ga/68542 McCalpin, J.P. (2009\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5090'], citation_context: \\\"10969837, 2021, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5090, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract The 20 May 2016  M W  6.1 Petermann earthquake in central Australia generated a 21\\u2009km surface rupture with 0.1 to 1 m vertical displacements across a low\\u2010relief landscape. No paleo\\u2010scarps or potentially analogous topographic features are evident in pre\\u2010earthquake Worldview\\u20101 and Worldview\\u20102 satellite data. Two excavations across the surface rupture expose near\\u2010surface fault geometry and mixed aeolian\\u2010sheetwash sediment faulted only in the 2016 earthquake. A 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture. Seven cosmogenic beryllium\\u201010 ( 10 Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging\\u2010wall and foot\\u2010wall, and three from samples 19 to 50\\u2009km from the surface rupture. No distinction is found between fault proximal rates (1.3 \\u00b1 0.1 to 2.6 \\u00b1 0.2 m Myr \\u22121 ) and distal samples (1.4 \\u00b1 0.1 to 2.3 \\u00b1 0.2 m Myr \\u22121 ). The thickness of rock fragments (2\\u20135 cm) coseismically displaced in the Petermann earthquake perturbs the steady\\u2010state bedrock erosion rate by only 1 to 3%, less than the erosion rate uncertainty estimated for each sample (7\\u201312%). Using  10 Be erosion rates and scarp height measurements we estimate approximately 0.5 to 1 Myr of differential erosion is required to return to pre\\u2010earthquake topography. By inference any pre\\u20102016 fault\\u2010related topography likely required a similar time for removal. We conclude that the Petermann earthquake was the first on this fault in the last ca. 0.5\\u20131 Myr. Extrapolating single nuclide erosion rates across this timescale introduces large uncertainties, and we cannot resolve whether 2016 represents the first ever surface rupture on this fault, or a\\u2009> 1 Myr interseismic period. Either option reinforces the importance of including distributed earthquake sources in fault displacement and seismic hazard analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], citation_context: \\\"Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low\\u2010T thermochronological data|Australian Journal of Earth Sciences|https://doi.org/10.1080/08120090802546977\\\"}\",\"{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], citation_context: \\\"Late\\u2010Holocene climatic variability indicated by three natural archives in arid southern Australia|Holocene|https://doi.org/10.1177/0959683613515732\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2\\n|, Mark, Tamarah Rosellen King, Albert, Dan, Abaz\",\"abstract\":\"R E S E A R C H A R T I C L E Paleoseismology of the 2016 MW 6.1 Petermann earthquake source: Implications for intraplate earthquake behaviour and the geomorphic longevity of bedrock fault scarps in a low strain-rate cratonic region Tamarah Rosellen King1,2 | Mark Quigley1 | Dan Clark3 | Albert Zondervan4 | Jan-Hendrik May5 | Abaz Alimanovic1 1School of Earth Sciences, The University of Melbourne, Melbourne, Victoria, Australia 2COMET, Department of Earth Sciences, University of Oxford, Oxford, UK 3Geoscience Australia, Canberra, Australian Capital Territory, Australia 4Rafter Radiocarbon Laboratory, GNS Science, Lower Hutt, New Zealand 5School of Geography, The University of Melbourne, Melbourne, Victoria, Australia Correspondence Tamarah Rosellen King, School of Earth Sciences, The University of Melbourne, Melbourne, VIC 3010, Australia.\\nEmail: tamarah.king@earth.ox.ac.uk Funding information British Geological Survey; University of Melbourne; Australian Research Council, Grant/Award Number: DP170103350\\nAbstract The 20 May 2016 MW 6.1 Petermann earthquake in central Australia generated a 21 km surface rupture with 0.1 to 1 m vertical displacements across a low-relief landscape.\\nNo paleo-scarps or potentially analogous topographic features are evident in pre-earthquake Worldview-1 and Worldview-2 satellite data.\\nTwo excavations across the surface rupture expose near-surface fault geometry and mixed aeolian-sheetwash sediment faulted only in the 2016 earthquake.\\nA 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture.\\nSeven cosmogenic beryllium-10 (10Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging-wall and foot-\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], citation_context: \\\"& Cummins, P.R. (2009) Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low-T thermochronological data.\\nAustralian Journal of Earth Sciences, 56(2), 99-110.\\nhttps://doi.org/10.1080/08120090802546977\\nCalais, E., Camelbeeck, T., Stein, S., Liu, M. & Craig, T.J. (2016)\\nA new para-digm for large earthquakes in stable continental plate interiors.\\nGeophysical Research Letters, 43(20), 10,621-10,637.\\nhttps://doi.org/10.1002/2016GL070815 Calais, E., Ma\\\"},{\\\"dataset_ids\\\": ['10.1038/nature04428'], citation_context: \\\"l Research Letters, 43(20), 10,621-10,637.\\nhttps://doi.org/10.1002/2016GL070815 Calais, E., Mattioli, G., DeMets, C., Nocquet, J.-M., Stein, S., Newman, A. & Rydelek, P. (2005) Tectonic strain in plate interiors?\\nNature, 438, E9-E10.\\nhttps://doi.org/10.1038/nature04428\\n[online] Available from: http://www.nature.com/articles/nature04428 Clark, D. (2012) Neotectonic Features Database.\\nCanberra, ACT: Geoscience Australia, Commonwealth of Australia.\\nClark, D. (2018)\\nWhat have we learned in the 50 years since the 1968 \\\"}\",\"{\\\"dataset_ids\\\": ['10.1111/j.1475-4754.1999.tb00987.x'], citation_context: \\\" Laslett, G.M., Yoshida, H. & Olley, J.M. (1999) Optical dating of single and multiple grains of quartz from Jinmium rock shelter, northern Australia: Part I, experimental design and statistical models.\\nArchaeometry, 41(2), 339-364.\\nhttps://doi.\\norg/10.1111/j.1475-4754.1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104-117. https://doi.org/10.1177/095968\\\"},{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], citation_context: \\\"1999.tb00987.x Gliganic, L.A., Cohen, T.J., May, J.H., Jansen, J.D., Nanson, G.C., Dosseto, A. et al. (2014)\\nLate-Holocene climatic variability indicated by three natural archives in arid southern Australia.\\nHolocene, 24(1), 104-117. https://doi.org/10.1177/0959683613515732\\nGold, R.D., Briggs, R.W., Crone, A.J. & DuRoss, C.B. (2017) Refining fault slip rates using multiple displaced terrace risers-an example from the Honey Lake fault, NE California, USA.\\nEarth and Planetary Science Letters, 477, 134-146. https://doi.or\\\"}\",\"{\\\"dataset_ids\\\": ['10.1016/S0264-3707'], citation_context: \\\" in stable continental regions (USGS Bulletin 2032-B).\\nWashington, DC: US Geological Survey.\\nMachette, M.N. (2000) Active, capable, and potentially active faults - a paleoseismic perspective.\\nJournal of Geodynamics, 29(3-5), 387-392.\\nhttps://doi.org/10.1016/S0264-3707(99)00060-5 Magee, J.W. (2009)\\nA Literature Review Palaeovalley Groundwater Resources in Arid and Semi-Arid Australia - A literature review.\\nCanberra, Australia: Geoscience Australia.\\nhttp://pid.geoscience.gov.au/dataset/ga/68542 McCalpin, J.P. (2009\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5090'], citation_context: \\\"10969837, 2021, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5090, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract The 20 May 2016  M W  6.1 Petermann earthquake in central Australia generated a 21\\u2009km surface rupture with 0.1 to 1 m vertical displacements across a low\\u2010relief landscape. No paleo\\u2010scarps or potentially analogous topographic features are evident in pre\\u2010earthquake Worldview\\u20101 and Worldview\\u20102 satellite data. Two excavations across the surface rupture expose near\\u2010surface fault geometry and mixed aeolian\\u2010sheetwash sediment faulted only in the 2016 earthquake. A 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture. Seven cosmogenic beryllium\\u201010 ( 10 Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging\\u2010wall and foot\\u2010wall, and three from samples 19 to 50\\u2009km from the surface rupture. No distinction is found between fault proximal rates (1.3 \\u00b1 0.1 to 2.6 \\u00b1 0.2 m Myr \\u22121 ) and distal samples (1.4 \\u00b1 0.1 to 2.3 \\u00b1 0.2 m Myr \\u22121 ). The thickness of rock fragments (2\\u20135 cm) coseismically displaced in the Petermann earthquake perturbs the steady\\u2010state bedrock erosion rate by only 1 to 3%, less than the erosion rate uncertainty estimated for each sample (7\\u201312%). Using  10 Be erosion rates and scarp height measurements we estimate approximately 0.5 to 1 Myr of differential erosion is required to return to pre\\u2010earthquake topography. By inference any pre\\u20102016 fault\\u2010related topography likely required a similar time for removal. We conclude that the Petermann earthquake was the first on this fault in the last ca. 0.5\\u20131 Myr. Extrapolating single nuclide erosion rates across this timescale introduces large uncertainties, and we cannot resolve whether 2016 represents the first ever surface rupture on this fault, or a\\u2009> 1 Myr interseismic period. Either option reinforces the importance of including distributed earthquake sources in fault displacement and seismic hazard analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], citation_context: \\\"Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low\\u2010T thermochronological data|Australian Journal of Earth Sciences|https://doi.org/10.1080/08120090802546977\\\"}\",\"{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], citation_context: \\\"Late\\u2010Holocene climatic variability indicated by three natural archives in arid southern Australia|Holocene|https://doi.org/10.1177/0959683613515732\\\"}\"]}"
        ],
        [
         "27",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_mp.14424.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brandon J. Wang, Hesham Elhalawani, W. Jim Zheng, Sara Ahmed, Nathan S. Doyle, Kendall J. Kisera, Peter Y. S. Park\",\"abstract\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines Kendall J. Kisera)\\nJohn P. and Kathrine G. McGovern Medical School, Houston, TX, USA Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Sara Ahmed and Sonja Stieb Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Abdallah S. R. Mohamed Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Hesham Elhalawani Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA Peter Y. S. Park, Nathan S. Doyle, and Brandon J. Wang Department of Diagnostic and Interventional Imaging, John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Arko Barman, Zhao Li, and W. Jim Zheng Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Clifton D. Fullera) Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Luca Giancardoa) Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA (Received 7 April 2020; revised 22 July 2020; accepted for publication 27 July 2020; published 28 August 2020)\\nThis manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non-small cell lung cancer.\\nThe segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction.\\nBilateral thoracic cavity volumes and pleural effusion volumes were manually seg-me\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"chive (TCIA): maintaining and operating a public information repository.\\nJ Digit Imag-ing. 2013;26:1045-1057.\\n22.\\nAerts HJWL, Wee L, Rios Velazquez E, et al.\\nData from NSCLC-Radio-mics\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2019.\\nhttps://doi.org/10.7937/K9/TCIA.2015.PF0M9REI 23.\\nAerts HJ, Velazquez ER, Leijenaar RT, et al.\\nDecoding tumour pheno-type by noninvasive imaging using a quantitative radiomics approach.\\nNat Commun.\\n2014;5:4006.\\n24.\\nLi X, Morgan PS, Ashburner J, Smith J, Rorden C. dcm2niix.exe [com-puter program\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"51.\\nKiser KJ, Ahmed S, Stieb SM, et al.\\nData from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2020.\\nhttps://doi.org/10.7937/tcia.2020.6c7y-gq39 52.\\nRonneberger O, Fischer P, Brox T. U-Net: Convolutional Networks for Biomedical Image Segmentation.\\nPaper presented at: Medical Image Computing and Computer-Assisted Intervention - MICCAI 2015; 5-9 October, 2015; Munich, Germany.\\nhttps://doi.org/\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39.51'], citation_context: \\\"We describe PleThora, a dataset of 402 expert-vetted tho-racic cavity segmentations, 78 expert-vetted pleural effusion segmentations, and corresponding clinical and technical metadata made available to the public through TCIA at https://doi.org/10.7937/tcia.2020.6c7y-gq39.51\\nThese seg-mentations have value for preprocessing steps in image analy-sis pipelines built for fundamental quantitative imaging tasks, including but not limited to pathologic lung segmentation, lesion detection, and radiomics feature extraction.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/mp.14424'], citation_context: \\\"24734209, 2020, 11, Downloaded from https://aapm.onlinelibrary.wiley.com/doi/10.1002/mp.14424 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brandon J. Wang, Hesham Elhalawani, W. Jim Zheng, Sara Ahmed, Nathan S. Doyle, Kendall J. Kisera, Peter Y. S. Park\",\"abstract\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines Kendall J. Kisera)\\nJohn P. and Kathrine G. McGovern Medical School, Houston, TX, USA Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Sara Ahmed and Sonja Stieb Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Abdallah S. R. Mohamed Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Hesham Elhalawani Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA Peter Y. S. Park, Nathan S. Doyle, and Brandon J. Wang Department of Diagnostic and Interventional Imaging, John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Arko Barman, Zhao Li, and W. Jim Zheng Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Clifton D. Fullera) Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Luca Giancardoa) Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA (Received 7 April 2020; revised 22 July 2020; accepted for publication 27 July 2020; published 28 August 2020)\\nThis manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non-small cell lung cancer.\\nThe segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction.\\nBilateral thoracic cavity volumes and pleural effusion volumes were manually seg-me\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"chive (TCIA): maintaining and operating a public information repository.\\nJ Digit Imag-ing. 2013;26:1045-1057.\\n22.\\nAerts HJWL, Wee L, Rios Velazquez E, et al.\\nData from NSCLC-Radio-mics\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2019.\\nhttps://doi.org/10.7937/K9/TCIA.2015.PF0M9REI 23.\\nAerts HJ, Velazquez ER, Leijenaar RT, et al.\\nDecoding tumour pheno-type by noninvasive imaging using a quantitative radiomics approach.\\nNat Commun.\\n2014;5:4006.\\n24.\\nLi X, Morgan PS, Ashburner J, Smith J, Rorden C. dcm2niix.exe [com-puter program\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"51.\\nKiser KJ, Ahmed S, Stieb SM, et al.\\nData from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2020.\\nhttps://doi.org/10.7937/tcia.2020.6c7y-gq39 52.\\nRonneberger O, Fischer P, Brox T. U-Net: Convolutional Networks for Biomedical Image Segmentation.\\nPaper presented at: Medical Image Computing and Computer-Assisted Intervention - MICCAI 2015; 5-9 October, 2015; Munich, Germany.\\nhttps://doi.org/\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39.51'], citation_context: \\\"We describe PleThora, a dataset of 402 expert-vetted tho-racic cavity segmentations, 78 expert-vetted pleural effusion segmentations, and corresponding clinical and technical metadata made available to the public through TCIA at https://doi.org/10.7937/tcia.2020.6c7y-gq39.51\\nThese seg-mentations have value for preprocessing steps in image analy-sis pipelines built for fundamental quantitative imaging tasks, including but not limited to pathologic lung segmentation, lesion detection, and radiomics feature extraction.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/mp.14424'], citation_context: \\\"24734209, 2020, 11, Downloaded from https://aapm.onlinelibrary.wiley.com/doi/10.1002/mp.14424 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}"
        ],
        [
         "28",
         "10.1002_nafm.10870",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_nafm.10870.pdf",
         null,
         "test",
         "[{'dataset_id': 'https://doi.org/10.5066/p9gtumay', 'type': 'Primary'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"John R. Farver3 |, Patrick M. Ko\\u010dovsk\\u00fd,6, Nathan D. Stott5, Wiley Periodicals LLC, James M. Long4, Revised, Sandusky, Jeffrey G. Miner5, Silver Chub\",\"abstract\":\"Abstract Objective: Silver Chub Macrhybopsis storeriana, a predominately riverine species  throughout its native range, exists within Lake Erie as the only known lake popu-lation.\\nIts population declined in the 1950s and never fully recovered.\\nCanada has  listed Silver Chub in the Great Lakes-\\u00adSt.\\nLawrence River as endangered and has  initiated a recovery plan that recognized the identification of spawning areas as a  critical component to inform Silver Chub's recovery potential.\\nMethods: We investigated potential spawning locations of Silver Chub using cap-ture records, otolith microchemistry, and daily age analysis.\\nLapillus otolith\\nSr:Ca ra-tios from 27 age-\\u00ad0 Silver Chub were used to identify potential spawning areas.\\nDaily  ages estimated from lapilli were used to calculate hatch dates, which then were com-pared with capture data of adults and river flows to further inform potential spawn-ing areas.\\nResult: The Detroit River (and its nearshore area) was all but ruled out as a potential  spawning location.\\nThe Maumee, Portage, and Sandusky rivers or their nearshore  areas were all possible spawning locations.\\nProjected hatch dates spanned the end  of May through the end of June and occurred across a wide range of flows, although  some peaks in hatch dates corresponded to flow peaks, indicating recruitment is po-tentially enhanced by high flows.\\nConclusion: Silver Chub spawning period and hypothesized spawning rivers or  lacustuaries overlap those of invasive Grass Carp Ctenopharyngodon idella, creat-ing a need to jointly consider Grass Carp control efforts with conservation of Silver  Chub when assessing management alternatives.\\nFurther research on spawning  guild and the use of rivers themselves or nearshore areas influenced by rivers as  spawning areas are required to maximize potential for conservation and recovery  of Silver Chub.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT U.S. Geological Survey data are publicly available at the  websites in the references section.\\nWestern Basin intera-gency August trawl survey data are available from the For-age Task Group of the Lake Erie Committee (http://www. glfc.org/lake-erie-committee.php) by request.\\nData col-lected by the U.S. Fish and Wildlife Service are available  upon request from Anjanette Bowen (anjanette_bowen@ fws.gov).\\nSilver Chub otolith microchemistry data\\nare  available upon request from John Farver or Jeff Miner  (jfarver@bgsu.edu or jminer@bgsu.edu, respectively).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5066/P9GTUMAY.'], citation_context: \\\"associated species.\\nDoctoral dissertation.\\nThe Ohio State  University, Columbus.\\nKo\\u010dovsk\\u00fd, P. M., and J. Farver. 2019.\\nWater chemistry of Great Lakes  tributaries, 2017-\\u00ad2018.\\nU.S. Geological Survey data release.  \\nAvailable: https://doi.org/10.5066/P9GTUMAY.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. o\\\"},{\\\"dataset_ids\\\": ['10.5066/P94BH3W0.'], citation_context: \\\"\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. org/10.5066/P94BH3W0.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, E. Weimer, C. Mayer, and S. S. Qian.  2021.\\nValidation of the model-\\u00adpredicted spawning area of Grass  Carp Ctenopharyngodon idella in the Sandusky River.\\nJournal  of Great Lakes Research 47:29-36.\\nKoel, T. M., and R. E.\\\"}\",\"{\\\"dataset_ids\\\": ['10.5066/F75M63X0.'], citation_context: \\\" Fisheries Society, Bethesda, Maryland.\\nUSGS (U.S. Geological Survey). 2019a.\\nGreat Lakes Science Center  Great Lakes research vessel operations 1958-\\u00ad2018.\\n(version  3.0, April 2019).\\nU.S. Geological Survey data release.\\nAvailable:  https://doi.org/10.5066/F75M63X0.\\nUSGS (U.S. Geological Survey). 2019b.\\nNewly hatched invasive Grass  Carp found in Maumee River, OH.\\nU.S. Geological Survey,  Office of Communications and Publishing, Reston, Virginia.  \\nAvailable:\\nwww.usgs.gov/news/newly-\\u00adhatched-\\u00adinvasive-\\u00adgrass -\\u00ad\\\"}\"]}",
         "",
         "{\"title\":\"\",\"authors\":\"John R. Farver3 |, Patrick M. Ko\\u010dovsk\\u00fd,6, Nathan D. Stott5, Wiley Periodicals LLC, James M. Long4, Revised, Sandusky, Jeffrey G. Miner5, Silver Chub\",\"abstract\":\"Abstract Objective: Silver Chub Macrhybopsis storeriana, a predominately riverine species  throughout its native range, exists within Lake Erie as the only known lake popu-lation.\\nIts population declined in the 1950s and never fully recovered.\\nCanada has  listed Silver Chub in the Great Lakes-\\u00adSt.\\nLawrence River as endangered and has  initiated a recovery plan that recognized the identification of spawning areas as a  critical component to inform Silver Chub's recovery potential.\\nMethods: We investigated potential spawning locations of Silver Chub using cap-ture records, otolith microchemistry, and daily age analysis.\\nLapillus otolith\\nSr:Ca ra-tios from 27 age-\\u00ad0 Silver Chub were used to identify potential spawning areas.\\nDaily  ages estimated from lapilli were used to calculate hatch dates, which then were com-pared with capture data of adults and river flows to further inform potential spawn-ing areas.\\nResult: The Detroit River (and its nearshore area) was all but ruled out as a potential  spawning location.\\nThe Maumee, Portage, and Sandusky rivers or their nearshore  areas were all possible spawning locations.\\nProjected hatch dates spanned the end  of May through the end of June and occurred across a wide range of flows, although  some peaks in hatch dates corresponded to flow peaks, indicating recruitment is po-tentially enhanced by high flows.\\nConclusion: Silver Chub spawning period and hypothesized spawning rivers or  lacustuaries overlap those of invasive Grass Carp Ctenopharyngodon idella, creat-ing a need to jointly consider Grass Carp control efforts with conservation of Silver  Chub when assessing management alternatives.\\nFurther research on spawning  guild and the use of rivers themselves or nearshore areas influenced by rivers as  spawning areas are required to maximize potential for conservation and recovery  of Silver Chub.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT U.S. Geological Survey data are publicly available at the  websites in the references section.\\nWestern Basin intera-gency August trawl survey data are available from the For-age Task Group of the Lake Erie Committee (http://www. glfc.org/lake-erie-committee.php) by request.\\nData col-lected by the U.S. Fish and Wildlife Service are available  upon request from Anjanette Bowen (anjanette_bowen@ fws.gov).\\nSilver Chub otolith microchemistry data\\nare  available upon request from John Farver or Jeff Miner  (jfarver@bgsu.edu or jminer@bgsu.edu, respectively).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5066/P9GTUMAY.'], citation_context: \\\"associated species.\\nDoctoral dissertation.\\nThe Ohio State  University, Columbus.\\nKo\\u010dovsk\\u00fd, P. M., and J. Farver. 2019.\\nWater chemistry of Great Lakes  tributaries, 2017-\\u00ad2018.\\nU.S. Geological Survey data release.  \\nAvailable: https://doi.org/10.5066/P9GTUMAY.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. o\\\"},{\\\"dataset_ids\\\": ['10.5066/P94BH3W0.'], citation_context: \\\"\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. org/10.5066/P94BH3W0.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, E. Weimer, C. Mayer, and S. S. Qian.  2021.\\nValidation of the model-\\u00adpredicted spawning area of Grass  Carp Ctenopharyngodon idella in the Sandusky River.\\nJournal  of Great Lakes Research 47:29-36.\\nKoel, T. M., and R. E.\\\"}\",\"{\\\"dataset_ids\\\": ['10.5066/F75M63X0.'], citation_context: \\\" Fisheries Society, Bethesda, Maryland.\\nUSGS (U.S. Geological Survey). 2019a.\\nGreat Lakes Science Center  Great Lakes research vessel operations 1958-\\u00ad2018.\\n(version  3.0, April 2019).\\nU.S. Geological Survey data release.\\nAvailable:  https://doi.org/10.5066/F75M63X0.\\nUSGS (U.S. Geological Survey). 2019b.\\nNewly hatched invasive Grass  Carp found in Maumee River, OH.\\nU.S. Geological Survey,  Office of Communications and Publishing, Reston, Virginia.  \\nAvailable:\\nwww.usgs.gov/news/newly-\\u00adhatched-\\u00adinvasive-\\u00adgrass -\\u00ad\\\"}\"]}\n"
        ],
        [
         "29",
         "10.1007_jhep07(2018)134",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1007_jhep07(2018)134.pdf",
         null,
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"abstract\":\"Abstract: The production cross-sections of \\u03a5(1S), \\u03a5(2S) and \\u03a5(3S) mesons in proton-proton collisions at \\u221as = 13 TeV are measured with a data sample corresponding to an integrated luminosity of 277 \\u00b1 11 pb\\u22121 recorded by the LHCb experiment in 2015.\\nThe \\u03a5 mesons are reconstructed in the decay mode \\u03a5 \\u2192\\u00b5+\\u00b5\\u2212.\\nThe differential production cross-sections times the dimuon branching fractions are measured as a function of the \\u03a5 transverse momentum, pT, and rapidity, y, over the range 0 < pT < 30 GeV/c and 2.0 < y < 4.5.\\nThe ratios of the cross-sections with respect to the LHCb measurement at \\u221as = 8 TeV are also determined.\\nThe measurements are compared with theoretical predictions based on NRQCD.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}",
         "",
         "{\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"abstract\":\"Abstract: The production cross-sections of \\u03a5(1S), \\u03a5(2S) and \\u03a5(3S) mesons in proton-proton collisions at \\u221as = 13 TeV are measured with a data sample corresponding to an integrated luminosity of 277 \\u00b1 11 pb\\u22121 recorded by the LHCb experiment in 2015.\\nThe \\u03a5 mesons are reconstructed in the decay mode \\u03a5 \\u2192\\u00b5+\\u00b5\\u2212.\\nThe differential production cross-sections times the dimuon branching fractions are measured as a function of the \\u03a5 transverse momentum, pT, and rapidity, y, over the range 0 < pT < 30 GeV/c and 2.0 < y < 4.5.\\nThe ratios of the cross-sections with respect to the LHCb measurement at \\u221as = 8 TeV are also determined.\\nThe measurements are compared with theoretical predictions based on NRQCD.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n"
        ]
       ],
       "shape": {
        "columns": 10,
        "rows": 30
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "      <td>{\"title\":\"Assessing the variability in the rel...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "      <td>{\"title\":\"Effective and Reversible Carbon Diox...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE collection SYSTEM \"BioC.dtd\"&gt;&lt;collec...</td>\n",
       "      <td>bioc</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "      <td>{\"title\":\"Trifluoromethyl Sulfoxides: Reagents...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximi...</td>\n",
       "      <td>{\"title\":\"Metal\\u2013Ligand Cooperativity of t...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, ...</td>\n",
       "      <td>{\"title\":\"A Synthetic Route Toward Tetrazoles:...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10.1002_chem.201903120</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Matthias, Philippe R. M...</td>\n",
       "      <td>{\"title\":\"Structure\\u2013Solubility Relationsh...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Matthias, Philippe R. M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10.1002_chem.202000235</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3,...</td>\n",
       "      <td>{\"title\":\"An Isolable Bis(Silanone\\u2013Borane...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10.1002_chem.202001412</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u0...</td>\n",
       "      <td>{\"title\":\"Iron(0)\\u2010Mediated Stereoselectiv...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10.1002_chem.202001668</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, ...</td>\n",
       "      <td>{\"title\":\"The First Use of a ReX 5  Synthon to...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10.1002_chem.202003167</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bo...</td>\n",
       "      <td>{\"title\":\"Metal\\u2010only Lewis Pairs of Rhodi...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10.1002_cssc.202201821</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u...</td>\n",
       "      <td>{\"title\":\"Development of a Mg/O ReaxFF Potenti...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10.1002_ece3.3985</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Jonathan P. Evans, Wats...</td>\n",
       "      <td>{\"title\":\"High levels of polyandry, but limite...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Jonathan P. Evans, Wats...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10.1002_ece3.4466</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Mats Olsson, Revised, S...</td>\n",
       "      <td>{\"title\":\"Incubation temperature and parental ...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Mats Olsson, Revised, S...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10.1002_ece3.5260</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, J...</td>\n",
       "      <td>{\"title\":\"Ultraconserved element (UCE) probe s...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, J...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10.1002_ece3.5395</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Evolution, Andrea Flack...</td>\n",
       "      <td>{\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Evolution, Andrea Flack...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10.1002_ece3.6144</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...</td>\n",
       "      <td>{\"title\":\"Efficacy of metabarcoding for identi...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10.1002_ece3.6303</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. ...</td>\n",
       "      <td>{\"title\":\"Can polyploidy confer invasive plant...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10.1002_ece3.6784</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Thomas, Evolutionsforsc...</td>\n",
       "      <td>{\"title\":\"Ecomorphology of the pectoral girdle...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Thomas, Evolutionsforsc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>10.1002_ece3.961</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney H...</td>\n",
       "      <td>{\"title\":\"The effects of insects, nutrients, a...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10.1002_ece3.9627</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2,...</td>\n",
       "      <td>{\"title\":\"Inferring predator\\u2013prey interac...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10.1002_ecs2.1280</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>10.1002_ecs2.4619</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.25349/d9qw...</td>\n",
       "      <td>&lt;html&gt;&lt;body&gt;&lt;tei xml:space=\"preserve\" xmlns=\"h...</td>\n",
       "      <td>tei</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis...</td>\n",
       "      <td>{\"title\":\"Interactions between temperature and...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10.1002_ejic.201900904</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;component xmlns=\"http://www.wiley.com/namespa...</td>\n",
       "      <td>wiley</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"A...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Chemical ...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10.1002_ejoc.202000139</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>10.1002_ejoc.202000916</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract:...</td>\n",
       "      <td>{\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10.1002_esp.5058</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>10.1002_esp.5090</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5066/p9353...</td>\n",
       "      <td>&lt;component xmlns=\"http://www.wiley.com/namespa...</td>\n",
       "      <td>wiley</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract ...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10.1002_mp.14424</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7937/k9/tc...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...</td>\n",
       "      <td>{\"title\":\"PleThora: Pleural effusion and thora...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>10.1002_nafm.10870</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5066/p9gtu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"John R. Farver3 |, Patr...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"John R. Farver3 |, Patr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>10.1007_jhep07(2018)134</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"a...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"a...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "0      10.1002_2017jc013030   \n",
       "1    10.1002_anie.201916483   \n",
       "2    10.1002_anie.202005531   \n",
       "3    10.1002_anie.202007717   \n",
       "4    10.1002_chem.201902131   \n",
       "5    10.1002_chem.201903120   \n",
       "6    10.1002_chem.202000235   \n",
       "7    10.1002_chem.202001412   \n",
       "8    10.1002_chem.202001668   \n",
       "9    10.1002_chem.202003167   \n",
       "10   10.1002_cssc.202201821   \n",
       "11        10.1002_ece3.3985   \n",
       "12        10.1002_ece3.4466   \n",
       "13        10.1002_ece3.5260   \n",
       "14        10.1002_ece3.5395   \n",
       "15        10.1002_ece3.6144   \n",
       "16        10.1002_ece3.6303   \n",
       "17        10.1002_ece3.6784   \n",
       "18         10.1002_ece3.961   \n",
       "19        10.1002_ece3.9627   \n",
       "20        10.1002_ecs2.1280   \n",
       "21        10.1002_ecs2.4619   \n",
       "22   10.1002_ejic.201900904   \n",
       "23   10.1002_ejoc.202000139   \n",
       "24   10.1002_ejoc.202000916   \n",
       "25         10.1002_esp.5058   \n",
       "26         10.1002_esp.5090   \n",
       "27         10.1002_mp.14424   \n",
       "28       10.1002_nafm.10870   \n",
       "29  10.1007_jhep07(2018)134   \n",
       "\n",
       "                                        pdf_file_path  \\\n",
       "0   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "3   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "4   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "5   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "6   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "7   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "8   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "9   ./kaggle/input/make-data-count-finding-data-re...   \n",
       "10  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "11  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "12  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "13  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "14  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "15  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "16  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "17  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "18  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "19  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "20  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "21  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "22  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "23  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "24  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "25  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "26  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "27  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "28  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "29  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                        xml_file_path dataset_type  \\\n",
       "0   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "1   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "3   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "4   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "5   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "6   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "7   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "8   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "9   ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "10  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "11  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "12  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "13  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "14  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "15  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "16  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "17  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "18  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "19  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "20                                                NaN         test   \n",
       "21  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "22  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "23                                                NaN         test   \n",
       "24  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "25                                                NaN         test   \n",
       "26  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "27  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "28                                                NaN         test   \n",
       "29                                                NaN         test   \n",
       "\n",
       "                                         dataset_info  \\\n",
       "0   [{'dataset_id': 'https://doi.org/10.17882/4938...   \n",
       "1      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "3      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "4      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "5      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "6      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "7      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "8      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "9      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "10                                                NaN   \n",
       "11     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "12  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "13  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "14     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "15  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "16  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "17     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "18     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "19  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "20  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "21  [{'dataset_id': 'https://doi.org/10.25349/d9qw...   \n",
       "22     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "23     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "24     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "25  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "26  [{'dataset_id': 'https://doi.org/10.5066/p9353...   \n",
       "27  [{'dataset_id': 'https://doi.org/10.7937/k9/tc...   \n",
       "28  [{'dataset_id': 'https://doi.org/10.5066/p9gtu...   \n",
       "29     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                       xml_first_line xml_type  \\\n",
       "0   <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "1   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2   <!DOCTYPE collection SYSTEM \"BioC.dtd\"><collec...     bioc   \n",
       "3   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "4   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "5   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "6   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "7   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "8   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "9   <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "10  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "11  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "12  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "13  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "14  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "15  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "16  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "17  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "18  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "19  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "20                                                NaN            \n",
       "21  <html><body><tei xml:space=\"preserve\" xmlns=\"h...      tei   \n",
       "22  <component xmlns=\"http://www.wiley.com/namespa...    wiley   \n",
       "23                                                NaN            \n",
       "24  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "25                                                NaN            \n",
       "26  <component xmlns=\"http://www.wiley.com/namespa...    wiley   \n",
       "27  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "28                                                NaN            \n",
       "29                                                NaN            \n",
       "\n",
       "                                             pdf_text  \\\n",
       "0   {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...   \n",
       "1   {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...   \n",
       "2   {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...   \n",
       "3   {\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximi...   \n",
       "4   {\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, ...   \n",
       "5   {\"title\":\"\",\"authors\":\"Matthias, Philippe R. M...   \n",
       "6   {\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3,...   \n",
       "7   {\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u0...   \n",
       "8   {\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, ...   \n",
       "9   {\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bo...   \n",
       "10  {\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u...   \n",
       "11  {\"title\":\"\",\"authors\":\"Jonathan P. Evans, Wats...   \n",
       "12  {\"title\":\"\",\"authors\":\"Mats Olsson, Revised, S...   \n",
       "13  {\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, J...   \n",
       "14  {\"title\":\"\",\"authors\":\"Evolution, Andrea Flack...   \n",
       "15  {\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...   \n",
       "16  {\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. ...   \n",
       "17  {\"title\":\"\",\"authors\":\"Thomas, Evolutionsforsc...   \n",
       "18  {\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney H...   \n",
       "19  {\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2,...   \n",
       "20  {\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De...   \n",
       "21  {\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis...   \n",
       "22  {\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"A...   \n",
       "23  {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...   \n",
       "24  {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract:...   \n",
       "25  {\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark...   \n",
       "26  {\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2...   \n",
       "27  {\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...   \n",
       "28  {\"title\":\"\",\"authors\":\"John R. Farver3 |, Patr...   \n",
       "29  {\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"a...   \n",
       "\n",
       "                                             xml_text  \\\n",
       "0   {\"title\":\"Assessing the variability in the rel...   \n",
       "1   {\"title\":\"Effective and Reversible Carbon Diox...   \n",
       "2   {\"title\":\"Trifluoromethyl Sulfoxides: Reagents...   \n",
       "3   {\"title\":\"Metal\\u2013Ligand Cooperativity of t...   \n",
       "4   {\"title\":\"A Synthetic Route Toward Tetrazoles:...   \n",
       "5   {\"title\":\"Structure\\u2013Solubility Relationsh...   \n",
       "6   {\"title\":\"An Isolable Bis(Silanone\\u2013Borane...   \n",
       "7   {\"title\":\"Iron(0)\\u2010Mediated Stereoselectiv...   \n",
       "8   {\"title\":\"The First Use of a ReX 5  Synthon to...   \n",
       "9   {\"title\":\"Metal\\u2010only Lewis Pairs of Rhodi...   \n",
       "10  {\"title\":\"Development of a Mg/O ReaxFF Potenti...   \n",
       "11  {\"title\":\"High levels of polyandry, but limite...   \n",
       "12  {\"title\":\"Incubation temperature and parental ...   \n",
       "13  {\"title\":\"Ultraconserved element (UCE) probe s...   \n",
       "14  {\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d...   \n",
       "15  {\"title\":\"Efficacy of metabarcoding for identi...   \n",
       "16  {\"title\":\"Can polyploidy confer invasive plant...   \n",
       "17  {\"title\":\"Ecomorphology of the pectoral girdle...   \n",
       "18  {\"title\":\"The effects of insects, nutrients, a...   \n",
       "19  {\"title\":\"Inferring predator\\u2013prey interac...   \n",
       "20                                                      \n",
       "21  {\"title\":\"Interactions between temperature and...   \n",
       "22  {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Chemical ...   \n",
       "23                                                      \n",
       "24  {\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2...   \n",
       "25                                                      \n",
       "26  {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract ...   \n",
       "27  {\"title\":\"PleThora: Pleural effusion and thora...   \n",
       "28                                                      \n",
       "29                                                      \n",
       "\n",
       "                                                 text  \n",
       "0   {\"title\":\"\",\"authors\":\"Annick Bricaud, Bernard...  \n",
       "1   {\"title\":\"\",\"authors\":\"Reiner Anwander, Daniel...  \n",
       "2   {\"title\":\"\",\"authors\":\"Gregory J. P. Perry, Tr...  \n",
       "3   {\"title\":\"\",\"authors\":\"Al\\ufffdO, Lukas Maximi...  \n",
       "4   {\"title\":\"\",\"authors\":\"Nitrogen Heterocycles, ...  \n",
       "5   {\"title\":\"\",\"authors\":\"Matthias, Philippe R. M...  \n",
       "6   {\"title\":\"\",\"authors\":\"Borane, Lewis acids.[3,...  \n",
       "7   {\"title\":\"\",\"authors\":\"R. Huisgen, Helmar G\\u0...  \n",
       "8   {\"title\":\"\",\"authors\":\"Anthony B. Carter+,[a, ...  \n",
       "9   {\"title\":\"\",\"authors\":\"Mar, RhI, bimetallic bo...  \n",
       "10  {\"title\":\"\",\"authors\":\"Timo Jacob*[a, Mg(101\\u...  \n",
       "11  {\"title\":\"\",\"authors\":\"Jonathan P. Evans, Wats...  \n",
       "12  {\"title\":\"\",\"authors\":\"Mats Olsson, Revised, S...  \n",
       "13  {\"title\":\"\",\"authors\":\"Andrew E. Z. Short,2, J...  \n",
       "14  {\"title\":\"\",\"authors\":\"Evolution, Andrea Flack...  \n",
       "15  {\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...  \n",
       "16  {\"title\":\"\",\"authors\":\"Jizhong Wan,2, Ayub M. ...  \n",
       "17  {\"title\":\"\",\"authors\":\"Thomas, Evolutionsforsc...  \n",
       "18  {\"title\":\"\",\"authors\":\"Phoebe Wright, Dabney H...  \n",
       "19  {\"title\":\"\",\"authors\":\"Matthew Scott Luskin,2,...  \n",
       "20  {\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De...  \n",
       "21  {\"title\":\"\",\"authors\":\"Manuel T. Lerdau, Davis...  \n",
       "22  {\"title\":\"\",\"authors\":\"Adelheid\",\"abstract\":\"A...  \n",
       "23  {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...  \n",
       "24  {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract:...  \n",
       "25  {\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark...  \n",
       "26  {\"title\":\"\",\"authors\":\"Tamarah Rosellen King,2...  \n",
       "27  {\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...  \n",
       "28  {\"title\":\"\",\"authors\":\"John R. Farver3 |, Patr...  \n",
       "29  {\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"a...  "
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Fill NaN values in the 'xml_type' column with None\n",
    "train_file_paths_df['xml_type'] = train_file_paths_df['xml_type'].fillna('')\n",
    "train_file_paths_df['xml_text'] = train_file_paths_df['xml_text'].fillna('')\n",
    "test_file_paths_df['xml_type'] = test_file_paths_df['xml_type'].fillna('')\n",
    "test_file_paths_df['xml_text'] = test_file_paths_df['xml_text'].fillna('')\n",
    "test_file_paths_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "76e021a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_file_path",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_file_path",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "dataset_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_first_line",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "xml_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "pdf_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "xml_text",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "48e02cf4-02a7-4107-81c1-3faf1fdb8522",
       "rows": [
        [
         "0",
         "10.1002_mp.14424",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_mp.14424.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_mp.14424.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brandon J. Wang, Hesham Elhalawani, W. Jim Zheng, Sara Ahmed, Nathan S. Doyle, Kendall J. Kisera, Peter Y. S. Park\",\"abstract\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines Kendall J. Kisera)\\nJohn P. and Kathrine G. McGovern Medical School, Houston, TX, USA Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Sara Ahmed and Sonja Stieb Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Abdallah S. R. Mohamed Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Hesham Elhalawani Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA Peter Y. S. Park, Nathan S. Doyle, and Brandon J. Wang Department of Diagnostic and Interventional Imaging, John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Arko Barman, Zhao Li, and W. Jim Zheng Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Clifton D. Fullera) Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Luca Giancardoa) Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA (Received 7 April 2020; revised 22 July 2020; accepted for publication 27 July 2020; published 28 August 2020)\\nThis manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non-small cell lung cancer.\\nThe segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction.\\nBilateral thoracic cavity volumes and pleural effusion volumes were manually seg-me\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"chive (TCIA): maintaining and operating a public information repository.\\nJ Digit Imag-ing. 2013;26:1045-1057.\\n22.\\nAerts HJWL, Wee L, Rios Velazquez E, et al.\\nData from NSCLC-Radio-mics\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2019.\\nhttps://doi.org/10.7937/K9/TCIA.2015.PF0M9REI 23.\\nAerts HJ, Velazquez ER, Leijenaar RT, et al.\\nDecoding tumour pheno-type by noninvasive imaging using a quantitative radiomics approach.\\nNat Commun.\\n2014;5:4006.\\n24.\\nLi X, Morgan PS, Ashburner J, Smith J, Rorden C. dcm2niix.exe [com-puter program\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"51.\\nKiser KJ, Ahmed S, Stieb SM, et al.\\nData from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2020.\\nhttps://doi.org/10.7937/tcia.2020.6c7y-gq39 52.\\nRonneberger O, Fischer P, Brox T. U-Net: Convolutional Networks for Biomedical Image Segmentation.\\nPaper presented at: Medical Image Computing and Computer-Assisted Intervention - MICCAI 2015; 5-9 October, 2015; Munich, Germany.\\nhttps://doi.org/\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39.51'], citation_context: \\\"We describe PleThora, a dataset of 402 expert-vetted tho-racic cavity segmentations, 78 expert-vetted pleural effusion segmentations, and corresponding clinical and technical metadata made available to the public through TCIA at https://doi.org/10.7937/tcia.2020.6c7y-gq39.51\\nThese seg-mentations have value for preprocessing steps in image analy-sis pipelines built for fundamental quantitative imaging tasks, including but not limited to pathologic lung segmentation, lesion detection, and radiomics feature extraction.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/mp.14424'], citation_context: \\\"24734209, 2020, 11, Downloaded from https://aapm.onlinelibrary.wiley.com/doi/10.1002/mp.14424 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}",
         "{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brandon J. Wang, Hesham Elhalawani, W. Jim Zheng, Sara Ahmed, Nathan S. Doyle, Kendall J. Kisera, Peter Y. S. Park\",\"abstract\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines Kendall J. Kisera)\\nJohn P. and Kathrine G. McGovern Medical School, Houston, TX, USA Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Sara Ahmed and Sonja Stieb Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA Abdallah S. R. Mohamed Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Hesham Elhalawani Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA Peter Y. S. Park, Nathan S. Doyle, and Brandon J. Wang Department of Diagnostic and Interventional Imaging, John P. and Kathrine G. McGovern Medical School, Houston, TX, USA Arko Barman, Zhao Li, and W. Jim Zheng Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Clifton D. Fullera) Department of Radiation Oncology, University of Texas MD Anderson Cancer Center, Houston, TX, USA MD Anderson Cancer Center-UTHealth Graduate School of Biomedical Sciences, Houston, TX, USA Luca Giancardoa) Center for Precision Health, UTHealth School of Biomedical Informatics, Houston, TX, USA Department of Radiation Oncology, Cleveland Clinic Taussig Cancer Center, Cleveland, OH, USA (Received 7 April 2020; revised 22 July 2020; accepted for publication 27 July 2020; published 28 August 2020)\\nThis manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non-small cell lung cancer.\\nThe segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction.\\nBilateral thoracic cavity volumes and pleural effusion volumes were manually seg-me\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"chive (TCIA): maintaining and operating a public information repository.\\nJ Digit Imag-ing. 2013;26:1045-1057.\\n22.\\nAerts HJWL, Wee L, Rios Velazquez E, et al.\\nData from NSCLC-Radio-mics\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2019.\\nhttps://doi.org/10.7937/K9/TCIA.2015.PF0M9REI 23.\\nAerts HJ, Velazquez ER, Leijenaar RT, et al.\\nDecoding tumour pheno-type by noninvasive imaging using a quantitative radiomics approach.\\nNat Commun.\\n2014;5:4006.\\n24.\\nLi X, Morgan PS, Ashburner J, Smith J, Rorden C. dcm2niix.exe [com-puter program\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"51.\\nKiser KJ, Ahmed S, Stieb SM, et al.\\nData from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines\\n[Dataset].\\nIn: The Cancer Imaging Archive; 2020.\\nhttps://doi.org/10.7937/tcia.2020.6c7y-gq39 52.\\nRonneberger O, Fischer P, Brox T. U-Net: Convolutional Networks for Biomedical Image Segmentation.\\nPaper presented at: Medical Image Computing and Computer-Assisted Intervention - MICCAI 2015; 5-9 October, 2015; Munich, Germany.\\nhttps://doi.org/\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39.51'], citation_context: \\\"We describe PleThora, a dataset of 402 expert-vetted tho-racic cavity segmentations, 78 expert-vetted pleural effusion segmentations, and corresponding clinical and technical metadata made available to the public through TCIA at https://doi.org/10.7937/tcia.2020.6c7y-gq39.51\\nThese seg-mentations have value for preprocessing steps in image analy-sis pipelines built for fundamental quantitative imaging tasks, including but not limited to pathologic lung segmentation, lesion detection, and radiomics feature extraction.\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/mp.14424'], citation_context: \\\"24734209, 2020, 11, Downloaded from https://aapm.onlinelibrary.wiley.com/doi/10.1002/mp.14424 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}"
        ],
        [
         "1",
         "10.1002_ece3.6144",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.6144.pdf",
         "./kaggle/input/make-data-count-finding-data-references\\test\\XML\\10.1002_ece3.6144.xml",
         "test",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z39.96) Journal Archiving and Interchange DTD ",
         "jats",
         "{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |, Ronald S. Burton, Evolution\",\"abstract\":\"Ecology and Evolution.\\n2020;10:3463-3476.  \\n\\ufeff\\b   |  3463 www.ecolevol.org 1 | INTRODUCTION Marine ecosystems can change rapidly in response to both natural  and anthropogenic changes in the environment.\\nMonitoring changes  in the abundance and distribution of marine organisms is a chal-lenging but critical component of conservation efforts and natural  resource management.\\nHuman impacts such as overfishing, coastal  development, and pollution can rapidly increase stress on ecosys-tems resulting in an urgent need to develop efficient methods for  monitoring spatial and temporal dynamics of biodiversity (Beng  et al., 2016).\\nFisheries management has long used ichthyoplankton sur-veys as a component of monitoring because sampling the early  life stages of fish reflects the reproductive activities of fish    Received: 17 July 2019 | Revised:\\n31 January 2020 |\\nAccepted: 4 February 2020 DOI: 10.1002/ece3.6144   O R\\nI G I N A L  R E S E A R C H Efficacy of metabarcoding for identification of fish eggs  evaluated with mock communities Elena M. Duke  |   Ronald S. Burton\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Marine Biology Research Division, Scripps  Institution of Oceanography, University of  California, San Diego, La Jolla, California Correspondence Elena M. Duke, Marine Biology  Research Division, Scripps Institution of  Oceanography, University of California, San  Diego, La Jolla, CA.\\nEmail: emduke@ucsd.edu\\nAbstract There is urgent need for effective and efficient monitoring of marine fish popula-tions.\\nMo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1371/journal.pone.0134647'], citation_context: \\\"ermsmeier, M. C., Rogowski, P. A., Terrill,  E., Burton, R. S., & Bernardi, G. (2015).\\nMonitoring spawning ac-tivity in a Southern California marine protected area using molec-ular identification of fish eggs.\\nPLoS ONE, 10(8), 1-21.\\nhttps://doi.\\norg/10.1371/journal.pone.0134647 Hastings, P. A., & Burton, R. S. (2008).\\nEstablishing a DNA Sequence  Database for the Marine Fish Fauna of California.\\nRetrieved from  https://escholarship.org/uc/item/1ck9b3qs Hatzenbuhler, C., Kelly, J. R., Martinson, J., Okum, S., & Pilgrim, E. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6144'], citation_context: \\\"20457758, 2020, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6144, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "{\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}",
         "{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |, Ronald S. Burton, Evolution\",\"abstract\":\"Ecology and Evolution.\\n2020;10:3463-3476.  \\n\\ufeff\\b   |  3463 www.ecolevol.org 1 | INTRODUCTION Marine ecosystems can change rapidly in response to both natural  and anthropogenic changes in the environment.\\nMonitoring changes  in the abundance and distribution of marine organisms is a chal-lenging but critical component of conservation efforts and natural  resource management.\\nHuman impacts such as overfishing, coastal  development, and pollution can rapidly increase stress on ecosys-tems resulting in an urgent need to develop efficient methods for  monitoring spatial and temporal dynamics of biodiversity (Beng  et al., 2016).\\nFisheries management has long used ichthyoplankton sur-veys as a component of monitoring because sampling the early  life stages of fish reflects the reproductive activities of fish    Received: 17 July 2019 | Revised:\\n31 January 2020 |\\nAccepted: 4 February 2020 DOI: 10.1002/ece3.6144   O R\\nI G I N A L  R E S E A R C H Efficacy of metabarcoding for identification of fish eggs  evaluated with mock communities Elena M. Duke  |   Ronald S. Burton\\nThis is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,  provided the original work is properly cited.\\n\\u00a9 2020 The Authors.\\nEcology and Evolution published by John Wiley & Sons Ltd. Marine Biology Research Division, Scripps  Institution of Oceanography, University of  California, San Diego, La Jolla, California Correspondence Elena M. Duke, Marine Biology  Research Division, Scripps Institution of  Oceanography, University of California, San  Diego, La Jolla, CA.\\nEmail: emduke@ucsd.edu\\nAbstract There is urgent need for effective and efficient monitoring of marine fish popula-tions.\\nMo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1371/journal.pone.0134647'], citation_context: \\\"ermsmeier, M. C., Rogowski, P. A., Terrill,  E., Burton, R. S., & Bernardi, G. (2015).\\nMonitoring spawning ac-tivity in a Southern California marine protected area using molec-ular identification of fish eggs.\\nPLoS ONE, 10(8), 1-21.\\nhttps://doi.\\norg/10.1371/journal.pone.0134647 Hastings, P. A., & Burton, R. S. (2008).\\nEstablishing a DNA Sequence  Database for the Marine Fish Fauna of California.\\nRetrieved from  https://escholarship.org/uc/item/1ck9b3qs Hatzenbuhler, C., Kelly, J. R., Martinson, J., Okum, S., & Pilgrim, E. \\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/ece3.6144'], citation_context: \\\"20457758, 2020, 7, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/ece3.6144, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n{\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}"
        ],
        [
         "2",
         "10.1002_ejoc.202000139",
         "./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ejoc.202000139.pdf",
         null,
         "test",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         null,
         "",
         "{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}",
         "",
         "{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n"
        ]
       ],
       "shape": {
        "columns": 10,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>pdf_file_path</th>\n",
       "      <th>xml_file_path</th>\n",
       "      <th>dataset_type</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>xml_first_line</th>\n",
       "      <th>xml_type</th>\n",
       "      <th>pdf_text</th>\n",
       "      <th>xml_text</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_mp.14424</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7937/k9/tc...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...</td>\n",
       "      <td>{\"title\":\"PleThora: Pleural effusion and thora...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_ece3.6144</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>&lt;!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...</td>\n",
       "      <td>jats</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...</td>\n",
       "      <td>{\"title\":\"Efficacy of metabarcoding for identi...</td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_ejoc.202000139</td>\n",
       "      <td>./kaggle/input/make-data-count-finding-data-re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>test</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...</td>\n",
       "      <td></td>\n",
       "      <td>{\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                      pdf_file_path  \\\n",
       "0        10.1002_mp.14424  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "1       10.1002_ece3.6144  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "2  10.1002_ejoc.202000139  ./kaggle/input/make-data-count-finding-data-re...   \n",
       "\n",
       "                                       xml_file_path dataset_type  \\\n",
       "0  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "1  ./kaggle/input/make-data-count-finding-data-re...         test   \n",
       "2                                                NaN         test   \n",
       "\n",
       "                                        dataset_info  \\\n",
       "0  [{'dataset_id': 'https://doi.org/10.7937/k9/tc...   \n",
       "1  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "2     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                      xml_first_line xml_type  \\\n",
       "0  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "1  <!DOCTYPE article PUBLIC \"-//NLM//DTD JATS (Z3...     jats   \n",
       "2                                                NaN            \n",
       "\n",
       "                                            pdf_text  \\\n",
       "0  {\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...   \n",
       "1  {\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...   \n",
       "2  {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...   \n",
       "\n",
       "                                            xml_text  \\\n",
       "0  {\"title\":\"PleThora: Pleural effusion and thora...   \n",
       "1  {\"title\":\"Efficacy of metabarcoding for identi...   \n",
       "2                                                      \n",
       "\n",
       "                                                text  \n",
       "0  {\"title\":\"\",\"authors\":\"Zhao Li, John P., Brand...  \n",
       "1  {\"title\":\"\",\"authors\":\"Beng, Elena M. Duke\\n |...  \n",
       "2  {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00...  "
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_file_paths_df_2 = test_file_paths_df.copy().sample(3, random_state=42).reset_index(drop=True)\n",
    "test_file_paths_df_2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "4d2524bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing article 0/3: 10.1002_mp.14424, type: XML\n",
      "Preparing input with length: 3045\n",
      "Parsing response from generated IDs...\n",
      "Processing article 1/3: 10.1002_ece3.6144, type: XML\n",
      "Preparing input with length: 2203\n",
      "Parsing response from generated IDs...\n",
      "Processing article 2/3: 10.1002_ejoc.202000139, type: PDF\n",
      "Preparing input with length: 985\n",
      "Parsing response from generated IDs...\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "text_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_input",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_response",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "llm_thinking_content",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "364afbfb-e145-4144-aa3a-03710acbbd79",
       "rows": [
        [
         "0",
         "10.1002_ece3.6144",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': '10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<think>\nOkay, let's start by looking at the input data. The user provided a JSON object with sections: title, abstract, data_availability, and other_dataset_citations. The task is to extract valid dataset IDs and classify them as Primary or Secondary.\n\nFirst, I need to check the data_availability section. The data_availability has a DOI: \"10.5061/dryad.zw3r22854\". The important rule says to extract DOIs only if they refer to a dataset. The citation context here is about the Dryad repository, which is a dataset. So that's a valid DOI. \n\nNext, the other_dataset_citations array is empty, so no other dataset IDs are mentioned. The other dataset citations are not present, so no need to check them.\n\nNow, I need to validate the DOI. The DOI is clearly from Dryad, which is a dataset repository. So that's a valid dataset ID. \n\nFor classification, the abstract mentions metabarcoding for identifying fish eggs in mock communities. The data_availability section mentions using two sequences from GenBank, which are datasets. The DOI is from Dryad, which is a data repository. \n\nSince there's a valid dataset ID (DOI) and no other entries in other_dataset_citations, both are valid. The classification should be Primary because the data is directly used in the study. The abstract and citations support that the data is part of the research. \n\nSo the final JSON array should include the valid dataset ID and classify it as Primary.\n</think>"
        ],
        [
         "1",
         "10.1002_ejoc.202000139",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n",
         "[{'dataset_id': '10.1002/ejoc.202000139', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string that represents a paper section, and I need to extract dataset IDs. First, I'll look through the sections to find any dataset-related information.\n\nStarting with the data_availability section, there's nothing there except an empty string. Then, the other_dataset_citations section has a list of dataset IDs. Let me check each one.\n\nThe first dataset ID is 10.1002/ejoc.202000139. The citation context mentions \"10990690, 2020, 19, Downloaded from...\" which looks like a DOI. But wait, the important rule says that only DOIs that are explicitly used as dataset IDs should be extracted. Since this is a DOI and the citation context includes a DOI, but the abstract mentions studies on gold complexes, maybe this is a primary dataset. However, the citation context includes a DOI, but the user's task says to extract only those that refer to datasets. So this might be a valid DOI. Then, the other dataset citation also has 10.1002/ejoc.202000139, but since there's only one entry, maybe process it once. \n\nWait, but the other dataset citations have two entries. Let me check again. The input has:\n\n\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"...\\\"}, ... ]\"\n\nSo there are two dataset entries. The first one is 10.1002/ejoc.202000139. The second one has 10.1002/ejoc.202000139 again? Or maybe there's a typo. Wait, the user's input shows that the first dataset ID is 10.1002/ejoc.202000139, and the second one is also 10.1002/ejoc.202000139? Or maybe a different ID? Wait, the user's input shows that the other_dataset_citations list has two entries, each with the same dataset_id. So both entries are the same. So I need to process only the first one. \n\nSo the first dataset ID is 10.1002/ejoc.202000139. The citation context mentions a DOI, but the important rule says to extract only those that are explicitly dataset IDs. Since this is a DOI, and the citation context includes a DOI, but the abstract mentions studies on gold complexes, perhaps this is a primary dataset. \n\nThen, the second dataset ID in the other_citations list is 10.1002/ejoc.202000139, but that's the same as the first one. So maybe there's a duplicate. But the task says to process the first instance. So the first entry is valid. \n\nNow, for classification. The abstract mentions studies on gold(III) coordination, and the dataset is part of that. So the dataset is directly related to the novel research. Therefore, it's Primary. \n\nThe other dataset citation also has the same dataset_id. But since the first one is processed, the second is redundant. \n\nSo the final JSON would have the dataset_id and type. But wait, the user's example shows that if no dataset IDs are found, the output is missing. But here, there are dataset IDs. So the answer should be an array with the two entries. \n\nWait, but looking at the input, the other_dataset_citations have two entries, both with the same dataset_id. So processing only the first one. So the dataset_id is 10.1002/ejoc.202000139, and the type is Primary. \n\nSo the final JSON would be:\n\n[\n    {\n        \"dataset_id\": \"10.1002/ejoc.202000139\",\n        \"type\": \"Primary\"\n    }\n]\n\nBut wait, the user's input shows that the other_dataset_citations list has two entries, but both have the same dataset_id. So the first one is processed, and the second is redundant. Therefore, the correct output is one entry. \n\nYes, that's correct. So the answer is that array with one entry.\n</think>"
        ],
        [
         "2",
         "10.1002_mp.14424",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}\n",
         "[{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI', 'type': 'Primary'}, {'dataset_id': '10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this dataset classification problem. First, I need to look at the input JSON and parse it. The title mentions \"PleThora: Pleural effusion and thoracic cavity segmentations...\" so the dataset is related to chest CT scans and lung diseases.\n\nLooking at the `data_availability` section, there's a DOI: 10.7937/K9/TCIA.2015.PF0M9REI. The DOI has a valid format starting with \"10.\" and a suffix. The citation context says it's from The Cancer Imaging Archive and a journal. Since the DOI is a dataset ID and not a publication reference, I need to check if it's a dataset. The abstract mentions that the thoracic cavity and pleural effusions are part of the dataset, so this seems like a primary dataset.\n\nNext, the `other_dataset_citations` have another DOI: 10.7937/tcia.2020.6c7y-gq39. Both these DOIs are valid. The citation contexts refer to the same dataset in different years, which aligns with the data availability section. Both are primary datasets since they are directly related to the research presented in the paper.\n\nNow, I need to check if there are any secondary datasets. The abstract mentions that thoracic cavity and pleural effusion segmentations are part of the dataset, which is primary. The data_availability section confirms the dataset's availability, so no secondary data is present here.\n\nThe classification for both DOIs would be Primary. Since both are valid and directly related, they should be included in the final output. The output should be a JSON array with these two dataset IDs and their respective types.\n</think>"
        ]
       ],
       "shape": {
        "columns": 6,
        "rows": 3
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>text_type</th>\n",
       "      <th>llm_input</th>\n",
       "      <th>llm_response</th>\n",
       "      <th>llm_thinking_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_ece3.6144</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Efficacy of metabarcod...</td>\n",
       "      <td>[{'dataset_id': '10.5061/dryad.zw3r22854', 'ty...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_ejoc.202000139</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"Ann Chris...</td>\n",
       "      <td>[{'dataset_id': '10.1002/ejoc.202000139', 'typ...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_mp.14424</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7937/k9/tc...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"PleThora: Pleural effu...</td>\n",
       "      <td>[{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this dataset class...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               article_id                                       dataset_info  \\\n",
       "0       10.1002_ece3.6144  [{'dataset_id': 'https://doi.org/10.5061/dryad...   \n",
       "1  10.1002_ejoc.202000139     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2        10.1002_mp.14424  [{'dataset_id': 'https://doi.org/10.7937/k9/tc...   \n",
       "\n",
       "  text_type                                          llm_input  \\\n",
       "0       XML  Text Content: {\"title\":\"Efficacy of metabarcod...   \n",
       "1       PDF  Text Content: {\"title\":\"\",\"authors\":\"Ann Chris...   \n",
       "2       XML  Text Content: {\"title\":\"PleThora: Pleural effu...   \n",
       "\n",
       "                                        llm_response  \\\n",
       "0  [{'dataset_id': '10.5061/dryad.zw3r22854', 'ty...   \n",
       "1  [{'dataset_id': '10.1002/ejoc.202000139', 'typ...   \n",
       "2  [{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI...   \n",
       "\n",
       "                                llm_thinking_content  \n",
       "0  <think>\\nOkay, let's start by looking at the i...  \n",
       "1  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "2  <think>\\nOkay, let's tackle this dataset class...  "
      ]
     },
     "execution_count": 289,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "processed_articles_df = process_articles(test_file_paths_df_2, inference_model)\n",
    "processed_articles_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "id": "e9bb547b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save processed_articles_df to CSV\n",
    "processed_articles_df.to_csv(\"processed_articles_5.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "8ae6e4d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing article 0/30: 10.1002_2017jc013030, type: XML\n",
      "Preparing input with length: 1648\n",
      "Parsing response from generated IDs...\n",
      "Processing article 1/30: 10.1002_anie.201916483, type: XML\n",
      "Preparing input with length: 1139\n",
      "Parsing response from generated IDs...\n",
      "Processing article 2/30: 10.1002_anie.202005531, type: XML\n",
      "Preparing input with length: 249\n",
      "Parsing response from generated IDs...\n",
      "Processing article 3/30: 10.1002_anie.202007717, type: XML\n",
      "Preparing input with length: 1459\n",
      "Parsing response from generated IDs...\n",
      "Processing article 4/30: 10.1002_chem.201902131, type: XML\n",
      "Preparing input with length: 1277\n",
      "Parsing response from generated IDs...\n",
      "Processing article 5/30: 10.1002_chem.201903120, type: XML\n",
      "Preparing input with length: 2051\n",
      "Parsing response from generated IDs...\n",
      "Processing article 6/30: 10.1002_chem.202000235, type: XML\n",
      "Preparing input with length: 882\n",
      "Parsing response from generated IDs...\n",
      "Processing article 7/30: 10.1002_chem.202001412, type: XML\n",
      "Preparing input with length: 1397\n",
      "Parsing response from generated IDs...\n",
      "Processing article 8/30: 10.1002_chem.202001668, type: XML\n",
      "Preparing input with length: 1007\n",
      "Parsing response from generated IDs...\n",
      "Processing article 9/30: 10.1002_chem.202003167, type: XML\n",
      "Preparing input with length: 1998\n",
      "Parsing response from generated IDs...\n",
      "Processing article 10/30: 10.1002_cssc.202201821, type: XML\n",
      "Preparing input with length: 1705\n",
      "Parsing response from generated IDs...\n",
      "Processing article 11/30: 10.1002_ece3.3985, type: XML\n",
      "Preparing input with length: 1731\n",
      "Parsing response from generated IDs...\n",
      "Processing article 12/30: 10.1002_ece3.4466, type: XML\n",
      "Preparing input with length: 1809\n",
      "Parsing response from generated IDs...\n",
      "Processing article 13/30: 10.1002_ece3.5260, type: XML\n",
      "Preparing input with length: 2575\n",
      "Parsing response from generated IDs...\n",
      "Processing article 14/30: 10.1002_ece3.5395, type: XML\n",
      "Preparing input with length: 4041\n",
      "Parsing response from generated IDs...\n",
      "Processing article 15/30: 10.1002_ece3.6144, type: XML\n",
      "Preparing input with length: 2203\n",
      "Parsing response from generated IDs...\n",
      "Processing article 16/30: 10.1002_ece3.6303, type: XML\n",
      "Preparing input with length: 2687\n",
      "Parsing response from generated IDs...\n",
      "Processing article 17/30: 10.1002_ece3.6784, type: XML\n",
      "Preparing input with length: 2915\n",
      "Parsing response from generated IDs...\n",
      "Processing article 18/30: 10.1002_ece3.961, type: XML\n",
      "Preparing input with length: 1786\n",
      "Parsing response from generated IDs...\n",
      "Processing article 19/30: 10.1002_ece3.9627, type: XML\n",
      "Preparing input with length: 2520\n",
      "Parsing response from generated IDs...\n",
      "Processing article 20/30: 10.1002_ecs2.1280, type: PDF\n",
      "Preparing input with length: 3354\n",
      "Parsing response from generated IDs...\n",
      "Processing article 21/30: 10.1002_ecs2.4619, type: XML\n",
      "Preparing input with length: 2095\n",
      "Parsing response from generated IDs...\n",
      "Processing article 22/30: 10.1002_ejic.201900904, type: XML\n",
      "Preparing input with length: 1029\n",
      "Parsing response from generated IDs...\n",
      "Processing article 23/30: 10.1002_ejoc.202000139, type: PDF\n",
      "Preparing input with length: 985\n",
      "Parsing response from generated IDs...\n",
      "Processing article 24/30: 10.1002_ejoc.202000916, type: XML\n",
      "Preparing input with length: 684\n",
      "Parsing response from generated IDs...\n",
      "Processing article 25/30: 10.1002_esp.5058, type: PDF\n",
      "Preparing input with length: 4509\n",
      "Parsing response from generated IDs...\n",
      "Processing article 26/30: 10.1002_esp.5090, type: XML\n",
      "Preparing input with length: 2729\n",
      "Parsing response from generated IDs...\n",
      "Processing article 27/30: 10.1002_mp.14424, type: XML\n",
      "Preparing input with length: 3045\n",
      "Parsing response from generated IDs...\n",
      "Processing article 28/30: 10.1002_nafm.10870, type: PDF\n",
      "Preparing input with length: 4589\n",
      "Parsing response from generated IDs...\n",
      "Processing article 29/30: 10.1007_jhep07(2018)134, type: PDF\n",
      "Preparing input with length: 894\n",
      "Parsing response from generated IDs...\n"
     ]
    },
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "article_id",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "dataset_info",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "text_type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_input",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "llm_response",
         "rawType": "object",
         "type": "unknown"
        },
        {
         "name": "llm_thinking_content",
         "rawType": "object",
         "type": "string"
        }
       ],
       "ref": "749931ff-049e-4db9-beee-1a639716b64f",
       "rows": [
        [
         "0",
         "10.1002_2017jc013030",
         "[{'dataset_id': 'https://doi.org/10.17882/49388', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Assessing the variability in the relationship between the particulate backscattering coefficient and the chlorophyll a concentration from a global Biogeochemical-Argo database\",\"authors\":\"Marie Barbieux, Julia Uitz, Annick Bricaud, Emanuele Organelli, Antoine Poteau, Catherine Schmechtig, Bernard Gentili, Christophe Penkerc'h, Edouard Leymarie, Fabrizio D'ortenzio, Herv\\u00e9 Claustre\",\"abstract\":\"The particulate backscattering coefficient vs chlorophyll a concentration relationship varies along the water column, according to seasons and oceanic regions. -The b bp -to-Chla ratio is a valuable biogeochemical proxy for assessing the nature of the particulate assemblage and revealing photoacclimation processes. -The BGC-Argo float network yields an unprecedented amount of quality data for studying biogeochemical processes at a global scale and along the vertical dimension.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.17882/49388'], citation_context: \\\"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\\\"}\",\"{\\\"dataset_ids\\\": ['10.17882/47142'], citation_context: \\\"A global bio-optical database derived from Biogeochemical Argo float measurements within the layer of interest for field and remote ocean colour applications|SEANOE|10.17882/47142\\\"}\",\"{\\\"dataset_ids\\\": ['10.5194/essd-2017-58'], citation_context: \\\"Two databases derived from BGC-Argo float measurements for marine biogeochemical and biooptical applications|Earth Syst. Sci. Data|10.5194/essd-2017-58\\\"}\"]}\n",
         "[{'dataset_id': '10.17882/49388', 'type': 'Primary'}, {'dataset_id': '10.17882/47142', 'type': 'Primary'}, {'dataset_id': '10.5194/essd-2017-58', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem step by step. First, I need to parse the input JSON and extract the dataset IDs. The user provided a JSON object with sections like title, authors, abstract, data_availability, and other_dataset_citations. \n\nStarting with Task 1: Identify valid dataset IDs. The data_availability section is empty, so I'll check the other_dataset_citations. There are three entries here. Let me look at each one.\n\nFirst citation: {\"dataset_ids\": [\"10.17882/49388\"], \"citation_context\": ...}. The DOI here looks valid. The citation context mentions \"A global database of vertical profiles derived from Biogeochemical Argo float measurements for biogeochemical and bio-optical applications|SEANOE|10.17882/49388\". The prefix is \"10.\" which matches the required format. Since the citation context clearly indicates it's a dataset, this is a valid DOI. \n\nSecond entry: {\"dataset_ids\": [\"10.17882/47142\"], \"citation_context\": ...}. Similarly, the DOI here is valid. The context mentions a global bio-optical database, so that's another valid dataset.\n\nThird entry: {\"dataset_ids\": [\"10.5194/essd-2017-58\"], \"citation_context\": ...}. This is a DOI. The context says two databases derived from BGC-Argo float measurements. Since the DOI starts with \"10.\" and follows the pattern, it's valid. \n\nNow, I need to check if these IDs are actually datasets. The data_availability section is empty, so these are the only valid ones. There are no other citations or data mentions. \n\nNext, Task 2: Classify the types. The abstract mentions \"Biogeochemical-Argo database\" which points to the data_availability section. Since the data_availability is empty, but the citations refer to databases, I need to use the citation contexts. \n\nFor the first citation, the context mentions \"global database of vertical profiles derived from Biogeochemical Argo float measurements\". This is a primary dataset because it's directly related to the study's focus. Similarly, the second citation mentions a bio-optical database, which is also a primary dataset. The third citation refers to two databases, which could be primary or secondary. However, since the abstract mentions \"global Biogeochemical-Argo database\" and the data_availability is empty, it's likely that these are primary datasets. \n\nWait, but the data_availability is empty, but the citations are there. So, the data_availability section is empty, so the only valid datasets are the ones in the other_dataset_citations. \n\nTherefore, all three dataset IDs are valid. Now, Task 3: Format the results. Since there are three valid dataset IDs, each with their types. The types are Primary for all. \n\nI need to ensure that the output is in the correct JSON format. Each dataset is an object with dataset_id and type. The missing JSON structure is for when no dataset is found, but here all are present. \n\nSo, the final JSON array would have three entries, each with the dataset IDs and their types.\n</think>"
        ],
        [
         "1",
         "10.1002_anie.201916483",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\",\"authors\":\"Bayer Uwe, Werner Daniel Dr., Maichle\\u2010M\\u00f6ssmer C\\u00e4cilia Dr., Anwander Reiner Prof.\\u2005Dr.\",\"abstract\":\"Abstract The homoleptic pyrazolate complexes [Ce III \\n 4 (Me 2 pz) 12 ] and [Ce IV (Me 2 pz) 4 ] 2  quantitatively insert CO 2  to give [Ce III \\n 4 (Me 2 pz\\u22c5CO 2 ) 12 ] and [Ce IV (Me 2 pz\\u22c5CO 2 ) 4 ], respectively (Me 2 pz=3,5\\u2010dimethylpyrazolato). This process is reversible for both complexes, as observed by in\\u2005situ IR and NMR spectroscopy in solution and by TGA in the solid state. By adjusting the molar ratio, one molecule of CO 2  per [Ce IV (Me 2 pz) 4 ] complex could be inserted to give trimetallic [Ce 3 (Me 2 pz) 9 (Me 2 pz\\u22c5CO 2 ) 3 (thf)]. Both the cerous and ceric insertion products catalyze the formation of cyclic carbonates from epoxides and CO 2  under mild conditions. In the absence of epoxide, the ceric catalyst is prone to reduction by the co\\u2010catalyst tetra\\u2010 n \\u2010butylammonium bromide (TBAB).\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string with a paper section and wants me to extract dataset IDs and classify them. Let me start by understanding the input.\n\nFirst, the paper title is \"Effective and Reversible Carbon Dioxide Insertion into Cerium Pyrazolates\". The authors are listed, but the abstract mentions CO2 insertion into pyrazolate complexes. The data_availability is empty, and other_dataset_citations is an empty array.\n\nThe task is to extract valid dataset IDs. The key here is to look in the data_availability section. Since it's empty, maybe there's no dataset available. The other_dataset_citations is also empty, so no citations. \n\nBut wait, the abstract mentions CO2 insertion. Are there any dataset IDs related to this? The abstract talks about the reaction, but not specific dataset IDs. The data_availability section is empty, so no dataset available. \n\nSo, according to Task 1, since there are no valid dataset IDs found in both sections, we need to return the \"Missing\" scenario. The JSON structure should have a dataset_id as \"Missing\" and type as \"Missing\". \n\nI need to make sure I'm not extracting any other IDs. The data_availability and other_dataset_citations are empty. The abstract doesn't mention any dataset IDs. Therefore, the correct response is to return the missing dataset.\n</think>"
        ],
        [
         "2",
         "10.1002_anie.202005531",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Trifluoromethyl Sulfoxides: Reagents for Metal\\u00e2\\u0080\\u0090Free C\\u00e2\\u0088\\u0092H Trifluoromethylthiolation\",\"authors\":\"Wang Dong;prefix:Dr.\",\"abstract\":\"Abstract\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this. The user provided a JSON structure with some paper sections. The task is to extract dataset IDs and classify them.\n\nFirst, looking at the input data, the `data_availability` section is empty. That means there's no data available here. The `other_dataset_citations` is also empty. So, neither section has any dataset IDs. \n\nTask 1 says to check both sections. Since both are empty, no valid dataset IDs are found. Then, Task 3 says to return a \"Missing\" JSON structure. \n\nBut wait, I should double-check if there's any possible dataset IDs mentioned elsewhere. The abstract and citations are empty. The authors mention \"prefix:Dr.\" but that's probably a name, not a dataset ID. So, no valid dataset IDs found. \n\nTherefore, the correct response is to return the \"Missing\" scenario.\n</think>"
        ],
        [
         "3",
         "10.1002_anie.202007717",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Metal\\u2013Ligand Cooperativity of the Calix[4]pyrrolato Aluminate: Triggerable C\\u2212C Bond Formation and Rate Control in Catalysis\",\"authors\":\"Ebner Fabian, Sigmund Lukas Maximilian, Greb Lutz\",\"abstract\":\"Abstract Metal\\u2010ligand cooperativity (MLC) had a remarkable impact on transition metal chemistry and catalysis. By use of the calix[4]pyrrolato aluminate, [ 1 ] \\u2212 , which features a square\\u2010planar Al III , we transfer this concept into the  p \\u2010block and fully elucidate its mechanisms by experiment and theory. Complementary to transition metal\\u2010based MLC ( aromatization  upon substrate binding), substrate binding in [ 1 ] \\u2212  occurs by  dearomatization  of the ligand. The aluminate trapps carbonyls by the formation of C\\u2212C and Al\\u2212O bonds, but the products maintain full reversibility and outstanding dynamic exchange rates. Remarkably, the C\\u2212C bonds can be formed or cleaved by the addition or removal of lithium cations, permitting unprecedented control over the system's constitutional state. Moreover, the metal\\u2010ligand cooperative substrate interaction allows to twist the kinetics of catalytic hydroboration reactions in a unique sense. Ultimately, this work describes the evolution of an  anti\\u2010van't Hoff/Le Bel  species from their being as a structural curiosity to their application as a reagent and catalyst.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string with a paper section and wants me to extract dataset IDs and classify them. Let me start by reading through the input.\n\nFirst, the title is about metal-ligand cooperativity in calix[4]pyrrolato aluminate. The authors are named, and the abstract mentions catalysis, transition metals, and the formation of C-C bonds. The data_availability section is empty, and other_dataset_citations is also empty. \n\nLooking at the data_availability section, there's no information provided. So I need to check if any dataset IDs are mentioned. The other_dataset_citations list is empty. According to the task, I should only extract dataset IDs from data_availability and other_dataset_citations. Since both are empty, there are no valid dataset IDs to extract.\n\nNow, for classification. The task says to use the abstract and citation context. The abstract mentions catalysts, transition metals, and the formation of bonds. The keywords here might be related to catalysis, which could be primary or secondary. But since there's no explicit mention of a dataset, and both sections are empty, there's no way to determine the type. \n\nThe fallback rule says if ambiguous, default to Primary. But since there's no dataset mentioned, maybe the answer is that there are no dataset IDs found, so the output is the \"Missing\" scenario. \n\nI need to make sure I'm not extracting any other data. The user specified not to extract other academic papers or the paper itself. Since both sections are empty, the result is no dataset IDs found. Therefore, the JSON array should have a single object with \"dataset_id\" as \"Missing\" and \"type\" as \"Missing\".\n</think>"
        ],
        [
         "4",
         "10.1002_chem.201902131",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"A Synthetic Route Toward Tetrazoles: The Thermolysis of Geminal Diazides\",\"authors\":\"Holzschneider Kristina, Tong My Linh, Mohr Fabian Prof.\\u2005Dr., Kirsch Stefan F. Prof.\\u2005Dr.\",\"abstract\":\"Abstract A new synthetic route toward the tetrazole core is described, which is based on a general fragmentation pattern that was found in a range of compounds featuring geminal diazido units. Through a simple two\\u2010step procedure, the synthesis of structurally diverse target compounds containing a tetrazole, such as tetrazoloquinoxalinones, benzoylaryltetrazoles, tetrazolotriazinones, and tetrazoloazepinones, was easily accomplished, starting from broadly accessible substrates (i.e., oxindoles, diarylethanones, pyrazolones, and phenanthrols). The initial oxidative diazidation reaction with iodine and sodium azide under mild conditions is followed by the thermal fragmentation under microwave irradiation, leading to the tetrazole products. Noteworthy, an experimental solution is presented in which the potentially hazardous diazide intermediates are not isolated and the concentration of crude reaction mixtures containing diazides is not required to achieve the tetrazoles in good yields.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem step by step. The user provided a JSON string representing a paper section and wants me to extract dataset IDs and classify them as Primary or Secondary. \n\nFirst, I need to check the data_availability section. The data_availability is empty here, so no dataset IDs are present there. Then, there's the other_dataset_citations array, which is empty as well. So according to Task 1, no valid dataset IDs were found.\n\nSince Task 1 didn't find any valid dataset IDs, I should follow Scenario A. The structure required for Scenario A is a JSON array with a single object. The dataset_id is \"Missing\" and the type is also \"Missing\". I need to make sure I don't include any other entries because there are none. \n\nI should double-check if there's any possibility of a DOI or accession ID being there, but the data_availability is empty, and other_dataset_citations is also empty. So no valid entries. Therefore, the correct output is the Missing JSON array.\n</think>"
        ],
        [
         "5",
         "10.1002_chem.201903120",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Structure\\u2013Solubility Relationship of 1,4\\u2010Dioxane Complexes of Di(hydrocarbyl)magnesium\",\"authors\":\"Fischer Reinald Dr., G\\u00f6rls Helmar Dr., Meisinger Philippe R., Suxdorf Regina, Westerhausen Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract Systematic variation of the 1,4\\u2010dioxane (dx) concentration during the precipitation of sparingly soluble [MgBr 2 (dx) 2 ] from ethereal Grignard solutions of RMgBr has allowed the structural investigation of crystallized [R 2 Mg(dx) n ] ( n =1, 1.5, 2, and 3), which form during this dioxane method, depending on the bulkiness of R. The numbering of the complexes explored in this study is based on the number  n  of dioxane molecules per magnesium atom, followed by the substituent R; an apostrophe denotes coordination polymers. The following derivatives were studied by X\\u2010ray crystal\\u2010structure determination and NMR spectroscopy:  n =1: [Me 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010Me ) and [ n Pr 2 Mg(\\u03bc\\u2010dx)] \\u221e  ( 1\\u2032\\u2010 n Pr );  n =1.5: [{ i Pr 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 i Pr ), [{ o Tol 2 Mg(dx)} 2 (\\u03bc\\u2010dx)] ( 1.5\\u2010 o Tol ), and [(Me 3 Si\\u2010C\\u2261C) 2 Mg(dx) 1.5 ] \\u221e  ( 1.5\\u2032\\u2010C 2 SiMe 3 );  n =2: [ t Bu 2 Mg(dx) 2 ] ( 2\\u2010 t Bu ) and [ o Tol 2 Mg(dx) 2 ] ( 2\\u2010 o Tol );  n =3: [Ph 2 Mg(dx) 3 ] ( 3\\u2010Ph ). In the structure types  1\\u2032 ,  1.5 , and  2 , the magnesium atom exhibits the coordination number 4, whereas pentacoordinate metal atoms are observed in types  3  and  1.5\\u2032 . The structure type  2\\u2032  is realized for [(Ph\\u2010C\\u2261C) 2 Mg(dx) 2 ] \\u221e  ( 2\\u2032\\u2010C 2 Ph ), [MgCl 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Cl ), and [MgBr 2 (dx) 2 ] \\u221e  ( 2\\u2032\\u2010Br ) with hexacoordinate metal atoms. The solubility of the dioxane adducts in common organic solvents strongly depends on the degree of aggregation with the solubility decreasing from molecular to strand to layer structures.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let me start by analyzing the provided JSON data. The user wants me to extract and classify dataset IDs from the paper's sections. \n\nFirst, I need to check the sections. The data_availability is empty, and other_dataset_citations is also empty. So, according to Task 1, there are no valid dataset IDs found. \n\nWait, but the abstract mentions \"data release\" and \"data availability\". However, the data_availability section here is empty. So maybe there's no dataset provided. Since both sections are empty, I should proceed to Task 3, which requires outputting a JSON array with \"Missing\" as the dataset_id and \"Missing\" as the type. \n\nI need to make sure I'm not extracting any IDs from the abstract, which is the abstract section. The abstract doesn't mention any dataset IDs. Also, the data_availability and other_dataset_citations are empty. \n\nSo, the final output should be a single JSON object indicating no datasets were found. That's Scenario A.\n</think>"
        ],
        [
         "6",
         "10.1002_chem.202000235",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"An Isolable Bis(Silanone\\u2013Borane) Adduct\",\"authors\":\"Luecke Marcel\\u2010Philip, Pens Elron, Yao Shenglai Dr., Driess Matthias Prof.\\u2005Dr.\",\"abstract\":\"Abstract The reaction of bis(silylenyl)\\u2010substituted ferrocene  1  with two molar equivalents of BPh 3  yields the corresponding bis(silylene\\u2013borane) Lewis adduct  2 . The latter is capable to activate CO 2  to furnish the borane\\u2010stabilized bis(silanone)  3  through mono\\u2010oxygenation of the dative Si II \\u2192B silicon centers under release of CO. Removal of BPh 3  from  3  with PMe 3  affords the corresponding 1,3,2,4\\u2010cyclodisiloxane and the Me 3 P\\u2212BPh 3  adduct. All isolated new compounds were characterized and their molecular structures were determined by single\\u2010crystal X\\u2010ray diffraction analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's start by analyzing the provided JSON input. The user wants me to extract and classify dataset IDs from the paper. First, I need to look through the sections specified.\n\nLooking at the \"data_availability\" section, there's an empty string. That might mean no data is available here. Then, the \"other_dataset_citations\" section is empty as well. So, according to Task 1, there are no valid dataset IDs found.\n\nTask 1 says to check the data_availability first. Since it's empty, I check the other_dataset_citations, which is also empty. Both sections have no dataset IDs. So, no valid dataset IDs were found.\n\nTask 3 requires returning a JSON array. Since no dataset IDs were found, I need to output the \"Missing\" scenario. The structure should have one object with dataset_id and type both missing. I need to make sure that the JSON is properly formatted and that the keys are correctly named. Let me double-check the JSON syntax to avoid any errors.\n</think>"
        ],
        [
         "7",
         "10.1002_chem.202001412",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Iron(0)\\u2010Mediated Stereoselective (3+2)\\u2010Cycloaddition of Thiochalcones via a Diradical Intermediate\",\"authors\":\"Buday Philipp, Seeber Phillip, Zens Clara, Abul\\u2010Futouh Hassan Dr., G\\u00f6rls Helmar Dr., Gr\\u00e4fe Stefanie Prof.\\u2005Dr., Matczak Piotr Dr., Kupfer Stephan Dr., Weigand Wolfgang Prof.\\u2005Dr., Mloston Grzegorz Prof.\\u2005Dr.\",\"abstract\":\"Abstract Reactions of \\u03b1,\\u03b2\\u2010unsaturated aromatic thioketones  1  (thiochalcones) with Fe 3 (CO) 12  leading to \\u03b7 4 \\u20101\\u2010thia\\u20101,3\\u2010diene iron tricarbonyl complexes\\u2005 2 , [FeFe] hydrogenase mimics  3 , and the thiopyrane adduct  4  are described. Obtained products have been characterized by X\\u2010ray crystallography and by computational methods. Completely regio\\u2010 and diastereoselective formation of the five\\u2010membered ring system in products\\u2005 3 , containing four stereogenic centers, can be explained by an unprecedented, stepwise (3+2)\\u2010cycloaddition of two thiochalcone molecules mediated by Fe 3 (CO) 12 . Quantum chemical calculations aimed at elucidation of the reaction mechanism, suggest that the formal (3+2)\\u2010cycloaddition proceeds via sequential intramolecular radical transfer events upon homolytic cleavage of one carbon\\u2010sulfur bond leading to a diradical intermediate.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a paper section and wants me to extract dataset IDs and classify them as Primary or Secondary. First, I need to look through the input JSON structure.\n\nLooking at the input, the `data_availability` section is empty. That means there's no data mentioned here. The `other_dataset_citations` section is also empty. So, according to Task 1, no valid dataset IDs were found. \n\nNow, Task 3 requires formatting the output. Since no dataset IDs were found, I need to return the \"Missing\" JSON structure. I should check if there's any other section that might have data, but in this case, both `data_availability` and `other_dataset_citations` are empty. \n\nI also need to ensure that I'm following all the guidelines. The key was to look for the DOI or Accession ID in the data availability or citation contexts. But since both are empty, there's nothing to extract. \n\nSo, the final output should be a JSON array with the missing dataset ID and type. The user might not have intended any dataset, so this is the correct approach.\n</think>"
        ],
        [
         "8",
         "10.1002_chem.202001668",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"The First Use of a ReX 5  Synthon to Modulate Fe III  Spin Crossover via Supramolecular Halogen\\u22c5\\u22c5\\u22c5Halogen Interactions\",\"authors\":\"Busch Rebecca Dr., Carter Anthony B., Konidaris Konstantis F. Dr., K\\u00fchne Irina A. Dr., Gonz\\u00e1lez Ricardo Dr., Anson Christopher E. Dr., Powell Annie K. Prof.\\u2005Dr.\",\"abstract\":\"Abstract We have added the {Re IV X 5 } \\u2212  (X=Br, Cl) synthon to a pocket\\u2010based ligand to provide supramolecular design using halogen\\u22c5\\u22c5\\u22c5halogen interactions within an Fe III  system that has the potential to undergo spin crossover (SCO). By removing the solvent from the crystal lattice, we \\u201cswitch on\\u201d halogen\\u22c5\\u22c5\\u22c5halogen interactions between neighboring molecules, providing a supramolecular cooperative pathway for SCO. Furthermore, changes to the halogen\\u2010based interaction allow us to modify the temperature and nature of the SCO event.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string with a paper section, and I need to extract valid dataset IDs and classify them as Primary or Secondary. \n\nFirst, I'll check the data_availability section. The abstract mentions \"supramolecular halogen halogen interactions\" and \"halogen-based interaction\", which are related to data. But the data_availability is empty here. So no dataset IDs are present in that section.\n\nNext, the other_dataset_citations is empty too. There's no mention of any datasets in that section. \n\nSince both sections have no dataset IDs, I need to proceed to Task 3. The instructions say to return a JSON array with \"Missing\" if no dataset IDs were found. \n\nWait, but the data_availability and other_dataset_citations are empty. So according to the tasks, if no valid dataset IDs are found after Task 1, output the \"Missing\" structure. \n\nSo the final answer should be the JSON array with the missing dataset_id and type.\n</think>"
        ],
        [
         "9",
         "10.1002_chem.202003167",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Metal\\u2010only Lewis Pairs of Rhodium with  s ,  p  and  d \\u2010Block Metals\",\"authors\":\"Bajo Sonia Dr., Alf\\u00e9rez Macarena G., Alcaide Mar\\u00eda M., L\\u00f3pez\\u2010Serrano Joaqu\\u00edn Dr., Campos Jes\\u00fas Dr.\",\"abstract\":\"Abstract Metal\\u2010only Lewis pairs (MOLPs) in which the two metal fragments are solely connected by a dative M\\u2192M bond represent privileged architectures to acquire fundamental understanding of bimetallic bonding. This has important implications in many catalytic processes or supramolecular systems that rely on synergistic effects between two metals. However, a systematic experimental/computational approach on a well\\u2010defined class of compounds is lacking. Here we report a family of MOLPs constructed around the Rh I  precursor [( \\u03b7 \\n 5 \\u2010C 5 Me 5 )Rh(PMe 3 ) 2 ] ( 1 ) with a series of  s ,  p  and  d \\u2010block metals, mostly from the main group elements, and investigate their bonding by computational means. Among the new MOLPs, we have structurally characterized those formed by dative bonding between  1  and MgMeBr, AlMe 3 , GeCl 2 , SnCl 2 , ZnMe 2  and Zn(C 6 F 5 ) 2,  as well as spectroscopically identified the ones resulting from coordination to MBAr F  (M=Na, Li; BAr F \\n \\u2212 =[B(C 6 H 2 \\u20103,5\\u2010(CF 3 ) 2 ) 4 ] \\u2212 ) and CuCl. Some of these compounds represent unique examples of bimetallic structures, such as the first unambiguous cases of Rh\\u2192Mg dative bonding or base\\u2010free rhodium bound germylene and stannylene species. Multinuclear NMR spectroscopy, including  103 Rh NMR, is used to probe the formation of Rh\\u2192M bonds. A comprehensive theoretical analysis of those provides clear trends. As anticipated, greater bond covalency is found for the more electronegative acids, whereas ionic character dominates for the least electronegative nuclei, though some degree of electron sharing is identified in all cases.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON structure with a paper section and wants me to extract dataset IDs. First, I need to check the sections for data availability and other citations.\n\nLooking at the abstract, there's a mention of \"MOLPs\" and \"s, p, d-block metals\". The data availability section is empty here. The other dataset citations are also empty. So, no dataset IDs are present in the given input. \n\nTask 1 says to check data_availability and other_dataset_citations. Both are empty. So, no valid dataset IDs. Then, Task 3 requires returning a JSON array with \"Missing\" if no datasets were found. \n\nI need to make sure I'm not extracting anything else, like the paper's title or authors. The data_availability and other_citations are the only sections that might have dataset IDs. Since both are empty, the answer should be the \"Missing\" scenario.\n</think>"
        ],
        [
         "10",
         "10.1002_cssc.202201821",
         null,
         "XML",
         "Text Content: {\"title\":\"Development of a Mg/O ReaxFF Potential to describe the Passivation Processes in Magnesium\\u2010Ion Batteries **\",\"authors\":\"Fiesinger Florian, Gaissmaier Daniel Dr., van den Borg Matthias, Be\\u00dfner Julian, van Duin Adri C. T. Prof.\\u2005Dr., Jacob Timo Prof.\\u2005Dr.\",\"abstract\":\"Abstract One of the key challenges preventing the breakthrough of magnesium\\u2010ion batteries (MIB) is the formation of a passivating boundary layer at the Mg anode. To describe the initial steps of Mg anode degradation by O 2  impurities, a Mg/O ReaxFF (force field for reactive systems) parameter set was developed capable of accurately modeling the bulk, surface, adsorption, and diffusion properties of metallic Mg and the salt MgO. It is shown that O 2  immediately dissociates upon first contact with the Mg anode (modeled as Mg(0001), Mg(10 1 \\u203e \\n0)A, and Mg(10 1 \\u203e \\n1)), heating the surface to several 1000\\u2005K. The high temperature assists the further oxidation and forms a rock salt interphase intersected by several grain boundaries. Among the Mg surface terminations, Mg(10 1 \\u203e \\n0)A is the most reactive, forming an MgO layer with a thickness of up to 25\\u2005\\u00c5. The trained force field can be used to model the ongoing reactions in Mg\\u2013air batteries but also to study the oxidation of magnesium metal in general.\",\"data_availability\":\"Data Availability Statement The data that support the findings of this study are openly available in zenodo at  https://doi.org/10.5281/zenodo.7074790 . Furthermore, the  KVIK  optimization routine is openly available in the GitHub repository at  https://github.com/shk11/KVIKOptimizer .\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': '10.5281/zenodo.7074790', 'type': 'Primary'}]",
         "<think>\nOkay, let's start by looking at the input data. The user provided a JSON string with sections for a paper. My task is to extract and classify dataset IDs.\n\nFirst, I need to check the `data_availability` section. The data availability states that the data is available in Zenodo with a DOI. The DOI is \"10.5281/zenodo.7074790\". Wait, but the important rule says if it's a citation for a publication, we shouldn't extract it. However, the DOI here seems to be for a dataset, so maybe it's valid. Then there's a GitHub repository with a KVIK optimization routine, which doesn't look like a dataset. \n\nNext, the `other_dataset_citations` list is empty. So, only the Zenodo DOI is a potential dataset ID. Now, I need to check if this DOI matches the criteria for a dataset. The DOI starts with \"10.\" and has a suffix. The suffix here is \"7074790\", which is valid. The citation context in the data availability mentions the data is available in Zenodo, which is a data repository. So this seems to be a valid dataset ID.\n\nNow, the classification. The abstract mentions the formation of a passivating boundary layer at the Mg anode. The dataset here is related to the Mg anode's passivation. The DOI is for a dataset, so it's a Primary dataset because it's generated by the authors as part of their research. The other part of the citation context is about data availability, which supports the classification.\n\nSo, the dataset_id is \"10.5281/zenodo.7074790\", and the type is Primary. Since there are no other valid dataset IDs, I proceed to format the output accordingly.\n</think>"
        ],
        [
         "11",
         "10.1002_ece3.3985",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"High levels of polyandry, but limited evidence for multiple paternity, in wild populations of the western rock lobster ( Panulirus cygnus )\",\"authors\":\"Loo Jacqueline, Kennington Winn Jason, de Lestang Simon, How Jason, Evans Jonathan P.\",\"abstract\":\"Abstract Polyandry, where multiple mating by females results in the temporal and spatial overlap of ejaculates from two or more males, is taxonomically widespread and occurs in varying frequencies within and among species. In decapods (crabs, lobsters, crayfish, and prawns), rates of polyandry are likely to be variable, but the extent to which patterns of multiple paternity reflect multiple mating, and thus are shaped by postmating processes that bias fertilization toward one or a subset of mated males, is unclear. Here, we use microsatellite markers to examine the frequency of multiple mating (the presence of spermatophores from two or more males) and patterns of paternity in wild populations of western rock lobster ( Panulirus cygnus ). Our data confirm that >45% of females had attached spermatophores arising from at least two males (i.e., confirming polyandry), but we found very limited evidence for multiple paternity; among 24 clutches sampled in this study, only two arose from fertilizations by two or more males. Single inferred paternal genotypes accounted for all remaining progeny genotypes in each clutch, including several instances when the mother had been shown to mate with two or more males. These findings highlight the need for further work to understand whether polyandry is adaptive and to uncover the mechanisms underlying postmating paternity biases in this system.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string with a paper section and wants me to extract dataset IDs. First, I need to check the data_availability section because that's where the main dataset information is. The data_availability is empty here, so there's no dataset mentioned. Then, the other_dataset_citations is also empty. So according to Task 1, there are no valid dataset IDs found. Therefore, I should return the \"Missing\" scenario as specified in the instructions. I need to make sure not to include any dataset IDs and structure the JSON correctly. Let me double-check the instructions to confirm that I'm not missing any other sections. Yep, the title, authors, abstract, and other dataset citations are all empty. So the final output should be a JSON array with the missing dataset ID and type.\n</think>"
        ],
        [
         "12",
         "10.1002_ece3.4466",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.r6nq870', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Incubation temperature and parental identity determine sex in the Australian agamid lizard  Ctenophorus pictus\",\"authors\":\"Hansson Alexander, Olsson Mats\",\"abstract\":\"Abstract Sex determination in Australian agamid lizards shows a complex framework of different mechanisms, varying even among closely related taxa. It is clear that discrete classification of these species as either having genetic ( GSD ) or environmental sex determination ( ESD ) does not agree with empirical data. Although many species in this group show temperature\\u2010dependent sex determination ( TSD ), recent evidence suggests additional genetic or epigenetic effects. A proposed model explaining the adaptive significance and evolution of  TSD  in short\\u2010lived agamids predicts that selection will favor temperature\\u2010biased sex ratios in species with intense male\\u2010male competition. Here, we use experimental incubation at (near) constant temperatures to test whether the sex of Australian painted dragons ( Ctenophorus pictus ) is influenced by temperature, building on previous research yet to have reached an agreement regarding the role of temperature in this species. In this study, incubation temperature and parental identity affected hatchling sex suggesting that environment and genetics may work in concert to determine sex in this species. Although our results are consistent with  TSD , our data cannot rule out a temperature\\u2010by\\u2010sex effect on egg or hatchling mortality. However, our findings together with the observed differences of sex determination systems in closely related species within this genus may provide novel opportunities to address fundamental questions in the evolution of sex determination systems.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's start by looking at the input JSON provided. The paper's title is about incubation temperature and parental identity determining sex in Australian agamids. The authors are Hansen and Olsson. The abstract mentions that discrete classification into genetic or environmental sex determination doesn't agree with empirical data. They propose a model where temperature affects sex ratios, and the study tested this with incubation at near constant temperatures.\n\nNow, the data_availability section is empty. The other_dataset_citations are also empty. So, according to Task 1, there are no valid dataset IDs to extract. Since Task 1 resulted in no valid IDs, we go to Task 3. The output should be a JSON array with \"Missing\" entries. The structure is as specified, so I need to create that array.\n</think>"
        ],
        [
         "13",
         "10.1002_ece3.5260",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Ultraconserved element (UCE) probe set design: Base genome and initial design parameters critical for optimization\",\"authors\":\"Gustafson Grey T., Alexander Alana, Sproul John S., Pflug James M., Maddison David R., Short Andrew E. Z.\",\"abstract\":\"Abstract Targeted capture and enrichment approaches have proven effective for phylogenetic study. Ultraconserved elements (UCEs) in particular have exhibited great utility for phylogenomic analyses, with the software package  phyluce  being among the most utilized pipelines for UCE phylogenomics, including probe design. Despite the success of UCEs, it is becoming increasing apparent that diverse lineages require probe sets tailored to focal taxa in order to improve locus recovery. However, factors affecting probe design and methods for optimizing probe sets to focal taxa remain underexplored. Here, we use newly available beetle (Coleoptera) genomic resources to investigate factors affecting UCE probe set design using  phyluce . In particular, we explore the effects of stringency during initial design steps, as well as base genome choice on resulting probe sets and locus recovery. We found that both base genome choice and initial bait design stringency parameters greatly alter the number of resultant probes included in final probe sets and strongly affect the number of loci detected and recovered during in silico testing of these probe sets. In addition, we identify attributes of base genomes that correlated with high performance in probe design. Ultimately, we provide a recommended workflow for using  phyluce  to design an optimized UCE probe set that will work across a targeted lineage, and use our findings to develop a new, open\\u2010source UCE probe set for beetles of the suborder Adephaga.\",\"data_availability\":\"DATA AVAILABILITY DNA sequences: GenBank MK838494 to MK838511;  phyluce  BED/BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad  https://doi.org/10.5061/dryad.2f62927 ; Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fastas available from;  https://github.com/laninsky/comparing_monolithic_UCE_fastas ; Alexander 2018b: Phase_target available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target ; Alexander 2018c: distance_calcs available from:  https://github.com/laninsky/distance_calcs ; Alexander 2018d: Phase_everyone available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone .\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': '10.5061/dryad.2f62927', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string representing a paper section and wants me to extract and classify dataset IDs. Let's start by understanding the input.\n\nFirst, looking at the `data_availability` section. The text mentions DNA sequences: GenBank MK838494 to MK838511, phyluce BED/BAM files, and a Dryad DOI. Also, there are custom scripts available. The other_dataset_citations is empty here. \n\nSo, the data_availability includes both a DNA sequence dataset and a BED/BAM file. The user's task is to extract valid dataset IDs. For DOIs, there's a DOI in the data_availability, but the citation context is not provided. However, the key point here is that the DOI might be a dataset_id. But wait, the important rule says if a DOI is presented as a citation, it's not a dataset_id. But here, the data_availability has a DOI. Wait, the input data says \"DATA AVAILABILITY DNA sequences: GenBank MK838494 to MK838511; phyluce  BED/BAM files...\" So the DOI is not in the data_availability section. Wait, the user's input data might have a typo here. Let me check again.\n\nLooking back, the data_availability section says \"DNA sequences: GenBank MK838494 to MK838511; phyluce  BED/BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad  https://doi.org/10.5061/dryad.2f62927 ; Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fastas available from;  https://github.com/laninsky/comparing_monolithic_UCE_fastas ; Alexander 2018b: Phase_target available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target ; Alexander 2018c: distance_calcs available from:  https://github.com/laninsky/distance_calcs ; Alexander 2018d: Phase_everyone available from  https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone .\"\n\nSo here, the DOI is in the Dryad link. The other dataset citations are empty. The data_availability includes a dataset with a DOI. But according to the important rule, a DOI is a dataset_id only if it's used as a citation. However, in this case, the DOI is part of the Dryad link, not a citation. So maybe the DOI is not a dataset_id here. Wait, the data_availability section says \"dataset\" in the context of DNA sequences, which are datasets. So perhaps the DOI is not part of the data_availability but part of the Dryad link. Wait, but the user's input data might have a typo here. Let me check again.\n\nThe data_availability says \"phyluce  BED/BAM files for probe design experiments and Adephaga_2.9Kv1 final probe set fasta file: Dryad  https://doi.org/10.5061/dryad.2f62927 ; Custom scripts...\" So the Dryad DOI is part of the data_availability. The other dataset citations are empty. Therefore, in this case, the dataset_id would be the Dryad DOI. But according to the important rule, if a DOI is presented as a citation, it's not a dataset_id. However, in this case, the Dryad DOI is part of the data_availability, not a citation. Therefore, it's a valid dataset_id. So the dataset_id here is 10.5061/dryad.2f62927.\n\nWait, but the other dataset citations are empty. So the only valid dataset_id here is the Dryad DOI. But the user's input data might have a typo where the DOI is actually a dataset_id. Alternatively, maybe there's another dataset_id. Let me check again.\n\nLooking at the other_dataset_citations array, it's empty. The data_availability includes a Dryad DOI. So that's one dataset_id. But according to the important rule, if a DOI is presented as a citation, it's not a dataset_id. But here, the Dryad DOI is part of the data_availability. So this is a valid dataset_id. So in Task 1, we have to extract this as a dataset_id.\n\nNow, for classification. The abstract mentions that the authors are using phyluce for UCE phylogenomics. The data_availability includes a Dryad DOI. The key words in the abstract are \"probe set design\", \"UEP\", \"phyluce\", which are related to the UCE design. The data_availability also includes a BED/BAM file for probe design experiments. So this seems like a dataset related to the research. Therefore, this would be a Primary dataset because it's used in the research. \n\nBut let's check the other_dataset_citations. They are empty. So only the Dryad DOI is considered here. Therefore, the dataset_id is 10.5061/dryad.2f62927, and the type is Primary. \n\nBut wait, the data_availability also includes a custom script. The script is part of the data_availability, so that's another dataset. However, the user's input data shows that the other_dataset_citations is empty. The data_availability includes the Dryad DOI and the custom scripts. But according to the task, we only need to extract dataset_ids. So the dataset_id here is the Dryad DOI. \n\nTherefore, in Task 1, we have one valid dataset_id. Then, in Task 2, we classify it as Primary. \n\nSo the final answer would be a JSON array with that dataset_id and type.\n</think>"
        ],
        [
         "14",
         "10.1002_ece3.5395",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"\\u201cCloser\\u2010to\\u2010home\\u201d strategy benefits juvenile survival in a long\\u2010distance migratory bird\",\"authors\":\"Cheng Yachang, Fiedler Wolfgang, Wikelski Martin, Flack Andrea\",\"abstract\":\"Abstract Human\\u2010induced changes in the climate and environment that occur at an unprecedented speed are challenging the existence of migratory species. Faced with these new challenges, species with diverse and flexible migratory behaviors may suffer less from population decline, as they may be better at responding to these changes by altering their migratory behavior. At the individual level, variations in migratory behavior may lead to differences in fitness and subsequently influence the population's demographic dynamics. Using lifetime GPS bio\\u2010logging data from 169 white storks ( Ciconia ciconia ), we explore whether the recently shortened migration distance of storks affects their survival during different stages of their juvenile life. We also explore how other variations in migratory decisions (i.e., time, destination), movement activity (measured using overall body dynamic acceleration), and early life conditions influence juvenile survival. We observed that their first autumn migration was the riskiest period for juvenile white storks. Individuals that migrated shorter distances and fledged earlier experienced lower mortality risks. In addition, higher movement activity and overwintering \\u201ccloser\\u2010to\\u2010home\\u201d (with 84.21% of the tracked individuals stayed Europe or North Africa) were associated with higher survival. Our study shows how avian migrants can change life history decisions over only a few decades, and thus it helps us to understand and predict how migrants respond to the rapidly changing world.\",\"data_availability\":\"DATA ACCESSIBILITY The data used for this study are available through the Movebank Data Repository ( https://www.movebank.org ): with  https://doi.org/10.5441/001/1.v1cs4nn0 ,  https://doi.org/10.5441/001/1.c42j3js7 ,  https://doi.org/10.5441/001/1.4192t2j4 ,  https://doi.org/10.5441/001/1.ck04mn78 ,  https://doi.org/10.5441/001/1.71r7pp6q  (Fiedler, Flack, Sch\\u00e4fle, et al.,  2019 ; Fiedler, Flack, Schmid, Reinhard, & Wikelski,  2019 ; Fiedler, Hilsendegen, et al.,  2019 ; Fiedler, Leppelsack, et al.,  2019 ; Fiedler, Niederer, Sch\\u00f6nenberger, Flack, & Wikelski,  2019 ).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5441/001/1.ck04mn78'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Sch\\u00e4fle ,  W. \\n ,  \\n Keeves ,  B. \\n ,  \\n Quetting ,  M. \\n ,  \\n Eid ,  B. \\n , \\u2026  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Southwest Germany .  Movebank Data Repository ,  10.5441/001/1.ck04mn78\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.c42j3js7'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Flack ,  A. \\n ,  \\n Schmid ,  H. \\n ,  \\n Reinhard ,  U. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Oberschwaben .  Movebank Data Repository ,  10.5441/001/1.c42j3js7\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.4192t2j4'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Hilsendegen ,  C. \\n ,  \\n Reis ,  C. \\n ,  \\n Lehmann ,  J. \\n ,  \\n Hilsendegen ,  P. \\n ,  \\n Schmid ,  H. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Rheinland\\u2010Pfalz .  Movebank Data Repository ,  10.5441/001/1.4192t2j4\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.v1cs4nn0'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Leppelsack ,  E. \\n ,  \\n Leppelsack ,  H. \\n ,  \\n Stahl ,  T. \\n ,  \\n Wieding ,  O. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Bavaria .  Movebank Data Repository ,  10.5441/001/1.v1cs4nn0\\\"}\",\"{\\\"dataset_ids\\\": ['10.5441/001/1.71r7pp6q'], citation_context: \\\"Fiedler ,  W. \\n ,  \\n Niederer ,  W. \\n ,  \\n Sch\\u00f6nenberger ,  A. \\n ,  \\n Flack ,  A. \\n , &  \\n Wikelski ,  M. \\n  ( 2019 ).  Data from: LifeTrack White Stork Vorarlberg .  Movebank Data Repository ,  10.5441/001/1.71r7pp6q\\\"}\"]}\n",
         "[{'dataset_id': '10.5441/001/1.v1cs4nn0', 'type': 'Primary'}, {'dataset_id': '10.5441/001/1.ck04mn78', 'type': 'Primary'}, {'dataset_id': '10.5441/001/1.c42j3js7', 'type': 'Primary'}, {'dataset_id': '10.5441/001/1.4192t2j4', 'type': 'Primary'}, {'dataset_id': '10.5441/001/1.v1cs4nn0', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string representing a paper section, and I need to extract and classify dataset IDs. Let's start by understanding the structure.\n\nFirst, I need to identify valid dataset IDs. The data_availability section has several DOIs listed. The important rule is to check if these DOIs are truly data-related. The user mentioned that DOIs should be extracted only if they refer to a dataset, not a publication. So I need to look at the citation contexts here.\n\nLooking at the data_availability, the first DOI is \"10.5441/001/1.v1cs4nn0\". The citation context says \"Fiedler, Leppelsack, Leppelsack, Stahl, Wieding, O. & Wikelski, M. (2019). Data from: LifeTrack White Stork Bavaria.\" This seems to indicate that this is a dataset related to the study, so it's a valid dataset ID. The other DOIs in the list also have similar citation contexts. \n\nNext, I need to check if there are any Accession IDs. The data_availability section mentions \"DATA ACCESSIBILITY The data used for this study are available through...\", but there's no explicit mention of an Accession ID. The other_dataset_citations have dataset IDs like \"10.5441/001/1.ck04mn78\", which are Accession IDs. So those are valid.\n\nNow, I need to classify each dataset. The abstract mentions that the study uses lifetime GPS data from 169 white storks. The keywords related to primary and secondary datasets are important here. The study's findings are about how migration distance affects survival, so the dataset likely relates to the research. Since the dataset is directly related, it's a Primary dataset.\n\nI should check if there are any Secondary datasets mentioned. The abstract doesn't explicitly state that, but the data availability includes the DOI, which is more likely a Primary dataset. The other citations also have dataset IDs that are Accession IDs, which are Primary. \n\nSo, all the valid dataset IDs are extracted. The classification for each should be Primary since they are directly related to the research. Finally, format the results as per the required JSON structure.\n</think>"
        ],
        [
         "15",
         "10.1002_ece3.6144",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Efficacy of metabarcoding for identification of fish eggs evaluated with mock communities\",\"authors\":\"Duke Elena M., Burton Ronald S.\",\"abstract\":\"Abstract There is urgent need for effective and efficient monitoring of marine fish populations. Monitoring eggs and larval fish may be more informative\\u00a0than that traditional fish surveys since ichthyoplankton surveys reveal the reproductive activities of fish populations, which directly impact their population trajectories. Ichthyoplankton surveys have turned to molecular methods (DNA barcoding & metabarcoding) for identification of eggs and larval fish due to challenges of morphological identification. In this study, we examine the effectiveness of using metabarcoding methods on mock communities of known fish egg DNA. We constructed six mock communities with known ratios of species. In addition, we analyzed two samples from a large field collection of fish eggs and compared metabarcoding results with traditional DNA barcoding results. We examine the ability of our metabarcoding methods to detect species and relative proportion of species identified in each mock community. We found that our metabarcoding methods were able to detect species at very low input proportions; however, levels of successful detection depended on the markers used in amplification, suggesting that the use of multiple markers is desirable. Variability in our quantitative results may result from amplification bias as well as interspecific variation in mitochondrial DNA copy number. Our results demonstrate that there remain significant challenges to using metabarcoding for estimating proportional species composition; however, the results provide important insights into understanding how to interpret metabarcoding data. This study will aid in the continuing development of efficient molecular methods of biological monitoring for fisheries management.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT Data are available at Dryad Digital Repository at:  https://doi.org/10.5061/dryad.zw3r22854 . Two sequences used in our reference library are on GenBank (accession numbers MH714866 and MH718435).\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': '10.5061/dryad.zw3r22854', 'type': 'Primary'}]",
         "<think>\nOkay, let's start by looking at the input JSON. The title is about the efficacy of metabarcoding for identifying fish eggs in mock communities. The authors are Duke Elena M. and Burton Ronald S. The abstract mentions using metabarcoding methods in a study, and they have data availability from Dryad. \n\nFirst, I need to check the data_availability section. The DOI is provided as 10.5061/dryad.zw3r22854. According to the guidelines, this is a valid dataset_id. The citation context here is the Dryad link, which clearly refers to a dataset. So that's one dataset.\n\nNext, the other_dataset_citations section is empty. So no additional citations. Now, I need to classify this dataset. The abstract mentions that metabarcoding is used in this study, and the data is available. Since the abstract talks about the study's effectiveness and the data is from Dryad, which is a dataset repository, this should be a primary dataset. \n\nWait, but the data_availability also mentions two sequences on GenBank. Are those part of the dataset? The data_availability says \"Data are available at Dryad Digital Repository...\" and the sequences are from GenBank. So the dataset is the Dryad DOI. \n\nSo, only one dataset is found. The classification should be Primary, as the study directly uses metabarcoding methods for this purpose. The other_dataset_citations are empty. \n\nNow, the final JSON structure. Since one dataset is found, I need to return a JSON array with one entry. The dataset_id is the Dryad DOI, and the type is Primary. No missing entries here.\n</think>"
        ],
        [
         "16",
         "10.1002_ece3.6303",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.37pvmcvgb', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Can polyploidy confer invasive plants with a wider climatic tolerance? A test using  Solidago canadensis\",\"authors\":\"Wan Jizhong, Oduor Ayub M. O., Pouteau Robin, Wang Beilei, Chen Luxi, Yang Beifen, Yu Feihai, Li Junmin\",\"abstract\":\"Abstract Polyploidy can cause variation in plant functional traits and thereby generate individuals that can adapt to fluctuating environments and exploit new environments. However, few empirical studies have tested for an association between ploidy level and climatic tolerance of invasive cytotypes relative to conspecific native\\u2010range cytotypes. Here, we used an invasive plant  Solidago canadensis  to test whether invasive populations had a higher proportion of polyploids, greater height and stem\\u2010base diameter, and occupied a wider range of climatic conditions than conspecific native\\u2010range populations. We also tested whether the invasive populations had overcome genetic founder effects. We sampled a total of 80 populations in parts of the invaded range in China and native range in North America for in situ measurements of plant height and stem\\u2010base diameter in the field and for population genetic and cytotype analyses. To examine climatic correlates, we augmented our field\\u2010sampled data with occurrence records obtained from Global Biodiversity Information Facility. All, except one, of the populations that we sampled in China occurred in a humid subtropical climate. In contrast, the North American populations occurred in humid continental, humid subtropical, and semi\\u2010arid climatic zones. All populations of  S. canadensis  in China were purely hexaploid, while the North American populations were diploid, tetraploid, and hexaploid. The invasive hexaploids were significantly taller and had a\\u00a0larger stem\\u2010base diameter than native hexaploids. Native hexaploids were significantly taller and had larger stem\\u2010base diameter than native diploids. Climatic correlate assessment found that invasive and native populations occupied different climatic envelopes, with invasive populations occurring in warmer and less seasonal climates than native populations. However, there was no significant correlation between ploidy level and climatic envelope of  S. canadensis . Molecular phylogeo\",\"data_availability\":\"DATA AVAILABILITY STATEMENT DNA sequences have been deposited in the National Center for Biotechnology Information (NCBI) database (accession numbers: MN107084\\u2013MN107151, MN335287\\u2013MN335290). The data that support the findings of this study have been deposited in Dryad with doi: https://doi.org/10.5061/dryad.37pvmcvgb .\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's start by looking at the input text. The title mentions a study on polyploidy and its effect on climate tolerance in Solidago canadensis. The authors are Wan Jizhong, Oduor Ayub M. O., Pouteau Robin, Wang Beilei, Chen Luxi, Yang Beifen, Yu Feihai, and Li Junmin.\n\nThe abstract states that polyploidy affects plant traits and adaptability. They tested whether invasive populations have higher ploidy, greater height, stem base diameter, and wider climate range than native ones. They used DNA sequences from NCBI and Dryad. The data_availability mentions depositing sequences in NCBI and Dryad, so that's a primary dataset.\n\nLooking at other_dataset_citations, there are no entries provided. The data_availability section clearly states the dataset IDs from NCBI and Dryad. Since there are no citations, maybe those are the only datasets. But wait, the abstract mentions \"DNA sequences have been deposited\" in NCBI and Dryad. The DOI or accession ID would be the dataset_id. But the input data here doesn't include any specific dataset IDs. However, the data_availability mentions depositing sequences, which might be part of a dataset. But according to the key definitions, the dataset_id is a unique identifier. Since there's no explicit DOI or accession ID in the input, maybe there are no valid dataset_ids. But the abstract mentions that the data is deposited in NCBI and Dryad, which might be considered as dataset_ids. However, the task says to extract only valid dataset_ids. Since the input doesn't include any specific IDs, perhaps the answer is that there are no valid dataset_ids. But the data_availability mentions depositing sequences, which could be a dataset_id. However, the input JSON structure doesn't include those IDs. Wait, looking back at the input structure, the other_dataset_citations are empty. The data_availability is a string with \"DATA AVAILABILITY STATEMENT DNA sequences...\", which might be part of the dataset. But according to the instructions, the dataset_ids should be extracted from the data_availability or other_dataset_citations. Since there are no entries in other_dataset_citations, maybe the answer is no dataset_ids found. However, the abstract mentions that the data is deposited in NCBI and Dryad, which could be considered as dataset_ids. But the input JSON structure doesn't have those entries. Therefore, perhaps the answer is that there are no valid dataset_ids. But maybe the user expects that the NCBI and Dryad IDs are included. But according to the input structure, those are in the data_availability section. Wait, the input data is structured as a JSON with \"data_availability\" as a string. The other_dataset_citations are empty. So the dataset_ids are from the data_availability. But the input doesn't have any specific dataset IDs. Therefore, there are no valid dataset_ids. Hence, the output would be a missing dataset JSON. But let me check again. The data_availability says DNA sequences have been deposited in NCBI and Dryad, but the input doesn't specify the actual dataset_ids. However, according to the key definitions, dataset_ids are either DOIs or accession IDs. Since the input doesn't have DOIs or accession IDs, maybe there are no valid dataset_ids. Therefore, the correct output is to return a missing dataset JSON.\n</think>"
        ],
        [
         "17",
         "10.1002_ece3.6784",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Ecomorphology of the pectoral girdle in anurans (Amphibia, Anura): Shape diversity and biomechanical considerations\",\"authors\":\"Engelkes Karolin, Kath Lena, Kleinteich Thomas, Hammel J\\u00f6rg U., Beerlink Andr\\u00e9, Haas Alexander\",\"abstract\":\"Abstract Frogs and toads (Lissamphibia: Anura) show a diversity of locomotor modes that allow them to inhabit a wide range of habitats. The different locomotor modes are likely to be linked to anatomical specializations of the skeleton within the typical frog Bauplan. While such anatomical adaptations of the hind limbs and the pelvic girdle are comparably well understood, the pectoral girdle received much less attention in the past. We tested for locomotor\\u2010mode\\u2010related shape differences in the pectoral girdle bones of 64 anuran species by means of micro\\u2010computed\\u2010tomography\\u2010based geometric morphometrics. The pectoral girdles of selected species were analyzed with regard to the effects of shape differences on muscle moment arms across the shoulder joint and stress dissipation within the coracoid. Phylogenetic relationships, size, and locomotor behavior have an effect on the shape of the pectoral girdle in anurans, but there are differences in the relative impact of these factors between the bones of this skeletal unit. Remarkable shape diversity has been observed within locomotor groups indicating many\\u2010to\\u2010one mapping of form onto function. Significant shape differences have mainly been related to the overall pectoral girdle geometry and the shape of the coracoid. Most prominent shape differences have been found between burrowing and nonburrowing species with headfirst and backward burrowing species significantly differing from one another and from the other locomotor groups. The pectoral girdle shapes of burrowing species have generally larger moment arms for (simulated) humerus retractor muscles across the shoulder joint, which might be an adaptation to the burrowing behavior. The mechanisms of how the moment arms were enlarged differed between species and were associated with differences in the reaction of the coracoid to simulated loading by physiologically relevant forces.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT CT volumes can be downloaded from  https://www.fdr.uni\\u2010hamburg.de/search?page=1&size=20&q=keywords:%22pectoral%20girdle%20morphometrics%20project%22 ; DOI numbers are provided in Appendix S1: Table A1.\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1007/s00435-017-0386-3'], citation_context: \\\"Engelkes ,  K. \\n ,  \\n Friedrich ,  F. \\n ,  \\n Hammel ,  J. U. \\n , &  \\n Haas ,  A. \\n  ( 2018 ).  A simple setup for episcopic microtomy and a digital image processing workflow to acquire high\\u2010quality volume data and 3D surface models of small vertebrates .  Zoomorphology ,  137 ( 1 ),  213 \\u2013 228 .  10.1007/s00435-017-0386-3\\\"}\"]}\n",
         "[{'dataset_id': '10.1007/s00435-017-0386-3', 'type': 'Primary'}, {'dataset_id': '10.1007/s00435-017-0386-3', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user wants me to extract and classify dataset IDs from the given paper. First, I need to parse the input JSON structure. The paper's title is \"Ecomorphology of the pectoral girdle in anurans...\" and the authors are listed. The abstract mentions data availability with a DOI and a citation context. The other_dataset_citations section has a dataset ID.\n\nStarting with Task 1: Identify valid dataset IDs. The data_availability mentions a DOI in Appendix S1, Table A1. The DOI is valid because it's explicitly stated as a dataset ID. The citation context includes the DOI, so that's a valid entry. Then, the other_dataset_citations has another dataset ID with a citation context. Both entries meet the criteria.\n\nNow, Task 2: Classify the types. The abstract mentions shape diversity in pectoral girdles and their impact on muscle moment arms. The DOI and other dataset citation are related to the research. Since the abstract clearly discusses shape differences and the data is used in the study, I should classify them as Primary. The other dataset citation's context is about a study using similar methods, which aligns with the research.\n\nTask 3: Format the results. Since both dataset IDs are valid, the output should include both entries. The JSON structure needs to reflect that each dataset is classified as Primary. I need to make sure to check the keywords in the abstract and citation context to confirm the classification. The fallback rule is applied here because the abstract clearly states the research focus.\n</think>"
        ],
        [
         "18",
         "10.1002_ece3.961",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"The effects of insects, nutrients, and plant invasion on community structure and function above-and belowground\",\"authors\":\"Wright Phoebe, Cregger Melissa A, Souza Lara, Sanders Nathan J, Classen Aim\\u00e9e T\",\"abstract\":\"Soil nutrient availability, invasive plants, and insect presence can directly alter ecosystem structure and function, but less is known about how these factors may interact. In this 6-year study in an old-field ecosystem, we manipulated insect abundance (reduced and control), the propagule pressure of an invasive nitrogen-fixing plant (propagules added and control), and soil nutrient availability (nitrogen added, nitrogen reduced and control) in a fully crossed, completely randomized plot design. We found that nutrient amendment and, occasionally, insect abundance interacted with the propagule pressure of an invasive plant to alter above-and belowground structure and function at our site. Not surprisingly, nutrient amendment had a direct effect on aboveground biomass and soil nutrient mineralization. The introduction of invasive nitrogen-fixing plant propagules interacted with nutrient amendment and insect presence to alter soil bacterial abundance and the activity of the microbial community. While the larger-scale, longer-term bulk measurements such as biomass production and nutrient mineralization responded to the direct effects of our treatments, the shorter-term and dynamic microbial communities tended to respond to interactions among our treatments. Our results indicate that soil nutrients, invasive plants, and insect herbivores determine both above-and belowground responses, but whether such effects are independent versus interdependent varies with scale.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string representing a paper section and wants me to extract valid dataset IDs and classify them as Primary or Secondary. \n\nFirst, I need to check the input data. The title mentions \"insects, nutrients, and plant invasion\", which might be related to the study. The authors are listed, but there's no mention of datasets. The abstract talks about manipulating variables like insect abundance, plant propagule pressure, and soil nutrients, but nothing about datasets.\n\nLooking at the data_availability section, it's empty. The other_dataset_citations is also empty. Since there are no dataset IDs provided, according to Task 1, there are no valid dataset IDs. So, I should return the \"Missing\" scenario.\n\nWait, but maybe I missed something? Let me double-check. The abstract doesn't mention any datasets. The data_availability and other_citations are empty. So, yes, no dataset IDs found. Therefore, the correct response is to output the JSON with \"Missing\" in both dataset_id and type.\n</think>"
        ],
        [
         "19",
         "10.1002_ece3.9627",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Inferring predator\\u2013prey interactions from camera traps: A\\u00a0Bayesian co\\u2010abundance modeling approach\",\"authors\":\"Amir Zachary, Sovie Adia, Luskin Matthew Scott\",\"abstract\":\"Abstract Predator\\u2013prey dynamics are a fundamental part of ecology, but directly studying interactions has proven difficult. The proliferation of camera trapping has enabled the collection of large datasets on wildlife, but researchers face hurdles inferring interactions from observational data. Recent advances in hierarchical co\\u2010abundance models infer species interactions while accounting for two species' detection probabilities, shared responses to environmental covariates, and propagate uncertainty throughout the entire modeling process. However, current approaches remain unsuitable for interacting species whose natural densities differ by an order of magnitude and have contrasting detection probabilities, such as predator\\u2013prey interactions, which introduce zero inflation and overdispersion in count histories. Here, we developed a Bayesian hierarchical N\\u2010mixture co\\u2010abundance model that is suitable for inferring predator\\u2013prey interactions. We accounted for excessive zeros in count histories using an informed zero\\u2010inflated Poisson distribution in the abundance formula and accounted for overdispersion in count histories by including a random effect per sampling unit and sampling occasion in the detection probability formula. We demonstrate that models with these modifications outperform alternative approaches, improve model goodness\\u2010of\\u2010fit, and overcome parameter convergence failures. We highlight its utility using 20 camera trapping datasets from 10 tropical forest landscapes in Southeast Asia and estimate four predator\\u2013prey relationships between tigers, clouded leopards, and muntjac and sambar deer. Tigers had a negative effect on muntjac abundance, providing support for top\\u2010down regulation, while clouded leopards had a positive effect on muntjac and sambar deer, likely driven by shared responses to unmodelled covariates like hunting. This Bayesian co\\u2010abundance modeling approach to quantify predator\\u2013prey relationships is widely applicable across species, ecosystems\",\"data_availability\":\"DATA AVAILABILITY STATEMENT All R code and data used to implement our Bayesian co\\u2010abundance models can be accessed at this Figshare repository:  https://doi.org/10.5061/dryad.b8gtht7h3 .\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': '10.5061/dryad.b8gtht7h3', 'type': 'Primary'}]",
         "<think>\nOkay, let's start by looking at the input data. The user provided a JSON string with sections for the paper. My task is to extract valid dataset IDs and classify them as Primary or Secondary.\n\nFirst, I need to check the `data_availability` section. The abstract mentions \"DATA AVAILABILITY STATEMENT\" and provides a DOI: 10.5061/dryad.b8gtht7h3. That looks like a DOI. But wait, the IMPORTANT DOI VALIDATION RULE says I shouldn't extract DOIs from publications. However, the citation context here is about the dataset's availability, so maybe it's valid. Let me check the citation context again. The abstract says \"R code and data...\" which are part of the dataset. So the DOI might be a valid dataset ID here.\n\nNext, the `other_dataset_citations` section is empty. So only the DOI is considered. Now, I need to validate this DOI. Since it's provided in the data_availability, and the citation context is about the dataset's availability, it's likely a dataset ID. So the dataset_id would be 10.5061/dryad.b8gtht7h3.\n\nNow, checking the classification. The abstract mentions using Bayesian models, which are part of the study's approach. The keywords related to Primary are \"generated in this study\" or \"newly sequenced\". The model they developed is specifically for predator-prey interactions, which is a novel study. So the dataset is directly related to the research, making it Primary.\n\nSo the final answer would have this dataset_id and type as Primary.\n</think>"
        ],
        [
         "20",
         "10.1002_ecs2.1280",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.p3fg9', 'type': 'Primary'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"Catton, Rosemarie A. De \\u00adClerck-Floate3, R. G. Lalonde, Yvonne M. Buckley,2, R. A. De Clerck-Floate, Haley A. Catton,,\\u2020 Robert G. Lalonde,1, Y. M. Buckley\",\"abstract\":\"Abstract.   \\nUnderstanding the impact of herbivory on plant populations is a fundamental goal of ecology.  \\nDamage to individual plants can be visually striking and affect the fates of individuals, but these impacts  do not necessarily translate into population-\\u00adlevel differences in vital rates (survival, growth, or fecundity)  or population growth rates.\\nIn biological control of weeds, quantitative assessments of population-\\u00adlevel  impacts of released agents on both target invasive plants and native, nontarget plants are needed to inform  evaluations of the benefits and risks of releasing agents into new regions.\\nHere we present a 3-\\u00adyr exper-imental demographic field study using the European root-\\u00adfeeding biocontrol weevil, Mogulones crucifer,  first released in Canada in 1997 to control the invasive weed Cynoglossum officinale (Boraginaceae).\\nMogu-lones crucifer is an effective \\u201csearch and destroy\\u201d agent in Canada, but sporadically feeds, oviposits, and  develops on native nontarget Boraginaceae.\\nWe investigated the population-\\u00adlevel impacts of this biocon-trol insect on its target weed and a native nontarget plant, Hackelia micrantha (Boraginaceae), by releasing  large numbers of weevils into naturally occurring patches of H. micrantha growing isolated from or inter-spersed with C. officinale.\\nWe followed the fates of individual plants on release and nonrelease (control)  sites for two transition years, developed matrix models to project population growth rates (\\u03bb) for each  plant species, and examined the contributions from differences in vital rates to changes in \\u03bb using life table  response experiments (LTRE).\\nIn contrast to studies of the insect-plant interaction in its native range, as a  biocontrol agent, M. crucifer increased mortality of C. officinale rosettes in the year immediately following  release, depressing the weed's \\u03bb to below the population replacement level.\\nHowever, \\u03bb for H. micrantha  was never depressed below the replacement level, and any differences between release and nonrelease  sites in the nontarget could not be explained by significant contributions from vital rates in the LTRE.\\nThis  study is the first to simultaneously and experimentally examine target and nontarget population-\\u00adlevel im-pacts of a weed biocontrol insect in the field, and supports the theoretical prediction that plant life history  characteristics and uneven herbivore host preferences can interact to produce differences in population-\\u00ad level impacts between target and nontarget plant species.\",\"data_availability\":\"Data Availability\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ecs2.1280'], citation_context: \\\"21508925, 2016, 5, Downloaded from https://esajournals.onlinelibrary.wiley.com/doi/10.1002/ecs2.1280 by California Digital Library University Of California, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n",
         "[{'dataset_id': '10.1002/ecs2.1280', 'type': 'Primary'}, {'dataset_id': '10.1002/ecs2.1280', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON structure with sections of a paper and wants me to extract dataset IDs and classify them as Primary or Secondary.\n\nFirst, I need to look at the `data_availability` section. The example here is a DOI: \"10.1002/ecs2.1280\". The important rule says if a DOI is used as a dataset ID, it must be valid. The citation context here mentions \"Downloaded from...\" and \"Terms and Conditions\", which might indicate it's a legitimate dataset. So I'll extract that DOI.\n\nNext, check the `other_dataset_citations` section. The first entry has a dataset_id of \"10.1002/ecs2.1280\" and a citation context. Since this is the same DOI as the data_availability section, it's the same dataset. So that's one dataset.\n\nNow, classifying them. The abstract mentions \"population-level impacts\" and \"vital rates\", which points to Primary. The DOI and citation context support that the dataset is directly related to the study's findings. So both datasets are Primary.\n\nNo other dataset IDs are present. The task says to extract only valid ones, so both are valid. The output should be an array with both dataset IDs and their types.\n</think>"
        ],
        [
         "21",
         "10.1002_ecs2.4619",
         "[{'dataset_id': 'https://doi.org/10.25349/d9qw5x', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"Interactions between temperature and predation impact insect emergence in alpine lakes\",\"authors\":\"Caroline H Owens, Michelle J Lee, | Melissa Grim, John Schroeder, Hillary S Young, Manuel T Lerdau\",\"abstract\":\"While insect populations are simultaneously threatened by many local-and global-scale stressors, the interacting effects of these factors remain poorly understood. These interactions between stressors, whether additive or antagonistic, may have profound effects on our assumptions about and predictions for any given system. Here we address this gap by exploring the interactive effects of introduced predators and elevated temperatures on emerging aquatic insects across alpine lakes. Using a crossed factorial field survey, we examine whether warmer temperatures either directly affect the magnitude or diversity of insect emergence, or mediate the impacts of predation via interactions with predator presence. Based on data from more than 11,000 insects collected across a two-year period, we find a 71% reduction in insect emergence associated with an additional 3.6 C of water temperature in oligotrophic alpine lakes. While our work confirms prior findings that predator presence drives strong reductions in insect emergence, we find that the effects of predation are significantly weaker in warmer lakes (2% reduction in warmest lakes studied vs. 75% reduction in coldest). Combining our results with regional studies of predator communities to identify mechanisms suggests that the interaction effects between temperature and predator presence are likely due to changes in predator behavior across temperatures. Critically, to fully understand the multiple stressors that impact insect populations and predict their future effects, we must consider the interactions between stressors across spatial scales.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.25349/D9QW5X'], citation_context: \\\"Data for: Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes|Dryad. Dataset|10.25349/D9QW5X\\\"}\"]}\n",
         "[{'dataset_id': '10.25349/D9QW5X', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem. The user provided a JSON string with a paper section and wants me to extract dataset identifiers (dataset_ids) and classify them as Primary or Secondary.\n\nFirst, I need to check the data_availability section. The data_availability is empty here, so that's no data. Then, the other_dataset_citations section has a dataset_id: \"10.25349/D9QW5X\" with a citation context. The citation context mentions \"Interactions between Temperature and Predation Impact Insect Emergence in Alpine Lakes|Dryad. Dataset|10.25349/D9QW5X\". \n\nSince the citation context clearly refers to a dataset (Dryad), and the DOI is valid, I should extract this dataset_id. Now, I need to check if this dataset is directly related to the paper's research. The abstract mentions interactions between temperature and predators affecting insect emergence, and the dataset is related to that. The citation context also mentions the dataset, so it's directly linked. \n\nNext, I should classify this dataset as Primary because it's directly used for the research. The other dataset citation doesn't have the same context, but the presence of the citation context confirms it's part of the study. \n\nSo, the only dataset found is this one. Since there are no other valid dataset_ids in the other sections, I proceed to Task 3. The final output should be a JSON array with the dataset_id and type.\n</think>"
        ],
        [
         "22",
         "10.1002_ejic.201900904",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Chemical properties and biological activity of Au(III) compounds obtained from dichlorido[2\\u2010(dimethylaminomethyl)phenyl\\u2010C 1 ,N]gold(III), [Au(damp\\u2010C 1 ,N)Cl 2 ], and halogenated, potentially tridentate thiosemicarbazones have been studied. The results of this work show that the complexation of the halogenated thiosemicarbazones with Au(III) enhances their stability against hydrolysis and retains or enhances their anti\\u2010parasitic activity. Fluorination in the periphery of the ligands has expectedly no influence on the structural chemistry of the obtained Au(III) complexes, but modulates their biological behaviour. Best results with a remarkably high selectivity index for the trypomastigote form of  Trypanosoma cruzi  were obtained with the complex containing the ligand, which presents a 3,5\\u2010fluorine substitution in meta\\u2010position of an aromatic ring, [Au(dampH)(L\\u20103,5\\u2010F)]Cl.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this. The input is a JSON object with the paper details. The user wants me to extract valid dataset IDs and classify them as Primary or Secondary.\n\nFirst, looking at the `data_availability` section, there's nothing there. The `other_dataset_citations` is also empty. So, no dataset IDs were found in either section. According to Task 1, if no valid IDs are found, return the \"Missing\" JSON structure.\n\nWait, but maybe I missed something? Let me double-check. The abstract mentions chemical properties and Au(III) complexes, but no specific dataset IDs. The data_availability is empty, and other_dataset_citations is empty. So yes, no dataset IDs. Therefore, the correct output is the \"Missing\" JSON array with both dataset_id and type missing.\n</think>"
        ],
        [
         "23",
         "10.1002_ejoc.202000139",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"Ann Christin Reiers\\u00f8lmoen[a\",\"abstract\":\"Abstract: Studies on gold(III) coordination of a series of pre-pared polydentate pyridine and quinoline based ligands are re-ported.\\nCharacterization (1H, 13C, 15N NMR, and XRD) of the novel gold(III) complexes, prepared in 31-98 % yield, revealed different coordination ability of the pyridine and quinoline nitrogen atoms.\\nTesting of catalytic activity in cyclopropanation\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/ejoc.202000139'], citation_context: \\\"10990690, 2020, 19, Downloaded from https://chemistry-europe.onlinelibrary.wiley.com/doi/10.1002/ejoc.202000139 by California Digital Library University Of California, Wiley Online Library on [09/05/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n",
         "[{'dataset_id': '10.1002/ejoc.202000139', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this problem step by step. The user provided a JSON string with a paper section and wants me to extract and classify dataset IDs. Let me start by understanding the structure of the input.\n\nFirst, the title is empty, which probably means the paper isn't about datasets. The authors list is \"Ann Christin Reiers\" with an [a] note, so maybe she's the author. The abstract mentions studies on gold(III) coordination and testing of catalytic activity. The data availability is empty, so I'll check the other_dataset_citations section.\n\nLooking at the other_dataset_citations, there's a dataset with dataset_ids [\"10.1002/ejoc.202000139\"] and a citation context. The citation context includes a DOI. Now, I need to check if this DOI is a dataset ID. The important rule says only extract DOIs if they're explicitly used as dataset IDs. The DOI here seems to be a valid dataset ID. The citation context mentions \"Downloaded from\" and \"Caltech Digital Library,\" which suggests it's a dataset. So that's a valid dataset ID.\n\nWait, but the data availability section is empty. The other dataset citation is the only one. So I should process that. Now, check if this dataset is related to the paper's research. The abstract talks about gold(III) coordination and catalytic activity, which are part of the study. The dataset might be related. Now, according to the classification rules, if the dataset is directly related to the novel research, it's Primary. The keywords in the abstract might help. Since the dataset is related, I'll classify it as Primary.\n\nSo the dataset_id is 10.1002/ejoc.202000139, and the type is Primary. The other citation is a DOI, which we've already considered. The user wants only valid dataset IDs, so this one is valid. The other dataset citation is a DOI, which we also process. So the final JSON array should include this dataset.\n</think>"
        ],
        [
         "24",
         "10.1002_ejoc.202000916",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "XML",
         "Text Content: {\"title\":\"Ammonium Salt\\u2010Catalyzed Ring\\u2010Opening of Aryl\\u2010Aziridines with \\u03b2\\u2010Keto Esters\",\"authors\":\"Haider Victoria, Kreuzer Viktoria, Tiffner Maximilian Dr., Spingler Bernhard Dr., Waser Mario Dr.\",\"abstract\":\"We herein report an ammonium salt\\u2010catalyzed protocol for the regioselective ring opening of aryl\\u2010aziridines with \\u03b2\\u2010keto esters. The reaction gives access to a variety of highly functionalized target molecules with two consecutive stereogenic centers and can be rendered enantioselective (up to  e.r . = 91:9) by using bifunctional chiral ammonium salt catalysts.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's start by looking at the provided JSON data. The title mentions ammonium salt-catalyzed ring-opening of aryl-aziridines with keto esters. The authors list is Haider Victoria, Kreuzer Viktoria, etc. The abstract talks about a reaction using chiral ammonium salts for enantioselectivity. \n\nLooking at the data_availability section, it's empty. The other_dataset_citations are also empty. So, according to Task 1, there are no valid dataset IDs found here. \n\nNow, Task 3 requires formatting. Since no dataset IDs are present, we need to return the \"Missing\" scenario. The JSON structure should have a single object with dataset_id and type both missing. \n\nI need to make sure I'm not extracting any other data, like the authors or abstract, which are not dataset IDs. Also, the DOI validation rule is strictly applied here, but since there are no DOIs or Accession IDs, I just proceed to the missing case.\n</think>"
        ],
        [
         "25",
         "10.1002_esp.5058",
         "[{'dataset_id': 'https://doi.org/10.5061/dryad.jh9w0vt9t', 'type': 'Primary'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"Yasuyuki, Jonathan Mark Nelson2\",\"abstract\":\"R E S E A R C H A R T I C L E Past, present and future of a meandering river in the Bolivian Amazon basin Kattia Rubi Arnez Ferrel1 |\\nJonathan Mark Nelson2 | Yasuyuki Shimizu1 | Tomoko Kyuka1 1Graduate School of Engineering, Hokkaido University, Sapporo, Japan 2U.S. Geological Survey, Golden, Colorado, USA Correspondence Kattia Rubi Arnez Ferrel, Graduate School of Engineering, Hokkaido University, Sapporo 060-0808, Japan.\\nEmail: rubikraf@gmail.com\\nFunding information Nitobe School Project Abstract Field observations on small rivers of the Amazon basin are less common due to their remote location and difficult accessibility.\\nHere we show, through remote sensing analysis and field works, the planform evolution and riverbed topography of a small river located in the upper foreland Amazon basin, the Ichilo River.\\nBy tracking plan-form changes over 30 years, we identified the factors that control meander migration rates in the Ichilo River: cutoffs, climate and human interventions.\\nThe data suggest that neck cutoffs are the main controls in the Ichilo River, with an annual density of 0.022 cutoffs/km.\\nIn addition, climate controls have been identified in the form of high-precipitation events that may have promoted cutoffs, an increase in meander migration rate and channel widening.\\nThe width distribution of the Ichilo River is well represented by general extreme value and inverse Gaussian distributions.\\nThe spatio-temporal variability of meandering migration rates in the Ichilo River is analysed in two locations where neck cutoffs are expected.\\nAnalysing the distance across the neck in these two points, we predict the occurrence of a new cutoff.\\nThe combined methodology of bathymetric surveys and structure from motion photogrammetry shows us the Ichilo riverbed topography and banks at high resolution, where two scour holes were identified.\\nFinally, we discuss the impact of planform changes of the Ichilo River on communities that are established along its riverbanks.\\nK E Y W O R D S bathymetric surveys, Bolivian Amazon basin, Ichilo River, meanders, remote sensing, UAV 1 | INTRODUCTION\\nThe Amazon basin exhibits a constantly changing landscap\",\"data_availability\":\"DATA AVAILABILITY STATEMENT\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1002/esp.4637'], citation_context: \\\", M.R., Chandler, J.H., Eltner, A., Fraser, C., Miller, P.E., Mills, J.P. et al. (2019) Guidelines on the use of structure-from-motion photo-grammetry in geomorphic research.\\nEarth Surface Processes and Land-forms, 44(10), 2081-2084.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Semin\\\"},{\\\"dataset_ids\\\": ['10.5066/P9FW6E8K'], citation_context: \\\"84.\\nhttps://doi.org/10.1002/esp.4637\\n[Dataset] Kinzel, P., Nelson, J.M. & Kattia, R.A.F. (2019)\\nBathymetric survey of the Ichilo and Sajta Rivers, near Puerto Villarroel, Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-d\\\"},{\\\"dataset_ids\\\": ['10.1029/2005JF000416'], citation_context: \\\", Bolivia, May 23-24, 2019.\\nU.S. Geological Survey data release.\\nhttps://doi.org/10.5066/P9FW6E8K Lanzoni, S. & Seminara, G. (2006)\\nOn the nature of meander instability.\\nJournal of Geophysical Research - Earth Surface, 111(F4), 1-14.\\nhttps://doi.org/10.1029/2005JF000416 Li, J., Grenfell, M.C., Wei, H., Tooth, S. & Ngiem, S. (2020) Chute cutoff-driven abandonment and sedimentation of meander bends along a fine-grained, non-vegetated, ephemeral river on the Bolivian Alti-plano.\\nGeomorphology, 350, 106917.\\nhttps://doi\\\"}\",\"{\\\"dataset_ids\\\": ['10.1002/esp.5058'], citation_context: \\\"10969837, 2021, 4, Downloaded from https://onlinelibrary.wiley.com/doi/10.1002/esp.5058, Wiley Online Library on [31/03/2025].\\nSee the Terms and Conditions (https://onlinelibrary.wiley.com/terms-and-conditions) on Wiley Online Library for rules of use; OA articles are governed by the applicable Creative Commons License\\\"}\"]}\n",
         "[{'dataset_id': '10.1002/esp.4637', 'type': 'Primary'}, {'dataset_id': '10.5066/P9FW6E8K', 'type': 'Secondary'}]",
         "<think>\nOkay, let's tackle this problem step by step. First, I need to parse the provided JSON input and extract the dataset IDs. The user wants only valid dataset IDs, so I have to check both the data_availability and other_dataset_citations sections.\n\nLooking at the abstract, there's a mention of a dataset in the data_availability section. The data_availability key has a value that includes a DOI. The important rule says if a DOI is used as a dataset_id, it must be validated. In this case, the DOI is provided, so it's valid. Also, the citation_context for that DOI is empty, which is allowed.\n\nNext, I check the other_dataset_citations. There are two entries. The first one has a dataset_id with a DOI. Again, the citation context is empty, so it's valid. The second dataset_id is an Accession ID, which fits the definition. The citation context for that is \"84.\", which seems to be a placeholder. \n\nNow, I need to ensure there are no extraneous entries. The other entries have citation contexts that might be placeholders or other references, but those are not dataset IDs. So, only the first two entries are valid. \n\nNow, for classification. The abstract mentions the Ichilo River's data, so it's related to the study. The dataset IDs are from the U.S. Geological Survey, which is the authors' institution. The data_availability section mentions data release, which aligns with primary data. The other dataset citations are from previous studies, which are secondary. \n\nApplying the primary/secondary classification rules: the first dataset is primary because it's the U.S. Geological Survey data, and the second is secondary since it's a citation from another study. \n\nFinally, format the results. Since both dataset IDs are valid, return them in an array. If there were no dataset IDs, return the missing structure. In this case, both are present, so the output is the array with both dataset entries.\n</think>"
        ],
        [
         "26",
         "10.1002_esp.5090",
         "[{'dataset_id': 'https://doi.org/10.5066/p9353101', 'type': 'Secondary'}]",
         "XML",
         "Text Content: {\"title\":\"\",\"authors\":\"\",\"abstract\":\"Abstract The 20 May 2016  M W  6.1 Petermann earthquake in central Australia generated a 21\\u2009km surface rupture with 0.1 to 1 m vertical displacements across a low\\u2010relief landscape. No paleo\\u2010scarps or potentially analogous topographic features are evident in pre\\u2010earthquake Worldview\\u20101 and Worldview\\u20102 satellite data. Two excavations across the surface rupture expose near\\u2010surface fault geometry and mixed aeolian\\u2010sheetwash sediment faulted only in the 2016 earthquake. A 10.6 \\u00b1 0.4 ka optically stimulated luminescence (OSL) age of sheetwash sediment provides a minimum estimate for the period of quiescence prior to 2016 rupture. Seven cosmogenic beryllium\\u201010 ( 10 Be) bedrock erosion rates are derived for samples < 5 km distance from the surface rupture on the hanging\\u2010wall and foot\\u2010wall, and three from samples 19 to 50\\u2009km from the surface rupture. No distinction is found between fault proximal rates (1.3 \\u00b1 0.1 to 2.6 \\u00b1 0.2 m Myr \\u22121 ) and distal samples (1.4 \\u00b1 0.1 to 2.3 \\u00b1 0.2 m Myr \\u22121 ). The thickness of rock fragments (2\\u20135 cm) coseismically displaced in the Petermann earthquake perturbs the steady\\u2010state bedrock erosion rate by only 1 to 3%, less than the erosion rate uncertainty estimated for each sample (7\\u201312%). Using  10 Be erosion rates and scarp height measurements we estimate approximately 0.5 to 1 Myr of differential erosion is required to return to pre\\u2010earthquake topography. By inference any pre\\u20102016 fault\\u2010related topography likely required a similar time for removal. We conclude that the Petermann earthquake was the first on this fault in the last ca. 0.5\\u20131 Myr. Extrapolating single nuclide erosion rates across this timescale introduces large uncertainties, and we cannot resolve whether 2016 represents the first ever surface rupture on this fault, or a\\u2009> 1 Myr interseismic period. Either option reinforces the importance of including distributed earthquake sources in fault displacement and seismic hazard analyses.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.1080/08120090802546977'], citation_context: \\\"Constraints on the current rate of deformation and surface uplift of the Australian continent from a new seismic database and low\\u2010T thermochronological data|Australian Journal of Earth Sciences|https://doi.org/10.1080/08120090802546977\\\"}\",\"{\\\"dataset_ids\\\": ['10.1177/0959683613515732'], citation_context: \\\"Late\\u2010Holocene climatic variability indicated by three natural archives in arid southern Australia|Holocene|https://doi.org/10.1177/0959683613515732\\\"}\"]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's tackle this problem. The user wants me to extract dataset IDs from the provided paper and classify them as Primary or Secondary. Let me start by reading through the input JSON to understand the sections.\n\nFirst, the title is empty, so maybe that's just the paper's title. The authors are also empty, which might mean no authors listed. The abstract mentions a 2016 earthquake and some geological data. The data availability section has two citations. The other dataset citations also have dataset IDs.\n\nLooking at the abstract, there's a mention of \"sheetwash sediment\" and \"cosmogenic beryllium-10 erosion rates.\" That sounds like data related to the earthquake. The data availability has two DOI entries. The first one is \"10.1080/08120090802546977.\" I need to check if this is a dataset ID. The DOI format is [prefix]/[suffix], and the prefix is always 10. So the first part is 10. followed by 80... So the suffix is 080... which matches. The citation context says it's from a new seismic database and low-T thermochronological data. Since the DOI is used as a reference to a publication, not a dataset, I should ignore this. The second DOI is 10.1177/0959683613515732. The format here is 10. followed by 1177... which also fits. The citation context mentions a climatic archive, so again, it's a publication, so not a dataset ID. \n\nThe other_dataset_citations list has two entries. Both have dataset IDs. The first one's citation context mentions a new seismic database and low-T data. The second mentions a climatic archive. Since they're both references to publications, not datasets, I need to check if they are dataset IDs. The dataset IDs provided are the ones in the citations. The user's task is to extract only dataset IDs. So even if the DOI is a reference, we just take the dataset IDs. \n\nNow, looking at the classification. The abstract mentions \"sheetwash sediment\" and \"cosmogenic beryllium-10 erosion rates,\" which are data related to the earthquake. These should be Primary because they are data generated by the authors for this study. The other dataset citations also have similar data. Since both entries are in the data availability section, they are Primary. \n\nSo, the dataset IDs are the ones in the citations. The first one is 10.1080/... which is a DOI, but according to the rules, if the citation context is a publication, it's not a dataset ID. However, the user's task is to extract only dataset IDs. So even if the DOI is a reference, we need to consider it as a dataset ID. Wait, but the important rule says that only extract DOIs that are explicitly used as dataset IDs. The citation context here is a publication, so it's not a dataset. Therefore, the first DOI should be excluded. The second one is 10.1177/..., which is a DOI, but again, the citation context is a publication. So both are excluded. \n\nWait, but the user's task says to extract only dataset IDs. So even if the DOI is a reference, we just take the dataset IDs. But according to the rules, if the citation context is a publication, it's not a dataset. So maybe the other dataset citations are dataset IDs. Let me check again. The other_dataset_citations have the dataset_ids as [\"10.1080/...\"] and [\"10.1177/...\"]. Both are DOIs, and the citation contexts mention publications. Therefore, they are not dataset IDs. \n\nSo in this case, there are no valid dataset IDs found. Therefore, the output should be a missing dataset entry. Wait, but the data_availability is empty. The abstract mentions data, but the data_availability section is empty. The other dataset citations have dataset IDs, but the citation contexts are publications. So the answer is that no dataset IDs were found. Therefore, the JSON array would have a missing dataset entry.\n</think>"
        ],
        [
         "27",
         "10.1002_mp.14424",
         "[{'dataset_id': 'https://doi.org/10.7937/k9/tcia.2015.pf0m9rei', 'type': 'Secondary'}, {'dataset_id': 'https://doi.org/10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "XML",
         "Text Content: {\"title\":\"PleThora: Pleural effusion and thoracic cavity segmentations in diseased lungs for benchmarking chest CT processing pipelines\",\"authors\":\"Kiser Kendall J., Ahmed Sara, Stieb Sonja, Mohamed Abdallah S. R., Elhalawani Hesham, Park Peter Y. S., Doyle Nathan S., Wang Brandon J., Barman Arko, Li Zhao, Zheng W. Jim, Fuller Clifton D., Giancardo Luca\",\"abstract\":\"This manuscript describes a dataset of thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 computed tomography (CT) scans acquired from patients with non\\u2010small cell lung cancer. The segmentation of these anatomic regions precedes fundamental tasks in image analysis pipelines such as lung structure segmentation, lesion detection, and radiomics feature extraction. Bilateral thoracic cavity volumes and pleural effusion volumes were manually segmented on CT scans acquired from The Cancer Imaging Archive \\u201cNSCLC Radiomics\\u201d data collection. Four hundred and two thoracic segmentations were first generated automatically by a U\\u2010Net based algorithm trained on chest CTs without cancer, manually corrected by a medical student to include the complete thoracic cavity (normal, pathologic, and atelectatic lung parenchyma, lung hilum, pleural effusion, fibrosis, nodules, tumor, and other anatomic anomalies), and revised by a radiation oncologist or a radiologist. Seventy\\u2010eight pleural effusions were manually segmented by a medical student and revised by a radiologist or radiation oncologist. Interobserver agreement between the radiation oncologist and radiologist corrections was acceptable. All expert\\u2010vetted segmentations are publicly available in NIfTI format through The Cancer Imaging Archive at  https://doi.org/10.7937/tcia.2020.6c7y\\u2010gq39 . Tabular data detailing clinical and technical metadata linked to segmentation cases are also available. Thoracic cavity segmentations will be valuable for developing image analysis pipelines on pathologic lungs \\u2014 where current automated algorithms struggle most. In conjunction with gross tumor volume segmentations already available from \\u201cNSCLC Radiomics,\\u201d pleural effusion segmentations may be valuable for investigating radiomics profile differences between effusion and primary tumor or training algorithms to discriminate between them.\",\"data_availability\":\"\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.7937/K9/TCIA.2015.PF0M9REI'], citation_context: \\\"Aerts \\n HJWL \\n ,  \\n Wee \\n L \\n ,  \\n Rios Velazquez \\n E \\n , et al.  Data from NSCLC\\u2010Radiomics [Dataset] . In:  The Cancer Imaging Archive ;  2019 \\n 10.7937/K9/TCIA.2015.PF0M9REI\\\"}\",\"{\\\"dataset_ids\\\": ['10.7937/tcia.2020.6c7y-gq39'], citation_context: \\\"Kiser \\n KJ \\n ,  \\n Ahmed \\n S \\n ,  \\n Stieb \\n SM \\n , et al.  Data from the thoracic volume and pleural effusion segmentations in diseased lungs for benchmarking chest CT processing pipelines [Dataset] . In:  The Cancer Imaging Archive ;  2020 \\n 10.7937/tcia.2020.6c7y-gq39\\\"}\"]}\n",
         "[{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI', 'type': 'Primary'}, {'dataset_id': '10.7937/tcia.2020.6c7y-gq39', 'type': 'Primary'}]",
         "<think>\nOkay, let's tackle this dataset classification problem. First, I need to look through the provided JSON data to find valid dataset IDs and then determine their types.\n\nStarting with the `data_availability` section. The text mentions \"Thoracic cavity segmentations and discrete pleural effusion segmentations we have annotated on 402 CT scans acquired from patients with non-small cell lung cancer.\" The DOI here is 10.7937/K9/TCIA.2015.PF0M9REI. I need to check if this is a valid dataset ID. The DOI starts with 10.7937... which matches the format. The citation context says it's from the NSCLC Radiomics dataset, which is a valid source. So that's a primary dataset.\n\nNext, the other dataset citation. The second one has 10.7937/tcia.2020.6c7y-gq39. This looks like an Accession ID. The citation context mentions it's for the thoracic volume and pleural effusion segmentations in diseased lungs. Since the data is publicly available in NIfTI format, this is also a primary dataset.\n\nNow, checking for any other potential IDs. The abstract mentions \"Thoracic cavity segmentations will be valuable...\" and \"pleural effusion segmentations may be valuable...\" which points to these datasets. Both have proper DOI and Citation Context, so they are primary.\n\nNow, classifying each. The first dataset uses the DOI, which fits Primary as it's generated and used for the study. The second dataset is an Accession ID, also Primary. The other_dataset_citations have their own citations, but the main data is already covered.\n\nNo missing dataset IDs. So the final output should be an array with both dataset IDs and their types.\n</think>"
        ],
        [
         "28",
         "10.1002_nafm.10870",
         "[{'dataset_id': 'https://doi.org/10.5066/p9gtumay', 'type': 'Primary'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"John R. Farver3 |, Patrick M. Ko\\u010dovsk\\u00fd,6, Nathan D. Stott5, Wiley Periodicals LLC, James M. Long4, Revised, Sandusky, Jeffrey G. Miner5, Silver Chub\",\"abstract\":\"Abstract Objective: Silver Chub Macrhybopsis storeriana, a predominately riverine species  throughout its native range, exists within Lake Erie as the only known lake popu-lation.\\nIts population declined in the 1950s and never fully recovered.\\nCanada has  listed Silver Chub in the Great Lakes-\\u00adSt.\\nLawrence River as endangered and has  initiated a recovery plan that recognized the identification of spawning areas as a  critical component to inform Silver Chub's recovery potential.\\nMethods: We investigated potential spawning locations of Silver Chub using cap-ture records, otolith microchemistry, and daily age analysis.\\nLapillus otolith\\nSr:Ca ra-tios from 27 age-\\u00ad0 Silver Chub were used to identify potential spawning areas.\\nDaily  ages estimated from lapilli were used to calculate hatch dates, which then were com-pared with capture data of adults and river flows to further inform potential spawn-ing areas.\\nResult: The Detroit River (and its nearshore area) was all but ruled out as a potential  spawning location.\\nThe Maumee, Portage, and Sandusky rivers or their nearshore  areas were all possible spawning locations.\\nProjected hatch dates spanned the end  of May through the end of June and occurred across a wide range of flows, although  some peaks in hatch dates corresponded to flow peaks, indicating recruitment is po-tentially enhanced by high flows.\\nConclusion: Silver Chub spawning period and hypothesized spawning rivers or  lacustuaries overlap those of invasive Grass Carp Ctenopharyngodon idella, creat-ing a need to jointly consider Grass Carp control efforts with conservation of Silver  Chub when assessing management alternatives.\\nFurther research on spawning  guild and the use of rivers themselves or nearshore areas influenced by rivers as  spawning areas are required to maximize potential for conservation and recovery  of Silver Chub.\",\"data_availability\":\"DATA AVAILABILITY STATEMENT U.S. Geological Survey data are publicly available at the  websites in the references section.\\nWestern Basin intera-gency August trawl survey data are available from the For-age Task Group of the Lake Erie Committee (http://www. glfc.org/lake-erie-committee.php) by request.\\nData col-lected by the U.S. Fish and Wildlife Service are available  upon request from Anjanette Bowen (anjanette_bowen@ fws.gov).\\nSilver Chub otolith microchemistry data\\nare  available upon request from John Farver or Jeff Miner  (jfarver@bgsu.edu or jminer@bgsu.edu, respectively).\",\"other_dataset_citations\":[\"{\\\"dataset_ids\\\": ['10.5066/P9GTUMAY.'], citation_context: \\\"associated species.\\nDoctoral dissertation.\\nThe Ohio State  University, Columbus.\\nKo\\u010dovsk\\u00fd, P. M., and J. Farver. 2019.\\nWater chemistry of Great Lakes  tributaries, 2017-\\u00ad2018.\\nU.S. Geological Survey data release.  \\nAvailable: https://doi.org/10.5066/P9GTUMAY.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. o\\\"},{\\\"dataset_ids\\\": ['10.5066/P94BH3W0.'], citation_context: \\\"\\u010dovsk\\u00fd, P. M., N. R. King, C. M. Merkes, and T. M. Schreier.  2018.\\nDevelopmental stages of Grass Carp (Ctenopharyngodon  idella) eggs in the Sandusky River (version 3.0, July 2020).  \\nU.S. Geological Survey data release.\\nAvailable: https://doi. org/10.5066/P94BH3W0.\\nKo\\u010dovsk\\u00fd, P. M., N. R. King, E. Weimer, C. Mayer, and S. S. Qian.  2021.\\nValidation of the model-\\u00adpredicted spawning area of Grass  Carp Ctenopharyngodon idella in the Sandusky River.\\nJournal  of Great Lakes Research 47:29-36.\\nKoel, T. M., and R. E.\\\"}\",\"{\\\"dataset_ids\\\": ['10.5066/F75M63X0.'], citation_context: \\\" Fisheries Society, Bethesda, Maryland.\\nUSGS (U.S. Geological Survey). 2019a.\\nGreat Lakes Science Center  Great Lakes research vessel operations 1958-\\u00ad2018.\\n(version  3.0, April 2019).\\nU.S. Geological Survey data release.\\nAvailable:  https://doi.org/10.5066/F75M63X0.\\nUSGS (U.S. Geological Survey). 2019b.\\nNewly hatched invasive Grass  Carp found in Maumee River, OH.\\nU.S. Geological Survey,  Office of Communications and Publishing, Reston, Virginia.  \\nAvailable:\\nwww.usgs.gov/news/newly-\\u00adhatched-\\u00adinvasive-\\u00adgrass -\\u00ad\\\"}\"]}\n",
         "[{'dataset_id': '10.5066/P9GTUMAY', 'type': 'Primary'}, {'dataset_id': '10.5066/F75M63X0', 'type': 'Secondary'}]",
         "<think>\nOkay, let's tackle this dataset classification problem. First, I need to parse the input JSON and extract the relevant sections. The user provided a JSON string with sections like title, authors, abstract, data_availability, and other_dataset_citations.\n\nStarting with the data_availability section, there's a statement about U.S. Geological Survey data. I should check if that's a dataset. The data_availability mentions \"Silver Chub otolith microchemistry data\" and \"Western Basin intera-gency August trawl survey data\". These look like data sources, so they might be datasets. But I need to verify if they are actual dataset IDs.\n\nLooking at the citation_context for the first dataset: \"associated species.\" That doesn't seem to indicate a dataset ID. The DOI is 10.5066/P9GTUMAY. The citation says \"associated species,\" which might be a citation context, not the dataset ID. So, maybe the DOI is valid but the citation context isn't. The other dataset citation has a DOI of 10.5066/F75M63X0, which also looks like a dataset ID. But again, the citation context here is \"Fisheries Society, Bethesda, Maryland.\" Not sure if that's relevant.\n\nNow, the other_dataset_citations include two entries. The first one has a citation context mentioning \"associated species,\" which is probably a citation. The second one has a citation context about \"newly hatched invasive Grass Carp found in Maumee River,\" which is a study on the same topic. Both these entries might be dataset citations. \n\nNext, I need to check if these are actually dataset IDs. The first dataset has a DOI, which is a valid format. The second one has a DOI, which is also valid. Both seem to be dataset IDs. The citation contexts are part of the data availability, but they don't directly indicate the dataset IDs themselves.\n\nNow, applying the classification rules. The abstract mentions the study on Silver Chub and its spawning, which is directly related to the research. The data_availability mentions U.S. Geological Survey data, which are datasets. The other_dataset_citations refer to studies on the same topic, so they are secondary datasets as they are reused from existing data.\n\nWait, the abstract says that the data was used for investigating potential spawning locations. So the data_availability should be the primary dataset. The other citations are secondary because they are from studies that were used, not generated by the authors. \n\nSo, the valid dataset IDs are the two DOIs found in the other_dataset_citations. The citation contexts don't provide the exact dataset IDs, but the DOIs are valid. The other dataset citations have their own DOIs, which are valid. \n\nTherefore, the results should list both dataset IDs as Primary and Secondary, based on their relevance to the research. The missing dataset ID would be if none were found, but in this case, both are present. \n\nI need to make sure not to extract any other information beyond the dataset IDs. Also, check for any other possible dataset IDs in the data_availability section, but the text doesn't mention any other dataset IDs besides the ones in the citations. \n\nFinally, format the JSON as specified, ensuring that each dataset is correctly classified as Primary or Secondary.\n</think>"
        ],
        [
         "29",
         "10.1007_jhep07(2018)134",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "PDF",
         "Text Content: {\"title\":\"\",\"authors\":\"TeV, Copyright CERN\",\"abstract\":\"Abstract: The production cross-sections of \\u03a5(1S), \\u03a5(2S) and \\u03a5(3S) mesons in proton-proton collisions at \\u221as = 13 TeV are measured with a data sample corresponding to an integrated luminosity of 277 \\u00b1 11 pb\\u22121 recorded by the LHCb experiment in 2015.\\nThe \\u03a5 mesons are reconstructed in the decay mode \\u03a5 \\u2192\\u00b5+\\u00b5\\u2212.\\nThe differential production cross-sections times the dimuon branching fractions are measured as a function of the \\u03a5 transverse momentum, pT, and rapidity, y, over the range 0 < pT < 30 GeV/c and 2.0 < y < 4.5.\\nThe ratios of the cross-sections with respect to the LHCb measurement at \\u221as = 8 TeV are also determined.\\nThe measurements are compared with theoretical predictions based on NRQCD.\",\"data_availability\":\"\",\"other_dataset_citations\":[]}\n",
         "[{'dataset_id': 'Missing', 'type': 'Missing'}]",
         "<think>\nOkay, let's see. The user provided a JSON object with the text content of a paper. My task is to extract valid dataset IDs and classify them as Primary or Secondary. \n\nFirst, I need to check the data_availability section. The abstract mentions data from the LHCb experiment, but the data_availability is empty here. Then, there are no other_dataset_citations entries. \n\nLooking at the data_availability, since it's empty, there are no datasets mentioned. The abstract talks about measurements of meson cross-sections, but no dataset IDs. The other sections are empty. \n\nSo, according to Task 1, there are no valid dataset IDs found. Therefore, I should output a JSON array with the dataset_id as \"Missing\" and type as \"Missing\". \n\nWait, the user's instructions say to check the data_availability and other_dataset_citations. In this case, both are empty. So no dataset IDs. Hence, the answer is just the \"Missing\" JSON.\n</think>"
        ]
       ],
       "shape": {
        "columns": 6,
        "rows": 30
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>article_id</th>\n",
       "      <th>dataset_info</th>\n",
       "      <th>text_type</th>\n",
       "      <th>llm_input</th>\n",
       "      <th>llm_response</th>\n",
       "      <th>llm_thinking_content</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002_2017jc013030</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.17882/4938...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Assessing the variabil...</td>\n",
       "      <td>[{'dataset_id': '10.17882/49388', 'type': 'Pri...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem step ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002_anie.201916483</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Effective and Reversib...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002_anie.202005531</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Trifluoromethyl Sulfox...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this. The user pro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002_anie.202007717</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Metal\\u2013Ligand Coop...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002_chem.201902131</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"A Synthetic Route Towa...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem step ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10.1002_chem.201903120</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Structure\\u2013Solubil...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let me start by analyzing the p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10.1002_chem.202000235</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"An Isolable Bis(Silano...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by analyzing the pr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10.1002_chem.202001412</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Iron(0)\\u2010Mediated ...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>10.1002_chem.202001668</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"The First Use of a ReX...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10.1002_chem.202003167</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Metal\\u2010only Lewis ...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10.1002_cssc.202201821</td>\n",
       "      <td>NaN</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Development of a Mg/O ...</td>\n",
       "      <td>[{'dataset_id': '10.5281/zenodo.7074790', 'typ...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10.1002_ece3.3985</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"High levels of polyand...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>10.1002_ece3.4466</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Incubation temperature...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>10.1002_ece3.5260</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Ultraconserved element...</td>\n",
       "      <td>[{'dataset_id': '10.5061/dryad.2f62927', 'type...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>10.1002_ece3.5395</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"\\u201cCloser\\u2010to\\u...</td>\n",
       "      <td>[{'dataset_id': '10.5441/001/1.v1cs4nn0', 'typ...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>10.1002_ece3.6144</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Efficacy of metabarcod...</td>\n",
       "      <td>[{'dataset_id': '10.5061/dryad.zw3r22854', 'ty...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>10.1002_ece3.6303</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Can polyploidy confer ...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>10.1002_ece3.6784</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Ecomorphology of the p...</td>\n",
       "      <td>[{'dataset_id': '10.1007/s00435-017-0386-3', '...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>10.1002_ece3.961</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"The effects of insects...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>10.1002_ece3.9627</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Inferring predator\\u20...</td>\n",
       "      <td>[{'dataset_id': '10.5061/dryad.b8gtht7h3', 'ty...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>10.1002_ecs2.1280</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"Catton, R...</td>\n",
       "      <td>[{'dataset_id': '10.1002/ecs2.1280', 'type': '...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>10.1002_ecs2.4619</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.25349/d9qw...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Interactions between t...</td>\n",
       "      <td>[{'dataset_id': '10.25349/D9QW5X', 'type': 'Pr...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>10.1002_ejic.201900904</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"\",\"abstra...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this. The input is...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>10.1002_ejoc.202000139</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"Ann Chris...</td>\n",
       "      <td>[{'dataset_id': '10.1002/ejoc.202000139', 'typ...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem step ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>10.1002_ejoc.202000916</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"Ammonium Salt\\u2010Cat...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's start by looking at the p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>10.1002_esp.5058</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5061/dryad...</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"Yasuyuki,...</td>\n",
       "      <td>[{'dataset_id': '10.1002/esp.4637', 'type': 'P...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem step ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>10.1002_esp.5090</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5066/p9353...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"\",\"abstra...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this problem. The ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>10.1002_mp.14424</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.7937/k9/tc...</td>\n",
       "      <td>XML</td>\n",
       "      <td>Text Content: {\"title\":\"PleThora: Pleural effu...</td>\n",
       "      <td>[{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this dataset class...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>10.1002_nafm.10870</td>\n",
       "      <td>[{'dataset_id': 'https://doi.org/10.5066/p9gtu...</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"John R. F...</td>\n",
       "      <td>[{'dataset_id': '10.5066/P9GTUMAY', 'type': 'P...</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's tackle this dataset class...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>10.1007_jhep07(2018)134</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>PDF</td>\n",
       "      <td>Text Content: {\"title\":\"\",\"authors\":\"TeV, Copy...</td>\n",
       "      <td>[{'dataset_id': 'Missing', 'type': 'Missing'}]</td>\n",
       "      <td>&lt;think&gt;\\nOkay, let's see. The user provided a ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 article_id  \\\n",
       "0      10.1002_2017jc013030   \n",
       "1    10.1002_anie.201916483   \n",
       "2    10.1002_anie.202005531   \n",
       "3    10.1002_anie.202007717   \n",
       "4    10.1002_chem.201902131   \n",
       "5    10.1002_chem.201903120   \n",
       "6    10.1002_chem.202000235   \n",
       "7    10.1002_chem.202001412   \n",
       "8    10.1002_chem.202001668   \n",
       "9    10.1002_chem.202003167   \n",
       "10   10.1002_cssc.202201821   \n",
       "11        10.1002_ece3.3985   \n",
       "12        10.1002_ece3.4466   \n",
       "13        10.1002_ece3.5260   \n",
       "14        10.1002_ece3.5395   \n",
       "15        10.1002_ece3.6144   \n",
       "16        10.1002_ece3.6303   \n",
       "17        10.1002_ece3.6784   \n",
       "18         10.1002_ece3.961   \n",
       "19        10.1002_ece3.9627   \n",
       "20        10.1002_ecs2.1280   \n",
       "21        10.1002_ecs2.4619   \n",
       "22   10.1002_ejic.201900904   \n",
       "23   10.1002_ejoc.202000139   \n",
       "24   10.1002_ejoc.202000916   \n",
       "25         10.1002_esp.5058   \n",
       "26         10.1002_esp.5090   \n",
       "27         10.1002_mp.14424   \n",
       "28       10.1002_nafm.10870   \n",
       "29  10.1007_jhep07(2018)134   \n",
       "\n",
       "                                         dataset_info text_type  \\\n",
       "0   [{'dataset_id': 'https://doi.org/10.17882/4938...       XML   \n",
       "1      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "2      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "3      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "4      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "5      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "6      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "7      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "8      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "9      [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "10                                                NaN       XML   \n",
       "11     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "12  [{'dataset_id': 'https://doi.org/10.5061/dryad...       XML   \n",
       "13  [{'dataset_id': 'https://doi.org/10.5061/dryad...       XML   \n",
       "14     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "15  [{'dataset_id': 'https://doi.org/10.5061/dryad...       XML   \n",
       "16  [{'dataset_id': 'https://doi.org/10.5061/dryad...       XML   \n",
       "17     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "18     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "19  [{'dataset_id': 'https://doi.org/10.5061/dryad...       XML   \n",
       "20  [{'dataset_id': 'https://doi.org/10.5061/dryad...       PDF   \n",
       "21  [{'dataset_id': 'https://doi.org/10.25349/d9qw...       XML   \n",
       "22     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "23     [{'dataset_id': 'Missing', 'type': 'Missing'}]       PDF   \n",
       "24     [{'dataset_id': 'Missing', 'type': 'Missing'}]       XML   \n",
       "25  [{'dataset_id': 'https://doi.org/10.5061/dryad...       PDF   \n",
       "26  [{'dataset_id': 'https://doi.org/10.5066/p9353...       XML   \n",
       "27  [{'dataset_id': 'https://doi.org/10.7937/k9/tc...       XML   \n",
       "28  [{'dataset_id': 'https://doi.org/10.5066/p9gtu...       PDF   \n",
       "29     [{'dataset_id': 'Missing', 'type': 'Missing'}]       PDF   \n",
       "\n",
       "                                            llm_input  \\\n",
       "0   Text Content: {\"title\":\"Assessing the variabil...   \n",
       "1   Text Content: {\"title\":\"Effective and Reversib...   \n",
       "2   Text Content: {\"title\":\"Trifluoromethyl Sulfox...   \n",
       "3   Text Content: {\"title\":\"Metal\\u2013Ligand Coop...   \n",
       "4   Text Content: {\"title\":\"A Synthetic Route Towa...   \n",
       "5   Text Content: {\"title\":\"Structure\\u2013Solubil...   \n",
       "6   Text Content: {\"title\":\"An Isolable Bis(Silano...   \n",
       "7   Text Content: {\"title\":\"Iron(0)\\u2010Mediated ...   \n",
       "8   Text Content: {\"title\":\"The First Use of a ReX...   \n",
       "9   Text Content: {\"title\":\"Metal\\u2010only Lewis ...   \n",
       "10  Text Content: {\"title\":\"Development of a Mg/O ...   \n",
       "11  Text Content: {\"title\":\"High levels of polyand...   \n",
       "12  Text Content: {\"title\":\"Incubation temperature...   \n",
       "13  Text Content: {\"title\":\"Ultraconserved element...   \n",
       "14  Text Content: {\"title\":\"\\u201cCloser\\u2010to\\u...   \n",
       "15  Text Content: {\"title\":\"Efficacy of metabarcod...   \n",
       "16  Text Content: {\"title\":\"Can polyploidy confer ...   \n",
       "17  Text Content: {\"title\":\"Ecomorphology of the p...   \n",
       "18  Text Content: {\"title\":\"The effects of insects...   \n",
       "19  Text Content: {\"title\":\"Inferring predator\\u20...   \n",
       "20  Text Content: {\"title\":\"\",\"authors\":\"Catton, R...   \n",
       "21  Text Content: {\"title\":\"Interactions between t...   \n",
       "22  Text Content: {\"title\":\"\",\"authors\":\"\",\"abstra...   \n",
       "23  Text Content: {\"title\":\"\",\"authors\":\"Ann Chris...   \n",
       "24  Text Content: {\"title\":\"Ammonium Salt\\u2010Cat...   \n",
       "25  Text Content: {\"title\":\"\",\"authors\":\"Yasuyuki,...   \n",
       "26  Text Content: {\"title\":\"\",\"authors\":\"\",\"abstra...   \n",
       "27  Text Content: {\"title\":\"PleThora: Pleural effu...   \n",
       "28  Text Content: {\"title\":\"\",\"authors\":\"John R. F...   \n",
       "29  Text Content: {\"title\":\"\",\"authors\":\"TeV, Copy...   \n",
       "\n",
       "                                         llm_response  \\\n",
       "0   [{'dataset_id': '10.17882/49388', 'type': 'Pri...   \n",
       "1      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "2      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "3      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "4      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "5      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "6      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "7      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "8      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "9      [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "10  [{'dataset_id': '10.5281/zenodo.7074790', 'typ...   \n",
       "11     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "12     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "13  [{'dataset_id': '10.5061/dryad.2f62927', 'type...   \n",
       "14  [{'dataset_id': '10.5441/001/1.v1cs4nn0', 'typ...   \n",
       "15  [{'dataset_id': '10.5061/dryad.zw3r22854', 'ty...   \n",
       "16     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "17  [{'dataset_id': '10.1007/s00435-017-0386-3', '...   \n",
       "18     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "19  [{'dataset_id': '10.5061/dryad.b8gtht7h3', 'ty...   \n",
       "20  [{'dataset_id': '10.1002/ecs2.1280', 'type': '...   \n",
       "21  [{'dataset_id': '10.25349/D9QW5X', 'type': 'Pr...   \n",
       "22     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "23  [{'dataset_id': '10.1002/ejoc.202000139', 'typ...   \n",
       "24     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "25  [{'dataset_id': '10.1002/esp.4637', 'type': 'P...   \n",
       "26     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "27  [{'dataset_id': '10.7937/K9/TCIA.2015.PF0M9REI...   \n",
       "28  [{'dataset_id': '10.5066/P9GTUMAY', 'type': 'P...   \n",
       "29     [{'dataset_id': 'Missing', 'type': 'Missing'}]   \n",
       "\n",
       "                                 llm_thinking_content  \n",
       "0   <think>\\nOkay, let's tackle this problem step ...  \n",
       "1   <think>\\nOkay, let's tackle this problem. The ...  \n",
       "2   <think>\\nOkay, let's tackle this. The user pro...  \n",
       "3   <think>\\nOkay, let's tackle this problem. The ...  \n",
       "4   <think>\\nOkay, let's tackle this problem step ...  \n",
       "5   <think>\\nOkay, let me start by analyzing the p...  \n",
       "6   <think>\\nOkay, let's start by analyzing the pr...  \n",
       "7   <think>\\nOkay, let's tackle this problem. The ...  \n",
       "8   <think>\\nOkay, let's tackle this problem. The ...  \n",
       "9   <think>\\nOkay, let's tackle this problem. The ...  \n",
       "10  <think>\\nOkay, let's start by looking at the i...  \n",
       "11  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "12  <think>\\nOkay, let's start by looking at the i...  \n",
       "13  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "14  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "15  <think>\\nOkay, let's start by looking at the i...  \n",
       "16  <think>\\nOkay, let's start by looking at the i...  \n",
       "17  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "18  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "19  <think>\\nOkay, let's start by looking at the i...  \n",
       "20  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "21  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "22  <think>\\nOkay, let's tackle this. The input is...  \n",
       "23  <think>\\nOkay, let's tackle this problem step ...  \n",
       "24  <think>\\nOkay, let's start by looking at the p...  \n",
       "25  <think>\\nOkay, let's tackle this problem step ...  \n",
       "26  <think>\\nOkay, let's tackle this problem. The ...  \n",
       "27  <think>\\nOkay, let's tackle this dataset class...  \n",
       "28  <think>\\nOkay, let's tackle this dataset class...  \n",
       "29  <think>\\nOkay, let's see. The user provided a ...  "
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_set_processed_articles_df = process_articles(test_file_paths_df, inference_model)\n",
    "test_set_processed_articles_df.to_csv(\"test_set_processed_articles.csv\", index=False)\n",
    "test_set_processed_articles_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4f2525aa",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading Qwen model and tokenizer from: C:\\Users\\jim\\.cache\\kagglehub\\models\\qwen-lm\\qwen-3\\transformers\\0.6b\\1\n"
     ]
    }
   ],
   "source": [
    "eval_model = QwenModelEval(QWEN_BASE_MODEL_PATH, sys_prompt=\"You are a chatbot.\",)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "959c69e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# --- 2. Data Preparation for LLM Training (Revised for Combined Task) ---\n",
    "\n",
    "def load_base_llm_for_training():\n",
    "    \"\"\"Loads the base Qwen model and tokenizer for fine-tuning.\"\"\"\n",
    "    global llm_tokenizer, llm_model\n",
    "    if not AutoModelForCausalLM or not QWEN_BASE_MODEL_PATH:\n",
    "        print(\"LLM components not available or base model path not set. Skipping LLM loading.\")\n",
    "        return False\n",
    "    try:\n",
    "        print(f\"Loading Qwen tokenizer from: {QWEN_BASE_MODEL_PATH}\")\n",
    "        llm_tokenizer = AutoTokenizer.from_pretrained(QWEN_BASE_MODEL_PATH, trust_remote_code=True)\n",
    "        if llm_tokenizer.pad_token is None:\n",
    "            llm_tokenizer.pad_token = llm_tokenizer.eos_token\n",
    "            print(\"Set tokenizer.pad_token to tokenizer.eos_token\")\n",
    "\n",
    "        print(f\"Loading Qwen model from: {QWEN_BASE_MODEL_PATH}\")\n",
    "        llm_model = AutoModelForCausalLM.from_pretrained(\n",
    "            QWEN_BASE_MODEL_PATH,\n",
    "            torch_dtype=torch.bfloat16 if torch.cuda.is_available() and torch.cuda.is_bf16_supported() else torch.float32,\n",
    "            device_map=\"auto\", # Automatically uses GPU if available\n",
    "            trust_remote_code=True,\n",
    "            # load_in_8bit=True if bnb else False # Uncomment if bitsandbytes is used\n",
    "        )\n",
    "        print(f\"Base LLM loaded successfully on {llm_model.device}.\")\n",
    "        return True\n",
    "    except Exception as e:\n",
    "        print(f\"Error loading base LLM for training: {e}\")\n",
    "        llm_tokenizer, llm_model = None, None # Reset to None on failure\n",
    "        return False\n",
    "\n",
    "def prepare_training_data_for_llm(\n",
    "    training_df: pd.DataFrame,\n",
    "    all_article_texts: dict[str, str],\n",
    "    tokenizer_max_length: int\n",
    ") -> Dataset:\n",
    "    \"\"\"\n",
    "    Prepares training data for LLM fine-tuning, aggregating dataset IDs and classifications\n",
    "    per article and formatting into ChatML JSON output.\n",
    "    \"\"\"\n",
    "    formatted_examples = []\n",
    "\n",
    "    # Group training data by article_id to get all datasets for each article\n",
    "    # This creates a dictionary where keys are article_ids and values are lists of dataset dicts\n",
    "    grouped_training_data = training_df.groupby('article_id').apply(\n",
    "        lambda x: [{\"dataset_id\": row['dataset_id'], \"classification\": row['label']} for _, row in x]\n",
    "    ).to_dict()\n",
    "\n",
    "    # Get all article IDs for which we have text content\n",
    "    all_article_ids_with_text = set(all_article_texts.keys())\n",
    "    \n",
    "    # Iterate through all articles for which we have text (these are our potential training examples)\n",
    "    for article_id in all_article_ids_with_text:\n",
    "        article_text = all_article_texts.get(article_id, \"\")\n",
    "        if not article_text:\n",
    "            print(f\"Warning: Article text for {article_id} not found. Skipping training example.\")\n",
    "            continue\n",
    "\n",
    "        # Truncate article text to fit within the model's context window\n",
    "        # Reserve tokens for the prompt and the expected JSON response.\n",
    "        # A typical Qwen 1.5 model has 32768 max_seq_length.\n",
    "        # 512 tokens for prompt/response is a safe estimate.\n",
    "        truncated_article_text = article_text[:tokenizer_max_length - 512] \n",
    "\n",
    "        # Determine the ground truth output for this article\n",
    "        if article_id in grouped_training_data:\n",
    "            # Article has datasets, format them as JSON\n",
    "            ground_truth_datasets = grouped_training_data[article_id]\n",
    "            assistant_response_json = json.dumps(ground_truth_datasets, ensure_ascii=False)\n",
    "        else:\n",
    "            # Article has no datasets in training data, so the model should output an empty list.\n",
    "            # This explicitly trains the model to output '[]' for \"Missing\" cases.\n",
    "            assistant_response_json = \"[]\"\n",
    "            # print(f\"Info: Article {article_id} has no datasets in training data. Training to output '[]'.\")\n",
    "\n",
    "        # Construct the user message for the LLM\n",
    "        user_message = f\"\"\"\n",
    "Article Text:\n",
    "{truncated_article_text}\n",
    "\n",
    "Task: Identify all datasets or databases used in this research article and classify each as \"Primary\" (if created by the authors for this research) or \"Secondary\" (if an existing dataset used in this research).\n",
    "\n",
    "Output Format: Provide a JSON list of objects. Each object should have \"dataset_id\" and \"classification\" keys. If no datasets are identified, return an empty JSON list: [].\n",
    "\"\"\"\n",
    "        # Construct the full ChatML formatted string for SFTTrainer\n",
    "        # The trainer will use this entire string as the 'text' field.\n",
    "        chatml_formatted_string = f\"<|im_start|>system\\nYou are an expert research assistant. Your task is to extract and classify datasets from scientific articles.<|im_end|>\\n<|im_start|>user\\n{user_message.strip()}<|im_end|>\\n<|im_start|>assistant\\n{assistant_response_json}<|im_end|>\"\n",
    "        \n",
    "        formatted_examples.append({\"text\": chatml_formatted_string})\n",
    "\n",
    "    if not formatted_examples:\n",
    "        raise ValueError(\"No training examples could be prepared. Check your data and article texts.\")\n",
    "\n",
    "    return Dataset.from_list(formatted_examples)\n",
    "\n",
    "# --- 3. LLM Model Training (Fine-tuning) ---\n",
    "\n",
    "# Attempt to load tokenizer and model if not already loaded (e.g., if previous training failed or was skipped)\n",
    "if llm_model is None:\n",
    "    load_base_llm_for_training()\n",
    "\n",
    "if llm_model and not training_df.empty and Dataset: # Ensure Dataset is imported\n",
    "    print(\"\\n--- Preparing data for Fine-tuning (Combined Task) ---\")\n",
    "    # Use the model's max_length for context, or a reasonable default if tokenizer isn't loaded\n",
    "    max_len = llm_tokenizer.model_max_length if llm_tokenizer else 4096 \n",
    "    train_dataset = prepare_training_data_for_llm(training_df, all_article_texts, max_len)\n",
    "    \n",
    "    print(f\"Prepared {len(train_dataset)} examples for fine-tuning.\")\n",
    "    print(\"Example formatted training instance (first 500 chars):\")\n",
    "    print(train_dataset[0]['text'][:500])\n",
    "\n",
    "    print(\"\\n--- Starting Fine-tuning (Combined Task) ---\")\n",
    "    try:\n",
    "        training_args = TrainingArguments(\n",
    "            output_dir=f\"{FINE_TUNED_MODEL_OUTPUT_DIR}/checkpoints\",\n",
    "            num_train_epochs=1,  # Start with 1 epoch, adjust as needed\n",
    "            per_device_train_batch_size=1, # Adjust based on VRAM\n",
    "            gradient_accumulation_steps=4, # Effective batch size = 1 * 4 = 4\n",
    "            learning_rate=2e-5,\n",
    "            logging_steps=10,\n",
    "            save_steps=50, # Save checkpoints periodically\n",
    "            fp16=torch.cuda.is_available() and not torch.cuda.is_bf16_supported(),\n",
    "            bf16=torch.cuda.is_available() and torch.cuda.is_bf16_supported(),\n",
    "            optim=\"paged_adamw_8bit\", # Good for memory efficiency if bitsandbytes is installed\n",
    "            # report_to=\"none\", # Disable logging to external services\n",
    "            # max_steps=100, # For quick testing\n",
    "        )\n",
    "\n",
    "        trainer = SFTTrainer(\n",
    "            model=llm_model,\n",
    "            tokenizer=llm_tokenizer,\n",
    "            train_dataset=train_dataset,\n",
    "            dataset_text_field=\"text\", # This field contains the full ChatML string\n",
    "            args=training_args,\n",
    "            max_seq_length=max_len, # Use the model's full context length\n",
    "            packing=False, # Set to True if your inputs are much shorter than max_seq_length\n",
    "        )\n",
    "\n",
    "        trainer.train()\n",
    "        print(\"Fine-tuning completed.\")\n",
    "\n",
    "        print(f\"Saving fine-tuned model to: {FINE_TUNED_MODEL_OUTPUT_DIR}\")\n",
    "        trainer.save_model(FINE_TUNED_MODEL_OUTPUT_DIR)\n",
    "        print(\"Model and tokenizer saved.\")\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"An error occurred during fine-tuning: {e}\")\n",
    "        import traceback\n",
    "        traceback.print_exc()\n",
    "        llm_model = None # Mark model as failed to load/train\n",
    "else:\n",
    "    print(\"Skipping LLM fine-tuning due to missing training data or LLM components.\")\n",
    "\n",
    "\n",
    "# --- 4. LLM-based Extraction & Classification (Inference) ---\n",
    "\n",
    "# Load the fine-tuned model for inference (if training was successful)\n",
    "# If training was skipped or failed, this will attempt to load from the base path or fail.\n",
    "if inference_model is None: # Only load if not already loaded\n",
    "    if AutoModelForCausalLM: # Check if transformers is available\n",
    "        if os.path.exists(FINE_TUNED_MODEL_OUTPUT_DIR) and os.path.isdir(FINE_TUNED_MODEL_OUTPUT_DIR):\n",
    "            MODEL_TO_LOAD = FINE_TUNED_MODEL_OUTPUT_DIR\n",
    "            print(f\"Loading fine-tuned model for inference from: {MODEL_TO_LOAD}\")\n",
    "        else:\n",
    "            MODEL_TO_LOAD = QWEN_BASE_MODEL_PATH\n",
    "            print(f\"Fine-tuned model not found. Loading base model for inference from: {MODEL_TO_LOAD}\")\n",
    "\n",
    "        try:\n",
    "            inference_tokenizer = AutoTokenizer.from_pretrained(MODEL_TO_LOAD, trust_remote_code=True)\n",
    "            if inference_tokenizer.pad_token is None:\n",
    "                inference_tokenizer.pad_token = inference_tokenizer.eos_token\n",
    "            inference_model = AutoModelForCausalLM.from_pretrained(\n",
    "                MODEL_TO_LOAD,\n",
    "                torch_dtype=torch.bfloat16 if torch.cuda.is_available() and torch.cuda.is_bf16_supported() else torch.float32,\n",
    "                device_map=\"auto\",\n",
    "                trust_remote_code=True\n",
    "            ).eval() # Set to evaluation mode\n",
    "            print(f\"Inference LLM loaded successfully on {inference_model.device}.\")\n",
    "        except Exception as e:\n",
    "            print(f\"Error loading inference LLM from {MODEL_TO_LOAD}: {e}\")\n",
    "            inference_model, inference_tokenizer = None, None\n",
    "    else:\n",
    "        print(\"Transformers library not available. Cannot load LLM for inference.\")\n",
    "\n",
    "\n",
    "def extract_and_classify_with_llm(article_text: str) -> list[dict]:\n",
    "    \"\"\"\n",
    "    Uses the loaded LLM to extract dataset IDs and classify them.\n",
    "    Returns a list of dictionaries like [{\"dataset_id\": \"...\", \"classification\": \"...\"}].\n",
    "    Returns an empty list if LLM is unavailable or parsing fails.\n",
    "    \"\"\"\n",
    "    if not inference_model or not inference_tokenizer:\n",
    "        print(\"  LLM unavailable for extraction/classification.\")\n",
    "        return [] # Return empty list if LLM is not loaded\n",
    "\n",
    "    # Truncate article text for inference if it exceeds model's context window\n",
    "    # Use the same max_length as during training for consistency\n",
    "    max_inference_context_length = inference_tokenizer.model_max_length - 256 # Reserve tokens for prompt and response\n",
    "    truncated_article_text = article_text[:max_inference_context_length]\n",
    "\n",
    "    user_message = f\"\"\"\n",
    "Article Text:\n",
    "{truncated_article_text}\n",
    "\n",
    "Task: Identify all datasets or databases used in this research article and classify each as \"Primary\" (if created by the authors for this research) or \"Secondary\" (if an existing dataset used in this research).\n",
    "\n",
    "Output Format: Provide a JSON list of objects. Each object should have \"dataset_id\" and \"classification\" keys. If no datasets are identified, return an empty JSON list: [].\n",
    "\"\"\"\n",
    "    messages = [\n",
    "        {\"role\": \"system\", \"content\": \"You are an expert research assistant. Your task is to extract and classify datasets from scientific articles.\"},\n",
    "        {\"role\": \"user\", \"content\": user_message.strip()}\n",
    "    ]\n",
    "    \n",
    "    input_ids = inference_tokenizer.apply_chat_template(\n",
    "        messages,\n",
    "        tokenize=True,\n",
    "        add_generation_prompt=True,\n",
    "        return_tensors=\"pt\"\n",
    "    ).to(inference_model.device)\n",
    "\n",
    "    try:\n",
    "        with torch.no_grad():\n",
    "            outputs = inference_model.generate(\n",
    "                input_ids,\n",
    "                max_new_tokens=512, # Allow more tokens for multiple dataset outputs\n",
    "                pad_token_id=inference_tokenizer.eos_token_id,\n",
    "                eos_token_id=inference_tokenizer.convert_tokens_to_ids(\"<|im_end|>\")\n",
    "            )\n",
    "        \n",
    "        response_text = inference_tokenizer.decode(\n",
    "            outputs[0][input_ids.shape[1]:],\n",
    "            skip_special_tokens=False # Keep special tokens to remove <|im_end|> explicitly\n",
    "        ).strip()\n",
    "        response_text = response_text.replace(\"<|im_end|>\", \"\").strip()\n",
    "        \n",
    "        print(f\"  LLM raw response: '{response_text}'\")\n",
    "\n",
    "        # Attempt to parse the JSON output\n",
    "        try:\n",
    "            parsed_data = json.loads(response_text)\n",
    "            if isinstance(parsed_data, list):\n",
    "                # Validate structure: each item should be a dict with 'dataset_id' and 'classification'\n",
    "                valid_datasets = []\n",
    "                for item in parsed_data:\n",
    "                    if isinstance(item, dict) and 'dataset_id' in item and 'classification' in item:\n",
    "                        # Basic validation for classification label\n",
    "                        if item['classification'] in [\"Primary\", \"Secondary\"]:\n",
    "                            valid_datasets.append(item)\n",
    "                        else:\n",
    "                            print(f\"  Warning: Invalid classification '{item['classification']}' for dataset '{item.get('dataset_id', 'N/A')}'. Skipping.\")\n",
    "                    else:\n",
    "                        print(f\"  Warning: Malformed JSON object: {item}. Skipping.\")\n",
    "                return valid_datasets\n",
    "            else:\n",
    "                print(f\"  Warning: LLM did not return a JSON list: {response_text}\")\n",
    "                return []\n",
    "        except json.JSONDecodeError as jde:\n",
    "            print(f\"  Error decoding JSON from LLM response: {jde}. Raw response: '{response_text}'\")\n",
    "            return []\n",
    "\n",
    "    except Exception as e:\n",
    "        print(f\"  Error during LLM generation: {e}\")\n",
    "        return []\n",
    "\n",
    "# --- Main Processing Loop for all articles (Revised) ---\n",
    "print(\"\\n--- Starting Article Processing and Classification (LLM-driven) ---\")\n",
    "final_results = []\n",
    "\n",
    "for article_id, article_text in all_article_texts.items():\n",
    "    print(f\"\\nProcessing article: {article_id}\")\n",
    "    \n",
    "    # LLM directly extracts and classifies\n",
    "    identified_datasets = extract_and_classify_with_llm(article_text)\n",
    "    \n",
    "    if not identified_datasets:\n",
    "        # If LLM returns an empty list, classify the article as \"Missing\"\n",
    "        print(f\"  LLM identified no datasets for {article_id}. Classifying as 'Missing'.\")\n",
    "        final_results.append({\n",
    "            \"article_id\": article_id,\n",
    "            \"dataset_id\": \"N/A\", # Indicate no specific dataset ID\n",
    "            \"classification_label\": \"Missing\"\n",
    "        })\n",
    "    else:\n",
    "        print(f\"  LLM identified {len(identified_datasets)} dataset(s) for {article_id}.\")\n",
    "        for item in identified_datasets:\n",
    "            final_results.append({\n",
    "                \"article_id\": article_id,\n",
    "                \"dataset_id\": item.get(\"dataset_id\", \"Unknown\"), # Use .get() for safety\n",
    "                \"classification_label\": item.get(\"classification\", \"Uncertain_LLM\")\n",
    "            })\n",
    "\n",
    "\n",
    "# --- 5. Results & Output ---\n",
    "\n",
    "print(\"\\n--- Final Results ---\")\n",
    "if final_results:\n",
    "    results_df = pd.DataFrame(final_results)\n",
    "    print(results_df.head(10)) # Print first 10 rows\n",
    "    \n",
    "    # Save to CSV\n",
    "    results_df.to_csv(FINAL_RESULTS_CSV_PATH, index=False)\n",
    "    print(f\"\\nResults saved to: {FINAL_RESULTS_CSV_PATH}\")\n",
    "else:\n",
    "    print(\"No results generated.\")\n",
    "\n",
    "print(\"\\nProcessing complete, Jim!\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
