{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6ec93052",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pymupdf4llm\n",
    "import os\n",
    "import re\n",
    "import json\n",
    "import spacy\n",
    "\n",
    "# Import classes from local utility file\n",
    "import mdc_data_processing_utils\n",
    "\n",
    "# If mdc_data_processing_utils.py has been changed and saved.\n",
    "# To load the changes without restarting the kernel:\n",
    "import importlib\n",
    "importlib.reload(mdc_data_processing_utils)\n",
    "\n",
    "# Now, any calls to functions from mdc_data_processing_utils\n",
    "# will use the newly reloaded code.\n",
    "from mdc_data_processing_utils import (\n",
    "    ArticleData,\n",
    "    DatasetCitation,\n",
    "    LlmTrainingData,\n",
    "    SubmissionData,\n",
    "    MdcFileTextExtractor,\n",
    ")\n",
    "\n",
    "# Load spaCy model for sentence segmentation and potentially other NLP tasks\n",
    "# python -m spacy download en_core_web_sm \n",
    "NLP_SPACY = spacy.load(\"en_core_web_sm\")\n",
    "\n",
    "\n",
    "\n",
    "# Define constants for file paths and model configurations\n",
    "BASE_INPUT_DIR = './kaggle/input/make-data-count-finding-data-references'\n",
    "BASE_OUTPUT_DIR = \"./kaggle/working\"\n",
    "\n",
    "# Define directories for articles in train and test sets\n",
    "TRAIN_DATA_DIR = os.path.join(BASE_INPUT_DIR, 'train')\n",
    "TEST_DATA_DIR = os.path.join(BASE_INPUT_DIR, 'test')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dd70b36a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Extracting md text from file: ./kaggle/input/make-data-count-finding-data-references\\test\\PDF\\10.1002_ece3.5260.pdf\n",
      "Received: 8 April 2019 | Revised: 26 April 2019 | Accepted: 28 April 2019\n",
      "\n",
      "DOI: 10.1002/ece3.5260\n",
      "\n",
      "O R I G I N A L R E S E A R C H\n",
      "# **Ultraconserved element (UCE) probe set design: Base genome ** **and initial design parameters critical for optimization**\n",
      "\n",
      "**[Grey T. Gustafson](mailto:﻿)** **[1,2]** | **Alana Alexander** **[2,3]** | **John S. Sproul** **[4,5]** | **James M. Pflug** **[4]** |\n",
      "\n",
      "**David R. Maddison** **[4]** | **Andrew E. Z. Short** **[1,2]**\n",
      "\n",
      "\n",
      "1 Department of Ecology and Evolutionary\n",
      "\n",
      "Biology, University of Kansas, Lawrence,\n",
      "\n",
      "Kansas\n",
      "\n",
      "2 Biodiversity Institute, University of Kansas,\n",
      "\n",
      "Lawrence, Kansas\n",
      "\n",
      "3 Department of Anatomy, School of\n",
      "\n",
      "Biomedical Sciences, University of Otago,\n",
      "\n",
      "Dunedin, New Zealand\n",
      "\n",
      "4 Department of Integrative Biology, Oregon\n",
      "\n",
      "State University, Corvallis, Oregon\n",
      "\n",
      "5 Department of Biology, University of\n",
      "\n",
      "Rochester, Rochester, New York\n",
      "\n",
      "**Correspondence**\n",
      "\n",
      "Grey T. Gustafson, Department of Ecology\n",
      "\n",
      "and Evolutionary Biology, University of\n",
      "\n",
      "Kansas, Lawrence, Kansas.\n",
      "\n",
      "[Email: gtgustafson@gmail.com](mailto:gtgustafson@gmail.com)\n",
      "\n",
      "**Funding information**\n",
      "\n",
      "National Institutes of Health, Grant/Award\n",
      "\n",
      "Number: 5K12GM063651\n",
      "\n",
      "**1** | **INTRODUCTION**\n",
      "\n",
      "\n",
      "**Abstract**\n",
      "\n",
      "Targeted capture and enrichment approaches have proven effective for phylogenetic\n",
      "\n",
      "study. Ultraconserved elements (UCEs) in particular have exhibited great utility for\n",
      "\n",
      "phylogenomic analyses, with the software package phyluce being among the most uti‐\n",
      "\n",
      "lized pipelines for UCE phylogenomics, including probe design. Despite the success\n",
      "\n",
      "of UCEs, it is becoming increasing apparent that diverse lineages require probe sets\n",
      "\n",
      "tailored to focal taxa in order to improve locus recovery. However, factors affecting\n",
      "\n",
      "probe design and methods for optimizing probe sets to focal taxa remain underex‐\n",
      "\n",
      "plored. Here, we use newly available beetle (Coleoptera) genomic resources to inves‐\n",
      "\n",
      "tigate factors affecting UCE probe set design using phyluce . In particular, we explore\n",
      "\n",
      "the effects of stringency during initial design steps, as well as base genome choice on\n",
      "\n",
      "resulting probe sets and locus recovery. We found that both base genome choice and\n",
      "\n",
      "initial bait design stringency parameters greatly alter the number of resultant probes\n",
      "\n",
      "included in final probe sets and strongly affect the number of loci detected and re‐\n",
      "\n",
      "covered during in silico testing of these probe sets. In addition, we identify attributes\n",
      "\n",
      "of base genomes that correlated with high performance in probe design. Ultimately,\n",
      "\n",
      "we provide a recommended workflow for using P hyluce to design an optimized UCE\n",
      "\n",
      "probe set that will work across a targeted lineage, and use our findings to develop a\n",
      "\n",
      "new, open‐source UCE probe set for beetles of the suborder Adephaga.\n",
      "\n",
      "**K E Y W O R D S**\n",
      "\n",
      "bait design, genomics, phylogenetics, phylogenomics, probe design, ultraconserved elements\n",
      "\n",
      "components that have been targeted, ultraconserved elements\n",
      "\n",
      "(UCEs sensu Faircloth et al., 2012) have demonstrated their util‐\n",
      "\n",
      "\n",
      "The phylogenomics revolution is underway with targeted capture\n",
      "\n",
      "and enrichment approaches (Mamanova et al., 2010; McCormack,\n",
      "\n",
      "Hird, Zellmer, Carstens, & Brumfield, 2013) proving an effective\n",
      "\n",
      "and improved method for phylogenetic studies relative to multilo‐\n",
      "\n",
      "cus Sanger techniques (Blaimer et al., 2015). Among the genomic\n",
      "\n",
      "\n",
      "ity in reconstructing phylogenies across diverse vertebrate [e.g.,\n",
      "\n",
      "fishes (Faircloth, Sorenson, Santini, & Alfaro, 2013), birds (Hosner,\n",
      "\n",
      "Tobias, Braun, & Kimball, 2017; McCormack et al., 2013), reptiles\n",
      "\n",
      "(Crawford et al., 2012; Grismer et al., 2016; Streicher & Wiens,\n",
      "\n",
      "2017), amphibians (Alexander et al., 2017; Streicher et al., 2018),\n",
      "\n",
      "\n",
      "[This is an open access article under the terms of the Creat​ive Commo​ns Attri​bution License, which permits use, distribution and reproduction in any medium,](http://creativecommons.org/licenses/by/4.0/)\n",
      "\n",
      "provided the original work is properly cited.\n",
      "\n",
      "© 2019 The Authors. _Ecology and Evolution_ published by John Wiley & Sons Ltd.\n",
      "\n",
      "_Ecology and Evolution._ 2019;9:6933–6948. www.ecolevol.org ~~[|]~~ **6933**\n",
      "\n",
      "**6934** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "and mammals (Esselstyn, Oliveros, Swanson, & Faircloth, 2017;\n",
      "\n",
      "McCormack et al., 2012)] and other animal lineages [e.g., cnidarians\n",
      "\n",
      "(Quattrini et al., 2018), arachnids (Starrett et al., 2017), hymenopter‐\n",
      "\n",
      "ans (Faircloth, Branstetter, White, & Brady, 2015), and coleopterans\n",
      "\n",
      "(Baca, Alexander, Gustafson, & Short, 2017; Van Dam et al., 2017)],\n",
      "\n",
      "providing resolution for notoriously difficult‐to‐reconstruct evolu‐\n",
      "\n",
      "tionary histories, such as those of placental mammals (Esselstyn et\n",
      "\n",
      "al., 2017) or the rapid radiation of modern birds (McCormack et al.,\n",
      "\n",
      "2013). Additionally, UCEs are phylogenetically informative at mul‐\n",
      "\n",
      "tiple timescales (Faircloth et al., 2012): able to reconstruct both\n",
      "\n",
      "deep (Quattrini et al., 2018) and shallow (Manthey, Campillo, Burns,\n",
      "\n",
      "& Moyle, 2016; Musher & Cracraft, 2018; Smith, Harvey, Faircloth,\n",
      "\n",
      "Glenn, & Brumfield, 2014) evolutionary relationships. Furthermore,\n",
      "\n",
      "UCEs allow for the inclusion of museum specimens (Blaimer, Lloyd,\n",
      "\n",
      "Guillory, & Brady, 2016; McCormack, Tsai, & Faircloth, 2016; Van\n",
      "\n",
      "Dam et al., 2017), even those preserved in formalin (Ruane & Austin,\n",
      "\n",
      "2017), affording improved taxon sampling by removing limitations\n",
      "\n",
      "associated with specimen preservation style.\n",
      "\n",
      "Another advantage to the use of UCEs is the ability to develop\n",
      "\n",
      "probe sets that target thousands of orthologous loci across members\n",
      "\n",
      "of a select organismal group, based on low coverage genomic reads\n",
      "\n",
      "from relatively few exemplar taxa (Faircloth, 2017). While universal\n",
      "\n",
      "probe sets designed to work across larger taxonomic groups such as\n",
      "\n",
      "tetrapods (Faircloth et al., 2012; Sun et al., 2014) have proven suc‐\n",
      "\n",
      "cessful for birds (Hosner et al., 2017; Manthey et al., 2016) and mam‐\n",
      "\n",
      "mals (Esselstyn et al., 2017), it is becoming increasingly apparent that\n",
      "\n",
      "for other diverse animal lineages probe sets tailored to focal taxa re‐\n",
      "\n",
      "sult in considerably improved locus recovery. For example, the first\n",
      "\n",
      "Hymenoptera UCE probe set, designed largely using the genome\n",
      "\n",
      "of a single parasitoid wasp species, targeted a total of 1,500 UCE\n",
      "\n",
      "loci, but captured decreasing numbers of loci with increasing phy‐\n",
      "\n",
      "logenetic distance from this taxon, with one highly divergent study\n",
      "\n",
      "species within Hymenoptera having as few as 341 loci enriched by\n",
      "\n",
      "the probe set (Faircloth et al., 2015). Thus, Branstetter, Longino,\n",
      "\n",
      "Ward, and Faircloth (2017) subsequently developed an expanded\n",
      "\n",
      "Hymenoptera UCE probe set tailored for use in ants, which success‐\n",
      "\n",
      "fully increased locus capture for target ant taxa, but similarly showed\n",
      "\n",
      "decreased capture rates for more distantly related ants, bees, and\n",
      "\n",
      "wasps. In another example, Faircloth (2017) designed a Coleoptera\n",
      "\n",
      "UCE probe set, primarily using genomes from the beetle suborder\n",
      "\n",
      "Polyphaga. This probe set, targeting 1,172 UCE loci, only recovered\n",
      "\n",
      "305 loci in 50% of the taxa studied in a phylogenomic analysis of\n",
      "\n",
      "the distantly related beetle suborder Adephaga (Baca et al., 2017). A\n",
      "\n",
      "second investigation by Van Dam et al., 2017 utilized this same probe\n",
      "\n",
      "set for a phylogenomic analysis of weevils (family Curculionidae),\n",
      "\n",
      "members of the same suborder of beetles used during the probe\n",
      "\n",
      "design (Faircloth, 2017). Although capturing considerably more loci\n",
      "\n",
      "(537 loci at 50% data matrix completeness) than the prior study, it\n",
      "\n",
      "still resulted in significantly fewer UCE loci for phylogenetic analysis\n",
      "\n",
      "(368 in the 70% complete data matrix) (Van Dam et al., 2017) than\n",
      "\n",
      "the probe set was designed to provide. This is likely due to the sin‐\n",
      "\n",
      "gle representative weevil genome used ( _Dendroctonus ponderosae_ )\n",
      "\n",
      "in the design of the Coleoptera UCE probe set by Faircloth (2017)\n",
      "\n",
      "\n",
      "being only distantly related to those in the Van Dam et al. (2017)\n",
      "\n",
      "study.\n",
      "\n",
      "In spite of the apparent need for probe sets tailored to focal taxa,\n",
      "\n",
      "factors affecting probe design and methods for optimizing probe\n",
      "\n",
      "sets to focal taxa remain largely underexplored (Branstetter et al.,\n",
      "\n",
      "2017; Glenn & Faircloth, 2016). The need for investigation of fac‐\n",
      "\n",
      "tors affecting probe design is made even greater by the increasing\n",
      "\n",
      "number of genomic resources becoming readily available for diverse\n",
      "\n",
      "organisms, and the development of new software enabling probe\n",
      "\n",
      "design.\n",
      "\n",
      "In this study, we investigate factors affecting probe set design\n",
      "\n",
      "of UCE loci. In particular, we investigate the effect of base genome\n",
      "\n",
      "choice and initial bait design parameters on the resulting probe set,\n",
      "\n",
      "and the number of UCE loci recovered during in silico tests. Currently,\n",
      "\n",
      "there are relatively few thorough workflows or pipelines and associ‐\n",
      "\n",
      "ated programs for identifying conserved loci and designing baits tar‐\n",
      "\n",
      "geting them (Campana, 2018; Faircloth, 2017; Johnson et al., 2016;\n",
      "\n",
      "Mayer et al., 2016). We opted to use one of the most widely utilized\n",
      "\n",
      "software packages for UCE phylogenomics, phyluce (Faircloth, 2016),\n",
      "\n",
      "and to follow the workflow outlined in detail by Faircloth (2017) for\n",
      "\n",
      "using phyluce to identify and design baits targeting UCE loci. Based\n",
      "\n",
      "on our results, we provide recommendations and discuss consider‐\n",
      "\n",
      "ations for base genome choice and parameter selections to optimize\n",
      "\n",
      "the design of UCE probe sets using phyluce . Our findings are then\n",
      "\n",
      "applied to develop a new UCE probe set, optimized for beetles of the\n",
      "\n",
      "suborder Adephaga, using newly available genomic resources and\n",
      "\n",
      "incorporating previously published Coleoptera UCE loci.\n",
      "\n",
      "**2** | **MATERIALS AND METHODS**\n",
      "\n",
      "**2.1** | **Genomic resources and taxon information**\n",
      "\n",
      "Genomic assemblies used in this study were generated using\n",
      "\n",
      "Illumina reads obtained from seven beetle species from the subor‐\n",
      "\n",
      "der Adephaga (Table 1) (Pflug, Holmes, Johnston, and Maddison, in\n",
      "\n",
      "prep). Full details are given in Pflug, Holmes, Johnston, and Maddison\n",
      "\n",
      "(in prep); in brief, poor quality reads and adapter sequences were\n",
      "\n",
      "trimmed from Illumina reads using the “Trim Sequences” tool in clc\n",
      "\n",
      "genomics workbench version 9.5.3 (CLC Bio, referred to below as CLC\n",
      "\n",
      "GW), with the trim quality score limit set to 0.5, allowing for a maxi‐\n",
      "\n",
      "mum of two ambiguities per read, and searching on both strands\n",
      "\n",
      "to remove adapters, retaining broken pairs. De novo assemblies of\n",
      "\n",
      "paired, trimmed reads in CLC GW were generated using automatic\n",
      "\n",
      "word and bubble size, with the minimum contig length set to 200\n",
      "\n",
      "bases. Genomic assembly metrics for the genomes used in this study\n",
      "\n",
      "are given in Table 1 (methods for obtaining these metrics are pro‐\n",
      "\n",
      "vided in Section 2.9).\n",
      "\n",
      "Our study species come from four different families within the\n",
      "\n",
      "beetle suborder Adephaga (Table 1). The suborder is diverse, con‐\n",
      "\n",
      "sisting of ~45,000 species (Ślipiński, Leschen, & Lawrence, 2011)\n",
      "\n",
      "exhibiting a wide range of ecologies with both terrestrial and aquatic\n",
      "\n",
      "members. These ecological differences have been used to divide\n",
      "\n",
      "the suborder into two groups: the Hydradephaga containing aquatic\n",
      "\n",
      "GUSTAFSON et al . **6935**\n",
      "**|**\n",
      "\n",
      "TA B LE 1 Study taxa and associated genomic assembly information. OSAC, Oregon State Arthropod Collection located at Oregon State\n",
      "\n",
      "University\n",
      "\n",
      "**Voucher #** **Estimated**\n",
      "\n",
      "**Species** **Depository** **Family** **read depth** **N50** **L50** **N90** **L90** **GC%** **BUSCO C%**\n",
      "\n",
      "\n",
      "_Bembidion_\n",
      "\n",
      "_haplogonum_\n",
      "\n",
      "Chaudoir\n",
      "\n",
      "\n",
      "DNA2544 Carabidae 58× 591 211,693 231 972,875 30.6 59.3\n",
      "\n",
      "OSAC\n",
      "\n",
      "\n",
      "_Chlaenius sericeus_ DNA4821 Carabidae 166× 1,765 41,747 408 190,237 29 78.7\n",
      "\n",
      "(Forster) OSAC\n",
      "\n",
      "_Lionepha_ DNA3782 Carabidae 8× 5,477 9,102 518 47,974 28.6 68.5\n",
      "\n",
      "“Waterfalls” OSAC\n",
      "\n",
      "_Pterostichus mela-_ DNA3787 Carabidae 33× 346 216,023 137 864,146 48.8 30.7\n",
      "\n",
      "_narius_ (Illiger) OSAC\n",
      "\n",
      "\n",
      "_Omoglymmius_\n",
      "\n",
      "_hamatus_\n",
      "\n",
      "(LeConte)\n",
      "\n",
      "\n",
      "DNA3783 Rhysodidae 13× 370 234,264 143 963,963 50 7.3\n",
      "\n",
      "OSAC\n",
      "\n",
      "\n",
      "_Trachypachus gibb-_ DNA3786 Trachypachidae 85× 1,758 34,686 213 233,955 32.8 45.8\n",
      "\n",
      "_sii_ LeConte OSAC\n",
      "\n",
      "_Amphizoa insolens_ DNA3784 Amphizoidae 8× 364 160,074 141 668,749 32.3 34.7\n",
      "\n",
      "LeConte OSAC\n",
      "\n",
      "\n",
      "members and the Geadephaga for terrestrial species. There has yet\n",
      "\n",
      "to be a comprehensive phylogenetic analysis of the suborder using\n",
      "\n",
      "molecular data, and recent studies with limited taxon sampling have\n",
      "\n",
      "either found these two groups reciprocally monophyletic (McKenna\n",
      "\n",
      "et al., 2015), or with Geadephaga sister to all aquatic families with the\n",
      "\n",
      "exception of the whirligig (the family Gyrinidae) beetles (Baca et al.,\n",
      "\n",
      "2017), or sister to both Gyrinidae and Haliplidae (Zhang et al., 2018).\n",
      "\n",
      "Our study taxa consist largely of families within the Geadephaga\n",
      "\n",
      "with a single hydradephagan representative, _Amphizoa_, belonging to\n",
      "\n",
      "one of the aquatic families consistently supported as sister to the\n",
      "\n",
      "Geadephaga (Baca et al., 2017; McKenna et al., 2015).\n",
      "\n",
      "The lack of a comprehensive time‐calibrated phylogeny of\n",
      "\n",
      "Adephaga prevents precise estimates of the evolutionary time and\n",
      "\n",
      "phylogenetic distance separating the species in our study. However,\n",
      "\n",
      "the maximum evolutionary time separating our study species is\n",
      "\n",
      "approximately 250 million years, which is similar in age to the esti‐\n",
      "\n",
      "mated split between turtles and archosaurs (Chiari, Cahais, Galtier,\n",
      "\n",
      "& Delsuc, 2012; Shen, Liang, Wen, & Zhang, 2011; Toussaint et al.,\n",
      "\n",
      "2017).\n",
      "\n",
      "**2.2** | **Overview**\n",
      "\n",
      "Using the above genomic resources and the software package phy \n",
      "luce, we developed multiple UCE probe sets while systematically al‐\n",
      "\n",
      "tering key aspects of the design strategy. Throughout this study, we\n",
      "\n",
      "use the term “bait” to refer to the temporary bait sets targeting theo‐\n",
      "\n",
      "retical conserved loci (an intermediate stage in probe design), with\n",
      "\n",
      "“probe” referring to the final products of probe design (i.e., the set of\n",
      "\n",
      "RNA probes targeting UCE loci that would actually be synthesized,\n",
      "\n",
      "subject to in silico testing). First, we explored the effects of chang‐\n",
      "\n",
      "ing which taxon served as the base genome during probe design, a\n",
      "\n",
      "critical decision, as the genomic reads of all other taxa are mapped to\n",
      "\n",
      "\n",
      "this genome in order to identify conserved loci. The second round of\n",
      "\n",
      "experiments focused on the effects of changing the stringency pa‐\n",
      "\n",
      "rameters used during temporary bait design in phyluce . Our general\n",
      "\n",
      "workflow for using phyluce to develop probe sets, and the specific\n",
      "\n",
      "methods for our experimentation are provided below. We evaluated\n",
      "\n",
      "the success of the different probe sets by tracking the number of\n",
      "\n",
      "loci identified and/or recovered during the different stages of probe\n",
      "\n",
      "design within phyluce, the number of probes developed, and finally\n",
      "\n",
      "by comparison of the UCE loci recovered during in silico tests. After\n",
      "\n",
      "producing optimized probes, we filtered the probe set to a subset\n",
      "\n",
      "of probes targeting only those loci found within all seven taxa that\n",
      "\n",
      "appear paralogy free. We then used baitstools as an additional check\n",
      "\n",
      "of our final subset of optimized probes in order to identify physical\n",
      "\n",
      "attributes that could affect their synthesis and in vitro utility. Finally,\n",
      "\n",
      "we investigated attributes of base genomes (e.g., assembly metrics\n",
      "\n",
      "and genetic distance from other study taxa) potentially correlated\n",
      "\n",
      "with improved probe design.\n",
      "\n",
      "**2.3** | **Phyluce workflow overview**\n",
      "\n",
      "A detailed workflow for using phyluce to identify UCE loci and to\n",
      "\n",
      "design probes targeting these loci is described by Faircloth (2017).\n",
      "\n",
      "The general phyluce work flow followed for our experiments is given\n",
      "\n",
      "in Figure 1. The first step in our experiments involved selection\n",
      "\n",
      "of a base genome (Figure 1, step 1) against which genomic reads\n",
      "\n",
      "of the remaining taxa were aligned (Figure 1, step 2). Following\n",
      "\n",
      "removal of duplicate reads, the next step (Figure 1, step 3) iden‐\n",
      "\n",
      "tified _putative loci_ : highly conserved loci shared among taxa (see\n",
      "\n",
      "Supporting Information for corresponding phyluce terminology and\n",
      "\n",
      "commands). _Putative loci_ are those present in the base genome as\n",
      "\n",
      "well as a specified number of additional taxa. The stringency at this\n",
      "\n",
      "step is varied (Figure 1b, step 3) by changing the number of taxa\n",
      "\n",
      "**6936** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "FI G U R E 1 General phyluce workflow for the two sets of probe design experiments conducted using phyluce . Workflow proceeds\n",
      "from left to right with the numbers above text blocks indicating steps referenced in text. (a) depicts the base genome choice experiments\n",
      "described in Section 2.4. (b) depicts the temporary bait design stringency experiments described in Section 2.5. A key for the figure is given\n",
      "in the bottom left corner. “Baits” refer to the temporary bait sets targeting theoretical conserved loci, with “probes” referring to the final\n",
      "products of probe design (i.e., the set of RNA probes targeting UCE loci that would actually be synthesized). Areas highlighted in red show\n",
      "targeted loci/baits that were found across the specified number of taxa. Results depicting the _putative loci_ (highly conserved loci shared\n",
      "among taxa), _candidate loci_ (loci where temporary baits align across genomes of multiple taxa), and the _recovered loci_ (UCE loci recovered\n",
      "during the in silico test) are shown in Figures 2‒6. The red boxes in the final probe set section indicate the combination of parameters\n",
      "leading to the largest number of in silico recovered probes for each set of experiments\n",
      "\n",
      "\n",
      "the locus must be found within, ranging from _putative loci_ found\n",
      "\n",
      "within the base genome and at least one other taxon (“+1”, the least\n",
      "\n",
      "stringent, designated as “at least one other taxon” on Figure 1b), to\n",
      "\n",
      "those found within the base genome and all remaining taxa (“+6”,\n",
      "\n",
      "the most stringent, designated as “all 6 taxa” on Figure 1b, that is,\n",
      "\n",
      "_putative loci_ are defined as those found in the base genome and all\n",
      "\n",
      "six additional taxa). Temporary baits were then developed target‐\n",
      "\n",
      "ing these _putative loci_ . These temporary baits were then aligned\n",
      "\n",
      "back to the genomic assemblies of all study taxa (Figure 1, step 4).\n",
      "\n",
      "This allowed the identification of _candidate loci_ : loci for which tem‐\n",
      "\n",
      "porary baits consistently aligned across the genomes of multiple\n",
      "\n",
      "taxa (Figure 1, step 5). By varying the number of taxa within which\n",
      "\n",
      "the candidate loci must be found within, an additional stringency\n",
      "\n",
      "filter can be applied. In the example shown in step 5 of Figure 1, the\n",
      "\n",
      "\n",
      "candidate loci chosen were those for which the temporary baits\n",
      "\n",
      "matched regions in all seven taxa (the base genome plus the other\n",
      "\n",
      "six taxa)—the most stringent option. The final probe set was then\n",
      "\n",
      "created by designing probes customized specifically for each taxon\n",
      "\n",
      "to allow enrichment of the same locus, across the specified num‐\n",
      "\n",
      "ber of taxa, using the previously identified _candidate loci_ . Finally,\n",
      "\n",
      "an in silico test (Figure 1, step 6) of the resulting probe set was\n",
      "\n",
      "performed using the genomic assemblies of all study taxa in order\n",
      "\n",
      "to assess the number of UCE loci recovered by the final probe set\n",
      "\n",
      "in semirealistic conditions. The purpose of using _recovered loci_ (loci\n",
      "\n",
      "recovered in the in silico test) to evaluate the choices involved in\n",
      "\n",
      "probe set design rather than the final probe sets themselves was in\n",
      "\n",
      "part to emulate some of the stochastic variation likely to be seen\n",
      "\n",
      "attempting to recover these loci in vitro.\n",
      "\n",
      "GUSTAFSON et al . **6937**\n",
      "**|**\n",
      "\n",
      "FI G U R E 2 Number of _putative loci_\n",
      "(highly conserved loci shared among taxa)\n",
      "recovered using different base genomes\n",
      "during probe design\n",
      "\n",
      "~~FI~~ ~~G~~ ~~U~~ ~~R~~ ~~E~~ ~~3~~ ~~Number~~ ~~of~~ ~~_candidate_~~ ~~_loci_~~ ~~(loci~~ ~~where~~ ~~temporary~~ ~~baits~~ ~~align~~ ~~across~~ ~~genomes~~ ~~of~~ ~~multiple~~ ~~taxa)~~ ~~shared~~ ~~among~~ ~~different~~ ~~numbers~~\n",
      "of taxa using different base genomes during probe design\n",
      "\n",
      "FI G U R E 4 Number of _recovered loci_\n",
      "\n",
      "(UCE loci recovered during the in silico\n",
      "test) shared among different taxa for\n",
      "probe sets designed using different base\n",
      "genomes. All categories provide the sum\n",
      "\n",
      "total of unique loci recovered for that\n",
      "\n",
      "base genome across all taxa\n",
      "\n",
      "**6938** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "We applied this workflow to two separate experiments: first,\n",
      "\n",
      "evaluating the use of different taxa as the base genome (Figure 1a,\n",
      "\n",
      "see Section 2.4 for additional methods), and second, varying differ‐\n",
      "\n",
      "ent parameters for temporary probe design (Figure 1b, see Section\n",
      "\n",
      "2.5 for additional methods).\n",
      "\n",
      "**2.4** | **Testing the effect of base genome choice on**\n",
      "**probe design and locus recovery**\n",
      "\n",
      "To test the influence of base genome choice on probe design\n",
      "\n",
      "and locus recovery, we performed the following steps in phyluce\n",
      "\n",
      "(Faircloth, 2016) (as outlined in the phyluce Tutorial IV, available at\n",
      "\n",
      "[http://phylu​ce.readt​hedocs.io/en/lates​t/tutor​ial-four.html). First, a](http://phyluce.readthedocs.io/en/latest/tutorial-four.html)\n",
      "\n",
      "single taxon was selected to serve as the base genome (Faircloth,\n",
      "\n",
      "2017) (Figure 1a, step 1) and the genomic reads of the remaining\n",
      "\n",
      "taxa were aligned to this genome using stampy (Lunter & Goodson,\n",
      "\n",
      "2011) (Figure 1a, step 2) under the following mapping parameters:\n",
      "\n",
      "‐‐maxbasequal 93, ‐‐substitutionrate = 0.05, and insertsize = 400.\n",
      "\n",
      "From the resulting alignment, we identified _putative loci_ shared\n",
      "\n",
      "among the base taxon and the remaining study taxa. In this base\n",
      "\n",
      "genome choice experiment, we then proceeded with the most strin‐\n",
      "\n",
      "gent approach to temporary bait design, targeting only those _puta-_\n",
      "\n",
      "_tive loci_ shared across the base genome as well as all six other taxa\n",
      "\n",
      "(Figure 1a, step 3). The influence of modifying the stringency of this\n",
      "\n",
      "step is evaluated by the experiment presented in the next section.\n",
      "\n",
      "We then determined the number of _candidate loci_ (where temporary\n",
      "\n",
      "baits consistently aligned across multiple genomes; Figure 1a, step\n",
      "\n",
      "4). Following this, we took a stringent approach to designing the final\n",
      "\n",
      "probe set by targeting only _candidate loci_ found within all seven of\n",
      "\n",
      "the study taxa (the base genome plus all six other taxa) (Figure 1a,\n",
      "\n",
      "step 5). After design of the final probe set, we performed an in silico\n",
      "\n",
      "test (Figure 1a, step 6) of the resulting probes using the genomic as‐\n",
      "\n",
      "semblies of all the study taxa to determine the number of _recovered_\n",
      "\n",
      "_loci_ by the probe set.\n",
      "\n",
      "Finally, the UCE loci extracted from each species during the in\n",
      "\n",
      "silico test were aligned using mafft (Katoh & Standley, 2013), am‐\n",
      "\n",
      "biguous regions removed via gblocks (Castresana, 2000; Talavera &\n",
      "\n",
      "Castresana, 2007), and a 75% complete matrix generated in phyluce\n",
      "\n",
      "using the command “phyluce_align_get_only_loci_with_min_taxa.”\n",
      "\n",
      "A maximum‐likelihood phylogenetic analysis was performed on this\n",
      "\n",
      "matrix using RA x ML 8.8.1 (Stamatakis, 2014) and the commands ‐m\n",
      "\n",
      "GTRGAMMA, ‐# autoMRE ‐n BOOT ‐T 24, to investigate any vari‐\n",
      "\n",
      "ance among the phylogenetic reconstructions obtained using loci\n",
      "\n",
      "recovered from probe sets designed with different base genomes.\n",
      "\n",
      "These steps were repeated changing only the taxon that served\n",
      "\n",
      "as the base genome (Figure 1a, step 1).\n",
      "\n",
      "Following comparison of probe sets designed using different\n",
      "\n",
      "base genomes and their success at locus recovery, we investigated\n",
      "\n",
      "attributes of base genomes potentially correlated with their probe\n",
      "\n",
      "design success. These attributes included genomic assembly metrics\n",
      "\n",
      "such as average estimated read depth, N50, L50, N90, L90, Guanine\n",
      "\n",
      "and Cytosine (GC) percent content, and BUSCO completeness per‐\n",
      "\n",
      "centage (Simão, Waterhouse, Ioannidis, Kriventseva, & Zdobnov,\n",
      "\n",
      "\n",
      "2015), as well as average relative genetic distance to the other taxa\n",
      "\n",
      "included in this study. These methods are presented in Sections 2.8\n",
      "\n",
      "and 2.9.\n",
      "\n",
      "**2.5** | **Testing temporary bait design stringency's**\n",
      "**influence on probe design and locus recovery**\n",
      "\n",
      "Following investigation into how base genome choice influences\n",
      "\n",
      "probe design and locus recovery, we experimented with relaxing\n",
      "\n",
      "temporary bait design stringency. This involved modifying the\n",
      "\n",
      "step in the phyluce workflow in which temporary baits are con‐\n",
      "\n",
      "structed based on the identification of _putative loci_ shared among\n",
      "\n",
      "the aligned genomic data (Figure 1b, step 3). We incrementally\n",
      "\n",
      "decreased the number of taxa _putative loci_ were required to be\n",
      "\n",
      "identified within when designing temporary baits (Figure 1b, step\n",
      "\n",
      "3). We began first with the most stringent requirement for tem‐\n",
      "\n",
      "porary bait design by targeting _putative loci_ found across the base\n",
      "\n",
      "genome and all six other taxa and ended by targeting those found\n",
      "\n",
      "across the base genome and at least one other taxon, the least\n",
      "\n",
      "stringent temporary bait design. We then proceeded to follow the\n",
      "\n",
      "same steps as in Section 2.4 through to the in silico test. It should\n",
      "\n",
      "be emphasized that we maintained the stringent requirement of\n",
      "\n",
      "designing the final probe set based only on _candidate loci_ present\n",
      "\n",
      "within all seven study taxa (Figure 1b, step 5). We selected the\n",
      "\n",
      "taxon _Lionepha_ to serve as the base genome throughout these\n",
      "\n",
      "tests as it performed moderately well during the base genome\n",
      "\n",
      "choice experiments, being neither the best, nor worst perform‐\n",
      "\n",
      "ing genome. After establishing that _Pterostichus_ appeared to be\n",
      "\n",
      "the optimum base genome, we also repeated the temporary bait\n",
      "\n",
      "stringency experiment on this base genome, to ensure the results\n",
      "\n",
      "were robust to base genome choice.\n",
      "\n",
      "**2.6** | **Comparison of resulting probe sets**\n",
      "\n",
      "To establish which UCE loci were shared between the probe sets de‐\n",
      "\n",
      "signed using different base genomes in Section 2.4, we used custom\n",
      "\n",
      "bash and R‐scripts to BLAST (at 95% and 99% similarity) between\n",
      "\n",
      "the monolithic files (containing all UCE loci identified across all taxa\n",
      "\n",
      "for each base genome) produced by phyluce (see Alexander 2018a\n",
      "\n",
      "under Data Accessibility for step‐by‐step commands). Loci with sin‐\n",
      "\n",
      "gle BLAST matches across at least two base genomes for at least\n",
      "\n",
      "one taxon were characterized as: “good” (UCE loci that appeared to\n",
      "\n",
      "be single copy across all taxa across all base genomes), “problematic\n",
      "\n",
      "within taxa” (evidence of potential paralogy within one or more taxa\n",
      "\n",
      "where loci identified within a taxon using one base genome matched\n",
      "\n",
      "to multiple loci within that same taxon identified using a different\n",
      "\n",
      "base genome), and “problematic between taxa” (potential lineage‐\n",
      "\n",
      "specific paralogy: a UCE locus that appeared to be single copy across\n",
      "\n",
      "all base genomes within one taxon appeared to be present in mul‐\n",
      "\n",
      "tiple copies in at least one other taxon). We used the “good loci” to\n",
      "\n",
      "identify the base genomes that had the greatest success recovering\n",
      "\n",
      "loci across the seven taxa and that gave the longest alignment in\n",
      "\n",
      "each taxon.\n",
      "\n",
      "GUSTAFSON et al . **6939**\n",
      "**|**\n",
      "\n",
      "FI G U R E 5 Number of _candidate_\n",
      "\n",
      "_loci_ (loci where temporary baits align\n",
      "across genomes of multiple taxa) found\n",
      "across increasing numbers of taxa for\n",
      "each temporary bait design stringency\n",
      "(requiring loci to be found in the base\n",
      "genome +1 other taxon through to the\n",
      "base genome +6 taxa)\n",
      "\n",
      "FI G U R E 6 Number of _recovered_\n",
      "\n",
      "_loci_ (UCE loci recovered during the in\n",
      "silico test) in different taxa from probe\n",
      "\n",
      "sets designed using _Lionepha_ and\n",
      "_Pterostichus_ as base genomes and varying\n",
      "the temporary bait design stringency\n",
      "requirements (requiring loci to be found in\n",
      "the base genome +1 other taxon through\n",
      "to the base genome +6 taxa). The all\n",
      "category provides the sum total of unique\n",
      "loci recovered for that base genome\n",
      "\n",
      "across all taxa\n",
      "\n",
      "\n",
      "**2.7** | **In vitro attributes of final subset of**\n",
      "**optimized probes**\n",
      "\n",
      "We examined aspects of the final subset of optimized probes de‐\n",
      "\n",
      "signed during this study that could potentially affect synthesis and\n",
      "\n",
      "in vitro utility. We gathered relevant information on the final opti‐\n",
      "\n",
      "mized subset of probes designed during this study relevant to in vitro\n",
      "\n",
      "performance including GC% content, melting temperature (Tm), and\n",
      "\n",
      "sequence complexity, using baitstools (Campana, 2018) and the\n",
      "\n",
      "“checkbaits” command under the following parameters: ‐L 120 ‐c ‐n\n",
      "\n",
      "25 ‐x 70 ‐q 80 ‐z 120 ‐w.\n",
      "\n",
      "**2.8** | **Genome assembly metrics**\n",
      "\n",
      "We estimated depth of coverage for each genome by mapping reads\n",
      "\n",
      "to the subset of “good” UCE loci (identified using the methods in\n",
      "\n",
      "Section 2.6) that were also found in all seven taxa. The specific ref‐\n",
      "\n",
      "erence sequences used for each locus depended on how many base\n",
      "\n",
      "genomes were able to characterize that UCE locus. If a UCE locus was\n",
      "\n",
      "detected only with one base genome, then the UCE locus sequences\n",
      "\n",
      "\n",
      "from this base genome were used as references. If a given locus was\n",
      "\n",
      "recovered using more than one base, then the base genome that gave\n",
      "\n",
      "the longest average alignment across all taxa was selected to provide\n",
      "\n",
      "the reference sequences. We extracted the UCE locus sequences for\n",
      "\n",
      "each taxon using custom bash scripts (see Alexander 2018a under\n",
      "\n",
      "Data Accessibility for step‐by‐step commands). Then, we used GATK v\n",
      "\n",
      "3.8/4.0, picard, bwa, MAFFT, R, samtools, stringr with custom scripts\n",
      "\n",
      "(see Alexander 2018d under Data Accessibility for step‐by‐step com‐\n",
      "\n",
      "mands) to calculate depth of coverage on a per‐locus basis.\n",
      "\n",
      "Additional genomic assembly metrics such as N50, L50, N90,\n",
      "\n",
      "L90, Guanine and Cytosine (GC) percent content, and completeness\n",
      "\n",
      "of conserved protein‐coding genes were generated using CLC GW\n",
      "\n",
      "v9.5.3 (available from https​://www.qiage​nbioi​nform​atics.com/),\n",
      "\n",
      "BBM ap (Bushnell, 2014) using the stats.sh tool, and BUSCO (Simão\n",
      "\n",
      "et al., 2015).\n",
      "\n",
      "**2.9** | **Estimating relatedness of study taxa**\n",
      "\n",
      "We inferred the relative divergence between the taxa in our study\n",
      "\n",
      "for several markers commonly used in phylogenetic investigations\n",
      "\n",
      "**6940** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "of Coleoptera: 18S, 28S, ArgK, CAD2, CAD4, _wg_, and COI. These\n",
      "\n",
      "markers were obtained either for species included in this study, or\n",
      "\n",
      "for their congeners. Some of these sequences were obtained from\n",
      "\n",
      "the base genomes (CAD2 for _Trachypachus_ and _Lionepha_, and COI of\n",
      "\n",
      "all taxa except _Bembidion haplogonum_ ). The rest were obtained from\n",
      "\n",
      "the literature (Gomez, Will, & Maddison, 2016; Kanda, Pflug, Sproul,\n",
      "\n",
      "Dasenko, & Maddison, 2015; Maddison, 2012; Maddison Baker, &\n",
      "\n",
      "Ober, 1999a, 1999b; Maddison et al., 2009; Maddison & Swanson,\n",
      "\n",
      "2010; McKenna et al., 2015; Ober, 2002; Sproul & Maddison, 2017),\n",
      "\n",
      "or were newly obtained from Sanger sequencing of PCR products, as\n",
      "\n",
      "specified in Supplemental Information Table S1. Methods for obtain‐\n",
      "\n",
      "ing the new sequences from PCR/Sanger sequencing are given in\n",
      "\n",
      "Maddison (2012). New sequences have been submitted to GenBank\n",
      "\n",
      "[with accession numbers MK838494 to MK838511.](info:ddbj-embl-genbank/MK838494)\n",
      "\n",
      "We used Geneious [®] 11.0.5 to calculate average pairwise genetic\n",
      "\n",
      "distances, average proportion of differences (using both Tamura–\n",
      "\n",
      "Nei and HKY corrections), and patristic distances from a neighbor‐\n",
      "\n",
      "joining tree using the Tamura–Nei correction. We ranked taxa from\n",
      "\n",
      "the smallest average genetic distance across markers and methods\n",
      "\n",
      "to the largest average distance.\n",
      "\n",
      "Additionally, we attempted to extract 50 nuclear protein‐\n",
      "\n",
      "coding markers developed for Coleoptera phylogenetics from\n",
      "\n",
      "genomewide surveys (Che et al., 2017) directly from our study\n",
      "\n",
      "genomes using bait sequences from related taxa available on\n",
      "\n",
      "GenBank via the Zhang et al. (2018) study (Appendix S1). We ex‐\n",
      "\n",
      "tracted these nuclear protein‐coding markers from our genomic\n",
      "\n",
      "assemblies using bwa (Li & Durbin, 2009, 2010), data.table (Dowle\n",
      "\n",
      "& Srinivasan, 2017), GATK (McKenna et al., 2010), MAFFT (Katoh\n",
      "\n",
      "[& Standley, 2013), picard (available from: http://broad​insti​tute.](http://broadinstitute.github.io/picard/)\n",
      "\n",
      "[github.io/picar​d/), R (R Core Team, 2017), samtools (Li et al., 2009),](http://broadinstitute.github.io/picard/)\n",
      "\n",
      "and stringr (Wickham, 2017) via custom scripts (see Alexander\n",
      "\n",
      "2018b under Data Availability for step‐by‐step commands). This\n",
      "\n",
      "was done in an attempt to better approximate divergences of\n",
      "\n",
      "the entire nuclear genomes among our taxa. We then generated\n",
      "\n",
      "relative genetic distance measures among these nuclear protein‐\n",
      "\n",
      "coding markers for our study taxa using EMBOSS distmat (Rice,\n",
      "\n",
      "Longden, & Bleasby, 2000) with a Tamura correction and R via\n",
      "\n",
      "custom scripts (see Alexander 2018c under Data Availability for\n",
      "\n",
      "step‐by‐step commands). Finally, we compared these findings with\n",
      "\n",
      "those of the more commonly used phylogenetic loci represented\n",
      "\n",
      "in our six gene dataset, run through the same distance measures\n",
      "\n",
      "pipeline.\n",
      "\n",
      "**3** | **RESULTS**\n",
      "\n",
      "**3.1** | **The effect of base genome choice on probe**\n",
      "**design and locus recovery**\n",
      "\n",
      "The effect of base genome choice was evident at each stage of\n",
      "\n",
      "the workflow (Figure 1a) from the number of _putative loci_ identi‐\n",
      "\n",
      "fied (Figure 2, Data S1), to the number of _candidate loci_ (Figure 3,\n",
      "\n",
      "Data S2), and finally the number of _recovered loci_ during the in silico\n",
      "\n",
      "test (Figure 4, Data S3). There were also consistent trends in the\n",
      "\n",
      "\n",
      "performance of taxa selected to serve as the base genome. For ex‐\n",
      "\n",
      "ample, the genomes of _Amphizoa_ and _Omoglymmius_ resulted in the\n",
      "\n",
      "fewest number of _putative loci_ (Figure 2), _candidate loci_ (Figure 3),\n",
      "\n",
      "and _recovered loci_ (Figure 4). These base genomes also resulted in\n",
      "\n",
      "the fewest number of probes designed, as well as the shortest over‐\n",
      "\n",
      "all sequence lengths recovered across taxa (see Data S2 and S3).\n",
      "\n",
      "The taxa _Bembidion_, _Lionepha_, and _Trachypachus_ performed\n",
      "\n",
      "considerably better as base genomes, resulting in more _puta-_\n",
      "\n",
      "_tive loci_ (Figure 2), _candidate loci_ (Figure 3), and _recovered loci_\n",
      "\n",
      "(Figure 4), as well as the design of tens of thousands more probes\n",
      "\n",
      "(Data S2). _Chlaenius_ performed even better as the base genome\n",
      "\n",
      "and was the base genome that resulted in the largest number\n",
      "\n",
      "of _putative loci_ shared among all six of the other taxa (Figure 2).\n",
      "\n",
      "However, _Chlaenius_ did not result in the most _candidate loci_ found\n",
      "\n",
      "across all seven taxa (Figure 3) nor _recovered loci_ during the in\n",
      "\n",
      "silico test (including _Chlaenius_ itself; Figure 4). This suggests the\n",
      "\n",
      "number of _putative loci_, while tracking general trends in a taxon's\n",
      "\n",
      "performance as the base genome, is not an accurate predictor of\n",
      "\n",
      "the best performing base genome for downstream locus detection\n",
      "\n",
      "and recovery.\n",
      "\n",
      "The best performing base genome was _Pterostichus_ . Using\n",
      "\n",
      "_Pterostichus_ as the base genome resulted in the largest number of\n",
      "\n",
      "_candidate loci_ (Figure 3) found across all seven taxa, and the larg‐\n",
      "\n",
      "est number of _recovered loci_ (including the most _recovered loci_ for\n",
      "\n",
      "_Pterostichus_ itself; Figure 4). The use of _Pterostichus_ as a base ge‐\n",
      "\n",
      "nome resulted in baits targeting several hundred (i.e., 8%) more loci\n",
      "\n",
      "than the second‐best performing genome, _Chlaenius_, concordantly\n",
      "\n",
      "reflected in the design of several thousand (i.e., ~7%) more probes\n",
      "\n",
      "(Data S2). _Pterostichus_ resulted in several hundred (i.e., 6%) more _re-_\n",
      "\n",
      "_covered loci_ during the in silico test (Figure 4) than _Chlaenius_, and\n",
      "\n",
      "the resulting concatenated final alignment across taxa was several\n",
      "\n",
      "hundred thousand base pairs (i.e., ~6%) longer than that of _Chlaenius_\n",
      "\n",
      "(Data S3). Despite these differences in performance at recovering\n",
      "\n",
      "and detecting UCE loci, base genome choice did not affect the topol‐\n",
      "\n",
      "ogy, nor support values of the final maximum‐likelihood phylogenies\n",
      "\n",
      "produced (Figure S1).\n",
      "\n",
      "**3.2** | **Stringency of temporary bait design's effect on**\n",
      "**probe design and locus recovery**\n",
      "\n",
      "The temporary bait design stringency experiments revealed less\n",
      "\n",
      "stringent parameters for bait design (i.e., requiring sharing of _puta-_\n",
      "\n",
      "_tive loci_ across fewer taxa) greatly improved the number of _candidate_\n",
      "\n",
      "_loci_ identified (Figure 5, Data S4), as well as the number of _recovered_\n",
      "\n",
      "_loci_ during the in silico test (Figure 6, Data S5).\n",
      "\n",
      "**3.3** | **Optimized probe design**\n",
      "\n",
      "The base genome choice experiments identified _Pterostichus_ as the\n",
      "\n",
      "optimal base genome. The results from the stringency in temporary\n",
      "\n",
      "bait design experiments showed decreasing the number of taxa\n",
      "\n",
      "_putative loci_ were required to be shared across increased both the\n",
      "\n",
      "number of final probes and the number of _recovered loci_ during the\n",
      "\n",
      "GUSTAFSON et al . **6941**\n",
      "**|**\n",
      "\n",
      "FI G U R E 7 Number of “good loci”\n",
      "(UCE loci that appeared to be single copy\n",
      "across all taxa across all base genomes)\n",
      "\n",
      "recovered in different taxa from probe\n",
      "\n",
      "sets designed using different base\n",
      "\n",
      "genomes\n",
      "\n",
      "\n",
      "in silico test (Figure 6). Therefore, to design an initial optimized UCE\n",
      "\n",
      "probe set, we selected _Pterostichus_ to serve as the base genome and\n",
      "\n",
      "designed temporary probes under the least stringent requirements,\n",
      "\n",
      "targeting _putative loci_ found within the base genome and at least one\n",
      "\n",
      "other taxon.\n",
      "\n",
      "**3.4** | **Comparison of probe sets based on loci found**\n",
      "**across different probe designs**\n",
      "\n",
      "Using the _recovered loci_ (i.e., those identified during the in silico\n",
      "\n",
      "test) from the seven base genome choice experiments, as well as\n",
      "\n",
      "the final optimized _Pterostichus_ +1 probe design, we identified\n",
      "\n",
      "8,159 loci through BLAST that matched at 99% similarity across\n",
      "\n",
      "more than one base genome for at least one taxon (95% similar‐\n",
      "\n",
      "ity results revealed very similar patterns, so these results are pre‐\n",
      "\n",
      "sented only in Data S6 and S7). Of these, 3,501 “good loci” showed\n",
      "\n",
      "no sign of paralogy within or among taxa across the various base\n",
      "\n",
      "genomes used. Comparisons of base genome performance based\n",
      "\n",
      "on recovery of these “good loci” largely reflected the previously\n",
      "\n",
      "presented results. When _Pterostichus_ served as the base genome\n",
      "\n",
      "with temporary baits designed from _putative loci_ found in just one\n",
      "\n",
      "other taxon ( _Pterostichus_ +1 taxon), the largest number of “good\n",
      "\n",
      "loci” were recovered for each taxon in comparison with all other\n",
      "\n",
      "base genomes (Figure 7, Data S8). This pattern persisted even\n",
      "\n",
      "when restricting the comparison to the 2,702 “good loci” detected\n",
      "\n",
      "in all seven taxa and thus does not seem to be an artifact of over‐\n",
      "\n",
      "representation of loci found only in _Pterostichus_ but absent in other\n",
      "\n",
      "lineages (Data S9). We also compared the impact of base genome\n",
      "\n",
      "selection on sequence length of the “good loci” found across all\n",
      "\n",
      "seven study taxa. This could be an important metric, as phyloge‐\n",
      "\n",
      "netically informative variation in UCEs tends to be greater in the\n",
      "\n",
      "DNA sequences flanking the core conserved element (Faircloth\n",
      "\n",
      "et al., 2012; Van Dam et al., 2017). Longer sequence lengths are\n",
      "\n",
      "therefore more likely to recover these more variable regions. For\n",
      "\n",
      "each “good locus” found in all seven taxa, we recorded the number\n",
      "\n",
      "\n",
      "of taxa where a base genome gave the longest or equally long‐\n",
      "\n",
      "est sequence lengths across all base genomes. These results\n",
      "\n",
      "again showed that _Pterostichus_ (specifically _Pterostichus_ +1 taxon)\n",
      "\n",
      "appeared to be the optimal base genome/stringency choice: It\n",
      "\n",
      "resulted in a larger number of loci where it gave the longest align‐\n",
      "\n",
      "ment for more taxa than the other base genomes (Data S9).\n",
      "\n",
      "**3.5** | **Final optimized probe set**\n",
      "\n",
      "The optimized probe design ( _Pterostichus_ +1 taxon) resulted in a\n",
      "\n",
      "probe set consisting of 145,599 probes, targeting 11,162 UCE loci.\n",
      "\n",
      "However, synthesizing and enriching this large number of probes\n",
      "\n",
      "and loci would be cost prohibitive for phylogenomic studies with\n",
      "\n",
      "large numbers of taxa. We therefore needed a way to naturally limit\n",
      "\n",
      "the number of probes included in the final probe set. The initial\n",
      "\n",
      "optimized probe set was filtered to a final set of probes targeting\n",
      "\n",
      "the 2,702 “good loci” detected in all seven taxa discussed in Section\n",
      "\n",
      "3.4. This was done using custom scripts (see Alexander 2018d\n",
      "\n",
      "under Data Accessibility for step‐by‐step commands) and was fur‐\n",
      "\n",
      "ther limited to only those loci covered by the _Ptersotichus_ +1 probes\n",
      "\n",
      "(i.e., the optimized probe design), as 858 loci were derived under\n",
      "\n",
      "other probe design parameters (i.e., different base genomes and\n",
      "\n",
      "temporary bait stringency). Therefore, the final subset of optimized\n",
      "\n",
      "probes targets 2,643 loci that appear paralogy free and are present\n",
      "\n",
      "within all seven study taxa. As part of the final probe set, we also\n",
      "\n",
      "included probes for 305 loci from the previous Coleoptera 1.1Kv1\n",
      "\n",
      "probe set (Faircloth, 2017) that were successfully recovered during\n",
      "\n",
      "the Baca et al. (2017) study (see Section 4 section below). The final\n",
      "\n",
      "optimized probe set targets a total of 2,948 UCE loci.\n",
      "\n",
      "In vitro attributes of the final subset of optimized probes\n",
      "\n",
      "(Appendix S2) include a GC% content of all probes between 20%\n",
      "\n",
      "and 70%, with a median of 40.83% and mean of 42%; and a Tm\n",
      "\n",
      "between 80°C and 120°C, with a median of 96.83°C and a mean\n",
      "\n",
      "of 97.69°C. The sequence complexity measure of the probes was\n",
      "\n",
      "very high with a bait linguistic complexity mean and median of\n",
      "\n",
      "**6942** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "FI G U R E 8 Attributes of study taxa\n",
      "\n",
      "and their genomic assemblies in relation\n",
      "to their performance as base genome. (a)\n",
      "Relationship between average pairwise\n",
      "genetic distance based on six gene\n",
      "fragments regularly use for phylogenetics\n",
      "\n",
      "and total number of UCE loci recovered\n",
      "\n",
      "during in silico test. (b) Relationship\n",
      "between contiguity measure N90 of\n",
      "genomic assembly and total number of\n",
      "UCE loci recovered during in silico test. (c)\n",
      "\n",
      "Relationship between BUSCO %C value\n",
      "\n",
      "of genomic assembly and total number of\n",
      "UCE loci recovered during in silico test\n",
      "\n",
      "\n",
      "0.99. The GC% content is well within the optimal range for ge‐\n",
      "\n",
      "nomic probes (Cruz‐Dávalos et al., 2017; Tewhey et al., 2009)\n",
      "\n",
      "and high‐sequence complexity and Tms increase the likelihood\n",
      "\n",
      "of target specificity.\n",
      "\n",
      "\n",
      "**3.6** | **Assembly metrics of genomic resources used**\n",
      "\n",
      "We used the 2,702 “good” UCE loci found across all seven taxa\n",
      "\n",
      "to estimate coverage depth for the genomes utilized in this study.\n",
      "\n",
      "GUSTAFSON et al . **6943**\n",
      "**|**\n",
      "\n",
      "S10), with BUSCO scores (Figure 8c, Table 1, Data S10) supporting\n",
      "\n",
      "the _Chlaenius_ genome as being the most complete. Interestingly,\n",
      "\n",
      "_Pterostichus_ in addition to _Omoglymmius_ and _Amphizoa_ had the\n",
      "\n",
      "least complete genomic assemblies and lowest assembly metrics\n",
      "\n",
      "(Figure 8c, Table 1), yet these taxa represent both the best and worst\n",
      "\n",
      "performing base genomes, respectively. Therefore, genomic assem‐\n",
      "\n",
      "bly metrics do not appear to correlate well with base genome perfor‐\n",
      "\n",
      "mance (Figure 8b,c). Similarly, GC% content of the genome does not\n",
      "\n",
      "appear correlated with suitability as a base genome, in that the best\n",
      "\n",
      "performing base genome, _Pterostichus_, had a comparable GC per‐\n",
      "\n",
      "centage (nearly 50%, Table 1) to that of one of the worst performing\n",
      "\n",
      "genomes, _Omoglymmius_, while _Amphizoa_ (another poorly performing\n",
      "\n",
      "genome) had a GC percentage similar to those of better performing\n",
      "\n",
      "genomes (around 30%, Table 1).\n",
      "\n",
      "**3.7** | **Pairwise genetic distances between taxa**\n",
      "\n",
      "Pairwise genetic distance rankings showed _Pterostichus_ as having\n",
      "\n",
      "the smallest genetic distance from all other taxa in the six gene frag‐\n",
      "\n",
      "ments commonly used in phylogenetic studies (28S, COI, CAD2,\n",
      "\n",
      "CAD4, _wg_, ArgK), with _Amphizoa_ and _Omoglymmius_ the largest ge‐\n",
      "\n",
      "netic distances (Figure 8a, Data S11–S13 for average‐, raw‐, and\n",
      "\n",
      "standardized pairwise genetic distance rankings).\n",
      "\n",
      "We found average genetic distance ranking based on the six\n",
      "\n",
      "gene dataset to be negatively correlated with a taxon's performance\n",
      "\n",
      "as the base genome as assessed by the total number of UCE loci\n",
      "\n",
      "recovered during in silico testing (Figure 8a), and reflected general\n",
      "\n",
      "trends in genome performance tracked elsewhere (Figures 3‒5).\n",
      "\n",
      "We attempted to validate the six gene genetic distance measure\n",
      "\n",
      "\n",
      "FI G U R E 9 Proposed workflow for designing an optimized\n",
      "universal UCE probe set using P hyluce for taxa without a well‐\n",
      "annotated genome available\n",
      "\n",
      "Genomic read coverage did not appear to correlate with base ge‐\n",
      "\n",
      "nome performance (Table 1). _Pterostichus_, the best performing base\n",
      "\n",
      "genome, showed intermediate coverage depth (33×) relative to other\n",
      "\n",
      "taxa (Table 1). The worst performing base genomes, _Omoglymmius_\n",
      "\n",
      "and _Amphizoa_, while estimated as having relatively low cover‐\n",
      "\n",
      "age depths (13× and 8× respectively), were comparable to that of\n",
      "\n",
      "_Lionepha_ (8×) which performed considerably better as a base ge‐\n",
      "\n",
      "nome (Figure 4). Finally, _Chlaenius_ had the highest estimated cover‐\n",
      "\n",
      "age depth (166×), yet did not outperform _Pterostichus_ which had a\n",
      "\n",
      "much lower coverage depth (33×).\n",
      "\n",
      "Additional genomic assembly metrics such as N50, L50, N90,\n",
      "\n",
      "L90 identified _Chlaenius_, _Lionepha_, and _Trachypachus_ as the best\n",
      "\n",
      "genomic assemblies in terms of contiguity (Figure 8b, Table 1, Data\n",
      "\n",
      "\n",
      "results with a larger data set by extracting 50 different nuclear pro‐\n",
      "\n",
      "tein‐coding loci directly from genomic assemblies of our study taxa\n",
      "\n",
      "(Appendix S1 and Section 2.9). However, this approach proved inef‐\n",
      "\n",
      "fective due to unevenness in genome assembly quality of our study\n",
      "\n",
      "taxa (Table 1) which resulted in an excess of missing or low‐quality\n",
      "\n",
      "data for taxa with assemblies having a low average depth of cov‐\n",
      "\n",
      "erage (e.g., _Lionepha_, _Omoglymmius_, and _Amphizoa_ ). We determined\n",
      "\n",
      "that low coverage and missing data affected data quality for these\n",
      "\n",
      "taxa based on an evident correlation between low sequencing cov‐\n",
      "\n",
      "erage and large genetic distance (Data S14–S19), as well as a maxi‐\n",
      "\n",
      "mum‐likelihood phylogenetic analysis of the data set using IQTree\n",
      "\n",
      "(Nguyen, Schmidt, Haeseler, & Minh, 2015), which failed to recover\n",
      "\n",
      "well‐established sister relationships as explained in more detail in\n",
      "\n",
      "Data S20.\n",
      "\n",
      "**4** | **DISCUSSION**\n",
      "\n",
      "Our study using adephagan beetles as an example demonstrates\n",
      "\n",
      "that altering the base genome used during probe design can greatly\n",
      "\n",
      "alter the number of resultant probes included in the final probe\n",
      "\n",
      "set, and more importantly, strongly affects the number of loci\n",
      "\n",
      "detected and recovered during in silico tests. The base genome\n",
      "\n",
      "that recovered the largest number of in silico loci, _Pterostichus_,\n",
      "\n",
      "**6944** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "also performed better at recovering the longest sequences across\n",
      "\n",
      "the seven taxa included in this study. As variable regions flank‐\n",
      "\n",
      "ing core UCEs contain phylogenetically informative sequence data\n",
      "\n",
      "(Faircloth et al., 2012; Van Dam et al., 2017), the ability to recover\n",
      "\n",
      "longer sequences for each locus is likely particularly important for\n",
      "\n",
      "their use in phylogenetic analysis. In addition, we demonstrated\n",
      "\n",
      "that being less stringent during the temporary bait design stage,\n",
      "\n",
      "by focusing on _putative loci_ shared across the base genome with\n",
      "\n",
      "as few as one other taxon, resulted in a significantly better final\n",
      "\n",
      "probe set design that detected and recovered more UCE loci dur‐\n",
      "\n",
      "ing in silico testing.\n",
      "\n",
      "**4.1** | **Considerations for selecting a base genome for**\n",
      "**probe design**\n",
      "\n",
      "Although we expected at the outset that higher quality base ge‐\n",
      "\n",
      "nomes would result in better final probe sets, genomic assembly\n",
      "\n",
      "metrics such as measures of completeness, average read depth, con‐\n",
      "\n",
      "tiguity, and GC content did not appear to correlate well with base ge‐\n",
      "\n",
      "nome performance (Figure 8, Table 1). The best performing genome\n",
      "\n",
      "for universal adephagan bait design, _Pterostichus_, had relatively low\n",
      "\n",
      "genomic assembly metrics compared to most other taxa in the study\n",
      "\n",
      "(Table 1). This pattern suggests that genome assembly quality may\n",
      "\n",
      "not be a critical criterion in selecting a base genome. In reporting this\n",
      "\n",
      "finding, we acknowledge that it is possible all our assemblies were\n",
      "\n",
      "above a critical coverage threshold, below which assembly metrics\n",
      "\n",
      "may begin to show stronger correlation with probe design outcomes.\n",
      "\n",
      "Similarly, all our assemblies are limited in quality due to being gener‐\n",
      "\n",
      "ated using short‐read Illumina data, and it is possible that assemblies\n",
      "\n",
      "of much higher quality could correlate with greater probe design\n",
      "\n",
      "success. Our results are a useful reference point in that they were\n",
      "\n",
      "generated using a type (e.g., Illumina reads) and quantity (e.g., tens to\n",
      "\n",
      "hundreds of millions of reads) of sequence data that are representa‐\n",
      "\n",
      "tive of increasing numbers of unpublished genomes, but the field will\n",
      "\n",
      "benefit from further exploring how assembly quality affects probe\n",
      "\n",
      "design outcomes.\n",
      "\n",
      "When introducing the workflow for using phyluce to identify\n",
      "\n",
      "and design baits targeting UCE loci, Faircloth (2017) recommends\n",
      "\n",
      "that selection of an appropriate base genome not be focused on\n",
      "\n",
      "ingroup versus outgroup status, but rather annotation and genomic\n",
      "\n",
      "assembly, in order to aid later in selection of probes targeting loci\n",
      "\n",
      "with desirable properties for phylogenetic inference. This is cer‐\n",
      "\n",
      "tainly a good recommendation for groups that have well‐anno‐\n",
      "\n",
      "tated genomic assemblies available; however, most invertebrate\n",
      "\n",
      "groups lack such genomic resources, including our study taxa.\n",
      "\n",
      "Furthermore, one of the most powerful aspects of phyluce is its\n",
      "\n",
      "ability to identify and design probe sets for such nonmodel organ‐\n",
      "\n",
      "isms, without the necessity of a complete and annotated genome.\n",
      "\n",
      "Our results demonstrated that UCE probe design can be success‐\n",
      "\n",
      "fully optimized even while using a base genome whose genomic\n",
      "\n",
      "assembly is of lower coverage and limited quality as phyluce targets\n",
      "\n",
      "conserved regions that are present and shared among the study\n",
      "\n",
      "taxa, rather than searching for a priori known regions to target.\n",
      "\n",
      "\n",
      "We provide a framework for conducting base genome experiments\n",
      "\n",
      "as outlined above (Figure 1) for selecting a base genome in cases\n",
      "\n",
      "where maximizing the number of loci in the probe set is critical. As a\n",
      "\n",
      "less involved alternative to conducting base genome experiments,\n",
      "\n",
      "we provide evidence from a six gene data set (generated through\n",
      "\n",
      "Sanger sequencing) that selecting the taxon with the smallest av‐\n",
      "\n",
      "erage genetic distance from the suite of taxa included in probe de‐\n",
      "\n",
      "sign to serve as the base genome can correlate with optimal probe\n",
      "\n",
      "design outcomes (Figure 8a). However, our conclusion that genetic\n",
      "\n",
      "distance patterns can guide base genome selection would bene‐\n",
      "\n",
      "fit from further testing given that low sequencing coverage from\n",
      "\n",
      "a subset of our assemblies prevented our validation of this finding\n",
      "\n",
      "with a larger data set (Data S14–S20).\n",
      "\n",
      "**4.2** | **Why does a “relaxed” approach at the**\n",
      "**temporary bait design stage result in a better**\n",
      "**probe set?**\n",
      "\n",
      "Perhaps most surprising among the results of our experiments re‐\n",
      "\n",
      "garding optimal probe set design are those pertaining to temporary\n",
      "\n",
      "bait design stringency. We anticipated a diminishing return in the\n",
      "\n",
      "number of _candidate loci_ as the number of taxa _putative loci_ were\n",
      "\n",
      "required to share decreased. This is because we expected _putative_\n",
      "\n",
      "_loci_ initially identified as being shared among the base and only one,\n",
      "\n",
      "or a few other taxa, would be unlikely to also be present or detect‐\n",
      "\n",
      "able in all of the remaining taxa, thus preventing their inclusion as\n",
      "\n",
      "_candidate loci_ (the step where we required the baits to target loci\n",
      "\n",
      "found in all seven taxa). However, we did not find this to be the case.\n",
      "\n",
      "Instead, we found that designing temporary baits based on _putative_\n",
      "\n",
      "_loci_ shared among the base genome and as few as one other taxon\n",
      "\n",
      "resulted in significantly better probe design, detecting and recov‐\n",
      "\n",
      "ering more UCE loci, as well as in a better performing probe set.\n",
      "\n",
      "Why this would be the case is not entirely clear. Faircloth, in his in‐\n",
      "\n",
      "formal discussion within the phyluce [Tutorial IV (available at: http://](http://phyluce.readthedocs.io/en/latest/tutorial-four.html)\n",
      "\n",
      "[phylu​ce.readt​hedocs.io/en/lates​t/tutor​ial-four.html), suggests loci](http://phyluce.readthedocs.io/en/latest/tutorial-four.html)\n",
      "\n",
      "that are actually present can randomly fail to be detected by phy \n",
      "luce . Perhaps having a temporary bait design protocol with relaxed\n",
      "\n",
      "stringency enables the discovery of such “cryptic” loci shared among\n",
      "\n",
      "taxa. In this case, relaxed temporary bait design would be analogous\n",
      "\n",
      "to casting a wider net which captures both the “cryptic” and initially\n",
      "\n",
      "detected loci, which, when narrowed down later by more stringent\n",
      "\n",
      "parameters applied during final probe design (such as only targeting\n",
      "\n",
      "_candidate loci_ present in all study taxa), results in more loci detected\n",
      "\n",
      "and recovered by the final probe set.\n",
      "\n",
      "It is also possible that this result is an artifact of the two differ‐\n",
      "\n",
      "ent read mapping programs used by the phyluce pipeline (Faircloth,\n",
      "\n",
      "2017; Faircloth pers. comm.). Genomic reads are initially aligned to\n",
      "\n",
      "the base genome (Figure 1, step 2) using stampy (Lunter & Goodson,\n",
      "\n",
      "2011), which has different parameters than that of the LASTZ\n",
      "\n",
      "(Harris, 2007) alignment used by phyluce when aligning temporary\n",
      "\n",
      "baits back to entire genomic assemblies (Figure 1, step 4). The more\n",
      "\n",
      "permissive LASTZ alignment could therefore potentially be able to\n",
      "\n",
      "detect loci ignored initially by stampy (Faircloth pers. comm.).\n",
      "\n",
      "GUSTAFSON et al . **6945**\n",
      "**|**\n",
      "\n",
      "\n",
      "**4.3** | **In vitro considerations**\n",
      "\n",
      "While we focused here on the optimization of probes based on in\n",
      "\n",
      "silico testing, optimal parameters for in vitro enrichment of UCE\n",
      "\n",
      "probes have not been investigated thoroughly. Certain physical at‐\n",
      "\n",
      "tributes are well known to affect probe efficacy, such as GC% con‐\n",
      "\n",
      "tent (Tewhey et al., 2009), annealing temperature (Cruz‐Dávalos et\n",
      "\n",
      "al., 2017), and Gibbs free energy change (delta G) (Matveeva et al.,\n",
      "\n",
      "2016). Additional in‐solution parameters such as buffer stringency\n",
      "\n",
      "(Dabney & Meyer, 2012), washing procedures (Li, Hofreiter, Straube,\n",
      "\n",
      "Corrigan, & Naylor, 2013), and touchdown approaches to hybridiza‐\n",
      "\n",
      "tion temperature (Li et al., 2013; Paijmans, Fickel, Courtiol, Hofreiter,\n",
      "\n",
      "& Förster, 2016) also affect enrichment of targeted DNA loci. While\n",
      "\n",
      "in silico testing has proven comparable to in vitro testing (Branstetter\n",
      "\n",
      "et al., 2017; Starrett et al., 2017), and the resulting probes designed\n",
      "\n",
      "here should prove suitable for in vitro capture and enrichment (see\n",
      "\n",
      "Section 3.5), there will always be certain loci that remain incalcitrant\n",
      "\n",
      "to enrichment. For this reason, as has previously been emphasized\n",
      "\n",
      "(i.e., Faircloth, 2017), all probe sets should be regarded as experi‐\n",
      "\n",
      "mental until validated through in vitro testing. Future investigation\n",
      "\n",
      "of optimal enrichment parameters of UCE probes will likely help to\n",
      "\n",
      "normalize the number of loci recovered between in silico and in vitro\n",
      "\n",
      "testing.\n",
      "\n",
      "**4.4** | **Tailored UCE probe sets and universal utility**\n",
      "\n",
      "When first introduced, one of the appeals of UCEs was their ability\n",
      "\n",
      "to enrich hundreds to thousands of homologous loci across a clade\n",
      "\n",
      "(Faircloth et al., 2012). To a large extent, UCE probe sets such as\n",
      "\n",
      "those designed for tetrapods (Faircloth et al., 2012) have accom‐\n",
      "\n",
      "plished this goal with the same probes working well across the major\n",
      "\n",
      "amniote lineages (Crawford et al., 2012; McCormack et al., 2012,\n",
      "\n",
      "2013; Streicher et al., 2018). However, a probe set with similar\n",
      "\n",
      "universal utility has proven harder to attain for diverse arthropod\n",
      "\n",
      "groups like Coleoptera (see Baca et al., 2017; Van Dam et al., 2017),\n",
      "\n",
      "which while being younger in age (estimated origin during the mid‐\n",
      "\n",
      "Carboniferous (Toussaint et al., 2017)) likely has over ten times the\n",
      "\n",
      "number of species (~380,000 named species; Ślipiński et al., 2011);\n",
      "\n",
      "or Hymenoptera (see Branstetter et al., 2017), a similarly hyperdi‐\n",
      "\n",
      "verse insect order which may in reality be larger than Coleoptera\n",
      "\n",
      "by as much as 3.2 times (Forbes, Bagley, Beer, Hippee, & Widmayer,\n",
      "\n",
      "2018).\n",
      "\n",
      "The first Coleoptera probe set designed by Faircloth (2017) re‐\n",
      "\n",
      "covered hundreds of loci and provided phylogenetic signal when\n",
      "\n",
      "applied to Adephaga, a suborder of beetles not originally included\n",
      "\n",
      "during probe design (Baca et al., 2017). However, there was con‐\n",
      "\n",
      "siderable conflict between different phylogenetic reconstruction\n",
      "\n",
      "methods with regard to the basal nodes of the phylogeny. Therefore,\n",
      "\n",
      "improved phylogenetic reconstruction of Adephaga would be en‐\n",
      "\n",
      "hanced by a probe set tailored to this lineage to target more UCE\n",
      "\n",
      "loci and increase phylogenetic resolution. While the probe set devel‐\n",
      "\n",
      "oped here was specifically optimized for universal application across\n",
      "\n",
      "Adephaga, our design notably lacked a whirligig beetle genome, as\n",
      "\n",
      "\n",
      "none are currently available. Whirligig beetles (Gyrinidae) comprise\n",
      "\n",
      "~900 species (Short, 2018), are a relatively old family with an esti‐\n",
      "\n",
      "mated early Triassic origin (Gustafson, Prokin, Bukontaite, Bergsten,\n",
      "\n",
      "& Miller, 2017), and may represent the sister lineage to all other ade‐\n",
      "\n",
      "phagans (Baca et al., 2017). Thus, the probe set designed here could\n",
      "\n",
      "similarly result in fewer loci enriched for members of the Gyrinidae\n",
      "\n",
      "relative to other adephagan families used during probe design.\n",
      "\n",
      "However, this remains to be seen through in vitro testing and, alter‐\n",
      "\n",
      "natively, this probe set could prove effective across all of Adephaga,\n",
      "\n",
      "preventing need for additional probes tailored to the Gyrinidae.\n",
      "\n",
      "With an increase in the number of tailored probe sets, concern\n",
      "\n",
      "could be raised that this deflates the value of UCEs as different\n",
      "\n",
      "probe sets result in the recovery of different homologous loci, caus‐\n",
      "\n",
      "ing incompatibility between datasets and decreased “universality.”\n",
      "\n",
      "However, UCE probe set development and synthesis affords the\n",
      "\n",
      "ability to mix probes from different designs, diversifying the loci\n",
      "\n",
      "targeted while providing backward compatibility with older data\n",
      "\n",
      "sets (Branstetter et al., 2017). For example, Branstetter et al. (2017)\n",
      "\n",
      "developed a probe set tailored for ant lineages that included newly\n",
      "\n",
      "developed probes targeting ant‐specific loci, while retaining probes\n",
      "\n",
      "from the original universal Hymenoptera probe set designed by\n",
      "\n",
      "Faircloth et al. (2015) that worked well within ant lineages. We sim‐\n",
      "\n",
      "ilarly follow this approach, and in addition to our newly developed\n",
      "\n",
      "probes (Section 3.5), we include as part of our final probe set 3,667\n",
      "\n",
      "probes from the prior Coleoptera 1.1Kv1 probe set (Faircloth, 2017)\n",
      "\n",
      "targeting the 305 loci successfully recovered for Adephaga during\n",
      "\n",
      "the Baca et al. (2017) study.\n",
      "\n",
      "**5** | **CONCLUSION**\n",
      "\n",
      "Given the need to tailor UCE probe sets to focal taxa to improve\n",
      "\n",
      "locus recovery for use in phylogenomic studies and the increasing\n",
      "\n",
      "availability of genomic resources for diverse organisms, information\n",
      "\n",
      "about factors affecting probe design and methods for optimizing\n",
      "\n",
      "locus recovery are becoming increasingly important. We found an\n",
      "\n",
      "optimized UCE probe set depends on selection of an appropriate\n",
      "\n",
      "base genome and beginning probe design with less stringent tem‐\n",
      "\n",
      "porary bait design parameters. We recommend the following work‐\n",
      "\n",
      "flow (Figure 9) for using phyluce to design an optimized UCE probe\n",
      "\n",
      "set for organisms without a complete and well‐annotated genome\n",
      "\n",
      "available. First, the base genome experiments shown in Figure 1\n",
      "\n",
      "should be conducted in order to identify an optimal base genome\n",
      "\n",
      "or, alternatively, the base genome can be selected from the taxon\n",
      "\n",
      "with the smallest average genetic distance to the other study taxa\n",
      "\n",
      "as measured by independently generated molecular markers. Then,\n",
      "\n",
      "we recommend that temporary baits are designed based on _puta-_\n",
      "\n",
      "_tive loci_ shared between the base genome and as few as one other\n",
      "\n",
      "taxon. Combined, these factors resulted in an optimized probe set\n",
      "\n",
      "for adephagan beetles, recovering the most UCE loci in in silico tests\n",
      "\n",
      "as well as the longest per‐locus alignments. We make this final op‐\n",
      "\n",
      "timized adephagan beetle probe set “Adephaga_2.9Kv1,” consisting\n",
      "\n",
      "of 38,948 probes, targeting 2,948 UCE loci, available under a public\n",
      "\n",
      "**6946** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "domain license (CC‐0), similar to the prior Coleoptera probe set de‐\n",
      "\n",
      "signed by Faircloth (2017).\n",
      "\n",
      "**ACKNOWLEDGMENTS**\n",
      "\n",
      "This manuscript was greatly improved by the thoughtful review\n",
      "\n",
      "of Brant Faircloth and two anonymous reviewers. We thank the\n",
      "\n",
      "University of Kansas Center for Research Computing and the\n",
      "\n",
      "Biodiversity Institute for computational resources used during this\n",
      "\n",
      "project. GTG is supported by an NIH IRACDA postdoctoral fel‐\n",
      "\n",
      "lowship (5K12GM063651). This project was funded in part by the\n",
      "\n",
      "Harold E. and Leona M. Rice Endowment Fund at Oregon State\n",
      "\n",
      "University.\n",
      "\n",
      "**CONFLICT OF INTEREST**\n",
      "\n",
      "None declared.\n",
      "\n",
      "**AUTHOR CONTRIBUTIONS**\n",
      "\n",
      "G.T.G., A.A., and A.E.Z.S. conceived of the study. G.T.G. and A.A.\n",
      "\n",
      "conducted probe design and analyses. J.S.S., J.M.P., and D.R.M. ob‐\n",
      "\n",
      "tained and provided the Illumina and Sanger sequence data, assem‐\n",
      "\n",
      "bled the base genomes, and produced genomic assembly metrics. All\n",
      "\n",
      "authors contributed to writing of the manuscript.\n",
      "\n",
      "**DATA AVAILABILITY**\n",
      "\n",
      "[DNA sequences: GenBank MK838494 to MK838511; P](info:x-wiley/peptideatlas/MK838494) hyluce BED/\n",
      "\n",
      "BAM files for probe design experiments and Adephaga_2.9Kv1 final\n",
      "\n",
      "[probe set fasta file: Dryad https​://doi.org/10.5061/dryad.2f62927;](https://doi.org/10.5061/dryad.2f62927)\n",
      "\n",
      "Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fas‐\n",
      "\n",
      "[tas available from; https​://github.com/lanin​sky/compa​ring_monol​](https://github.com/laninsky/comparing_monolithic_UCE_fastas)\n",
      "\n",
      "[ithic_UCE_fastas; Alexander 2018b: Phase_target available from](https://github.com/laninsky/comparing_monolithic_UCE_fastas)\n",
      "\n",
      "[https​://github.com/lanin​sky/refer​ence_align​ing_to_estab​lished_loci/](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target)\n",
      "\n",
      "[tree/maste​r/phase_target; Alexander 2018c: distance_calcs available](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target)\n",
      "\n",
      "[from: https​://github.com/lanin​sky/dista​nce_calcs​; Alexander 2018d:](https://github.com/laninsky/distance_calcs)\n",
      "\n",
      "[Phase_everyone available from https​://github.com/lanin​sky/refer​](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone)\n",
      "\n",
      "[ence_align​ing_to_estab​lished_loci/tree/maste​r/phase_everyone.](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone)\n",
      "\n",
      "**ORCID**\n",
      "\n",
      "_Grey T. Gustafson_ [https://orcid.org/0000-0002-2014-7128](https://orcid.org/0000-0002-2014-7128)\n",
      "\n",
      "_Alana Alexander_ [https://orcid.org/0000-0002-6456-7757](https://orcid.org/0000-0002-6456-7757)\n",
      "\n",
      "_John S. Sproul_ [https://orcid.org/0000-0002-6747-3537](https://orcid.org/0000-0002-6747-3537)\n",
      "\n",
      "_David R. Maddison_ [https://orcid.org/0000-0002-7152-3824](https://orcid.org/0000-0002-7152-3824)\n",
      "\n",
      "_Andrew E. Z. Short_ [https://orcid.org/0000-0002-7467-7116](https://orcid.org/0000-0002-7467-7116)\n",
      "\n",
      "**REFERENCES**\n",
      "\n",
      "Alexander, A. M., Su, Y. C., Oliveros, C. H., Olson, K. V., Travers, S.\n",
      "\n",
      "L., & Brown, R. M. (2017). Genomic data reveals potential for\n",
      "\n",
      "\n",
      "hybridization, introgression, and incomplete lineage sorting to\n",
      "\n",
      "confound phylogenetic relationships in an adaptive radiation\n",
      "of narrow‐mouth frogs. _Evolution_, _71_ [(2), 475–488. https​://doi.](https://doi.org/10.1111/evo.13133)\n",
      "\n",
      "[org/10.1111/evo.13133​](https://doi.org/10.1111/evo.13133)\n",
      "Baca, S. M., Alexander, A., Gustafson, G. T., & Short, A. E. Z. (2017).\n",
      "\n",
      "Ultraconserved elements show utility in phylogenetic inference of\n",
      "Adephaga (Coleoptera) and suggest paraphyly of 'Hydradephaga'.\n",
      "_Systematic Entomology_, _42_ [(4), 786–795. https​://doi.org/10.1111/](https://doi.org/10.1111/syen.12244)\n",
      "\n",
      "[syen.12244​](https://doi.org/10.1111/syen.12244)\n",
      "\n",
      "Blaimer, B. B., Brady, S. G., Schultz, T. R., Lloyd, M. W., Fisher, B. L., &\n",
      "Ward, P. S. (2015). Phylogenomic methods outperform traditional\n",
      "\n",
      "multi‐locus approaches in resolving deep evolutionary history: A\n",
      "\n",
      "case study of formicine ants. _BMC Evolutionary Biology_, _15_ [, 271. https​](https://doi.org/10.1186/s12862-015-0552-5)\n",
      "\n",
      "[://doi.org/10.1186/s12862-015-0552-5](https://doi.org/10.1186/s12862-015-0552-5)\n",
      "Blaimer, B. B., Lloyd, M. W., Guillory, W. X., & Brady, S. G. (2016).\n",
      "\n",
      "Sequence capture and phylogenetic utility of genomic ultracon‐\n",
      "\n",
      "served elements obtained from pinned insect specimens. _PLoS ONE_,\n",
      "\n",
      "_11_ [, e0161531. https​://doi.org/10.1371/journ​al.pone.0161531](https://doi.org/10.1371/journal.pone.0161531)\n",
      "Branstetter, M. G., Longino, J. T., Ward, P. S., & Faircloth, B. C. (2017).\n",
      "\n",
      "Enriching the ant tree of life: Enhanced UCE bait set for genome‐scale\n",
      "\n",
      "phylogenetics of ants and other Hymenoptera. _Methods in Ecology and_\n",
      "\n",
      "_Evolution_, _8_ [, 768–776. https​://doi.org/10.1111/2041-210X.12742​](https://doi.org/10.1111/2041-210X.12742)\n",
      "Bushnell, B. (2014). _BBTools software package_ [. Retrieved from http://](http://sourceforge.net/projects/bbmap)\n",
      "\n",
      "[sourc​eforge.net/proje​cts/bbmap​](http://sourceforge.net/projects/bbmap)\n",
      "Campana, M. G. (2018). BaitsTools: Software for hybridization capture\n",
      "bait design. _Molecular Ecology Resources_, _18_ [(2), 356–361. https​://doi.](https://doi.org/10.1111/1755-0998.12721)\n",
      "\n",
      "[org/10.1111/1755-0998.12721​](https://doi.org/10.1111/1755-0998.12721)\n",
      "Castresana, J. (2000). Selection of conserved blocks from multiple align‐\n",
      "\n",
      "ments for their use in phylogenetic analysis. _Molecular Biology and_\n",
      "\n",
      "_Evolution_, _17_ [, 540–552. https​://doi.org/10.1093/oxfor​djour​nals.mol‐](https://doi.org/10.1093/oxfordjournals.molbev.a026334)\n",
      "\n",
      "[bev.a026334](https://doi.org/10.1093/oxfordjournals.molbev.a026334)\n",
      "Che, L. H., Zhang, S.‐Q., Li, Y., Liang, D., Pang, H., Ślipiński, A., & Zhang, P.\n",
      "\n",
      "(2017). Genome‐wide survey of nuclear protein‐coding markers for\n",
      "\n",
      "beetle phylogenetics and their application in resolving both deep and\n",
      "shallow‐level divergences. _Molecular Ecology Resources_, _17_ (6), 1342–\n",
      "\n",
      "[1358. https​://doi.org/10.1111/1755-0998.12664​](https://doi.org/10.1111/1755-0998.12664)\n",
      "Chiari, Y., Cahais, V., Galtier, N., & Delsuc, F. (2012). Phylogenomic analy‐\n",
      "\n",
      "ses support the position of turtles as sister group of birds and croco‐\n",
      "diles (Archosauria). _BMC Biology_, _10_, 65.\n",
      "Core Team, R. (2017). _R: A language and environment for statistical com-_\n",
      "\n",
      "_puting_ . Vienna, Austria: R Foundation for Statistical Computing.\n",
      "\n",
      "[Retrieved from http://www.R-proje​ct.org/](http://www.R-project.org/)\n",
      "\n",
      "Crawford, N. G., Faircloth, B. C., McCormack, J. E., Brumfield, R. T.,\n",
      "\n",
      "Winker, K., & Glenn, T. C. (2012). More than 1000 ultraconserved\n",
      "\n",
      "elements provide evidence that turtles are the sister group of ar‐\n",
      "chosaurs. _Biology Letters_, _8_ [(5), 783–786. https​://doi.org/10.1098/](https://doi.org/10.1098/rsbl.2012.0331)\n",
      "\n",
      "[rsbl.2012.0331](https://doi.org/10.1098/rsbl.2012.0331)\n",
      "\n",
      "Cruz‐Dávalos, D. I., Llamas, B., Gaunitz, C., Fages, A., Gamba, C., Soubrier,\n",
      "J., … Orlando, L. (2017). Experimental conditions improving in‐solu‐\n",
      "\n",
      "tion target enrichment for ancient DNA. _Molecular Ecology Resources_,\n",
      "_17_ [(3), 508–522. https​://doi.org/10.1111/1755-0998.12595​](https://doi.org/10.1111/1755-0998.12595)\n",
      "Dabney, J., & Meyer, M. (2012). Length and GC‐biases during sequenc‐\n",
      "\n",
      "ing library amplification: A comparison of various polymerase‐buf‐\n",
      "\n",
      "fer systems with ancient and modern DNA sequencing libraries.\n",
      "\n",
      "_BioTechniques_, _52_ [, 87–94. https​://doi.org/10.2144/00011​3809](https://doi.org/10.2144/000113809)\n",
      "Dowle, M., & Srinivasan, A. (2017). _data.table: Extension of ‘data.frame’_ . R\n",
      "\n",
      "[package version 1.10.4. Retrieved from https​://CRAN.R-proje​ct.org/](https://CRAN.R-project.org/package=data.table)\n",
      "\n",
      "[packa​ge=data.table​](https://CRAN.R-project.org/package=data.table)\n",
      "Esselstyn, J. A., Oliveros, C. H., Swanson, M. T., & Faircloth, B. C. (2017).\n",
      "\n",
      "Investigating difficult nodes in the placental mammal tree with\n",
      "\n",
      "expanded taxon sampling and thousands of ultraconserved loci.\n",
      "_Genome Biology and Evolution_, _9_ (9), 2308–2321.\n",
      "Faircloth, B. C. (2016). PHYLUCE is a software package for the analysis of\n",
      "conserved genomic loci. _Bioinformatics_, _32_ [(5), 786–788. https​://doi.](https://doi.org/10.1093/bioinformatics/btv646)\n",
      "\n",
      "[org/10.1093/bioin​forma​tics/btv646](https://doi.org/10.1093/bioinformatics/btv646)\n",
      "\n",
      "GUSTAFSON et al . **6947**\n",
      "**|**\n",
      "\n",
      "\n",
      "Faircloth, B. C. (2017). Identifying conserved genomic elements and design‐\n",
      "\n",
      "ing universal bait sets to enrich them. _Methods in Ecology and Evolution_,\n",
      "_8_ [(9), 1103–1112. https​://doi.org/10.1111/2041-210X.12754​](https://doi.org/10.1111/2041-210X.12754)\n",
      "Faircloth, B. C., Branstetter, M. G., White, N. D., & Brady, S. G. (2015).\n",
      "\n",
      "Target enrichment of ultraconserved elements from arthro‐\n",
      "\n",
      "pods provides a genomic perspective on relationships among\n",
      "Hymenoptera. _Molecular Ecology Resources_, _15_ [(3), 489–501. https​://](https://doi.org/10.1111/1755-0998.12328)\n",
      "\n",
      "[doi.org/10.1111/1755-0998.12328​](https://doi.org/10.1111/1755-0998.12328)\n",
      "\n",
      "Faircloth, B. C., McCormack, J. E., Crawford, N. G., Harvey, M. G.,\n",
      "Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements\n",
      "\n",
      "anchor thousands of genetic markers spanning multiple evolu‐\n",
      "tionary timescales. _Systematic Biology_, _61_ [(5), 717–726. https​://doi.](https://doi.org/10.1093/sysbio/sys004)\n",
      "\n",
      "[org/10.1093/sysbi​o/sys004](https://doi.org/10.1093/sysbio/sys004)\n",
      "Faircloth, B. C., Sorenson, L., Santini, F., & Alfaro, M. E. (2013). A phylog‐\n",
      "\n",
      "enomic perspective on the radiation of ray‐finned fishes based upon\n",
      "targeted sequencing of ultraconserved elements (UCEs). _PLoS ONE_,\n",
      "\n",
      "_8_ [, e65923. https​://doi.org/10.1371/journ​al.pone.0065923](https://doi.org/10.1371/journal.pone.0065923)\n",
      "\n",
      "Forbes, A. A., Bagley, R. K., Beer, M. A., Hippee, A. C., & Widmayer, H.\n",
      "A. (2018). Quantifying the unquantifiable: Why Hymenoptera, not\n",
      "\n",
      "Coleoptera, is the most speciose animal order. _BMC Ecology_, _18_, 21.\n",
      "\n",
      "[https​://doi.org/10.1186/s12898-018-0176-x](https://doi.org/10.1186/s12898-018-0176-x)\n",
      "Glenn, T. C., & Faircloth, B. C. (2016). Capturing Darwin’s dream.\n",
      "_Molecular_ _Ecology_ _Resources_, _16_ (5), 1051–1058. [https​://doi.](https://doi.org/10.1111/1755-0998.12574)\n",
      "\n",
      "[org/10.1111/1755-0998.12574​](https://doi.org/10.1111/1755-0998.12574)\n",
      "Gomez, R. A., Will, K. W., & Maddison, D. R. (2016). Are _Miquihuana_\n",
      "_rhadiniformis_ Barr, 1982 and _Pseudamara arenaria_ (LeConte, 1847)\n",
      "(Coleoptera, Carabidae) sphodrines? Phylogenetic analysis of data\n",
      "\n",
      "from next‐generation sequencing of museum specimens resolves the\n",
      "\n",
      "tribal‐group relationships of these enigmatic taxa. _Entomologische_\n",
      "\n",
      "_Blätter Und Coleoptera_, _112_, 149–168.\n",
      "\n",
      "Grismer, J. L., Schulte, J. A., Alexander, A., Wagner, P., Travers, S. L.,\n",
      "Buehler, M. D., … Brown, R. M. (2016). The Eurasian invasion:\n",
      "\n",
      "Phylogenomic data reveal multiple Southeast Asian origins for Indian\n",
      "Dragon Lizards. _BMC Evolutionary Biology_, _16_ [(1), 43. https​://doi.](https://doi.org/10.1186/s12862-016-0611-6)\n",
      "\n",
      "[org/10.1186/s12862-016-0611-6](https://doi.org/10.1186/s12862-016-0611-6)\n",
      "\n",
      "Gustafson, G. T., Prokin, A. A., Bukontaite, A., Bergsten, J., & Miller, K.\n",
      "B. (2017). Tip‐dated phylogeny of whirligig beetles reveals ancient\n",
      "\n",
      "lineage surviving on Madagascar. _Scientific Reports_, _7_ [, 8619. https​://](https://doi.org/10.1038/s41598-017-08403-1)\n",
      "\n",
      "[doi.org/10.1038/s41598-017-08403-1](https://doi.org/10.1038/s41598-017-08403-1)\n",
      "Harris, R. S. (2007). _Improved pairwise alignment of genomic DNA_ . PhD\n",
      "\n",
      "thesis, Pennsylvania State University.\n",
      "Hosner, P. A., Tobias, J. A., Braun, E. L., & Kimball, R. T. (2017). How do\n",
      "\n",
      "seemingly non‐vagile clades accomplish trans‐marine dispersal?\n",
      "Trait and dispersal evolution in the landfowl (Aves: Galliformes).\n",
      "_Proceedings of the Royal Society B: Biological Sciences_, _284_ (1854),\n",
      "\n",
      "[20170210. https​://doi.org/10.1098/rspb.2017.0210](https://doi.org/10.1098/rspb.2017.0210)\n",
      "\n",
      "Johnson, M. G., Gardner, E. M., Liu, Y., Medina, R., Goffinet, B., Shaw,\n",
      "\n",
      "A. J., … Wickett, N. J. (2016). HybPiper: Extracting coding sequence\n",
      "\n",
      "and introns for phylogenetics from high‐throughput sequencing\n",
      "reads using target enrichment. _Applications in Plant Sciences_, _4_ (7),\n",
      "\n",
      "[1600016. https​://doi.org/10.3732/apps.1600016](https://doi.org/10.3732/apps.1600016)\n",
      "\n",
      "Kanda, K., Pflug, J. M., Sproul, J. S., Dasenko, M. A., & Maddison, D. R.\n",
      "(2015). Successful recovery of nuclear protein‐coding genes from\n",
      "\n",
      "small insects in museums using illumina sequencing. _PLoS ONE_,\n",
      "_10_ [(12), e0143929. https​://doi.org/10.1371/journ​al.pone.0143929](https://doi.org/10.1371/journal.pone.0143929)\n",
      "Katoh, K., & Standley, D. M. (2013). MAFFT multiple sequence align‐\n",
      "\n",
      "ment software version 7: Improvements in performance and us‐\n",
      "ability. _Molecular Biology and Evolution_, _30_ [(4), 772–780. https​://doi.](https://doi.org/10.1093/molbev/mst010)\n",
      "\n",
      "[org/10.1093/molbe​v/mst010](https://doi.org/10.1093/molbev/mst010)\n",
      "Li, C., Hofreiter, M., Straube, N., Corrigan, S., & Naylor, G. J. (2013).\n",
      "\n",
      "Capturing protein‐coding genes across highly divergent species.\n",
      "\n",
      "_BioTechniques_, _54_ [, 321–326. https​://doi.org/10.2144/00011​4039](https://doi.org/10.2144/000114039)\n",
      "Li, H., & Durbin, R. (2009). Fast and accurate short read alignment with\n",
      "\n",
      "Burrows‐Wheeler transform. _Bioinformatics_, _25_ [, 1754–1760. https​://](https://doi.org/10.1093/bioinformatics/btp324)\n",
      "\n",
      "[doi.org/10.1093/bioin​forma​tics/btp324](https://doi.org/10.1093/bioinformatics/btp324)\n",
      "\n",
      "\n",
      "Li, H., & Durbin, R. (2010). Fast and accurate long‐read alignment with\n",
      "\n",
      "Burrows‐Wheeler transform. _Bioinformatics_, _26_ [, 589–595. https​://](https://doi.org/10.1093/bioinformatics/btp698)\n",
      "\n",
      "[doi.org/10.1093/bioin​forma​tics/btp698](https://doi.org/10.1093/bioinformatics/btp698)\n",
      "\n",
      "Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., …\n",
      "1000 Genome Project Data Processing Subgroup (2009). The se‐\n",
      "quence alignment/map (SAM) format and SAMtools. _Bioinformatics_,\n",
      "\n",
      "_25_, 2078–2079.\n",
      "\n",
      "Lunter, G., & Goodson, M. (2011). Stampy: A statistical algorithm for sen‐\n",
      "\n",
      "sitive and fast mapping of Illumina sequence reads. _Genome Research_,\n",
      "\n",
      "_21_ [, 936–939. https​://doi.org/10.1101/gr.111120.110](https://doi.org/10.1101/gr.111120.110)\n",
      "Maddison, D. R. (2012). Phylogeny of _Bembidion_ and related ground\n",
      "beetles (Coleoptera: Carabidae: Trechinae: Bembidiini: Bembidiina).\n",
      "\n",
      "_Molecular Phylogenetics and Evolution_, _63_ [, 533–576. https​://doi.](https://doi.org/10.1016/j.ympev.2012.01.015)\n",
      "\n",
      "[org/10.1016/j.ympev.2012.01.015](https://doi.org/10.1016/j.ympev.2012.01.015)\n",
      "Maddison, D. R., Baker, M. D., & Ober, K. A. (1999a). Phylogeny of ca‐\n",
      "rabid beetles as inferred from 18S ribosomal DNA (Coleoptera:\n",
      "Carabidae). _Systematic Entomology_, _24_ [, 103–138. https​://doi.](https://doi.org/10.1046/j.1365-3113.1999.00088.x)\n",
      "\n",
      "[org/10.1046/j.1365-3113.1999.00088.x](https://doi.org/10.1046/j.1365-3113.1999.00088.x)\n",
      "Maddison, D. R., Baker, M. D., & Ober, K. A.. (1999b). A preliminary phy‐\n",
      "logenetic analysis of 18S ribosomal DNA of carabid beetles (Insecta:\n",
      "Coleoptera). In Ball, G. E., Casale, A., & Vigna‐Taglianti, A. (Eds.),\n",
      "_Phylogeny and classification of Caraboidea (Coleoptera: Adephaga)_, pp.\n",
      "\n",
      "229–250. Torino, Italy: Museo Regionale di Scienze Naturali, Atti.\n",
      "\n",
      "517 p.\n",
      "\n",
      "Maddison, D. R., Moore, W., Baker, M. D., Ellis, T. M., Ober, K. A.,\n",
      "\n",
      "Cannone, J. J., & Gutell, R. R. (2009). Monophyly of terrestrial ade‐\n",
      "phagan beetles as indicated by three nuclear genes (Coleoptera:\n",
      "Carabidae and Trachypachidae). _Zoologica Scripta_, _38_ [(1), 43–62. https​](https://doi.org/10.1111/j.1463-6409.2008.00359.x)\n",
      "\n",
      "[://doi.org/10.1111/j.1463-6409.2008.00359.x](https://doi.org/10.1111/j.1463-6409.2008.00359.x)\n",
      "Maddison, D. R., & Swanson, A. P. (2010). A preliminary characterization\n",
      "\n",
      "of Bembidion perspicuum LeConte, with a reclassification of related\n",
      "species (Coleoptera, Carabidae) north of México. _ZooKeys_, _43_, 15–\n",
      "\n",
      "[31. https​://doi.org/10.3897/zooke​ys.43.390](https://doi.org/10.3897/zookeys.43.390)\n",
      "\n",
      "Mamanova, L., Coffey, A. J., Scott, C. E., Kozarewa, I., Turner, E. H.,\n",
      "Kumar, A., … Turner, D. J. (2010). Target‐enrichment strategies for\n",
      "next‐generation sequencing. _Nature Methods_, _7_ [(2), 111–118. https​://](https://doi.org/10.1038/nmeth.1419)\n",
      "\n",
      "[doi.org/10.1038/nmeth.1419](https://doi.org/10.1038/nmeth.1419)\n",
      "Manthey, J. D., Campillo, L. C., Burns, K. J., & Moyle, R. G. (2016).\n",
      "\n",
      "Comparison of target‐capture and restriction‐site associated DNA\n",
      "sequencing for phylogenomics: A test in cardinalid Tanagers (Aves,\n",
      "Genus: _Piranga_ ). _Systematic Biology_, _65_ (4), 640–650.\n",
      "\n",
      "Matveeva, O. V., Nechipurenko, Y. D., Riabenko, E., Ragan, C., Nazipova,\n",
      "N. N., Ogurtsov, A. Y., & Shabalina, S. A. (2016). Optimization\n",
      "\n",
      "of signal‐to‐noise ratio for efficient microarray probe design.\n",
      "_Bioinformatics_, _30_ [(17), i552–i558. https​://doi.org/10.1093/bioin​](https://doi.org/10.1093/bioinformatics/btw451)\n",
      "\n",
      "[forma​tics/btw451](https://doi.org/10.1093/bioinformatics/btw451)\n",
      "\n",
      "Mayer, C., Sann, M., Donath, A., Meixner, M., Podsiadlowski, L., Peters,\n",
      "R. S., … Niehuis, O. (2016). BaitFisher: A software package for mul‐\n",
      "\n",
      "tispecies target DNA enrichment probe design. _Molecular Biology_\n",
      "_and Evolution_, _33_ [(7), 1875–1886. https​://doi.org/10.1093/molbe​v/](https://doi.org/10.1093/molbev/msw056)\n",
      "\n",
      "[msw056](https://doi.org/10.1093/molbev/msw056)\n",
      "\n",
      "McCormack, J. E., Faircloth, B. C., Crawford, N. G., Gowaty, P. A.,\n",
      "Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements are\n",
      "\n",
      "novel phylogenomic markers that resolved placental mammal phy‐\n",
      "\n",
      "logeny when combined with species‐tree analysis. _Genome Research_,\n",
      "_22_ (4), 746–754.\n",
      "\n",
      "McCormack, J. E., Harvey, M. G., Faircloth, B. C., Crawford, N. G., Glenn,\n",
      "T. C., & Brumfield, R. T. (2013). A phylogeny of birds based on over\n",
      "\n",
      "1,500 loci collected by target enrichment and high‐throughput se‐\n",
      "\n",
      "quencing. _PLoS ONE_, _8_ [, e54848. https​://doi.org/10.1371/journ​](https://doi.org/10.1371/journal.pone.0054848)\n",
      "\n",
      "[al.pone.0054848](https://doi.org/10.1371/journal.pone.0054848)\n",
      "\n",
      "McCormack, J. E., Hird, S. M., Zellmer, A. J., Carstens, B. C., & Brumfield,\n",
      "\n",
      "R. T. (2013). Applications of next‐generation sequencing to phylo‐\n",
      "\n",
      "geography and phylogenetics. _Molecular Phylogenetics and Evolution_,\n",
      "_66_ [(2), 526–538. https​://doi.org/10.1016/j.ympev.2011.12.007](https://doi.org/10.1016/j.ympev.2011.12.007)\n",
      "\n",
      "**6948** GUSTAFSON et al .\n",
      "**|**\n",
      "\n",
      "\n",
      "McCormack, J. E., Tsai, W. L., & Faircloth, B. C. (2016). Sequence\n",
      "\n",
      "capture of ultraconserved elements from bird museum speci‐\n",
      "mens. _Molecular Ecology Resources_, _16_ [(5), 1189–1203. https​://doi.](https://doi.org/10.1111/1755-0998.12466)\n",
      "\n",
      "[org/10.1111/1755-0998.12466​](https://doi.org/10.1111/1755-0998.12466)\n",
      "\n",
      "McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K.,\n",
      "\n",
      "Kernytsky, A., … DePristo, M. A. (2010). The Genome Analysis\n",
      "\n",
      "Toolkit: A MapReduce framework for analyzing next‐generation\n",
      "DNA sequencing data. _Genome Resources_, _20_ [(9), 1297–1303. https​://](https://doi.org/10.1101/gr.107524.110)\n",
      "\n",
      "[doi.org/10.1101/gr.107524.110](https://doi.org/10.1101/gr.107524.110)\n",
      "\n",
      "McKenna, D. D., Wild, A. L., Kanda, K., Bellamy, C. L., Beutel, R. G.,\n",
      "Caterino, M. S., … Farrell, B. D. (2015). The beetle tree of life reveals\n",
      "\n",
      "that Coleoptera survived end‐Permian mass extinction to diversify\n",
      "\n",
      "during the Cretaceous terrestrial revolution. _Systematic Entomology_,\n",
      "\n",
      "_40_ [, 835–880. https​://doi.org/10.1111/syen.12132​](https://doi.org/10.1111/syen.12132)\n",
      "Musher, L. J., & Cracraft, J. (2018). Phylogenomics and species de‐\n",
      "\n",
      "limitation of a complex radiation of Neotropical suboscine birds\n",
      "( _Pachyramphus_ ). _Molecular Phylogenetics and Evolution_, _118_, 204–221.\n",
      "\n",
      "[https​://doi.org/10.1016/j.ympev.2017.09.013](https://doi.org/10.1016/j.ympev.2017.09.013)\n",
      "Nguyen, L.‐T., Schmidt, H. A., von Haeseler, A., & Minh, B. Q. (2015). IQ‐\n",
      "\n",
      "TREE: A fast and effective stochastic algorithm for estimating max‐\n",
      "imum likelihood phylogenies. _Molecular Biology and Evolution_, _32_ (1),\n",
      "\n",
      "[268–274. https​://doi.org/10.1093/molbe​v/msu300](https://doi.org/10.1093/molbev/msu300)\n",
      "Ober, K. (2002). Phylogenetic relationships of the carabid subfam‐\n",
      "ily Harpalinae (Coleoptera) based on molecular sequence data.\n",
      "\n",
      "_Molecular Phylogenetics and Evolution_, _24_ [, 228–248. https​://doi.](https://doi.org/10.1016/S1055-7903(02)00251-8)\n",
      "[org/10.1016/S1055-7903(02)00251-8](https://doi.org/10.1016/S1055-7903(02)00251-8)\n",
      "\n",
      "Paijmans, J. L. A., Fickel, J., Courtiol, A., Hofreiter, M., & Förster, D. W.\n",
      "(2016). Impact of enrichment conditions on cross‐species capture of\n",
      "fresh and degraded DNA. _Molecular Ecology Resources_, _16_ (1), 42–55.\n",
      "\n",
      "[https​://doi.org/10.1111/1755-0998.12420​](https://doi.org/10.1111/1755-0998.12420)\n",
      "\n",
      "Quattrini, A. M., Faircloth, B. C., Dueñas, L. F., Bridge, T. C. L., Brugler,\n",
      "M. R., Calixto‐Botía, I. F., … McFadden, C. S. (2018). Universal tar‐\n",
      "get‐enrichment baits for anthozoan (Cnidaria) phylogenomics: New\n",
      "\n",
      "approaches to long‐standing problems. _Molecular Ecology Resources_,\n",
      "_18_ [(2), 281–295. https​://doi.org/10.1111/1755-0998.12736​](https://doi.org/10.1111/1755-0998.12736)\n",
      "Rice, P., Longden, I., & Bleasby, A. (2000). EMBOSS: The European mo‐\n",
      "lecular biology open software suite. _Trends in Genetics_, _16_ (6), 276–\n",
      "[277. https​://doi.org/10.1016/S0168-9525(00)02024-2](https://doi.org/10.1016/S0168-9525(00)02024-2)\n",
      "Ruane, S., & Austin, C. C. (2017). Phylogenomics using formalin‐\n",
      "\n",
      "fixed and 100+ year‐old intractable natural history specimens.\n",
      "_Molecular_ _Ecology_ _Resources_, _17_ (5), 1003–1008. [https​://doi.](https://doi.org/10.1111/1755-0998.12655)\n",
      "\n",
      "[org/10.1111/1755-0998.12655​](https://doi.org/10.1111/1755-0998.12655)\n",
      "Shen, X.‐X., Liang, D., Wen, J.‐Z., & Zhang, P. (2011). Multiple genome\n",
      "\n",
      "alignments facilitate development of NPCL markers: A case study\n",
      "\n",
      "of tetrapod phylogeny focusing on the position of turtles. _Molecular_\n",
      "_Biology and Evolution_, _28_ [(12), 3237–3252. https​://doi.org/10.1093/](https://doi.org/10.1093/molbev/msr148)\n",
      "\n",
      "[molbe​v/msr148](https://doi.org/10.1093/molbev/msr148)\n",
      "\n",
      "Short, A. E. Z. (2018). Systematics of aquatic beetles (Coleoptera):\n",
      "\n",
      "Current state and future directions. _Systematic Entomology_, _43_, 1–18.\n",
      "\n",
      "[https​://doi.org/10.1111/syen.12270​](https://doi.org/10.1111/syen.12270)\n",
      "\n",
      "Simão, F. A., Waterhouse, R. M., Ioannidis, P., Kriventseva, E. V., &\n",
      "\n",
      "Zdobnov, E. M. (2015). BUSCO: Assessing genome assembly and\n",
      "\n",
      "annotation completeness with single‐copy orthologs. _Bioinformatics_,\n",
      "_31_ [(19), 3210–3212. https​://doi.org/10.1093/bioin​forma​tics/btv351](https://doi.org/10.1093/bioinformatics/btv351)\n",
      "Ślipiński, A., Leschen, R. A. B., & Lawrence, J. F. (2011). Order Coleoptera\n",
      "\n",
      "Linnaeus, 1758. Animal biodiversity: An outline of higher‐level clas‐\n",
      "sification and survey of taxonomic richness (ed. By Z.‐Q. Zhang).\n",
      "\n",
      "_Zootaxa_, _3148_, 203–208.\n",
      "\n",
      "Smith, B. T., Harvey, M. G., Faircloth, B. C., Glenn, T. C., & Brumfield,\n",
      "R. T. (2014). Target capture and massively parallel sequencing of\n",
      "\n",
      "ultraconserved elements for comparative studies at shallow evo‐\n",
      "lutionary time scales. _Systematic Biology_, _63_ [(1), 83–95. https​://doi.](https://doi.org/10.1093/sysbio/syt061)\n",
      "\n",
      "[org/10.1093/sysbi​o/syt061](https://doi.org/10.1093/sysbio/syt061)\n",
      "\n",
      "\n",
      "Sproul, J. S., & Maddison, D. R. (2017). Sequencing historical specimens:\n",
      "\n",
      "Successful preparation of small specimens with low amounts of de‐\n",
      "\n",
      "graded DNA. _Molecular Ecology Resources_, _17_ [, 1183–1201. https​://](https://doi.org/10.1111/1755-0998.12660)\n",
      "\n",
      "[doi.org/10.1111/1755-0998.12660​](https://doi.org/10.1111/1755-0998.12660)\n",
      "Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis\n",
      "and post‐analysis of large phylogenies. _Bioinformatics_, _30_ (9), 1312–\n",
      "\n",
      "[1313. https​://doi.org/10.1093/bioin​forma​tics/btu033](https://doi.org/10.1093/bioinformatics/btu033)\n",
      "\n",
      "Starrett, J., Derkarabetian, S., Hedin, M., Byrson, R. W. Jr, McCormack,\n",
      "J., & Fairclotch, B. C. (2017). High phylogenetic utility of an ul‐\n",
      "\n",
      "traconserved element probe set designed for Arachnida.\n",
      "_Molecular_ _Ecology_ _Resources_, _17_ (4), 812–823. [https​://doi.](https://doi.org/10.1111/1755-0998.12621)\n",
      "\n",
      "[org/10.1111/1755-0998.12621​](https://doi.org/10.1111/1755-0998.12621)\n",
      "\n",
      "Streicher, J. W., Miller, E. C., Guerrero, P. C., Correa, C., Ortiz, J. C.,\n",
      "\n",
      "Crawford, A. J., … Wiens, J. J. (2018). Evaluating methods for phy‐\n",
      "\n",
      "logenomic analyses, and a new phylogeny for a major frog clade\n",
      "(Hyloidea) based on 2214 loci. _Molecular Phylogenetics and Evolution_,\n",
      "\n",
      "_119_ [, 128–143. https​://doi.org/10.1016/j.ympev.2017.10.013](https://doi.org/10.1016/j.ympev.2017.10.013)\n",
      "Streicher, J. W., & Wiens, J. J. (2017). Phylogenomic analyses of more\n",
      "\n",
      "than 4000 nuclear loci resolve the origin of snakes among lizard\n",
      "families. _Biology Letters_, _13_ [(9), 20170393. https​://doi.org/10.1098/](https://doi.org/10.1098/rsbl.2017.0393)\n",
      "\n",
      "[rsbl.2017.0393](https://doi.org/10.1098/rsbl.2017.0393)\n",
      "\n",
      "Sun, K., Meiklejohn, K. A., Faircloth, B. C., Glenn, T. C., Braun, E. L.,\n",
      "& Kimball, R. T. (2014). The evolution of peafowl and other taxa\n",
      "with ocelli (eyespots): A phylogenomic approach. _Proceedings of_\n",
      "\n",
      "_the Royal Society B: Biological Sciences_, _281_ [, 20140823. https​://doi.](https://doi.org/10.1098/rspb.2014.0823)\n",
      "\n",
      "[org/10.1098/rspb.2014.0823](https://doi.org/10.1098/rspb.2014.0823)\n",
      "Talavera, G., & Castresana, J. (2007). Improvement of phylogenies after\n",
      "\n",
      "removing divergent and ambiguously aligned blocks from protein\n",
      "\n",
      "sequence alignments. _Systematic Biology_, _56_ [, 564–577. https​://doi.](https://doi.org/10.1080/10635150701472164)\n",
      "\n",
      "[org/10.1080/10635​15070​1472164](https://doi.org/10.1080/10635150701472164)\n",
      "\n",
      "Tewhey, R., Nakano, M., Wang, X., Pabón‐Peña, C., Novak, B., Giuffre,\n",
      "A., … Frazer, K. A. (2009). Enrichment of sequencing targets from the\n",
      "\n",
      "human genome by solution hybridization. _Genome Biology_, _10_, R116.\n",
      "\n",
      "[https​://doi.org/10.1186/gb-2009-10-10-r116](https://doi.org/10.1186/gb-2009-10-10-r116)\n",
      "\n",
      "Toussaint, E. F. A., Seidel, M., Arriaga‐Varela, E., Hájek, J., Král, D., Sekerka,\n",
      "L., … Fikáček, M. (2017). The peril of dating beetles. _Systematic_\n",
      "_Entomology_, _42_ [(1), 1–10. https​://doi.org/10.1111/syen.12198​](https://doi.org/10.1111/syen.12198)\n",
      "\n",
      "Van Dam, M. H., Lam, A. W., Sagata, K., Gewa, B., Laufa, R., Balke, M., …\n",
      "Reidel, A. (2017). Ultraconserved elements (UCEs) resolve the phy‐\n",
      "\n",
      "logeny of Australasian smurf‐weevils. _PLoS ONE_, _12_, e0188044.\n",
      "Wickham, H. (2017). _stringr: Simple, consistent wrappers for common string_\n",
      "\n",
      "_operations_ . R package version 1.2.0. Retrieved from http://CRAN.R\n",
      "proje​ct.org/packa​ge=stringr\n",
      "Zhang, S.‐Q., Che, L.‐H., Li, Y., Liang, D., Pang, H., Ślipiński, A., & Zhang,\n",
      "\n",
      "P. (2018). Evolutionary history of Coleoptera revealed by extensive\n",
      "\n",
      "sampling of genes and species. _Nature Communications_, _9_ [, 205. https​](https://doi.org/10.1038/s41467-017-02644-4)\n",
      "\n",
      "[://doi.org/10.1038/s41467-017-02644-4](https://doi.org/10.1038/s41467-017-02644-4)\n",
      "\n",
      "**SUPPORTING INFORMATION**\n",
      "\n",
      "Additional supporting information may be found online in the\n",
      "\n",
      "Supporting Information section at the end of the article.\n",
      "\n",
      "**How to cite this article:** Gustafson GT, Alexander A, Sproul\n",
      "\n",
      "JS, Pflug JM, Maddison DR, Short AEZ. Ultraconserved\n",
      "\n",
      "~~element~~ ~~(UCE)~~ ~~probe~~ ~~set~~ ~~design:~~ ~~Base~~ ~~genome~~ ~~and~~ ~~initial~~\n",
      "\n",
      "design parameters critical for optimization. _Ecol Evol_ .\n",
      "\n",
      "2019;9:6933–6948. [https​://doi.org/10.1002/ece3.5260](https://doi.org/10.1002/ece3.5260)\n",
      "\n",
      "\n",
      "md_text: Received: 8 April 2019 | Revised: 26 April 2019 | Accepted: 28 April 2019 DOI: 10.1002/ece3.5260 O R I G I N A L R E S E A R C H # **Ultraconserved element (UCE) probe set design: Base genome ** **and initial design parameters critical for optimization** **[Grey T. Gustafson](mailto:﻿)** **[1,2]** | **Alana Alexander** **[2,3]** | **John S. Sproul** **[4,5]** | **James M. Pflug** **[4]** | **David R. Maddison** **[4]** | **Andrew E. Z. Short** **[1,2]** 1 Department of Ecology and Evolutionary Biology, University of Kansas, Lawrence, Kansas 2 Biodiversity Institute, University of Kansas, Lawrence, Kansas 3 Department of Anatomy, School of Biomedical Sciences, University of Otago, Dunedin, New Zealand 4 Department of Integrative Biology, Oregon State University, Corvallis, Oregon 5 Department of Biology, University of Rochester, Rochester, New York **Correspondence** Grey T. Gustafson, Department of Ecology and Evolutionary Biology, University of Kansas, Lawrence, Kansas. [Email: gtgustafson@gmail.com](mailto:gtgustafson@gmail.com) **Funding information** National Institutes of Health, Grant/Award Number: 5K12GM063651 **1** | **INTRODUCTION** **Abstract** Targeted capture and enrichment approaches have proven effective for phylogenetic study. Ultraconserved elements (UCEs) in particular have exhibited great utility for phylogenomic analyses, with the software package phyluce being among the most uti- lized pipelines for UCE phylogenomics, including probe design. Despite the success of UCEs, it is becoming increasing apparent that diverse lineages require probe sets tailored to focal taxa in order to improve locus recovery. However, factors affecting probe design and methods for optimizing probe sets to focal taxa remain underex- plored. Here, we use newly available beetle (Coleoptera) genomic resources to inves- tigate factors affecting UCE probe set design using phyluce . In particular, we explore the effects of stringency during initial design steps, as well as base genome choice on resulting probe sets and locus recovery. We found that both base genome choice and initial bait design stringency parameters greatly alter the number of resultant probes included in final probe sets and strongly affect the number of loci detected and re- covered during in silico testing of these probe sets. In addition, we identify attributes of base genomes that correlated with high performance in probe design. Ultimately, we provide a recommended workflow for using P hyluce to design an optimized UCE probe set that will work across a targeted lineage, and use our findings to develop a new, open-source UCE probe set for beetles of the suborder Adephaga. **K E Y W O R D S** bait design, genomics, phylogenetics, phylogenomics, probe design, ultraconserved elements components that have been targeted, ultraconserved elements (UCEs sensu Faircloth et al., 2012) have demonstrated their util- The phylogenomics revolution is underway with targeted capture and enrichment approaches (Mamanova et al., 2010; McCormack, Hird, Zellmer, Carstens, & Brumfield, 2013) proving an effective and improved method for phylogenetic studies relative to multilo- cus Sanger techniques (Blaimer et al., 2015). Among the genomic ity in reconstructing phylogenies across diverse vertebrate [e.g., fishes (Faircloth, Sorenson, Santini, & Alfaro, 2013), birds (Hosner, Tobias, Braun, & Kimball, 2017; McCormack et al., 2013), reptiles (Crawford et al., 2012; Grismer et al., 2016; Streicher & Wiens, 2017), amphibians (Alexander et al., 2017; Streicher et al., 2018), [This is an open access article under the terms of the Creative Commons Attribution License, which permits use, distribution and reproduction in any medium,](http://creativecommons.org/licenses/by/4.0/) provided the original work is properly cited. © 2019 The Authors. _Ecology and Evolution_ published by John Wiley & Sons Ltd. _Ecology and Evolution._ 2019;9:6933-6948. www.ecolevol.org ~~[|]~~ **6933** **6934** GUSTAFSON et al . **|** and mammals (Esselstyn, Oliveros, Swanson, & Faircloth, 2017; McCormack et al., 2012)] and other animal lineages [e.g., cnidarians (Quattrini et al., 2018), arachnids (Starrett et al., 2017), hymenopter- ans (Faircloth, Branstetter, White, & Brady, 2015), and coleopterans (Baca, Alexander, Gustafson, & Short, 2017; Van Dam et al., 2017)], providing resolution for notoriously difficult-to-reconstruct evolu- tionary histories, such as those of placental mammals (Esselstyn et al., 2017) or the rapid radiation of modern birds (McCormack et al., 2013). Additionally, UCEs are phylogenetically informative at mul- tiple timescales (Faircloth et al., 2012): able to reconstruct both deep (Quattrini et al., 2018) and shallow (Manthey, Campillo, Burns, & Moyle, 2016; Musher & Cracraft, 2018; Smith, Harvey, Faircloth, Glenn, & Brumfield, 2014) evolutionary relationships. Furthermore, UCEs allow for the inclusion of museum specimens (Blaimer, Lloyd, Guillory, & Brady, 2016; McCormack, Tsai, & Faircloth, 2016; Van Dam et al., 2017), even those preserved in formalin (Ruane & Austin, 2017), affording improved taxon sampling by removing limitations associated with specimen preservation style. Another advantage to the use of UCEs is the ability to develop probe sets that target thousands of orthologous loci across members of a select organismal group, based on low coverage genomic reads from relatively few exemplar taxa (Faircloth, 2017). While universal probe sets designed to work across larger taxonomic groups such as tetrapods (Faircloth et al., 2012; Sun et al., 2014) have proven suc- cessful for birds (Hosner et al., 2017; Manthey et al., 2016) and mam- mals (Esselstyn et al., 2017), it is becoming increasingly apparent that for other diverse animal lineages probe sets tailored to focal taxa re- sult in considerably improved locus recovery. For example, the first Hymenoptera UCE probe set, designed largely using the genome of a single parasitoid wasp species, targeted a total of 1,500 UCE loci, but captured decreasing numbers of loci with increasing phy- logenetic distance from this taxon, with one highly divergent study species within Hymenoptera having as few as 341 loci enriched by the probe set (Faircloth et al., 2015). Thus, Branstetter, Longino, Ward, and Faircloth (2017) subsequently developed an expanded Hymenoptera UCE probe set tailored for use in ants, which success- fully increased locus capture for target ant taxa, but similarly showed decreased capture rates for more distantly related ants, bees, and wasps. In another example, Faircloth (2017) designed a Coleoptera UCE probe set, primarily using genomes from the beetle suborder Polyphaga. This probe set, targeting 1,172 UCE loci, only recovered 305 loci in 50% of the taxa studied in a phylogenomic analysis of the distantly related beetle suborder Adephaga (Baca et al., 2017). A second investigation by Van Dam et al., 2017 utilized this same probe set for a phylogenomic analysis of weevils (family Curculionidae), members of the same suborder of beetles used during the probe design (Faircloth, 2017). Although capturing considerably more loci (537 loci at 50% data matrix completeness) than the prior study, it still resulted in significantly fewer UCE loci for phylogenetic analysis (368 in the 70% complete data matrix) (Van Dam et al., 2017) than the probe set was designed to provide. This is likely due to the sin- gle representative weevil genome used ( _Dendroctonus ponderosae_ ) in the design of the Coleoptera UCE probe set by Faircloth (2017) being only distantly related to those in the Van Dam et al. (2017) study. In spite of the apparent need for probe sets tailored to focal taxa, factors affecting probe design and methods for optimizing probe sets to focal taxa remain largely underexplored (Branstetter et al., 2017; Glenn & Faircloth, 2016). The need for investigation of fac- tors affecting probe design is made even greater by the increasing number of genomic resources becoming readily available for diverse organisms, and the development of new software enabling probe design. In this study, we investigate factors affecting probe set design of UCE loci. In particular, we investigate the effect of base genome choice and initial bait design parameters on the resulting probe set, and the number of UCE loci recovered during in silico tests. Currently, there are relatively few thorough workflows or pipelines and associ- ated programs for identifying conserved loci and designing baits tar- geting them (Campana, 2018; Faircloth, 2017; Johnson et al., 2016; Mayer et al., 2016). We opted to use one of the most widely utilized software packages for UCE phylogenomics, phyluce (Faircloth, 2016), and to follow the workflow outlined in detail by Faircloth (2017) for using phyluce to identify and design baits targeting UCE loci. Based on our results, we provide recommendations and discuss consider- ations for base genome choice and parameter selections to optimize the design of UCE probe sets using phyluce . Our findings are then applied to develop a new UCE probe set, optimized for beetles of the suborder Adephaga, using newly available genomic resources and incorporating previously published Coleoptera UCE loci. **2** | **MATERIALS AND METHODS** **2.1** | **Genomic resources and taxon information** Genomic assemblies used in this study were generated using Illumina reads obtained from seven beetle species from the subor- der Adephaga (Table 1) (Pflug, Holmes, Johnston, and Maddison, in prep). Full details are given in Pflug, Holmes, Johnston, and Maddison (in prep); in brief, poor quality reads and adapter sequences were trimmed from Illumina reads using the 'Trim Sequences' tool in clc genomics workbench version 9.5.3 (CLC Bio, referred to below as CLC GW), with the trim quality score limit set to 0.5, allowing for a maxi- mum of two ambiguities per read, and searching on both strands to remove adapters, retaining broken pairs. De novo assemblies of paired, trimmed reads in CLC GW were generated using automatic word and bubble size, with the minimum contig length set to 200 bases. Genomic assembly metrics for the genomes used in this study are given in Table 1 (methods for obtaining these metrics are pro- vided in Section 2.9). Our study species come from four different families within the beetle suborder Adephaga (Table 1). The suborder is diverse, con- sisting of ~45,000 species (Ślipiński, Leschen, & Lawrence, 2011) exhibiting a wide range of ecologies with both terrestrial and aquatic members. These ecological differences have been used to divide the suborder into two groups: the Hydradephaga containing aquatic GUSTAFSON et al . **6935** **|** TA B LE 1 Study taxa and associated genomic assembly information. OSAC, Oregon State Arthropod Collection located at Oregon State University **Voucher #** **Estimated** **Species** **Depository** **Family** **read depth** **N50** **L50** **N90** **L90** **GC%** **BUSCO C%** _Bembidion_ _haplogonum_ Chaudoir DNA2544 Carabidae 58× 591 211,693 231 972,875 30.6 59.3 OSAC _Chlaenius sericeus_ DNA4821 Carabidae 166× 1,765 41,747 408 190,237 29 78.7 (Forster) OSAC _Lionepha_ DNA3782 Carabidae 8× 5,477 9,102 518 47,974 28.6 68.5 'Waterfalls' OSAC _Pterostichus mela-_ DNA3787 Carabidae 33× 346 216,023 137 864,146 48.8 30.7 _narius_ (Illiger) OSAC _Omoglymmius_ _hamatus_ (LeConte) DNA3783 Rhysodidae 13× 370 234,264 143 963,963 50 7.3 OSAC _Trachypachus gibb-_ DNA3786 Trachypachidae 85× 1,758 34,686 213 233,955 32.8 45.8 _sii_ LeConte OSAC _Amphizoa insolens_ DNA3784 Amphizoidae 8× 364 160,074 141 668,749 32.3 34.7 LeConte OSAC members and the Geadephaga for terrestrial species. There has yet to be a comprehensive phylogenetic analysis of the suborder using molecular data, and recent studies with limited taxon sampling have either found these two groups reciprocally monophyletic (McKenna et al., 2015), or with Geadephaga sister to all aquatic families with the exception of the whirligig (the family Gyrinidae) beetles (Baca et al., 2017), or sister to both Gyrinidae and Haliplidae (Zhang et al., 2018). Our study taxa consist largely of families within the Geadephaga with a single hydradephagan representative, _Amphizoa_, belonging to one of the aquatic families consistently supported as sister to the Geadephaga (Baca et al., 2017; McKenna et al., 2015). The lack of a comprehensive time-calibrated phylogeny of Adephaga prevents precise estimates of the evolutionary time and phylogenetic distance separating the species in our study. However, the maximum evolutionary time separating our study species is approximately 250 million years, which is similar in age to the esti- mated split between turtles and archosaurs (Chiari, Cahais, Galtier, & Delsuc, 2012; Shen, Liang, Wen, & Zhang, 2011; Toussaint et al., 2017). **2.2** | **Overview** Using the above genomic resources and the software package phy luce, we developed multiple UCE probe sets while systematically al- tering key aspects of the design strategy. Throughout this study, we use the term 'bait' to refer to the temporary bait sets targeting theo- retical conserved loci (an intermediate stage in probe design), with 'probe' referring to the final products of probe design (i.e., the set of RNA probes targeting UCE loci that would actually be synthesized, subject to in silico testing). First, we explored the effects of chang- ing which taxon served as the base genome during probe design, a critical decision, as the genomic reads of all other taxa are mapped to this genome in order to identify conserved loci. The second round of experiments focused on the effects of changing the stringency pa- rameters used during temporary bait design in phyluce . Our general workflow for using phyluce to develop probe sets, and the specific methods for our experimentation are provided below. We evaluated the success of the different probe sets by tracking the number of loci identified and/or recovered during the different stages of probe design within phyluce, the number of probes developed, and finally by comparison of the UCE loci recovered during in silico tests. After producing optimized probes, we filtered the probe set to a subset of probes targeting only those loci found within all seven taxa that appear paralogy free. We then used baitstools as an additional check of our final subset of optimized probes in order to identify physical attributes that could affect their synthesis and in vitro utility. Finally, we investigated attributes of base genomes (e.g., assembly metrics and genetic distance from other study taxa) potentially correlated with improved probe design. **2.3** | **Phyluce workflow overview** A detailed workflow for using phyluce to identify UCE loci and to design probes targeting these loci is described by Faircloth (2017). The general phyluce work flow followed for our experiments is given in Figure 1. The first step in our experiments involved selection of a base genome (Figure 1, step 1) against which genomic reads of the remaining taxa were aligned (Figure 1, step 2). Following removal of duplicate reads, the next step (Figure 1, step 3) iden- tified _putative loci_ : highly conserved loci shared among taxa (see Supporting Information for corresponding phyluce terminology and commands). _Putative loci_ are those present in the base genome as well as a specified number of additional taxa. The stringency at this step is varied (Figure 1b, step 3) by changing the number of taxa **6936** GUSTAFSON et al . **|** FI G U R E 1 General phyluce workflow for the two sets of probe design experiments conducted using phyluce . Workflow proceeds from left to right with the numbers above text blocks indicating steps referenced in text. (a) depicts the base genome choice experiments described in Section 2.4. (b) depicts the temporary bait design stringency experiments described in Section 2.5. A key for the figure is given in the bottom left corner. 'Baits' refer to the temporary bait sets targeting theoretical conserved loci, with 'probes' referring to the final products of probe design (i.e., the set of RNA probes targeting UCE loci that would actually be synthesized). Areas highlighted in red show targeted loci/baits that were found across the specified number of taxa. Results depicting the _putative loci_ (highly conserved loci shared among taxa), _candidate loci_ (loci where temporary baits align across genomes of multiple taxa), and the _recovered loci_ (UCE loci recovered during the in silico test) are shown in Figures 2-6. The red boxes in the final probe set section indicate the combination of parameters leading to the largest number of in silico recovered probes for each set of experiments the locus must be found within, ranging from _putative loci_ found within the base genome and at least one other taxon ('+1', the least stringent, designated as 'at least one other taxon' on Figure 1b), to those found within the base genome and all remaining taxa ('+6', the most stringent, designated as 'all 6 taxa' on Figure 1b, that is, _putative loci_ are defined as those found in the base genome and all six additional taxa). Temporary baits were then developed target- ing these _putative loci_ . These temporary baits were then aligned back to the genomic assemblies of all study taxa (Figure 1, step 4). This allowed the identification of _candidate loci_ : loci for which tem- porary baits consistently aligned across the genomes of multiple taxa (Figure 1, step 5). By varying the number of taxa within which the candidate loci must be found within, an additional stringency filter can be applied. In the example shown in step 5 of Figure 1, the candidate loci chosen were those for which the temporary baits matched regions in all seven taxa (the base genome plus the other six taxa)-the most stringent option. The final probe set was then created by designing probes customized specifically for each taxon to allow enrichment of the same locus, across the specified num- ber of taxa, using the previously identified _candidate loci_ . Finally, an in silico test (Figure 1, step 6) of the resulting probe set was performed using the genomic assemblies of all study taxa in order to assess the number of UCE loci recovered by the final probe set in semirealistic conditions. The purpose of using _recovered loci_ (loci recovered in the in silico test) to evaluate the choices involved in probe set design rather than the final probe sets themselves was in part to emulate some of the stochastic variation likely to be seen attempting to recover these loci in vitro. GUSTAFSON et al . **6937** **|** FI G U R E 2 Number of _putative loci_(highly conserved loci shared among taxa) recovered using different base genomes during probe design ~~FI~~ ~~G~~ ~~U~~ ~~R~~ ~~E~~ ~~3~~ ~~Number~~ ~~of~~ ~~_candidate_~~ ~~_loci_~~ ~~(loci~~ ~~where~~ ~~temporary~~ ~~baits~~ ~~align~~ ~~across~~ ~~genomes~~ ~~of~~ ~~multiple~~ ~~taxa)~~ ~~shared~~ ~~among~~ ~~different~~ ~~numbers~~ of taxa using different base genomes during probe design FI G U R E 4 Number of _recovered loci_ (UCE loci recovered during the in silico test) shared among different taxa for probe sets designed using different base genomes. All categories provide the sum total of unique loci recovered for that base genome across all taxa **6938** GUSTAFSON et al . **|** We applied this workflow to two separate experiments: first, evaluating the use of different taxa as the base genome (Figure 1a, see Section 2.4 for additional methods), and second, varying differ- ent parameters for temporary probe design (Figure 1b, see Section 2.5 for additional methods). **2.4** | **Testing the effect of base genome choice on** **probe design and locus recovery** To test the influence of base genome choice on probe design and locus recovery, we performed the following steps in phyluce (Faircloth, 2016) (as outlined in the phyluce Tutorial IV, available at [http://phyluce.readthedocs.io/en/latest/tutorial-four.html). First, a](http://phyluce.readthedocs.io/en/latest/tutorial-four.html) single taxon was selected to serve as the base genome (Faircloth, 2017) (Figure 1a, step 1) and the genomic reads of the remaining taxa were aligned to this genome using stampy (Lunter & Goodson, 2011) (Figure 1a, step 2) under the following mapping parameters: --maxbasequal 93, --substitutionrate = 0.05, and insertsize = 400. From the resulting alignment, we identified _putative loci_ shared among the base taxon and the remaining study taxa. In this base genome choice experiment, we then proceeded with the most strin- gent approach to temporary bait design, targeting only those _puta-_ _tive loci_ shared across the base genome as well as all six other taxa (Figure 1a, step 3). The influence of modifying the stringency of this step is evaluated by the experiment presented in the next section. We then determined the number of _candidate loci_ (where temporary baits consistently aligned across multiple genomes; Figure 1a, step 4). Following this, we took a stringent approach to designing the final probe set by targeting only _candidate loci_ found within all seven of the study taxa (the base genome plus all six other taxa) (Figure 1a, step 5). After design of the final probe set, we performed an in silico test (Figure 1a, step 6) of the resulting probes using the genomic as- semblies of all the study taxa to determine the number of _recovered_ _loci_ by the probe set. Finally, the UCE loci extracted from each species during the in silico test were aligned using mafft (Katoh & Standley, 2013), am- biguous regions removed via gblocks (Castresana, 2000; Talavera & Castresana, 2007), and a 75% complete matrix generated in phyluce using the command 'phyluce_align_get_only_loci_with_min_taxa.' A maximum-likelihood phylogenetic analysis was performed on this matrix using RA x ML 8.8.1 (Stamatakis, 2014) and the commands -m GTRGAMMA, -# autoMRE -n BOOT -T 24, to investigate any vari- ance among the phylogenetic reconstructions obtained using loci recovered from probe sets designed with different base genomes. These steps were repeated changing only the taxon that served as the base genome (Figure 1a, step 1). Following comparison of probe sets designed using different base genomes and their success at locus recovery, we investigated attributes of base genomes potentially correlated with their probe design success. These attributes included genomic assembly metrics such as average estimated read depth, N50, L50, N90, L90, Guanine and Cytosine (GC) percent content, and BUSCO completeness per- centage (Simão, Waterhouse, Ioannidis, Kriventseva, & Zdobnov, 2015), as well as average relative genetic distance to the other taxa included in this study. These methods are presented in Sections 2.8 and 2.9. **2.5** | **Testing temporary bait design stringency's** **influence on probe design and locus recovery** Following investigation into how base genome choice influences probe design and locus recovery, we experimented with relaxing temporary bait design stringency. This involved modifying the step in the phyluce workflow in which temporary baits are con- structed based on the identification of _putative loci_ shared among the aligned genomic data (Figure 1b, step 3). We incrementally decreased the number of taxa _putative loci_ were required to be identified within when designing temporary baits (Figure 1b, step 3). We began first with the most stringent requirement for tem- porary bait design by targeting _putative loci_ found across the base genome and all six other taxa and ended by targeting those found across the base genome and at least one other taxon, the least stringent temporary bait design. We then proceeded to follow the same steps as in Section 2.4 through to the in silico test. It should be emphasized that we maintained the stringent requirement of designing the final probe set based only on _candidate loci_ present within all seven study taxa (Figure 1b, step 5). We selected the taxon _Lionepha_ to serve as the base genome throughout these tests as it performed moderately well during the base genome choice experiments, being neither the best, nor worst perform- ing genome. After establishing that _Pterostichus_ appeared to be the optimum base genome, we also repeated the temporary bait stringency experiment on this base genome, to ensure the results were robust to base genome choice. **2.6** | **Comparison of resulting probe sets** To establish which UCE loci were shared between the probe sets de- signed using different base genomes in Section 2.4, we used custom bash and R-scripts to BLAST (at 95% and 99% similarity) between the monolithic files (containing all UCE loci identified across all taxa for each base genome) produced by phyluce (see Alexander 2018a under Data Accessibility for step-by-step commands). Loci with sin- gle BLAST matches across at least two base genomes for at least one taxon were characterized as: 'good' (UCE loci that appeared to be single copy across all taxa across all base genomes), 'problematic within taxa' (evidence of potential paralogy within one or more taxa where loci identified within a taxon using one base genome matched to multiple loci within that same taxon identified using a different base genome), and 'problematic between taxa' (potential lineage- specific paralogy: a UCE locus that appeared to be single copy across all base genomes within one taxon appeared to be present in mul- tiple copies in at least one other taxon). We used the 'good loci' to identify the base genomes that had the greatest success recovering loci across the seven taxa and that gave the longest alignment in each taxon. GUSTAFSON et al . **6939** **|** FI G U R E 5 Number of _candidate_ _loci_ (loci where temporary baits align across genomes of multiple taxa) found across increasing numbers of taxa for each temporary bait design stringency (requiring loci to be found in the base genome +1 other taxon through to the base genome +6 taxa) FI G U R E 6 Number of _recovered_ _loci_ (UCE loci recovered during the in silico test) in different taxa from probe sets designed using _Lionepha_ and _Pterostichus_ as base genomes and varying the temporary bait design stringency requirements (requiring loci to be found in the base genome +1 other taxon through to the base genome +6 taxa). The all category provides the sum total of unique loci recovered for that base genome across all taxa **2.7** | **In vitro attributes of final subset of** **optimized probes** We examined aspects of the final subset of optimized probes de- signed during this study that could potentially affect synthesis and in vitro utility. We gathered relevant information on the final opti- mized subset of probes designed during this study relevant to in vitro performance including GC% content, melting temperature (Tm), and sequence complexity, using baitstools (Campana, 2018) and the 'checkbaits' command under the following parameters: -L 120 -c -n 25 -x 70 -q 80 -z 120 -w. **2.8** | **Genome assembly metrics** We estimated depth of coverage for each genome by mapping reads to the subset of 'good' UCE loci (identified using the methods in Section 2.6) that were also found in all seven taxa. The specific ref- erence sequences used for each locus depended on how many base genomes were able to characterize that UCE locus. If a UCE locus was detected only with one base genome, then the UCE locus sequences from this base genome were used as references. If a given locus was recovered using more than one base, then the base genome that gave the longest average alignment across all taxa was selected to provide the reference sequences. We extracted the UCE locus sequences for each taxon using custom bash scripts (see Alexander 2018a under Data Accessibility for step-by-step commands). Then, we used GATK v 3.8/4.0, picard, bwa, MAFFT, R, samtools, stringr with custom scripts (see Alexander 2018d under Data Accessibility for step-by-step com- mands) to calculate depth of coverage on a per-locus basis. Additional genomic assembly metrics such as N50, L50, N90, L90, Guanine and Cytosine (GC) percent content, and completeness of conserved protein-coding genes were generated using CLC GW v9.5.3 (available from https://www.qiagenbioinformatics.com/), BBM ap (Bushnell, 2014) using the stats.sh tool, and BUSCO (Simão et al., 2015). **2.9** | **Estimating relatedness of study taxa** We inferred the relative divergence between the taxa in our study for several markers commonly used in phylogenetic investigations **6940** GUSTAFSON et al . **|** of Coleoptera: 18S, 28S, ArgK, CAD2, CAD4, _wg_, and COI. These markers were obtained either for species included in this study, or for their congeners. Some of these sequences were obtained from the base genomes (CAD2 for _Trachypachus_ and _Lionepha_, and COI of all taxa except _Bembidion haplogonum_ ). The rest were obtained from the literature (Gomez, Will, & Maddison, 2016; Kanda, Pflug, Sproul, Dasenko, & Maddison, 2015; Maddison, 2012; Maddison Baker, & Ober, 1999a, 1999b; Maddison et al., 2009; Maddison & Swanson, 2010; McKenna et al., 2015; Ober, 2002; Sproul & Maddison, 2017), or were newly obtained from Sanger sequencing of PCR products, as specified in Supplemental Information Table S1. Methods for obtain- ing the new sequences from PCR/Sanger sequencing are given in Maddison (2012). New sequences have been submitted to GenBank [with accession numbers MK838494 to MK838511.](info:ddbj-embl-genbank/MK838494) We used Geneious [®] 11.0.5 to calculate average pairwise genetic distances, average proportion of differences (using both Tamura- Nei and HKY corrections), and patristic distances from a neighbor- joining tree using the Tamura-Nei correction. We ranked taxa from the smallest average genetic distance across markers and methods to the largest average distance. Additionally, we attempted to extract 50 nuclear protein- coding markers developed for Coleoptera phylogenetics from genomewide surveys (Che et al., 2017) directly from our study genomes using bait sequences from related taxa available on GenBank via the Zhang et al. (2018) study (Appendix S1). We ex- tracted these nuclear protein-coding markers from our genomic assemblies using bwa (Li & Durbin, 2009, 2010), data.table (Dowle & Srinivasan, 2017), GATK (McKenna et al., 2010), MAFFT (Katoh [& Standley, 2013), picard (available from: http://broadinstitute.](http://broadinstitute.github.io/picard/) [github.io/picard/), R (R Core Team, 2017), samtools (Li et al., 2009),](http://broadinstitute.github.io/picard/) and stringr (Wickham, 2017) via custom scripts (see Alexander 2018b under Data Availability for step-by-step commands). This was done in an attempt to better approximate divergences of the entire nuclear genomes among our taxa. We then generated relative genetic distance measures among these nuclear protein- coding markers for our study taxa using EMBOSS distmat (Rice, Longden, & Bleasby, 2000) with a Tamura correction and R via custom scripts (see Alexander 2018c under Data Availability for step-by-step commands). Finally, we compared these findings with those of the more commonly used phylogenetic loci represented in our six gene dataset, run through the same distance measures pipeline. **3** | **RESULTS** **3.1** | **The effect of base genome choice on probe** **design and locus recovery** The effect of base genome choice was evident at each stage of the workflow (Figure 1a) from the number of _putative loci_ identi- fied (Figure 2, Data S1), to the number of _candidate loci_ (Figure 3, Data S2), and finally the number of _recovered loci_ during the in silico test (Figure 4, Data S3). There were also consistent trends in the performance of taxa selected to serve as the base genome. For ex- ample, the genomes of _Amphizoa_ and _Omoglymmius_ resulted in the fewest number of _putative loci_ (Figure 2), _candidate loci_ (Figure 3), and _recovered loci_ (Figure 4). These base genomes also resulted in the fewest number of probes designed, as well as the shortest over- all sequence lengths recovered across taxa (see Data S2 and S3). The taxa _Bembidion_, _Lionepha_, and _Trachypachus_ performed considerably better as base genomes, resulting in more _puta-_ _tive loci_ (Figure 2), _candidate loci_ (Figure 3), and _recovered loci_ (Figure 4), as well as the design of tens of thousands more probes (Data S2). _Chlaenius_ performed even better as the base genome and was the base genome that resulted in the largest number of _putative loci_ shared among all six of the other taxa (Figure 2). However, _Chlaenius_ did not result in the most _candidate loci_ found across all seven taxa (Figure 3) nor _recovered loci_ during the in silico test (including _Chlaenius_ itself; Figure 4). This suggests the number of _putative loci_, while tracking general trends in a taxon's performance as the base genome, is not an accurate predictor of the best performing base genome for downstream locus detection and recovery. The best performing base genome was _Pterostichus_ . Using _Pterostichus_ as the base genome resulted in the largest number of _candidate loci_ (Figure 3) found across all seven taxa, and the larg- est number of _recovered loci_ (including the most _recovered loci_ for _Pterostichus_ itself; Figure 4). The use of _Pterostichus_ as a base ge- nome resulted in baits targeting several hundred (i.e., 8%) more loci than the second-best performing genome, _Chlaenius_, concordantly reflected in the design of several thousand (i.e., ~7%) more probes (Data S2). _Pterostichus_ resulted in several hundred (i.e., 6%) more _re-_ _covered loci_ during the in silico test (Figure 4) than _Chlaenius_, and the resulting concatenated final alignment across taxa was several hundred thousand base pairs (i.e., ~6%) longer than that of _Chlaenius_ (Data S3). Despite these differences in performance at recovering and detecting UCE loci, base genome choice did not affect the topol- ogy, nor support values of the final maximum-likelihood phylogenies produced (Figure S1). **3.2** | **Stringency of temporary bait design's effect on** **probe design and locus recovery** The temporary bait design stringency experiments revealed less stringent parameters for bait design (i.e., requiring sharing of _puta-_ _tive loci_ across fewer taxa) greatly improved the number of _candidate_ _loci_ identified (Figure 5, Data S4), as well as the number of _recovered_ _loci_ during the in silico test (Figure 6, Data S5). **3.3** | **Optimized probe design** The base genome choice experiments identified _Pterostichus_ as the optimal base genome. The results from the stringency in temporary bait design experiments showed decreasing the number of taxa _putative loci_ were required to be shared across increased both the number of final probes and the number of _recovered loci_ during the GUSTAFSON et al . **6941** **|** FI G U R E 7 Number of 'good loci' (UCE loci that appeared to be single copy across all taxa across all base genomes) recovered in different taxa from probe sets designed using different base genomes in silico test (Figure 6). Therefore, to design an initial optimized UCE probe set, we selected _Pterostichus_ to serve as the base genome and designed temporary probes under the least stringent requirements, targeting _putative loci_ found within the base genome and at least one other taxon. **3.4** | **Comparison of probe sets based on loci found** **across different probe designs** Using the _recovered loci_ (i.e., those identified during the in silico test) from the seven base genome choice experiments, as well as the final optimized _Pterostichus_ +1 probe design, we identified 8,159 loci through BLAST that matched at 99% similarity across more than one base genome for at least one taxon (95% similar- ity results revealed very similar patterns, so these results are pre- sented only in Data S6 and S7). Of these, 3,501 'good loci' showed no sign of paralogy within or among taxa across the various base genomes used. Comparisons of base genome performance based on recovery of these 'good loci' largely reflected the previously presented results. When _Pterostichus_ served as the base genome with temporary baits designed from _putative loci_ found in just one other taxon ( _Pterostichus_ +1 taxon), the largest number of 'good loci' were recovered for each taxon in comparison with all other base genomes (Figure 7, Data S8). This pattern persisted even when restricting the comparison to the 2,702 'good loci' detected in all seven taxa and thus does not seem to be an artifact of over- representation of loci found only in _Pterostichus_ but absent in other lineages (Data S9). We also compared the impact of base genome selection on sequence length of the 'good loci' found across all seven study taxa. This could be an important metric, as phyloge- netically informative variation in UCEs tends to be greater in the DNA sequences flanking the core conserved element (Faircloth et al., 2012; Van Dam et al., 2017). Longer sequence lengths are therefore more likely to recover these more variable regions. For each 'good locus' found in all seven taxa, we recorded the number of taxa where a base genome gave the longest or equally long- est sequence lengths across all base genomes. These results again showed that _Pterostichus_ (specifically _Pterostichus_ +1 taxon) appeared to be the optimal base genome/stringency choice: It resulted in a larger number of loci where it gave the longest align- ment for more taxa than the other base genomes (Data S9). **3.5** | **Final optimized probe set** The optimized probe design ( _Pterostichus_ +1 taxon) resulted in a probe set consisting of 145,599 probes, targeting 11,162 UCE loci. However, synthesizing and enriching this large number of probes and loci would be cost prohibitive for phylogenomic studies with large numbers of taxa. We therefore needed a way to naturally limit the number of probes included in the final probe set. The initial optimized probe set was filtered to a final set of probes targeting the 2,702 'good loci' detected in all seven taxa discussed in Section 3.4. This was done using custom scripts (see Alexander 2018d under Data Accessibility for step-by-step commands) and was fur- ther limited to only those loci covered by the _Ptersotichus_ +1 probes (i.e., the optimized probe design), as 858 loci were derived under other probe design parameters (i.e., different base genomes and temporary bait stringency). Therefore, the final subset of optimized probes targets 2,643 loci that appear paralogy free and are present within all seven study taxa. As part of the final probe set, we also included probes for 305 loci from the previous Coleoptera 1.1Kv1 probe set (Faircloth, 2017) that were successfully recovered during the Baca et al. (2017) study (see Section 4 section below). The final optimized probe set targets a total of 2,948 UCE loci. In vitro attributes of the final subset of optimized probes (Appendix S2) include a GC% content of all probes between 20% and 70%, with a median of 40.83% and mean of 42%; and a Tm between 80°C and 120°C, with a median of 96.83°C and a mean of 97.69°C. The sequence complexity measure of the probes was very high with a bait linguistic complexity mean and median of **6942** GUSTAFSON et al . **|** FI G U R E 8 Attributes of study taxa and their genomic assemblies in relation to their performance as base genome. (a) Relationship between average pairwise genetic distance based on six gene fragments regularly use for phylogenetics and total number of UCE loci recovered during in silico test. (b) Relationship between contiguity measure N90 of genomic assembly and total number of UCE loci recovered during in silico test. (c) Relationship between BUSCO %C value of genomic assembly and total number of UCE loci recovered during in silico test 0.99. The GC% content is well within the optimal range for ge- nomic probes (Cruz-Dávalos et al., 2017; Tewhey et al., 2009) and high-sequence complexity and Tms increase the likelihood of target specificity. **3.6** | **Assembly metrics of genomic resources used** We used the 2,702 'good' UCE loci found across all seven taxa to estimate coverage depth for the genomes utilized in this study. GUSTAFSON et al . **6943** **|** S10), with BUSCO scores (Figure 8c, Table 1, Data S10) supporting the _Chlaenius_ genome as being the most complete. Interestingly, _Pterostichus_ in addition to _Omoglymmius_ and _Amphizoa_ had the least complete genomic assemblies and lowest assembly metrics (Figure 8c, Table 1), yet these taxa represent both the best and worst performing base genomes, respectively. Therefore, genomic assem- bly metrics do not appear to correlate well with base genome perfor- mance (Figure 8b,c). Similarly, GC% content of the genome does not appear correlated with suitability as a base genome, in that the best performing base genome, _Pterostichus_, had a comparable GC per- centage (nearly 50%, Table 1) to that of one of the worst performing genomes, _Omoglymmius_, while _Amphizoa_ (another poorly performing genome) had a GC percentage similar to those of better performing genomes (around 30%, Table 1). **3.7** | **Pairwise genetic distances between taxa** Pairwise genetic distance rankings showed _Pterostichus_ as having the smallest genetic distance from all other taxa in the six gene frag- ments commonly used in phylogenetic studies (28S, COI, CAD2, CAD4, _wg_, ArgK), with _Amphizoa_ and _Omoglymmius_ the largest ge- netic distances (Figure 8a, Data S11-S13 for average-, raw-, and standardized pairwise genetic distance rankings). We found average genetic distance ranking based on the six gene dataset to be negatively correlated with a taxon's performance as the base genome as assessed by the total number of UCE loci recovered during in silico testing (Figure 8a), and reflected general trends in genome performance tracked elsewhere (Figures 3-5). We attempted to validate the six gene genetic distance measure FI G U R E 9 Proposed workflow for designing an optimized universal UCE probe set using P hyluce for taxa without a well-annotated genome available Genomic read coverage did not appear to correlate with base ge- nome performance (Table 1). _Pterostichus_, the best performing base genome, showed intermediate coverage depth (33×) relative to other taxa (Table 1). The worst performing base genomes, _Omoglymmius_ and _Amphizoa_, while estimated as having relatively low cover- age depths (13× and 8× respectively), were comparable to that of _Lionepha_ (8×) which performed considerably better as a base ge- nome (Figure 4). Finally, _Chlaenius_ had the highest estimated cover- age depth (166×), yet did not outperform _Pterostichus_ which had a much lower coverage depth (33×). Additional genomic assembly metrics such as N50, L50, N90, L90 identified _Chlaenius_, _Lionepha_, and _Trachypachus_ as the best genomic assemblies in terms of contiguity (Figure 8b, Table 1, Data results with a larger data set by extracting 50 different nuclear pro- tein-coding loci directly from genomic assemblies of our study taxa (Appendix S1 and Section 2.9). However, this approach proved inef- fective due to unevenness in genome assembly quality of our study taxa (Table 1) which resulted in an excess of missing or low-quality data for taxa with assemblies having a low average depth of cov- erage (e.g., _Lionepha_, _Omoglymmius_, and _Amphizoa_ ). We determined that low coverage and missing data affected data quality for these taxa based on an evident correlation between low sequencing cov- erage and large genetic distance (Data S14-S19), as well as a maxi- mum-likelihood phylogenetic analysis of the data set using IQTree (Nguyen, Schmidt, Haeseler, & Minh, 2015), which failed to recover well-established sister relationships as explained in more detail in Data S20. **4** | **DISCUSSION** Our study using adephagan beetles as an example demonstrates that altering the base genome used during probe design can greatly alter the number of resultant probes included in the final probe set, and more importantly, strongly affects the number of loci detected and recovered during in silico tests. The base genome that recovered the largest number of in silico loci, _Pterostichus_, **6944** GUSTAFSON et al . **|** also performed better at recovering the longest sequences across the seven taxa included in this study. As variable regions flank- ing core UCEs contain phylogenetically informative sequence data (Faircloth et al., 2012; Van Dam et al., 2017), the ability to recover longer sequences for each locus is likely particularly important for their use in phylogenetic analysis. In addition, we demonstrated that being less stringent during the temporary bait design stage, by focusing on _putative loci_ shared across the base genome with as few as one other taxon, resulted in a significantly better final probe set design that detected and recovered more UCE loci dur- ing in silico testing. **4.1** | **Considerations for selecting a base genome for** **probe design** Although we expected at the outset that higher quality base ge- nomes would result in better final probe sets, genomic assembly metrics such as measures of completeness, average read depth, con- tiguity, and GC content did not appear to correlate well with base ge- nome performance (Figure 8, Table 1). The best performing genome for universal adephagan bait design, _Pterostichus_, had relatively low genomic assembly metrics compared to most other taxa in the study (Table 1). This pattern suggests that genome assembly quality may not be a critical criterion in selecting a base genome. In reporting this finding, we acknowledge that it is possible all our assemblies were above a critical coverage threshold, below which assembly metrics may begin to show stronger correlation with probe design outcomes. Similarly, all our assemblies are limited in quality due to being gener- ated using short-read Illumina data, and it is possible that assemblies of much higher quality could correlate with greater probe design success. Our results are a useful reference point in that they were generated using a type (e.g., Illumina reads) and quantity (e.g., tens to hundreds of millions of reads) of sequence data that are representa- tive of increasing numbers of unpublished genomes, but the field will benefit from further exploring how assembly quality affects probe design outcomes. When introducing the workflow for using phyluce to identify and design baits targeting UCE loci, Faircloth (2017) recommends that selection of an appropriate base genome not be focused on ingroup versus outgroup status, but rather annotation and genomic assembly, in order to aid later in selection of probes targeting loci with desirable properties for phylogenetic inference. This is cer- tainly a good recommendation for groups that have well-anno- tated genomic assemblies available; however, most invertebrate groups lack such genomic resources, including our study taxa. Furthermore, one of the most powerful aspects of phyluce is its ability to identify and design probe sets for such nonmodel organ- isms, without the necessity of a complete and annotated genome. Our results demonstrated that UCE probe design can be success- fully optimized even while using a base genome whose genomic assembly is of lower coverage and limited quality as phyluce targets conserved regions that are present and shared among the study taxa, rather than searching for a priori known regions to target. We provide a framework for conducting base genome experiments as outlined above (Figure 1) for selecting a base genome in cases where maximizing the number of loci in the probe set is critical. As a less involved alternative to conducting base genome experiments, we provide evidence from a six gene data set (generated through Sanger sequencing) that selecting the taxon with the smallest av- erage genetic distance from the suite of taxa included in probe de- sign to serve as the base genome can correlate with optimal probe design outcomes (Figure 8a). However, our conclusion that genetic distance patterns can guide base genome selection would bene- fit from further testing given that low sequencing coverage from a subset of our assemblies prevented our validation of this finding with a larger data set (Data S14-S20). **4.2** | **Why does a 'relaxed' approach at the** **temporary bait design stage result in a better** **probe set?** Perhaps most surprising among the results of our experiments re- garding optimal probe set design are those pertaining to temporary bait design stringency. We anticipated a diminishing return in the number of _candidate loci_ as the number of taxa _putative loci_ were required to share decreased. This is because we expected _putative_ _loci_ initially identified as being shared among the base and only one, or a few other taxa, would be unlikely to also be present or detect- able in all of the remaining taxa, thus preventing their inclusion as _candidate loci_ (the step where we required the baits to target loci found in all seven taxa). However, we did not find this to be the case. Instead, we found that designing temporary baits based on _putative_ _loci_ shared among the base genome and as few as one other taxon resulted in significantly better probe design, detecting and recov- ering more UCE loci, as well as in a better performing probe set. Why this would be the case is not entirely clear. Faircloth, in his in- formal discussion within the phyluce [Tutorial IV (available at: http://](http://phyluce.readthedocs.io/en/latest/tutorial-four.html) [phyluce.readthedocs.io/en/latest/tutorial-four.html), suggests loci](http://phyluce.readthedocs.io/en/latest/tutorial-four.html) that are actually present can randomly fail to be detected by phy luce . Perhaps having a temporary bait design protocol with relaxed stringency enables the discovery of such 'cryptic' loci shared among taxa. In this case, relaxed temporary bait design would be analogous to casting a wider net which captures both the 'cryptic' and initially detected loci, which, when narrowed down later by more stringent parameters applied during final probe design (such as only targeting _candidate loci_ present in all study taxa), results in more loci detected and recovered by the final probe set. It is also possible that this result is an artifact of the two differ- ent read mapping programs used by the phyluce pipeline (Faircloth, 2017; Faircloth pers. comm.). Genomic reads are initially aligned to the base genome (Figure 1, step 2) using stampy (Lunter & Goodson, 2011), which has different parameters than that of the LASTZ (Harris, 2007) alignment used by phyluce when aligning temporary baits back to entire genomic assemblies (Figure 1, step 4). The more permissive LASTZ alignment could therefore potentially be able to detect loci ignored initially by stampy (Faircloth pers. comm.). GUSTAFSON et al . **6945** **|** **4.3** | **In vitro considerations** While we focused here on the optimization of probes based on in silico testing, optimal parameters for in vitro enrichment of UCE probes have not been investigated thoroughly. Certain physical at- tributes are well known to affect probe efficacy, such as GC% con- tent (Tewhey et al., 2009), annealing temperature (Cruz-Dávalos et al., 2017), and Gibbs free energy change (delta G) (Matveeva et al., 2016). Additional in-solution parameters such as buffer stringency (Dabney & Meyer, 2012), washing procedures (Li, Hofreiter, Straube, Corrigan, & Naylor, 2013), and touchdown approaches to hybridiza- tion temperature (Li et al., 2013; Paijmans, Fickel, Courtiol, Hofreiter, & Förster, 2016) also affect enrichment of targeted DNA loci. While in silico testing has proven comparable to in vitro testing (Branstetter et al., 2017; Starrett et al., 2017), and the resulting probes designed here should prove suitable for in vitro capture and enrichment (see Section 3.5), there will always be certain loci that remain incalcitrant to enrichment. For this reason, as has previously been emphasized (i.e., Faircloth, 2017), all probe sets should be regarded as experi- mental until validated through in vitro testing. Future investigation of optimal enrichment parameters of UCE probes will likely help to normalize the number of loci recovered between in silico and in vitro testing. **4.4** | **Tailored UCE probe sets and universal utility** When first introduced, one of the appeals of UCEs was their ability to enrich hundreds to thousands of homologous loci across a clade (Faircloth et al., 2012). To a large extent, UCE probe sets such as those designed for tetrapods (Faircloth et al., 2012) have accom- plished this goal with the same probes working well across the major amniote lineages (Crawford et al., 2012; McCormack et al., 2012, 2013; Streicher et al., 2018). However, a probe set with similar universal utility has proven harder to attain for diverse arthropod groups like Coleoptera (see Baca et al., 2017; Van Dam et al., 2017), which while being younger in age (estimated origin during the mid- Carboniferous (Toussaint et al., 2017)) likely has over ten times the number of species (~380,000 named species; Ślipiński et al., 2011); or Hymenoptera (see Branstetter et al., 2017), a similarly hyperdi- verse insect order which may in reality be larger than Coleoptera by as much as 3.2 times (Forbes, Bagley, Beer, Hippee, & Widmayer, 2018). The first Coleoptera probe set designed by Faircloth (2017) re- covered hundreds of loci and provided phylogenetic signal when applied to Adephaga, a suborder of beetles not originally included during probe design (Baca et al., 2017). However, there was con- siderable conflict between different phylogenetic reconstruction methods with regard to the basal nodes of the phylogeny. Therefore, improved phylogenetic reconstruction of Adephaga would be en- hanced by a probe set tailored to this lineage to target more UCE loci and increase phylogenetic resolution. While the probe set devel- oped here was specifically optimized for universal application across Adephaga, our design notably lacked a whirligig beetle genome, as none are currently available. Whirligig beetles (Gyrinidae) comprise ~900 species (Short, 2018), are a relatively old family with an esti- mated early Triassic origin (Gustafson, Prokin, Bukontaite, Bergsten, & Miller, 2017), and may represent the sister lineage to all other ade- phagans (Baca et al., 2017). Thus, the probe set designed here could similarly result in fewer loci enriched for members of the Gyrinidae relative to other adephagan families used during probe design. However, this remains to be seen through in vitro testing and, alter- natively, this probe set could prove effective across all of Adephaga, preventing need for additional probes tailored to the Gyrinidae. With an increase in the number of tailored probe sets, concern could be raised that this deflates the value of UCEs as different probe sets result in the recovery of different homologous loci, caus- ing incompatibility between datasets and decreased 'universality.' However, UCE probe set development and synthesis affords the ability to mix probes from different designs, diversifying the loci targeted while providing backward compatibility with older data sets (Branstetter et al., 2017). For example, Branstetter et al. (2017) developed a probe set tailored for ant lineages that included newly developed probes targeting ant-specific loci, while retaining probes from the original universal Hymenoptera probe set designed by Faircloth et al. (2015) that worked well within ant lineages. We sim- ilarly follow this approach, and in addition to our newly developed probes (Section 3.5), we include as part of our final probe set 3,667 probes from the prior Coleoptera 1.1Kv1 probe set (Faircloth, 2017) targeting the 305 loci successfully recovered for Adephaga during the Baca et al. (2017) study. **5** | **CONCLUSION** Given the need to tailor UCE probe sets to focal taxa to improve locus recovery for use in phylogenomic studies and the increasing availability of genomic resources for diverse organisms, information about factors affecting probe design and methods for optimizing locus recovery are becoming increasingly important. We found an optimized UCE probe set depends on selection of an appropriate base genome and beginning probe design with less stringent tem- porary bait design parameters. We recommend the following work- flow (Figure 9) for using phyluce to design an optimized UCE probe set for organisms without a complete and well-annotated genome available. First, the base genome experiments shown in Figure 1 should be conducted in order to identify an optimal base genome or, alternatively, the base genome can be selected from the taxon with the smallest average genetic distance to the other study taxa as measured by independently generated molecular markers. Then, we recommend that temporary baits are designed based on _puta-_ _tive loci_ shared between the base genome and as few as one other taxon. Combined, these factors resulted in an optimized probe set for adephagan beetles, recovering the most UCE loci in in silico tests as well as the longest per-locus alignments. We make this final op- timized adephagan beetle probe set 'Adephaga_2.9Kv1,' consisting of 38,948 probes, targeting 2,948 UCE loci, available under a public **6946** GUSTAFSON et al . **|** domain license (CC-0), similar to the prior Coleoptera probe set de- signed by Faircloth (2017). **ACKNOWLEDGMENTS** This manuscript was greatly improved by the thoughtful review of Brant Faircloth and two anonymous reviewers. We thank the University of Kansas Center for Research Computing and the Biodiversity Institute for computational resources used during this project. GTG is supported by an NIH IRACDA postdoctoral fel- lowship (5K12GM063651). This project was funded in part by the Harold E. and Leona M. Rice Endowment Fund at Oregon State University. **CONFLICT OF INTEREST** None declared. **AUTHOR CONTRIBUTIONS** G.T.G., A.A., and A.E.Z.S. conceived of the study. G.T.G. and A.A. conducted probe design and analyses. J.S.S., J.M.P., and D.R.M. ob- tained and provided the Illumina and Sanger sequence data, assem- bled the base genomes, and produced genomic assembly metrics. All authors contributed to writing of the manuscript. **DATA AVAILABILITY** [DNA sequences: GenBank MK838494 to MK838511; P](info:x-wiley/peptideatlas/MK838494) hyluce BED/ BAM files for probe design experiments and Adephaga_2.9Kv1 final [probe set fasta file: Dryad https://doi.org/10.5061/dryad.2f62927;](https://doi.org/10.5061/dryad.2f62927) Custom scripts: Alexander 2018a: Comparing_monolithic_UCE_fas- [tas available from; https://github.com/laninsky/comparing_monol](https://github.com/laninsky/comparing_monolithic_UCE_fastas) [ithic_UCE_fastas; Alexander 2018b: Phase_target available from](https://github.com/laninsky/comparing_monolithic_UCE_fastas) [https://github.com/laninsky/reference_aligning_to_established_loci/](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target) [tree/master/phase_target; Alexander 2018c: distance_calcs available](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_target) [from: https://github.com/laninsky/distance_calcs; Alexander 2018d:](https://github.com/laninsky/distance_calcs) [Phase_everyone available from https://github.com/laninsky/refer](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone) [ence_aligning_to_established_loci/tree/master/phase_everyone.](https://github.com/laninsky/reference_aligning_to_established_loci/tree/master/phase_everyone) **ORCID** _Grey T. Gustafson_ [https://orcid.org/0000-0002-2014-7128](https://orcid.org/0000-0002-2014-7128) _Alana Alexander_ [https://orcid.org/0000-0002-6456-7757](https://orcid.org/0000-0002-6456-7757) _John S. Sproul_ [https://orcid.org/0000-0002-6747-3537](https://orcid.org/0000-0002-6747-3537) _David R. Maddison_ [https://orcid.org/0000-0002-7152-3824](https://orcid.org/0000-0002-7152-3824) _Andrew E. Z. Short_ [https://orcid.org/0000-0002-7467-7116](https://orcid.org/0000-0002-7467-7116) **REFERENCES** Alexander, A. M., Su, Y. C., Oliveros, C. H., Olson, K. V., Travers, S. L., & Brown, R. M. (2017). Genomic data reveals potential for hybridization, introgression, and incomplete lineage sorting to confound phylogenetic relationships in an adaptive radiation of narrow-mouth frogs. _Evolution_, _71_ [(2), 475-488. https://doi.](https://doi.org/10.1111/evo.13133) [org/10.1111/evo.13133](https://doi.org/10.1111/evo.13133) Baca, S. M., Alexander, A., Gustafson, G. T., & Short, A. E. Z. (2017). Ultraconserved elements show utility in phylogenetic inference of Adephaga (Coleoptera) and suggest paraphyly of 'Hydradephaga'. _Systematic Entomology_, _42_ [(4), 786-795. https://doi.org/10.1111/](https://doi.org/10.1111/syen.12244) [syen.12244](https://doi.org/10.1111/syen.12244) Blaimer, B. B., Brady, S. G., Schultz, T. R., Lloyd, M. W., Fisher, B. L., & Ward, P. S. (2015). Phylogenomic methods outperform traditional multi-locus approaches in resolving deep evolutionary history: A case study of formicine ants. _BMC Evolutionary Biology_, _15_ [, 271. https](https://doi.org/10.1186/s12862-015-0552-5) [://doi.org/10.1186/s12862-015-0552-5](https://doi.org/10.1186/s12862-015-0552-5) Blaimer, B. B., Lloyd, M. W., Guillory, W. X., & Brady, S. G. (2016). Sequence capture and phylogenetic utility of genomic ultracon- served elements obtained from pinned insect specimens. _PLoS ONE_, _11_ [, e0161531. https://doi.org/10.1371/journal.pone.0161531](https://doi.org/10.1371/journal.pone.0161531) Branstetter, M. G., Longino, J. T., Ward, P. S., & Faircloth, B. C. (2017). Enriching the ant tree of life: Enhanced UCE bait set for genome-scale phylogenetics of ants and other Hymenoptera. _Methods in Ecology and_ _Evolution_, _8_ [, 768-776. https://doi.org/10.1111/2041-210X.12742](https://doi.org/10.1111/2041-210X.12742) Bushnell, B. (2014). _BBTools software package_ [. Retrieved from http://](http://sourceforge.net/projects/bbmap) [sourceforge.net/projects/bbmap](http://sourceforge.net/projects/bbmap) Campana, M. G. (2018). BaitsTools: Software for hybridization capture bait design. _Molecular Ecology Resources_, _18_ [(2), 356-361. https://doi.](https://doi.org/10.1111/1755-0998.12721) [org/10.1111/1755-0998.12721](https://doi.org/10.1111/1755-0998.12721) Castresana, J. (2000). Selection of conserved blocks from multiple align- ments for their use in phylogenetic analysis. _Molecular Biology and_ _Evolution_, _17_ [, 540-552. https://doi.org/10.1093/oxfordjournals.mol-](https://doi.org/10.1093/oxfordjournals.molbev.a026334) [bev.a026334](https://doi.org/10.1093/oxfordjournals.molbev.a026334) Che, L. H., Zhang, S.-Q., Li, Y., Liang, D., Pang, H., Ślipiński, A., & Zhang, P. (2017). Genome-wide survey of nuclear protein-coding markers for beetle phylogenetics and their application in resolving both deep and shallow-level divergences. _Molecular Ecology Resources_, _17_ (6), 1342- [1358. https://doi.org/10.1111/1755-0998.12664](https://doi.org/10.1111/1755-0998.12664) Chiari, Y., Cahais, V., Galtier, N., & Delsuc, F. (2012). Phylogenomic analy- ses support the position of turtles as sister group of birds and croco-diles (Archosauria). _BMC Biology_, _10_, 65. Core Team, R. (2017). _R: A language and environment for statistical com-_ _puting_ . Vienna, Austria: R Foundation for Statistical Computing. [Retrieved from http://www.R-project.org/](http://www.R-project.org/) Crawford, N. G., Faircloth, B. C., McCormack, J. E., Brumfield, R. T., Winker, K., & Glenn, T. C. (2012). More than 1000 ultraconserved elements provide evidence that turtles are the sister group of ar-chosaurs. _Biology Letters_, _8_ [(5), 783-786. https://doi.org/10.1098/](https://doi.org/10.1098/rsbl.2012.0331) [rsbl.2012.0331](https://doi.org/10.1098/rsbl.2012.0331) Cruz-Dávalos, D. I., Llamas, B., Gaunitz, C., Fages, A., Gamba, C., Soubrier, J., … Orlando, L. (2017). Experimental conditions improving in-solu- tion target enrichment for ancient DNA. _Molecular Ecology Resources_, _17_ [(3), 508-522. https://doi.org/10.1111/1755-0998.12595](https://doi.org/10.1111/1755-0998.12595) Dabney, J., & Meyer, M. (2012). Length and GC-biases during sequenc- ing library amplification: A comparison of various polymerase-buf- fer systems with ancient and modern DNA sequencing libraries. _BioTechniques_, _52_ [, 87-94. https://doi.org/10.2144/000113809](https://doi.org/10.2144/000113809) Dowle, M., & Srinivasan, A. (2017). _data.table: Extension of ‘data.frame’_ . R [package version 1.10.4. Retrieved from https://CRAN.R-project.org/](https://CRAN.R-project.org/package=data.table) [package=data.table](https://CRAN.R-project.org/package=data.table) Esselstyn, J. A., Oliveros, C. H., Swanson, M. T., & Faircloth, B. C. (2017). Investigating difficult nodes in the placental mammal tree with expanded taxon sampling and thousands of ultraconserved loci. _Genome Biology and Evolution_, _9_ (9), 2308-2321. Faircloth, B. C. (2016). PHYLUCE is a software package for the analysis of conserved genomic loci. _Bioinformatics_, _32_ [(5), 786-788. https://doi.](https://doi.org/10.1093/bioinformatics/btv646) [org/10.1093/bioinformatics/btv646](https://doi.org/10.1093/bioinformatics/btv646) GUSTAFSON et al . **6947** **|** Faircloth, B. C. (2017). Identifying conserved genomic elements and design- ing universal bait sets to enrich them. _Methods in Ecology and Evolution_, _8_ [(9), 1103-1112. https://doi.org/10.1111/2041-210X.12754](https://doi.org/10.1111/2041-210X.12754) Faircloth, B. C., Branstetter, M. G., White, N. D., & Brady, S. G. (2015). Target enrichment of ultraconserved elements from arthro- pods provides a genomic perspective on relationships among Hymenoptera. _Molecular Ecology Resources_, _15_ [(3), 489-501. https://](https://doi.org/10.1111/1755-0998.12328) [doi.org/10.1111/1755-0998.12328](https://doi.org/10.1111/1755-0998.12328) Faircloth, B. C., McCormack, J. E., Crawford, N. G., Harvey, M. G., Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements anchor thousands of genetic markers spanning multiple evolu-tionary timescales. _Systematic Biology_, _61_ [(5), 717-726. https://doi.](https://doi.org/10.1093/sysbio/sys004) [org/10.1093/sysbio/sys004](https://doi.org/10.1093/sysbio/sys004) Faircloth, B. C., Sorenson, L., Santini, F., & Alfaro, M. E. (2013). A phylog- enomic perspective on the radiation of ray-finned fishes based upon targeted sequencing of ultraconserved elements (UCEs). _PLoS ONE_, _8_ [, e65923. https://doi.org/10.1371/journal.pone.0065923](https://doi.org/10.1371/journal.pone.0065923) Forbes, A. A., Bagley, R. K., Beer, M. A., Hippee, A. C., & Widmayer, H. A. (2018). Quantifying the unquantifiable: Why Hymenoptera, not Coleoptera, is the most speciose animal order. _BMC Ecology_, _18_, 21. [https://doi.org/10.1186/s12898-018-0176-x](https://doi.org/10.1186/s12898-018-0176-x) Glenn, T. C., & Faircloth, B. C. (2016). Capturing Darwin’s dream. _Molecular_ _Ecology_ _Resources_, _16_ (5), 1051-1058. [https://doi.](https://doi.org/10.1111/1755-0998.12574) [org/10.1111/1755-0998.12574](https://doi.org/10.1111/1755-0998.12574) Gomez, R. A., Will, K. W., & Maddison, D. R. (2016). Are _Miquihuana__rhadiniformis_ Barr, 1982 and _Pseudamara arenaria_ (LeConte, 1847) (Coleoptera, Carabidae) sphodrines? Phylogenetic analysis of data from next-generation sequencing of museum specimens resolves the tribal-group relationships of these enigmatic taxa. _Entomologische_ _Blätter Und Coleoptera_, _112_, 149-168. Grismer, J. L., Schulte, J. A., Alexander, A., Wagner, P., Travers, S. L., Buehler, M. D., … Brown, R. M. (2016). The Eurasian invasion: Phylogenomic data reveal multiple Southeast Asian origins for Indian Dragon Lizards. _BMC Evolutionary Biology_, _16_ [(1), 43. https://doi.](https://doi.org/10.1186/s12862-016-0611-6) [org/10.1186/s12862-016-0611-6](https://doi.org/10.1186/s12862-016-0611-6) Gustafson, G. T., Prokin, A. A., Bukontaite, A., Bergsten, J., & Miller, K. B. (2017). Tip-dated phylogeny of whirligig beetles reveals ancient lineage surviving on Madagascar. _Scientific Reports_, _7_ [, 8619. https://](https://doi.org/10.1038/s41598-017-08403-1) [doi.org/10.1038/s41598-017-08403-1](https://doi.org/10.1038/s41598-017-08403-1) Harris, R. S. (2007). _Improved pairwise alignment of genomic DNA_ . PhD thesis, Pennsylvania State University. Hosner, P. A., Tobias, J. A., Braun, E. L., & Kimball, R. T. (2017). How do seemingly non-vagile clades accomplish trans-marine dispersal? Trait and dispersal evolution in the landfowl (Aves: Galliformes). _Proceedings of the Royal Society B: Biological Sciences_, _284_ (1854), [20170210. https://doi.org/10.1098/rspb.2017.0210](https://doi.org/10.1098/rspb.2017.0210) Johnson, M. G., Gardner, E. M., Liu, Y., Medina, R., Goffinet, B., Shaw, A. J., … Wickett, N. J. (2016). HybPiper: Extracting coding sequence and introns for phylogenetics from high-throughput sequencing reads using target enrichment. _Applications in Plant Sciences_, _4_ (7), [1600016. https://doi.org/10.3732/apps.1600016](https://doi.org/10.3732/apps.1600016) Kanda, K., Pflug, J. M., Sproul, J. S., Dasenko, M. A., & Maddison, D. R. (2015). Successful recovery of nuclear protein-coding genes from small insects in museums using illumina sequencing. _PLoS ONE_, _10_ [(12), e0143929. https://doi.org/10.1371/journal.pone.0143929](https://doi.org/10.1371/journal.pone.0143929) Katoh, K., & Standley, D. M. (2013). MAFFT multiple sequence align- ment software version 7: Improvements in performance and us-ability. _Molecular Biology and Evolution_, _30_ [(4), 772-780. https://doi.](https://doi.org/10.1093/molbev/mst010) [org/10.1093/molbev/mst010](https://doi.org/10.1093/molbev/mst010) Li, C., Hofreiter, M., Straube, N., Corrigan, S., & Naylor, G. J. (2013). Capturing protein-coding genes across highly divergent species. _BioTechniques_, _54_ [, 321-326. https://doi.org/10.2144/000114039](https://doi.org/10.2144/000114039) Li, H., & Durbin, R. (2009). Fast and accurate short read alignment with Burrows-Wheeler transform. _Bioinformatics_, _25_ [, 1754-1760. https://](https://doi.org/10.1093/bioinformatics/btp324) [doi.org/10.1093/bioinformatics/btp324](https://doi.org/10.1093/bioinformatics/btp324) Li, H., & Durbin, R. (2010). Fast and accurate long-read alignment with Burrows-Wheeler transform. _Bioinformatics_, _26_ [, 589-595. https://](https://doi.org/10.1093/bioinformatics/btp698) [doi.org/10.1093/bioinformatics/btp698](https://doi.org/10.1093/bioinformatics/btp698) Li, H., Handsaker, B., Wysoker, A., Fennell, T., Ruan, J., Homer, N., … 1000 Genome Project Data Processing Subgroup (2009). The se-quence alignment/map (SAM) format and SAMtools. _Bioinformatics_, _25_, 2078-2079. Lunter, G., & Goodson, M. (2011). Stampy: A statistical algorithm for sen- sitive and fast mapping of Illumina sequence reads. _Genome Research_, _21_ [, 936-939. https://doi.org/10.1101/gr.111120.110](https://doi.org/10.1101/gr.111120.110) Maddison, D. R. (2012). Phylogeny of _Bembidion_ and related ground beetles (Coleoptera: Carabidae: Trechinae: Bembidiini: Bembidiina). _Molecular Phylogenetics and Evolution_, _63_ [, 533-576. https://doi.](https://doi.org/10.1016/j.ympev.2012.01.015) [org/10.1016/j.ympev.2012.01.015](https://doi.org/10.1016/j.ympev.2012.01.015) Maddison, D. R., Baker, M. D., & Ober, K. A. (1999a). Phylogeny of ca-rabid beetles as inferred from 18S ribosomal DNA (Coleoptera: Carabidae). _Systematic Entomology_, _24_ [, 103-138. https://doi.](https://doi.org/10.1046/j.1365-3113.1999.00088.x) [org/10.1046/j.1365-3113.1999.00088.x](https://doi.org/10.1046/j.1365-3113.1999.00088.x) Maddison, D. R., Baker, M. D., & Ober, K. A.. (1999b). A preliminary phy-logenetic analysis of 18S ribosomal DNA of carabid beetles (Insecta: Coleoptera). In Ball, G. E., Casale, A., & Vigna-Taglianti, A. (Eds.), _Phylogeny and classification of Caraboidea (Coleoptera: Adephaga)_, pp. 229-250. Torino, Italy: Museo Regionale di Scienze Naturali, Atti. 517 p. Maddison, D. R., Moore, W., Baker, M. D., Ellis, T. M., Ober, K. A., Cannone, J. J., & Gutell, R. R. (2009). Monophyly of terrestrial ade-phagan beetles as indicated by three nuclear genes (Coleoptera: Carabidae and Trachypachidae). _Zoologica Scripta_, _38_ [(1), 43-62. https](https://doi.org/10.1111/j.1463-6409.2008.00359.x) [://doi.org/10.1111/j.1463-6409.2008.00359.x](https://doi.org/10.1111/j.1463-6409.2008.00359.x) Maddison, D. R., & Swanson, A. P. (2010). A preliminary characterization of Bembidion perspicuum LeConte, with a reclassification of related species (Coleoptera, Carabidae) north of México. _ZooKeys_, _43_, 15- [31. https://doi.org/10.3897/zookeys.43.390](https://doi.org/10.3897/zookeys.43.390) Mamanova, L., Coffey, A. J., Scott, C. E., Kozarewa, I., Turner, E. H., Kumar, A., … Turner, D. J. (2010). Target-enrichment strategies for next-generation sequencing. _Nature Methods_, _7_ [(2), 111-118. https://](https://doi.org/10.1038/nmeth.1419) [doi.org/10.1038/nmeth.1419](https://doi.org/10.1038/nmeth.1419) Manthey, J. D., Campillo, L. C., Burns, K. J., & Moyle, R. G. (2016). Comparison of target-capture and restriction-site associated DNA sequencing for phylogenomics: A test in cardinalid Tanagers (Aves, Genus: _Piranga_ ). _Systematic Biology_, _65_ (4), 640-650. Matveeva, O. V., Nechipurenko, Y. D., Riabenko, E., Ragan, C., Nazipova, N. N., Ogurtsov, A. Y., & Shabalina, S. A. (2016). Optimization of signal-to-noise ratio for efficient microarray probe design. _Bioinformatics_, _30_ [(17), i552-i558. https://doi.org/10.1093/bioin](https://doi.org/10.1093/bioinformatics/btw451) [formatics/btw451](https://doi.org/10.1093/bioinformatics/btw451) Mayer, C., Sann, M., Donath, A., Meixner, M., Podsiadlowski, L., Peters, R. S., … Niehuis, O. (2016). BaitFisher: A software package for mul- tispecies target DNA enrichment probe design. _Molecular Biology__and Evolution_, _33_ [(7), 1875-1886. https://doi.org/10.1093/molbev/](https://doi.org/10.1093/molbev/msw056) [msw056](https://doi.org/10.1093/molbev/msw056) McCormack, J. E., Faircloth, B. C., Crawford, N. G., Gowaty, P. A., Brumfield, R. T., & Glenn, T. C. (2012). Ultraconserved elements are novel phylogenomic markers that resolved placental mammal phy- logeny when combined with species-tree analysis. _Genome Research_, _22_ (4), 746-754. McCormack, J. E., Harvey, M. G., Faircloth, B. C., Crawford, N. G., Glenn, T. C., & Brumfield, R. T. (2013). A phylogeny of birds based on over 1,500 loci collected by target enrichment and high-throughput se- quencing. _PLoS ONE_, _8_ [, e54848. https://doi.org/10.1371/journ](https://doi.org/10.1371/journal.pone.0054848) [al.pone.0054848](https://doi.org/10.1371/journal.pone.0054848) McCormack, J. E., Hird, S. M., Zellmer, A. J., Carstens, B. C., & Brumfield, R. T. (2013). Applications of next-generation sequencing to phylo- geography and phylogenetics. _Molecular Phylogenetics and Evolution_, _66_ [(2), 526-538. https://doi.org/10.1016/j.ympev.2011.12.007](https://doi.org/10.1016/j.ympev.2011.12.007) **6948** GUSTAFSON et al . **|** McCormack, J. E., Tsai, W. L., & Faircloth, B. C. (2016). Sequence capture of ultraconserved elements from bird museum speci-mens. _Molecular Ecology Resources_, _16_ [(5), 1189-1203. https://doi.](https://doi.org/10.1111/1755-0998.12466) [org/10.1111/1755-0998.12466](https://doi.org/10.1111/1755-0998.12466) McKenna, A., Hanna, M., Banks, E., Sivachenko, A., Cibulskis, K., Kernytsky, A., … DePristo, M. A. (2010). The Genome Analysis Toolkit: A MapReduce framework for analyzing next-generation DNA sequencing data. _Genome Resources_, _20_ [(9), 1297-1303. https://](https://doi.org/10.1101/gr.107524.110) [doi.org/10.1101/gr.107524.110](https://doi.org/10.1101/gr.107524.110) McKenna, D. D., Wild, A. L., Kanda, K., Bellamy, C. L., Beutel, R. G., Caterino, M. S., … Farrell, B. D. (2015). The beetle tree of life reveals that Coleoptera survived end-Permian mass extinction to diversify during the Cretaceous terrestrial revolution. _Systematic Entomology_, _40_ [, 835-880. https://doi.org/10.1111/syen.12132](https://doi.org/10.1111/syen.12132) Musher, L. J., & Cracraft, J. (2018). Phylogenomics and species de- limitation of a complex radiation of Neotropical suboscine birds ( _Pachyramphus_ ). _Molecular Phylogenetics and Evolution_, _118_, 204-221. [https://doi.org/10.1016/j.ympev.2017.09.013](https://doi.org/10.1016/j.ympev.2017.09.013) Nguyen, L.-T., Schmidt, H. A., von Haeseler, A., & Minh, B. Q. (2015). IQ- TREE: A fast and effective stochastic algorithm for estimating max-imum likelihood phylogenies. _Molecular Biology and Evolution_, _32_ (1), [268-274. https://doi.org/10.1093/molbev/msu300](https://doi.org/10.1093/molbev/msu300) Ober, K. (2002). Phylogenetic relationships of the carabid subfam-ily Harpalinae (Coleoptera) based on molecular sequence data. _Molecular Phylogenetics and Evolution_, _24_ [, 228-248. https://doi.](https://doi.org/10.1016/S1055-7903(02)00251-8) [org/10.1016/S1055-7903(02)00251-8](https://doi.org/10.1016/S1055-7903(02)00251-8) Paijmans, J. L. A., Fickel, J., Courtiol, A., Hofreiter, M., & Förster, D. W. (2016). Impact of enrichment conditions on cross-species capture of fresh and degraded DNA. _Molecular Ecology Resources_, _16_ (1), 42-55. [https://doi.org/10.1111/1755-0998.12420](https://doi.org/10.1111/1755-0998.12420) Quattrini, A. M., Faircloth, B. C., Dueñas, L. F., Bridge, T. C. L., Brugler, M. R., Calixto-Botía, I. F., … McFadden, C. S. (2018). Universal tar-get-enrichment baits for anthozoan (Cnidaria) phylogenomics: New approaches to long-standing problems. _Molecular Ecology Resources_, _18_ [(2), 281-295. https://doi.org/10.1111/1755-0998.12736](https://doi.org/10.1111/1755-0998.12736) Rice, P., Longden, I., & Bleasby, A. (2000). EMBOSS: The European mo-lecular biology open software suite. _Trends in Genetics_, _16_ (6), 276-[277. https://doi.org/10.1016/S0168-9525(00)02024-2](https://doi.org/10.1016/S0168-9525(00)02024-2) Ruane, S., & Austin, C. C. (2017). Phylogenomics using formalin- fixed and 100+ year-old intractable natural history specimens. _Molecular_ _Ecology_ _Resources_, _17_ (5), 1003-1008. [https://doi.](https://doi.org/10.1111/1755-0998.12655) [org/10.1111/1755-0998.12655](https://doi.org/10.1111/1755-0998.12655) Shen, X.-X., Liang, D., Wen, J.-Z., & Zhang, P. (2011). Multiple genome alignments facilitate development of NPCL markers: A case study of tetrapod phylogeny focusing on the position of turtles. _Molecular__Biology and Evolution_, _28_ [(12), 3237-3252. https://doi.org/10.1093/](https://doi.org/10.1093/molbev/msr148) [molbev/msr148](https://doi.org/10.1093/molbev/msr148) Short, A. E. Z. (2018). Systematics of aquatic beetles (Coleoptera): Current state and future directions. _Systematic Entomology_, _43_, 1-18. [https://doi.org/10.1111/syen.12270](https://doi.org/10.1111/syen.12270) Simão, F. A., Waterhouse, R. M., Ioannidis, P., Kriventseva, E. V., & Zdobnov, E. M. (2015). BUSCO: Assessing genome assembly and annotation completeness with single-copy orthologs. _Bioinformatics_, _31_ [(19), 3210-3212. https://doi.org/10.1093/bioinformatics/btv351](https://doi.org/10.1093/bioinformatics/btv351) Ślipiński, A., Leschen, R. A. B., & Lawrence, J. F. (2011). Order Coleoptera Linnaeus, 1758. Animal biodiversity: An outline of higher-level clas-sification and survey of taxonomic richness (ed. By Z.-Q. Zhang). _Zootaxa_, _3148_, 203-208. Smith, B. T., Harvey, M. G., Faircloth, B. C., Glenn, T. C., & Brumfield, R. T. (2014). Target capture and massively parallel sequencing of ultraconserved elements for comparative studies at shallow evo-lutionary time scales. _Systematic Biology_, _63_ [(1), 83-95. https://doi.](https://doi.org/10.1093/sysbio/syt061) [org/10.1093/sysbio/syt061](https://doi.org/10.1093/sysbio/syt061) Sproul, J. S., & Maddison, D. R. (2017). Sequencing historical specimens: Successful preparation of small specimens with low amounts of de- graded DNA. _Molecular Ecology Resources_, _17_ [, 1183-1201. https://](https://doi.org/10.1111/1755-0998.12660) [doi.org/10.1111/1755-0998.12660](https://doi.org/10.1111/1755-0998.12660) Stamatakis, A. (2014). RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. _Bioinformatics_, _30_ (9), 1312- [1313. https://doi.org/10.1093/bioinformatics/btu033](https://doi.org/10.1093/bioinformatics/btu033) Starrett, J., Derkarabetian, S., Hedin, M., Byrson, R. W. Jr, McCormack, J., & Fairclotch, B. C. (2017). High phylogenetic utility of an ul- traconserved element probe set designed for Arachnida. _Molecular_ _Ecology_ _Resources_, _17_ (4), 812-823. [https://doi.](https://doi.org/10.1111/1755-0998.12621) [org/10.1111/1755-0998.12621](https://doi.org/10.1111/1755-0998.12621) Streicher, J. W., Miller, E. C., Guerrero, P. C., Correa, C., Ortiz, J. C., Crawford, A. J., … Wiens, J. J. (2018). Evaluating methods for phy- logenomic analyses, and a new phylogeny for a major frog clade (Hyloidea) based on 2214 loci. _Molecular Phylogenetics and Evolution_, _119_ [, 128-143. https://doi.org/10.1016/j.ympev.2017.10.013](https://doi.org/10.1016/j.ympev.2017.10.013) Streicher, J. W., & Wiens, J. J. (2017). Phylogenomic analyses of more than 4000 nuclear loci resolve the origin of snakes among lizard families. _Biology Letters_, _13_ [(9), 20170393. https://doi.org/10.1098/](https://doi.org/10.1098/rsbl.2017.0393) [rsbl.2017.0393](https://doi.org/10.1098/rsbl.2017.0393) Sun, K., Meiklejohn, K. A., Faircloth, B. C., Glenn, T. C., Braun, E. L., & Kimball, R. T. (2014). The evolution of peafowl and other taxa with ocelli (eyespots): A phylogenomic approach. _Proceedings of_ _the Royal Society B: Biological Sciences_, _281_ [, 20140823. https://doi.](https://doi.org/10.1098/rspb.2014.0823) [org/10.1098/rspb.2014.0823](https://doi.org/10.1098/rspb.2014.0823) Talavera, G., & Castresana, J. (2007). Improvement of phylogenies after removing divergent and ambiguously aligned blocks from protein sequence alignments. _Systematic Biology_, _56_ [, 564-577. https://doi.](https://doi.org/10.1080/10635150701472164) [org/10.1080/10635150701472164](https://doi.org/10.1080/10635150701472164) Tewhey, R., Nakano, M., Wang, X., Pabón-Peña, C., Novak, B., Giuffre, A., … Frazer, K. A. (2009). Enrichment of sequencing targets from the human genome by solution hybridization. _Genome Biology_, _10_, R116. [https://doi.org/10.1186/gb-2009-10-10-r116](https://doi.org/10.1186/gb-2009-10-10-r116) Toussaint, E. F. A., Seidel, M., Arriaga-Varela, E., Hájek, J., Král, D., Sekerka, L., … Fikáček, M. (2017). The peril of dating beetles. _Systematic__Entomology_, _42_ [(1), 1-10. https://doi.org/10.1111/syen.12198](https://doi.org/10.1111/syen.12198) Van Dam, M. H., Lam, A. W., Sagata, K., Gewa, B., Laufa, R., Balke, M., … Reidel, A. (2017). Ultraconserved elements (UCEs) resolve the phy- logeny of Australasian smurf-weevils. _PLoS ONE_, _12_, e0188044. Wickham, H. (2017). _stringr: Simple, consistent wrappers for common string_ _operations_ . R package version 1.2.0. Retrieved from http://CRAN.R project.org/package=stringr Zhang, S.-Q., Che, L.-H., Li, Y., Liang, D., Pang, H., Ślipiński, A., & Zhang, P. (2018). Evolutionary history of Coleoptera revealed by extensive sampling of genes and species. _Nature Communications_, _9_ [, 205. https](https://doi.org/10.1038/s41467-017-02644-4) [://doi.org/10.1038/s41467-017-02644-4](https://doi.org/10.1038/s41467-017-02644-4) **SUPPORTING INFORMATION** Additional supporting information may be found online in the Supporting Information section at the end of the article. **How to cite this article:** Gustafson GT, Alexander A, Sproul JS, Pflug JM, Maddison DR, Short AEZ. Ultraconserved ~~element~~ ~~(UCE)~~ ~~probe~~ ~~set~~ ~~design:~~ ~~Base~~ ~~genome~~ ~~and~~ ~~initial~~ design parameters critical for optimization. _Ecol Evol_ . 2019;9:6933-6948. [https://doi.org/10.1002/ece3.5260](https://doi.org/10.1002/ece3.5260)\n",
      "['10.1038/nmeth.1419', '10.1186/s12862-015-0552-5', '10.1093/bioinformatics/btp698', '10.1111/1755-0998.12660', '10.1101/gr.107524.110', '10.1111/j.1463-6409.2008.00359.x', '10.1371/journ', '10.1371/journal.pone.0161531', '10.1093/molbev/', '10.2144/000113809', '10.1111/1755-0998.12736', '10.1093/molbev/msr148', '10.1186/gb-2009-10-10-r116', '10.3897/zookeys.43.390', '10.1111/1755-0998.12721', '10.1093/molbev/msw056', '10.1016/S1055-7903', '10.1093/oxfordjournals.molbev.a026334', '10.1093/bioin', '10.1093/molbev/mst010', '10.1016/j.ympev.2017.10.013', '10.1371/journal.pone.0054848', '10.1111/1755-0998.12621', '10.1093/molbev/msu300', '10.1093/bioinformatics/btv646', '10.1111/1755-0998.12466', '10.1093/bioinformatics/btu033', '10.1080/10635150701472164', '10.1111/syen.12244', '10.1093/sysbio/sys004', '10.3732/apps.1600016', '10.1038/s41467-017-02644-4', '10.1002/ece3.5260', '10.1016/j.ympev.2011.12.007', 'MK838494', '10.1093/sysbio/syt061', '10.1111/syen.12270', '10.1111/syen.12198', '10.1098/rsbl.2017.0393', '10.1111/1755-0998.12595', '10.1371/journal.pone.0143929', '10.1111/syen.12132', '10.1111/1755-0998.12574', '10.1016/j.ympev.2012.01.015', '10.1016/S0168-9525', '10.1098/rspb.2014.0823', '10.1186/s12898-018-0176-x', '10.1186/s12862-016-0611-6', '10.1111/evo.13133', '10.1111/1755-0998.12664', '10.1016/j.ympev.2017.09.013', '10.1111/2041-210X.12742', '10.1093/bioinformatics/btw451', 'MK838511', '10.1098/rspb.2017.0210', '10.1111/1755-0998.12420', '10.1038/s41598-017-08403-1', '10.1093/bioinformatics/btp324', '10.1111/1755-0998.12655', '10.1101/gr.111120.110', '10.1093/bioinformatics/btv351', '10.1111/2041-210X.12754', '10.1093/oxfordjournals.mol-', '10.1371/journal.pone.0065923', '10.1046/j.1365-3113.1999.00088.x', '10.1098/rsbl.2012.0331', '10.2144/000114039', '10.5061/dryad.2f62927', '10.1111/1755-0998.12328']\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#pdf_file_path = os.path.join(TRAIN_DATA_DIR, 'PDF', '10.1002_2017jc013030.pdf')\n",
    "pdf_file_path = os.path.join(TEST_DATA_DIR, 'PDF', '10.1002_ece3.5260.pdf')\n",
    "# pdf_file_path = os.path.join(TRAIN_DATA_DIR, 'PDF', '10.7554_elife.29944.pdf')\n",
    "# pdf_file_path = os.path.join(TRAIN_DATA_DIR, 'PDF', '10.20944_preprints202009.0353.v1.pdf')\n",
    "# pdf_file_path = os.path.join(TRAIN_DATA_DIR, 'PDF', '10.1038_s41396-020-00885-8.pdf')\n",
    "\n",
    "\n",
    "file_extractor = MdcFileTextExtractor('10.20944_preprints202009.0353.v1', pdf_file_path)\n",
    "md_text = file_extractor.extract_text_from_file()\n",
    "print(f\"md_text: {md_text}\")\n",
    "\n",
    "# article_data = file_extractor.extract_article_data_from_text(full_text=md_text, nlp=NLP_SPACY)\n",
    "# article_data = file_extractor.extract_article_data_for_inference(NLP_SPACY)\n",
    "# print(f\"article_data: {article_data}\")\n",
    "\n",
    "potential_ds_ids = file_extractor.find_potential_dataset_ids(md_text)\n",
    "print(potential_ds_ids)\n",
    "\n",
    "\n",
    "# abstract_match = re.search(r\"Abstract\\s*(.*?)(?=\\n\\n|\\Z)\", md_text, re.IGNORECASE | re.DOTALL)\n",
    "# abstract = abstract_match.group(1).split('\\n\\n')[0] if abstract_match else \"No Abstract\"\n",
    "# print(f\"abstract_match: {abstract[:400]}\")\n",
    "\n",
    "# idx_1 = md_text.find('###')\n",
    "# idx_2 = md_text.find('Abstract')\n",
    "# idx_2e = md_text.find('\\n\\n', idx_2)\n",
    "# abstract = md_text[idx_2:idx_2e]\n",
    "# print(f\"abstract_idx: {abstract[:400]}\")\n",
    "      \n",
    "# idx_3 = md_text.find('References')\n",
    "# print(f\"{idx_1},{idx_2},{idx_3}\")\n",
    "# pre_ref_text = md_text[idx_2: idx_3]\n",
    "\n",
    "# # md_text = md_text[idx_1:idx_3]\n",
    "# print(f\"pre ref Text len: {len(pre_ref_text)}\")\n",
    "# print(f\"pre ref Text: {pre_ref_text[:400]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "07d47e6e",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'temp' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mNameError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[3]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[43mtemp\u001b[49m\n",
      "\u001b[31mNameError\u001b[39m: name 'temp' is not defined"
     ]
    }
   ],
   "source": [
    "temp"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
